# Comparing `tmp/gemmapy-1.0.0.tar.gz` & `tmp/gemmapy-1.0.1.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "gemmapy-1.0.0.tar", last modified: Wed May  8 23:49:00 2024, max compression
+gzip compressed data, was "gemmapy-1.0.1.tar", last modified: Fri May 17 06:30:06 2024, max compression
```

## Comparing `gemmapy-1.0.0.tar` & `gemmapy-1.0.1.tar`

### file list

```diff
@@ -1,157 +1,163 @@
-drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-08 23:49:00.396396 gemmapy-1.0.0/
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    11357 2024-05-04 04:09:08.000000 gemmapy-1.0.0/LICENSE.md
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     1730 2024-05-08 23:49:00.396396 gemmapy-1.0.0/PKG-INFO
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     1306 2024-05-07 21:06:20.000000 gemmapy-1.0.0/README.rst
-drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-08 23:49:00.344395 gemmapy-1.0.0/gemmapy/
--rw-r--r--   0 omancarci (30067) pavlab   (30005)       61 2024-05-04 04:09:08.000000 gemmapy-1.0.0/gemmapy/__init__.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    21504 2024-05-08 23:46:56.000000 gemmapy-1.0.0/gemmapy/_processors.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     6768 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/_subprocessors.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)      936 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/_validators.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    89458 2024-05-08 00:42:56.000000 gemmapy-1.0.0/gemmapy/gemmapy_api.py
-drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-08 23:49:00.348395 gemmapy-1.0.0/gemmapy/sdk/
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    12986 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/__init__.py
-drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-08 23:49:00.348395 gemmapy-1.0.0/gemmapy/sdk/api/
--rw-r--r--   0 omancarci (30067) pavlab   (30005)      138 2024-05-04 04:09:08.000000 gemmapy-1.0.0/gemmapy/sdk/api/__init__.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)   214827 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/api/default_api.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    25975 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/api_client.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9213 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/configuration.py
-drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-08 23:49:00.392396 gemmapy-1.0.0/gemmapy/sdk/models/
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    12789 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/__init__.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     6156 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/annotation_search_result_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8608 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/annotation_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9891 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/annotation_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5987 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/api_info_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    24020 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/array_design_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    24987 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/array_design_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8819 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/audit_event_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7478 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/bibliographic_phenotypes_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    16799 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/bibliographic_reference_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    14982 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/bio_assay_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9742 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/bio_material_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    10036 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/bio_sequence_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5992 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/category_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7293 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/characteristic_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     6624 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/citation_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3104 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/composite_sequence_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     6014 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/composite_sequence_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8528 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/contrast_result_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3088 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/database_entry_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5384 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/database_entry_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3052 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset1.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset10.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset2.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset3.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset4.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset5.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset6.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset7.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset8.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset9.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3064 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/dataset_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    15728 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/diff_ex_result_set_summary_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7917 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/differential_expression_analysis_result_set_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9659 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/differential_expression_analysis_result_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    14048 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/differential_expression_analysis_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5109 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/experiment_expression_levels_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9358 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/experimental_factor_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    15105 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/expression_experiment_set_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    28654 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/expression_experiment_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    31623 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/expression_experiment_with_search_result_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9207 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/external_database_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    10910 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/factor_value_basic_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    16266 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/factor_value_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3104 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filter_arg_array_design.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3168 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filter_arg_expression_analysis_result_set.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3140 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filter_arg_expression_experiment.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9152 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_array_design_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9362 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_composite_sequence_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    10132 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_differential_expression_analysis_result_set_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9467 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_expression_experiment_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8907 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_gene_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    25633 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/geeq_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3040 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene1.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene2.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene3.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene4.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3052 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5754 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene_element_expressions_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7870 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene_ontology_term_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     6956 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene_set_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     9642 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/gene_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8358 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/limited_response_data_object_annotation_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8389 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/limited_response_data_object_array_design_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/measurement_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3208 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/one_of_search_result_value_object_object_result_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5194 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/ontology_term_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8221 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/paginated_response_data_object_composite_sequence_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8314 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/paginated_response_data_object_expression_experiment_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8127 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/physical_location_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/platform.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/platform1.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/platform2.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/platform3.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/platform4.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/platform5.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3068 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/platform_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/probe.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3088 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/quantitation_type.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3100 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/quantitation_type_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    17466 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/quantitation_type_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    10652 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    11276 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_with_search_result_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7723 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_response_data_object_category_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7642 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_response_data_object_taxon_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/query_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3884 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_api_info_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4236 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_differential_expression_analysis_result_set_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4099 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_annotation_search_result_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3978 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_array_design_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3945 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_bio_assay_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4187 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_differential_expression_analysis_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4143 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_experiment_expression_levels_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4033 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_gene_ontology_term_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3901 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_gene_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4033 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_physical_location_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3912 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_taxon_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3727 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_long.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3959 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_set_annotation_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4025 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_set_quantitation_type_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3906 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_simple_svd_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4521 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/response_error_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3088 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/search_result_type.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7682 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/search_result_value_object_expression_experiment_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7127 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/search_result_value_object_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4833 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/search_results_response_data_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     6838 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/search_settings_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5411 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/simple_svd_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3096 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_array_design.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3160 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_expression_analysis_result_set.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3132 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_expression_experiment.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3072 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_taxon.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     4439 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/sort_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    10286 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/statement_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon1.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon2.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon3.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon4.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon5.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon6.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon_arg.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     6759 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     8441 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/taxon_with_usage_statistics_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5162 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/vector_element_value_object.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     5078 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/models/well_composed_error_body.py
--rw-r--r--   0 omancarci (30067) pavlab   (30005)    13649 2024-05-07 20:41:44.000000 gemmapy-1.0.0/gemmapy/sdk/rest.py
-drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-08 23:49:00.396396 gemmapy-1.0.0/gemmapy.egg-info/
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     1730 2024-05-08 23:49:00.000000 gemmapy-1.0.0/gemmapy.egg-info/PKG-INFO
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     7474 2024-05-08 23:49:00.000000 gemmapy-1.0.0/gemmapy.egg-info/SOURCES.txt
--rw-r--r--   0 omancarci (30067) pavlab   (30005)        1 2024-05-08 23:49:00.000000 gemmapy-1.0.0/gemmapy.egg-info/dependency_links.txt
--rw-r--r--   0 omancarci (30067) pavlab   (30005)      114 2024-05-08 23:49:00.000000 gemmapy-1.0.0/gemmapy.egg-info/requires.txt
--rw-r--r--   0 omancarci (30067) pavlab   (30005)        8 2024-05-08 23:49:00.000000 gemmapy-1.0.0/gemmapy.egg-info/top_level.txt
--rw-r--r--   0 omancarci (30067) pavlab   (30005)       81 2024-05-04 04:09:08.000000 gemmapy-1.0.0/pyproject.toml
--rw-r--r--   0 omancarci (30067) pavlab   (30005)      404 2024-05-08 23:49:00.396396 gemmapy-1.0.0/setup.cfg
-drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-08 23:49:00.392396 gemmapy-1.0.0/tests/
--rw-r--r--   0 omancarci (30067) pavlab   (30005)     3766 2024-05-08 05:16:40.000000 gemmapy-1.0.0/tests/test_basic.py
+drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-17 06:30:06.541129 gemmapy-1.0.1/
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    11357 2024-05-04 04:09:08.000000 gemmapy-1.0.1/LICENSE.md
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     1730 2024-05-17 06:30:06.541129 gemmapy-1.0.1/PKG-INFO
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     1306 2024-05-07 21:06:20.000000 gemmapy-1.0.1/README.rst
+drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-17 06:30:06.513129 gemmapy-1.0.1/gemmapy/
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)       61 2024-05-04 04:09:08.000000 gemmapy-1.0.1/gemmapy/__init__.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    21504 2024-05-08 23:46:56.000000 gemmapy-1.0.1/gemmapy/_processors.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     6768 2024-05-07 20:41:44.000000 gemmapy-1.0.1/gemmapy/_subprocessors.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)      936 2024-05-07 20:41:44.000000 gemmapy-1.0.1/gemmapy/_validators.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    89515 2024-05-17 06:20:47.000000 gemmapy-1.0.1/gemmapy/gemmapy_api.py
+drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-17 06:30:06.517129 gemmapy-1.0.1/gemmapy/sdk/
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    13741 2024-05-17 06:26:34.000000 gemmapy-1.0.1/gemmapy/sdk/__init__.py
+drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-17 06:30:06.517129 gemmapy-1.0.1/gemmapy/sdk/api/
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)      138 2024-05-17 06:26:34.000000 gemmapy-1.0.1/gemmapy/sdk/api/__init__.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)   249425 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/api/default_api.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    25975 2024-05-17 06:26:34.000000 gemmapy-1.0.1/gemmapy/sdk/api_client.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9213 2024-05-17 06:26:34.000000 gemmapy-1.0.1/gemmapy/sdk/configuration.py
+drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-17 06:30:06.537129 gemmapy-1.0.1/gemmapy/sdk/models/
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    13544 2024-05-17 06:26:34.000000 gemmapy-1.0.1/gemmapy/sdk/models/__init__.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     6156 2024-05-17 06:26:31.000000 gemmapy-1.0.1/gemmapy/sdk/models/annotation_search_result_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8608 2024-05-17 06:26:31.000000 gemmapy-1.0.1/gemmapy/sdk/models/annotation_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9891 2024-05-17 06:26:31.000000 gemmapy-1.0.1/gemmapy/sdk/models/annotation_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5987 2024-05-17 06:26:31.000000 gemmapy-1.0.1/gemmapy/sdk/models/api_info_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    24020 2024-05-17 06:26:31.000000 gemmapy-1.0.1/gemmapy/sdk/models/array_design_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    24987 2024-05-17 06:26:31.000000 gemmapy-1.0.1/gemmapy/sdk/models/array_design_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8819 2024-05-17 06:26:31.000000 gemmapy-1.0.1/gemmapy/sdk/models/audit_event_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7478 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/bibliographic_phenotypes_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    16799 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/bibliographic_reference_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    14982 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/bio_assay_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9742 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/bio_material_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10036 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/bio_sequence_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5992 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/category_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7293 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/characteristic_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     6624 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/citation_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3104 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/composite_sequence_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     6014 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/composite_sequence_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8528 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/contrast_result_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3088 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/database_entry_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5384 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/database_entry_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3052 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset1.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset10.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset2.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset3.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset4.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset5.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset6.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset7.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset8.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset9.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3064 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/dataset_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    15728 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/diff_ex_result_set_summary_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9056 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/differential_expression_analysis_result_set_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9659 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/differential_expression_analysis_result_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    14048 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/differential_expression_analysis_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5109 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/experiment_expression_levels_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9358 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/experimental_factor_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    15105 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/expression_experiment_set_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    28654 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/expression_experiment_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    31623 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/expression_experiment_with_search_result_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9207 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/external_database_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10910 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/factor_value_basic_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    16266 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/factor_value_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3104 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filter_arg_array_design.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3168 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filter_arg_expression_analysis_result_set.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3140 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filter_arg_expression_experiment.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10951 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_inferred_and_paginated_response_data_object_expression_experiment_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9152 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_array_design_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9362 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_composite_sequence_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10132 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_differential_expression_analysis_result_set_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9467 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_expression_experiment_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8907 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_gene_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    25633 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/geeq_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3040 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene1.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene2.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene3.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene4.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene5.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3052 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5754 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene_element_expressions_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7870 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene_ontology_term_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     6956 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene_set_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9642 2024-05-17 06:26:32.000000 gemmapy-1.0.1/gemmapy/sdk/models/gene_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7056 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/measurement_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3208 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/one_of_search_result_value_object_object_result_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5194 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/ontology_term_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8221 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/paginated_response_data_object_composite_sequence_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8314 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/paginated_response_data_object_expression_experiment_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7818 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/paginated_response_data_object_gene_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9986 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/paginated_results_response_data_object_differential_expression_analysis_result_set_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8127 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/physical_location_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/platform.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/platform1.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/platform2.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/platform3.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/platform4.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3060 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/platform5.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3068 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/platform_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/probe.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3088 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/quantitation_type.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3100 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/quantitation_type_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    17466 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/quantitation_type_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10454 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_and_limited_response_data_object_annotation_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10489 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_and_limited_response_data_object_array_design_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10384 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_and_limited_response_data_object_category_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    12908 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_and_paginated_response_data_object_expression_experiment_with_search_result_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     9082 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_response_data_object_taxon_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10652 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/query_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3884 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_api_info_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4099 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_annotation_search_result_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3978 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_array_design_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3945 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_bio_assay_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4187 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_differential_expression_analysis_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4143 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_experiment_expression_levels_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4033 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_gene_ontology_term_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3901 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_gene_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4033 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_physical_location_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3912 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_taxon_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3727 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_long.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3959 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_set_annotation_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4025 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_set_quantitation_type_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3906 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_simple_svd_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4521 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/response_error_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3088 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/search_result_type.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7682 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/search_result_value_object_expression_experiment_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7127 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/search_result_value_object_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4833 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/search_results_response_data_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     6838 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/search_settings_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5411 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/simple_svd_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3096 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/sort_arg_array_design.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3160 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/sort_arg_expression_analysis_result_set.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3132 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/sort_arg_expression_experiment.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     4439 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/sort_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    10286 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/statement_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3040 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxa.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3044 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon1.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon2.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon3.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon4.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon5.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon6.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon7.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon8.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3048 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon9.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3056 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon_arg.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     6759 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     8441 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/taxon_with_usage_statistics_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5162 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/vector_element_value_object.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     5078 2024-05-17 06:26:33.000000 gemmapy-1.0.1/gemmapy/sdk/models/well_composed_error_body.py
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)    13649 2024-05-17 06:26:34.000000 gemmapy-1.0.1/gemmapy/sdk/rest.py
+drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-17 06:30:06.541129 gemmapy-1.0.1/gemmapy.egg-info/
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     1730 2024-05-17 06:30:06.000000 gemmapy-1.0.1/gemmapy.egg-info/PKG-INFO
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     7909 2024-05-17 06:30:06.000000 gemmapy-1.0.1/gemmapy.egg-info/SOURCES.txt
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)        1 2024-05-17 06:30:06.000000 gemmapy-1.0.1/gemmapy.egg-info/dependency_links.txt
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)      114 2024-05-17 06:30:06.000000 gemmapy-1.0.1/gemmapy.egg-info/requires.txt
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)        8 2024-05-17 06:30:06.000000 gemmapy-1.0.1/gemmapy.egg-info/top_level.txt
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)       81 2024-05-04 04:09:08.000000 gemmapy-1.0.1/pyproject.toml
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)      404 2024-05-17 06:30:06.541129 gemmapy-1.0.1/setup.cfg
+drwxr-xr-x   0 omancarci (30067) pavlab   (30005)        0 2024-05-17 06:30:06.537129 gemmapy-1.0.1/tests/
+-rw-r--r--   0 omancarci (30067) pavlab   (30005)     3766 2024-05-08 05:16:40.000000 gemmapy-1.0.1/tests/test_basic.py
```

### Comparing `gemmapy-1.0.0/LICENSE.md` & `gemmapy-1.0.1/LICENSE.md`

 * *Files identical despite different names*

### Comparing `gemmapy-1.0.0/PKG-INFO` & `gemmapy-1.0.1/PKG-INFO`

 * *Files 0% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: gemmapy
-Version: 1.0.0
+Version: 1.0.1
 Summary: a Python Wrapper for the Gemma API
 Keywords: gemma,bioinformatics
 Requires-Python: >3.10
 License-File: LICENSE.md
 Requires-Dist: certifi>=14.05.14
 Requires-Dist: six>=1.10
 Requires-Dist: python_dateutil>=2.5.3
```

### Comparing `gemmapy-1.0.0/README.rst` & `gemmapy-1.0.1/README.rst`

 * *Files identical despite different names*

### Comparing `gemmapy-1.0.0/gemmapy/_processors.py` & `gemmapy-1.0.1/gemmapy/_processors.py`

 * *Files identical despite different names*

### Comparing `gemmapy-1.0.0/gemmapy/_subprocessors.py` & `gemmapy-1.0.1/gemmapy/_subprocessors.py`

 * *Files identical despite different names*

### Comparing `gemmapy-1.0.0/gemmapy/_validators.py` & `gemmapy-1.0.1/gemmapy/_validators.py`

 * *Files identical despite different names*

### Comparing `gemmapy-1.0.0/gemmapy/gemmapy_api.py` & `gemmapy-1.0.1/gemmapy/gemmapy_api.py`

 * *Files 0% similar despite different names*

```diff
@@ -141,5452 +141,5455 @@
 000008c0: 2020 2020 2020 3a72 6574 7572 6e3a 2044        :return: D
 000008d0: 4553 4352 4950 5449 4f4e 0a20 2020 2020  ESCRIPTION.     
 000008e0: 2020 203a 7274 7970 653a 2054 5950 450a     :rtype: TYPE.
 000008f0: 0a20 2020 2020 2020 2022 2222 0a20 2020  .        """.   
 00000900: 2020 2020 200a 2020 2020 2020 2020 7265       .        re
 00000910: 7370 6f6e 7365 203d 2073 656c 662e 7261  sponse = self.ra
 00000920: 772e 6765 745f 7265 7375 6c74 5f73 6574  w.get_result_set
-00000930: 5f61 735f 7473 7628 7265 7375 6c74 5f73  _as_tsv(result_s
-00000940: 6574 2c20 2a2a 6b77 6172 6773 290a 2020  et, **kwargs).  
-00000950: 2020 2020 2020 0a20 2020 2020 2020 2064        .        d
-00000960: 6620 3d20 7073 2e70 726f 6365 7373 5f64  f = ps.process_d
-00000970: 655f 6d61 7472 6978 2872 6573 706f 6e73  e_matrix(respons
-00000980: 652c 2072 6573 756c 745f 7365 742c 7365  e, result_set,se
-00000990: 6c66 290a 2020 2020 2020 2020 0a20 2020  lf).        .   
-000009a0: 2020 2020 2072 6574 7572 6e20 6466 0a20       return df. 
-000009b0: 2020 2020 2020 200a 0a20 2020 200a 2020         ..    .  
-000009c0: 2020 2320 2f72 6573 756c 7453 6574 732c    # /resultSets,
-000009d0: 2067 6574 5f72 6573 756c 745f 7365 7473   get_result_sets
-000009e0: 202d 2d2d 2d2d 0a20 2020 200a 2020 2020   -----.    .    
-000009f0: 6465 6620 6765 745f 7265 7375 6c74 5f73  def get_result_s
-00000a00: 6574 7328 7365 6c66 2c0a 2020 2020 2020  ets(self,.      
-00000a10: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00000a20: 2020 6461 7461 7365 7473 3a4f 7074 696f    datasets:Optio
-00000a30: 6e61 6c5b 4c69 7374 5b73 7472 7c69 6e74  nal[List[str|int
-00000a40: 5d5d 203d 204e 6f6e 652c 0a20 2020 2020  ]] = None,.     
-00000a50: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00000a60: 2020 2072 6573 756c 745f 7365 7473 3a4f     result_sets:O
-00000a70: 7074 696f 6e61 6c5b 4c69 7374 5b69 6e74  ptional[List[int
-00000a80: 5d5d 203d 204e 6f6e 652c 0a20 2020 2020  ]] = None,.     
-00000a90: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00000aa0: 2020 2066 696c 7465 723a 7374 7220 3d20     filter:str = 
-00000ab0: 4e6f 6e65 2c0a 2020 2020 2020 2020 2020  None,.          
-00000ac0: 2020 2020 2020 2020 2020 2020 2020 6f66                of
-00000ad0: 6673 6574 3a69 6e74 203d 2030 2c0a 2020  fset:int = 0,.  
-00000ae0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00000af0: 2020 2020 2020 6c69 6d69 743a 696e 7420        limit:int 
-00000b00: 3d20 3230 2c0a 2020 2020 2020 2020 2020  = 20,.          
-00000b10: 2020 2020 2020 2020 2020 2020 2020 736f                so
-00000b20: 7274 3a73 7472 203d 2022 2b69 6422 2c0a  rt:str = "+id",.
-00000b30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00000b40: 2020 2020 2020 2020 2a2a 6b77 6172 6773          **kwargs
-00000b50: 292d 3e44 6174 6146 7261 6d65 3a0a 2020  )->DataFrame:.  
-00000b60: 2020 2020 2020 2222 2252 6574 7572 6e73        """Returns
-00000b70: 2071 7565 7269 6564 2072 6573 756c 7420   queried result 
-00000b80: 7365 740a 0a20 2020 2020 2020 204f 7574  set..        Out
-00000b90: 7075 7420 616e 6420 7573 6167 6520 6f66  put and usage of
-00000ba0: 2074 6869 7320 6675 6e63 7469 6f6e 2069   this function i
-00000bb0: 7320 6d6f 7374 6c79 2069 6465 6e74 6963  s mostly identic
-00000bc0: 616c 2074 6f20 0a20 2020 2020 2020 2067  al to .        g
-00000bd0: 6574 5f64 6174 6173 6574 5f64 6966 6665  et_dataset_diffe
-00000be0: 7265 6e74 6961 6c5f 6578 7072 6573 7369  rential_expressi
-00000bf0: 6f6e 5f61 6e61 6c79 7365 732e 2054 6865  on_analyses. The
-00000c00: 2070 7269 6e63 6970 616c 2064 6966 6665   principal diffe
-00000c10: 7265 6e63 650a 2020 2020 2020 2020 6265  rence.        be
-00000c20: 696e 6720 7468 6520 6162 696c 6974 7920  ing the ability 
-00000c30: 746f 2072 6573 7472 6963 7420 796f 7572  to restrict your
-00000c40: 2072 6573 756c 7420 7365 7473 2c20 6265   result sets, be
-00000c50: 696e 6720 6162 6c65 2074 6f20 7175 6572  ing able to quer
-00000c60: 7920 0a20 2020 2020 2020 2061 6372 6f73  y .        acros
-00000c70: 7320 6d75 6c74 6970 6c65 2064 6174 6173  s multiple datas
-00000c80: 6574 7320 616e 6420 6265 696e 6720 6162  ets and being ab
-00000c90: 6c65 2074 6f20 7573 6520 7468 6520 6669  le to use the fi
-00000ca0: 6c74 6572 2061 7267 756d 656e 7420 746f  lter argument to
-00000cb0: 200a 2020 2020 2020 2020 7365 6172 6368   .        search
-00000cc0: 2062 6173 6564 206f 6e20 7265 7375 6c74   based on result
-00000cd0: 2073 6574 2070 726f 7065 7274 6965 732e   set properties.
-00000ce0: 0a0a 0a20 2020 2020 2020 200a 2020 2020  ...        .    
-00000cf0: 2020 2020 3a70 6172 616d 2064 6174 6173      :param datas
-00000d00: 6574 733a 2041 206e 756d 6572 6963 616c  ets: A numerical
-00000d10: 2064 6174 6173 6574 2069 6465 6e74 6966   dataset identif
-00000d20: 6965 7220 6f72 2061 2064 6174 6173 6574  ier or a dataset
-00000d30: 2073 686f 7274 206e 616d 652c 2064 6566   short name, def
-00000d40: 6175 6c74 7320 746f 204e 6f6e 650a 2020  aults to None.  
-00000d50: 2020 2020 2020 3a74 7970 6520 6461 7461        :type data
-00000d60: 7365 7473 3a20 4f70 7469 6f6e 616c 5b4c  sets: Optional[L
-00000d70: 6973 745b 7374 727c 696e 745d 5d2c 206f  ist[str|int]], o
-00000d80: 7074 696f 6e61 6c0a 2020 2020 2020 2020  ptional.        
-00000d90: 3a70 6172 616d 2072 6573 756c 745f 7365  :param result_se
-00000da0: 7473 3a20 0941 2072 6573 756c 7420 7365  ts: .A result se
-00000db0: 7420 6964 656e 7469 6669 6572 2e20 4e6f  t identifier. No
-00000dc0: 7465 2074 6861 7420 7265 7375 6c74 2073  te that result s
-00000dd0: 6574 2069 6465 6e74 6966 6965 7273 2061  et identifiers a
-00000de0: 7265 206e 6f74 2073 7461 7469 6320 616e  re not static an
-00000df0: 6420 6361 6e20 6368 616e 6765 2077 6865  d can change whe
-00000e00: 6e20 4765 6d6d 6120 7265 2d72 756e 7320  n Gemma re-runs 
-00000e10: 616e 616c 7973 6573 2069 6e74 6572 6e61  analyses interna
-00000e20: 6c6c 792e 2057 6865 6d20 7573 696e 6720  lly. Whem using 
-00000e30: 7468 6573 6520 6173 2069 6e70 7574 732c  these as inputs,
-00000e40: 2074 7279 2074 6f20 6d61 6b65 2073 7572   try to make sur
-00000e50: 6520 796f 7520 6163 6365 7373 2061 2063  e you access a c
-00000e60: 7572 7265 6e74 6c79 2065 7869 7374 696e  urrently existin
-00000e70: 6720 7265 7375 6c74 2073 6574 2049 4420  g result set ID 
-00000e80: 6279 2062 6173 696e 6720 7468 656d 206f  by basing them o
-00000e90: 6e20 7265 7375 6c74 2073 6574 7320 7265  n result sets re
-00000ea0: 7475 726e 6564 2066 6f72 2061 2070 6172  turned for a par
-00000eb0: 7469 6375 6c61 7220 6461 7461 7365 7420  ticular dataset 
-00000ec0: 6f72 2066 696c 7465 7220 7573 6564 2069  or filter used i
-00000ed0: 6e20 6765 745f 7265 7375 6c74 5f73 6574  n get_result_set
-00000ee0: 732c 2064 6566 6175 6c74 7320 746f 204e  s, defaults to N
-00000ef0: 6f6e 650a 2020 2020 2020 2020 3a74 7970  one.        :typ
-00000f00: 6520 7265 7375 6c74 5f73 6574 733a 204f  e result_sets: O
-00000f10: 7074 696f 6e61 6c5b 4c69 7374 5b69 6e74  ptional[List[int
-00000f20: 5d5d 2c20 6f70 7469 6f6e 616c 0a20 2020  ]], optional.   
-00000f30: 2020 2020 203a 7061 7261 6d20 6669 6c74       :param filt
-00000f40: 6572 3a20 4669 6c74 6572 2072 6573 756c  er: Filter resul
-00000f50: 7473 2062 7920 6d61 7463 6869 6e67 2065  ts by matching e
-00000f60: 7870 7265 7373 696f 6e2e 2055 7365 200a  xpression. Use .
-00000f70: 2020 2020 2020 2020 2020 6669 6c74 6572            filter
-00000f80: 5f70 726f 7065 7274 6965 7320 6675 6e63  _properties func
-00000f90: 7469 6f6e 2074 6f20 6765 7420 6120 6c69  tion to get a li
-00000fa0: 7374 206f 6620 616c 6c20 6176 6169 6c61  st of all availa
-00000fb0: 626c 6520 7061 7261 6d65 7465 7273 2e20  ble parameters. 
-00000fc0: 0a20 2020 2020 2020 2020 2054 6865 7365  .          These
-00000fd0: 2070 726f 7065 7274 6965 7320 6361 6e20   properties can 
-00000fe0: 6265 2063 6f6d 6269 6e65 6420 7573 696e  be combined usin
-00000ff0: 6720 2261 6e64 2220 226f 7222 2063 6c61  g "and" "or" cla
-00001000: 7573 6573 2061 6e64 206d 6179 200a 2020  uses and may .  
-00001010: 2020 2020 2020 2020 636f 6e74 6169 6e20          contain 
-00001020: 636f 6d6d 6f6e 206f 7065 7261 746f 7273  common operators
-00001030: 2073 7563 6820 6173 2022 3d22 2c20 223c   such as "=", "<
-00001040: 2220 6f72 2022 696e 222e 2028 652e 672e  " or "in". (e.g.
-00001050: 200a 2020 2020 2020 2020 2020 2274 6178   .          "tax
-00001060: 6f6e 2e63 6f6d 6d6f 6e4e 616d 6520 3d20  on.commonName = 
-00001070: 6875 6d61 6e22 2c20 2274 6178 6f6e 2e63  human", "taxon.c
-00001080: 6f6d 6d6f 6e4e 616d 6520 696e 2028 6875  ommonName in (hu
-00001090: 6d61 6e2c 6d6f 7573 6529 2c20 0a20 2020  man,mouse), .   
-000010a0: 2020 2020 2020 2022 6964 203c 2031 3030         "id < 100
-000010b0: 3022 292c 2064 6566 6175 6c74 7320 746f  0"), defaults to
-000010c0: 204e 6f6e 650a 2020 2020 2020 2020 3a74   None.        :t
-000010d0: 7970 6520 6669 6c74 6572 3a20 7374 722c  ype filter: str,
-000010e0: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
-000010f0: 2020 3a70 6172 616d 206f 6666 7365 743a    :param offset:
-00001100: 2054 6865 206f 6666 7365 7420 6f66 2074   The offset of t
-00001110: 6865 2066 6972 7374 2072 6574 7269 6576  he first retriev
-00001120: 6564 2072 6573 756c 742e 2c20 6465 6661  ed result., defa
-00001130: 756c 7473 2074 6f20 300a 2020 2020 2020  ults to 0.      
-00001140: 2020 3a74 7970 6520 6f66 6673 6574 3a20    :type offset: 
-00001150: 696e 742c 206f 7074 696f 6e61 6c0a 2020  int, optional.  
-00001160: 2020 2020 2020 3a70 6172 616d 206c 696d        :param lim
-00001170: 6974 3a20 4c69 6d69 7473 2074 6865 2072  it: Limits the r
-00001180: 6573 756c 7420 746f 2073 7065 6369 6669  esult to specifi
-00001190: 6564 2061 6d6f 756e 7420 6f66 206f 626a  ed amount of obj
-000011a0: 6563 7473 2e0a 2020 2020 2020 2020 2020  ects..          
-000011b0: 4861 7320 6120 6d61 7869 6d75 6d20 7661  Has a maximum va
-000011c0: 6c75 6520 6f66 2031 3030 2e20 5573 6520  lue of 100. Use 
-000011d0: 746f 6765 7468 6572 2077 6974 6820 6f66  together with of
-000011e0: 6673 6574 2061 6e64 2074 6865 200a 2020  fset and the .  
-000011f0: 2020 2020 2020 2020 746f 7461 6c5f 656c          total_el
-00001200: 656d 656e 7473 2061 7474 7269 6275 7465  ements attribute
-00001210: 2069 6e20 7468 6520 6f75 7470 7574 2074   in the output t
-00001220: 6f20 636f 6d70 696c 6520 616c 6c20 6461  o compile all da
-00001230: 7461 2069 6620 6e65 6564 6564 2e0a 2020  ta if needed..  
-00001240: 2020 2020 2020 2020 416c 7465 726e 6174          Alternat
-00001250: 6976 656c 7920 6765 745f 616c 6c5f 7061  ively get_all_pa
-00001260: 6765 7320 6675 6e63 7469 6f6e 2063 616e  ges function can
-00001270: 2062 6520 7573 6564 2077 6974 6820 616c   be used with al
-00001280: 6c20 6675 6e63 7469 6f6e 730a 2020 2020  l functions.    
-00001290: 2020 2020 2020 696e 636c 7564 696e 6720        including 
-000012a0: 6f66 6673 6574 2061 6e64 206c 696d 6974  offset and limit
-000012b0: 2070 6172 616d 6574 6572 732c 2064 6566   parameters, def
-000012c0: 6175 6c74 7320 746f 2032 300a 2020 2020  aults to 20.    
-000012d0: 2020 2020 3a74 7970 6520 6c69 6d69 743a      :type limit:
-000012e0: 2069 6e74 2c20 6f70 7469 6f6e 616c 0a20   int, optional. 
-000012f0: 2020 2020 2020 203a 7061 7261 6d20 736f         :param so
-00001300: 7274 3a20 4f72 6465 7220 7265 7375 6c74  rt: Order result
-00001310: 7320 6279 2074 6865 2067 6976 656e 2070  s by the given p
-00001320: 726f 7065 7274 7920 616e 6420 6469 7265  roperty and dire
-00001330: 6374 696f 6e2e 2054 6865 2027 2b27 0a20  ction. The '+'. 
-00001340: 2020 2020 2020 2020 2073 6967 6e20 696e           sign in
-00001350: 6469 6361 7465 2061 7363 656e 6469 6e67  dicate ascending
-00001360: 206f 7264 6572 2077 6865 7265 6173 2074   order whereas t
-00001370: 6865 2027 2d27 2069 6e64 6963 6174 6520  he '-' indicate 
-00001380: 6465 7363 656e 6469 6e67 2c0a 2020 2020  descending,.    
-00001390: 2020 2020 2020 6465 6661 756c 7473 2074        defaults t
-000013a0: 6f20 222b 6964 220a 2020 2020 2020 2020  o "+id".        
-000013b0: 3a74 7970 6520 736f 7274 3a20 7374 722c  :type sort: str,
-000013c0: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
-000013d0: 2020 3a70 6172 616d 202a 2a6b 7761 7267    :param **kwarg
-000013e0: 733a 2041 6464 6974 696f 6e61 6c20 6172  s: Additional ar
-000013f0: 6775 6d65 6e74 7320 746f 2070 6173 7320  guments to pass 
-00001400: 746f 2072 6177 2e67 6574 5f72 6573 756c  to raw.get_resul
-00001410: 745f 7365 7473 0a20 2020 2020 2020 203a  t_sets.        :
-00001420: 7265 7475 726e 3a20 4120 4461 7461 4672  return: A DataFr
-00001430: 616d 6520 7769 7468 2069 6e66 6f72 6d61  ame with informa
-00001440: 7469 6f6e 2061 626f 7574 2074 6865 2071  tion about the q
-00001450: 7565 7269 6564 2072 6573 756c 7420 7365  ueried result se
-00001460: 7473 2e20 0a20 2020 2020 2020 2020 204e  ts. .          N
-00001470: 6f74 6520 7468 6174 2074 6869 7320 6675  ote that this fu
-00001480: 6e63 7469 6f6e 2064 6f65 7320 6e6f 7420  nction does not 
-00001490: 7265 7475 726e 2064 6966 6665 7265 6e74  return different
-000014a0: 6961 6c20 6578 7072 6573 7369 6f6e 2076  ial expression v
-000014b0: 616c 7565 730a 2020 2020 2020 2020 2020  alues.          
-000014c0: 7468 656d 7365 6c76 6573 0a20 2020 2020  themselves.     
-000014d0: 2020 200a 2020 2020 2020 2020 2020 5468     .          Th
-000014e0: 6520 6669 656c 6473 206f 6620 7468 6520  e fields of the 
-000014f0: 4461 7461 4672 616d 6520 6172 653a 0a20  DataFrame are:. 
-00001500: 2020 2020 2020 2020 2020 202d 2072 6573             - res
-00001510: 756c 745f 4944 3a20 5265 7375 6c74 2073  ult_ID: Result s
-00001520: 6574 2049 4420 6f66 2074 6865 2064 6966  et ID of the dif
-00001530: 6665 7265 6e74 6961 6c20 6578 7072 6573  ferential expres
-00001540: 7369 6f6e 2061 6e61 6c79 7369 732e 204d  sion analysis. M
-00001550: 6179 2072 6570 7265 7365 6e74 206d 756c  ay represent mul
-00001560: 7469 706c 6520 6661 6374 6f72 7320 696e  tiple factors in
-00001570: 2061 2073 696e 676c 6520 6d6f 6465 6c2e   a single model.
-00001580: 0a20 2020 2020 2020 2020 2020 202d 2063  .            - c
-00001590: 6f6e 7472 6173 745f 4944 3a20 4964 206f  ontrast_ID: Id o
-000015a0: 6620 7468 6520 7370 6563 6966 6963 2063  f the specific c
-000015b0: 6f6e 7472 6173 7420 6661 6374 6f72 2e20  ontrast factor. 
-000015c0: 546f 6765 7468 6572 2077 6974 6820 7468  Together with th
-000015d0: 6520 7265 7375 6c74 2e49 4420 7468 6579  e result.ID they
-000015e0: 2075 6e69 7175 656c 7920 7265 7072 6573   uniquely repres
-000015f0: 656e 7420 6120 6769 7665 6e20 636f 6e74  ent a given cont
-00001600: 7261 7374 2e0a 2020 2020 2020 2020 2020  rast..          
-00001610: 2020 2d20 6578 7065 7269 6d65 6e74 5f49    - experiment_I
-00001620: 443a 2049 6420 6f66 2074 6865 2073 6f75  D: Id of the sou
-00001630: 7263 6520 6578 7065 7269 6d65 6e74 0a20  rce experiment. 
-00001640: 2020 2020 2020 2020 2020 202d 2066 6163             - fac
-00001650: 746f 725f 6361 7465 676f 7279 3a20 4361  tor_category: Ca
-00001660: 7465 676f 7279 2066 6f72 2074 6865 2063  tegory for the c
-00001670: 6f6e 7472 6173 740a 2020 2020 2020 2020  ontrast.        
+00000930: 2872 6573 756c 745f 7365 742c 202a 2a6b  (result_set, **k
+00000940: 7761 7267 732c 0a20 2020 2020 2020 2020  wargs,.         
+00000950: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00000960: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00000970: 2020 5f66 6f72 6365 5f74 6162 6c65 203d    _force_table =
+00000980: 2054 7275 6529 0a20 2020 2020 2020 200a   True).        .
+00000990: 2020 2020 2020 2020 6466 203d 2070 732e          df = ps.
+000009a0: 7072 6f63 6573 735f 6465 5f6d 6174 7269  process_de_matri
+000009b0: 7828 7265 7370 6f6e 7365 2c20 7265 7375  x(response, resu
+000009c0: 6c74 5f73 6574 2c73 656c 6629 0a20 2020  lt_set,self).   
+000009d0: 2020 2020 200a 2020 2020 2020 2020 7265       .        re
+000009e0: 7475 726e 2064 660a 2020 2020 2020 2020  turn df.        
+000009f0: 0a0a 2020 2020 0a20 2020 2023 202f 7265  ..    .    # /re
+00000a00: 7375 6c74 5365 7473 2c20 6765 745f 7265  sultSets, get_re
+00000a10: 7375 6c74 5f73 6574 7320 2d2d 2d2d 2d0a  sult_sets -----.
+00000a20: 2020 2020 0a20 2020 2064 6566 2067 6574      .    def get
+00000a30: 5f72 6573 756c 745f 7365 7473 2873 656c  _result_sets(sel
+00000a40: 662c 0a20 2020 2020 2020 2020 2020 2020  f,.             
+00000a50: 2020 2020 2020 2020 2020 2064 6174 6173             datas
+00000a60: 6574 733a 4f70 7469 6f6e 616c 5b4c 6973  ets:Optional[Lis
+00000a70: 745b 7374 727c 696e 745d 5d20 3d20 4e6f  t[str|int]] = No
+00000a80: 6e65 2c0a 2020 2020 2020 2020 2020 2020  ne,.            
+00000a90: 2020 2020 2020 2020 2020 2020 7265 7375              resu
+00000aa0: 6c74 5f73 6574 733a 4f70 7469 6f6e 616c  lt_sets:Optional
+00000ab0: 5b4c 6973 745b 696e 745d 5d20 3d20 4e6f  [List[int]] = No
+00000ac0: 6e65 2c0a 2020 2020 2020 2020 2020 2020  ne,.            
+00000ad0: 2020 2020 2020 2020 2020 2020 6669 6c74              filt
+00000ae0: 6572 3a73 7472 203d 204e 6f6e 652c 0a20  er:str = None,. 
+00000af0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00000b00: 2020 2020 2020 206f 6666 7365 743a 696e         offset:in
+00000b10: 7420 3d20 302c 0a20 2020 2020 2020 2020  t = 0,.         
+00000b20: 2020 2020 2020 2020 2020 2020 2020 206c                 l
+00000b30: 696d 6974 3a69 6e74 203d 2032 302c 0a20  imit:int = 20,. 
+00000b40: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00000b50: 2020 2020 2020 2073 6f72 743a 7374 7220         sort:str 
+00000b60: 3d20 222b 6964 222c 0a20 2020 2020 2020  = "+id",.       
+00000b70: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00000b80: 202a 2a6b 7761 7267 7329 2d3e 4461 7461   **kwargs)->Data
+00000b90: 4672 616d 653a 0a20 2020 2020 2020 2022  Frame:.        "
+00000ba0: 2222 5265 7475 726e 7320 7175 6572 6965  ""Returns querie
+00000bb0: 6420 7265 7375 6c74 2073 6574 0a0a 2020  d result set..  
+00000bc0: 2020 2020 2020 4f75 7470 7574 2061 6e64        Output and
+00000bd0: 2075 7361 6765 206f 6620 7468 6973 2066   usage of this f
+00000be0: 756e 6374 696f 6e20 6973 206d 6f73 746c  unction is mostl
+00000bf0: 7920 6964 656e 7469 6361 6c20 746f 200a  y identical to .
+00000c00: 2020 2020 2020 2020 6765 745f 6461 7461          get_data
+00000c10: 7365 745f 6469 6666 6572 656e 7469 616c  set_differential
+00000c20: 5f65 7870 7265 7373 696f 6e5f 616e 616c  _expression_anal
+00000c30: 7973 6573 2e20 5468 6520 7072 696e 6369  yses. The princi
+00000c40: 7061 6c20 6469 6666 6572 656e 6365 0a20  pal difference. 
+00000c50: 2020 2020 2020 2062 6569 6e67 2074 6865         being the
+00000c60: 2061 6269 6c69 7479 2074 6f20 7265 7374   ability to rest
+00000c70: 7269 6374 2079 6f75 7220 7265 7375 6c74  rict your result
+00000c80: 2073 6574 732c 2062 6569 6e67 2061 626c   sets, being abl
+00000c90: 6520 746f 2071 7565 7279 200a 2020 2020  e to query .    
+00000ca0: 2020 2020 6163 726f 7373 206d 756c 7469      across multi
+00000cb0: 706c 6520 6461 7461 7365 7473 2061 6e64  ple datasets and
+00000cc0: 2062 6569 6e67 2061 626c 6520 746f 2075   being able to u
+00000cd0: 7365 2074 6865 2066 696c 7465 7220 6172  se the filter ar
+00000ce0: 6775 6d65 6e74 2074 6f20 0a20 2020 2020  gument to .     
+00000cf0: 2020 2073 6561 7263 6820 6261 7365 6420     search based 
+00000d00: 6f6e 2072 6573 756c 7420 7365 7420 7072  on result set pr
+00000d10: 6f70 6572 7469 6573 2e0a 0a0a 2020 2020  operties....    
+00000d20: 2020 2020 0a20 2020 2020 2020 203a 7061      .        :pa
+00000d30: 7261 6d20 6461 7461 7365 7473 3a20 4120  ram datasets: A 
+00000d40: 6e75 6d65 7269 6361 6c20 6461 7461 7365  numerical datase
+00000d50: 7420 6964 656e 7469 6669 6572 206f 7220  t identifier or 
+00000d60: 6120 6461 7461 7365 7420 7368 6f72 7420  a dataset short 
+00000d70: 6e61 6d65 2c20 6465 6661 756c 7473 2074  name, defaults t
+00000d80: 6f20 4e6f 6e65 0a20 2020 2020 2020 203a  o None.        :
+00000d90: 7479 7065 2064 6174 6173 6574 733a 204f  type datasets: O
+00000da0: 7074 696f 6e61 6c5b 4c69 7374 5b73 7472  ptional[List[str
+00000db0: 7c69 6e74 5d5d 2c20 6f70 7469 6f6e 616c  |int]], optional
+00000dc0: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+00000dd0: 7265 7375 6c74 5f73 6574 733a 2009 4120  result_sets: .A 
+00000de0: 7265 7375 6c74 2073 6574 2069 6465 6e74  result set ident
+00000df0: 6966 6965 722e 204e 6f74 6520 7468 6174  ifier. Note that
+00000e00: 2072 6573 756c 7420 7365 7420 6964 656e   result set iden
+00000e10: 7469 6669 6572 7320 6172 6520 6e6f 7420  tifiers are not 
+00000e20: 7374 6174 6963 2061 6e64 2063 616e 2063  static and can c
+00000e30: 6861 6e67 6520 7768 656e 2047 656d 6d61  hange when Gemma
+00000e40: 2072 652d 7275 6e73 2061 6e61 6c79 7365   re-runs analyse
+00000e50: 7320 696e 7465 726e 616c 6c79 2e20 5768  s internally. Wh
+00000e60: 656d 2075 7369 6e67 2074 6865 7365 2061  em using these a
+00000e70: 7320 696e 7075 7473 2c20 7472 7920 746f  s inputs, try to
+00000e80: 206d 616b 6520 7375 7265 2079 6f75 2061   make sure you a
+00000e90: 6363 6573 7320 6120 6375 7272 656e 746c  ccess a currentl
+00000ea0: 7920 6578 6973 7469 6e67 2072 6573 756c  y existing resul
+00000eb0: 7420 7365 7420 4944 2062 7920 6261 7369  t set ID by basi
+00000ec0: 6e67 2074 6865 6d20 6f6e 2072 6573 756c  ng them on resul
+00000ed0: 7420 7365 7473 2072 6574 7572 6e65 6420  t sets returned 
+00000ee0: 666f 7220 6120 7061 7274 6963 756c 6172  for a particular
+00000ef0: 2064 6174 6173 6574 206f 7220 6669 6c74   dataset or filt
+00000f00: 6572 2075 7365 6420 696e 2067 6574 5f72  er used in get_r
+00000f10: 6573 756c 745f 7365 7473 2c20 6465 6661  esult_sets, defa
+00000f20: 756c 7473 2074 6f20 4e6f 6e65 0a20 2020  ults to None.   
+00000f30: 2020 2020 203a 7479 7065 2072 6573 756c       :type resul
+00000f40: 745f 7365 7473 3a20 4f70 7469 6f6e 616c  t_sets: Optional
+00000f50: 5b4c 6973 745b 696e 745d 5d2c 206f 7074  [List[int]], opt
+00000f60: 696f 6e61 6c0a 2020 2020 2020 2020 3a70  ional.        :p
+00000f70: 6172 616d 2066 696c 7465 723a 2046 696c  aram filter: Fil
+00000f80: 7465 7220 7265 7375 6c74 7320 6279 206d  ter results by m
+00000f90: 6174 6368 696e 6720 6578 7072 6573 7369  atching expressi
+00000fa0: 6f6e 2e20 5573 6520 0a20 2020 2020 2020  on. Use .       
+00000fb0: 2020 2066 696c 7465 725f 7072 6f70 6572     filter_proper
+00000fc0: 7469 6573 2066 756e 6374 696f 6e20 746f  ties function to
+00000fd0: 2067 6574 2061 206c 6973 7420 6f66 2061   get a list of a
+00000fe0: 6c6c 2061 7661 696c 6162 6c65 2070 6172  ll available par
+00000ff0: 616d 6574 6572 732e 200a 2020 2020 2020  ameters. .      
+00001000: 2020 2020 5468 6573 6520 7072 6f70 6572      These proper
+00001010: 7469 6573 2063 616e 2062 6520 636f 6d62  ties can be comb
+00001020: 696e 6564 2075 7369 6e67 2022 616e 6422  ined using "and"
+00001030: 2022 6f72 2220 636c 6175 7365 7320 616e   "or" clauses an
+00001040: 6420 6d61 7920 0a20 2020 2020 2020 2020  d may .         
+00001050: 2063 6f6e 7461 696e 2063 6f6d 6d6f 6e20   contain common 
+00001060: 6f70 6572 6174 6f72 7320 7375 6368 2061  operators such a
+00001070: 7320 223d 222c 2022 3c22 206f 7220 2269  s "=", "<" or "i
+00001080: 6e22 2e20 2865 2e67 2e20 0a20 2020 2020  n". (e.g. .     
+00001090: 2020 2020 2022 7461 786f 6e2e 636f 6d6d       "taxon.comm
+000010a0: 6f6e 4e61 6d65 203d 2068 756d 616e 222c  onName = human",
+000010b0: 2022 7461 786f 6e2e 636f 6d6d 6f6e 4e61   "taxon.commonNa
+000010c0: 6d65 2069 6e20 2868 756d 616e 2c6d 6f75  me in (human,mou
+000010d0: 7365 292c 200a 2020 2020 2020 2020 2020  se), .          
+000010e0: 2269 6420 3c20 3130 3030 2229 2c20 6465  "id < 1000"), de
+000010f0: 6661 756c 7473 2074 6f20 4e6f 6e65 0a20  faults to None. 
+00001100: 2020 2020 2020 203a 7479 7065 2066 696c         :type fil
+00001110: 7465 723a 2073 7472 2c20 6f70 7469 6f6e  ter: str, option
+00001120: 616c 0a20 2020 2020 2020 203a 7061 7261  al.        :para
+00001130: 6d20 6f66 6673 6574 3a20 5468 6520 6f66  m offset: The of
+00001140: 6673 6574 206f 6620 7468 6520 6669 7273  fset of the firs
+00001150: 7420 7265 7472 6965 7665 6420 7265 7375  t retrieved resu
+00001160: 6c74 2e2c 2064 6566 6175 6c74 7320 746f  lt., defaults to
+00001170: 2030 0a20 2020 2020 2020 203a 7479 7065   0.        :type
+00001180: 206f 6666 7365 743a 2069 6e74 2c20 6f70   offset: int, op
+00001190: 7469 6f6e 616c 0a20 2020 2020 2020 203a  tional.        :
+000011a0: 7061 7261 6d20 6c69 6d69 743a 204c 696d  param limit: Lim
+000011b0: 6974 7320 7468 6520 7265 7375 6c74 2074  its the result t
+000011c0: 6f20 7370 6563 6966 6965 6420 616d 6f75  o specified amou
+000011d0: 6e74 206f 6620 6f62 6a65 6374 732e 0a20  nt of objects.. 
+000011e0: 2020 2020 2020 2020 2048 6173 2061 206d           Has a m
+000011f0: 6178 696d 756d 2076 616c 7565 206f 6620  aximum value of 
+00001200: 3130 302e 2055 7365 2074 6f67 6574 6865  100. Use togethe
+00001210: 7220 7769 7468 206f 6666 7365 7420 616e  r with offset an
+00001220: 6420 7468 6520 0a20 2020 2020 2020 2020  d the .         
+00001230: 2074 6f74 616c 5f65 6c65 6d65 6e74 7320   total_elements 
+00001240: 6174 7472 6962 7574 6520 696e 2074 6865  attribute in the
+00001250: 206f 7574 7075 7420 746f 2063 6f6d 7069   output to compi
+00001260: 6c65 2061 6c6c 2064 6174 6120 6966 206e  le all data if n
+00001270: 6565 6465 642e 0a20 2020 2020 2020 2020  eeded..         
+00001280: 2041 6c74 6572 6e61 7469 7665 6c79 2067   Alternatively g
+00001290: 6574 5f61 6c6c 5f70 6167 6573 2066 756e  et_all_pages fun
+000012a0: 6374 696f 6e20 6361 6e20 6265 2075 7365  ction can be use
+000012b0: 6420 7769 7468 2061 6c6c 2066 756e 6374  d with all funct
+000012c0: 696f 6e73 0a20 2020 2020 2020 2020 2069  ions.          i
+000012d0: 6e63 6c75 6469 6e67 206f 6666 7365 7420  ncluding offset 
+000012e0: 616e 6420 6c69 6d69 7420 7061 7261 6d65  and limit parame
+000012f0: 7465 7273 2c20 6465 6661 756c 7473 2074  ters, defaults t
+00001300: 6f20 3230 0a20 2020 2020 2020 203a 7479  o 20.        :ty
+00001310: 7065 206c 696d 6974 3a20 696e 742c 206f  pe limit: int, o
+00001320: 7074 696f 6e61 6c0a 2020 2020 2020 2020  ptional.        
+00001330: 3a70 6172 616d 2073 6f72 743a 204f 7264  :param sort: Ord
+00001340: 6572 2072 6573 756c 7473 2062 7920 7468  er results by th
+00001350: 6520 6769 7665 6e20 7072 6f70 6572 7479  e given property
+00001360: 2061 6e64 2064 6972 6563 7469 6f6e 2e20   and direction. 
+00001370: 5468 6520 272b 270a 2020 2020 2020 2020  The '+'.        
+00001380: 2020 7369 676e 2069 6e64 6963 6174 6520    sign indicate 
+00001390: 6173 6365 6e64 696e 6720 6f72 6465 7220  ascending order 
+000013a0: 7768 6572 6561 7320 7468 6520 272d 2720  whereas the '-' 
+000013b0: 696e 6469 6361 7465 2064 6573 6365 6e64  indicate descend
+000013c0: 696e 672c 0a20 2020 2020 2020 2020 2064  ing,.          d
+000013d0: 6566 6175 6c74 7320 746f 2022 2b69 6422  efaults to "+id"
+000013e0: 0a20 2020 2020 2020 203a 7479 7065 2073  .        :type s
+000013f0: 6f72 743a 2073 7472 2c20 6f70 7469 6f6e  ort: str, option
+00001400: 616c 0a20 2020 2020 2020 203a 7061 7261  al.        :para
+00001410: 6d20 2a2a 6b77 6172 6773 3a20 4164 6469  m **kwargs: Addi
+00001420: 7469 6f6e 616c 2061 7267 756d 656e 7473  tional arguments
+00001430: 2074 6f20 7061 7373 2074 6f20 7261 772e   to pass to raw.
+00001440: 6765 745f 7265 7375 6c74 5f73 6574 730a  get_result_sets.
+00001450: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
+00001460: 2041 2044 6174 6146 7261 6d65 2077 6974   A DataFrame wit
+00001470: 6820 696e 666f 726d 6174 696f 6e20 6162  h information ab
+00001480: 6f75 7420 7468 6520 7175 6572 6965 6420  out the queried 
+00001490: 7265 7375 6c74 2073 6574 732e 200a 2020  result sets. .  
+000014a0: 2020 2020 2020 2020 4e6f 7465 2074 6861          Note tha
+000014b0: 7420 7468 6973 2066 756e 6374 696f 6e20  t this function 
+000014c0: 646f 6573 206e 6f74 2072 6574 7572 6e20  does not return 
+000014d0: 6469 6666 6572 656e 7469 616c 2065 7870  differential exp
+000014e0: 7265 7373 696f 6e20 7661 6c75 6573 0a20  ression values. 
+000014f0: 2020 2020 2020 2020 2074 6865 6d73 656c           themsel
+00001500: 7665 730a 2020 2020 2020 2020 0a20 2020  ves.        .   
+00001510: 2020 2020 2020 2054 6865 2066 6965 6c64         The field
+00001520: 7320 6f66 2074 6865 2044 6174 6146 7261  s of the DataFra
+00001530: 6d65 2061 7265 3a0a 2020 2020 2020 2020  me are:.        
+00001540: 2020 2020 2d20 7265 7375 6c74 5f49 443a      - result_ID:
+00001550: 2052 6573 756c 7420 7365 7420 4944 206f   Result set ID o
+00001560: 6620 7468 6520 6469 6666 6572 656e 7469  f the differenti
+00001570: 616c 2065 7870 7265 7373 696f 6e20 616e  al expression an
+00001580: 616c 7973 6973 2e20 4d61 7920 7265 7072  alysis. May repr
+00001590: 6573 656e 7420 6d75 6c74 6970 6c65 2066  esent multiple f
+000015a0: 6163 746f 7273 2069 6e20 6120 7369 6e67  actors in a sing
+000015b0: 6c65 206d 6f64 656c 2e0a 2020 2020 2020  le model..      
+000015c0: 2020 2020 2020 2d20 636f 6e74 7261 7374        - contrast
+000015d0: 5f49 443a 2049 6420 6f66 2074 6865 2073  _ID: Id of the s
+000015e0: 7065 6369 6669 6320 636f 6e74 7261 7374  pecific contrast
+000015f0: 2066 6163 746f 722e 2054 6f67 6574 6865   factor. Togethe
+00001600: 7220 7769 7468 2074 6865 2072 6573 756c  r with the resul
+00001610: 742e 4944 2074 6865 7920 756e 6971 7565  t.ID they unique
+00001620: 6c79 2072 6570 7265 7365 6e74 2061 2067  ly represent a g
+00001630: 6976 656e 2063 6f6e 7472 6173 742e 0a20  iven contrast.. 
+00001640: 2020 2020 2020 2020 2020 202d 2065 7870             - exp
+00001650: 6572 696d 656e 745f 4944 3a20 4964 206f  eriment_ID: Id o
+00001660: 6620 7468 6520 736f 7572 6365 2065 7870  f the source exp
+00001670: 6572 696d 656e 740a 2020 2020 2020 2020  eriment.        
 00001680: 2020 2020 2d20 6661 6374 6f72 5f63 6174      - factor_cat
-00001690: 6567 6f72 795f 5552 493a 2055 5249 2066  egory_URI: URI f
-000016a0: 6f72 2074 6865 2062 6173 656c 696e 6520  or the baseline 
-000016b0: 6361 7465 676f 7279 0a20 2020 2020 2020  category.       
-000016c0: 2020 2020 202d 2066 6163 746f 725f 4944       - factor_ID
-000016d0: 3a20 4944 206f 6620 7468 6520 6661 6374  : ID of the fact
-000016e0: 6f72 0a20 2020 2020 2020 2020 2020 202d  or.            -
-000016f0: 2062 6173 656c 696e 655f 6661 6374 6f72   baseline_factor
-00001700: 733a 2043 6861 7261 6374 6572 6973 7469  s: Characteristi
-00001710: 6373 206f 6620 7468 6520 6261 7365 6c69  cs of the baseli
-00001720: 6e65 2e20 5468 6973 2066 6965 6c64 2069  ne. This field i
-00001730: 7320 6120 4461 7461 4672 616d 650a 2020  s a DataFrame.  
-00001740: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
-00001750: 7269 6d65 6e74 616c 5f66 6163 746f 7273  rimental_factors
-00001760: 3a20 4368 6172 6163 7465 7269 7374 6963  : Characteristic
-00001770: 7320 6f66 2074 6865 2065 7870 6572 696d  s of the experim
-00001780: 656e 7461 6c20 6772 6f75 702e 2054 6869  ental group. Thi
-00001790: 7320 6669 656c 6420 6973 2061 2044 6174  s field is a Dat
-000017a0: 6146 7261 6d65 0a20 2020 2020 2020 2020  aFrame.         
-000017b0: 2020 202d 2069 735f 7375 6273 6574 3a20     - is_subset: 
-000017c0: 5472 7565 2069 6620 7468 6520 7265 7375  True if the resu
-000017d0: 6c74 2073 6574 2062 656c 6f6e 6720 746f  lt set belong to
-000017e0: 2061 2073 7562 7365 742c 2046 616c 7365   a subset, False
-000017f0: 2069 6620 6e6f 742e 2053 7562 7365 7473   if not. Subsets
-00001800: 2061 7265 2063 7265 6174 6564 2077 6865   are created whe
-00001810: 6e20 7065 7266 6f72 6d69 6e67 2064 6966  n performing dif
-00001820: 6665 7265 6e74 6961 6c20 6578 7072 6573  ferential expres
-00001830: 7369 6f6e 2074 6f20 6176 6f69 6420 756e  sion to avoid un
-00001840: 6865 6c70 6675 6c20 636f 6d70 6172 6973  helpful comparis
-00001850: 6f6e 732e 0a20 2020 2020 2020 2020 2020  ons..           
-00001860: 202d 2073 7562 7365 745f 6661 6374 6f72   - subset_factor
-00001870: 3a20 4368 6172 6163 7465 7269 7374 6963  : Characteristic
-00001880: 7320 6f66 2074 6865 2073 7562 7365 742e  s of the subset.
-00001890: 2054 6869 7320 6669 656c 6420 6973 2061   This field is a
-000018a0: 2044 6174 6146 7261 6d65 0a20 2020 2020   DataFrame.     
-000018b0: 2020 200a 2020 2020 2020 2020 3a72 7479     .        :rty
-000018c0: 7065 3a20 4461 7461 4672 616d 650a 0a20  pe: DataFrame.. 
-000018d0: 2020 2020 2020 2022 2222 0a20 2020 2020         """.     
-000018e0: 2020 200a 2020 2020 2020 2020 6669 6c74     .        filt
-000018f0: 6572 203d 2076 732e 6164 645f 746f 5f66  er = vs.add_to_f
-00001900: 696c 7465 7228 6669 6c74 6572 2c20 2769  ilter(filter, 'i
-00001910: 6427 2c20 7265 7375 6c74 5f73 6574 7329  d', result_sets)
-00001920: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
-00001930: 2020 0a20 2020 2020 2020 206b 7761 7267    .        kwarg
-00001940: 7320 3d20 7673 2e72 656d 6f76 655f 6e6f  s = vs.remove_no
-00001950: 6e65 7328 0a20 2020 2020 2020 2020 2020  nes(.           
-00001960: 2064 6174 6173 6574 7320 3d20 6461 7461   datasets = data
-00001970: 7365 7473 2c0a 2020 2020 2020 2020 2020  sets,.          
-00001980: 2020 6669 6c74 6572 203d 2066 696c 7465    filter = filte
-00001990: 722c 0a20 2020 2020 2020 2020 2020 206f  r,.            o
-000019a0: 6666 7365 7420 3d20 6f66 6673 6574 2c0a  ffset = offset,.
-000019b0: 2020 2020 2020 2020 2020 2020 6c69 6d69              limi
-000019c0: 7420 3d20 6c69 6d69 742c 0a20 2020 2020  t = limit,.     
-000019d0: 2020 2020 2020 2073 6f72 7420 3d20 736f         sort = so
-000019e0: 7274 2c0a 2020 2020 2020 2020 2020 2020  rt,.            
-000019f0: 2a2a 6b77 6172 6773 290a 2020 2020 2020  **kwargs).      
-00001a00: 2020 0a20 2020 2020 2020 2072 6573 706f    .        respo
-00001a10: 6e73 6520 3d20 7365 6c66 2e72 6177 2e67  nse = self.raw.g
-00001a20: 6574 5f72 6573 756c 745f 7365 7473 282a  et_result_sets(*
-00001a30: 2a6b 7761 7267 7329 0a20 2020 2020 2020  *kwargs).       
-00001a40: 2064 6620 3d20 7073 2e70 726f 6365 7373   df = ps.process
-00001a50: 5f44 6966 6665 7265 6e74 6961 6c45 7870  _DifferentialExp
-00001a60: 7265 7373 696f 6e41 6e61 6c79 7369 7352  ressionAnalysisR
-00001a70: 6573 756c 7453 6574 5661 6c75 654f 626a  esultSetValueObj
-00001a80: 6563 7428 7265 7370 6f6e 7365 2e64 6174  ect(response.dat
-00001a90: 612c 7365 6c66 290a 2020 2020 2020 2020  a,self).        
-00001aa0: 7073 2e61 7474 6163 685f 6174 7472 6962  ps.attach_attrib
-00001ab0: 7574 6573 2864 662c 2072 6573 706f 6e73  utes(df, respons
-00001ac0: 652e 746f 5f64 6963 7428 2929 0a0a 2020  e.to_dict())..  
-00001ad0: 2020 2020 2020 7265 7475 726e 2064 660a        return df.
-00001ae0: 0a20 2020 200a 2020 2020 2320 2f61 6e6e  .    .    # /ann
-00001af0: 6f74 6174 696f 6e73 2f73 6561 7263 682c  otations/search,
-00001b00: 2073 6561 7263 685f 616e 6e6f 7461 7469   search_annotati
-00001b10: 6f6e 7320 2d2d 2d2d 2d2d 2d2d 0a20 2020  ons --------.   
-00001b20: 2064 6566 2073 6561 7263 685f 616e 6e6f   def search_anno
-00001b30: 7461 7469 6f6e 7328 7365 6c66 2c20 7175  tations(self, qu
-00001b40: 6572 793a 4c69 7374 5b73 7472 5d2c 202a  ery:List[str], *
-00001b50: 2a6b 7761 7267 7329 2d3e 4461 7461 4672  *kwargs)->DataFr
-00001b60: 616d 653a 0a20 2020 2020 2020 2022 2222  ame:.        """
-00001b70: 0a20 2020 2020 2020 2053 6561 7263 6820  .        Search 
-00001b80: 666f 7220 616e 6e6f 7461 7469 6f6e 2074  for annotation t
-00001b90: 6167 730a 0a0a 2020 2020 2020 2020 3a70  ags...        :p
-00001ba0: 6172 616d 2071 7565 7279 3a20 5468 6520  aram query: The 
-00001bb0: 7365 6172 6368 2071 7565 7279 0a20 2020  search query.   
-00001bc0: 2020 2020 203a 7479 7065 2071 7565 7279       :type query
-00001bd0: 3a20 4c69 7374 5b73 7472 5d0a 2020 2020  : List[str].    
-00001be0: 2020 2020 3a70 6172 616d 202a 2a6b 7761      :param **kwa
-00001bf0: 7267 733a 2041 6464 6974 696f 6e61 6c20  rgs: Additional 
-00001c00: 6172 6775 6d65 6e74 7320 746f 2070 6173  arguments to pas
-00001c10: 7320 746f 2072 6177 2e73 6561 7263 685f  s to raw.search_
-00001c20: 616e 6e6f 7461 7469 6f6e 730a 2020 2020  annotations.    
-00001c30: 2020 2020 3a72 6574 7572 6e3a 2041 2044      :return: A D
-00001c40: 6174 6146 7261 6d65 2077 6974 6820 616e  ataFrame with an
-00001c50: 6e6f 7461 7469 6f6e 7320 6d61 7463 6869  notations matchi
-00001c60: 6e67 2074 6865 2067 6976 656e 2069 6465  ng the given ide
-00001c70: 6e74 6966 6965 7273 2e20 0a20 2020 2020  ntifiers. .     
-00001c80: 2020 2020 2054 6865 2066 6965 6c64 7320       The fields 
-00001c90: 6f66 2074 6865 2044 6174 6146 7261 6d65  of the DataFrame
-00001ca0: 2061 7265 3a0a 2020 2020 2020 2020 2020   are:.          
-00001cb0: 2020 2d20 6361 7465 676f 7279 5f6e 616d    - category_nam
-00001cc0: 653a 2043 6174 6567 6f72 7920 7468 6174  e: Category that
-00001cd0: 2074 6865 2061 6e6e 6f74 6174 696f 6e20   the annotation 
-00001ce0: 6265 6c6f 6e67 7320 746f 0a20 2020 2020  belongs to.     
-00001cf0: 2020 2020 2020 202d 2063 6174 6567 6f72         - categor
-00001d00: 795f 5552 493a 2055 5249 2066 6f72 2074  y_URI: URI for t
-00001d10: 6865 2063 6174 6567 6f72 795f 6e61 6d65  he category_name
-00001d20: 0a20 2020 2020 2020 2020 2020 202d 2076  .            - v
-00001d30: 616c 7565 5f6e 616d 653a 2041 6e6e 6f74  alue_name: Annot
-00001d40: 6174 696f 6e20 7465 726d 0a20 2020 2020  ation term.     
-00001d50: 2020 2020 2020 202d 2076 616c 7565 5f55         - value_U
-00001d60: 5249 3a20 5552 4920 666f 7220 7468 6520  RI: URI for the 
-00001d70: 7661 6c75 655f 6e61 6d65 0a20 2020 2020  value_name.     
-00001d80: 2020 203a 7274 7970 653a 2044 6174 6146     :rtype: DataF
-00001d90: 7261 6d65 0a0a 2020 2020 2020 2020 2222  rame..        ""
-00001da0: 220a 0a20 2020 2020 2020 2072 6573 706f  "..        respo
-00001db0: 6e73 6520 3d20 7365 6c66 2e72 6177 2e73  nse = self.raw.s
-00001dc0: 6561 7263 685f 616e 6e6f 7461 7469 6f6e  earch_annotation
-00001dd0: 7328 7175 6572 793d 7175 6572 792c 202a  s(query=query, *
-00001de0: 2a6b 7761 7267 7329 0a20 2020 2020 2020  *kwargs).       
-00001df0: 2072 6574 7572 6e20 7073 2e70 726f 6365   return ps.proce
-00001e00: 7373 5f73 6561 7263 685f 616e 6e6f 7461  ss_search_annota
-00001e10: 7469 6f6e 7328 7265 7370 6f6e 7365 2e64  tions(response.d
-00001e20: 6174 6129 0a20 2020 200a 0a20 2020 2023  ata).    ..    #
-00001e30: 202f 6461 7461 7365 7473 2f7b 6461 7461   /datasets/{data
-00001e40: 7365 747d 2f61 6e6e 6f74 6174 696f 6e73  set}/annotations
-00001e50: 2c20 6765 745f 6461 7461 7365 745f 616e  , get_dataset_an
-00001e60: 6e6f 7461 7469 6f6e 7320 2d2d 2d2d 2d2d  notations ------
-00001e70: 2d2d 2d2d 0a20 2020 2064 6566 2067 6574  ----.    def get
-00001e80: 5f64 6174 6173 6574 5f61 6e6e 6f74 6174  _dataset_annotat
-00001e90: 696f 6e73 2873 656c 662c 2064 6174 6173  ions(self, datas
-00001ea0: 6574 3a73 7472 7c69 6e74 2c20 2a2a 6b77  et:str|int, **kw
-00001eb0: 6172 6773 292d 3e44 6174 6146 7261 6d65  args)->DataFrame
-00001ec0: 3a0a 2020 2020 2020 2020 2222 220a 2020  :.        """.  
-00001ed0: 2020 2020 2020 5265 7472 6965 7665 2074        Retrieve t
-00001ee0: 6865 2061 6e6e 6f74 6174 696f 6e73 206f  he annotations o
-00001ef0: 6620 6120 6461 7461 7365 740a 0a0a 2020  f a dataset...  
-00001f00: 2020 2020 2020 3a70 6172 616d 2064 6174        :param dat
-00001f10: 6173 6574 3a20 0941 206e 756d 6572 6963  aset: .A numeric
-00001f20: 616c 2064 6174 6173 6574 2069 6465 6e74  al dataset ident
-00001f30: 6966 6965 7220 6f72 2061 2064 6174 6173  ifier or a datas
-00001f40: 6574 2073 686f 7274 206e 616d 650a 2020  et short name.  
-00001f50: 2020 2020 2020 3a74 7970 6520 6461 7461        :type data
-00001f60: 7365 743a 2073 7472 7c69 6e74 0a20 2020  set: str|int.   
-00001f70: 2020 2020 203a 7061 7261 6d20 2a2a 6b77       :param **kw
-00001f80: 6172 6773 3a20 4164 6469 7469 6f6e 616c  args: Additional
-00001f90: 2061 7267 756d 656e 7473 2074 6f20 7061   arguments to pa
-00001fa0: 7373 2074 6f20 7261 772e 6765 745f 6461  ss to raw.get_da
-00001fb0: 7461 7365 745f 616e 6e6f 7461 7469 6f6e  taset_annotation
-00001fc0: 730a 2020 2020 2020 2020 3a72 6574 7572  s.        :retur
-00001fd0: 6e3a 2041 2044 6174 6146 7261 6d65 2077  n: A DataFrame w
-00001fe0: 6974 6820 696e 666f 726d 6174 696f 6e20  ith information 
-00001ff0: 6162 6f75 7420 7468 6520 616e 6e6f 7461  about the annota
-00002000: 7469 6f6e 7320 6f66 2074 6865 2071 7565  tions of the que
-00002010: 7269 6564 2064 6174 6173 6574 2e20 0a20  ried dataset. . 
-00002020: 2020 2020 2020 2020 2054 6865 2066 6965           The fie
-00002030: 6c64 7320 6f66 2074 6865 2044 6174 6146  lds of the DataF
-00002040: 7261 6d65 2061 7265 3a0a 2020 2020 2020  rame are:.      
-00002050: 2020 2020 2020 2d20 636c 6173 735f 6e61        - class_na
-00002060: 6d65 3a20 4e61 6d65 206f 6620 7468 6520  me: Name of the 
-00002070: 616e 6e6f 7461 7469 6f6e 2063 6c61 7373  annotation class
-00002080: 2028 652e 672e 206f 7267 616e 6973 6d20   (e.g. organism 
-00002090: 7061 7274 290a 2020 2020 2020 2020 2020  part).          
-000020a0: 2020 2d20 636c 6173 735f 5552 493a 2055    - class_URI: U
-000020b0: 5249 2066 6f72 2074 6865 2061 6e6e 6f74  RI for the annot
-000020c0: 6174 696f 6e20 636c 6173 730a 2020 2020  ation class.    
-000020d0: 2020 2020 2020 2020 2d20 7465 726d 5f6e          - term_n
-000020e0: 616d 653a 204e 616d 6520 6f66 2074 6865  ame: Name of the
-000020f0: 2061 6e6e 6f74 6174 696f 6e20 7465 726d   annotation term
-00002100: 2028 652e 672e 206c 756e 6729 0a20 2020   (e.g. lung).   
-00002110: 2020 2020 2020 2020 202d 2074 6572 6d5f           - term_
-00002120: 5552 493a 2055 5249 2066 6f72 2074 6865  URI: URI for the
-00002130: 2061 6e6e 6f74 6174 696f 6e20 7465 726d   annotation term
-00002140: 0a20 2020 2020 2020 2020 2020 202d 206f  .            - o
-00002150: 626a 6563 745f 636c 6173 733a 2043 6c61  bject_class: Cla
-00002160: 7373 206f 6620 6f62 6a65 6374 2074 6861  ss of object tha
-00002170: 7420 7468 6520 7465 726d 206f 7269 6769  t the term origi
-00002180: 6e61 7465 6420 6672 6f6d 2e20 2020 2020  nated from.     
-00002190: 2020 0a20 2020 2020 2020 203a 7274 7970    .        :rtyp
-000021a0: 653a 2044 6174 6146 7261 6d65 0a0a 2020  e: DataFrame..  
-000021b0: 2020 2020 2020 2222 220a 0a20 2020 2020        """..     
-000021c0: 2020 2072 6573 706f 6e73 6520 3d20 7365     response = se
-000021d0: 6c66 2e72 6177 2e67 6574 5f64 6174 6173  lf.raw.get_datas
-000021e0: 6574 5f61 6e6e 6f74 6174 696f 6e73 2864  et_annotations(d
-000021f0: 6174 6173 6574 2c20 2a2a 6b77 6172 6773  ataset, **kwargs
-00002200: 290a 2020 2020 2020 2020 6466 203d 2070  ).        df = p
-00002210: 732e 7072 6f63 6573 735f 616e 6e6f 7461  s.process_annota
-00002220: 7469 6f6e 7328 7265 7370 6f6e 7365 2e64  tions(response.d
-00002230: 6174 6129 0a20 2020 2020 2020 2070 732e  ata).        ps.
-00002240: 6174 7461 6368 5f61 7474 7269 6275 7465  attach_attribute
-00002250: 7328 6466 2c20 7265 7370 6f6e 7365 2e74  s(df, response.t
-00002260: 6f5f 6469 6374 2829 290a 2020 2020 2020  o_dict()).      
-00002270: 2020 0a20 2020 2020 2020 2072 6574 7572    .        retur
-00002280: 6e20 6466 0a20 2020 200a 2020 2020 2320  n df.    .    # 
-00002290: 2f64 6174 6173 6574 732f 7b64 6174 6173  /datasets/{datas
-000022a0: 6574 7d2f 6465 7369 676e 2c20 6765 745f  et}/design, get_
-000022b0: 6461 7461 7365 745f 6465 7369 676e 202d  dataset_design -
-000022c0: 2d2d 2d2d 0a20 2020 2023 2074 6869 7320  ----.    # this 
-000022d0: 656e 6470 6f69 6e74 2069 7320 6e6f 7420  endpoint is not 
-000022e0: 7665 7279 2075 7365 6675 6c20 7369 6e63  very useful sinc
-000022f0: 6520 7468 6520 6e61 6d65 7320 6974 2063  e the names it c
-00002300: 6f6d 6573 2077 6974 680a 2020 2020 2320  omes with.    # 
-00002310: 6973 2061 6e6e 6f79 696e 6720 746f 206d  is annoying to m
-00002320: 6174 6368 206e 616d 6573 2070 726f 7669  atch names provi
-00002330: 6465 6420 696e 2074 6865 2073 616d 706c  ded in the sampl
-00002340: 6573 2065 6e64 706f 696e 740a 2020 2020  es endpoint.    
-00002350: 2320 6d61 6b65 5f64 6573 6967 6e20 7265  # make_design re
-00002360: 706c 6163 6573 2074 6869 730a 2020 2020  places this.    
-00002370: 6465 6620 5f5f 6765 745f 6461 7461 7365  def __get_datase
-00002380: 745f 6465 7369 676e 2873 656c 662c 2064  t_design(self, d
-00002390: 6174 6173 6574 3a73 7472 7c69 6e74 2c20  ataset:str|int, 
-000023a0: 2a2a 6b77 6172 6773 293a 0a20 2020 2020  **kwargs):.     
-000023b0: 2020 2022 2222 0a20 2020 2020 2020 200a     """.        .
-000023c0: 2020 2020 2020 2020 3a70 6172 616d 2064          :param d
-000023d0: 6174 6173 6574 3a20 4445 5343 5249 5054  ataset: DESCRIPT
-000023e0: 494f 4e0a 2020 2020 2020 2020 3a74 7970  ION.        :typ
-000023f0: 6520 6461 7461 7365 743a 2073 7472 7c69  e dataset: str|i
-00002400: 6e74 0a20 2020 2020 2020 203a 7061 7261  nt.        :para
-00002410: 6d20 2a2a 6b77 6172 6773 3a20 4164 6469  m **kwargs: Addi
-00002420: 7469 6f6e 616c 2061 7267 756d 656e 7473  tional arguments
-00002430: 2074 6f20 7061 7373 2074 6f20 7261 772e   to pass to raw.
-00002440: 6765 745f 6461 7461 7365 745f 6465 7369  get_dataset_desi
-00002450: 676e 0a20 2020 2020 2020 203a 7479 7065  gn.        :type
-00002460: 202a 2a6b 7761 7267 733a 2054 5950 450a   **kwargs: TYPE.
-00002470: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
-00002480: 2044 4553 4352 4950 5449 4f4e 0a20 2020   DESCRIPTION.   
-00002490: 2020 2020 203a 7274 7970 653a 2054 5950       :rtype: TYP
-000024a0: 450a 0a20 2020 2020 2020 2022 2222 0a0a  E..        """..
-000024b0: 2020 2020 2020 2020 7265 7370 6f6e 7365          response
-000024c0: 203d 2073 656c 662e 7261 772e 6765 745f   = self.raw.get_
-000024d0: 6461 7461 7365 745f 6465 7369 676e 2864  dataset_design(d
-000024e0: 6174 6173 6574 2c20 2a2a 6b77 6172 6773  ataset, **kwargs
-000024f0: 290a 2020 2020 2020 2020 6466 203d 2070  ).        df = p
-00002500: 732e 7072 6f63 6573 735f 6461 7461 7365  s.process_datase
-00002510: 745f 6465 7369 676e 2872 6573 706f 6e73  t_design(respons
-00002520: 6529 0a20 2020 2020 2020 200a 2020 2020  e).        .    
-00002530: 2020 2020 7265 7475 726e 2064 660a 2020      return df.  
-00002540: 2020 0a20 2020 2023 202f 6461 7461 7365    .    # /datase
-00002550: 7473 2f7b 6461 7461 7365 7473 7d2f 6578  ts/{datasets}/ex
-00002560: 7072 6573 7369 6f6e 732f 6469 6666 6572  pressions/differ
-00002570: 656e 7469 616c 202d 2d2d 2d2d 2d0a 2020  ential ------.  
-00002580: 2020 2320 756e 696d 706c 656d 656e 7465    # unimplemente
-00002590: 640a 2020 2020 2320 6e6f 7420 7375 7265  d.    # not sure
-000025a0: 2068 6f77 2074 6865 2070 6172 616d 6574   how the paramet
-000025b0: 6572 7320 666f 7220 7468 6973 2065 6e64  ers for this end
-000025c0: 706f 696e 7420 776f 726b 7320 616e 6420  point works and 
-000025d0: 646f 6573 6e27 7420 7365 656d 2065 7373  doesn't seem ess
-000025e0: 656e 7469 616c 0a20 2020 200a 2020 2020  ential.    .    
-000025f0: 2320 2f64 6174 6173 6574 732f 7b64 6174  # /datasets/{dat
-00002600: 6173 6574 7d2f 616e 616c 7973 6573 2f64  aset}/analyses/d
-00002610: 6966 6665 7265 6e74 6961 6c2c 2067 6574  ifferential, get
-00002620: 5f64 6174 6173 6574 5f64 6966 6665 7265  _dataset_differe
-00002630: 6e74 6961 6c5f 6578 7072 6573 7369 6f6e  ntial_expression
-00002640: 5f61 6e61 6c79 7365 7320 2d2d 2d2d 2d2d  _analyses ------
-00002650: 0a20 2020 2064 6566 2067 6574 5f64 6174  .    def get_dat
-00002660: 6173 6574 5f64 6966 6665 7265 6e74 6961  aset_differentia
-00002670: 6c5f 6578 7072 6573 7369 6f6e 5f61 6e61  l_expression_ana
-00002680: 6c79 7365 7328 7365 6c66 2c20 0a20 2020  lyses(self, .   
-00002690: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000026a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000026b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000026c0: 2020 6461 7461 7365 743a 7374 727c 696e    dataset:str|in
-000026d0: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
+00001690: 6567 6f72 793a 2043 6174 6567 6f72 7920  egory: Category 
+000016a0: 666f 7220 7468 6520 636f 6e74 7261 7374  for the contrast
+000016b0: 0a20 2020 2020 2020 2020 2020 202d 2066  .            - f
+000016c0: 6163 746f 725f 6361 7465 676f 7279 5f55  actor_category_U
+000016d0: 5249 3a20 5552 4920 666f 7220 7468 6520  RI: URI for the 
+000016e0: 6261 7365 6c69 6e65 2063 6174 6567 6f72  baseline categor
+000016f0: 790a 2020 2020 2020 2020 2020 2020 2d20  y.            - 
+00001700: 6661 6374 6f72 5f49 443a 2049 4420 6f66  factor_ID: ID of
+00001710: 2074 6865 2066 6163 746f 720a 2020 2020   the factor.    
+00001720: 2020 2020 2020 2020 2d20 6261 7365 6c69          - baseli
+00001730: 6e65 5f66 6163 746f 7273 3a20 4368 6172  ne_factors: Char
+00001740: 6163 7465 7269 7374 6963 7320 6f66 2074  acteristics of t
+00001750: 6865 2062 6173 656c 696e 652e 2054 6869  he baseline. Thi
+00001760: 7320 6669 656c 6420 6973 2061 2044 6174  s field is a Dat
+00001770: 6146 7261 6d65 0a20 2020 2020 2020 2020  aFrame.         
+00001780: 2020 202d 2065 7870 6572 696d 656e 7461     - experimenta
+00001790: 6c5f 6661 6374 6f72 733a 2043 6861 7261  l_factors: Chara
+000017a0: 6374 6572 6973 7469 6373 206f 6620 7468  cteristics of th
+000017b0: 6520 6578 7065 7269 6d65 6e74 616c 2067  e experimental g
+000017c0: 726f 7570 2e20 5468 6973 2066 6965 6c64  roup. This field
+000017d0: 2069 7320 6120 4461 7461 4672 616d 650a   is a DataFrame.
+000017e0: 2020 2020 2020 2020 2020 2020 2d20 6973              - is
+000017f0: 5f73 7562 7365 743a 2054 7275 6520 6966  _subset: True if
+00001800: 2074 6865 2072 6573 756c 7420 7365 7420   the result set 
+00001810: 6265 6c6f 6e67 2074 6f20 6120 7375 6273  belong to a subs
+00001820: 6574 2c20 4661 6c73 6520 6966 206e 6f74  et, False if not
+00001830: 2e20 5375 6273 6574 7320 6172 6520 6372  . Subsets are cr
+00001840: 6561 7465 6420 7768 656e 2070 6572 666f  eated when perfo
+00001850: 726d 696e 6720 6469 6666 6572 656e 7469  rming differenti
+00001860: 616c 2065 7870 7265 7373 696f 6e20 746f  al expression to
+00001870: 2061 766f 6964 2075 6e68 656c 7066 756c   avoid unhelpful
+00001880: 2063 6f6d 7061 7269 736f 6e73 2e0a 2020   comparisons..  
+00001890: 2020 2020 2020 2020 2020 2d20 7375 6273            - subs
+000018a0: 6574 5f66 6163 746f 723a 2043 6861 7261  et_factor: Chara
+000018b0: 6374 6572 6973 7469 6373 206f 6620 7468  cteristics of th
+000018c0: 6520 7375 6273 6574 2e20 5468 6973 2066  e subset. This f
+000018d0: 6965 6c64 2069 7320 6120 4461 7461 4672  ield is a DataFr
+000018e0: 616d 650a 2020 2020 2020 2020 0a20 2020  ame.        .   
+000018f0: 2020 2020 203a 7274 7970 653a 2044 6174       :rtype: Dat
+00001900: 6146 7261 6d65 0a0a 2020 2020 2020 2020  aFrame..        
+00001910: 2222 220a 2020 2020 2020 2020 0a20 2020  """.        .   
+00001920: 2020 2020 2066 696c 7465 7220 3d20 7673       filter = vs
+00001930: 2e61 6464 5f74 6f5f 6669 6c74 6572 2866  .add_to_filter(f
+00001940: 696c 7465 722c 2027 6964 272c 2072 6573  ilter, 'id', res
+00001950: 756c 745f 7365 7473 290a 2020 2020 2020  ult_sets).      
+00001960: 2020 0a20 2020 2020 2020 200a 2020 2020    .        .    
+00001970: 2020 2020 6b77 6172 6773 203d 2076 732e      kwargs = vs.
+00001980: 7265 6d6f 7665 5f6e 6f6e 6573 280a 2020  remove_nones(.  
+00001990: 2020 2020 2020 2020 2020 6461 7461 7365            datase
+000019a0: 7473 203d 2064 6174 6173 6574 732c 0a20  ts = datasets,. 
+000019b0: 2020 2020 2020 2020 2020 2066 696c 7465             filte
+000019c0: 7220 3d20 6669 6c74 6572 2c0a 2020 2020  r = filter,.    
+000019d0: 2020 2020 2020 2020 6f66 6673 6574 203d          offset =
+000019e0: 206f 6666 7365 742c 0a20 2020 2020 2020   offset,.       
+000019f0: 2020 2020 206c 696d 6974 203d 206c 696d       limit = lim
+00001a00: 6974 2c0a 2020 2020 2020 2020 2020 2020  it,.            
+00001a10: 736f 7274 203d 2073 6f72 742c 0a20 2020  sort = sort,.   
+00001a20: 2020 2020 2020 2020 202a 2a6b 7761 7267           **kwarg
+00001a30: 7329 0a20 2020 2020 2020 200a 2020 2020  s).        .    
+00001a40: 2020 2020 7265 7370 6f6e 7365 203d 2073      response = s
+00001a50: 656c 662e 7261 772e 6765 745f 7265 7375  elf.raw.get_resu
+00001a60: 6c74 5f73 6574 7328 2a2a 6b77 6172 6773  lt_sets(**kwargs
+00001a70: 290a 2020 2020 2020 2020 6466 203d 2070  ).        df = p
+00001a80: 732e 7072 6f63 6573 735f 4469 6666 6572  s.process_Differ
+00001a90: 656e 7469 616c 4578 7072 6573 7369 6f6e  entialExpression
+00001aa0: 416e 616c 7973 6973 5265 7375 6c74 5365  AnalysisResultSe
+00001ab0: 7456 616c 7565 4f62 6a65 6374 2872 6573  tValueObject(res
+00001ac0: 706f 6e73 652e 6461 7461 2c73 656c 6629  ponse.data,self)
+00001ad0: 0a20 2020 2020 2020 2070 732e 6174 7461  .        ps.atta
+00001ae0: 6368 5f61 7474 7269 6275 7465 7328 6466  ch_attributes(df
+00001af0: 2c20 7265 7370 6f6e 7365 2e74 6f5f 6469  , response.to_di
+00001b00: 6374 2829 290a 0a20 2020 2020 2020 2072  ct())..        r
+00001b10: 6574 7572 6e20 6466 0a0a 2020 2020 0a20  eturn df..    . 
+00001b20: 2020 2023 202f 616e 6e6f 7461 7469 6f6e     # /annotation
+00001b30: 732f 7365 6172 6368 2c20 7365 6172 6368  s/search, search
+00001b40: 5f61 6e6e 6f74 6174 696f 6e73 202d 2d2d  _annotations ---
+00001b50: 2d2d 2d2d 2d0a 2020 2020 6465 6620 7365  -----.    def se
+00001b60: 6172 6368 5f61 6e6e 6f74 6174 696f 6e73  arch_annotations
+00001b70: 2873 656c 662c 2071 7565 7279 3a4c 6973  (self, query:Lis
+00001b80: 745b 7374 725d 2c20 2a2a 6b77 6172 6773  t[str], **kwargs
+00001b90: 292d 3e44 6174 6146 7261 6d65 3a0a 2020  )->DataFrame:.  
+00001ba0: 2020 2020 2020 2222 220a 2020 2020 2020        """.      
+00001bb0: 2020 5365 6172 6368 2066 6f72 2061 6e6e    Search for ann
+00001bc0: 6f74 6174 696f 6e20 7461 6773 0a0a 0a20  otation tags... 
+00001bd0: 2020 2020 2020 203a 7061 7261 6d20 7175         :param qu
+00001be0: 6572 793a 2054 6865 2073 6561 7263 6820  ery: The search 
+00001bf0: 7175 6572 790a 2020 2020 2020 2020 3a74  query.        :t
+00001c00: 7970 6520 7175 6572 793a 204c 6973 745b  ype query: List[
+00001c10: 7374 725d 0a20 2020 2020 2020 203a 7061  str].        :pa
+00001c20: 7261 6d20 2a2a 6b77 6172 6773 3a20 4164  ram **kwargs: Ad
+00001c30: 6469 7469 6f6e 616c 2061 7267 756d 656e  ditional argumen
+00001c40: 7473 2074 6f20 7061 7373 2074 6f20 7261  ts to pass to ra
+00001c50: 772e 7365 6172 6368 5f61 6e6e 6f74 6174  w.search_annotat
+00001c60: 696f 6e73 0a20 2020 2020 2020 203a 7265  ions.        :re
+00001c70: 7475 726e 3a20 4120 4461 7461 4672 616d  turn: A DataFram
+00001c80: 6520 7769 7468 2061 6e6e 6f74 6174 696f  e with annotatio
+00001c90: 6e73 206d 6174 6368 696e 6720 7468 6520  ns matching the 
+00001ca0: 6769 7665 6e20 6964 656e 7469 6669 6572  given identifier
+00001cb0: 732e 200a 2020 2020 2020 2020 2020 5468  s. .          Th
+00001cc0: 6520 6669 656c 6473 206f 6620 7468 6520  e fields of the 
+00001cd0: 4461 7461 4672 616d 6520 6172 653a 0a20  DataFrame are:. 
+00001ce0: 2020 2020 2020 2020 2020 202d 2063 6174             - cat
+00001cf0: 6567 6f72 795f 6e61 6d65 3a20 4361 7465  egory_name: Cate
+00001d00: 676f 7279 2074 6861 7420 7468 6520 616e  gory that the an
+00001d10: 6e6f 7461 7469 6f6e 2062 656c 6f6e 6773  notation belongs
+00001d20: 2074 6f0a 2020 2020 2020 2020 2020 2020   to.            
+00001d30: 2d20 6361 7465 676f 7279 5f55 5249 3a20  - category_URI: 
+00001d40: 5552 4920 666f 7220 7468 6520 6361 7465  URI for the cate
+00001d50: 676f 7279 5f6e 616d 650a 2020 2020 2020  gory_name.      
+00001d60: 2020 2020 2020 2d20 7661 6c75 655f 6e61        - value_na
+00001d70: 6d65 3a20 416e 6e6f 7461 7469 6f6e 2074  me: Annotation t
+00001d80: 6572 6d0a 2020 2020 2020 2020 2020 2020  erm.            
+00001d90: 2d20 7661 6c75 655f 5552 493a 2055 5249  - value_URI: URI
+00001da0: 2066 6f72 2074 6865 2076 616c 7565 5f6e   for the value_n
+00001db0: 616d 650a 2020 2020 2020 2020 3a72 7479  ame.        :rty
+00001dc0: 7065 3a20 4461 7461 4672 616d 650a 0a20  pe: DataFrame.. 
+00001dd0: 2020 2020 2020 2022 2222 0a0a 2020 2020         """..    
+00001de0: 2020 2020 7265 7370 6f6e 7365 203d 2073      response = s
+00001df0: 656c 662e 7261 772e 7365 6172 6368 5f61  elf.raw.search_a
+00001e00: 6e6e 6f74 6174 696f 6e73 2871 7565 7279  nnotations(query
+00001e10: 3d71 7565 7279 2c20 2a2a 6b77 6172 6773  =query, **kwargs
+00001e20: 290a 2020 2020 2020 2020 7265 7475 726e  ).        return
+00001e30: 2070 732e 7072 6f63 6573 735f 7365 6172   ps.process_sear
+00001e40: 6368 5f61 6e6e 6f74 6174 696f 6e73 2872  ch_annotations(r
+00001e50: 6573 706f 6e73 652e 6461 7461 290a 2020  esponse.data).  
+00001e60: 2020 0a0a 2020 2020 2320 2f64 6174 6173    ..    # /datas
+00001e70: 6574 732f 7b64 6174 6173 6574 7d2f 616e  ets/{dataset}/an
+00001e80: 6e6f 7461 7469 6f6e 732c 2067 6574 5f64  notations, get_d
+00001e90: 6174 6173 6574 5f61 6e6e 6f74 6174 696f  ataset_annotatio
+00001ea0: 6e73 202d 2d2d 2d2d 2d2d 2d2d 2d0a 2020  ns ----------.  
+00001eb0: 2020 6465 6620 6765 745f 6461 7461 7365    def get_datase
+00001ec0: 745f 616e 6e6f 7461 7469 6f6e 7328 7365  t_annotations(se
+00001ed0: 6c66 2c20 6461 7461 7365 743a 7374 727c  lf, dataset:str|
+00001ee0: 696e 742c 202a 2a6b 7761 7267 7329 2d3e  int, **kwargs)->
+00001ef0: 4461 7461 4672 616d 653a 0a20 2020 2020  DataFrame:.     
+00001f00: 2020 2022 2222 0a20 2020 2020 2020 2052     """.        R
+00001f10: 6574 7269 6576 6520 7468 6520 616e 6e6f  etrieve the anno
+00001f20: 7461 7469 6f6e 7320 6f66 2061 2064 6174  tations of a dat
+00001f30: 6173 6574 0a0a 0a20 2020 2020 2020 203a  aset...        :
+00001f40: 7061 7261 6d20 6461 7461 7365 743a 2009  param dataset: .
+00001f50: 4120 6e75 6d65 7269 6361 6c20 6461 7461  A numerical data
+00001f60: 7365 7420 6964 656e 7469 6669 6572 206f  set identifier o
+00001f70: 7220 6120 6461 7461 7365 7420 7368 6f72  r a dataset shor
+00001f80: 7420 6e61 6d65 0a20 2020 2020 2020 203a  t name.        :
+00001f90: 7479 7065 2064 6174 6173 6574 3a20 7374  type dataset: st
+00001fa0: 727c 696e 740a 2020 2020 2020 2020 3a70  r|int.        :p
+00001fb0: 6172 616d 202a 2a6b 7761 7267 733a 2041  aram **kwargs: A
+00001fc0: 6464 6974 696f 6e61 6c20 6172 6775 6d65  dditional argume
+00001fd0: 6e74 7320 746f 2070 6173 7320 746f 2072  nts to pass to r
+00001fe0: 6177 2e67 6574 5f64 6174 6173 6574 5f61  aw.get_dataset_a
+00001ff0: 6e6e 6f74 6174 696f 6e73 0a20 2020 2020  nnotations.     
+00002000: 2020 203a 7265 7475 726e 3a20 4120 4461     :return: A Da
+00002010: 7461 4672 616d 6520 7769 7468 2069 6e66  taFrame with inf
+00002020: 6f72 6d61 7469 6f6e 2061 626f 7574 2074  ormation about t
+00002030: 6865 2061 6e6e 6f74 6174 696f 6e73 206f  he annotations o
+00002040: 6620 7468 6520 7175 6572 6965 6420 6461  f the queried da
+00002050: 7461 7365 742e 200a 2020 2020 2020 2020  taset. .        
+00002060: 2020 5468 6520 6669 656c 6473 206f 6620    The fields of 
+00002070: 7468 6520 4461 7461 4672 616d 6520 6172  the DataFrame ar
+00002080: 653a 0a20 2020 2020 2020 2020 2020 202d  e:.            -
+00002090: 2063 6c61 7373 5f6e 616d 653a 204e 616d   class_name: Nam
+000020a0: 6520 6f66 2074 6865 2061 6e6e 6f74 6174  e of the annotat
+000020b0: 696f 6e20 636c 6173 7320 2865 2e67 2e20  ion class (e.g. 
+000020c0: 6f72 6761 6e69 736d 2070 6172 7429 0a20  organism part). 
+000020d0: 2020 2020 2020 2020 2020 202d 2063 6c61             - cla
+000020e0: 7373 5f55 5249 3a20 5552 4920 666f 7220  ss_URI: URI for 
+000020f0: 7468 6520 616e 6e6f 7461 7469 6f6e 2063  the annotation c
+00002100: 6c61 7373 0a20 2020 2020 2020 2020 2020  lass.           
+00002110: 202d 2074 6572 6d5f 6e61 6d65 3a20 4e61   - term_name: Na
+00002120: 6d65 206f 6620 7468 6520 616e 6e6f 7461  me of the annota
+00002130: 7469 6f6e 2074 6572 6d20 2865 2e67 2e20  tion term (e.g. 
+00002140: 6c75 6e67 290a 2020 2020 2020 2020 2020  lung).          
+00002150: 2020 2d20 7465 726d 5f55 5249 3a20 5552    - term_URI: UR
+00002160: 4920 666f 7220 7468 6520 616e 6e6f 7461  I for the annota
+00002170: 7469 6f6e 2074 6572 6d0a 2020 2020 2020  tion term.      
+00002180: 2020 2020 2020 2d20 6f62 6a65 6374 5f63        - object_c
+00002190: 6c61 7373 3a20 436c 6173 7320 6f66 206f  lass: Class of o
+000021a0: 626a 6563 7420 7468 6174 2074 6865 2074  bject that the t
+000021b0: 6572 6d20 6f72 6967 696e 6174 6564 2066  erm originated f
+000021c0: 726f 6d2e 2020 2020 2020 200a 2020 2020  rom.       .    
+000021d0: 2020 2020 3a72 7479 7065 3a20 4461 7461      :rtype: Data
+000021e0: 4672 616d 650a 0a20 2020 2020 2020 2022  Frame..        "
+000021f0: 2222 0a0a 2020 2020 2020 2020 7265 7370  ""..        resp
+00002200: 6f6e 7365 203d 2073 656c 662e 7261 772e  onse = self.raw.
+00002210: 6765 745f 6461 7461 7365 745f 616e 6e6f  get_dataset_anno
+00002220: 7461 7469 6f6e 7328 6461 7461 7365 742c  tations(dataset,
+00002230: 202a 2a6b 7761 7267 7329 0a20 2020 2020   **kwargs).     
+00002240: 2020 2064 6620 3d20 7073 2e70 726f 6365     df = ps.proce
+00002250: 7373 5f61 6e6e 6f74 6174 696f 6e73 2872  ss_annotations(r
+00002260: 6573 706f 6e73 652e 6461 7461 290a 2020  esponse.data).  
+00002270: 2020 2020 2020 7073 2e61 7474 6163 685f        ps.attach_
+00002280: 6174 7472 6962 7574 6573 2864 662c 2072  attributes(df, r
+00002290: 6573 706f 6e73 652e 746f 5f64 6963 7428  esponse.to_dict(
+000022a0: 2929 0a20 2020 2020 2020 200a 2020 2020  )).        .    
+000022b0: 2020 2020 7265 7475 726e 2064 660a 2020      return df.  
+000022c0: 2020 0a20 2020 2023 202f 6461 7461 7365    .    # /datase
+000022d0: 7473 2f7b 6461 7461 7365 747d 2f64 6573  ts/{dataset}/des
+000022e0: 6967 6e2c 2067 6574 5f64 6174 6173 6574  ign, get_dataset
+000022f0: 5f64 6573 6967 6e20 2d2d 2d2d 2d0a 2020  _design -----.  
+00002300: 2020 2320 7468 6973 2065 6e64 706f 696e    # this endpoin
+00002310: 7420 6973 206e 6f74 2076 6572 7920 7573  t is not very us
+00002320: 6566 756c 2073 696e 6365 2074 6865 206e  eful since the n
+00002330: 616d 6573 2069 7420 636f 6d65 7320 7769  ames it comes wi
+00002340: 7468 0a20 2020 2023 2069 7320 616e 6e6f  th.    # is anno
+00002350: 7969 6e67 2074 6f20 6d61 7463 6820 6e61  ying to match na
+00002360: 6d65 7320 7072 6f76 6964 6564 2069 6e20  mes provided in 
+00002370: 7468 6520 7361 6d70 6c65 7320 656e 6470  the samples endp
+00002380: 6f69 6e74 0a20 2020 2023 206d 616b 655f  oint.    # make_
+00002390: 6465 7369 676e 2072 6570 6c61 6365 7320  design replaces 
+000023a0: 7468 6973 0a20 2020 2064 6566 205f 5f67  this.    def __g
+000023b0: 6574 5f64 6174 6173 6574 5f64 6573 6967  et_dataset_desig
+000023c0: 6e28 7365 6c66 2c20 6461 7461 7365 743a  n(self, dataset:
+000023d0: 7374 727c 696e 742c 202a 2a6b 7761 7267  str|int, **kwarg
+000023e0: 7329 3a0a 2020 2020 2020 2020 2222 220a  s):.        """.
+000023f0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00002400: 203a 7061 7261 6d20 6461 7461 7365 743a   :param dataset:
+00002410: 2044 4553 4352 4950 5449 4f4e 0a20 2020   DESCRIPTION.   
+00002420: 2020 2020 203a 7479 7065 2064 6174 6173       :type datas
+00002430: 6574 3a20 7374 727c 696e 740a 2020 2020  et: str|int.    
+00002440: 2020 2020 3a70 6172 616d 202a 2a6b 7761      :param **kwa
+00002450: 7267 733a 2041 6464 6974 696f 6e61 6c20  rgs: Additional 
+00002460: 6172 6775 6d65 6e74 7320 746f 2070 6173  arguments to pas
+00002470: 7320 746f 2072 6177 2e67 6574 5f64 6174  s to raw.get_dat
+00002480: 6173 6574 5f64 6573 6967 6e0a 2020 2020  aset_design.    
+00002490: 2020 2020 3a74 7970 6520 2a2a 6b77 6172      :type **kwar
+000024a0: 6773 3a20 5459 5045 0a20 2020 2020 2020  gs: TYPE.       
+000024b0: 203a 7265 7475 726e 3a20 4445 5343 5249   :return: DESCRI
+000024c0: 5054 494f 4e0a 2020 2020 2020 2020 3a72  PTION.        :r
+000024d0: 7479 7065 3a20 5459 5045 0a0a 2020 2020  type: TYPE..    
+000024e0: 2020 2020 2222 220a 0a20 2020 2020 2020      """..       
+000024f0: 2072 6573 706f 6e73 6520 3d20 7365 6c66   response = self
+00002500: 2e72 6177 2e67 6574 5f64 6174 6173 6574  .raw.get_dataset
+00002510: 5f64 6573 6967 6e28 6461 7461 7365 742c  _design(dataset,
+00002520: 202a 2a6b 7761 7267 7329 0a20 2020 2020   **kwargs).     
+00002530: 2020 2064 6620 3d20 7073 2e70 726f 6365     df = ps.proce
+00002540: 7373 5f64 6174 6173 6574 5f64 6573 6967  ss_dataset_desig
+00002550: 6e28 7265 7370 6f6e 7365 290a 2020 2020  n(response).    
+00002560: 2020 2020 0a20 2020 2020 2020 2072 6574      .        ret
+00002570: 7572 6e20 6466 0a20 2020 200a 2020 2020  urn df.    .    
+00002580: 2320 2f64 6174 6173 6574 732f 7b64 6174  # /datasets/{dat
+00002590: 6173 6574 737d 2f65 7870 7265 7373 696f  asets}/expressio
+000025a0: 6e73 2f64 6966 6665 7265 6e74 6961 6c20  ns/differential 
+000025b0: 2d2d 2d2d 2d2d 0a20 2020 2023 2075 6e69  ------.    # uni
+000025c0: 6d70 6c65 6d65 6e74 6564 0a20 2020 2023  mplemented.    #
+000025d0: 206e 6f74 2073 7572 6520 686f 7720 7468   not sure how th
+000025e0: 6520 7061 7261 6d65 7465 7273 2066 6f72  e parameters for
+000025f0: 2074 6869 7320 656e 6470 6f69 6e74 2077   this endpoint w
+00002600: 6f72 6b73 2061 6e64 2064 6f65 736e 2774  orks and doesn't
+00002610: 2073 6565 6d20 6573 7365 6e74 6961 6c0a   seem essential.
+00002620: 2020 2020 0a20 2020 2023 202f 6461 7461      .    # /data
+00002630: 7365 7473 2f7b 6461 7461 7365 747d 2f61  sets/{dataset}/a
+00002640: 6e61 6c79 7365 732f 6469 6666 6572 656e  nalyses/differen
+00002650: 7469 616c 2c20 6765 745f 6461 7461 7365  tial, get_datase
+00002660: 745f 6469 6666 6572 656e 7469 616c 5f65  t_differential_e
+00002670: 7870 7265 7373 696f 6e5f 616e 616c 7973  xpression_analys
+00002680: 6573 202d 2d2d 2d2d 2d0a 2020 2020 6465  es ------.    de
+00002690: 6620 6765 745f 6461 7461 7365 745f 6469  f get_dataset_di
+000026a0: 6666 6572 656e 7469 616c 5f65 7870 7265  fferential_expre
+000026b0: 7373 696f 6e5f 616e 616c 7973 6573 2873  ssion_analyses(s
+000026c0: 656c 662c 200a 2020 2020 2020 2020 2020  elf, .          
+000026d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
 000026e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000026f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00002700: 2020 2020 2020 2020 2a2a 6b77 6172 6773          **kwargs
-00002710: 292d 3e44 6174 6146 7261 6d65 3a0a 2020  )->DataFrame:.  
-00002720: 2020 2020 2020 2222 2252 6574 7269 6576        """Retriev
-00002730: 6520 616e 6e6f 7461 7469 6f6e 7320 616e  e annotations an
-00002740: 6420 7375 7266 6163 6520 6c65 7665 6c20  d surface level 
-00002750: 7374 6174 7320 666f 7220 6120 6461 7461  stats for a data
-00002760: 7365 7427 7320 6469 6666 6572 656e 7469  set's differenti
-00002770: 616c 2061 6e61 6c79 7365 730a 2020 2020  al analyses.    
-00002780: 2020 2020 0a20 2020 2020 2020 203a 7061      .        :pa
-00002790: 7261 6d20 6461 7461 7365 743a 2041 206e  ram dataset: A n
-000027a0: 756d 6572 6963 616c 2064 6174 6173 6574  umerical dataset
-000027b0: 2069 6465 6e74 6966 6965 7220 6f72 2061   identifier or a
-000027c0: 2064 6174 6173 6574 2073 686f 7274 206e   dataset short n
-000027d0: 616d 650a 2020 2020 2020 2020 3a74 7970  ame.        :typ
-000027e0: 6520 6461 7461 7365 743a 2073 7472 7c69  e dataset: str|i
-000027f0: 6e74 0a20 2020 2020 2020 203a 7061 7261  nt.        :para
-00002800: 6d20 2a2a 6b77 6172 6773 3a20 4164 6469  m **kwargs: Addi
-00002810: 7469 6f6e 616c 2061 7267 756d 656e 7473  tional arguments
-00002820: 2074 6f20 7061 7373 2074 6f20 7261 772e   to pass to raw.
-00002830: 6765 745f 6461 7461 7365 745f 6469 6666  get_dataset_diff
-00002840: 6572 656e 7469 616c 5f65 7870 7265 7373  erential_express
-00002850: 696f 6e5f 616e 616c 7973 6573 0a20 2020  ion_analyses.   
-00002860: 2020 2020 203a 7265 7475 726e 3a20 4120       :return: A 
-00002870: 6461 7461 2074 6162 6c65 2077 6974 6820  data table with 
-00002880: 696e 666f 726d 6174 696f 6e20 6162 6f75  information abou
-00002890: 7420 7468 6520 6469 6666 6572 656e 7469  t the differenti
-000028a0: 616c 2065 7870 7265 7373 696f 6e0a 2020  al expression.  
-000028b0: 2020 2020 2020 2020 616e 616c 7973 6973          analysis
-000028c0: 206f 6620 7468 6520 7175 6572 6965 6420   of the queried 
-000028d0: 6461 7461 7365 742e 204e 6f74 6520 7468  dataset. Note th
-000028e0: 6174 2074 6869 7320 6675 6e63 6974 6f6e  at this funciton
-000028f0: 2064 6f65 7320 6e6f 7420 7265 7475 726e   does not return
-00002900: 200a 2020 2020 2020 2020 2020 6469 6666   .          diff
-00002910: 6572 656e 7469 616c 2065 7870 7265 7373  erential express
-00002920: 696f 6e20 7661 6c75 6573 2074 6865 6d73  ion values thems
-00002930: 656c 7665 732e 2055 7365 0a20 2020 2020  elves. Use.     
-00002940: 2020 2020 2067 6574 5f64 6966 6665 7265       get_differe
-00002950: 6e74 6961 6c5f 6578 7072 6573 7369 6f6e  ntial_expression
-00002960: 5f76 616c 7565 7320 746f 2067 6574 2064  _values to get d
-00002970: 6966 6665 7265 6e74 6961 6c20 6578 7072  ifferential expr
-00002980: 6573 7369 6f6e 0a20 2020 2020 2020 2020  ession.         
-00002990: 2076 616c 7565 7320 2873 6565 2065 7861   values (see exa
-000029a0: 6d70 6c65 7329 2e0a 2020 2020 2020 2020  mples)..        
-000029b0: 2020 0a20 2020 2020 2020 2020 2054 6865    .          The
-000029c0: 2066 6965 6c64 7320 6f66 2074 6865 2044   fields of the D
-000029d0: 6174 6146 7261 6d65 2061 7265 3a0a 2020  ataFrame are:.  
-000029e0: 2020 2020 2020 2020 2020 2d20 7265 7375            - resu
-000029f0: 6c74 5f49 443a 2052 6573 756c 7420 7365  lt_ID: Result se
-00002a00: 7420 4944 206f 6620 7468 6520 6469 6666  t ID of the diff
-00002a10: 6572 656e 7469 616c 2065 7870 7265 7373  erential express
-00002a20: 696f 6e20 616e 616c 7973 6973 2e20 4d61  ion analysis. Ma
-00002a30: 7920 7265 7072 6573 656e 7420 6d75 6c74  y represent mult
-00002a40: 6970 6c65 2066 6163 746f 7273 2069 6e20  iple factors in 
-00002a50: 6120 7369 6e67 6c65 206d 6f64 656c 2e0a  a single model..
-00002a60: 2020 2020 2020 2020 2020 2020 2d20 636f              - co
-00002a70: 6e74 7261 7374 5f49 443a 2049 6420 6f66  ntrast_ID: Id of
-00002a80: 2074 6865 2073 7065 6369 6669 6320 636f   the specific co
-00002a90: 6e74 7261 7374 2066 6163 746f 722e 2054  ntrast factor. T
-00002aa0: 6f67 6574 6865 7220 7769 7468 2074 6865  ogether with the
-00002ab0: 2072 6573 756c 742e 4944 2074 6865 7920   result.ID they 
-00002ac0: 756e 6971 7565 6c79 2072 6570 7265 7365  uniquely represe
-00002ad0: 6e74 2061 2067 6976 656e 2063 6f6e 7472  nt a given contr
-00002ae0: 6173 742e 0a20 2020 2020 2020 2020 2020  ast..           
-00002af0: 202d 2065 7870 6572 696d 656e 745f 4944   - experiment_ID
-00002b00: 3a20 4964 206f 6620 7468 6520 736f 7572  : Id of the sour
-00002b10: 6365 2065 7870 6572 696d 656e 740a 2020  ce experiment.  
-00002b20: 2020 2020 2020 2020 2020 2d20 6661 6374            - fact
-00002b30: 6f72 5f63 6174 6567 6f72 793a 2043 6174  or_category: Cat
-00002b40: 6567 6f72 7920 666f 7220 7468 6520 636f  egory for the co
-00002b50: 6e74 7261 7374 0a20 2020 2020 2020 2020  ntrast.         
+000026f0: 2020 2020 2020 2020 2020 2064 6174 6173             datas
+00002700: 6574 3a73 7472 7c69 6e74 2c0a 2020 2020  et:str|int,.    
+00002710: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00002720: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00002730: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00002740: 202a 2a6b 7761 7267 7329 2d3e 4461 7461   **kwargs)->Data
+00002750: 4672 616d 653a 0a20 2020 2020 2020 2022  Frame:.        "
+00002760: 2222 5265 7472 6965 7665 2061 6e6e 6f74  ""Retrieve annot
+00002770: 6174 696f 6e73 2061 6e64 2073 7572 6661  ations and surfa
+00002780: 6365 206c 6576 656c 2073 7461 7473 2066  ce level stats f
+00002790: 6f72 2061 2064 6174 6173 6574 2773 2064  or a dataset's d
+000027a0: 6966 6665 7265 6e74 6961 6c20 616e 616c  ifferential anal
+000027b0: 7973 6573 0a20 2020 2020 2020 200a 2020  yses.        .  
+000027c0: 2020 2020 2020 3a70 6172 616d 2064 6174        :param dat
+000027d0: 6173 6574 3a20 4120 6e75 6d65 7269 6361  aset: A numerica
+000027e0: 6c20 6461 7461 7365 7420 6964 656e 7469  l dataset identi
+000027f0: 6669 6572 206f 7220 6120 6461 7461 7365  fier or a datase
+00002800: 7420 7368 6f72 7420 6e61 6d65 0a20 2020  t short name.   
+00002810: 2020 2020 203a 7479 7065 2064 6174 6173       :type datas
+00002820: 6574 3a20 7374 727c 696e 740a 2020 2020  et: str|int.    
+00002830: 2020 2020 3a70 6172 616d 202a 2a6b 7761      :param **kwa
+00002840: 7267 733a 2041 6464 6974 696f 6e61 6c20  rgs: Additional 
+00002850: 6172 6775 6d65 6e74 7320 746f 2070 6173  arguments to pas
+00002860: 7320 746f 2072 6177 2e67 6574 5f64 6174  s to raw.get_dat
+00002870: 6173 6574 5f64 6966 6665 7265 6e74 6961  aset_differentia
+00002880: 6c5f 6578 7072 6573 7369 6f6e 5f61 6e61  l_expression_ana
+00002890: 6c79 7365 730a 2020 2020 2020 2020 3a72  lyses.        :r
+000028a0: 6574 7572 6e3a 2041 2064 6174 6120 7461  eturn: A data ta
+000028b0: 626c 6520 7769 7468 2069 6e66 6f72 6d61  ble with informa
+000028c0: 7469 6f6e 2061 626f 7574 2074 6865 2064  tion about the d
+000028d0: 6966 6665 7265 6e74 6961 6c20 6578 7072  ifferential expr
+000028e0: 6573 7369 6f6e 0a20 2020 2020 2020 2020  ession.         
+000028f0: 2061 6e61 6c79 7369 7320 6f66 2074 6865   analysis of the
+00002900: 2071 7565 7269 6564 2064 6174 6173 6574   queried dataset
+00002910: 2e20 4e6f 7465 2074 6861 7420 7468 6973  . Note that this
+00002920: 2066 756e 6369 746f 6e20 646f 6573 206e   funciton does n
+00002930: 6f74 2072 6574 7572 6e20 0a20 2020 2020  ot return .     
+00002940: 2020 2020 2064 6966 6665 7265 6e74 6961       differentia
+00002950: 6c20 6578 7072 6573 7369 6f6e 2076 616c  l expression val
+00002960: 7565 7320 7468 656d 7365 6c76 6573 2e20  ues themselves. 
+00002970: 5573 650a 2020 2020 2020 2020 2020 6765  Use.          ge
+00002980: 745f 6469 6666 6572 656e 7469 616c 5f65  t_differential_e
+00002990: 7870 7265 7373 696f 6e5f 7661 6c75 6573  xpression_values
+000029a0: 2074 6f20 6765 7420 6469 6666 6572 656e   to get differen
+000029b0: 7469 616c 2065 7870 7265 7373 696f 6e0a  tial expression.
+000029c0: 2020 2020 2020 2020 2020 7661 6c75 6573            values
+000029d0: 2028 7365 6520 6578 616d 706c 6573 292e   (see examples).
+000029e0: 0a20 2020 2020 2020 2020 200a 2020 2020  .          .    
+000029f0: 2020 2020 2020 5468 6520 6669 656c 6473        The fields
+00002a00: 206f 6620 7468 6520 4461 7461 4672 616d   of the DataFram
+00002a10: 6520 6172 653a 0a20 2020 2020 2020 2020  e are:.         
+00002a20: 2020 202d 2072 6573 756c 745f 4944 3a20     - result_ID: 
+00002a30: 5265 7375 6c74 2073 6574 2049 4420 6f66  Result set ID of
+00002a40: 2074 6865 2064 6966 6665 7265 6e74 6961   the differentia
+00002a50: 6c20 6578 7072 6573 7369 6f6e 2061 6e61  l expression ana
+00002a60: 6c79 7369 732e 204d 6179 2072 6570 7265  lysis. May repre
+00002a70: 7365 6e74 206d 756c 7469 706c 6520 6661  sent multiple fa
+00002a80: 6374 6f72 7320 696e 2061 2073 696e 676c  ctors in a singl
+00002a90: 6520 6d6f 6465 6c2e 0a20 2020 2020 2020  e model..       
+00002aa0: 2020 2020 202d 2063 6f6e 7472 6173 745f       - contrast_
+00002ab0: 4944 3a20 4964 206f 6620 7468 6520 7370  ID: Id of the sp
+00002ac0: 6563 6966 6963 2063 6f6e 7472 6173 7420  ecific contrast 
+00002ad0: 6661 6374 6f72 2e20 546f 6765 7468 6572  factor. Together
+00002ae0: 2077 6974 6820 7468 6520 7265 7375 6c74   with the result
+00002af0: 2e49 4420 7468 6579 2075 6e69 7175 656c  .ID they uniquel
+00002b00: 7920 7265 7072 6573 656e 7420 6120 6769  y represent a gi
+00002b10: 7665 6e20 636f 6e74 7261 7374 2e0a 2020  ven contrast..  
+00002b20: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
+00002b30: 7269 6d65 6e74 5f49 443a 2049 6420 6f66  riment_ID: Id of
+00002b40: 2074 6865 2073 6f75 7263 6520 6578 7065   the source expe
+00002b50: 7269 6d65 6e74 0a20 2020 2020 2020 2020  riment.         
 00002b60: 2020 202d 2066 6163 746f 725f 6361 7465     - factor_cate
-00002b70: 676f 7279 5f55 5249 3a20 5552 4920 666f  gory_URI: URI fo
-00002b80: 7220 7468 6520 636f 6e74 7261 7374 2063  r the contrast c
-00002b90: 6174 6567 6f72 790a 2020 2020 2020 2020  ategory.        
-00002ba0: 2020 2020 2d20 6661 6374 6f72 5f49 443a      - factor_ID:
-00002bb0: 2049 4420 6f66 2074 6865 2066 6163 746f   ID of the facto
-00002bc0: 720a 2020 2020 2020 2020 2020 2020 2d20  r.            - 
-00002bd0: 6261 7365 6c69 6e65 5f66 6163 746f 7273  baseline_factors
-00002be0: 3a20 4368 6172 6163 7465 7269 7374 6963  : Characteristic
-00002bf0: 7320 6f66 2074 6865 2062 6173 656c 696e  s of the baselin
-00002c00: 652e 2054 6869 7320 6669 656c 6420 6973  e. This field is
-00002c10: 2061 2044 6174 6146 7261 6d65 0a20 2020   a DataFrame.   
-00002c20: 2020 2020 2020 2020 202d 2065 7870 6572           - exper
-00002c30: 696d 656e 7461 6c5f 6661 6374 6f72 733a  imental_factors:
-00002c40: 2043 6861 7261 6374 6572 6973 7469 6373   Characteristics
-00002c50: 206f 6620 7468 6520 6578 7065 7269 6d65   of the experime
-00002c60: 6e74 616c 2067 726f 7570 2e20 5468 6973  ntal group. This
-00002c70: 2066 6965 6c64 2069 7320 6120 4461 7461   field is a Data
-00002c80: 4672 616d 650a 2020 2020 2020 2020 2020  Frame.          
-00002c90: 2020 2d20 6973 5375 6273 6574 3a20 5472    - isSubset: Tr
-00002ca0: 7565 2069 6620 7468 6520 7265 7375 6c74  ue if the result
-00002cb0: 2073 6574 2062 656c 6f6e 6720 746f 2061   set belong to a
-00002cc0: 2073 7562 7365 742c 2046 616c 7365 2069   subset, False i
-00002cd0: 6620 6e6f 742e 200a 2020 2020 2020 2020  f not. .        
-00002ce0: 2020 2020 2020 5375 6273 6574 7320 6172        Subsets ar
-00002cf0: 6520 6372 6561 7465 6420 7768 656e 2070  e created when p
-00002d00: 6572 666f 726d 696e 6720 6469 6666 6572  erforming differ
-00002d10: 656e 7469 616c 2065 7870 7265 7373 696f  ential expressio
-00002d20: 6e20 746f 2061 766f 6964 200a 2020 2020  n to avoid .    
-00002d30: 2020 2020 2020 2020 2020 756e 6865 6c70            unhelp
-00002d40: 6675 6c20 636f 6d70 6172 6973 6f6e 732e  ful comparisons.
-00002d50: 0a20 2020 2020 2020 2020 2020 202d 2073  .            - s
-00002d60: 7562 7365 745f 6661 6374 6f72 3a20 4368  ubset_factor: Ch
-00002d70: 6172 6163 7465 7269 7374 6963 7320 6f66  aracteristics of
-00002d80: 2074 6865 2073 7562 7365 742e 2054 6869   the subset. Thi
-00002d90: 7320 6669 656c 6420 6973 2061 2044 6174  s field is a Dat
-00002da0: 6146 7261 6d65 0a20 2020 2020 2020 2020  aFrame.         
-00002db0: 2020 202d 2070 726f 6265 735f 616e 616c     - probes_anal
-00002dc0: 797a 6564 3a20 4e75 6d62 6572 206f 6620  yzed: Number of 
-00002dd0: 7072 6f62 6573 6574 7320 7265 7072 6573  probesets repres
-00002de0: 656e 7465 6420 696e 2074 6865 2063 6f6e  ented in the con
-00002df0: 7472 6173 740a 2020 2020 2020 2020 2020  trast.          
-00002e00: 2020 2d20 6765 6e65 735f 616e 616c 797a    - genes_analyz
-00002e10: 6564 3a20 4e75 6d62 6572 206f 6620 6765  ed: Number of ge
-00002e20: 6e65 7320 7265 7072 6573 656e 7465 6420  nes represented 
-00002e30: 696e 2074 6865 2063 6f6e 7472 6173 740a  in the contrast.
-00002e40: 2020 2020 2020 2020 3a72 7479 7065 3a20          :rtype: 
-00002e50: 4461 7461 4672 616d 650a 0a20 2020 2020  DataFrame..     
-00002e60: 2020 2022 2222 0a20 2020 2020 2020 2072     """.        r
-00002e70: 6573 706f 6e73 6520 3d20 7365 6c66 2e72  esponse = self.r
-00002e80: 6177 2e67 6574 5f64 6174 6173 6574 5f64  aw.get_dataset_d
-00002e90: 6966 6665 7265 6e74 6961 6c5f 6578 7072  ifferential_expr
-00002ea0: 6573 7369 6f6e 5f61 6e61 6c79 7365 7328  ession_analyses(
-00002eb0: 6461 7461 7365 742c 0a20 2020 2020 2020  dataset,.       
-00002ec0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00002ed0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00002ee0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00002ef0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00002f00: 2020 2a2a 6b77 6172 6773 290a 2020 2020    **kwargs).    
-00002f10: 2020 2020 6466 203d 2070 732e 7072 6f63      df = ps.proc
-00002f20: 6573 735f 6465 6128 7265 7370 6f6e 7365  ess_dea(response
-00002f30: 2e64 6174 6129 0a20 2020 2020 2020 200a  .data).        .
-00002f40: 2020 2020 2020 2020 7265 7475 726e 2064          return d
-00002f50: 660a 2020 2020 0a20 2020 2023 202f 6461  f.    .    # /da
-00002f60: 7461 7365 7473 2f7b 6461 7461 7365 747d  tasets/{dataset}
-00002f70: 2f61 6e61 6c79 7365 732f 6469 6666 6572  /analyses/differ
-00002f80: 656e 7469 616c 2f72 6573 756c 7453 6574  ential/resultSet
-00002f90: 7320 2d2d 2d2d 2d0a 2020 2020 2320 756e  s -----.    # un
-00002fa0: 696d 706c 656d 656e 7465 640a 2020 2020  implemented.    
-00002fb0: 2320 756e 7375 7265 2061 626f 7574 2074  # unsure about t
-00002fc0: 6865 2064 6973 7469 6e63 7469 6f6e 2062  he distinction b
-00002fd0: 6574 7765 656e 2074 6869 7320 616e 6420  etween this and 
-00002fe0: 7468 6520 6765 745f 6461 7461 7365 745f  the get_dataset_
-00002ff0: 6469 6666 6572 656e 7469 616c 5f65 7870  differential_exp
-00003000: 7265 7373 696f 6e5f 616e 616c 7973 6573  ression_analyses
-00003010: 2e20 0a20 2020 2023 2073 6565 6d20 746f  . .    # seem to
-00003020: 2063 6f6e 7461 696e 2074 6865 2072 6564   contain the red
-00003030: 756e 7461 6e74 2069 6e66 6f72 6d61 7469  untant informati
-00003040: 6f6e 0a20 2020 200a 2020 2020 0a20 2020  on.    .    .   
-00003050: 2023 202f 6461 7461 7365 7473 2f7b 6461   # /datasets/{da
-00003060: 7461 7365 747d 2f64 6174 6120 2d2d 2d2d  taset}/data ----
-00003070: 2d0a 2020 2020 2320 6465 7072 6563 6174  -.    # deprecat
-00003080: 6564 2c20 7265 6d6f 7665 206c 6174 6572  ed, remove later
-00003090: 0a20 2020 2064 6566 2067 6574 5f64 6174  .    def get_dat
-000030a0: 6173 6574 5f65 7870 7265 7373 696f 6e28  aset_expression(
-000030b0: 7365 6c66 2c20 6461 7461 7365 743a 7374  self, dataset:st
-000030c0: 727c 696e 742c 202a 2a6b 7761 7267 7329  r|int, **kwargs)
-000030d0: 2d3e 4461 7461 4672 616d 653a 0a20 2020  ->DataFrame:.   
-000030e0: 2020 2020 2022 2222 0a20 2020 2020 2020       """.       
-000030f0: 2044 6570 7265 6361 7465 6420 696e 2066   Deprecated in f
-00003100: 6176 6f75 7220 6f66 2067 6574 5f64 6174  avour of get_dat
-00003110: 6173 6574 5f65 7870 7265 7373 696f 6e0a  aset_expression.
-00003120: 2020 2020 2020 2020 2222 220a 2020 2020          """.    
-00003130: 2020 2020 7761 726e 696e 6773 2e77 6172      warnings.war
-00003140: 6e28 2767 6574 5f64 6174 6173 6574 5f65  n('get_dataset_e
-00003150: 7870 7265 7373 696f 6e20 6973 2064 6570  xpression is dep
-00003160: 7265 6361 7465 642c 2070 6c65 6173 6520  recated, please 
-00003170: 7573 6520 6765 745f 6461 7461 7365 745f  use get_dataset_
-00003180: 7072 6f63 6573 7365 645f 6578 7072 6573  processed_expres
-00003190: 7369 6f6e 2069 6e73 7465 6164 2729 0a20  sion instead'). 
-000031a0: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-000031b0: 7265 7475 726e 2073 656c 662e 6765 745f  return self.get_
-000031c0: 6461 7461 7365 745f 7072 6f63 6573 7365  dataset_processe
-000031d0: 645f 6578 7072 6573 7369 6f6e 2864 6174  d_expression(dat
-000031e0: 6173 6574 2c2a 2a6b 7761 7267 7329 0a0a  aset,**kwargs)..
-000031f0: 2020 2020 0a20 2020 200a 2020 2020 0a20      .    .    . 
-00003200: 2020 2023 202f 6461 7461 7365 7473 2f7b     # /datasets/{
-00003210: 6461 7461 7365 7473 7d2f 6578 7072 6573  datasets}/expres
-00003220: 7369 6f6e 732f 6765 6e65 732f 7b67 656e  sions/genes/{gen
-00003230: 6573 7d2c 2067 6574 5f64 6174 6173 6574  es}, get_dataset
-00003240: 5f65 7870 7265 7373 696f 6e5f 666f 725f  _expression_for_
-00003250: 6765 6e65 7320 2d2d 2d2d 2d2d 0a20 2020  genes ------.   
-00003260: 2064 6566 2067 6574 5f64 6174 6173 6574   def get_dataset
-00003270: 5f65 7870 7265 7373 696f 6e5f 666f 725f  _expression_for_
-00003280: 6765 6e65 7328 7365 6c66 2c0a 2020 2020  genes(self,.    
-00003290: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000032a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000032b0: 2020 2020 2064 6174 6173 6574 733a 4c69       datasets:Li
-000032c0: 7374 5b73 7472 7c69 6e74 5d2c 0a20 2020  st[str|int],.   
+00002b70: 676f 7279 3a20 4361 7465 676f 7279 2066  gory: Category f
+00002b80: 6f72 2074 6865 2063 6f6e 7472 6173 740a  or the contrast.
+00002b90: 2020 2020 2020 2020 2020 2020 2d20 6661              - fa
+00002ba0: 6374 6f72 5f63 6174 6567 6f72 795f 5552  ctor_category_UR
+00002bb0: 493a 2055 5249 2066 6f72 2074 6865 2063  I: URI for the c
+00002bc0: 6f6e 7472 6173 7420 6361 7465 676f 7279  ontrast category
+00002bd0: 0a20 2020 2020 2020 2020 2020 202d 2066  .            - f
+00002be0: 6163 746f 725f 4944 3a20 4944 206f 6620  actor_ID: ID of 
+00002bf0: 7468 6520 6661 6374 6f72 0a20 2020 2020  the factor.     
+00002c00: 2020 2020 2020 202d 2062 6173 656c 696e         - baselin
+00002c10: 655f 6661 6374 6f72 733a 2043 6861 7261  e_factors: Chara
+00002c20: 6374 6572 6973 7469 6373 206f 6620 7468  cteristics of th
+00002c30: 6520 6261 7365 6c69 6e65 2e20 5468 6973  e baseline. This
+00002c40: 2066 6965 6c64 2069 7320 6120 4461 7461   field is a Data
+00002c50: 4672 616d 650a 2020 2020 2020 2020 2020  Frame.          
+00002c60: 2020 2d20 6578 7065 7269 6d65 6e74 616c    - experimental
+00002c70: 5f66 6163 746f 7273 3a20 4368 6172 6163  _factors: Charac
+00002c80: 7465 7269 7374 6963 7320 6f66 2074 6865  teristics of the
+00002c90: 2065 7870 6572 696d 656e 7461 6c20 6772   experimental gr
+00002ca0: 6f75 702e 2054 6869 7320 6669 656c 6420  oup. This field 
+00002cb0: 6973 2061 2044 6174 6146 7261 6d65 0a20  is a DataFrame. 
+00002cc0: 2020 2020 2020 2020 2020 202d 2069 7353             - isS
+00002cd0: 7562 7365 743a 2054 7275 6520 6966 2074  ubset: True if t
+00002ce0: 6865 2072 6573 756c 7420 7365 7420 6265  he result set be
+00002cf0: 6c6f 6e67 2074 6f20 6120 7375 6273 6574  long to a subset
+00002d00: 2c20 4661 6c73 6520 6966 206e 6f74 2e20  , False if not. 
+00002d10: 0a20 2020 2020 2020 2020 2020 2020 2053  .              S
+00002d20: 7562 7365 7473 2061 7265 2063 7265 6174  ubsets are creat
+00002d30: 6564 2077 6865 6e20 7065 7266 6f72 6d69  ed when performi
+00002d40: 6e67 2064 6966 6665 7265 6e74 6961 6c20  ng differential 
+00002d50: 6578 7072 6573 7369 6f6e 2074 6f20 6176  expression to av
+00002d60: 6f69 6420 0a20 2020 2020 2020 2020 2020  oid .           
+00002d70: 2020 2075 6e68 656c 7066 756c 2063 6f6d     unhelpful com
+00002d80: 7061 7269 736f 6e73 2e0a 2020 2020 2020  parisons..      
+00002d90: 2020 2020 2020 2d20 7375 6273 6574 5f66        - subset_f
+00002da0: 6163 746f 723a 2043 6861 7261 6374 6572  actor: Character
+00002db0: 6973 7469 6373 206f 6620 7468 6520 7375  istics of the su
+00002dc0: 6273 6574 2e20 5468 6973 2066 6965 6c64  bset. This field
+00002dd0: 2069 7320 6120 4461 7461 4672 616d 650a   is a DataFrame.
+00002de0: 2020 2020 2020 2020 2020 2020 2d20 7072              - pr
+00002df0: 6f62 6573 5f61 6e61 6c79 7a65 643a 204e  obes_analyzed: N
+00002e00: 756d 6265 7220 6f66 2070 726f 6265 7365  umber of probese
+00002e10: 7473 2072 6570 7265 7365 6e74 6564 2069  ts represented i
+00002e20: 6e20 7468 6520 636f 6e74 7261 7374 0a20  n the contrast. 
+00002e30: 2020 2020 2020 2020 2020 202d 2067 656e             - gen
+00002e40: 6573 5f61 6e61 6c79 7a65 643a 204e 756d  es_analyzed: Num
+00002e50: 6265 7220 6f66 2067 656e 6573 2072 6570  ber of genes rep
+00002e60: 7265 7365 6e74 6564 2069 6e20 7468 6520  resented in the 
+00002e70: 636f 6e74 7261 7374 0a20 2020 2020 2020  contrast.       
+00002e80: 203a 7274 7970 653a 2044 6174 6146 7261   :rtype: DataFra
+00002e90: 6d65 0a0a 2020 2020 2020 2020 2222 220a  me..        """.
+00002ea0: 2020 2020 2020 2020 7265 7370 6f6e 7365          response
+00002eb0: 203d 2073 656c 662e 7261 772e 6765 745f   = self.raw.get_
+00002ec0: 6461 7461 7365 745f 6469 6666 6572 656e  dataset_differen
+00002ed0: 7469 616c 5f65 7870 7265 7373 696f 6e5f  tial_expression_
+00002ee0: 616e 616c 7973 6573 2864 6174 6173 6574  analyses(dataset
+00002ef0: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+00002f00: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00002f10: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00002f20: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00002f30: 2020 2020 2020 2020 2020 202a 2a6b 7761             **kwa
+00002f40: 7267 7329 0a20 2020 2020 2020 2064 6620  rgs).        df 
+00002f50: 3d20 7073 2e70 726f 6365 7373 5f64 6561  = ps.process_dea
+00002f60: 2872 6573 706f 6e73 652e 6461 7461 290a  (response.data).
+00002f70: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00002f80: 2072 6574 7572 6e20 6466 0a20 2020 200a   return df.    .
+00002f90: 2020 2020 2320 2f64 6174 6173 6574 732f      # /datasets/
+00002fa0: 7b64 6174 6173 6574 7d2f 616e 616c 7973  {dataset}/analys
+00002fb0: 6573 2f64 6966 6665 7265 6e74 6961 6c2f  es/differential/
+00002fc0: 7265 7375 6c74 5365 7473 202d 2d2d 2d2d  resultSets -----
+00002fd0: 0a20 2020 2023 2075 6e69 6d70 6c65 6d65  .    # unimpleme
+00002fe0: 6e74 6564 0a20 2020 2023 2075 6e73 7572  nted.    # unsur
+00002ff0: 6520 6162 6f75 7420 7468 6520 6469 7374  e about the dist
+00003000: 696e 6374 696f 6e20 6265 7477 6565 6e20  inction between 
+00003010: 7468 6973 2061 6e64 2074 6865 2067 6574  this and the get
+00003020: 5f64 6174 6173 6574 5f64 6966 6665 7265  _dataset_differe
+00003030: 6e74 6961 6c5f 6578 7072 6573 7369 6f6e  ntial_expression
+00003040: 5f61 6e61 6c79 7365 732e 200a 2020 2020  _analyses. .    
+00003050: 2320 7365 656d 2074 6f20 636f 6e74 6169  # seem to contai
+00003060: 6e20 7468 6520 7265 6475 6e74 616e 7420  n the reduntant 
+00003070: 696e 666f 726d 6174 696f 6e0a 2020 2020  information.    
+00003080: 0a20 2020 200a 2020 2020 2320 2f64 6174  .    .    # /dat
+00003090: 6173 6574 732f 7b64 6174 6173 6574 7d2f  asets/{dataset}/
+000030a0: 6461 7461 202d 2d2d 2d2d 0a20 2020 2023  data -----.    #
+000030b0: 2064 6570 7265 6361 7465 642c 2072 656d   deprecated, rem
+000030c0: 6f76 6520 6c61 7465 720a 2020 2020 6465  ove later.    de
+000030d0: 6620 6765 745f 6461 7461 7365 745f 6578  f get_dataset_ex
+000030e0: 7072 6573 7369 6f6e 2873 656c 662c 2064  pression(self, d
+000030f0: 6174 6173 6574 3a73 7472 7c69 6e74 2c20  ataset:str|int, 
+00003100: 2a2a 6b77 6172 6773 292d 3e44 6174 6146  **kwargs)->DataF
+00003110: 7261 6d65 3a0a 2020 2020 2020 2020 2222  rame:.        ""
+00003120: 220a 2020 2020 2020 2020 4465 7072 6563  ".        Deprec
+00003130: 6174 6564 2069 6e20 6661 766f 7572 206f  ated in favour o
+00003140: 6620 6765 745f 6461 7461 7365 745f 6578  f get_dataset_ex
+00003150: 7072 6573 7369 6f6e 0a20 2020 2020 2020  pression.       
+00003160: 2022 2222 0a20 2020 2020 2020 2077 6172   """.        war
+00003170: 6e69 6e67 732e 7761 726e 2827 6765 745f  nings.warn('get_
+00003180: 6461 7461 7365 745f 6578 7072 6573 7369  dataset_expressi
+00003190: 6f6e 2069 7320 6465 7072 6563 6174 6564  on is deprecated
+000031a0: 2c20 706c 6561 7365 2075 7365 2067 6574  , please use get
+000031b0: 5f64 6174 6173 6574 5f70 726f 6365 7373  _dataset_process
+000031c0: 6564 5f65 7870 7265 7373 696f 6e20 696e  ed_expression in
+000031d0: 7374 6561 6427 290a 2020 2020 2020 2020  stead').        
+000031e0: 0a20 2020 2020 2020 2072 6574 7572 6e20  .        return 
+000031f0: 7365 6c66 2e67 6574 5f64 6174 6173 6574  self.get_dataset
+00003200: 5f70 726f 6365 7373 6564 5f65 7870 7265  _processed_expre
+00003210: 7373 696f 6e28 6461 7461 7365 742c 2a2a  ssion(dataset,**
+00003220: 6b77 6172 6773 290a 0a20 2020 200a 2020  kwargs)..    .  
+00003230: 2020 0a20 2020 200a 2020 2020 2320 2f64    .    .    # /d
+00003240: 6174 6173 6574 732f 7b64 6174 6173 6574  atasets/{dataset
+00003250: 737d 2f65 7870 7265 7373 696f 6e73 2f67  s}/expressions/g
+00003260: 656e 6573 2f7b 6765 6e65 737d 2c20 6765  enes/{genes}, ge
+00003270: 745f 6461 7461 7365 745f 6578 7072 6573  t_dataset_expres
+00003280: 7369 6f6e 5f66 6f72 5f67 656e 6573 202d  sion_for_genes -
+00003290: 2d2d 2d2d 2d0a 2020 2020 6465 6620 6765  -----.    def ge
+000032a0: 745f 6461 7461 7365 745f 6578 7072 6573  t_dataset_expres
+000032b0: 7369 6f6e 5f66 6f72 5f67 656e 6573 2873  sion_for_genes(s
+000032c0: 656c 662c 0a20 2020 2020 2020 2020 2020  elf,.           
 000032d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000032e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000032f0: 2020 2020 2020 6765 6e65 733a 4c69 7374        genes:List
-00003300: 5b69 6e74 5d2c 0a20 2020 2020 2020 2020  [int],.         
+000032e0: 2020 2020 2020 2020 2020 2020 2020 6461                da
+000032f0: 7461 7365 7473 3a4c 6973 745b 7374 727c  tasets:List[str|
+00003300: 696e 745d 2c0a 2020 2020 2020 2020 2020  int],.          
 00003310: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00003320: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00003330: 6b65 6570 5f6e 6f6e 5f73 7065 6369 6669  keep_non_specifi
-00003340: 633a 626f 6f6c 203d 2046 616c 7365 2c0a  c:bool = False,.
+00003320: 2020 2020 2020 2020 2020 2020 2020 2067                 g
+00003330: 656e 6573 3a4c 6973 745b 696e 745d 2c0a  enes:List[int],.
+00003340: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00003350: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00003360: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00003370: 2020 2020 2020 2020 2063 6f6e 736f 6c69           consoli
-00003380: 6461 7465 203d 204e 6f6e 652c 0a20 2020  date = None,.   
+00003360: 2020 2020 2020 2020 206b 6565 705f 6e6f           keep_no
+00003370: 6e5f 7370 6563 6966 6963 3a62 6f6f 6c20  n_specific:bool 
+00003380: 3d20 4661 6c73 652c 0a20 2020 2020 2020  = False,.       
 00003390: 2020 2020 2020 2020 2020 2020 2020 2020                  
 000033a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000033b0: 2020 2020 2020 2a2a 6b77 6172 6773 292d        **kwargs)-
-000033c0: 3e64 6963 745b 696e 743a 4461 7461 4672  >dict[int:DataFr
-000033d0: 616d 655d 3a0a 2020 2020 2020 2020 2222  ame]:.        ""
-000033e0: 2252 6574 7269 6576 6520 7468 6520 6578  "Retrieve the ex
-000033f0: 7072 6573 7369 6f6e 2064 6174 6120 6d61  pression data ma
-00003400: 7472 6978 206f 6620 6120 7365 7420 6f66  trix of a set of
-00003410: 2064 6174 6173 6574 7320 616e 6420 6765   datasets and ge
-00003420: 6e65 730a 0a0a 2020 2020 2020 2020 3a70  nes...        :p
-00003430: 6172 616d 2064 6174 6173 6574 733a 2041  aram datasets: A
-00003440: 206e 756d 6572 6963 616c 2064 6174 6173   numerical datas
-00003450: 6574 2069 6465 6e74 6966 6965 7220 6f72  et identifier or
-00003460: 2061 2064 6174 6173 6574 2073 686f 7274   a dataset short
-00003470: 206e 616d 650a 2020 2020 2020 2020 3a74   name.        :t
-00003480: 7970 6520 6461 7461 7365 7473 3a20 4c69  ype datasets: Li
-00003490: 7374 5b73 7472 7c69 6e74 5d0a 2020 2020  st[str|int].    
-000034a0: 2020 2020 3a70 6172 616d 2067 656e 6573      :param genes
-000034b0: 3a20 416e 2065 6e73 656d 626c 2067 656e  : An ensembl gen
-000034c0: 6520 6964 656e 7469 6669 6572 2077 6869  e identifier whi
-000034d0: 6368 2074 7970 6963 616c 6c79 2073 7461  ch typically sta
-000034e0: 7274 7320 7769 7468 200a 2020 2020 2020  rts with .      
-000034f0: 2020 2020 656e 7367 206f 7220 616e 206e      ensg or an n
-00003500: 6362 6920 6765 6e65 2069 6465 6e74 6966  cbi gene identif
-00003510: 6965 7220 6f72 2061 6e20 6f66 6669 6369  ier or an offici
-00003520: 616c 2067 656e 6520 7379 6d62 6f6c 2061  al gene symbol a
-00003530: 7070 726f 7665 6420 6279 200a 2020 2020  pproved by .    
-00003540: 2020 2020 2020 6867 6e63 0a20 2020 2020        hgnc.     
-00003550: 2020 203a 7479 7065 2067 656e 6573 3a20     :type genes: 
-00003560: 4c69 7374 5b69 6e74 5d0a 2020 2020 2020  List[int].      
-00003570: 2020 3a70 6172 616d 206b 6565 705f 6e6f    :param keep_no
-00003580: 6e5f 7370 6563 6966 6963 3a20 4966 2054  n_specific: If T
-00003590: 7275 652c 2072 6573 756c 7473 2066 726f  rue, results fro
-000035a0: 6d20 7072 6f62 6573 6574 7320 7468 6174  m probesets that
-000035b0: 2061 7265 206e 6f74 0a20 2020 2020 2020   are not.       
-000035c0: 2020 2073 7065 6369 6669 6320 746f 2074     specific to t
-000035d0: 6865 2067 656e 6520 7769 6c6c 2061 6c73  he gene will als
-000035e0: 6f20 6265 2072 6574 7572 6e65 642e 2c20  o be returned., 
-000035f0: 6465 6661 756c 7473 2074 6f20 4661 6c73  defaults to Fals
-00003600: 650a 2020 2020 2020 2020 3a74 7970 6520  e.        :type 
-00003610: 6b65 6570 5f6e 6f6e 5f73 7065 6369 6669  keep_non_specifi
-00003620: 633a 2062 6f6f 6c2c 206f 7074 696f 6e61  c: bool, optiona
-00003630: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
-00003640: 2063 6f6e 736f 6c69 6461 7465 3a20 416e   consolidate: An
-00003650: 206f 7074 696f 6e20 666f 7220 6765 6e65   option for gene
-00003660: 2065 7870 7265 7373 696f 6e20 6c65 7665   expression leve
-00003670: 6c20 636f 6e73 6f6c 6964 6174 696f 6e2e  l consolidation.
-00003680: 200a 2020 2020 2020 2020 2020 4966 2065   .          If e
-00003690: 6d70 7479 2c20 7769 6c6c 2072 6574 7572  mpty, will retur
-000036a0: 6e20 6576 6572 7920 7072 6f62 6520 666f  n every probe fo
-000036b0: 7220 7468 6520 6765 6e65 732e 2022 7069  r the genes. "pi
-000036c0: 636b 6d61 7822 2074 6f20 7069 636b 2074  ckmax" to pick t
-000036d0: 6865 0a20 2020 2020 2020 2020 2070 726f  he.          pro
-000036e0: 6265 2077 6974 6820 7468 6520 6869 6768  be with the high
-000036f0: 6573 7420 6578 7072 6573 7369 6f6e 2c20  est expression, 
-00003700: 2270 6963 6b76 6172 2220 746f 2070 6963  "pickvar" to pic
-00003710: 6b20 7468 6520 7072 6f76 6520 7769 7468  k the prove with
-00003720: 2074 6865 200a 2020 2020 2020 2020 2020   the .          
-00003730: 6869 6768 6573 7420 7661 7269 616e 6365  highest variance
-00003740: 2061 6e64 2022 6176 6572 6167 6522 2066   and "average" f
-00003750: 6f72 2072 6574 7572 6e69 6e67 2074 6865  or returning the
-00003760: 2061 7665 7261 6765 2065 7870 7265 7373   average express
-00003770: 696f 6e2c 200a 2020 2020 2020 2020 2020  ion, .          
-00003780: 6465 6661 756c 7473 2074 6f20 4e6f 6e65  defaults to None
-00003790: 0a20 2020 2020 2020 203a 7479 7065 2063  .        :type c
-000037a0: 6f6e 736f 6c69 6461 7465 3a20 5459 5045  onsolidate: TYPE
-000037b0: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
-000037c0: 2020 203a 7061 7261 6d20 2a2a 6b77 6172     :param **kwar
-000037d0: 6773 3a20 4164 6469 7469 6f6e 616c 2061  gs: Additional a
-000037e0: 7267 756d 656e 7473 2074 6f20 7061 7373  rguments to pass
-000037f0: 2074 6f20 7261 772e 6765 745f 6461 7461   to raw.get_data
-00003800: 7365 745f 6578 7072 6573 7369 6f6e 5f66  set_expression_f
-00003810: 6f72 5f67 656e 6573 0a20 2020 2020 2020  or_genes.       
-00003820: 203a 7265 7475 726e 3a20 4120 6469 6374   :return: A dict
-00003830: 206f 6620 4461 7461 4672 616d 6573 206b   of DataFrames k
-00003840: 6579 6564 2074 6f20 6461 7461 7365 7420  eyed to dataset 
-00003850: 6964 730a 2020 2020 2020 2020 3a72 7479  ids.        :rty
-00003860: 7065 3a20 6469 6374 5b69 6e74 3a44 6174  pe: dict[int:Dat
-00003870: 6146 7261 6d65 5d0a 0a20 2020 2020 2020  aFrame]..       
-00003880: 2022 2222 0a20 2020 2020 2020 200a 2020   """.        .  
-00003890: 2020 2020 2020 6b77 6172 6773 203d 2076        kwargs = v
-000038a0: 732e 7265 6d6f 7665 5f6e 6f6e 6573 280a  s.remove_nones(.
-000038b0: 2020 2020 2020 2020 2020 2020 6b65 6570              keep
-000038c0: 5f6e 6f6e 5f73 7065 6369 6669 6320 3d20  _non_specific = 
-000038d0: 6b65 6570 5f6e 6f6e 5f73 7065 6369 6669  keep_non_specifi
-000038e0: 632c 0a20 2020 2020 2020 2020 2020 2063  c,.            c
-000038f0: 6f6e 736f 6c69 6461 7465 203d 2063 6f6e  onsolidate = con
-00003900: 736f 6c69 6461 7465 2c0a 2020 2020 2020  solidate,.      
-00003910: 2020 2020 2020 2a2a 6b77 6172 6773 290a        **kwargs).
-00003920: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-00003930: 2072 6573 706f 6e73 6520 3d20 7365 6c66   response = self
-00003940: 2e72 6177 2e67 6574 5f64 6174 6173 6574  .raw.get_dataset
-00003950: 5f65 7870 7265 7373 696f 6e5f 666f 725f  _expression_for_
-00003960: 6765 6e65 7328 6461 7461 7365 7473 2c20  genes(datasets, 
-00003970: 6765 6e65 732c 200a 2020 2020 2020 2020  genes, .        
-00003980: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00003990: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000039a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000039b0: 2020 2020 202a 2a6b 7761 7267 7329 0a20       **kwargs). 
-000039c0: 2020 2020 2020 2064 6620 3d20 7073 2e70         df = ps.p
-000039d0: 726f 6365 7373 5f64 6174 6173 6574 5f67  rocess_dataset_g
-000039e0: 656e 655f 6578 7072 6573 7369 6f6e 2872  ene_expression(r
-000039f0: 6573 706f 6e73 652e 6461 7461 2c73 656c  esponse.data,sel
-00003a00: 6629 0a20 2020 2020 2020 200a 2020 2020  f).        .    
-00003a10: 2020 2020 7265 7475 726e 2064 660a 2020      return df.  
-00003a20: 2020 2020 2020 0a20 2020 2023 2064 6174        .    # dat
-00003a30: 6173 6574 732f 7b64 6174 6173 6574 737d  asets/{datasets}
-00003a40: 2f65 7870 7265 7373 696f 6e73 2f70 6361  /expressions/pca
-00003a50: 202d 2d2d 2d2d 0a20 2020 2023 2075 6e69   -----.    # uni
-00003a60: 6d70 6c65 6d65 6e74 6564 0a20 2020 200a  mplemented.    .
-00003a70: 2020 2020 0a20 2020 2023 2064 6174 6173      .    # datas
-00003a80: 6574 732f 7b64 6174 6173 6574 7d2f 706c  ets/{dataset}/pl
-00003a90: 6174 666f 726d 7320 2d2d 2d2d 2d2d 0a20  atforms ------. 
-00003aa0: 2020 2064 6566 2067 6574 5f64 6174 6173     def get_datas
-00003ab0: 6574 5f70 6c61 7466 6f72 6d73 2873 656c  et_platforms(sel
-00003ac0: 662c 2064 6174 6173 6574 3a73 7472 7c69  f, dataset:str|i
-00003ad0: 6e74 2c20 2a2a 6b77 6172 6773 292d 3e44  nt, **kwargs)->D
-00003ae0: 6174 6146 7261 6d65 3a0a 2020 2020 2020  ataFrame:.      
-00003af0: 2020 2222 220a 2020 2020 2020 2020 0a20    """.        . 
-00003b00: 2020 2020 2020 203a 7061 7261 6d20 6461         :param da
-00003b10: 7461 7365 743a 2041 206e 756d 6572 6963  taset: A numeric
-00003b20: 616c 2064 6174 6173 6574 2069 6465 6e74  al dataset ident
-00003b30: 6966 6965 7220 6f72 2061 2064 6174 6173  ifier or a datas
-00003b40: 6574 2073 686f 7274 206e 616d 650a 2020  et short name.  
-00003b50: 2020 2020 2020 3a74 7970 6520 6461 7461        :type data
-00003b60: 7365 743a 2073 7472 7c69 6e74 0a20 2020  set: str|int.   
-00003b70: 2020 2020 203a 7061 7261 6d20 2a2a 6b77       :param **kw
-00003b80: 6172 6773 3a20 4164 6469 7469 6f6e 616c  args: Additional
-00003b90: 2061 7267 756d 656e 7473 2074 6f20 7061   arguments to pa
-00003ba0: 7373 2074 6f20 7261 772e 6765 745f 6461  ss to raw.get_da
-00003bb0: 7461 7365 745f 706c 6174 666f 726d 730a  taset_platforms.
-00003bc0: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
-00003bd0: 2041 2044 6174 6146 7261 6d65 2077 6974   A DataFrame wit
-00003be0: 6820 696e 666f 726d 6174 696f 6e20 6162  h information ab
-00003bf0: 6f75 7420 7468 6520 706c 6174 666f 726d  out the platform
-00003c00: 732e 0a20 2020 2020 2020 2020 2054 6865  s..          The
-00003c10: 2066 6965 6c64 7320 6f66 2074 6865 2044   fields of the D
-00003c20: 6174 6146 7261 6d65 2061 7265 3a0a 2020  ataFrame are:.  
-00003c30: 2020 2020 2020 2020 2020 2020 2020 0a20                . 
-00003c40: 2020 2020 2020 2020 2020 202d 2070 6c61             - pla
-00003c50: 7466 6f72 6d5f 4944 3a20 4964 206e 756d  tform_ID: Id num
-00003c60: 6265 7220 6f66 2074 6865 2070 6c61 7466  ber of the platf
-00003c70: 6f72 6d20 6769 7665 6e20 6279 2047 656d  orm given by Gem
-00003c80: 6d61 0a20 2020 2020 2020 2020 2020 202d  ma.            -
-00003c90: 2070 6c61 7466 6f72 6d5f 7479 7065 3a20   platform_type: 
-00003ca0: 5479 7065 206f 6620 7468 6520 706c 6174  Type of the plat
-00003cb0: 666f 726d 2e0a 2020 2020 2020 2020 2020  form..          
-00003cc0: 2020 2d20 706c 6174 666f 726d 5f64 6573    - platform_des
-00003cd0: 6372 6970 7469 6f6e 3a20 4672 6565 2074  cription: Free t
-00003ce0: 6578 7420 6669 656c 6420 6465 7363 7269  ext field descri
-00003cf0: 6269 6e67 2074 6865 2070 6c61 7466 6f72  bing the platfor
-00003d00: 6d2e 0a20 2020 2020 2020 2020 2020 202d  m..            -
-00003d10: 2070 6c61 7466 6f72 6d5f 7472 6f75 626c   platform_troubl
-00003d20: 6564 3a20 5768 6574 6865 7220 7468 6520  ed: Whether the 
-00003d30: 706c 6174 666f 726d 2069 7320 6d61 726b  platform is mark
-00003d40: 6564 2061 7320 7472 6f75 626c 6564 2062  ed as troubled b
-00003d50: 7920 6120 4765 6d6d 6120 6375 7261 746f  y a Gemma curato
-00003d60: 722e 0a20 2020 2020 2020 2020 2020 202d  r..            -
-00003d70: 2074 6178 6f6e 5f6e 616d 653a 204e 616d   taxon_name: Nam
-00003d80: 6520 6f66 2074 6865 2073 7065 6369 6573  e of the species
-00003d90: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
-00003da0: 6178 6f6e 5f73 6369 656e 7469 6669 633a  axon_scientific:
-00003db0: 2053 6369 656e 7469 6669 6320 6e61 6d65   Scientific name
-00003dc0: 2066 6f72 2074 6865 2074 6178 6f6e 0a20   for the taxon. 
-00003dd0: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
-00003de0: 6f6e 5f49 443a 2049 6e74 6572 6e61 6c20  on_ID: Internal 
-00003df0: 6964 656e 7469 6669 6572 2067 6976 656e  identifier given
-00003e00: 2074 6f20 7468 6520 7370 6563 6965 7320   to the species 
-00003e10: 6279 2047 656d 6d61 0a20 2020 2020 2020  by Gemma.       
-00003e20: 2020 2020 202d 2074 6178 6f6e 5f4e 4342       - taxon_NCB
-00003e30: 493a 204e 4342 4920 4944 206f 6620 7468  I: NCBI ID of th
-00003e40: 6520 7461 786f 6e0a 2020 2020 2020 2020  e taxon.        
-00003e50: 2020 2020 2d20 7461 786f 6e5f 6461 7461      - taxon_data
-00003e60: 6261 7365 5f6e 616d 653a 2055 6e64 6572  base_name: Under
-00003e70: 6c79 696e 6720 6461 7461 6261 7365 2075  lying database u
-00003e80: 7365 6420 696e 2047 656d 6d61 2066 6f72  sed in Gemma for
-00003e90: 2074 6865 2074 6178 6f6e 0a20 2020 2020   the taxon.     
-00003ea0: 2020 2020 2020 202d 2074 6178 6f6e 5f64         - taxon_d
-00003eb0: 6174 6162 6173 655f 4944 3a20 4944 206f  atabase_ID: ID o
-00003ec0: 6620 7468 6520 756e 6465 726c 7969 6e67  f the underlying
-00003ed0: 2064 6174 6162 6173 6520 7573 6564 2069   database used i
-00003ee0: 6e20 4765 6d6d 6120 666f 7220 7468 6520  n Gemma for the 
-00003ef0: 7461 786f 6e20 200a 2020 2020 2020 2020  taxon  .        
-00003f00: 3a72 7479 7065 3a20 4461 7461 4672 616d  :rtype: DataFram
-00003f10: 650a 0a20 2020 2020 2020 2022 2222 0a20  e..        """. 
-00003f20: 2020 2020 2020 2072 6573 706f 6e73 6520         response 
-00003f30: 3d20 7365 6c66 2e72 6177 2e67 6574 5f64  = self.raw.get_d
-00003f40: 6174 6173 6574 5f70 6c61 7466 6f72 6d73  ataset_platforms
-00003f50: 2864 6174 6173 6574 2c20 2a2a 6b77 6172  (dataset, **kwar
-00003f60: 6773 290a 2020 2020 2020 2020 6466 203d  gs).        df =
-00003f70: 2070 732e 7072 6f63 6573 735f 706c 6174   ps.process_plat
-00003f80: 666f 726d 7328 7265 7370 6f6e 7365 2e64  forms(response.d
-00003f90: 6174 6129 0a20 2020 2020 2020 200a 2020  ata).        .  
-00003fa0: 2020 2020 2020 7265 7475 726e 2864 6629        return(df)
-00003fb0: 0a20 2020 200a 2020 2020 0a20 2020 2023  .    .    .    #
-00003fc0: 2064 6174 6173 6574 732f 7b64 6174 6173   datasets/{datas
-00003fd0: 6574 7d2f 6461 7461 2f70 726f 6365 7373  et}/data/process
-00003fe0: 6564 202d 2d2d 2d2d 2d0a 2020 2020 0a20  ed ------.    . 
-00003ff0: 2020 2064 6566 2067 6574 5f64 6174 6173     def get_datas
-00004000: 6574 5f70 726f 6365 7373 6564 5f65 7870  et_processed_exp
-00004010: 7265 7373 696f 6e28 7365 6c66 2c64 6174  ression(self,dat
-00004020: 6173 6574 3a73 7472 7c69 6e74 2c2a 2a6b  aset:str|int,**k
-00004030: 7761 7267 7329 2d3e 4461 7461 4672 616d  wargs)->DataFram
-00004040: 653a 0a20 2020 2020 2020 2022 2222 5265  e:.        """Re
-00004050: 7472 6965 7665 2070 726f 6365 7373 6564  trieve processed
-00004060: 2065 7870 7265 7373 696f 6e20 6461 7461   expression data
-00004070: 206f 6620 6120 6461 7461 7365 740a 0a20   of a dataset.. 
-00004080: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-00004090: 3a70 6172 616d 2064 6174 6173 6574 3a20  :param dataset: 
-000040a0: 206e 756d 6572 6963 616c 2064 6174 6173   numerical datas
-000040b0: 6574 2069 6465 6e74 6966 6965 7220 6f72  et identifier or
-000040c0: 2061 2064 6174 6173 6574 2073 686f 7274   a dataset short
-000040d0: 206e 616d 650a 2020 2020 2020 2020 3a74   name.        :t
-000040e0: 7970 6520 6461 7461 7365 743a 2073 7472  ype dataset: str
-000040f0: 7c69 6e74 0a20 2020 2020 2020 203a 7061  |int.        :pa
-00004100: 7261 6d20 2a2a 6b77 6172 6773 3a20 4164  ram **kwargs: Ad
-00004110: 6469 7469 6f6e 616c 2061 7267 756d 656e  ditional argumen
-00004120: 7473 2074 6f20 7061 7373 2074 6f20 7261  ts to pass to ra
-00004130: 772e 6765 745f 6461 7461 7365 745f 7072  w.get_dataset_pr
-00004140: 6f63 6573 7365 645f 6578 7072 6573 7369  ocessed_expressi
-00004150: 6f6e 0a20 2020 2020 2020 203a 7265 7475  on.        :retu
-00004160: 726e 3a20 4120 4461 7461 4672 616d 6520  rn: A DataFrame 
-00004170: 6f66 2074 6865 2065 7870 7265 7373 696f  of the expressio
-00004180: 6e20 6d61 7472 6978 2066 6f72 2074 6865  n matrix for the
-00004190: 2071 7565 7269 6564 2064 6174 6173 6574   queried dataset
-000041a0: 0a20 2020 2020 2020 203a 7274 7970 653a  .        :rtype:
-000041b0: 2044 6174 6146 7261 6d65 0a0a 2020 2020   DataFrame..    
-000041c0: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
-000041d0: 7265 7370 6f6e 7365 203d 2073 656c 662e  response = self.
-000041e0: 7261 772e 6765 745f 6461 7461 7365 745f  raw.get_dataset_
-000041f0: 7072 6f63 6573 7365 645f 6578 7072 6573  processed_expres
-00004200: 7369 6f6e 2864 6174 6173 6574 2c20 2a2a  sion(dataset, **
-00004210: 6b77 6172 6773 290a 2020 2020 2020 2020  kwargs).        
-00004220: 0a20 2020 2020 2020 2064 6620 3d20 7073  .        df = ps
-00004230: 2e70 726f 6365 7373 5f65 7870 7265 7373  .process_express
-00004240: 696f 6e28 7265 7370 6f6e 7365 2c64 6174  ion(response,dat
-00004250: 6173 6574 2c73 656c 6629 0a20 2020 2020  aset,self).     
-00004260: 2020 200a 2020 2020 2020 2020 7265 7475     .        retu
-00004270: 726e 2064 660a 2020 2020 0a20 2020 2023  rn df.    .    #
-00004280: 2064 6174 6173 6574 732f 7b64 6174 6173   datasets/{datas
-00004290: 6574 7d2f 7175 616e 7469 7461 7469 6f6e  et}/quantitation
-000042a0: 5479 7065 7320 6765 745f 6461 7461 7365  Types get_datase
-000042b0: 745f 7175 616e 7469 7461 7469 6f6e 5f74  t_quantitation_t
-000042c0: 7970 6573 202d 2d2d 2d2d 2d2d 2d2d 2d0a  ypes ----------.
-000042d0: 2020 2020 0a20 2020 2064 6566 2067 6574      .    def get
-000042e0: 5f64 6174 6173 6574 5f71 7561 6e74 6974  _dataset_quantit
-000042f0: 6174 696f 6e5f 7479 7065 7328 7365 6c66  ation_types(self
-00004300: 2c64 6174 6173 6574 3a69 6e74 7c73 7472  ,dataset:int|str
-00004310: 2c2a 2a6b 7761 7267 7329 2d3e 4461 7461  ,**kwargs)->Data
-00004320: 4672 616d 653a 0a20 2020 2020 2020 2022  Frame:.        "
-00004330: 2222 5265 7472 6965 7665 2071 7561 6e74  ""Retrieve quant
-00004340: 6974 6174 696f 6e20 7479 7065 7320 6f66  itation types of
-00004350: 2061 2064 6174 6173 6574 0a0a 2020 2020   a dataset..    
-00004360: 2020 2020 0a20 2020 2020 2020 203a 7061      .        :pa
-00004370: 7261 6d20 6461 7461 7365 743a 2009 4120  ram dataset: .A 
-00004380: 6e75 6d65 7269 6361 6c20 6461 7461 7365  numerical datase
-00004390: 7420 6964 656e 7469 6669 6572 206f 7220  t identifier or 
-000043a0: 6120 6461 7461 7365 7420 7368 6f72 7420  a dataset short 
-000043b0: 6e61 6d65 0a20 2020 2020 2020 203a 7479  name.        :ty
-000043c0: 7065 2064 6174 6173 6574 3a20 696e 747c  pe dataset: int|
-000043d0: 7374 720a 2020 2020 2020 2020 3a70 6172  str.        :par
-000043e0: 616d 202a 2a6b 7761 7267 733a 2041 6464  am **kwargs: Add
-000043f0: 6974 696f 6e61 6c20 6172 6775 6d65 6e74  itional argument
-00004400: 7320 746f 2070 6173 7320 746f 2072 6177  s to pass to raw
-00004410: 2e67 6574 5f64 6174 6173 6574 5f71 7561  .get_dataset_qua
-00004420: 6e74 6974 6174 696f 6e5f 7479 7065 730a  ntitation_types.
-00004430: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
-00004440: 2041 2044 6174 6146 7261 6d65 2063 6f6e   A DataFrame con
-00004450: 7461 696e 696e 6720 7468 6520 7175 616e  taining the quan
-00004460: 7469 7461 7469 6f6e 2074 7970 6573 0a20  titation types. 
-00004470: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-00004480: 2020 5468 6520 6669 656c 6473 206f 6620    The fields of 
-00004490: 7468 6520 6f75 7470 7574 2044 6174 6146  the output DataF
-000044a0: 7261 6d65 2061 7265 3a0a 2020 2020 2020  rame are:.      
-000044b0: 2020 2020 2020 2d20 6964 3a20 4966 206f        - id: If o
-000044c0: 6620 7468 6520 7175 616e 7469 7461 7469  f the quantitati
-000044d0: 6f6e 2074 7970 652e 2041 6e79 2072 6177  on type. Any raw
-000044e0: 2071 7561 6e74 6974 6174 696f 6e20 7479   quantitation ty
-000044f0: 7065 2063 616e 2062 650a 2020 2020 2020  pe can be.      
-00004500: 2020 2020 2020 2020 2062 7920 6765 745f           by get_
-00004510: 6461 7461 7365 745f 7261 775f 6578 7072  dataset_raw_expr
-00004520: 6573 7369 6f6e 2066 756e 6374 696f 6e20  ession function 
-00004530: 7573 696e 6720 7468 6973 2069 642e 0a20  using this id.. 
-00004540: 2020 2020 2020 2020 2020 202d 206e 616d             - nam
-00004550: 653a 204e 616d 6520 6f66 2074 6865 2071  e: Name of the q
-00004560: 7561 6e74 6974 6174 696f 6e20 7479 7065  uantitation type
-00004570: 0a20 2020 2020 2020 2020 2020 202d 2064  .            - d
-00004580: 6573 6372 6970 7469 6f6e 3a20 4465 7363  escription: Desc
-00004590: 7269 7074 696f 6e20 6f66 2074 6865 2071  ription of the q
-000045a0: 7561 6e74 6974 6174 696f 6e20 7479 7065  uantitation type
-000045b0: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
-000045c0: 7970 653a 2054 7970 6520 6f66 2074 6865  ype: Type of the
-000045d0: 2071 7561 6e74 6974 6174 696f 6e20 7479   quantitation ty
-000045e0: 7065 2e20 4569 7468 6572 2072 6177 206f  pe. Either raw o
-000045f0: 7220 7072 6f63 6573 7365 642e 200a 2020  r processed. .  
-00004600: 2020 2020 2020 2020 2020 2020 4561 6368              Each
-00004610: 2064 6174 6173 6574 2077 696c 6c20 6861   dataset will ha
-00004620: 7665 206f 6e65 2070 726f 6365 7373 6564  ve one processed
-00004630: 2071 7561 6e74 6974 6174 696f 6e20 7479   quantitation ty
-00004640: 7065 2077 6869 6368 2069 7320 7468 6520  pe which is the 
-00004650: 0a20 2020 2020 2020 2020 2020 2020 2064  .              d
-00004660: 6174 6120 7265 7475 726e 6564 2075 7369  ata returned usi
-00004670: 6e67 2067 6574 5f64 6174 6173 6574 5f70  ng get_dataset_p
-00004680: 726f 6365 7373 6564 5f65 7870 7265 7373  rocessed_express
-00004690: 696f 6e0a 2020 2020 2020 2020 2020 2020  ion.            
-000046a0: 2d20 7261 7469 6f3a 2057 6865 7468 6572  - ratio: Whether
-000046b0: 206f 7220 6e6f 7420 7468 6520 7175 616e   or not the quan
-000046c0: 6974 6174 696f 6e20 7479 7065 2069 7320  itation type is 
-000046d0: 6120 7261 7469 6f20 6f66 206d 756c 7469  a ratio of multi
-000046e0: 706c 650a 2020 2020 2020 2020 2020 2020  ple.            
-000046f0: 2020 7175 616e 7469 7461 7469 6f6e 2074    quantitation t
-00004700: 7970 6573 2e20 5479 7069 6361 6c6c 7920  ypes. Typically 
-00004710: 5452 5545 2066 6f72 2070 726f 6365 7373  TRUE for process
-00004720: 6564 2054 574f 434f 4c4f 5220 7175 616e  ed TWOCOLOR quan
-00004730: 7469 7461 7469 6f6e 2074 7970 652e 0a20  titation type.. 
-00004740: 2020 2020 2020 2020 2020 202d 2070 7265             - pre
-00004750: 6665 7272 6564 3a20 5468 6520 7072 6566  ferred: The pref
-00004760: 6572 7265 6420 7261 7720 7175 616e 7469  erred raw quanti
-00004770: 7461 7469 6f6e 2074 7970 652e 2054 6869  tation type. Thi
-00004780: 7320 7665 7273 696f 6e20 6973 200a 2020  s version is .  
-00004790: 2020 2020 2020 2020 2020 2020 7573 6564              used
-000047a0: 2069 6e20 6765 6e65 7261 7469 6f6e 206f   in generation o
-000047b0: 6620 7468 6520 7072 6f63 6573 7365 6420  f the processed 
-000047c0: 6461 7461 2077 6974 6869 6e20 6765 6d6d  data within gemm
-000047d0: 612e 0a20 2020 2020 2020 2020 2020 202d  a..            -
-000047e0: 2072 6563 6f6d 7075 7465 643a 2049 6620   recomputed: If 
-000047f0: 5452 5545 2074 6869 7320 7175 616e 7469  TRUE this quanti
-00004800: 7461 7469 6f6e 2074 7970 6520 6973 2067  tation type is g
-00004810: 656e 6572 6174 6564 2062 790a 2020 2020  enerated by.    
-00004820: 2020 2020 2020 2020 2020 7265 636f 6d70            recomp
-00004830: 7574 696e 6720 7261 7720 6461 7461 2066  uting raw data f
-00004840: 696c 6573 2047 656d 6d61 2068 6164 2061  iles Gemma had a
-00004850: 6363 6573 7320 746f 0a20 2020 2020 2020  ccess to.       
-00004860: 203a 7274 7970 653a 2044 6174 6146 7261   :rtype: DataFra
-00004870: 6d65 0a0a 2020 2020 2020 2020 2222 220a  me..        """.
-00004880: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-00004890: 2072 6573 706f 6e73 6520 3d20 7365 6c66   response = self
-000048a0: 2e72 6177 2e67 6574 5f64 6174 6173 6574  .raw.get_dataset
-000048b0: 5f71 7561 6e74 6974 6174 696f 6e5f 7479  _quantitation_ty
-000048c0: 7065 7328 6461 7461 7365 742c 202a 2a6b  pes(dataset, **k
-000048d0: 7761 7267 7329 0a20 2020 2020 2020 2064  wargs).        d
-000048e0: 6620 3d20 7073 2e70 726f 6365 7373 5f51  f = ps.process_Q
-000048f0: 7561 6e74 6974 6174 696f 6e54 7970 6556  uantitationTypeV
-00004900: 616c 7565 4f62 6a65 6374 2872 6573 706f  alueObject(respo
-00004910: 6e73 652e 6461 7461 290a 2020 2020 2020  nse.data).      
-00004920: 2020 0a20 2020 2020 2020 200a 2020 2020    .        .    
-00004930: 2020 2020 7265 7475 726e 2064 660a 0a20      return df.. 
-00004940: 2020 2023 2064 6174 6173 6574 732f 7b64     # datasets/{d
-00004950: 6174 6173 6574 7d2f 6461 7461 2f72 6177  ataset}/data/raw
-00004960: 2c20 6765 745f 6461 7461 7365 745f 7261  , get_dataset_ra
-00004970: 775f 6578 7072 6573 7369 6f6e 202d 2d2d  w_expression ---
-00004980: 2d2d 2d2d 2d2d 0a20 2020 2064 6566 2067  ------.    def g
-00004990: 6574 5f64 6174 6173 6574 5f72 6177 5f65  et_dataset_raw_e
-000049a0: 7870 7265 7373 696f 6e28 7365 6c66 2c64  xpression(self,d
-000049b0: 6174 6173 6574 3a69 6e74 7c73 7472 2c0a  ataset:int|str,.
-000049c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000049d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000049e0: 2020 2071 7561 6e74 6974 6174 696f 6e5f     quantitation_
-000049f0: 7479 7065 3a5b 696e 745d 2c2a 2a6b 7761  type:[int],**kwa
-00004a00: 7267 7329 2d3e 4461 7461 4672 616d 653a  rgs)->DataFrame:
-00004a10: 0a20 2020 2020 2020 2022 2222 0a20 2020  .        """.   
-00004a20: 2020 2020 200a 2020 2020 2020 2020 3a70       .        :p
-00004a30: 6172 616d 2064 6174 6173 6574 3a20 4120  aram dataset: A 
-00004a40: 6e75 6d65 7269 6361 6c20 6461 7461 7365  numerical datase
-00004a50: 7420 6964 656e 7469 6669 6572 206f 7220  t identifier or 
-00004a60: 6120 6461 7461 7365 7420 7368 6f72 7420  a dataset short 
-00004a70: 6e61 6d65 0a20 2020 2020 2020 203a 7479  name.        :ty
-00004a80: 7065 2064 6174 6173 6574 3a20 696e 747c  pe dataset: int|
-00004a90: 7374 720a 2020 2020 2020 2020 3a70 6172  str.        :par
-00004aa0: 616d 2071 7561 6e74 6974 6174 696f 6e5f  am quantitation_
-00004ab0: 7479 7065 3a20 5175 616e 7469 7461 7469  type: Quantitati
-00004ac0: 6f6e 2074 7970 6520 6964 2e20 5468 6573  on type id. Thes
-00004ad0: 6520 6361 6e20 6265 2061 6371 7569 7265  e can be acquire
-00004ae0: 6420 0a20 2020 2020 2020 2020 2075 7369  d .          usi
-00004af0: 6e67 2067 6574 5f64 6174 6173 6574 5f71  ng get_dataset_q
-00004b00: 7561 6e74 6974 6174 696f 6e5f 7479 7065  uantitation_type
-00004b10: 7320 6675 6e63 7469 6f6e 2e20 5468 6973  s function. This
-00004b20: 2065 6e64 706f 696e 7420 6361 6e20 6f6e   endpoint can on
-00004b30: 6c79 200a 2020 2020 2020 2020 2020 7265  ly .          re
-00004b40: 7475 726e 206e 6f6e 2d70 726f 6365 7373  turn non-process
-00004b50: 6564 2071 7561 6e74 6974 6174 696f 6e20  ed quantitation 
-00004b60: 7479 7065 732e 0a20 2020 2020 2020 203a  types..        :
-00004b70: 7479 7065 2071 7561 6e74 6974 6174 696f  type quantitatio
-00004b80: 6e5f 7479 7065 3a20 5b69 6e74 5d0a 2020  n_type: [int].  
-00004b90: 2020 2020 2020 3a70 6172 616d 202a 2a6b        :param **k
-00004ba0: 7761 7267 733a 2041 6464 6974 696f 6e61  wargs: Additiona
-00004bb0: 6c20 6172 6775 6d65 6e74 7320 746f 2070  l arguments to p
-00004bc0: 6173 7320 746f 2072 6177 2e67 6574 5f64  ass to raw.get_d
-00004bd0: 6174 6173 6574 5f72 6177 5f65 7870 7265  ataset_raw_expre
-00004be0: 7373 696f 6e0a 2020 2020 2020 2020 3a72  ssion.        :r
-00004bf0: 6574 7572 6e3a 2041 2044 6174 6146 7261  eturn: A DataFra
-00004c00: 6d65 206f 6620 7468 6520 6578 7072 6573  me of the expres
-00004c10: 7369 6f6e 206d 6174 7269 7820 666f 7220  sion matrix for 
-00004c20: 7468 6520 7175 6572 6965 6420 6461 7461  the queried data
-00004c30: 7365 740a 2020 2020 2020 2020 3a72 7479  set.        :rty
-00004c40: 7065 3a20 4461 7461 4672 616d 650a 0a20  pe: DataFrame.. 
-00004c50: 2020 2020 2020 2022 2222 0a20 2020 2020         """.     
-00004c60: 2020 200a 2020 2020 2020 2020 6b77 6172     .        kwar
-00004c70: 6773 203d 2076 732e 7265 6d6f 7665 5f6e  gs = vs.remove_n
-00004c80: 6f6e 6573 280a 2020 2020 2020 2020 2020  ones(.          
-00004c90: 2020 7175 616e 7469 7461 7469 6f6e 5f74    quantitation_t
-00004ca0: 7970 6520 3d20 7175 616e 7469 7461 7469  ype = quantitati
-00004cb0: 6f6e 5f74 7970 652c 0a20 2020 2020 2020  on_type,.       
-00004cc0: 2020 2020 202a 2a6b 7761 7267 7329 0a20       **kwargs). 
-00004cd0: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-00004ce0: 7265 7370 6f6e 7365 203d 2073 656c 662e  response = self.
-00004cf0: 7261 772e 6765 745f 6461 7461 7365 745f  raw.get_dataset_
-00004d00: 7261 775f 6578 7072 6573 7369 6f6e 2864  raw_expression(d
-00004d10: 6174 6173 6574 2c20 2a2a 6b77 6172 6773  ataset, **kwargs
-00004d20: 290a 2020 2020 2020 2020 0a20 2020 2020  ).        .     
-00004d30: 2020 2064 6620 3d20 7073 2e70 726f 6365     df = ps.proce
-00004d40: 7373 5f65 7870 7265 7373 696f 6e28 7265  ss_expression(re
-00004d50: 7370 6f6e 7365 2c64 6174 6173 6574 2c73  sponse,dataset,s
-00004d60: 656c 6629 0a20 2020 2020 2020 200a 2020  elf).        .  
-00004d70: 2020 2020 2020 7265 7475 726e 2064 660a        return df.
-00004d80: 2020 2020 0a20 2020 200a 2020 2020 2320      .    .    # 
-00004d90: 6461 7461 7365 7473 2f7b 6461 7461 7365  datasets/{datase
-00004da0: 747d 2f73 616d 706c 6573 2c20 6765 745f  t}/samples, get_
-00004db0: 6461 7461 7365 745f 7361 6d70 6c65 7320  dataset_samples 
-00004dc0: 2d2d 2d2d 2d2d 2d2d 0a20 2020 2064 6566  --------.    def
-00004dd0: 2067 6574 5f64 6174 6173 6574 5f73 616d   get_dataset_sam
-00004de0: 706c 6573 2873 656c 662c 2064 6174 6173  ples(self, datas
-00004df0: 6574 3a69 6e74 7c73 7472 2c20 2a2a 6b77  et:int|str, **kw
-00004e00: 6172 6773 292d 3e44 6174 6146 7261 6d65  args)->DataFrame
-00004e10: 3a0a 2020 2020 2020 2020 2222 220a 2020  :.        """.  
-00004e20: 2020 2020 2020 5265 7472 6965 7665 2074        Retrieve t
-00004e30: 6865 2073 616d 706c 6573 206f 6620 6120  he samples of a 
-00004e40: 6461 7461 7365 740a 0a0a 2020 2020 2020  dataset...      
-00004e50: 2020 3a70 6172 616d 2064 6174 6173 6574    :param dataset
-00004e60: 3a20 4120 6e75 6d65 7269 6361 6c20 6461  : A numerical da
-00004e70: 7461 7365 7420 6964 656e 7469 6669 6572  taset identifier
-00004e80: 206f 7220 6120 6461 7461 7365 7420 7368   or a dataset sh
-00004e90: 6f72 7420 6e61 6d65 0a20 2020 2020 2020  ort name.       
-00004ea0: 203a 7479 7065 2064 6174 6173 6574 3a20   :type dataset: 
-00004eb0: 696e 747c 7374 720a 2020 2020 2020 2020  int|str.        
-00004ec0: 3a70 6172 616d 202a 2a6b 7761 7267 733a  :param **kwargs:
-00004ed0: 2041 6464 6974 696f 6e61 6c20 6172 6775   Additional argu
-00004ee0: 6d65 6e74 7320 746f 2070 6173 7320 746f  ments to pass to
-00004ef0: 2072 6177 2e67 6574 5f64 6174 6173 6574   raw.get_dataset
-00004f00: 5f73 616d 706c 6573 0a20 2020 2020 2020  _samples.       
-00004f10: 203a 7265 7475 726e 3a20 4120 4461 7461   :return: A Data
-00004f20: 4672 616d 6520 7769 7468 2069 6e66 6f72  Frame with infor
-00004f30: 6d61 7469 6f6e 2061 626f 7574 2074 6865  mation about the
-00004f40: 2073 616d 706c 6573 206f 6620 7468 6520   samples of the 
-00004f50: 7175 6572 6965 6420 6461 7461 7365 742e  queried dataset.
-00004f60: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
-00004f70: 2020 2020 5468 6520 6669 656c 6473 206f      The fields o
-00004f80: 6620 7468 6520 4461 7461 4672 616d 6520  f the DataFrame 
-00004f90: 6172 653a 0a20 2020 2020 2020 2020 2020  are:.           
-00004fa0: 202d 2073 616d 706c 655f 6e61 6d65 3a20   - sample_name: 
-00004fb0: 496e 7465 726e 616c 206e 616d 6520 6769  Internal name gi
-00004fc0: 7665 6e20 746f 2074 6865 2073 616d 706c  ven to the sampl
-00004fd0: 652e 0a20 2020 2020 2020 2020 2020 202d  e..            -
-00004fe0: 2073 616d 706c 655f 4944 3a20 496e 7465   sample_ID: Inte
-00004ff0: 726e 616c 2049 4420 6f66 2074 6865 2073  rnal ID of the s
-00005000: 616d 706c 650a 2020 2020 2020 2020 2020  ample.          
-00005010: 2020 2d20 7361 6d70 6c65 5f64 6573 6372    - sample_descr
-00005020: 6970 7469 6f6e 3a20 4672 6565 2074 6578  iption: Free tex
-00005030: 7420 6465 7363 7269 7074 696f 6e20 6f66  t description of
-00005040: 2074 6865 2073 616d 706c 650a 2020 2020   the sample.    
-00005050: 2020 2020 2020 2020 2d20 7361 6d70 6c65          - sample
-00005060: 5f6f 7574 6c69 6572 3a20 5768 6574 6865  _outlier: Whethe
-00005070: 7220 6f72 206e 6f74 2074 6865 2073 616d  r or not the sam
-00005080: 706c 6520 6973 206d 6172 6b65 6420 6173  ple is marked as
-00005090: 2061 6e20 6f75 746c 6965 720a 2020 2020   an outlier.    
-000050a0: 2020 2020 2020 2020 2d20 7361 6d70 6c65          - sample
-000050b0: 5f61 6363 6573 7369 6f6e 3a20 4163 6365  _accession: Acce
-000050c0: 7373 696f 6e20 4944 206f 6620 7468 6520  ssion ID of the 
-000050d0: 7361 6d70 6c65 2069 6e20 6974 2773 206f  sample in it's o
-000050e0: 7269 6769 6e61 6c20 6461 7461 6261 7365  riginal database
-000050f0: 0a20 2020 2020 2020 2020 2020 202d 2073  .            - s
-00005100: 616d 706c 655f 6461 7461 6261 7365 3a20  ample_database: 
-00005110: 4461 7461 6261 7365 206f 6620 6f72 6967  Database of orig
-00005120: 696e 2066 6f72 2074 6865 2073 616d 706c  in for the sampl
-00005130: 650a 2020 2020 2020 2020 2020 2020 2d20  e.            - 
-00005140: 7361 6d70 6c65 5f63 6861 7261 6374 6572  sample_character
-00005150: 6973 7469 6373 3a20 4368 6172 6163 7465  istics: Characte
-00005160: 7269 7374 6963 7320 6f66 2074 6865 2073  ristics of the s
-00005170: 616d 706c 652e 2054 6869 7320 6669 656c  ample. This fiel
-00005180: 6420 6973 2061 2064 6174 6120 7461 626c  d is a data tabl
-00005190: 650a 2020 2020 2020 2020 2020 2020 2d20  e.            - 
-000051a0: 7361 6d70 6c65 5f66 6163 746f 725f 7661  sample_factor_va
-000051b0: 6c75 6573 3a20 4578 7065 7269 6d65 6e74  lues: Experiment
-000051c0: 616c 2066 6163 746f 7220 7661 6c75 6573  al factor values
-000051d0: 206f 6620 7468 6520 7361 6d70 6c65 2e20   of the sample. 
-000051e0: 5468 6973 2066 6965 6c64 2069 7320 6120  This field is a 
-000051f0: 6461 7461 2074 6162 6c65 0a20 2020 2020  data table.     
-00005200: 2020 203a 7274 7970 653a 2044 6174 6146     :rtype: DataF
-00005210: 7261 6d65 0a0a 2020 2020 2020 2020 2222  rame..        ""
-00005220: 220a 2020 2020 2020 2020 7265 7370 6f6e  ".        respon
-00005230: 7365 203d 2073 656c 662e 7261 772e 6765  se = self.raw.ge
-00005240: 745f 6461 7461 7365 745f 7361 6d70 6c65  t_dataset_sample
-00005250: 7328 6461 7461 7365 742c 202a 2a6b 7761  s(dataset, **kwa
-00005260: 7267 7329 0a20 2020 2020 2020 2064 6620  rgs).        df 
-00005270: 3d20 7073 2e70 726f 6365 7373 5f73 616d  = ps.process_sam
-00005280: 706c 6573 2872 6573 706f 6e73 652e 6461  ples(response.da
-00005290: 7461 290a 2020 2020 2020 2020 7265 7475  ta).        retu
-000052a0: 726e 2064 660a 2020 2020 2020 2020 0a20  rn df.        . 
-000052b0: 2020 2023 2064 6174 6173 6574 732f 7b64     # datasets/{d
-000052c0: 6174 6173 6574 7d2f 7376 6420 2d2d 2d20  ataset}/svd --- 
-000052d0: 0a20 2020 2023 206e 6f74 2069 6d70 6c65  .    # not imple
-000052e0: 6d65 6e74 6564 0a20 2020 200a 2020 2020  mented.    .    
-000052f0: 2320 6461 7461 7365 7473 2c20 6765 745f  # datasets, get_
-00005300: 6461 7461 7365 7473 202d 2d2d 2d2d 2d0a  datasets ------.
-00005310: 2020 2020 6465 6620 6765 745f 6461 7461      def get_data
-00005320: 7365 7473 2873 656c 662c 7175 6572 793a  sets(self,query:
-00005330: 4f70 7469 6f6e 616c 5b73 7472 5d20 3d20  Optional[str] = 
-00005340: 4e6f 6e65 2c20 0a20 2020 2020 2020 2020  None, .         
-00005350: 2020 2020 2020 2020 2020 2020 6669 6c74              filt
-00005360: 6572 3a4f 7074 696f 6e61 6c5b 7374 725d  er:Optional[str]
-00005370: 203d 204e 6f6e 652c 200a 2020 2020 2020   = None, .      
-00005380: 2020 2020 2020 2020 2020 2020 2020 2074                 t
-00005390: 6178 613a 4f70 7469 6f6e 616c 5b4c 6973  axa:Optional[Lis
-000053a0: 745b 7374 725d 5d20 3d20 4e6f 6e65 2c20  t[str]] = None, 
-000053b0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-000053c0: 2020 2020 2020 7572 6973 3a4f 7074 696f        uris:Optio
-000053d0: 6e61 6c5b 4c69 7374 5b73 7472 5d5d 203d  nal[List[str]] =
-000053e0: 204e 6f6e 652c 0a20 2020 2020 2020 2020   None,.         
-000053f0: 2020 2020 2020 2020 2020 2020 6f66 6673              offs
-00005400: 6574 3a69 6e74 203d 2030 2c0a 2020 2020  et:int = 0,.    
-00005410: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00005420: 206c 696d 6974 3a69 6e74 203d 2032 302c   limit:int = 20,
-00005430: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00005440: 2020 2020 2020 736f 7274 3a73 7472 203d        sort:str =
-00005450: 2022 2b69 6422 2c0a 2020 2020 2020 2020   "+id",.        
-00005460: 2020 2020 2020 2020 2020 2020 202a 2a6b               **k
-00005470: 7761 7267 7329 2d3e 4461 7461 4672 616d  wargs)->DataFram
-00005480: 653a 0a20 2020 2020 2020 2022 2222 0a20  e:.        """. 
-00005490: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-000054a0: 3a70 6172 616d 2071 7565 7279 3a20 5468  :param query: Th
-000054b0: 6520 7365 6172 6368 2071 7565 7279 2e20  e search query. 
-000054c0: 4569 7468 6572 2070 6c61 696e 2074 6578  Either plain tex
-000054d0: 7420 2827 7472 6175 6d61 7469 6327 292c  t ('traumatic'),
-000054e0: 206f 7220 616e 200a 2020 2020 2020 2020   or an .        
-000054f0: 2020 6f6e 746f 6c6f 6779 2074 6572 6d20    ontology term 
-00005500: 5552 4920 2827 6874 7470 3a2f 2f70 7572  URI ('http://pur
-00005510: 6c2e 6f62 6f6c 6962 7261 7279 2e6f 7267  l.obolibrary.org
-00005520: 2f6f 626f 2f55 4245 524f 4e5f 3030 3032  /obo/UBERON_0002
-00005530: 3034 3827 292e 0a20 2020 2020 2020 2020  048')..         
-00005540: 2044 6174 6173 6574 7320 7468 6174 2063   Datasets that c
-00005550: 6f6e 7461 696e 2074 6865 2067 6976 656e  ontain the given
-00005560: 2073 7472 696e 6720 696e 2074 6865 6972   string in their
-00005570: 2073 686f 7274 206f 6620 6675 6c6c 206e   short of full n
-00005580: 616d 6520 7769 6c6c 200a 2020 2020 2020  ame will .      
-00005590: 2020 2020 616c 736f 2062 6520 6d61 7463      also be matc
-000055a0: 6865 642e 2c20 6465 6661 756c 7473 2074  hed., defaults t
-000055b0: 6f20 4e6f 6e65 0a20 2020 2020 2020 203a  o None.        :
-000055c0: 7479 7065 2071 7565 7279 3a20 4f70 7469  type query: Opti
-000055d0: 6f6e 616c 5b73 7472 5d2c 206f 7074 696f  onal[str], optio
-000055e0: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
-000055f0: 616d 2066 696c 7465 723a 2046 696c 7465  am filter: Filte
-00005600: 7220 7265 7375 6c74 7320 6279 206d 6174  r results by mat
-00005610: 6368 696e 6720 6578 7072 6573 7369 6f6e  ching expression
-00005620: 2e20 5573 6520 0a20 2020 2020 2020 2020  . Use .         
-00005630: 2066 696c 7465 725f 7072 6f70 6572 7469   filter_properti
-00005640: 6573 2066 756e 6374 696f 6e20 746f 2067  es function to g
-00005650: 6574 2061 206c 6973 7420 6f66 2061 6c6c  et a list of all
-00005660: 2061 7661 696c 6162 6c65 2070 6172 616d   available param
-00005670: 6574 6572 732e 200a 2020 2020 2020 2020  eters. .        
-00005680: 2020 5468 6573 6520 7072 6f70 6572 7469    These properti
-00005690: 6573 2063 616e 2062 6520 636f 6d62 696e  es can be combin
-000056a0: 6564 2075 7369 6e67 2022 616e 6422 2022  ed using "and" "
-000056b0: 6f72 2220 636c 6175 7365 7320 616e 6420  or" clauses and 
-000056c0: 6d61 7920 0a20 2020 2020 2020 2020 2063  may .          c
-000056d0: 6f6e 7461 696e 2063 6f6d 6d6f 6e20 6f70  ontain common op
-000056e0: 6572 6174 6f72 7320 7375 6368 2061 7320  erators such as 
-000056f0: 223d 222c 2022 3c22 206f 7220 2269 6e22  "=", "<" or "in"
-00005700: 2e20 2865 2e67 2e20 0a20 2020 2020 2020  . (e.g. .       
-00005710: 2020 2022 7461 786f 6e2e 636f 6d6d 6f6e     "taxon.common
-00005720: 4e61 6d65 203d 2068 756d 616e 222c 2022  Name = human", "
-00005730: 7461 786f 6e2e 636f 6d6d 6f6e 4e61 6d65  taxon.commonName
-00005740: 2069 6e20 2868 756d 616e 2c6d 6f75 7365   in (human,mouse
-00005750: 292c 200a 2020 2020 2020 2020 2020 2269  ), .          "i
-00005760: 6420 3c20 3130 3030 2229 2c20 6465 6661  d < 1000"), defa
-00005770: 756c 7473 2074 6f20 4e6f 6e65 0a20 2020  ults to None.   
-00005780: 2020 2020 203a 7479 7065 2066 696c 7465       :type filte
-00005790: 723a 204f 7074 696f 6e61 6c5b 7374 725d  r: Optional[str]
-000057a0: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
-000057b0: 2020 203a 7061 7261 6d20 7461 7861 3a20     :param taxa: 
-000057c0: 4120 7665 6374 6f72 206f 6620 7461 786f  A vector of taxo
-000057d0: 6e20 636f 6d6d 6f6e 206e 616d 6573 2028  n common names (
-000057e0: 652e 672e 2068 756d 616e 2c20 6d6f 7573  e.g. human, mous
-000057f0: 652c 2072 6174 292e 0a20 2020 2020 2020  e, rat)..       
-00005800: 2020 2050 726f 7669 6469 6e67 206d 756c     Providing mul
-00005810: 7469 706c 6520 7370 6563 6965 7320 7769  tiple species wi
-00005820: 6c6c 2072 6574 7572 6e20 7265 7375 6c74  ll return result
-00005830: 7320 666f 7220 616c 6c20 7370 6563 6965  s for all specie
-00005840: 732e 2054 6865 7365 200a 2020 2020 2020  s. These .      
-00005850: 2020 2020 6172 6520 6170 7065 6e64 6564      are appended
-00005860: 2074 6f20 7468 6520 6669 6c74 6572 2061   to the filter a
-00005870: 6e64 2065 7175 6976 616c 656e 7420 746f  nd equivalent to
-00005880: 2066 696c 7465 7269 6e67 2066 6f72 200a   filtering for .
-00005890: 2020 2020 2020 2020 2020 7461 786f 6e2e            taxon.
-000058a0: 636f 6d6d 6f6e 4e61 6d65 2070 726f 7065  commonName prope
-000058b0: 7274 792c 2064 6566 6175 6c74 7320 746f  rty, defaults to
-000058c0: 204e 6f6e 650a 2020 2020 2020 2020 3a70   None.        :p
-000058d0: 6172 616d 2074 6178 613a 2041 206c 6973  aram taxa: A lis
-000058e0: 7420 6f66 2074 6178 6f6e 2063 6f6d 6d6f  t of taxon commo
-000058f0: 6e20 6e61 6d65 7320 2865 2e67 2e20 6875  n names (e.g. hu
-00005900: 6d61 6e2c 206d 6f75 7365 2c20 7261 7429  man, mouse, rat)
-00005910: 2e20 0a20 2020 2020 2020 2020 2050 726f  . .          Pro
-00005920: 7669 6469 6e67 206d 756c 7469 706c 6520  viding multiple 
-00005930: 7370 6563 6965 7320 7769 6c6c 2072 6574  species will ret
-00005940: 7572 6e20 7265 7375 6c74 7320 666f 7220  urn results for 
-00005950: 616c 6c20 7370 6563 6965 732e 200a 2020  all species. .  
-00005960: 2020 2020 2020 2020 5468 6573 6520 6172          These ar
-00005970: 6520 6170 7065 6e64 6564 2074 6f20 7468  e appended to th
-00005980: 6520 6669 6c74 6572 2061 6e64 2065 7175  e filter and equ
-00005990: 6976 616c 656e 7420 746f 2066 696c 7465  ivalent to filte
-000059a0: 7269 6e67 2066 6f72 200a 2020 2020 2020  ring for .      
-000059b0: 2020 2020 7461 786f 6e2e 636f 6d6d 6f6e      taxon.common
-000059c0: 4e61 6d65 2070 726f 7065 7274 792c 2064  Name property, d
-000059d0: 6566 6175 6c74 7320 746f 204e 6f6e 650a  efaults to None.
-000059e0: 2020 2020 2020 2020 3a74 7970 6520 7461          :type ta
-000059f0: 7861 3a20 4f70 7469 6f6e 616c 5b4c 6973  xa: Optional[Lis
-00005a00: 745b 7374 725d 5d2c 206f 7074 696f 6e61  t[str]], optiona
-00005a10: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
-00005a20: 2075 7269 733a 2041 2076 6563 746f 7220   uris: A vector 
-00005a30: 6f66 206f 6e74 6f6c 6f67 7920 7465 726d  of ontology term
-00005a40: 2055 5249 732e 2050 726f 7669 6469 6e67   URIs. Providing
-00005a50: 206d 756c 7469 706c 6520 7465 726d 730a   multiple terms.
-00005a60: 2020 2020 2020 2020 2020 7769 6c6c 2072            will r
-00005a70: 6574 7572 6e20 7265 7375 6c74 7320 636f  eturn results co
-00005a80: 6e74 6169 6e69 6e67 2061 6e79 206f 6620  ntaining any of 
-00005a90: 7468 6520 7465 726d 7320 616e 6420 7468  the terms and th
-00005aa0: 6569 7220 6368 696c 6472 656e 2e20 0a20  eir children. . 
-00005ab0: 2020 2020 2020 2020 2054 6865 7365 2061           These a
-00005ac0: 7265 2061 7070 656e 6465 6420 746f 2074  re appended to t
-00005ad0: 6865 2066 696c 7465 7220 616e 6420 6571  he filter and eq
-00005ae0: 7569 7661 6c65 6e74 2074 6f20 6669 6c74  uivalent to filt
-00005af0: 6572 696e 6720 666f 7220 0a20 2020 2020  ering for .     
-00005b00: 2020 2020 2061 6c6c 4368 6172 6163 7465       allCharacte
-00005b10: 7269 7374 6963 732e 7661 6c75 6555 7269  ristics.valueUri
-00005b20: 2c20 6465 6661 756c 7473 2074 6f20 4e6f  , defaults to No
-00005b30: 6e65 0a20 2020 2020 2020 203a 7479 7065  ne.        :type
-00005b40: 2075 7269 733a 204f 7074 696f 6e61 6c5b   uris: Optional[
-00005b50: 4c69 7374 5b73 7472 5d5d 2c20 6f70 7469  List[str]], opti
-00005b60: 6f6e 616c 0a20 2020 2020 2020 203a 7061  onal.        :pa
-00005b70: 7261 6d20 6f66 6673 6574 3a20 5468 6520  ram offset: The 
-00005b80: 6f66 6673 6574 206f 6620 7468 6520 6669  offset of the fi
-00005b90: 7273 7420 7265 7472 6965 7665 6420 7265  rst retrieved re
-00005ba0: 7375 6c74 2e2c 2064 6566 6175 6c74 7320  sult., defaults 
-00005bb0: 746f 2030 0a20 2020 2020 2020 203a 7479  to 0.        :ty
-00005bc0: 7065 206f 6666 7365 743a 2069 6e74 2c20  pe offset: int, 
-00005bd0: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
-00005be0: 203a 7061 7261 6d20 6c69 6d69 743a 204c   :param limit: L
-00005bf0: 696d 6974 7320 7468 6520 7265 7375 6c74  imits the result
-00005c00: 2074 6f20 7370 6563 6966 6965 6420 616d   to specified am
-00005c10: 6f75 6e74 206f 6620 6f62 6a65 6374 732e  ount of objects.
-00005c20: 0a20 2020 2020 2020 2020 2048 6173 2061  .          Has a
-00005c30: 206d 6178 696d 756d 2076 616c 7565 206f   maximum value o
-00005c40: 6620 3130 302e 2055 7365 2074 6f67 6574  f 100. Use toget
-00005c50: 6865 7220 7769 7468 206f 6666 7365 7420  her with offset 
-00005c60: 616e 6420 7468 6520 0a20 2020 2020 2020  and the .       
-00005c70: 2020 2074 6f74 616c 5f65 6c65 6d65 6e74     total_element
-00005c80: 7320 6174 7472 6962 7574 6520 696e 2074  s attribute in t
-00005c90: 6865 206f 7574 7075 7420 746f 2063 6f6d  he output to com
-00005ca0: 7069 6c65 2061 6c6c 2064 6174 6120 6966  pile all data if
-00005cb0: 206e 6565 6465 642e 0a20 2020 2020 2020   needed..       
-00005cc0: 2020 2041 6c74 6572 6e61 7469 7665 6c79     Alternatively
-00005cd0: 2067 6574 5f61 6c6c 5f70 6167 6573 2066   get_all_pages f
-00005ce0: 756e 6374 696f 6e20 6361 6e20 6265 2075  unction can be u
-00005cf0: 7365 6420 7769 7468 2061 6c6c 2066 756e  sed with all fun
-00005d00: 6374 696f 6e73 0a20 2020 2020 2020 2020  ctions.         
-00005d10: 2069 6e63 6c75 6469 6e67 206f 6666 7365   including offse
-00005d20: 7420 616e 6420 6c69 6d69 7420 7061 7261  t and limit para
-00005d30: 6d65 7465 7273 2c20 6465 6661 756c 7473  meters, defaults
-00005d40: 2074 6f20 3230 0a20 2020 2020 2020 203a   to 20.        :
-00005d50: 7479 7065 206c 696d 6974 3a20 696e 742c  type limit: int,
-00005d60: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
-00005d70: 2020 3a70 6172 616d 2073 6f72 743a 204f    :param sort: O
-00005d80: 7264 6572 2072 6573 756c 7473 2062 7920  rder results by 
-00005d90: 7468 6520 6769 7665 6e20 7072 6f70 6572  the given proper
-00005da0: 7479 2061 6e64 2064 6972 6563 7469 6f6e  ty and direction
-00005db0: 2e20 5468 6520 272b 270a 2020 2020 2020  . The '+'.      
-00005dc0: 2020 2020 7369 676e 2069 6e64 6963 6174      sign indicat
-00005dd0: 6520 6173 6365 6e64 696e 6720 6f72 6465  e ascending orde
-00005de0: 7220 7768 6572 6561 7320 7468 6520 272d  r whereas the '-
-00005df0: 2720 696e 6469 6361 7465 2064 6573 6365  ' indicate desce
-00005e00: 6e64 696e 672c 0a20 2020 2020 2020 2020  nding,.         
-00005e10: 2064 6566 6175 6c74 7320 746f 2022 2b69   defaults to "+i
-00005e20: 6422 0a20 2020 2020 2020 203a 7479 7065  d".        :type
-00005e30: 2073 6f72 743a 2073 7472 2c20 6f70 7469   sort: str, opti
-00005e40: 6f6e 616c 0a20 2020 2020 2020 203a 7061  onal.        :pa
-00005e50: 7261 6d20 2a2a 6b77 6172 6773 3a20 4164  ram **kwargs: Ad
-00005e60: 6469 7469 6f6e 616c 2061 7267 756d 656e  ditional argumen
-00005e70: 7473 2074 6f20 7061 7373 2074 6f20 7261  ts to pass to ra
-00005e80: 772e 6765 745f 6461 7461 7365 7473 0a20  w.get_datasets. 
-00005e90: 2020 2020 2020 203a 7265 7475 726e 3a20         :return: 
-00005ea0: 4120 4461 7461 4672 616d 6520 7769 7468  A DataFrame with
-00005eb0: 2069 6e66 6f72 6d61 7469 6f6e 2061 626f   information abo
-00005ec0: 7574 2074 6865 2071 7565 7269 6564 2064  ut the queried d
-00005ed0: 6174 6173 6574 2873 292e 0a20 2020 2020  ataset(s)..     
-00005ee0: 2020 200a 2020 2020 2020 2020 2020 5468     .          Th
-00005ef0: 6520 6669 656c 6473 206f 6620 7468 6520  e fields of the 
-00005f00: 4461 7461 4672 616d 6520 6172 653a 0a20  DataFrame are:. 
-00005f10: 2020 2020 2020 2020 2020 202d 2065 7870             - exp
-00005f20: 6572 696d 656e 745f 7368 6f72 745f 6e61  eriment_short_na
-00005f30: 6d65 3a20 5368 6f72 746e 616d 6520 6769  me: Shortname gi
-00005f40: 7665 6e20 746f 2074 6865 2064 6174 6173  ven to the datas
-00005f50: 6574 2077 6974 6869 6e20 4765 6d6d 612e  et within Gemma.
-00005f60: 204f 6674 656e 2063 6f72 7265 7370 6f6e   Often correspon
-00005f70: 6473 2074 6f20 6163 6365 7373 696f 6e20  ds to accession 
-00005f80: 4944 0a20 2020 2020 2020 2020 2020 202d  ID.            -
-00005f90: 2065 7870 6572 696d 656e 745f 6e61 6d65   experiment_name
-00005fa0: 3a20 4675 6c6c 2074 6974 6c65 206f 6620  : Full title of 
-00005fb0: 7468 6520 6461 7461 7365 740a 2020 2020  the dataset.    
+000033b0: 2020 636f 6e73 6f6c 6964 6174 6520 3d20    consolidate = 
+000033c0: 4e6f 6e65 2c0a 2020 2020 2020 2020 2020  None,.          
+000033d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000033e0: 2020 2020 2020 2020 2020 2020 2020 202a                 *
+000033f0: 2a6b 7761 7267 7329 2d3e 6469 6374 5b69  *kwargs)->dict[i
+00003400: 6e74 3a44 6174 6146 7261 6d65 5d3a 0a20  nt:DataFrame]:. 
+00003410: 2020 2020 2020 2022 2222 5265 7472 6965         """Retrie
+00003420: 7665 2074 6865 2065 7870 7265 7373 696f  ve the expressio
+00003430: 6e20 6461 7461 206d 6174 7269 7820 6f66  n data matrix of
+00003440: 2061 2073 6574 206f 6620 6461 7461 7365   a set of datase
+00003450: 7473 2061 6e64 2067 656e 6573 0a0a 0a20  ts and genes... 
+00003460: 2020 2020 2020 203a 7061 7261 6d20 6461         :param da
+00003470: 7461 7365 7473 3a20 4120 6e75 6d65 7269  tasets: A numeri
+00003480: 6361 6c20 6461 7461 7365 7420 6964 656e  cal dataset iden
+00003490: 7469 6669 6572 206f 7220 6120 6461 7461  tifier or a data
+000034a0: 7365 7420 7368 6f72 7420 6e61 6d65 0a20  set short name. 
+000034b0: 2020 2020 2020 203a 7479 7065 2064 6174         :type dat
+000034c0: 6173 6574 733a 204c 6973 745b 7374 727c  asets: List[str|
+000034d0: 696e 745d 0a20 2020 2020 2020 203a 7061  int].        :pa
+000034e0: 7261 6d20 6765 6e65 733a 2041 6e20 656e  ram genes: An en
+000034f0: 7365 6d62 6c20 6765 6e65 2069 6465 6e74  sembl gene ident
+00003500: 6966 6965 7220 7768 6963 6820 7479 7069  ifier which typi
+00003510: 6361 6c6c 7920 7374 6172 7473 2077 6974  cally starts wit
+00003520: 6820 0a20 2020 2020 2020 2020 2065 6e73  h .          ens
+00003530: 6720 6f72 2061 6e20 6e63 6269 2067 656e  g or an ncbi gen
+00003540: 6520 6964 656e 7469 6669 6572 206f 7220  e identifier or 
+00003550: 616e 206f 6666 6963 6961 6c20 6765 6e65  an official gene
+00003560: 2073 796d 626f 6c20 6170 7072 6f76 6564   symbol approved
+00003570: 2062 7920 0a20 2020 2020 2020 2020 2068   by .          h
+00003580: 676e 630a 2020 2020 2020 2020 3a74 7970  gnc.        :typ
+00003590: 6520 6765 6e65 733a 204c 6973 745b 696e  e genes: List[in
+000035a0: 745d 0a20 2020 2020 2020 203a 7061 7261  t].        :para
+000035b0: 6d20 6b65 6570 5f6e 6f6e 5f73 7065 6369  m keep_non_speci
+000035c0: 6669 633a 2049 6620 5472 7565 2c20 7265  fic: If True, re
+000035d0: 7375 6c74 7320 6672 6f6d 2070 726f 6265  sults from probe
+000035e0: 7365 7473 2074 6861 7420 6172 6520 6e6f  sets that are no
+000035f0: 740a 2020 2020 2020 2020 2020 7370 6563  t.          spec
+00003600: 6966 6963 2074 6f20 7468 6520 6765 6e65  ific to the gene
+00003610: 2077 696c 6c20 616c 736f 2062 6520 7265   will also be re
+00003620: 7475 726e 6564 2e2c 2064 6566 6175 6c74  turned., default
+00003630: 7320 746f 2046 616c 7365 0a20 2020 2020  s to False.     
+00003640: 2020 203a 7479 7065 206b 6565 705f 6e6f     :type keep_no
+00003650: 6e5f 7370 6563 6966 6963 3a20 626f 6f6c  n_specific: bool
+00003660: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
+00003670: 2020 203a 7061 7261 6d20 636f 6e73 6f6c     :param consol
+00003680: 6964 6174 653a 2041 6e20 6f70 7469 6f6e  idate: An option
+00003690: 2066 6f72 2067 656e 6520 6578 7072 6573   for gene expres
+000036a0: 7369 6f6e 206c 6576 656c 2063 6f6e 736f  sion level conso
+000036b0: 6c69 6461 7469 6f6e 2e20 0a20 2020 2020  lidation. .     
+000036c0: 2020 2020 2049 6620 656d 7074 792c 2077       If empty, w
+000036d0: 696c 6c20 7265 7475 726e 2065 7665 7279  ill return every
+000036e0: 2070 726f 6265 2066 6f72 2074 6865 2067   probe for the g
+000036f0: 656e 6573 2e20 2270 6963 6b6d 6178 2220  enes. "pickmax" 
+00003700: 746f 2070 6963 6b20 7468 650a 2020 2020  to pick the.    
+00003710: 2020 2020 2020 7072 6f62 6520 7769 7468        probe with
+00003720: 2074 6865 2068 6967 6865 7374 2065 7870   the highest exp
+00003730: 7265 7373 696f 6e2c 2022 7069 636b 7661  ression, "pickva
+00003740: 7222 2074 6f20 7069 636b 2074 6865 2070  r" to pick the p
+00003750: 726f 7665 2077 6974 6820 7468 6520 0a20  rove with the . 
+00003760: 2020 2020 2020 2020 2068 6967 6865 7374           highest
+00003770: 2076 6172 6961 6e63 6520 616e 6420 2261   variance and "a
+00003780: 7665 7261 6765 2220 666f 7220 7265 7475  verage" for retu
+00003790: 726e 696e 6720 7468 6520 6176 6572 6167  rning the averag
+000037a0: 6520 6578 7072 6573 7369 6f6e 2c20 0a20  e expression, . 
+000037b0: 2020 2020 2020 2020 2064 6566 6175 6c74           default
+000037c0: 7320 746f 204e 6f6e 650a 2020 2020 2020  s to None.      
+000037d0: 2020 3a74 7970 6520 636f 6e73 6f6c 6964    :type consolid
+000037e0: 6174 653a 2054 5950 452c 206f 7074 696f  ate: TYPE, optio
+000037f0: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
+00003800: 616d 202a 2a6b 7761 7267 733a 2041 6464  am **kwargs: Add
+00003810: 6974 696f 6e61 6c20 6172 6775 6d65 6e74  itional argument
+00003820: 7320 746f 2070 6173 7320 746f 2072 6177  s to pass to raw
+00003830: 2e67 6574 5f64 6174 6173 6574 5f65 7870  .get_dataset_exp
+00003840: 7265 7373 696f 6e5f 666f 725f 6765 6e65  ression_for_gene
+00003850: 730a 2020 2020 2020 2020 3a72 6574 7572  s.        :retur
+00003860: 6e3a 2041 2064 6963 7420 6f66 2044 6174  n: A dict of Dat
+00003870: 6146 7261 6d65 7320 6b65 7965 6420 746f  aFrames keyed to
+00003880: 2064 6174 6173 6574 2069 6473 0a20 2020   dataset ids.   
+00003890: 2020 2020 203a 7274 7970 653a 2064 6963       :rtype: dic
+000038a0: 745b 696e 743a 4461 7461 4672 616d 655d  t[int:DataFrame]
+000038b0: 0a0a 2020 2020 2020 2020 2222 220a 2020  ..        """.  
+000038c0: 2020 2020 2020 0a20 2020 2020 2020 206b        .        k
+000038d0: 7761 7267 7320 3d20 7673 2e72 656d 6f76  wargs = vs.remov
+000038e0: 655f 6e6f 6e65 7328 0a20 2020 2020 2020  e_nones(.       
+000038f0: 2020 2020 206b 6565 705f 6e6f 6e5f 7370       keep_non_sp
+00003900: 6563 6966 6963 203d 206b 6565 705f 6e6f  ecific = keep_no
+00003910: 6e5f 7370 6563 6966 6963 2c0a 2020 2020  n_specific,.    
+00003920: 2020 2020 2020 2020 636f 6e73 6f6c 6964          consolid
+00003930: 6174 6520 3d20 636f 6e73 6f6c 6964 6174  ate = consolidat
+00003940: 652c 0a20 2020 2020 2020 2020 2020 202a  e,.            *
+00003950: 2a6b 7761 7267 7329 0a20 2020 2020 2020  *kwargs).       
+00003960: 200a 2020 2020 2020 2020 7265 7370 6f6e   .        respon
+00003970: 7365 203d 2073 656c 662e 7261 772e 6765  se = self.raw.ge
+00003980: 745f 6461 7461 7365 745f 6578 7072 6573  t_dataset_expres
+00003990: 7369 6f6e 5f66 6f72 5f67 656e 6573 2864  sion_for_genes(d
+000039a0: 6174 6173 6574 732c 2067 656e 6573 2c20  atasets, genes, 
+000039b0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+000039c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000039d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000039e0: 2020 2020 2020 2020 2020 2020 2020 2a2a                **
+000039f0: 6b77 6172 6773 290a 2020 2020 2020 2020  kwargs).        
+00003a00: 6466 203d 2070 732e 7072 6f63 6573 735f  df = ps.process_
+00003a10: 6461 7461 7365 745f 6765 6e65 5f65 7870  dataset_gene_exp
+00003a20: 7265 7373 696f 6e28 7265 7370 6f6e 7365  ression(response
+00003a30: 2e64 6174 612c 7365 6c66 290a 2020 2020  .data,self).    
+00003a40: 2020 2020 0a20 2020 2020 2020 2072 6574      .        ret
+00003a50: 7572 6e20 6466 0a20 2020 2020 2020 200a  urn df.        .
+00003a60: 2020 2020 2320 6461 7461 7365 7473 2f7b      # datasets/{
+00003a70: 6461 7461 7365 7473 7d2f 6578 7072 6573  datasets}/expres
+00003a80: 7369 6f6e 732f 7063 6120 2d2d 2d2d 2d0a  sions/pca -----.
+00003a90: 2020 2020 2320 756e 696d 706c 656d 656e      # unimplemen
+00003aa0: 7465 640a 2020 2020 0a20 2020 200a 2020  ted.    .    .  
+00003ab0: 2020 2320 6461 7461 7365 7473 2f7b 6461    # datasets/{da
+00003ac0: 7461 7365 747d 2f70 6c61 7466 6f72 6d73  taset}/platforms
+00003ad0: 202d 2d2d 2d2d 2d0a 2020 2020 6465 6620   ------.    def 
+00003ae0: 6765 745f 6461 7461 7365 745f 706c 6174  get_dataset_plat
+00003af0: 666f 726d 7328 7365 6c66 2c20 6461 7461  forms(self, data
+00003b00: 7365 743a 7374 727c 696e 742c 202a 2a6b  set:str|int, **k
+00003b10: 7761 7267 7329 2d3e 4461 7461 4672 616d  wargs)->DataFram
+00003b20: 653a 0a20 2020 2020 2020 2022 2222 0a20  e:.        """. 
+00003b30: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+00003b40: 3a70 6172 616d 2064 6174 6173 6574 3a20  :param dataset: 
+00003b50: 4120 6e75 6d65 7269 6361 6c20 6461 7461  A numerical data
+00003b60: 7365 7420 6964 656e 7469 6669 6572 206f  set identifier o
+00003b70: 7220 6120 6461 7461 7365 7420 7368 6f72  r a dataset shor
+00003b80: 7420 6e61 6d65 0a20 2020 2020 2020 203a  t name.        :
+00003b90: 7479 7065 2064 6174 6173 6574 3a20 7374  type dataset: st
+00003ba0: 727c 696e 740a 2020 2020 2020 2020 3a70  r|int.        :p
+00003bb0: 6172 616d 202a 2a6b 7761 7267 733a 2041  aram **kwargs: A
+00003bc0: 6464 6974 696f 6e61 6c20 6172 6775 6d65  dditional argume
+00003bd0: 6e74 7320 746f 2070 6173 7320 746f 2072  nts to pass to r
+00003be0: 6177 2e67 6574 5f64 6174 6173 6574 5f70  aw.get_dataset_p
+00003bf0: 6c61 7466 6f72 6d73 0a20 2020 2020 2020  latforms.       
+00003c00: 203a 7265 7475 726e 3a20 4120 4461 7461   :return: A Data
+00003c10: 4672 616d 6520 7769 7468 2069 6e66 6f72  Frame with infor
+00003c20: 6d61 7469 6f6e 2061 626f 7574 2074 6865  mation about the
+00003c30: 2070 6c61 7466 6f72 6d73 2e0a 2020 2020   platforms..    
+00003c40: 2020 2020 2020 5468 6520 6669 656c 6473        The fields
+00003c50: 206f 6620 7468 6520 4461 7461 4672 616d   of the DataFram
+00003c60: 6520 6172 653a 0a20 2020 2020 2020 2020  e are:.         
+00003c70: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+00003c80: 2020 2020 2d20 706c 6174 666f 726d 5f49      - platform_I
+00003c90: 443a 2049 6420 6e75 6d62 6572 206f 6620  D: Id number of 
+00003ca0: 7468 6520 706c 6174 666f 726d 2067 6976  the platform giv
+00003cb0: 656e 2062 7920 4765 6d6d 610a 2020 2020  en by Gemma.    
+00003cc0: 2020 2020 2020 2020 2d20 706c 6174 666f          - platfo
+00003cd0: 726d 5f74 7970 653a 2054 7970 6520 6f66  rm_type: Type of
+00003ce0: 2074 6865 2070 6c61 7466 6f72 6d2e 0a20   the platform.. 
+00003cf0: 2020 2020 2020 2020 2020 202d 2070 6c61             - pla
+00003d00: 7466 6f72 6d5f 6465 7363 7269 7074 696f  tform_descriptio
+00003d10: 6e3a 2046 7265 6520 7465 7874 2066 6965  n: Free text fie
+00003d20: 6c64 2064 6573 6372 6962 696e 6720 7468  ld describing th
+00003d30: 6520 706c 6174 666f 726d 2e0a 2020 2020  e platform..    
+00003d40: 2020 2020 2020 2020 2d20 706c 6174 666f          - platfo
+00003d50: 726d 5f74 726f 7562 6c65 643a 2057 6865  rm_troubled: Whe
+00003d60: 7468 6572 2074 6865 2070 6c61 7466 6f72  ther the platfor
+00003d70: 6d20 6973 206d 6172 6b65 6420 6173 2074  m is marked as t
+00003d80: 726f 7562 6c65 6420 6279 2061 2047 656d  roubled by a Gem
+00003d90: 6d61 2063 7572 6174 6f72 2e0a 2020 2020  ma curator..    
+00003da0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
+00003db0: 6e61 6d65 3a20 4e61 6d65 206f 6620 7468  name: Name of th
+00003dc0: 6520 7370 6563 6965 730a 2020 2020 2020  e species.      
+00003dd0: 2020 2020 2020 2d20 7461 786f 6e5f 7363        - taxon_sc
+00003de0: 6965 6e74 6966 6963 3a20 5363 6965 6e74  ientific: Scient
+00003df0: 6966 6963 206e 616d 6520 666f 7220 7468  ific name for th
+00003e00: 6520 7461 786f 6e0a 2020 2020 2020 2020  e taxon.        
+00003e10: 2020 2020 2d20 7461 786f 6e5f 4944 3a20      - taxon_ID: 
+00003e20: 496e 7465 726e 616c 2069 6465 6e74 6966  Internal identif
+00003e30: 6965 7220 6769 7665 6e20 746f 2074 6865  ier given to the
+00003e40: 2073 7065 6369 6573 2062 7920 4765 6d6d   species by Gemm
+00003e50: 610a 2020 2020 2020 2020 2020 2020 2d20  a.            - 
+00003e60: 7461 786f 6e5f 4e43 4249 3a20 4e43 4249  taxon_NCBI: NCBI
+00003e70: 2049 4420 6f66 2074 6865 2074 6178 6f6e   ID of the taxon
+00003e80: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
+00003e90: 6178 6f6e 5f64 6174 6162 6173 655f 6e61  axon_database_na
+00003ea0: 6d65 3a20 556e 6465 726c 7969 6e67 2064  me: Underlying d
+00003eb0: 6174 6162 6173 6520 7573 6564 2069 6e20  atabase used in 
+00003ec0: 4765 6d6d 6120 666f 7220 7468 6520 7461  Gemma for the ta
+00003ed0: 786f 6e0a 2020 2020 2020 2020 2020 2020  xon.            
+00003ee0: 2d20 7461 786f 6e5f 6461 7461 6261 7365  - taxon_database
+00003ef0: 5f49 443a 2049 4420 6f66 2074 6865 2075  _ID: ID of the u
+00003f00: 6e64 6572 6c79 696e 6720 6461 7461 6261  nderlying databa
+00003f10: 7365 2075 7365 6420 696e 2047 656d 6d61  se used in Gemma
+00003f20: 2066 6f72 2074 6865 2074 6178 6f6e 2020   for the taxon  
+00003f30: 0a20 2020 2020 2020 203a 7274 7970 653a  .        :rtype:
+00003f40: 2044 6174 6146 7261 6d65 0a0a 2020 2020   DataFrame..    
+00003f50: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
+00003f60: 7265 7370 6f6e 7365 203d 2073 656c 662e  response = self.
+00003f70: 7261 772e 6765 745f 6461 7461 7365 745f  raw.get_dataset_
+00003f80: 706c 6174 666f 726d 7328 6461 7461 7365  platforms(datase
+00003f90: 742c 202a 2a6b 7761 7267 7329 0a20 2020  t, **kwargs).   
+00003fa0: 2020 2020 2064 6620 3d20 7073 2e70 726f       df = ps.pro
+00003fb0: 6365 7373 5f70 6c61 7466 6f72 6d73 2872  cess_platforms(r
+00003fc0: 6573 706f 6e73 652e 6461 7461 290a 2020  esponse.data).  
+00003fd0: 2020 2020 2020 0a20 2020 2020 2020 2072        .        r
+00003fe0: 6574 7572 6e28 6466 290a 2020 2020 0a20  eturn(df).    . 
+00003ff0: 2020 200a 2020 2020 2320 6461 7461 7365     .    # datase
+00004000: 7473 2f7b 6461 7461 7365 747d 2f64 6174  ts/{dataset}/dat
+00004010: 612f 7072 6f63 6573 7365 6420 2d2d 2d2d  a/processed ----
+00004020: 2d2d 0a20 2020 200a 2020 2020 6465 6620  --.    .    def 
+00004030: 6765 745f 6461 7461 7365 745f 7072 6f63  get_dataset_proc
+00004040: 6573 7365 645f 6578 7072 6573 7369 6f6e  essed_expression
+00004050: 2873 656c 662c 6461 7461 7365 743a 7374  (self,dataset:st
+00004060: 727c 696e 742c 2a2a 6b77 6172 6773 292d  r|int,**kwargs)-
+00004070: 3e44 6174 6146 7261 6d65 3a0a 2020 2020  >DataFrame:.    
+00004080: 2020 2020 2222 2252 6574 7269 6576 6520      """Retrieve 
+00004090: 7072 6f63 6573 7365 6420 6578 7072 6573  processed expres
+000040a0: 7369 6f6e 2064 6174 6120 6f66 2061 2064  sion data of a d
+000040b0: 6174 6173 6574 0a0a 2020 2020 2020 2020  ataset..        
+000040c0: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+000040d0: 6461 7461 7365 743a 2020 6e75 6d65 7269  dataset:  numeri
+000040e0: 6361 6c20 6461 7461 7365 7420 6964 656e  cal dataset iden
+000040f0: 7469 6669 6572 206f 7220 6120 6461 7461  tifier or a data
+00004100: 7365 7420 7368 6f72 7420 6e61 6d65 0a20  set short name. 
+00004110: 2020 2020 2020 203a 7479 7065 2064 6174         :type dat
+00004120: 6173 6574 3a20 7374 727c 696e 740a 2020  aset: str|int.  
+00004130: 2020 2020 2020 3a70 6172 616d 202a 2a6b        :param **k
+00004140: 7761 7267 733a 2041 6464 6974 696f 6e61  wargs: Additiona
+00004150: 6c20 6172 6775 6d65 6e74 7320 746f 2070  l arguments to p
+00004160: 6173 7320 746f 2072 6177 2e67 6574 5f64  ass to raw.get_d
+00004170: 6174 6173 6574 5f70 726f 6365 7373 6564  ataset_processed
+00004180: 5f65 7870 7265 7373 696f 6e0a 2020 2020  _expression.    
+00004190: 2020 2020 3a72 6574 7572 6e3a 2041 2044      :return: A D
+000041a0: 6174 6146 7261 6d65 206f 6620 7468 6520  ataFrame of the 
+000041b0: 6578 7072 6573 7369 6f6e 206d 6174 7269  expression matri
+000041c0: 7820 666f 7220 7468 6520 7175 6572 6965  x for the querie
+000041d0: 6420 6461 7461 7365 740a 2020 2020 2020  d dataset.      
+000041e0: 2020 3a72 7479 7065 3a20 4461 7461 4672    :rtype: DataFr
+000041f0: 616d 650a 0a20 2020 2020 2020 2022 2222  ame..        """
+00004200: 0a20 2020 2020 2020 2072 6573 706f 6e73  .        respons
+00004210: 6520 3d20 7365 6c66 2e72 6177 2e67 6574  e = self.raw.get
+00004220: 5f64 6174 6173 6574 5f70 726f 6365 7373  _dataset_process
+00004230: 6564 5f65 7870 7265 7373 696f 6e28 6461  ed_expression(da
+00004240: 7461 7365 742c 202a 2a6b 7761 7267 7329  taset, **kwargs)
+00004250: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
+00004260: 2020 6466 203d 2070 732e 7072 6f63 6573    df = ps.proces
+00004270: 735f 6578 7072 6573 7369 6f6e 2872 6573  s_expression(res
+00004280: 706f 6e73 652c 6461 7461 7365 742c 7365  ponse,dataset,se
+00004290: 6c66 290a 2020 2020 2020 2020 0a20 2020  lf).        .   
+000042a0: 2020 2020 2072 6574 7572 6e20 6466 0a20       return df. 
+000042b0: 2020 200a 2020 2020 2320 6461 7461 7365     .    # datase
+000042c0: 7473 2f7b 6461 7461 7365 747d 2f71 7561  ts/{dataset}/qua
+000042d0: 6e74 6974 6174 696f 6e54 7970 6573 2067  ntitationTypes g
+000042e0: 6574 5f64 6174 6173 6574 5f71 7561 6e74  et_dataset_quant
+000042f0: 6974 6174 696f 6e5f 7479 7065 7320 2d2d  itation_types --
+00004300: 2d2d 2d2d 2d2d 2d2d 0a20 2020 200a 2020  --------.    .  
+00004310: 2020 6465 6620 6765 745f 6461 7461 7365    def get_datase
+00004320: 745f 7175 616e 7469 7461 7469 6f6e 5f74  t_quantitation_t
+00004330: 7970 6573 2873 656c 662c 6461 7461 7365  ypes(self,datase
+00004340: 743a 696e 747c 7374 722c 2a2a 6b77 6172  t:int|str,**kwar
+00004350: 6773 292d 3e44 6174 6146 7261 6d65 3a0a  gs)->DataFrame:.
+00004360: 2020 2020 2020 2020 2222 2252 6574 7269          """Retri
+00004370: 6576 6520 7175 616e 7469 7461 7469 6f6e  eve quantitation
+00004380: 2074 7970 6573 206f 6620 6120 6461 7461   types of a data
+00004390: 7365 740a 0a20 2020 2020 2020 200a 2020  set..        .  
+000043a0: 2020 2020 2020 3a70 6172 616d 2064 6174        :param dat
+000043b0: 6173 6574 3a20 0941 206e 756d 6572 6963  aset: .A numeric
+000043c0: 616c 2064 6174 6173 6574 2069 6465 6e74  al dataset ident
+000043d0: 6966 6965 7220 6f72 2061 2064 6174 6173  ifier or a datas
+000043e0: 6574 2073 686f 7274 206e 616d 650a 2020  et short name.  
+000043f0: 2020 2020 2020 3a74 7970 6520 6461 7461        :type data
+00004400: 7365 743a 2069 6e74 7c73 7472 0a20 2020  set: int|str.   
+00004410: 2020 2020 203a 7061 7261 6d20 2a2a 6b77       :param **kw
+00004420: 6172 6773 3a20 4164 6469 7469 6f6e 616c  args: Additional
+00004430: 2061 7267 756d 656e 7473 2074 6f20 7061   arguments to pa
+00004440: 7373 2074 6f20 7261 772e 6765 745f 6461  ss to raw.get_da
+00004450: 7461 7365 745f 7175 616e 7469 7461 7469  taset_quantitati
+00004460: 6f6e 5f74 7970 6573 0a20 2020 2020 2020  on_types.       
+00004470: 203a 7265 7475 726e 3a20 4120 4461 7461   :return: A Data
+00004480: 4672 616d 6520 636f 6e74 6169 6e69 6e67  Frame containing
+00004490: 2074 6865 2071 7561 6e74 6974 6174 696f   the quantitatio
+000044a0: 6e20 7479 7065 730a 2020 2020 2020 2020  n types.        
+000044b0: 0a20 2020 2020 2020 2020 2054 6865 2066  .          The f
+000044c0: 6965 6c64 7320 6f66 2074 6865 206f 7574  ields of the out
+000044d0: 7075 7420 4461 7461 4672 616d 6520 6172  put DataFrame ar
+000044e0: 653a 0a20 2020 2020 2020 2020 2020 202d  e:.            -
+000044f0: 2069 643a 2049 6620 6f66 2074 6865 2071   id: If of the q
+00004500: 7561 6e74 6974 6174 696f 6e20 7479 7065  uantitation type
+00004510: 2e20 416e 7920 7261 7720 7175 616e 7469  . Any raw quanti
+00004520: 7461 7469 6f6e 2074 7970 6520 6361 6e20  tation type can 
+00004530: 6265 0a20 2020 2020 2020 2020 2020 2020  be.             
+00004540: 2020 6279 2067 6574 5f64 6174 6173 6574    by get_dataset
+00004550: 5f72 6177 5f65 7870 7265 7373 696f 6e20  _raw_expression 
+00004560: 6675 6e63 7469 6f6e 2075 7369 6e67 2074  function using t
+00004570: 6869 7320 6964 2e0a 2020 2020 2020 2020  his id..        
+00004580: 2020 2020 2d20 6e61 6d65 3a20 4e61 6d65      - name: Name
+00004590: 206f 6620 7468 6520 7175 616e 7469 7461   of the quantita
+000045a0: 7469 6f6e 2074 7970 650a 2020 2020 2020  tion type.      
+000045b0: 2020 2020 2020 2d20 6465 7363 7269 7074        - descript
+000045c0: 696f 6e3a 2044 6573 6372 6970 7469 6f6e  ion: Description
+000045d0: 206f 6620 7468 6520 7175 616e 7469 7461   of the quantita
+000045e0: 7469 6f6e 2074 7970 650a 2020 2020 2020  tion type.      
+000045f0: 2020 2020 2020 2d20 7479 7065 3a20 5479        - type: Ty
+00004600: 7065 206f 6620 7468 6520 7175 616e 7469  pe of the quanti
+00004610: 7461 7469 6f6e 2074 7970 652e 2045 6974  tation type. Eit
+00004620: 6865 7220 7261 7720 6f72 2070 726f 6365  her raw or proce
+00004630: 7373 6564 2e20 0a20 2020 2020 2020 2020  ssed. .         
+00004640: 2020 2020 2045 6163 6820 6461 7461 7365       Each datase
+00004650: 7420 7769 6c6c 2068 6176 6520 6f6e 6520  t will have one 
+00004660: 7072 6f63 6573 7365 6420 7175 616e 7469  processed quanti
+00004670: 7461 7469 6f6e 2074 7970 6520 7768 6963  tation type whic
+00004680: 6820 6973 2074 6865 200a 2020 2020 2020  h is the .      
+00004690: 2020 2020 2020 2020 6461 7461 2072 6574          data ret
+000046a0: 7572 6e65 6420 7573 696e 6720 6765 745f  urned using get_
+000046b0: 6461 7461 7365 745f 7072 6f63 6573 7365  dataset_processe
+000046c0: 645f 6578 7072 6573 7369 6f6e 0a20 2020  d_expression.   
+000046d0: 2020 2020 2020 2020 202d 2072 6174 696f           - ratio
+000046e0: 3a20 5768 6574 6865 7220 6f72 206e 6f74  : Whether or not
+000046f0: 2074 6865 2071 7561 6e69 7461 7469 6f6e   the quanitation
+00004700: 2074 7970 6520 6973 2061 2072 6174 696f   type is a ratio
+00004710: 206f 6620 6d75 6c74 6970 6c65 0a20 2020   of multiple.   
+00004720: 2020 2020 2020 2020 2020 2071 7561 6e74             quant
+00004730: 6974 6174 696f 6e20 7479 7065 732e 2054  itation types. T
+00004740: 7970 6963 616c 6c79 2054 5255 4520 666f  ypically TRUE fo
+00004750: 7220 7072 6f63 6573 7365 6420 5457 4f43  r processed TWOC
+00004760: 4f4c 4f52 2071 7561 6e74 6974 6174 696f  OLOR quantitatio
+00004770: 6e20 7479 7065 2e0a 2020 2020 2020 2020  n type..        
+00004780: 2020 2020 2d20 7072 6566 6572 7265 643a      - preferred:
+00004790: 2054 6865 2070 7265 6665 7272 6564 2072   The preferred r
+000047a0: 6177 2071 7561 6e74 6974 6174 696f 6e20  aw quantitation 
+000047b0: 7479 7065 2e20 5468 6973 2076 6572 7369  type. This versi
+000047c0: 6f6e 2069 7320 0a20 2020 2020 2020 2020  on is .         
+000047d0: 2020 2020 2075 7365 6420 696e 2067 656e       used in gen
+000047e0: 6572 6174 696f 6e20 6f66 2074 6865 2070  eration of the p
+000047f0: 726f 6365 7373 6564 2064 6174 6120 7769  rocessed data wi
+00004800: 7468 696e 2067 656d 6d61 2e0a 2020 2020  thin gemma..    
+00004810: 2020 2020 2020 2020 2d20 7265 636f 6d70          - recomp
+00004820: 7574 6564 3a20 4966 2054 5255 4520 7468  uted: If TRUE th
+00004830: 6973 2071 7561 6e74 6974 6174 696f 6e20  is quantitation 
+00004840: 7479 7065 2069 7320 6765 6e65 7261 7465  type is generate
+00004850: 6420 6279 0a20 2020 2020 2020 2020 2020  d by.           
+00004860: 2020 2072 6563 6f6d 7075 7469 6e67 2072     recomputing r
+00004870: 6177 2064 6174 6120 6669 6c65 7320 4765  aw data files Ge
+00004880: 6d6d 6120 6861 6420 6163 6365 7373 2074  mma had access t
+00004890: 6f0a 2020 2020 2020 2020 3a72 7479 7065  o.        :rtype
+000048a0: 3a20 4461 7461 4672 616d 650a 0a20 2020  : DataFrame..   
+000048b0: 2020 2020 2022 2222 0a20 2020 2020 2020       """.       
+000048c0: 200a 2020 2020 2020 2020 7265 7370 6f6e   .        respon
+000048d0: 7365 203d 2073 656c 662e 7261 772e 6765  se = self.raw.ge
+000048e0: 745f 6461 7461 7365 745f 7175 616e 7469  t_dataset_quanti
+000048f0: 7461 7469 6f6e 5f74 7970 6573 2864 6174  tation_types(dat
+00004900: 6173 6574 2c20 2a2a 6b77 6172 6773 290a  aset, **kwargs).
+00004910: 2020 2020 2020 2020 6466 203d 2070 732e          df = ps.
+00004920: 7072 6f63 6573 735f 5175 616e 7469 7461  process_Quantita
+00004930: 7469 6f6e 5479 7065 5661 6c75 654f 626a  tionTypeValueObj
+00004940: 6563 7428 7265 7370 6f6e 7365 2e64 6174  ect(response.dat
+00004950: 6129 0a20 2020 2020 2020 200a 2020 2020  a).        .    
+00004960: 2020 2020 0a20 2020 2020 2020 2072 6574      .        ret
+00004970: 7572 6e20 6466 0a0a 2020 2020 2320 6461  urn df..    # da
+00004980: 7461 7365 7473 2f7b 6461 7461 7365 747d  tasets/{dataset}
+00004990: 2f64 6174 612f 7261 772c 2067 6574 5f64  /data/raw, get_d
+000049a0: 6174 6173 6574 5f72 6177 5f65 7870 7265  ataset_raw_expre
+000049b0: 7373 696f 6e20 2d2d 2d2d 2d2d 2d2d 2d0a  ssion ---------.
+000049c0: 2020 2020 6465 6620 6765 745f 6461 7461      def get_data
+000049d0: 7365 745f 7261 775f 6578 7072 6573 7369  set_raw_expressi
+000049e0: 6f6e 2873 656c 662c 6461 7461 7365 743a  on(self,dataset:
+000049f0: 696e 747c 7374 722c 0a20 2020 2020 2020  int|str,.       
+00004a00: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00004a10: 2020 2020 2020 2020 2020 2020 7175 616e              quan
+00004a20: 7469 7461 7469 6f6e 5f74 7970 653a 5b69  titation_type:[i
+00004a30: 6e74 5d2c 2a2a 6b77 6172 6773 292d 3e44  nt],**kwargs)->D
+00004a40: 6174 6146 7261 6d65 3a0a 2020 2020 2020  ataFrame:.      
+00004a50: 2020 2222 220a 2020 2020 2020 2020 0a20    """.        . 
+00004a60: 2020 2020 2020 203a 7061 7261 6d20 6461         :param da
+00004a70: 7461 7365 743a 2041 206e 756d 6572 6963  taset: A numeric
+00004a80: 616c 2064 6174 6173 6574 2069 6465 6e74  al dataset ident
+00004a90: 6966 6965 7220 6f72 2061 2064 6174 6173  ifier or a datas
+00004aa0: 6574 2073 686f 7274 206e 616d 650a 2020  et short name.  
+00004ab0: 2020 2020 2020 3a74 7970 6520 6461 7461        :type data
+00004ac0: 7365 743a 2069 6e74 7c73 7472 0a20 2020  set: int|str.   
+00004ad0: 2020 2020 203a 7061 7261 6d20 7175 616e       :param quan
+00004ae0: 7469 7461 7469 6f6e 5f74 7970 653a 2051  titation_type: Q
+00004af0: 7561 6e74 6974 6174 696f 6e20 7479 7065  uantitation type
+00004b00: 2069 642e 2054 6865 7365 2063 616e 2062   id. These can b
+00004b10: 6520 6163 7175 6972 6564 200a 2020 2020  e acquired .    
+00004b20: 2020 2020 2020 7573 696e 6720 6765 745f        using get_
+00004b30: 6461 7461 7365 745f 7175 616e 7469 7461  dataset_quantita
+00004b40: 7469 6f6e 5f74 7970 6573 2066 756e 6374  tion_types funct
+00004b50: 696f 6e2e 2054 6869 7320 656e 6470 6f69  ion. This endpoi
+00004b60: 6e74 2063 616e 206f 6e6c 7920 0a20 2020  nt can only .   
+00004b70: 2020 2020 2020 2072 6574 7572 6e20 6e6f         return no
+00004b80: 6e2d 7072 6f63 6573 7365 6420 7175 616e  n-processed quan
+00004b90: 7469 7461 7469 6f6e 2074 7970 6573 2e0a  titation types..
+00004ba0: 2020 2020 2020 2020 3a74 7970 6520 7175          :type qu
+00004bb0: 616e 7469 7461 7469 6f6e 5f74 7970 653a  antitation_type:
+00004bc0: 205b 696e 745d 0a20 2020 2020 2020 203a   [int].        :
+00004bd0: 7061 7261 6d20 2a2a 6b77 6172 6773 3a20  param **kwargs: 
+00004be0: 4164 6469 7469 6f6e 616c 2061 7267 756d  Additional argum
+00004bf0: 656e 7473 2074 6f20 7061 7373 2074 6f20  ents to pass to 
+00004c00: 7261 772e 6765 745f 6461 7461 7365 745f  raw.get_dataset_
+00004c10: 7261 775f 6578 7072 6573 7369 6f6e 0a20  raw_expression. 
+00004c20: 2020 2020 2020 203a 7265 7475 726e 3a20         :return: 
+00004c30: 4120 4461 7461 4672 616d 6520 6f66 2074  A DataFrame of t
+00004c40: 6865 2065 7870 7265 7373 696f 6e20 6d61  he expression ma
+00004c50: 7472 6978 2066 6f72 2074 6865 2071 7565  trix for the que
+00004c60: 7269 6564 2064 6174 6173 6574 0a20 2020  ried dataset.   
+00004c70: 2020 2020 203a 7274 7970 653a 2044 6174       :rtype: Dat
+00004c80: 6146 7261 6d65 0a0a 2020 2020 2020 2020  aFrame..        
+00004c90: 2222 220a 2020 2020 2020 2020 0a20 2020  """.        .   
+00004ca0: 2020 2020 206b 7761 7267 7320 3d20 7673       kwargs = vs
+00004cb0: 2e72 656d 6f76 655f 6e6f 6e65 7328 0a20  .remove_nones(. 
+00004cc0: 2020 2020 2020 2020 2020 2071 7561 6e74             quant
+00004cd0: 6974 6174 696f 6e5f 7479 7065 203d 2071  itation_type = q
+00004ce0: 7561 6e74 6974 6174 696f 6e5f 7479 7065  uantitation_type
+00004cf0: 2c0a 2020 2020 2020 2020 2020 2020 2a2a  ,.            **
+00004d00: 6b77 6172 6773 290a 2020 2020 2020 2020  kwargs).        
+00004d10: 0a20 2020 2020 2020 2072 6573 706f 6e73  .        respons
+00004d20: 6520 3d20 7365 6c66 2e72 6177 2e67 6574  e = self.raw.get
+00004d30: 5f64 6174 6173 6574 5f72 6177 5f65 7870  _dataset_raw_exp
+00004d40: 7265 7373 696f 6e28 6461 7461 7365 742c  ression(dataset,
+00004d50: 202a 2a6b 7761 7267 7329 0a20 2020 2020   **kwargs).     
+00004d60: 2020 200a 2020 2020 2020 2020 6466 203d     .        df =
+00004d70: 2070 732e 7072 6f63 6573 735f 6578 7072   ps.process_expr
+00004d80: 6573 7369 6f6e 2872 6573 706f 6e73 652c  ession(response,
+00004d90: 6461 7461 7365 742c 7365 6c66 290a 2020  dataset,self).  
+00004da0: 2020 2020 2020 0a20 2020 2020 2020 2072        .        r
+00004db0: 6574 7572 6e20 6466 0a20 2020 200a 2020  eturn df.    .  
+00004dc0: 2020 0a20 2020 2023 2064 6174 6173 6574    .    # dataset
+00004dd0: 732f 7b64 6174 6173 6574 7d2f 7361 6d70  s/{dataset}/samp
+00004de0: 6c65 732c 2067 6574 5f64 6174 6173 6574  les, get_dataset
+00004df0: 5f73 616d 706c 6573 202d 2d2d 2d2d 2d2d  _samples -------
+00004e00: 2d0a 2020 2020 6465 6620 6765 745f 6461  -.    def get_da
+00004e10: 7461 7365 745f 7361 6d70 6c65 7328 7365  taset_samples(se
+00004e20: 6c66 2c20 6461 7461 7365 743a 696e 747c  lf, dataset:int|
+00004e30: 7374 722c 202a 2a6b 7761 7267 7329 2d3e  str, **kwargs)->
+00004e40: 4461 7461 4672 616d 653a 0a20 2020 2020  DataFrame:.     
+00004e50: 2020 2022 2222 0a20 2020 2020 2020 2052     """.        R
+00004e60: 6574 7269 6576 6520 7468 6520 7361 6d70  etrieve the samp
+00004e70: 6c65 7320 6f66 2061 2064 6174 6173 6574  les of a dataset
+00004e80: 0a0a 0a20 2020 2020 2020 203a 7061 7261  ...        :para
+00004e90: 6d20 6461 7461 7365 743a 2041 206e 756d  m dataset: A num
+00004ea0: 6572 6963 616c 2064 6174 6173 6574 2069  erical dataset i
+00004eb0: 6465 6e74 6966 6965 7220 6f72 2061 2064  dentifier or a d
+00004ec0: 6174 6173 6574 2073 686f 7274 206e 616d  ataset short nam
+00004ed0: 650a 2020 2020 2020 2020 3a74 7970 6520  e.        :type 
+00004ee0: 6461 7461 7365 743a 2069 6e74 7c73 7472  dataset: int|str
+00004ef0: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+00004f00: 2a2a 6b77 6172 6773 3a20 4164 6469 7469  **kwargs: Additi
+00004f10: 6f6e 616c 2061 7267 756d 656e 7473 2074  onal arguments t
+00004f20: 6f20 7061 7373 2074 6f20 7261 772e 6765  o pass to raw.ge
+00004f30: 745f 6461 7461 7365 745f 7361 6d70 6c65  t_dataset_sample
+00004f40: 730a 2020 2020 2020 2020 3a72 6574 7572  s.        :retur
+00004f50: 6e3a 2041 2044 6174 6146 7261 6d65 2077  n: A DataFrame w
+00004f60: 6974 6820 696e 666f 726d 6174 696f 6e20  ith information 
+00004f70: 6162 6f75 7420 7468 6520 7361 6d70 6c65  about the sample
+00004f80: 7320 6f66 2074 6865 2071 7565 7269 6564  s of the queried
+00004f90: 2064 6174 6173 6574 2e0a 2020 2020 2020   dataset..      
+00004fa0: 2020 0a20 2020 2020 2020 2020 2054 6865    .          The
+00004fb0: 2066 6965 6c64 7320 6f66 2074 6865 2044   fields of the D
+00004fc0: 6174 6146 7261 6d65 2061 7265 3a0a 2020  ataFrame are:.  
+00004fd0: 2020 2020 2020 2020 2020 2d20 7361 6d70            - samp
+00004fe0: 6c65 5f6e 616d 653a 2049 6e74 6572 6e61  le_name: Interna
+00004ff0: 6c20 6e61 6d65 2067 6976 656e 2074 6f20  l name given to 
+00005000: 7468 6520 7361 6d70 6c65 2e0a 2020 2020  the sample..    
+00005010: 2020 2020 2020 2020 2d20 7361 6d70 6c65          - sample
+00005020: 5f49 443a 2049 6e74 6572 6e61 6c20 4944  _ID: Internal ID
+00005030: 206f 6620 7468 6520 7361 6d70 6c65 0a20   of the sample. 
+00005040: 2020 2020 2020 2020 2020 202d 2073 616d             - sam
+00005050: 706c 655f 6465 7363 7269 7074 696f 6e3a  ple_description:
+00005060: 2046 7265 6520 7465 7874 2064 6573 6372   Free text descr
+00005070: 6970 7469 6f6e 206f 6620 7468 6520 7361  iption of the sa
+00005080: 6d70 6c65 0a20 2020 2020 2020 2020 2020  mple.           
+00005090: 202d 2073 616d 706c 655f 6f75 746c 6965   - sample_outlie
+000050a0: 723a 2057 6865 7468 6572 206f 7220 6e6f  r: Whether or no
+000050b0: 7420 7468 6520 7361 6d70 6c65 2069 7320  t the sample is 
+000050c0: 6d61 726b 6564 2061 7320 616e 206f 7574  marked as an out
+000050d0: 6c69 6572 0a20 2020 2020 2020 2020 2020  lier.           
+000050e0: 202d 2073 616d 706c 655f 6163 6365 7373   - sample_access
+000050f0: 696f 6e3a 2041 6363 6573 7369 6f6e 2049  ion: Accession I
+00005100: 4420 6f66 2074 6865 2073 616d 706c 6520  D of the sample 
+00005110: 696e 2069 7427 7320 6f72 6967 696e 616c  in it's original
+00005120: 2064 6174 6162 6173 650a 2020 2020 2020   database.      
+00005130: 2020 2020 2020 2d20 7361 6d70 6c65 5f64        - sample_d
+00005140: 6174 6162 6173 653a 2044 6174 6162 6173  atabase: Databas
+00005150: 6520 6f66 206f 7269 6769 6e20 666f 7220  e of origin for 
+00005160: 7468 6520 7361 6d70 6c65 0a20 2020 2020  the sample.     
+00005170: 2020 2020 2020 202d 2073 616d 706c 655f         - sample_
+00005180: 6368 6172 6163 7465 7269 7374 6963 733a  characteristics:
+00005190: 2043 6861 7261 6374 6572 6973 7469 6373   Characteristics
+000051a0: 206f 6620 7468 6520 7361 6d70 6c65 2e20   of the sample. 
+000051b0: 5468 6973 2066 6965 6c64 2069 7320 6120  This field is a 
+000051c0: 6461 7461 2074 6162 6c65 0a20 2020 2020  data table.     
+000051d0: 2020 2020 2020 202d 2073 616d 706c 655f         - sample_
+000051e0: 6661 6374 6f72 5f76 616c 7565 733a 2045  factor_values: E
+000051f0: 7870 6572 696d 656e 7461 6c20 6661 6374  xperimental fact
+00005200: 6f72 2076 616c 7565 7320 6f66 2074 6865  or values of the
+00005210: 2073 616d 706c 652e 2054 6869 7320 6669   sample. This fi
+00005220: 656c 6420 6973 2061 2064 6174 6120 7461  eld is a data ta
+00005230: 626c 650a 2020 2020 2020 2020 3a72 7479  ble.        :rty
+00005240: 7065 3a20 4461 7461 4672 616d 650a 0a20  pe: DataFrame.. 
+00005250: 2020 2020 2020 2022 2222 0a20 2020 2020         """.     
+00005260: 2020 2072 6573 706f 6e73 6520 3d20 7365     response = se
+00005270: 6c66 2e72 6177 2e67 6574 5f64 6174 6173  lf.raw.get_datas
+00005280: 6574 5f73 616d 706c 6573 2864 6174 6173  et_samples(datas
+00005290: 6574 2c20 2a2a 6b77 6172 6773 290a 2020  et, **kwargs).  
+000052a0: 2020 2020 2020 6466 203d 2070 732e 7072        df = ps.pr
+000052b0: 6f63 6573 735f 7361 6d70 6c65 7328 7265  ocess_samples(re
+000052c0: 7370 6f6e 7365 2e64 6174 6129 0a20 2020  sponse.data).   
+000052d0: 2020 2020 2072 6574 7572 6e20 6466 0a20       return df. 
+000052e0: 2020 2020 2020 200a 2020 2020 2320 6461         .    # da
+000052f0: 7461 7365 7473 2f7b 6461 7461 7365 747d  tasets/{dataset}
+00005300: 2f73 7664 202d 2d2d 200a 2020 2020 2320  /svd --- .    # 
+00005310: 6e6f 7420 696d 706c 656d 656e 7465 640a  not implemented.
+00005320: 2020 2020 0a20 2020 2023 2064 6174 6173      .    # datas
+00005330: 6574 732c 2067 6574 5f64 6174 6173 6574  ets, get_dataset
+00005340: 7320 2d2d 2d2d 2d2d 0a20 2020 2064 6566  s ------.    def
+00005350: 2067 6574 5f64 6174 6173 6574 7328 7365   get_datasets(se
+00005360: 6c66 2c71 7565 7279 3a4f 7074 696f 6e61  lf,query:Optiona
+00005370: 6c5b 7374 725d 203d 204e 6f6e 652c 200a  l[str] = None, .
+00005380: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00005390: 2020 2020 2066 696c 7465 723a 4f70 7469       filter:Opti
+000053a0: 6f6e 616c 5b73 7472 5d20 3d20 4e6f 6e65  onal[str] = None
+000053b0: 2c20 0a20 2020 2020 2020 2020 2020 2020  , .             
+000053c0: 2020 2020 2020 2020 7461 7861 3a4f 7074          taxa:Opt
+000053d0: 696f 6e61 6c5b 4c69 7374 5b73 7472 5d5d  ional[List[str]]
+000053e0: 203d 204e 6f6e 652c 200a 2020 2020 2020   = None, .      
+000053f0: 2020 2020 2020 2020 2020 2020 2020 2075                 u
+00005400: 7269 733a 4f70 7469 6f6e 616c 5b4c 6973  ris:Optional[Lis
+00005410: 745b 7374 725d 5d20 3d20 4e6f 6e65 2c0a  t[str]] = None,.
+00005420: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00005430: 2020 2020 206f 6666 7365 743a 696e 7420       offset:int 
+00005440: 3d20 302c 0a20 2020 2020 2020 2020 2020  = 0,.           
+00005450: 2020 2020 2020 2020 2020 6c69 6d69 743a            limit:
+00005460: 696e 7420 3d20 3230 2c0a 2020 2020 2020  int = 20,.      
+00005470: 2020 2020 2020 2020 2020 2020 2020 2073                 s
+00005480: 6f72 743a 7374 7220 3d20 222b 6964 222c  ort:str = "+id",
+00005490: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+000054a0: 2020 2020 2020 2a2a 6b77 6172 6773 292d        **kwargs)-
+000054b0: 3e44 6174 6146 7261 6d65 3a0a 2020 2020  >DataFrame:.    
+000054c0: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
+000054d0: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+000054e0: 7175 6572 793a 2054 6865 2073 6561 7263  query: The searc
+000054f0: 6820 7175 6572 792e 2045 6974 6865 7220  h query. Either 
+00005500: 706c 6169 6e20 7465 7874 2028 2774 7261  plain text ('tra
+00005510: 756d 6174 6963 2729 2c20 6f72 2061 6e20  umatic'), or an 
+00005520: 0a20 2020 2020 2020 2020 206f 6e74 6f6c  .          ontol
+00005530: 6f67 7920 7465 726d 2055 5249 2028 2768  ogy term URI ('h
+00005540: 7474 703a 2f2f 7075 726c 2e6f 626f 6c69  ttp://purl.oboli
+00005550: 6272 6172 792e 6f72 672f 6f62 6f2f 5542  brary.org/obo/UB
+00005560: 4552 4f4e 5f30 3030 3230 3438 2729 2e0a  ERON_0002048')..
+00005570: 2020 2020 2020 2020 2020 4461 7461 7365            Datase
+00005580: 7473 2074 6861 7420 636f 6e74 6169 6e20  ts that contain 
+00005590: 7468 6520 6769 7665 6e20 7374 7269 6e67  the given string
+000055a0: 2069 6e20 7468 6569 7220 7368 6f72 7420   in their short 
+000055b0: 6f66 2066 756c 6c20 6e61 6d65 2077 696c  of full name wil
+000055c0: 6c20 0a20 2020 2020 2020 2020 2061 6c73  l .          als
+000055d0: 6f20 6265 206d 6174 6368 6564 2e2c 2064  o be matched., d
+000055e0: 6566 6175 6c74 7320 746f 204e 6f6e 650a  efaults to None.
+000055f0: 2020 2020 2020 2020 3a74 7970 6520 7175          :type qu
+00005600: 6572 793a 204f 7074 696f 6e61 6c5b 7374  ery: Optional[st
+00005610: 725d 2c20 6f70 7469 6f6e 616c 0a20 2020  r], optional.   
+00005620: 2020 2020 203a 7061 7261 6d20 6669 6c74       :param filt
+00005630: 6572 3a20 4669 6c74 6572 2072 6573 756c  er: Filter resul
+00005640: 7473 2062 7920 6d61 7463 6869 6e67 2065  ts by matching e
+00005650: 7870 7265 7373 696f 6e2e 2055 7365 200a  xpression. Use .
+00005660: 2020 2020 2020 2020 2020 6669 6c74 6572            filter
+00005670: 5f70 726f 7065 7274 6965 7320 6675 6e63  _properties func
+00005680: 7469 6f6e 2074 6f20 6765 7420 6120 6c69  tion to get a li
+00005690: 7374 206f 6620 616c 6c20 6176 6169 6c61  st of all availa
+000056a0: 626c 6520 7061 7261 6d65 7465 7273 2e20  ble parameters. 
+000056b0: 0a20 2020 2020 2020 2020 2054 6865 7365  .          These
+000056c0: 2070 726f 7065 7274 6965 7320 6361 6e20   properties can 
+000056d0: 6265 2063 6f6d 6269 6e65 6420 7573 696e  be combined usin
+000056e0: 6720 2261 6e64 2220 226f 7222 2063 6c61  g "and" "or" cla
+000056f0: 7573 6573 2061 6e64 206d 6179 200a 2020  uses and may .  
+00005700: 2020 2020 2020 2020 636f 6e74 6169 6e20          contain 
+00005710: 636f 6d6d 6f6e 206f 7065 7261 746f 7273  common operators
+00005720: 2073 7563 6820 6173 2022 3d22 2c20 223c   such as "=", "<
+00005730: 2220 6f72 2022 696e 222e 2028 652e 672e  " or "in". (e.g.
+00005740: 200a 2020 2020 2020 2020 2020 2274 6178   .          "tax
+00005750: 6f6e 2e63 6f6d 6d6f 6e4e 616d 6520 3d20  on.commonName = 
+00005760: 6875 6d61 6e22 2c20 2274 6178 6f6e 2e63  human", "taxon.c
+00005770: 6f6d 6d6f 6e4e 616d 6520 696e 2028 6875  ommonName in (hu
+00005780: 6d61 6e2c 6d6f 7573 6529 2c20 0a20 2020  man,mouse), .   
+00005790: 2020 2020 2020 2022 6964 203c 2031 3030         "id < 100
+000057a0: 3022 292c 2064 6566 6175 6c74 7320 746f  0"), defaults to
+000057b0: 204e 6f6e 650a 2020 2020 2020 2020 3a74   None.        :t
+000057c0: 7970 6520 6669 6c74 6572 3a20 4f70 7469  ype filter: Opti
+000057d0: 6f6e 616c 5b73 7472 5d2c 206f 7074 696f  onal[str], optio
+000057e0: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
+000057f0: 616d 2074 6178 613a 2041 2076 6563 746f  am taxa: A vecto
+00005800: 7220 6f66 2074 6178 6f6e 2063 6f6d 6d6f  r of taxon commo
+00005810: 6e20 6e61 6d65 7320 2865 2e67 2e20 6875  n names (e.g. hu
+00005820: 6d61 6e2c 206d 6f75 7365 2c20 7261 7429  man, mouse, rat)
+00005830: 2e0a 2020 2020 2020 2020 2020 5072 6f76  ..          Prov
+00005840: 6964 696e 6720 6d75 6c74 6970 6c65 2073  iding multiple s
+00005850: 7065 6369 6573 2077 696c 6c20 7265 7475  pecies will retu
+00005860: 726e 2072 6573 756c 7473 2066 6f72 2061  rn results for a
+00005870: 6c6c 2073 7065 6369 6573 2e20 5468 6573  ll species. Thes
+00005880: 6520 0a20 2020 2020 2020 2020 2061 7265  e .          are
+00005890: 2061 7070 656e 6465 6420 746f 2074 6865   appended to the
+000058a0: 2066 696c 7465 7220 616e 6420 6571 7569   filter and equi
+000058b0: 7661 6c65 6e74 2074 6f20 6669 6c74 6572  valent to filter
+000058c0: 696e 6720 666f 7220 0a20 2020 2020 2020  ing for .       
+000058d0: 2020 2074 6178 6f6e 2e63 6f6d 6d6f 6e4e     taxon.commonN
+000058e0: 616d 6520 7072 6f70 6572 7479 2c20 6465  ame property, de
+000058f0: 6661 756c 7473 2074 6f20 4e6f 6e65 0a20  faults to None. 
+00005900: 2020 2020 2020 203a 7061 7261 6d20 7461         :param ta
+00005910: 7861 3a20 4120 6c69 7374 206f 6620 7461  xa: A list of ta
+00005920: 786f 6e20 636f 6d6d 6f6e 206e 616d 6573  xon common names
+00005930: 2028 652e 672e 2068 756d 616e 2c20 6d6f   (e.g. human, mo
+00005940: 7573 652c 2072 6174 292e 200a 2020 2020  use, rat). .    
+00005950: 2020 2020 2020 5072 6f76 6964 696e 6720        Providing 
+00005960: 6d75 6c74 6970 6c65 2073 7065 6369 6573  multiple species
+00005970: 2077 696c 6c20 7265 7475 726e 2072 6573   will return res
+00005980: 756c 7473 2066 6f72 2061 6c6c 2073 7065  ults for all spe
+00005990: 6369 6573 2e20 0a20 2020 2020 2020 2020  cies. .         
+000059a0: 2054 6865 7365 2061 7265 2061 7070 656e   These are appen
+000059b0: 6465 6420 746f 2074 6865 2066 696c 7465  ded to the filte
+000059c0: 7220 616e 6420 6571 7569 7661 6c65 6e74  r and equivalent
+000059d0: 2074 6f20 6669 6c74 6572 696e 6720 666f   to filtering fo
+000059e0: 7220 0a20 2020 2020 2020 2020 2074 6178  r .          tax
+000059f0: 6f6e 2e63 6f6d 6d6f 6e4e 616d 6520 7072  on.commonName pr
+00005a00: 6f70 6572 7479 2c20 6465 6661 756c 7473  operty, defaults
+00005a10: 2074 6f20 4e6f 6e65 0a20 2020 2020 2020   to None.       
+00005a20: 203a 7479 7065 2074 6178 613a 204f 7074   :type taxa: Opt
+00005a30: 696f 6e61 6c5b 4c69 7374 5b73 7472 5d5d  ional[List[str]]
+00005a40: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
+00005a50: 2020 203a 7061 7261 6d20 7572 6973 3a20     :param uris: 
+00005a60: 4120 7665 6374 6f72 206f 6620 6f6e 746f  A vector of onto
+00005a70: 6c6f 6779 2074 6572 6d20 5552 4973 2e20  logy term URIs. 
+00005a80: 5072 6f76 6964 696e 6720 6d75 6c74 6970  Providing multip
+00005a90: 6c65 2074 6572 6d73 0a20 2020 2020 2020  le terms.       
+00005aa0: 2020 2077 696c 6c20 7265 7475 726e 2072     will return r
+00005ab0: 6573 756c 7473 2063 6f6e 7461 696e 696e  esults containin
+00005ac0: 6720 616e 7920 6f66 2074 6865 2074 6572  g any of the ter
+00005ad0: 6d73 2061 6e64 2074 6865 6972 2063 6869  ms and their chi
+00005ae0: 6c64 7265 6e2e 200a 2020 2020 2020 2020  ldren. .        
+00005af0: 2020 5468 6573 6520 6172 6520 6170 7065    These are appe
+00005b00: 6e64 6564 2074 6f20 7468 6520 6669 6c74  nded to the filt
+00005b10: 6572 2061 6e64 2065 7175 6976 616c 656e  er and equivalen
+00005b20: 7420 746f 2066 696c 7465 7269 6e67 2066  t to filtering f
+00005b30: 6f72 200a 2020 2020 2020 2020 2020 616c  or .          al
+00005b40: 6c43 6861 7261 6374 6572 6973 7469 6373  lCharacteristics
+00005b50: 2e76 616c 7565 5572 692c 2064 6566 6175  .valueUri, defau
+00005b60: 6c74 7320 746f 204e 6f6e 650a 2020 2020  lts to None.    
+00005b70: 2020 2020 3a74 7970 6520 7572 6973 3a20      :type uris: 
+00005b80: 4f70 7469 6f6e 616c 5b4c 6973 745b 7374  Optional[List[st
+00005b90: 725d 5d2c 206f 7074 696f 6e61 6c0a 2020  r]], optional.  
+00005ba0: 2020 2020 2020 3a70 6172 616d 206f 6666        :param off
+00005bb0: 7365 743a 2054 6865 206f 6666 7365 7420  set: The offset 
+00005bc0: 6f66 2074 6865 2066 6972 7374 2072 6574  of the first ret
+00005bd0: 7269 6576 6564 2072 6573 756c 742e 2c20  rieved result., 
+00005be0: 6465 6661 756c 7473 2074 6f20 300a 2020  defaults to 0.  
+00005bf0: 2020 2020 2020 3a74 7970 6520 6f66 6673        :type offs
+00005c00: 6574 3a20 696e 742c 206f 7074 696f 6e61  et: int, optiona
+00005c10: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
+00005c20: 206c 696d 6974 3a20 4c69 6d69 7473 2074   limit: Limits t
+00005c30: 6865 2072 6573 756c 7420 746f 2073 7065  he result to spe
+00005c40: 6369 6669 6564 2061 6d6f 756e 7420 6f66  cified amount of
+00005c50: 206f 626a 6563 7473 2e0a 2020 2020 2020   objects..      
+00005c60: 2020 2020 4861 7320 6120 6d61 7869 6d75      Has a maximu
+00005c70: 6d20 7661 6c75 6520 6f66 2031 3030 2e20  m value of 100. 
+00005c80: 5573 6520 746f 6765 7468 6572 2077 6974  Use together wit
+00005c90: 6820 6f66 6673 6574 2061 6e64 2074 6865  h offset and the
+00005ca0: 200a 2020 2020 2020 2020 2020 746f 7461   .          tota
+00005cb0: 6c5f 656c 656d 656e 7473 2061 7474 7269  l_elements attri
+00005cc0: 6275 7465 2069 6e20 7468 6520 6f75 7470  bute in the outp
+00005cd0: 7574 2074 6f20 636f 6d70 696c 6520 616c  ut to compile al
+00005ce0: 6c20 6461 7461 2069 6620 6e65 6564 6564  l data if needed
+00005cf0: 2e0a 2020 2020 2020 2020 2020 416c 7465  ..          Alte
+00005d00: 726e 6174 6976 656c 7920 6765 745f 616c  rnatively get_al
+00005d10: 6c5f 7061 6765 7320 6675 6e63 7469 6f6e  l_pages function
+00005d20: 2063 616e 2062 6520 7573 6564 2077 6974   can be used wit
+00005d30: 6820 616c 6c20 6675 6e63 7469 6f6e 730a  h all functions.
+00005d40: 2020 2020 2020 2020 2020 696e 636c 7564            includ
+00005d50: 696e 6720 6f66 6673 6574 2061 6e64 206c  ing offset and l
+00005d60: 696d 6974 2070 6172 616d 6574 6572 732c  imit parameters,
+00005d70: 2064 6566 6175 6c74 7320 746f 2032 300a   defaults to 20.
+00005d80: 2020 2020 2020 2020 3a74 7970 6520 6c69          :type li
+00005d90: 6d69 743a 2069 6e74 2c20 6f70 7469 6f6e  mit: int, option
+00005da0: 616c 0a20 2020 2020 2020 203a 7061 7261  al.        :para
+00005db0: 6d20 736f 7274 3a20 4f72 6465 7220 7265  m sort: Order re
+00005dc0: 7375 6c74 7320 6279 2074 6865 2067 6976  sults by the giv
+00005dd0: 656e 2070 726f 7065 7274 7920 616e 6420  en property and 
+00005de0: 6469 7265 6374 696f 6e2e 2054 6865 2027  direction. The '
+00005df0: 2b27 0a20 2020 2020 2020 2020 2073 6967  +'.          sig
+00005e00: 6e20 696e 6469 6361 7465 2061 7363 656e  n indicate ascen
+00005e10: 6469 6e67 206f 7264 6572 2077 6865 7265  ding order where
+00005e20: 6173 2074 6865 2027 2d27 2069 6e64 6963  as the '-' indic
+00005e30: 6174 6520 6465 7363 656e 6469 6e67 2c0a  ate descending,.
+00005e40: 2020 2020 2020 2020 2020 6465 6661 756c            defaul
+00005e50: 7473 2074 6f20 222b 6964 220a 2020 2020  ts to "+id".    
+00005e60: 2020 2020 3a74 7970 6520 736f 7274 3a20      :type sort: 
+00005e70: 7374 722c 206f 7074 696f 6e61 6c0a 2020  str, optional.  
+00005e80: 2020 2020 2020 3a70 6172 616d 202a 2a6b        :param **k
+00005e90: 7761 7267 733a 2041 6464 6974 696f 6e61  wargs: Additiona
+00005ea0: 6c20 6172 6775 6d65 6e74 7320 746f 2070  l arguments to p
+00005eb0: 6173 7320 746f 2072 6177 2e67 6574 5f64  ass to raw.get_d
+00005ec0: 6174 6173 6574 730a 2020 2020 2020 2020  atasets.        
+00005ed0: 3a72 6574 7572 6e3a 2041 2044 6174 6146  :return: A DataF
+00005ee0: 7261 6d65 2077 6974 6820 696e 666f 726d  rame with inform
+00005ef0: 6174 696f 6e20 6162 6f75 7420 7468 6520  ation about the 
+00005f00: 7175 6572 6965 6420 6461 7461 7365 7428  queried dataset(
+00005f10: 7329 2e0a 2020 2020 2020 2020 0a20 2020  s)..        .   
+00005f20: 2020 2020 2020 2054 6865 2066 6965 6c64         The field
+00005f30: 7320 6f66 2074 6865 2044 6174 6146 7261  s of the DataFra
+00005f40: 6d65 2061 7265 3a0a 2020 2020 2020 2020  me are:.        
+00005f50: 2020 2020 2d20 6578 7065 7269 6d65 6e74      - experiment
+00005f60: 5f73 686f 7274 5f6e 616d 653a 2053 686f  _short_name: Sho
+00005f70: 7274 6e61 6d65 2067 6976 656e 2074 6f20  rtname given to 
+00005f80: 7468 6520 6461 7461 7365 7420 7769 7468  the dataset with
+00005f90: 696e 2047 656d 6d61 2e20 4f66 7465 6e20  in Gemma. Often 
+00005fa0: 636f 7272 6573 706f 6e64 7320 746f 2061  corresponds to a
+00005fb0: 6363 6573 7369 6f6e 2049 440a 2020 2020  ccession ID.    
 00005fc0: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
-00005fd0: 6d65 6e74 5f49 443a 2049 6e74 6572 6e61  ment_ID: Interna
-00005fe0: 6c20 4944 206f 6620 7468 6520 6461 7461  l ID of the data
-00005ff0: 7365 742e 0a20 2020 2020 2020 2020 2020  set..           
-00006000: 202d 2065 7870 6572 696d 656e 745f 6465   - experiment_de
-00006010: 7363 7269 7074 696f 6e3a 2044 6573 6372  scription: Descr
-00006020: 6970 7469 6f6e 206f 6620 7468 6520 6461  iption of the da
-00006030: 7461 7365 740a 2020 2020 2020 2020 2020  taset.          
-00006040: 2020 2d20 6578 7065 7269 6d65 6e74 5f74    - experiment_t
-00006050: 726f 7562 6c65 643a 2044 6964 2061 6e20  roubled: Did an 
-00006060: 6175 746f 6d61 7469 6320 7072 6f63 6573  automatic proces
-00006070: 7320 7769 7468 696e 2067 656d 6d61 206f  s within gemma o
-00006080: 7220 6120 6375 7261 746f 7220 6d61 726b  r a curator mark
-00006090: 2074 6865 2064 6174 6173 6574 2061 7320   the dataset as 
-000060a0: 2274 726f 7562 6c65 6422 0a20 2020 2020  "troubled".     
-000060b0: 2020 2020 2020 202d 2065 7870 6572 696d         - experim
-000060c0: 656e 745f 6163 6365 7373 696f 6e3a 2041  ent_accession: A
-000060d0: 6363 6573 7369 6f6e 2049 4420 6f66 2074  ccession ID of t
-000060e0: 6865 2064 6174 6173 6574 2069 6e20 7468  he dataset in th
-000060f0: 6520 6578 7465 726e 616c 2064 6174 6162  e external datab
-00006100: 6173 6520 6974 2077 6173 2074 616b 656e  ase it was taken
-00006110: 2066 726f 6d0a 2020 2020 2020 2020 2020   from.          
-00006120: 2020 2d20 6578 7065 7269 6d65 6e74 5f64    - experiment_d
-00006130: 6174 6162 6173 653a 2054 6865 206e 616d  atabase: The nam
-00006140: 6520 6f66 2074 6865 2064 6174 6162 6173  e of the databas
-00006150: 6520 7768 6572 6520 7468 6520 6461 7461  e where the data
-00006160: 7365 7420 7761 7320 7461 6b65 6e20 6672  set was taken fr
-00006170: 6f6d 0a20 2020 2020 2020 2020 2020 202d  om.            -
-00006180: 2065 7870 6572 696d 656e 745f 5552 493a   experiment_URI:
-00006190: 2055 5249 206f 6620 7468 6520 6f72 6967   URI of the orig
-000061a0: 696e 616c 2064 6174 6162 6173 650a 2020  inal database.  
-000061b0: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
-000061c0: 7269 6d65 6e74 5f73 616d 706c 655f 636f  riment_sample_co
-000061d0: 756e 743a 204e 756d 6265 7220 6f66 2073  unt: Number of s
-000061e0: 616d 706c 6573 2069 6e20 7468 6520 6461  amples in the da
-000061f0: 7461 7365 740a 2020 2020 2020 2020 2020  taset.          
-00006200: 2020 2d20 6578 7065 7269 6d65 6e74 5f62    - experiment_b
-00006210: 6174 6368 5f65 6666 6563 745f 7465 7874  atch_effect_text
-00006220: 3a20 4120 7465 7874 2066 6965 6c64 2064  : A text field d
-00006230: 6573 6372 6962 696e 6720 7768 6574 6865  escribing whethe
-00006240: 7220 7468 6520 6461 7461 7365 7420 6861  r the dataset ha
-00006250: 7320 6261 7463 6820 6566 6665 6374 730a  s batch effects.
-00006260: 2020 2020 2020 2020 2020 2020 2d20 6578              - ex
-00006270: 7065 7269 6d65 6e5f 6261 7463 685f 636f  perimen_batch_co
-00006280: 7272 6563 7465 643a 2057 6865 7468 6572  rrected: Whether
-00006290: 2062 6174 6368 2063 6f72 7265 6374 696f   batch correctio
-000062a0: 6e20 6861 7320 6265 656e 2070 6572 666f  n has been perfo
-000062b0: 726d 6564 206f 6e20 7468 6520 6461 7461  rmed on the data
-000062c0: 7365 742e 0a20 2020 2020 2020 2020 2020  set..           
-000062d0: 202d 2065 7870 6572 696d 656e 5f62 6174   - experimen_bat
-000062e0: 6368 5f63 6f6e 666f 756e 643a 2030 2069  ch_confound: 0 i
-000062f0: 6620 6261 7463 6820 696e 666f 2069 736e  f batch info isn
-00006300: 2774 2061 7661 696c 6162 6c65 2c20 2d31  't available, -1
-00006310: 2069 6620 6261 7463 6820 636f 756e 666f   if batch counfo
-00006320: 7564 2069 7320 6465 7465 6374 6564 2c20  ud is detected, 
-00006330: 3120 6966 2062 6174 6368 2069 6e66 6f72  1 if batch infor
-00006340: 6d61 7469 6f6e 2069 7320 6176 6169 6c61  mation is availa
-00006350: 626c 6520 616e 6420 6e6f 2062 6174 6368  ble and no batch
-00006360: 2063 6f6e 666f 756e 6420 666f 756e 640a   confound found.
-00006370: 2020 2020 2020 2020 2020 2020 2d20 6578              - ex
-00006380: 7065 7269 6d65 6e5f 6261 7463 685f 6566  perimen_batch_ef
-00006390: 6665 6374 3a20 2d31 2069 6620 6261 7463  fect: -1 if batc
-000063a0: 6820 7020 7661 6c75 6520 3c20 302e 3030  h p value < 0.00
-000063b0: 3031 2c20 3120 6966 2062 6174 6368 2070  01, 1 if batch p
-000063c0: 2076 616c 7565 203e 2030 2e31 2c20 3020   value > 0.1, 0 
-000063d0: 6966 206f 7468 6572 7769 7365 2061 6e64  if otherwise and
-000063e0: 2077 6865 6e20 7468 6572 6520 6973 206e   when there is n
-000063f0: 6f20 6261 7463 6820 696e 666f 726d 6174  o batch informat
-00006400: 696f 6e20 6973 2061 7661 696c 6162 6c65  ion is available
-00006410: 206f 7220 7768 656e 2074 6865 2064 6174   or when the dat
-00006420: 6120 6973 2063 6f6e 666f 756e 6465 6420  a is confounded 
-00006430: 7769 7468 2062 6174 6368 6573 2e0a 2020  with batches..  
-00006440: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
-00006450: 7269 6d65 6e5f 7261 775f 6461 7461 3a20  rimen_raw_data: 
-00006460: 2d31 2069 6620 6e6f 2072 6177 2064 6174  -1 if no raw dat
-00006470: 6120 6176 6169 6c61 626c 652c 2031 2069  a available, 1 i
-00006480: 6620 7261 7720 6461 7461 2077 6173 2061  f raw data was a
-00006490: 7661 696c 6162 6c65 2e20 5768 656e 2061  vailable. When a
-000064a0: 7661 696c 6162 6c65 2c20 4765 6d6d 6120  vailable, Gemma 
-000064b0: 7265 7072 6f63 6573 7365 7320 7261 7720  reprocesses raw 
-000064c0: 6461 7461 2074 6f20 6765 7420 6578 7072  data to get expr
-000064d0: 6573 7369 6f6e 2076 616c 7565 7320 616e  ession values an
-000064e0: 6420 6261 7463 6865 730a 2020 2020 2020  d batches.      
-000064f0: 2020 2020 2020 2d20 6765 6571 5f71 5f73        - geeq_q_s
-00006500: 636f 7265 3a20 4461 7461 2071 7561 6c69  core: Data quali
-00006510: 7479 2073 636f 7265 2067 6976 656e 2074  ty score given t
-00006520: 6f20 7468 6520 6461 7461 7365 7420 6279  o the dataset by
-00006530: 2047 656d 6d61 2e0a 2020 2020 2020 2020   Gemma..        
-00006540: 2020 2020 2d20 6765 6571 5f73 5f73 636f      - geeq_s_sco
-00006550: 7265 3a20 5375 6974 6162 696c 6974 7920  re: Suitability 
-00006560: 7363 6f72 6520 6769 7665 6e20 746f 2074  score given to t
-00006570: 6865 2064 6174 6173 6574 2062 7920 4765  he dataset by Ge
-00006580: 6d6d 612e 2052 6566 6572 7320 746f 2066  mma. Refers to f
-00006590: 6163 746f 7273 206c 696b 6520 6261 7463  actors like batc
-000065a0: 6865 732c 2070 6c61 7466 6f72 6d73 2061  hes, platforms a
-000065b0: 6e64 206f 7468 6572 2061 7370 6563 7473  nd other aspects
-000065c0: 206f 6620 6578 7065 7269 6d65 6e74 616c   of experimental
-000065d0: 2064 6573 6967 6e0a 2020 2020 2020 2020   design.        
-000065e0: 2020 2020 2d20 7461 786f 6e5f 6e61 6d65      - taxon_name
-000065f0: 3a20 4e61 6d65 206f 6620 7468 6520 7370  : Name of the sp
-00006600: 6563 6965 730a 2020 2020 2020 2020 2020  ecies.          
-00006610: 2020 2d20 7461 786f 6e5f 7363 6965 6e74    - taxon_scient
-00006620: 6966 6963 3a20 5363 6965 6e74 6966 6963  ific: Scientific
-00006630: 206e 616d 6520 666f 7220 7468 6520 7461   name for the ta
-00006640: 786f 6e0a 2020 2020 2020 2020 2020 2020  xon.            
-00006650: 2d20 7461 786f 6e5f 4944 3a20 496e 7465  - taxon_ID: Inte
-00006660: 726e 616c 2069 6465 6e74 6966 6965 7220  rnal identifier 
-00006670: 6769 7665 6e20 746f 2074 6865 2073 7065  given to the spe
-00006680: 6369 6573 2062 7920 4765 6d6d 610a 2020  cies by Gemma.  
-00006690: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
-000066a0: 6e5f 4e43 4249 3a20 4e43 4249 2049 4420  n_NCBI: NCBI ID 
-000066b0: 6f66 2074 6865 2074 6178 6f6e 0a20 2020  of the taxon.   
-000066c0: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
-000066d0: 5f64 6174 6162 6173 655f 6e61 6d65 3a20  _database_name: 
-000066e0: 556e 6465 726c 7969 6e67 2064 6174 6162  Underlying datab
-000066f0: 6173 6520 7573 6564 2069 6e20 4765 6d6d  ase used in Gemm
-00006700: 6120 666f 7220 7468 6520 7461 786f 6e0a  a for the taxon.
-00006710: 2020 2020 2020 2020 2020 2020 2d20 7461              - ta
-00006720: 786f 6e5f 6461 7461 6261 7365 5f49 443a  xon_database_ID:
-00006730: 2049 4420 6f66 2074 6865 2075 6e64 6572   ID of the under
-00006740: 796c 696e 6720 6461 7461 6261 7365 2075  yling database u
-00006750: 7365 6420 696e 2047 656d 6d61 2066 6f72  sed in Gemma for
-00006760: 2074 6865 2074 6178 6f6e 0a20 2020 2020   the taxon.     
-00006770: 2020 203a 7274 7970 653a 2044 6174 6146     :rtype: DataF
-00006780: 7261 6d65 0a0a 2020 2020 2020 2020 2222  rame..        ""
-00006790: 220a 2020 2020 2020 2020 0a20 2020 2020  ".        .     
-000067a0: 2020 2066 696c 7465 7220 3d20 7673 2e61     filter = vs.a
-000067b0: 6464 5f74 6f5f 6669 6c74 6572 2866 696c  dd_to_filter(fil
-000067c0: 7465 722c 2027 616c 6c43 6861 7261 6374  ter, 'allCharact
-000067d0: 6572 6973 7469 6373 2e76 616c 7565 5572  eristics.valueUr
-000067e0: 6927 2c20 7572 6973 290a 2020 2020 2020  i', uris).      
-000067f0: 2020 6669 6c74 6572 203d 2076 732e 6164    filter = vs.ad
-00006800: 645f 746f 5f66 696c 7465 7228 6669 6c74  d_to_filter(filt
-00006810: 6572 2c20 2774 6178 6f6e 2e63 6f6d 6d6f  er, 'taxon.commo
-00006820: 6e4e 616d 6527 2c20 7461 7861 290a 2020  nName', taxa).  
-00006830: 2020 2020 2020 0a20 2020 2020 2020 206b        .        k
-00006840: 7761 7267 7320 3d20 7673 2e72 656d 6f76  wargs = vs.remov
-00006850: 655f 6e6f 6e65 7328 0a20 2020 2020 2020  e_nones(.       
-00006860: 2020 2020 2071 7565 7279 203d 2071 7565       query = que
-00006870: 7279 2c0a 2020 2020 2020 2020 2020 2020  ry,.            
-00006880: 6669 6c74 6572 203d 2066 696c 7465 722c  filter = filter,
-00006890: 0a20 2020 2020 2020 2020 2020 206f 6666  .            off
-000068a0: 7365 7420 3d20 6f66 6673 6574 2c0a 2020  set = offset,.  
-000068b0: 2020 2020 2020 2020 2020 6c69 6d69 7420            limit 
-000068c0: 3d20 6c69 6d69 742c 0a20 2020 2020 2020  = limit,.       
-000068d0: 2020 2020 2073 6f72 7420 3d20 736f 7274       sort = sort
-000068e0: 2c0a 2020 2020 2020 2020 2020 2020 2a2a  ,.            **
-000068f0: 6b77 6172 6773 290a 2020 2020 2020 2020  kwargs).        
-00006900: 0a20 2020 2020 2020 2072 6573 706f 6e73  .        respons
-00006910: 6520 3d20 7365 6c66 2e72 6177 2e67 6574  e = self.raw.get
-00006920: 5f64 6174 6173 6574 7328 2a2a 6b77 6172  _datasets(**kwar
-00006930: 6773 290a 2020 2020 2020 2020 6466 203d  gs).        df =
-00006940: 2070 732e 7072 6f63 6573 735f 6461 7461   ps.process_data
-00006950: 7365 7473 2872 6573 706f 6e73 652e 6461  sets(response.da
-00006960: 7461 290a 2020 2020 2020 2020 7073 2e61  ta).        ps.a
-00006970: 7474 6163 685f 6174 7472 6962 7574 6573  ttach_attributes
-00006980: 2864 662c 2072 6573 706f 6e73 652e 746f  (df, response.to
-00006990: 5f64 6963 7428 2929 0a0a 2020 2020 2020  _dict())..      
-000069a0: 2020 0a20 2020 2020 2020 2072 6574 7572    .        retur
-000069b0: 6e20 6466 0a20 2020 200a 2020 2020 2320  n df.    .    # 
-000069c0: 6461 7461 7365 7473 2f61 6e6e 6f74 6174  datasets/annotat
-000069d0: 696f 6e73 202d 2d2d 2d2d 0a20 2020 2023  ions -----.    #
-000069e0: 2063 7572 7265 6e74 6c79 2075 6e69 6d70   currently unimp
-000069f0: 6c65 6d65 6e74 6564 0a20 2020 200a 2020  lemented.    .  
-00006a00: 2020 0a20 2020 2023 2064 6174 6173 6574    .    # dataset
-00006a10: 732f 7b64 6174 6173 6574 737d 2c20 6765  s/{datasets}, ge
-00006a20: 745f 6461 7461 7365 7473 5f62 795f 6964  t_datasets_by_id
-00006a30: 7320 2d2d 2d2d 2d0a 2020 2020 6465 6620  s -----.    def 
-00006a40: 6765 745f 6461 7461 7365 7473 5f62 795f  get_datasets_by_
-00006a50: 6964 7328 7365 6c66 2c20 6461 7461 7365  ids(self, datase
-00006a60: 743a 4c69 7374 5b73 7472 7c69 6e74 5d2c  t:List[str|int],
-00006a70: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00006a80: 2020 2020 2020 2020 2020 2020 2066 696c               fil
-00006a90: 7465 723a 4f70 7469 6f6e 616c 5b73 7472  ter:Optional[str
-00006aa0: 5d20 3d20 4e6f 6e65 2c20 0a20 2020 2020  ] = None, .     
+00005fd0: 6d65 6e74 5f6e 616d 653a 2046 756c 6c20  ment_name: Full 
+00005fe0: 7469 746c 6520 6f66 2074 6865 2064 6174  title of the dat
+00005ff0: 6173 6574 0a20 2020 2020 2020 2020 2020  aset.           
+00006000: 202d 2065 7870 6572 696d 656e 745f 4944   - experiment_ID
+00006010: 3a20 496e 7465 726e 616c 2049 4420 6f66  : Internal ID of
+00006020: 2074 6865 2064 6174 6173 6574 2e0a 2020   the dataset..  
+00006030: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
+00006040: 7269 6d65 6e74 5f64 6573 6372 6970 7469  riment_descripti
+00006050: 6f6e 3a20 4465 7363 7269 7074 696f 6e20  on: Description 
+00006060: 6f66 2074 6865 2064 6174 6173 6574 0a20  of the dataset. 
+00006070: 2020 2020 2020 2020 2020 202d 2065 7870             - exp
+00006080: 6572 696d 656e 745f 7472 6f75 626c 6564  eriment_troubled
+00006090: 3a20 4469 6420 616e 2061 7574 6f6d 6174  : Did an automat
+000060a0: 6963 2070 726f 6365 7373 2077 6974 6869  ic process withi
+000060b0: 6e20 6765 6d6d 6120 6f72 2061 2063 7572  n gemma or a cur
+000060c0: 6174 6f72 206d 6172 6b20 7468 6520 6461  ator mark the da
+000060d0: 7461 7365 7420 6173 2022 7472 6f75 626c  taset as "troubl
+000060e0: 6564 220a 2020 2020 2020 2020 2020 2020  ed".            
+000060f0: 2d20 6578 7065 7269 6d65 6e74 5f61 6363  - experiment_acc
+00006100: 6573 7369 6f6e 3a20 4163 6365 7373 696f  ession: Accessio
+00006110: 6e20 4944 206f 6620 7468 6520 6461 7461  n ID of the data
+00006120: 7365 7420 696e 2074 6865 2065 7874 6572  set in the exter
+00006130: 6e61 6c20 6461 7461 6261 7365 2069 7420  nal database it 
+00006140: 7761 7320 7461 6b65 6e20 6672 6f6d 0a20  was taken from. 
+00006150: 2020 2020 2020 2020 2020 202d 2065 7870             - exp
+00006160: 6572 696d 656e 745f 6461 7461 6261 7365  eriment_database
+00006170: 3a20 5468 6520 6e61 6d65 206f 6620 7468  : The name of th
+00006180: 6520 6461 7461 6261 7365 2077 6865 7265  e database where
+00006190: 2074 6865 2064 6174 6173 6574 2077 6173   the dataset was
+000061a0: 2074 616b 656e 2066 726f 6d0a 2020 2020   taken from.    
+000061b0: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
+000061c0: 6d65 6e74 5f55 5249 3a20 5552 4920 6f66  ment_URI: URI of
+000061d0: 2074 6865 206f 7269 6769 6e61 6c20 6461   the original da
+000061e0: 7461 6261 7365 0a20 2020 2020 2020 2020  tabase.         
+000061f0: 2020 202d 2065 7870 6572 696d 656e 745f     - experiment_
+00006200: 7361 6d70 6c65 5f63 6f75 6e74 3a20 4e75  sample_count: Nu
+00006210: 6d62 6572 206f 6620 7361 6d70 6c65 7320  mber of samples 
+00006220: 696e 2074 6865 2064 6174 6173 6574 0a20  in the dataset. 
+00006230: 2020 2020 2020 2020 2020 202d 2065 7870             - exp
+00006240: 6572 696d 656e 745f 6261 7463 685f 6566  eriment_batch_ef
+00006250: 6665 6374 5f74 6578 743a 2041 2074 6578  fect_text: A tex
+00006260: 7420 6669 656c 6420 6465 7363 7269 6269  t field describi
+00006270: 6e67 2077 6865 7468 6572 2074 6865 2064  ng whether the d
+00006280: 6174 6173 6574 2068 6173 2062 6174 6368  ataset has batch
+00006290: 2065 6666 6563 7473 0a20 2020 2020 2020   effects.       
+000062a0: 2020 2020 202d 2065 7870 6572 696d 656e       - experimen
+000062b0: 5f62 6174 6368 5f63 6f72 7265 6374 6564  _batch_corrected
+000062c0: 3a20 5768 6574 6865 7220 6261 7463 6820  : Whether batch 
+000062d0: 636f 7272 6563 7469 6f6e 2068 6173 2062  correction has b
+000062e0: 6565 6e20 7065 7266 6f72 6d65 6420 6f6e  een performed on
+000062f0: 2074 6865 2064 6174 6173 6574 2e0a 2020   the dataset..  
+00006300: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
+00006310: 7269 6d65 6e5f 6261 7463 685f 636f 6e66  rimen_batch_conf
+00006320: 6f75 6e64 3a20 3020 6966 2062 6174 6368  ound: 0 if batch
+00006330: 2069 6e66 6f20 6973 6e27 7420 6176 6169   info isn't avai
+00006340: 6c61 626c 652c 202d 3120 6966 2062 6174  lable, -1 if bat
+00006350: 6368 2063 6f75 6e66 6f75 6420 6973 2064  ch counfoud is d
+00006360: 6574 6563 7465 642c 2031 2069 6620 6261  etected, 1 if ba
+00006370: 7463 6820 696e 666f 726d 6174 696f 6e20  tch information 
+00006380: 6973 2061 7661 696c 6162 6c65 2061 6e64  is available and
+00006390: 206e 6f20 6261 7463 6820 636f 6e66 6f75   no batch confou
+000063a0: 6e64 2066 6f75 6e64 0a20 2020 2020 2020  nd found.       
+000063b0: 2020 2020 202d 2065 7870 6572 696d 656e       - experimen
+000063c0: 5f62 6174 6368 5f65 6666 6563 743a 202d  _batch_effect: -
+000063d0: 3120 6966 2062 6174 6368 2070 2076 616c  1 if batch p val
+000063e0: 7565 203c 2030 2e30 3030 312c 2031 2069  ue < 0.0001, 1 i
+000063f0: 6620 6261 7463 6820 7020 7661 6c75 6520  f batch p value 
+00006400: 3e20 302e 312c 2030 2069 6620 6f74 6865  > 0.1, 0 if othe
+00006410: 7277 6973 6520 616e 6420 7768 656e 2074  rwise and when t
+00006420: 6865 7265 2069 7320 6e6f 2062 6174 6368  here is no batch
+00006430: 2069 6e66 6f72 6d61 7469 6f6e 2069 7320   information is 
+00006440: 6176 6169 6c61 626c 6520 6f72 2077 6865  available or whe
+00006450: 6e20 7468 6520 6461 7461 2069 7320 636f  n the data is co
+00006460: 6e66 6f75 6e64 6564 2077 6974 6820 6261  nfounded with ba
+00006470: 7463 6865 732e 0a20 2020 2020 2020 2020  tches..         
+00006480: 2020 202d 2065 7870 6572 696d 656e 5f72     - experimen_r
+00006490: 6177 5f64 6174 613a 202d 3120 6966 206e  aw_data: -1 if n
+000064a0: 6f20 7261 7720 6461 7461 2061 7661 696c  o raw data avail
+000064b0: 6162 6c65 2c20 3120 6966 2072 6177 2064  able, 1 if raw d
+000064c0: 6174 6120 7761 7320 6176 6169 6c61 626c  ata was availabl
+000064d0: 652e 2057 6865 6e20 6176 6169 6c61 626c  e. When availabl
+000064e0: 652c 2047 656d 6d61 2072 6570 726f 6365  e, Gemma reproce
+000064f0: 7373 6573 2072 6177 2064 6174 6120 746f  sses raw data to
+00006500: 2067 6574 2065 7870 7265 7373 696f 6e20   get expression 
+00006510: 7661 6c75 6573 2061 6e64 2062 6174 6368  values and batch
+00006520: 6573 0a20 2020 2020 2020 2020 2020 202d  es.            -
+00006530: 2067 6565 715f 715f 7363 6f72 653a 2044   geeq_q_score: D
+00006540: 6174 6120 7175 616c 6974 7920 7363 6f72  ata quality scor
+00006550: 6520 6769 7665 6e20 746f 2074 6865 2064  e given to the d
+00006560: 6174 6173 6574 2062 7920 4765 6d6d 612e  ataset by Gemma.
+00006570: 0a20 2020 2020 2020 2020 2020 202d 2067  .            - g
+00006580: 6565 715f 735f 7363 6f72 653a 2053 7569  eeq_s_score: Sui
+00006590: 7461 6269 6c69 7479 2073 636f 7265 2067  tability score g
+000065a0: 6976 656e 2074 6f20 7468 6520 6461 7461  iven to the data
+000065b0: 7365 7420 6279 2047 656d 6d61 2e20 5265  set by Gemma. Re
+000065c0: 6665 7273 2074 6f20 6661 6374 6f72 7320  fers to factors 
+000065d0: 6c69 6b65 2062 6174 6368 6573 2c20 706c  like batches, pl
+000065e0: 6174 666f 726d 7320 616e 6420 6f74 6865  atforms and othe
+000065f0: 7220 6173 7065 6374 7320 6f66 2065 7870  r aspects of exp
+00006600: 6572 696d 656e 7461 6c20 6465 7369 676e  erimental design
+00006610: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
+00006620: 6178 6f6e 5f6e 616d 653a 204e 616d 6520  axon_name: Name 
+00006630: 6f66 2074 6865 2073 7065 6369 6573 0a20  of the species. 
+00006640: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
+00006650: 6f6e 5f73 6369 656e 7469 6669 633a 2053  on_scientific: S
+00006660: 6369 656e 7469 6669 6320 6e61 6d65 2066  cientific name f
+00006670: 6f72 2074 6865 2074 6178 6f6e 0a20 2020  or the taxon.   
+00006680: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
+00006690: 5f49 443a 2049 6e74 6572 6e61 6c20 6964  _ID: Internal id
+000066a0: 656e 7469 6669 6572 2067 6976 656e 2074  entifier given t
+000066b0: 6f20 7468 6520 7370 6563 6965 7320 6279  o the species by
+000066c0: 2047 656d 6d61 0a20 2020 2020 2020 2020   Gemma.         
+000066d0: 2020 202d 2074 6178 6f6e 5f4e 4342 493a     - taxon_NCBI:
+000066e0: 204e 4342 4920 4944 206f 6620 7468 6520   NCBI ID of the 
+000066f0: 7461 786f 6e0a 2020 2020 2020 2020 2020  taxon.          
+00006700: 2020 2d20 7461 786f 6e5f 6461 7461 6261    - taxon_databa
+00006710: 7365 5f6e 616d 653a 2055 6e64 6572 6c79  se_name: Underly
+00006720: 696e 6720 6461 7461 6261 7365 2075 7365  ing database use
+00006730: 6420 696e 2047 656d 6d61 2066 6f72 2074  d in Gemma for t
+00006740: 6865 2074 6178 6f6e 0a20 2020 2020 2020  he taxon.       
+00006750: 2020 2020 202d 2074 6178 6f6e 5f64 6174       - taxon_dat
+00006760: 6162 6173 655f 4944 3a20 4944 206f 6620  abase_ID: ID of 
+00006770: 7468 6520 756e 6465 7279 6c69 6e67 2064  the underyling d
+00006780: 6174 6162 6173 6520 7573 6564 2069 6e20  atabase used in 
+00006790: 4765 6d6d 6120 666f 7220 7468 6520 7461  Gemma for the ta
+000067a0: 786f 6e0a 2020 2020 2020 2020 3a72 7479  xon.        :rty
+000067b0: 7065 3a20 4461 7461 4672 616d 650a 0a20  pe: DataFrame.. 
+000067c0: 2020 2020 2020 2022 2222 0a20 2020 2020         """.     
+000067d0: 2020 200a 2020 2020 2020 2020 6669 6c74     .        filt
+000067e0: 6572 203d 2076 732e 6164 645f 746f 5f66  er = vs.add_to_f
+000067f0: 696c 7465 7228 6669 6c74 6572 2c20 2761  ilter(filter, 'a
+00006800: 6c6c 4368 6172 6163 7465 7269 7374 6963  llCharacteristic
+00006810: 732e 7661 6c75 6555 7269 272c 2075 7269  s.valueUri', uri
+00006820: 7329 0a20 2020 2020 2020 2066 696c 7465  s).        filte
+00006830: 7220 3d20 7673 2e61 6464 5f74 6f5f 6669  r = vs.add_to_fi
+00006840: 6c74 6572 2866 696c 7465 722c 2027 7461  lter(filter, 'ta
+00006850: 786f 6e2e 636f 6d6d 6f6e 4e61 6d65 272c  xon.commonName',
+00006860: 2074 6178 6129 0a20 2020 2020 2020 200a   taxa).        .
+00006870: 2020 2020 2020 2020 6b77 6172 6773 203d          kwargs =
+00006880: 2076 732e 7265 6d6f 7665 5f6e 6f6e 6573   vs.remove_nones
+00006890: 280a 2020 2020 2020 2020 2020 2020 7175  (.            qu
+000068a0: 6572 7920 3d20 7175 6572 792c 0a20 2020  ery = query,.   
+000068b0: 2020 2020 2020 2020 2066 696c 7465 7220           filter 
+000068c0: 3d20 6669 6c74 6572 2c0a 2020 2020 2020  = filter,.      
+000068d0: 2020 2020 2020 6f66 6673 6574 203d 206f        offset = o
+000068e0: 6666 7365 742c 0a20 2020 2020 2020 2020  ffset,.         
+000068f0: 2020 206c 696d 6974 203d 206c 696d 6974     limit = limit
+00006900: 2c0a 2020 2020 2020 2020 2020 2020 736f  ,.            so
+00006910: 7274 203d 2073 6f72 742c 0a20 2020 2020  rt = sort,.     
+00006920: 2020 2020 2020 202a 2a6b 7761 7267 7329         **kwargs)
+00006930: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
+00006940: 2020 7265 7370 6f6e 7365 203d 2073 656c    response = sel
+00006950: 662e 7261 772e 6765 745f 6461 7461 7365  f.raw.get_datase
+00006960: 7473 282a 2a6b 7761 7267 7329 0a20 2020  ts(**kwargs).   
+00006970: 2020 2020 2064 6620 3d20 7073 2e70 726f       df = ps.pro
+00006980: 6365 7373 5f64 6174 6173 6574 7328 7265  cess_datasets(re
+00006990: 7370 6f6e 7365 2e64 6174 6129 0a20 2020  sponse.data).   
+000069a0: 2020 2020 2070 732e 6174 7461 6368 5f61       ps.attach_a
+000069b0: 7474 7269 6275 7465 7328 6466 2c20 7265  ttributes(df, re
+000069c0: 7370 6f6e 7365 2e74 6f5f 6469 6374 2829  sponse.to_dict()
+000069d0: 290a 0a20 2020 2020 2020 200a 2020 2020  )..        .    
+000069e0: 2020 2020 7265 7475 726e 2064 660a 2020      return df.  
+000069f0: 2020 0a20 2020 2023 2064 6174 6173 6574    .    # dataset
+00006a00: 732f 616e 6e6f 7461 7469 6f6e 7320 2d2d  s/annotations --
+00006a10: 2d2d 2d0a 2020 2020 2320 6375 7272 656e  ---.    # curren
+00006a20: 746c 7920 756e 696d 706c 656d 656e 7465  tly unimplemente
+00006a30: 640a 2020 2020 0a20 2020 200a 2020 2020  d.    .    .    
+00006a40: 2320 6461 7461 7365 7473 2f7b 6461 7461  # datasets/{data
+00006a50: 7365 7473 7d2c 2067 6574 5f64 6174 6173  sets}, get_datas
+00006a60: 6574 735f 6279 5f69 6473 202d 2d2d 2d2d  ets_by_ids -----
+00006a70: 0a20 2020 2064 6566 2067 6574 5f64 6174  .    def get_dat
+00006a80: 6173 6574 735f 6279 5f69 6473 2873 656c  asets_by_ids(sel
+00006a90: 662c 2064 6174 6173 6574 3a4c 6973 745b  f, dataset:List[
+00006aa0: 7374 727c 696e 745d 2c0a 2020 2020 2020  str|int],.      
 00006ab0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00006ac0: 2020 2020 2020 2074 6178 613a 4f70 7469         taxa:Opti
-00006ad0: 6f6e 616c 5b4c 6973 745b 7374 725d 5d20  onal[List[str]] 
-00006ae0: 3d20 4e6f 6e65 2c20 0a20 2020 2020 2020  = None, .       
+00006ac0: 2020 2020 2020 6669 6c74 6572 3a4f 7074        filter:Opt
+00006ad0: 696f 6e61 6c5b 7374 725d 203d 204e 6f6e  ional[str] = Non
+00006ae0: 652c 200a 2020 2020 2020 2020 2020 2020  e, .            
 00006af0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00006b00: 2020 2020 2075 7269 733a 4f70 7469 6f6e       uris:Option
-00006b10: 616c 5b4c 6973 745b 7374 725d 5d20 3d20  al[List[str]] = 
-00006b20: 4e6f 6e65 2c0a 2020 2020 2020 2020 2020  None,.          
-00006b30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00006b40: 2020 6f66 6673 6574 3a69 6e74 203d 2030    offset:int = 0
-00006b50: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
-00006b60: 2020 2020 2020 2020 2020 2020 2020 6c69                li
-00006b70: 6d69 743a 696e 7420 3d20 3230 2c0a 2020  mit:int = 20,.  
-00006b80: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00006b90: 2020 2020 2020 2020 2020 736f 7274 3a73            sort:s
-00006ba0: 7472 203d 2022 2b69 6422 2c0a 2020 2020  tr = "+id",.    
-00006bb0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00006bc0: 2020 2020 2020 2020 2a2a 6b77 6172 6773          **kwargs
-00006bd0: 292d 3e44 6174 6146 7261 6d65 3a0a 2020  )->DataFrame:.  
-00006be0: 2020 2020 2020 2222 220a 2020 2020 2020        """.      
-00006bf0: 2020 0a20 2020 2020 2020 203a 7061 7261    .        :para
-00006c00: 6d20 6461 7461 7365 743a 204e 756d 6572  m dataset: Numer
-00006c10: 6963 616c 2064 6174 6173 6574 2069 6465  ical dataset ide
-00006c20: 6e74 6966 6965 7273 206f 7220 6461 7461  ntifiers or data
-00006c30: 7365 7420 7368 6f72 7420 6e61 6d65 732e  set short names.
-00006c40: 0a20 2020 2020 2020 203a 7479 7065 2064  .        :type d
-00006c50: 6174 6173 6574 3a20 4c69 7374 5b73 7472  ataset: List[str
-00006c60: 7c69 6e74 5d0a 2020 2020 2020 2020 3a70  |int].        :p
-00006c70: 6172 616d 2066 696c 7465 723a 2046 696c  aram filter: Fil
-00006c80: 7465 7220 7265 7375 6c74 7320 6279 206d  ter results by m
-00006c90: 6174 6368 696e 6720 6578 7072 6573 7369  atching expressi
-00006ca0: 6f6e 2e20 5573 6520 0a20 2020 2020 2020  on. Use .       
-00006cb0: 2020 2066 696c 7465 725f 7072 6f70 6572     filter_proper
-00006cc0: 7469 6573 2066 756e 6374 696f 6e20 746f  ties function to
-00006cd0: 2067 6574 2061 206c 6973 7420 6f66 2061   get a list of a
-00006ce0: 6c6c 2061 7661 696c 6162 6c65 2070 6172  ll available par
-00006cf0: 616d 6574 6572 732e 200a 2020 2020 2020  ameters. .      
-00006d00: 2020 2020 5468 6573 6520 7072 6f70 6572      These proper
-00006d10: 7469 6573 2063 616e 2062 6520 636f 6d62  ties can be comb
-00006d20: 696e 6564 2075 7369 6e67 2022 616e 6422  ined using "and"
-00006d30: 2022 6f72 2220 636c 6175 7365 7320 616e   "or" clauses an
-00006d40: 6420 6d61 7920 0a20 2020 2020 2020 2020  d may .         
-00006d50: 2063 6f6e 7461 696e 2063 6f6d 6d6f 6e20   contain common 
-00006d60: 6f70 6572 6174 6f72 7320 7375 6368 2061  operators such a
-00006d70: 7320 223d 222c 2022 3c22 206f 7220 2269  s "=", "<" or "i
-00006d80: 6e22 2e20 2865 2e67 2e20 0a20 2020 2020  n". (e.g. .     
-00006d90: 2020 2020 2022 7461 786f 6e2e 636f 6d6d       "taxon.comm
-00006da0: 6f6e 4e61 6d65 203d 2068 756d 616e 222c  onName = human",
-00006db0: 2022 7461 786f 6e2e 636f 6d6d 6f6e 4e61   "taxon.commonNa
-00006dc0: 6d65 2069 6e20 2868 756d 616e 2c6d 6f75  me in (human,mou
-00006dd0: 7365 292c 200a 2020 2020 2020 2020 2020  se), .          
-00006de0: 2269 6420 3c20 3130 3030 2229 2c20 6465  "id < 1000"), de
-00006df0: 6661 756c 7473 2074 6f20 4e6f 6e65 0a20  faults to None. 
-00006e00: 2020 2020 2020 203a 7479 7065 2066 696c         :type fil
-00006e10: 7465 723a 204f 7074 696f 6e61 6c5b 7374  ter: Optional[st
-00006e20: 725d 2c20 6f70 7469 6f6e 616c 0a20 2020  r], optional.   
-00006e30: 2020 2020 203a 7061 7261 6d20 7461 7861       :param taxa
-00006e40: 3a20 4120 6c69 7374 206f 6620 7461 786f  : A list of taxo
-00006e50: 6e20 636f 6d6d 6f6e 206e 616d 6573 2028  n common names (
-00006e60: 652e 672e 2068 756d 616e 2c20 6d6f 7573  e.g. human, mous
-00006e70: 652c 2072 6174 292e 200a 2020 2020 2020  e, rat). .      
-00006e80: 2020 2020 5072 6f76 6964 696e 6720 6d75      Providing mu
-00006e90: 6c74 6970 6c65 2073 7065 6369 6573 2077  ltiple species w
-00006ea0: 696c 6c20 7265 7475 726e 2072 6573 756c  ill return resul
-00006eb0: 7473 2066 6f72 2061 6c6c 2073 7065 6369  ts for all speci
-00006ec0: 6573 2e20 0a20 2020 2020 2020 2020 2054  es. .          T
-00006ed0: 6865 7365 2061 7265 2061 7070 656e 6465  hese are appende
-00006ee0: 6420 746f 2074 6865 2066 696c 7465 7220  d to the filter 
-00006ef0: 616e 6420 6571 7569 7661 6c65 6e74 2074  and equivalent t
-00006f00: 6f20 6669 6c74 6572 696e 6720 666f 7220  o filtering for 
-00006f10: 0a20 2020 2020 2020 2020 2074 6178 6f6e  .          taxon
-00006f20: 2e63 6f6d 6d6f 6e4e 616d 6520 7072 6f70  .commonName prop
-00006f30: 6572 7479 2c20 6465 6661 756c 7473 2074  erty, defaults t
-00006f40: 6f20 4e6f 6e65 0a20 2020 2020 2020 203a  o None.        :
-00006f50: 7479 7065 2074 6178 613a 204f 7074 696f  type taxa: Optio
-00006f60: 6e61 6c5b 4c69 7374 5b73 7472 5d5d 2c20  nal[List[str]], 
-00006f70: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
-00006f80: 203a 7061 7261 6d20 7572 6973 3a20 4120   :param uris: A 
-00006f90: 7665 6374 6f72 206f 6620 6f6e 746f 6c6f  vector of ontolo
-00006fa0: 6779 2074 6572 6d20 5552 4973 2e20 5072  gy term URIs. Pr
-00006fb0: 6f76 6964 696e 6720 6d75 6c74 6970 6c65  oviding multiple
-00006fc0: 2074 6572 6d73 0a20 2020 2020 2020 2020   terms.         
-00006fd0: 2077 696c 6c20 7265 7475 726e 2072 6573   will return res
-00006fe0: 756c 7473 2063 6f6e 7461 696e 696e 6720  ults containing 
-00006ff0: 616e 7920 6f66 2074 6865 2074 6572 6d73  any of the terms
-00007000: 2061 6e64 2074 6865 6972 2063 6869 6c64   and their child
-00007010: 7265 6e2e 200a 2020 2020 2020 2020 2020  ren. .          
-00007020: 5468 6573 6520 6172 6520 6170 7065 6e64  These are append
-00007030: 6564 2074 6f20 7468 6520 6669 6c74 6572  ed to the filter
-00007040: 2061 6e64 2065 7175 6976 616c 656e 7420   and equivalent 
-00007050: 746f 2066 696c 7465 7269 6e67 2066 6f72  to filtering for
-00007060: 200a 2020 2020 2020 2020 2020 616c 6c43   .          allC
-00007070: 6861 7261 6374 6572 6973 7469 6373 2e76  haracteristics.v
-00007080: 616c 7565 5572 692c 2064 6566 6175 6c74  alueUri, default
-00007090: 7320 746f 204e 6f6e 650a 2020 2020 2020  s to None.      
-000070a0: 2020 3a74 7970 6520 7572 6973 3a20 4f70    :type uris: Op
-000070b0: 7469 6f6e 616c 5b4c 6973 745b 7374 725d  tional[List[str]
-000070c0: 5d2c 206f 7074 696f 6e61 6c0a 2020 2020  ], optional.    
-000070d0: 2020 2020 3a70 6172 616d 206f 6666 7365      :param offse
-000070e0: 743a 2054 6865 206f 6666 7365 7420 6f66  t: The offset of
-000070f0: 2074 6865 2066 6972 7374 2072 6574 7269   the first retri
-00007100: 6576 6564 2072 6573 756c 742e 2c20 6465  eved result., de
-00007110: 6661 756c 7473 2074 6f20 300a 2020 2020  faults to 0.    
-00007120: 2020 2020 3a74 7970 6520 6f66 6673 6574      :type offset
-00007130: 3a20 696e 742c 206f 7074 696f 6e61 6c0a  : int, optional.
-00007140: 2020 2020 2020 2020 3a70 6172 616d 206c          :param l
-00007150: 696d 6974 3a20 4c69 6d69 7473 2074 6865  imit: Limits the
-00007160: 2072 6573 756c 7420 746f 2073 7065 6369   result to speci
-00007170: 6669 6564 2061 6d6f 756e 7420 6f66 206f  fied amount of o
-00007180: 626a 6563 7473 2e0a 2020 2020 2020 2020  bjects..        
-00007190: 2020 4861 7320 6120 6d61 7869 6d75 6d20    Has a maximum 
-000071a0: 7661 6c75 6520 6f66 2031 3030 2e20 5573  value of 100. Us
-000071b0: 6520 746f 6765 7468 6572 2077 6974 6820  e together with 
-000071c0: 6f66 6673 6574 2061 6e64 2074 6865 200a  offset and the .
-000071d0: 2020 2020 2020 2020 2020 746f 7461 6c5f            total_
-000071e0: 656c 656d 656e 7473 2061 7474 7269 6275  elements attribu
-000071f0: 7465 2069 6e20 7468 6520 6f75 7470 7574  te in the output
-00007200: 2074 6f20 636f 6d70 696c 6520 616c 6c20   to compile all 
-00007210: 6461 7461 2069 6620 6e65 6564 6564 2e0a  data if needed..
-00007220: 2020 2020 2020 2020 2020 416c 7465 726e            Altern
-00007230: 6174 6976 656c 7920 6765 745f 616c 6c5f  atively get_all_
-00007240: 7061 6765 7320 6675 6e63 7469 6f6e 2063  pages function c
-00007250: 616e 2062 6520 7573 6564 2077 6974 6820  an be used with 
-00007260: 616c 6c20 6675 6e63 7469 6f6e 730a 2020  all functions.  
-00007270: 2020 2020 2020 2020 696e 636c 7564 696e          includin
-00007280: 6720 6f66 6673 6574 2061 6e64 206c 696d  g offset and lim
-00007290: 6974 2070 6172 616d 6574 6572 732c 2064  it parameters, d
-000072a0: 6566 6175 6c74 7320 746f 2032 300a 2020  efaults to 20.  
-000072b0: 2020 2020 2020 3a74 7970 6520 6c69 6d69        :type limi
-000072c0: 743a 2069 6e74 2c20 6f70 7469 6f6e 616c  t: int, optional
-000072d0: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
-000072e0: 736f 7274 3a20 4f72 6465 7220 7265 7375  sort: Order resu
-000072f0: 6c74 7320 6279 2074 6865 2067 6976 656e  lts by the given
-00007300: 2070 726f 7065 7274 7920 616e 6420 6469   property and di
-00007310: 7265 6374 696f 6e2e 2054 6865 2027 2b27  rection. The '+'
-00007320: 0a20 2020 2020 2020 2020 2073 6967 6e20  .          sign 
-00007330: 696e 6469 6361 7465 2061 7363 656e 6469  indicate ascendi
-00007340: 6e67 206f 7264 6572 2077 6865 7265 6173  ng order whereas
-00007350: 2074 6865 2027 2d27 2069 6e64 6963 6174   the '-' indicat
-00007360: 6520 6465 7363 656e 6469 6e67 2c0a 2020  e descending,.  
-00007370: 2020 2020 2020 2020 6465 6661 756c 7473          defaults
-00007380: 2074 6f20 222b 6964 220a 2020 2020 2020   to "+id".      
-00007390: 2020 3a74 7970 6520 736f 7274 3a20 7374    :type sort: st
-000073a0: 722c 206f 7074 696f 6e61 6c0a 2020 2020  r, optional.    
-000073b0: 2020 2020 3a70 6172 616d 202a 2a6b 7761      :param **kwa
-000073c0: 7267 733a 2041 6464 6974 696f 6e61 6c20  rgs: Additional 
-000073d0: 6172 6775 6d65 6e74 7320 746f 2070 6173  arguments to pas
-000073e0: 7320 746f 2072 6177 2e67 6574 5f64 6174  s to raw.get_dat
-000073f0: 6173 6574 735f 6279 5f69 6473 0a20 2020  asets_by_ids.   
-00007400: 2020 2020 203a 7265 7475 726e 3a20 4120       :return: A 
-00007410: 4461 7461 4672 616d 6520 7769 7468 2069  DataFrame with i
-00007420: 6e66 6f72 6d61 7469 6f6e 2061 626f 7574  nformation about
-00007430: 2074 6865 2071 7565 7269 6564 2064 6174   the queried dat
-00007440: 6173 6574 2873 292e 0a20 2020 2020 2020  aset(s)..       
-00007450: 200a 2020 2020 2020 2020 2020 5468 6520   .          The 
-00007460: 6669 656c 6473 206f 6620 7468 6520 4461  fields of the Da
-00007470: 7461 4672 616d 6520 6172 653a 0a20 2020  taFrame are:.   
-00007480: 2020 2020 2020 2020 202d 2065 7870 6572           - exper
-00007490: 696d 656e 745f 7368 6f72 745f 6e61 6d65  iment_short_name
-000074a0: 3a20 5368 6f72 746e 616d 6520 6769 7665  : Shortname give
-000074b0: 6e20 746f 2074 6865 2064 6174 6173 6574  n to the dataset
-000074c0: 2077 6974 6869 6e20 4765 6d6d 612e 204f   within Gemma. O
-000074d0: 6674 656e 2063 6f72 7265 7370 6f6e 6473  ften corresponds
-000074e0: 2074 6f20 6163 6365 7373 696f 6e20 4944   to accession ID
-000074f0: 0a20 2020 2020 2020 2020 2020 202d 2065  .            - e
-00007500: 7870 6572 696d 656e 745f 6e61 6d65 3a20  xperiment_name: 
-00007510: 4675 6c6c 2074 6974 6c65 206f 6620 7468  Full title of th
-00007520: 6520 6461 7461 7365 740a 2020 2020 2020  e dataset.      
+00006b00: 7461 7861 3a4f 7074 696f 6e61 6c5b 4c69  taxa:Optional[Li
+00006b10: 7374 5b73 7472 5d5d 203d 204e 6f6e 652c  st[str]] = None,
+00006b20: 200a 2020 2020 2020 2020 2020 2020 2020   .              
+00006b30: 2020 2020 2020 2020 2020 2020 2020 7572                ur
+00006b40: 6973 3a4f 7074 696f 6e61 6c5b 4c69 7374  is:Optional[List
+00006b50: 5b73 7472 5d5d 203d 204e 6f6e 652c 0a20  [str]] = None,. 
+00006b60: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00006b70: 2020 2020 2020 2020 2020 206f 6666 7365             offse
+00006b80: 743a 696e 7420 3d20 302c 0a20 2020 2020  t:int = 0,.     
+00006b90: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00006ba0: 2020 2020 2020 206c 696d 6974 3a69 6e74         limit:int
+00006bb0: 203d 2032 302c 0a20 2020 2020 2020 2020   = 20,.         
+00006bc0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00006bd0: 2020 2073 6f72 743a 7374 7220 3d20 222b     sort:str = "+
+00006be0: 6964 222c 0a20 2020 2020 2020 2020 2020  id",.           
+00006bf0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00006c00: 202a 2a6b 7761 7267 7329 2d3e 4461 7461   **kwargs)->Data
+00006c10: 4672 616d 653a 0a20 2020 2020 2020 2022  Frame:.        "
+00006c20: 2222 0a20 2020 2020 2020 200a 2020 2020  "".        .    
+00006c30: 2020 2020 3a70 6172 616d 2064 6174 6173      :param datas
+00006c40: 6574 3a20 4e75 6d65 7269 6361 6c20 6461  et: Numerical da
+00006c50: 7461 7365 7420 6964 656e 7469 6669 6572  taset identifier
+00006c60: 7320 6f72 2064 6174 6173 6574 2073 686f  s or dataset sho
+00006c70: 7274 206e 616d 6573 2e0a 2020 2020 2020  rt names..      
+00006c80: 2020 3a74 7970 6520 6461 7461 7365 743a    :type dataset:
+00006c90: 204c 6973 745b 7374 727c 696e 745d 0a20   List[str|int]. 
+00006ca0: 2020 2020 2020 203a 7061 7261 6d20 6669         :param fi
+00006cb0: 6c74 6572 3a20 4669 6c74 6572 2072 6573  lter: Filter res
+00006cc0: 756c 7473 2062 7920 6d61 7463 6869 6e67  ults by matching
+00006cd0: 2065 7870 7265 7373 696f 6e2e 2055 7365   expression. Use
+00006ce0: 200a 2020 2020 2020 2020 2020 6669 6c74   .          filt
+00006cf0: 6572 5f70 726f 7065 7274 6965 7320 6675  er_properties fu
+00006d00: 6e63 7469 6f6e 2074 6f20 6765 7420 6120  nction to get a 
+00006d10: 6c69 7374 206f 6620 616c 6c20 6176 6169  list of all avai
+00006d20: 6c61 626c 6520 7061 7261 6d65 7465 7273  lable parameters
+00006d30: 2e20 0a20 2020 2020 2020 2020 2054 6865  . .          The
+00006d40: 7365 2070 726f 7065 7274 6965 7320 6361  se properties ca
+00006d50: 6e20 6265 2063 6f6d 6269 6e65 6420 7573  n be combined us
+00006d60: 696e 6720 2261 6e64 2220 226f 7222 2063  ing "and" "or" c
+00006d70: 6c61 7573 6573 2061 6e64 206d 6179 200a  lauses and may .
+00006d80: 2020 2020 2020 2020 2020 636f 6e74 6169            contai
+00006d90: 6e20 636f 6d6d 6f6e 206f 7065 7261 746f  n common operato
+00006da0: 7273 2073 7563 6820 6173 2022 3d22 2c20  rs such as "=", 
+00006db0: 223c 2220 6f72 2022 696e 222e 2028 652e  "<" or "in". (e.
+00006dc0: 672e 200a 2020 2020 2020 2020 2020 2274  g. .          "t
+00006dd0: 6178 6f6e 2e63 6f6d 6d6f 6e4e 616d 6520  axon.commonName 
+00006de0: 3d20 6875 6d61 6e22 2c20 2274 6178 6f6e  = human", "taxon
+00006df0: 2e63 6f6d 6d6f 6e4e 616d 6520 696e 2028  .commonName in (
+00006e00: 6875 6d61 6e2c 6d6f 7573 6529 2c20 0a20  human,mouse), . 
+00006e10: 2020 2020 2020 2020 2022 6964 203c 2031           "id < 1
+00006e20: 3030 3022 292c 2064 6566 6175 6c74 7320  000"), defaults 
+00006e30: 746f 204e 6f6e 650a 2020 2020 2020 2020  to None.        
+00006e40: 3a74 7970 6520 6669 6c74 6572 3a20 4f70  :type filter: Op
+00006e50: 7469 6f6e 616c 5b73 7472 5d2c 206f 7074  tional[str], opt
+00006e60: 696f 6e61 6c0a 2020 2020 2020 2020 3a70  ional.        :p
+00006e70: 6172 616d 2074 6178 613a 2041 206c 6973  aram taxa: A lis
+00006e80: 7420 6f66 2074 6178 6f6e 2063 6f6d 6d6f  t of taxon commo
+00006e90: 6e20 6e61 6d65 7320 2865 2e67 2e20 6875  n names (e.g. hu
+00006ea0: 6d61 6e2c 206d 6f75 7365 2c20 7261 7429  man, mouse, rat)
+00006eb0: 2e20 0a20 2020 2020 2020 2020 2050 726f  . .          Pro
+00006ec0: 7669 6469 6e67 206d 756c 7469 706c 6520  viding multiple 
+00006ed0: 7370 6563 6965 7320 7769 6c6c 2072 6574  species will ret
+00006ee0: 7572 6e20 7265 7375 6c74 7320 666f 7220  urn results for 
+00006ef0: 616c 6c20 7370 6563 6965 732e 200a 2020  all species. .  
+00006f00: 2020 2020 2020 2020 5468 6573 6520 6172          These ar
+00006f10: 6520 6170 7065 6e64 6564 2074 6f20 7468  e appended to th
+00006f20: 6520 6669 6c74 6572 2061 6e64 2065 7175  e filter and equ
+00006f30: 6976 616c 656e 7420 746f 2066 696c 7465  ivalent to filte
+00006f40: 7269 6e67 2066 6f72 200a 2020 2020 2020  ring for .      
+00006f50: 2020 2020 7461 786f 6e2e 636f 6d6d 6f6e      taxon.common
+00006f60: 4e61 6d65 2070 726f 7065 7274 792c 2064  Name property, d
+00006f70: 6566 6175 6c74 7320 746f 204e 6f6e 650a  efaults to None.
+00006f80: 2020 2020 2020 2020 3a74 7970 6520 7461          :type ta
+00006f90: 7861 3a20 4f70 7469 6f6e 616c 5b4c 6973  xa: Optional[Lis
+00006fa0: 745b 7374 725d 5d2c 206f 7074 696f 6e61  t[str]], optiona
+00006fb0: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
+00006fc0: 2075 7269 733a 2041 2076 6563 746f 7220   uris: A vector 
+00006fd0: 6f66 206f 6e74 6f6c 6f67 7920 7465 726d  of ontology term
+00006fe0: 2055 5249 732e 2050 726f 7669 6469 6e67   URIs. Providing
+00006ff0: 206d 756c 7469 706c 6520 7465 726d 730a   multiple terms.
+00007000: 2020 2020 2020 2020 2020 7769 6c6c 2072            will r
+00007010: 6574 7572 6e20 7265 7375 6c74 7320 636f  eturn results co
+00007020: 6e74 6169 6e69 6e67 2061 6e79 206f 6620  ntaining any of 
+00007030: 7468 6520 7465 726d 7320 616e 6420 7468  the terms and th
+00007040: 6569 7220 6368 696c 6472 656e 2e20 0a20  eir children. . 
+00007050: 2020 2020 2020 2020 2054 6865 7365 2061           These a
+00007060: 7265 2061 7070 656e 6465 6420 746f 2074  re appended to t
+00007070: 6865 2066 696c 7465 7220 616e 6420 6571  he filter and eq
+00007080: 7569 7661 6c65 6e74 2074 6f20 6669 6c74  uivalent to filt
+00007090: 6572 696e 6720 666f 7220 0a20 2020 2020  ering for .     
+000070a0: 2020 2020 2061 6c6c 4368 6172 6163 7465       allCharacte
+000070b0: 7269 7374 6963 732e 7661 6c75 6555 7269  ristics.valueUri
+000070c0: 2c20 6465 6661 756c 7473 2074 6f20 4e6f  , defaults to No
+000070d0: 6e65 0a20 2020 2020 2020 203a 7479 7065  ne.        :type
+000070e0: 2075 7269 733a 204f 7074 696f 6e61 6c5b   uris: Optional[
+000070f0: 4c69 7374 5b73 7472 5d5d 2c20 6f70 7469  List[str]], opti
+00007100: 6f6e 616c 0a20 2020 2020 2020 203a 7061  onal.        :pa
+00007110: 7261 6d20 6f66 6673 6574 3a20 5468 6520  ram offset: The 
+00007120: 6f66 6673 6574 206f 6620 7468 6520 6669  offset of the fi
+00007130: 7273 7420 7265 7472 6965 7665 6420 7265  rst retrieved re
+00007140: 7375 6c74 2e2c 2064 6566 6175 6c74 7320  sult., defaults 
+00007150: 746f 2030 0a20 2020 2020 2020 203a 7479  to 0.        :ty
+00007160: 7065 206f 6666 7365 743a 2069 6e74 2c20  pe offset: int, 
+00007170: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
+00007180: 203a 7061 7261 6d20 6c69 6d69 743a 204c   :param limit: L
+00007190: 696d 6974 7320 7468 6520 7265 7375 6c74  imits the result
+000071a0: 2074 6f20 7370 6563 6966 6965 6420 616d   to specified am
+000071b0: 6f75 6e74 206f 6620 6f62 6a65 6374 732e  ount of objects.
+000071c0: 0a20 2020 2020 2020 2020 2048 6173 2061  .          Has a
+000071d0: 206d 6178 696d 756d 2076 616c 7565 206f   maximum value o
+000071e0: 6620 3130 302e 2055 7365 2074 6f67 6574  f 100. Use toget
+000071f0: 6865 7220 7769 7468 206f 6666 7365 7420  her with offset 
+00007200: 616e 6420 7468 6520 0a20 2020 2020 2020  and the .       
+00007210: 2020 2074 6f74 616c 5f65 6c65 6d65 6e74     total_element
+00007220: 7320 6174 7472 6962 7574 6520 696e 2074  s attribute in t
+00007230: 6865 206f 7574 7075 7420 746f 2063 6f6d  he output to com
+00007240: 7069 6c65 2061 6c6c 2064 6174 6120 6966  pile all data if
+00007250: 206e 6565 6465 642e 0a20 2020 2020 2020   needed..       
+00007260: 2020 2041 6c74 6572 6e61 7469 7665 6c79     Alternatively
+00007270: 2067 6574 5f61 6c6c 5f70 6167 6573 2066   get_all_pages f
+00007280: 756e 6374 696f 6e20 6361 6e20 6265 2075  unction can be u
+00007290: 7365 6420 7769 7468 2061 6c6c 2066 756e  sed with all fun
+000072a0: 6374 696f 6e73 0a20 2020 2020 2020 2020  ctions.         
+000072b0: 2069 6e63 6c75 6469 6e67 206f 6666 7365   including offse
+000072c0: 7420 616e 6420 6c69 6d69 7420 7061 7261  t and limit para
+000072d0: 6d65 7465 7273 2c20 6465 6661 756c 7473  meters, defaults
+000072e0: 2074 6f20 3230 0a20 2020 2020 2020 203a   to 20.        :
+000072f0: 7479 7065 206c 696d 6974 3a20 696e 742c  type limit: int,
+00007300: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
+00007310: 2020 3a70 6172 616d 2073 6f72 743a 204f    :param sort: O
+00007320: 7264 6572 2072 6573 756c 7473 2062 7920  rder results by 
+00007330: 7468 6520 6769 7665 6e20 7072 6f70 6572  the given proper
+00007340: 7479 2061 6e64 2064 6972 6563 7469 6f6e  ty and direction
+00007350: 2e20 5468 6520 272b 270a 2020 2020 2020  . The '+'.      
+00007360: 2020 2020 7369 676e 2069 6e64 6963 6174      sign indicat
+00007370: 6520 6173 6365 6e64 696e 6720 6f72 6465  e ascending orde
+00007380: 7220 7768 6572 6561 7320 7468 6520 272d  r whereas the '-
+00007390: 2720 696e 6469 6361 7465 2064 6573 6365  ' indicate desce
+000073a0: 6e64 696e 672c 0a20 2020 2020 2020 2020  nding,.         
+000073b0: 2064 6566 6175 6c74 7320 746f 2022 2b69   defaults to "+i
+000073c0: 6422 0a20 2020 2020 2020 203a 7479 7065  d".        :type
+000073d0: 2073 6f72 743a 2073 7472 2c20 6f70 7469   sort: str, opti
+000073e0: 6f6e 616c 0a20 2020 2020 2020 203a 7061  onal.        :pa
+000073f0: 7261 6d20 2a2a 6b77 6172 6773 3a20 4164  ram **kwargs: Ad
+00007400: 6469 7469 6f6e 616c 2061 7267 756d 656e  ditional argumen
+00007410: 7473 2074 6f20 7061 7373 2074 6f20 7261  ts to pass to ra
+00007420: 772e 6765 745f 6461 7461 7365 7473 5f62  w.get_datasets_b
+00007430: 795f 6964 730a 2020 2020 2020 2020 3a72  y_ids.        :r
+00007440: 6574 7572 6e3a 2041 2044 6174 6146 7261  eturn: A DataFra
+00007450: 6d65 2077 6974 6820 696e 666f 726d 6174  me with informat
+00007460: 696f 6e20 6162 6f75 7420 7468 6520 7175  ion about the qu
+00007470: 6572 6965 6420 6461 7461 7365 7428 7329  eried dataset(s)
+00007480: 2e0a 2020 2020 2020 2020 0a20 2020 2020  ..        .     
+00007490: 2020 2020 2054 6865 2066 6965 6c64 7320       The fields 
+000074a0: 6f66 2074 6865 2044 6174 6146 7261 6d65  of the DataFrame
+000074b0: 2061 7265 3a0a 2020 2020 2020 2020 2020   are:.          
+000074c0: 2020 2d20 6578 7065 7269 6d65 6e74 5f73    - experiment_s
+000074d0: 686f 7274 5f6e 616d 653a 2053 686f 7274  hort_name: Short
+000074e0: 6e61 6d65 2067 6976 656e 2074 6f20 7468  name given to th
+000074f0: 6520 6461 7461 7365 7420 7769 7468 696e  e dataset within
+00007500: 2047 656d 6d61 2e20 4f66 7465 6e20 636f   Gemma. Often co
+00007510: 7272 6573 706f 6e64 7320 746f 2061 6363  rresponds to acc
+00007520: 6573 7369 6f6e 2049 440a 2020 2020 2020  ession ID.      
 00007530: 2020 2020 2020 2d20 6578 7065 7269 6d65        - experime
-00007540: 6e74 5f49 443a 2049 6e74 6572 6e61 6c20  nt_ID: Internal 
-00007550: 4944 206f 6620 7468 6520 6461 7461 7365  ID of the datase
-00007560: 742e 0a20 2020 2020 2020 2020 2020 202d  t..            -
-00007570: 2065 7870 6572 696d 656e 745f 6465 7363   experiment_desc
-00007580: 7269 7074 696f 6e3a 2044 6573 6372 6970  ription: Descrip
-00007590: 7469 6f6e 206f 6620 7468 6520 6461 7461  tion of the data
-000075a0: 7365 740a 2020 2020 2020 2020 2020 2020  set.            
-000075b0: 2d20 6578 7065 7269 6d65 6e74 5f74 726f  - experiment_tro
-000075c0: 7562 6c65 643a 2044 6964 2061 6e20 6175  ubled: Did an au
-000075d0: 746f 6d61 7469 6320 7072 6f63 6573 7320  tomatic process 
-000075e0: 7769 7468 696e 2067 656d 6d61 206f 7220  within gemma or 
-000075f0: 6120 6375 7261 746f 7220 6d61 726b 2074  a curator mark t
-00007600: 6865 2064 6174 6173 6574 2061 7320 2274  he dataset as "t
-00007610: 726f 7562 6c65 6422 0a20 2020 2020 2020  roubled".       
-00007620: 2020 2020 202d 2065 7870 6572 696d 656e       - experimen
-00007630: 745f 6163 6365 7373 696f 6e3a 2041 6363  t_accession: Acc
-00007640: 6573 7369 6f6e 2049 4420 6f66 2074 6865  ession ID of the
-00007650: 2064 6174 6173 6574 2069 6e20 7468 6520   dataset in the 
-00007660: 6578 7465 726e 616c 2064 6174 6162 6173  external databas
-00007670: 6520 6974 2077 6173 2074 616b 656e 2066  e it was taken f
-00007680: 726f 6d0a 2020 2020 2020 2020 2020 2020  rom.            
-00007690: 2d20 6578 7065 7269 6d65 6e74 5f64 6174  - experiment_dat
-000076a0: 6162 6173 653a 2054 6865 206e 616d 6520  abase: The name 
-000076b0: 6f66 2074 6865 2064 6174 6162 6173 6520  of the database 
-000076c0: 7768 6572 6520 7468 6520 6461 7461 7365  where the datase
-000076d0: 7420 7761 7320 7461 6b65 6e20 6672 6f6d  t was taken from
-000076e0: 0a20 2020 2020 2020 2020 2020 202d 2065  .            - e
-000076f0: 7870 6572 696d 656e 745f 5552 493a 2055  xperiment_URI: U
-00007700: 5249 206f 6620 7468 6520 6f72 6967 696e  RI of the origin
-00007710: 616c 2064 6174 6162 6173 650a 2020 2020  al database.    
-00007720: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
-00007730: 6d65 6e74 5f73 616d 706c 655f 636f 756e  ment_sample_coun
-00007740: 743a 204e 756d 6265 7220 6f66 2073 616d  t: Number of sam
-00007750: 706c 6573 2069 6e20 7468 6520 6461 7461  ples in the data
-00007760: 7365 740a 2020 2020 2020 2020 2020 2020  set.            
-00007770: 2d20 6578 7065 7269 6d65 6e74 5f62 6174  - experiment_bat
-00007780: 6368 5f65 6666 6563 745f 7465 7874 3a20  ch_effect_text: 
-00007790: 4120 7465 7874 2066 6965 6c64 2064 6573  A text field des
-000077a0: 6372 6962 696e 6720 7768 6574 6865 7220  cribing whether 
-000077b0: 7468 6520 6461 7461 7365 7420 6861 7320  the dataset has 
-000077c0: 6261 7463 6820 6566 6665 6374 730a 2020  batch effects.  
-000077d0: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
-000077e0: 7269 6d65 6e5f 6261 7463 685f 636f 7272  rimen_batch_corr
-000077f0: 6563 7465 643a 2057 6865 7468 6572 2062  ected: Whether b
-00007800: 6174 6368 2063 6f72 7265 6374 696f 6e20  atch correction 
-00007810: 6861 7320 6265 656e 2070 6572 666f 726d  has been perform
-00007820: 6564 206f 6e20 7468 6520 6461 7461 7365  ed on the datase
-00007830: 742e 0a20 2020 2020 2020 2020 2020 202d  t..            -
-00007840: 2065 7870 6572 696d 656e 5f62 6174 6368   experimen_batch
-00007850: 5f63 6f6e 666f 756e 643a 2030 2069 6620  _confound: 0 if 
-00007860: 6261 7463 6820 696e 666f 2069 736e 2774  batch info isn't
-00007870: 2061 7661 696c 6162 6c65 2c20 2d31 2069   available, -1 i
-00007880: 6620 6261 7463 6820 636f 756e 666f 7564  f batch counfoud
-00007890: 2069 7320 6465 7465 6374 6564 2c20 3120   is detected, 1 
-000078a0: 6966 2062 6174 6368 2069 6e66 6f72 6d61  if batch informa
-000078b0: 7469 6f6e 2069 7320 6176 6169 6c61 626c  tion is availabl
-000078c0: 6520 616e 6420 6e6f 2062 6174 6368 2063  e and no batch c
-000078d0: 6f6e 666f 756e 6420 666f 756e 640a 2020  onfound found.  
-000078e0: 2020 2020 2020 2020 2020 2d20 6578 7065            - expe
-000078f0: 7269 6d65 6e5f 6261 7463 685f 6566 6665  rimen_batch_effe
-00007900: 6374 3a20 2d31 2069 6620 6261 7463 6820  ct: -1 if batch 
-00007910: 7020 7661 6c75 6520 3c20 302e 3030 3031  p value < 0.0001
-00007920: 2c20 3120 6966 2062 6174 6368 2070 2076  , 1 if batch p v
-00007930: 616c 7565 203e 2030 2e31 2c20 3020 6966  alue > 0.1, 0 if
-00007940: 206f 7468 6572 7769 7365 2061 6e64 2077   otherwise and w
-00007950: 6865 6e20 7468 6572 6520 6973 206e 6f20  hen there is no 
-00007960: 6261 7463 6820 696e 666f 726d 6174 696f  batch informatio
-00007970: 6e20 6973 2061 7661 696c 6162 6c65 206f  n is available o
-00007980: 7220 7768 656e 2074 6865 2064 6174 6120  r when the data 
-00007990: 6973 2063 6f6e 666f 756e 6465 6420 7769  is confounded wi
-000079a0: 7468 2062 6174 6368 6573 2e0a 2020 2020  th batches..    
-000079b0: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
-000079c0: 6d65 6e5f 7261 775f 6461 7461 3a20 2d31  men_raw_data: -1
-000079d0: 2069 6620 6e6f 2072 6177 2064 6174 6120   if no raw data 
-000079e0: 6176 6169 6c61 626c 652c 2031 2069 6620  available, 1 if 
-000079f0: 7261 7720 6461 7461 2077 6173 2061 7661  raw data was ava
-00007a00: 696c 6162 6c65 2e20 5768 656e 2061 7661  ilable. When ava
-00007a10: 696c 6162 6c65 2c20 4765 6d6d 6120 7265  ilable, Gemma re
-00007a20: 7072 6f63 6573 7365 7320 7261 7720 6461  processes raw da
-00007a30: 7461 2074 6f20 6765 7420 6578 7072 6573  ta to get expres
-00007a40: 7369 6f6e 2076 616c 7565 7320 616e 6420  sion values and 
-00007a50: 6261 7463 6865 730a 2020 2020 2020 2020  batches.        
-00007a60: 2020 2020 2d20 6765 6571 5f71 5f73 636f      - geeq_q_sco
-00007a70: 7265 3a20 4461 7461 2071 7561 6c69 7479  re: Data quality
-00007a80: 2073 636f 7265 2067 6976 656e 2074 6f20   score given to 
-00007a90: 7468 6520 6461 7461 7365 7420 6279 2047  the dataset by G
-00007aa0: 656d 6d61 2e0a 2020 2020 2020 2020 2020  emma..          
-00007ab0: 2020 2d20 6765 6571 5f73 5f73 636f 7265    - geeq_s_score
-00007ac0: 3a20 5375 6974 6162 696c 6974 7920 7363  : Suitability sc
-00007ad0: 6f72 6520 6769 7665 6e20 746f 2074 6865  ore given to the
-00007ae0: 2064 6174 6173 6574 2062 7920 4765 6d6d   dataset by Gemm
-00007af0: 612e 2052 6566 6572 7320 746f 2066 6163  a. Refers to fac
-00007b00: 746f 7273 206c 696b 6520 6261 7463 6865  tors like batche
-00007b10: 732c 2070 6c61 7466 6f72 6d73 2061 6e64  s, platforms and
-00007b20: 206f 7468 6572 2061 7370 6563 7473 206f   other aspects o
-00007b30: 6620 6578 7065 7269 6d65 6e74 616c 2064  f experimental d
-00007b40: 6573 6967 6e0a 2020 2020 2020 2020 2020  esign.          
-00007b50: 2020 2d20 7461 786f 6e5f 6e61 6d65 3a20    - taxon_name: 
-00007b60: 4e61 6d65 206f 6620 7468 6520 7370 6563  Name of the spec
-00007b70: 6965 730a 2020 2020 2020 2020 2020 2020  ies.            
-00007b80: 2d20 7461 786f 6e5f 7363 6965 6e74 6966  - taxon_scientif
-00007b90: 6963 3a20 5363 6965 6e74 6966 6963 206e  ic: Scientific n
-00007ba0: 616d 6520 666f 7220 7468 6520 7461 786f  ame for the taxo
-00007bb0: 6e0a 2020 2020 2020 2020 2020 2020 2d20  n.            - 
-00007bc0: 7461 786f 6e5f 4944 3a20 496e 7465 726e  taxon_ID: Intern
-00007bd0: 616c 2069 6465 6e74 6966 6965 7220 6769  al identifier gi
-00007be0: 7665 6e20 746f 2074 6865 2073 7065 6369  ven to the speci
-00007bf0: 6573 2062 7920 4765 6d6d 610a 2020 2020  es by Gemma.    
-00007c00: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
-00007c10: 4e43 4249 3a20 4e43 4249 2049 4420 6f66  NCBI: NCBI ID of
-00007c20: 2074 6865 2074 6178 6f6e 0a20 2020 2020   the taxon.     
-00007c30: 2020 2020 2020 202d 2074 6178 6f6e 5f64         - taxon_d
-00007c40: 6174 6162 6173 655f 6e61 6d65 3a20 556e  atabase_name: Un
-00007c50: 6465 726c 7969 6e67 2064 6174 6162 6173  derlying databas
-00007c60: 6520 7573 6564 2069 6e20 4765 6d6d 6120  e used in Gemma 
-00007c70: 666f 7220 7468 6520 7461 786f 6e0a 2020  for the taxon.  
-00007c80: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
-00007c90: 6e5f 6461 7461 6261 7365 5f49 443a 2049  n_database_ID: I
-00007ca0: 4420 6f66 2074 6865 2075 6e64 6572 796c  D of the underyl
-00007cb0: 696e 6720 6461 7461 6261 7365 2075 7365  ing database use
-00007cc0: 6420 696e 2047 656d 6d61 2066 6f72 2074  d in Gemma for t
-00007cd0: 6865 2074 6178 6f6e 0a20 2020 2020 2020  he taxon.       
-00007ce0: 203a 7274 7970 653a 2044 6174 6146 7261   :rtype: DataFra
-00007cf0: 6d65 0a0a 2020 2020 2020 2020 2222 220a  me..        """.
-00007d00: 0a20 2020 2020 2020 2066 696c 7465 7220  .        filter 
-00007d10: 3d20 7673 2e61 6464 5f74 6f5f 6669 6c74  = vs.add_to_filt
-00007d20: 6572 2866 696c 7465 722c 2027 616c 6c43  er(filter, 'allC
-00007d30: 6861 7261 6374 6572 6973 7469 6373 2e76  haracteristics.v
-00007d40: 616c 7565 5572 6927 2c20 7572 6973 290a  alueUri', uris).
-00007d50: 2020 2020 2020 2020 6669 6c74 6572 203d          filter =
-00007d60: 2076 732e 6164 645f 746f 5f66 696c 7465   vs.add_to_filte
-00007d70: 7228 6669 6c74 6572 2c20 2774 6178 6f6e  r(filter, 'taxon
-00007d80: 2e63 6f6d 6d6f 6e4e 616d 6527 2c20 7461  .commonName', ta
-00007d90: 7861 290a 2020 2020 2020 2020 0a20 2020  xa).        .   
-00007da0: 2020 2020 200a 2020 2020 2020 2020 6b77       .        kw
-00007db0: 6172 6773 203d 2076 732e 7265 6d6f 7665  args = vs.remove
-00007dc0: 5f6e 6f6e 6573 280a 2020 2020 2020 2020  _nones(.        
-00007dd0: 2020 2020 6669 6c74 6572 203d 2066 696c      filter = fil
-00007de0: 7465 722c 0a20 2020 2020 2020 2020 2020  ter,.           
-00007df0: 206f 6666 7365 7420 3d20 6f66 6673 6574   offset = offset
-00007e00: 2c0a 2020 2020 2020 2020 2020 2020 6c69  ,.            li
-00007e10: 6d69 7420 3d20 6c69 6d69 742c 0a20 2020  mit = limit,.   
-00007e20: 2020 2020 2020 2020 2073 6f72 7420 3d20           sort = 
-00007e30: 736f 7274 2c0a 2020 2020 2020 2020 2020  sort,.          
-00007e40: 2020 2a2a 6b77 6172 6773 290a 2020 2020    **kwargs).    
-00007e50: 2020 2020 0a20 2020 2020 2020 2072 6573      .        res
-00007e60: 706f 6e73 6520 3d20 7365 6c66 2e72 6177  ponse = self.raw
-00007e70: 2e67 6574 5f64 6174 6173 6574 735f 6279  .get_datasets_by
-00007e80: 5f69 6473 2864 6174 6173 6574 2c20 2a2a  _ids(dataset, **
-00007e90: 6b77 6172 6773 290a 2020 2020 2020 2020  kwargs).        
-00007ea0: 6466 203d 2070 732e 7072 6f63 6573 735f  df = ps.process_
-00007eb0: 6461 7461 7365 7473 2872 6573 706f 6e73  datasets(respons
-00007ec0: 652e 6461 7461 290a 2020 2020 2020 2020  e.data).        
-00007ed0: 7073 2e61 7474 6163 685f 6174 7472 6962  ps.attach_attrib
-00007ee0: 7574 6573 2864 662c 2072 6573 706f 6e73  utes(df, respons
-00007ef0: 652e 746f 5f64 6963 7428 2929 0a20 2020  e.to_dict()).   
-00007f00: 2020 2020 200a 2020 2020 2020 2020 7265       .        re
-00007f10: 7475 726e 2064 660a 0a20 2020 2023 2064  turn df..    # d
-00007f20: 6174 6173 6574 732f 6361 7465 676f 7269  atasets/categori
-00007f30: 6573 202d 2d2d 2d2d 0a20 2020 2023 2063  es -----.    # c
-00007f40: 7572 7265 6e74 6c79 2075 6e69 6d70 6c65  urrently unimple
-00007f50: 6d65 6e74 6564 0a0a 2020 2020 2320 6461  mented..    # da
-00007f60: 7461 7365 7473 2f74 6178 6120 2d2d 2d2d  tasets/taxa ----
+00007540: 6e74 5f6e 616d 653a 2046 756c 6c20 7469  nt_name: Full ti
+00007550: 746c 6520 6f66 2074 6865 2064 6174 6173  tle of the datas
+00007560: 6574 0a20 2020 2020 2020 2020 2020 202d  et.            -
+00007570: 2065 7870 6572 696d 656e 745f 4944 3a20   experiment_ID: 
+00007580: 496e 7465 726e 616c 2049 4420 6f66 2074  Internal ID of t
+00007590: 6865 2064 6174 6173 6574 2e0a 2020 2020  he dataset..    
+000075a0: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
+000075b0: 6d65 6e74 5f64 6573 6372 6970 7469 6f6e  ment_description
+000075c0: 3a20 4465 7363 7269 7074 696f 6e20 6f66  : Description of
+000075d0: 2074 6865 2064 6174 6173 6574 0a20 2020   the dataset.   
+000075e0: 2020 2020 2020 2020 202d 2065 7870 6572           - exper
+000075f0: 696d 656e 745f 7472 6f75 626c 6564 3a20  iment_troubled: 
+00007600: 4469 6420 616e 2061 7574 6f6d 6174 6963  Did an automatic
+00007610: 2070 726f 6365 7373 2077 6974 6869 6e20   process within 
+00007620: 6765 6d6d 6120 6f72 2061 2063 7572 6174  gemma or a curat
+00007630: 6f72 206d 6172 6b20 7468 6520 6461 7461  or mark the data
+00007640: 7365 7420 6173 2022 7472 6f75 626c 6564  set as "troubled
+00007650: 220a 2020 2020 2020 2020 2020 2020 2d20  ".            - 
+00007660: 6578 7065 7269 6d65 6e74 5f61 6363 6573  experiment_acces
+00007670: 7369 6f6e 3a20 4163 6365 7373 696f 6e20  sion: Accession 
+00007680: 4944 206f 6620 7468 6520 6461 7461 7365  ID of the datase
+00007690: 7420 696e 2074 6865 2065 7874 6572 6e61  t in the externa
+000076a0: 6c20 6461 7461 6261 7365 2069 7420 7761  l database it wa
+000076b0: 7320 7461 6b65 6e20 6672 6f6d 0a20 2020  s taken from.   
+000076c0: 2020 2020 2020 2020 202d 2065 7870 6572           - exper
+000076d0: 696d 656e 745f 6461 7461 6261 7365 3a20  iment_database: 
+000076e0: 5468 6520 6e61 6d65 206f 6620 7468 6520  The name of the 
+000076f0: 6461 7461 6261 7365 2077 6865 7265 2074  database where t
+00007700: 6865 2064 6174 6173 6574 2077 6173 2074  he dataset was t
+00007710: 616b 656e 2066 726f 6d0a 2020 2020 2020  aken from.      
+00007720: 2020 2020 2020 2d20 6578 7065 7269 6d65        - experime
+00007730: 6e74 5f55 5249 3a20 5552 4920 6f66 2074  nt_URI: URI of t
+00007740: 6865 206f 7269 6769 6e61 6c20 6461 7461  he original data
+00007750: 6261 7365 0a20 2020 2020 2020 2020 2020  base.           
+00007760: 202d 2065 7870 6572 696d 656e 745f 7361   - experiment_sa
+00007770: 6d70 6c65 5f63 6f75 6e74 3a20 4e75 6d62  mple_count: Numb
+00007780: 6572 206f 6620 7361 6d70 6c65 7320 696e  er of samples in
+00007790: 2074 6865 2064 6174 6173 6574 0a20 2020   the dataset.   
+000077a0: 2020 2020 2020 2020 202d 2065 7870 6572           - exper
+000077b0: 696d 656e 745f 6261 7463 685f 6566 6665  iment_batch_effe
+000077c0: 6374 5f74 6578 743a 2041 2074 6578 7420  ct_text: A text 
+000077d0: 6669 656c 6420 6465 7363 7269 6269 6e67  field describing
+000077e0: 2077 6865 7468 6572 2074 6865 2064 6174   whether the dat
+000077f0: 6173 6574 2068 6173 2062 6174 6368 2065  aset has batch e
+00007800: 6666 6563 7473 0a20 2020 2020 2020 2020  ffects.         
+00007810: 2020 202d 2065 7870 6572 696d 656e 5f62     - experimen_b
+00007820: 6174 6368 5f63 6f72 7265 6374 6564 3a20  atch_corrected: 
+00007830: 5768 6574 6865 7220 6261 7463 6820 636f  Whether batch co
+00007840: 7272 6563 7469 6f6e 2068 6173 2062 6565  rrection has bee
+00007850: 6e20 7065 7266 6f72 6d65 6420 6f6e 2074  n performed on t
+00007860: 6865 2064 6174 6173 6574 2e0a 2020 2020  he dataset..    
+00007870: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
+00007880: 6d65 6e5f 6261 7463 685f 636f 6e66 6f75  men_batch_confou
+00007890: 6e64 3a20 3020 6966 2062 6174 6368 2069  nd: 0 if batch i
+000078a0: 6e66 6f20 6973 6e27 7420 6176 6169 6c61  nfo isn't availa
+000078b0: 626c 652c 202d 3120 6966 2062 6174 6368  ble, -1 if batch
+000078c0: 2063 6f75 6e66 6f75 6420 6973 2064 6574   counfoud is det
+000078d0: 6563 7465 642c 2031 2069 6620 6261 7463  ected, 1 if batc
+000078e0: 6820 696e 666f 726d 6174 696f 6e20 6973  h information is
+000078f0: 2061 7661 696c 6162 6c65 2061 6e64 206e   available and n
+00007900: 6f20 6261 7463 6820 636f 6e66 6f75 6e64  o batch confound
+00007910: 2066 6f75 6e64 0a20 2020 2020 2020 2020   found.         
+00007920: 2020 202d 2065 7870 6572 696d 656e 5f62     - experimen_b
+00007930: 6174 6368 5f65 6666 6563 743a 202d 3120  atch_effect: -1 
+00007940: 6966 2062 6174 6368 2070 2076 616c 7565  if batch p value
+00007950: 203c 2030 2e30 3030 312c 2031 2069 6620   < 0.0001, 1 if 
+00007960: 6261 7463 6820 7020 7661 6c75 6520 3e20  batch p value > 
+00007970: 302e 312c 2030 2069 6620 6f74 6865 7277  0.1, 0 if otherw
+00007980: 6973 6520 616e 6420 7768 656e 2074 6865  ise and when the
+00007990: 7265 2069 7320 6e6f 2062 6174 6368 2069  re is no batch i
+000079a0: 6e66 6f72 6d61 7469 6f6e 2069 7320 6176  nformation is av
+000079b0: 6169 6c61 626c 6520 6f72 2077 6865 6e20  ailable or when 
+000079c0: 7468 6520 6461 7461 2069 7320 636f 6e66  the data is conf
+000079d0: 6f75 6e64 6564 2077 6974 6820 6261 7463  ounded with batc
+000079e0: 6865 732e 0a20 2020 2020 2020 2020 2020  hes..           
+000079f0: 202d 2065 7870 6572 696d 656e 5f72 6177   - experimen_raw
+00007a00: 5f64 6174 613a 202d 3120 6966 206e 6f20  _data: -1 if no 
+00007a10: 7261 7720 6461 7461 2061 7661 696c 6162  raw data availab
+00007a20: 6c65 2c20 3120 6966 2072 6177 2064 6174  le, 1 if raw dat
+00007a30: 6120 7761 7320 6176 6169 6c61 626c 652e  a was available.
+00007a40: 2057 6865 6e20 6176 6169 6c61 626c 652c   When available,
+00007a50: 2047 656d 6d61 2072 6570 726f 6365 7373   Gemma reprocess
+00007a60: 6573 2072 6177 2064 6174 6120 746f 2067  es raw data to g
+00007a70: 6574 2065 7870 7265 7373 696f 6e20 7661  et expression va
+00007a80: 6c75 6573 2061 6e64 2062 6174 6368 6573  lues and batches
+00007a90: 0a20 2020 2020 2020 2020 2020 202d 2067  .            - g
+00007aa0: 6565 715f 715f 7363 6f72 653a 2044 6174  eeq_q_score: Dat
+00007ab0: 6120 7175 616c 6974 7920 7363 6f72 6520  a quality score 
+00007ac0: 6769 7665 6e20 746f 2074 6865 2064 6174  given to the dat
+00007ad0: 6173 6574 2062 7920 4765 6d6d 612e 0a20  aset by Gemma.. 
+00007ae0: 2020 2020 2020 2020 2020 202d 2067 6565             - gee
+00007af0: 715f 735f 7363 6f72 653a 2053 7569 7461  q_s_score: Suita
+00007b00: 6269 6c69 7479 2073 636f 7265 2067 6976  bility score giv
+00007b10: 656e 2074 6f20 7468 6520 6461 7461 7365  en to the datase
+00007b20: 7420 6279 2047 656d 6d61 2e20 5265 6665  t by Gemma. Refe
+00007b30: 7273 2074 6f20 6661 6374 6f72 7320 6c69  rs to factors li
+00007b40: 6b65 2062 6174 6368 6573 2c20 706c 6174  ke batches, plat
+00007b50: 666f 726d 7320 616e 6420 6f74 6865 7220  forms and other 
+00007b60: 6173 7065 6374 7320 6f66 2065 7870 6572  aspects of exper
+00007b70: 696d 656e 7461 6c20 6465 7369 676e 0a20  imental design. 
+00007b80: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
+00007b90: 6f6e 5f6e 616d 653a 204e 616d 6520 6f66  on_name: Name of
+00007ba0: 2074 6865 2073 7065 6369 6573 0a20 2020   the species.   
+00007bb0: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
+00007bc0: 5f73 6369 656e 7469 6669 633a 2053 6369  _scientific: Sci
+00007bd0: 656e 7469 6669 6320 6e61 6d65 2066 6f72  entific name for
+00007be0: 2074 6865 2074 6178 6f6e 0a20 2020 2020   the taxon.     
+00007bf0: 2020 2020 2020 202d 2074 6178 6f6e 5f49         - taxon_I
+00007c00: 443a 2049 6e74 6572 6e61 6c20 6964 656e  D: Internal iden
+00007c10: 7469 6669 6572 2067 6976 656e 2074 6f20  tifier given to 
+00007c20: 7468 6520 7370 6563 6965 7320 6279 2047  the species by G
+00007c30: 656d 6d61 0a20 2020 2020 2020 2020 2020  emma.           
+00007c40: 202d 2074 6178 6f6e 5f4e 4342 493a 204e   - taxon_NCBI: N
+00007c50: 4342 4920 4944 206f 6620 7468 6520 7461  CBI ID of the ta
+00007c60: 786f 6e0a 2020 2020 2020 2020 2020 2020  xon.            
+00007c70: 2d20 7461 786f 6e5f 6461 7461 6261 7365  - taxon_database
+00007c80: 5f6e 616d 653a 2055 6e64 6572 6c79 696e  _name: Underlyin
+00007c90: 6720 6461 7461 6261 7365 2075 7365 6420  g database used 
+00007ca0: 696e 2047 656d 6d61 2066 6f72 2074 6865  in Gemma for the
+00007cb0: 2074 6178 6f6e 0a20 2020 2020 2020 2020   taxon.         
+00007cc0: 2020 202d 2074 6178 6f6e 5f64 6174 6162     - taxon_datab
+00007cd0: 6173 655f 4944 3a20 4944 206f 6620 7468  ase_ID: ID of th
+00007ce0: 6520 756e 6465 7279 6c69 6e67 2064 6174  e underyling dat
+00007cf0: 6162 6173 6520 7573 6564 2069 6e20 4765  abase used in Ge
+00007d00: 6d6d 6120 666f 7220 7468 6520 7461 786f  mma for the taxo
+00007d10: 6e0a 2020 2020 2020 2020 3a72 7479 7065  n.        :rtype
+00007d20: 3a20 4461 7461 4672 616d 650a 0a20 2020  : DataFrame..   
+00007d30: 2020 2020 2022 2222 0a0a 2020 2020 2020       """..      
+00007d40: 2020 6669 6c74 6572 203d 2076 732e 6164    filter = vs.ad
+00007d50: 645f 746f 5f66 696c 7465 7228 6669 6c74  d_to_filter(filt
+00007d60: 6572 2c20 2761 6c6c 4368 6172 6163 7465  er, 'allCharacte
+00007d70: 7269 7374 6963 732e 7661 6c75 6555 7269  ristics.valueUri
+00007d80: 272c 2075 7269 7329 0a20 2020 2020 2020  ', uris).       
+00007d90: 2066 696c 7465 7220 3d20 7673 2e61 6464   filter = vs.add
+00007da0: 5f74 6f5f 6669 6c74 6572 2866 696c 7465  _to_filter(filte
+00007db0: 722c 2027 7461 786f 6e2e 636f 6d6d 6f6e  r, 'taxon.common
+00007dc0: 4e61 6d65 272c 2074 6178 6129 0a20 2020  Name', taxa).   
+00007dd0: 2020 2020 200a 2020 2020 2020 2020 0a20       .        . 
+00007de0: 2020 2020 2020 206b 7761 7267 7320 3d20         kwargs = 
+00007df0: 7673 2e72 656d 6f76 655f 6e6f 6e65 7328  vs.remove_nones(
+00007e00: 0a20 2020 2020 2020 2020 2020 2066 696c  .            fil
+00007e10: 7465 7220 3d20 6669 6c74 6572 2c0a 2020  ter = filter,.  
+00007e20: 2020 2020 2020 2020 2020 6f66 6673 6574            offset
+00007e30: 203d 206f 6666 7365 742c 0a20 2020 2020   = offset,.     
+00007e40: 2020 2020 2020 206c 696d 6974 203d 206c         limit = l
+00007e50: 696d 6974 2c0a 2020 2020 2020 2020 2020  imit,.          
+00007e60: 2020 736f 7274 203d 2073 6f72 742c 0a20    sort = sort,. 
+00007e70: 2020 2020 2020 2020 2020 202a 2a6b 7761             **kwa
+00007e80: 7267 7329 0a20 2020 2020 2020 200a 2020  rgs).        .  
+00007e90: 2020 2020 2020 7265 7370 6f6e 7365 203d        response =
+00007ea0: 2073 656c 662e 7261 772e 6765 745f 6461   self.raw.get_da
+00007eb0: 7461 7365 7473 5f62 795f 6964 7328 6461  tasets_by_ids(da
+00007ec0: 7461 7365 742c 202a 2a6b 7761 7267 7329  taset, **kwargs)
+00007ed0: 0a20 2020 2020 2020 2064 6620 3d20 7073  .        df = ps
+00007ee0: 2e70 726f 6365 7373 5f64 6174 6173 6574  .process_dataset
+00007ef0: 7328 7265 7370 6f6e 7365 2e64 6174 6129  s(response.data)
+00007f00: 0a20 2020 2020 2020 2070 732e 6174 7461  .        ps.atta
+00007f10: 6368 5f61 7474 7269 6275 7465 7328 6466  ch_attributes(df
+00007f20: 2c20 7265 7370 6f6e 7365 2e74 6f5f 6469  , response.to_di
+00007f30: 6374 2829 290a 2020 2020 2020 2020 0a20  ct()).        . 
+00007f40: 2020 2020 2020 2072 6574 7572 6e20 6466         return df
+00007f50: 0a0a 2020 2020 2320 6461 7461 7365 7473  ..    # datasets
+00007f60: 2f63 6174 6567 6f72 6965 7320 2d2d 2d2d  /categories ----
 00007f70: 2d0a 2020 2020 2320 6375 7272 656e 746c  -.    # currentl
 00007f80: 7920 756e 696d 706c 656d 656e 7465 640a  y unimplemented.
 00007f90: 0a20 2020 2023 2064 6174 6173 6574 732f  .    # datasets/
-00007fa0: 636f 756e 7420 2d2d 2d2d 2d0a 2020 2020  count -----.    
-00007fb0: 2320 6375 7272 656e 746c 7920 756e 696d  # currently unim
-00007fc0: 706c 656d 656e 7465 640a 0a20 2020 2023  plemented..    #
-00007fd0: 2067 656e 6573 2f7b 6765 6e65 7d2f 676f   genes/{gene}/go
-00007fe0: 5465 726d 7320 2d2d 2d2d 2d2d 2d20 2020  Terms -------   
-00007ff0: 0a20 2020 200a 2020 2020 6465 6620 6765  .    .    def ge
-00008000: 745f 6765 6e65 5f67 6f5f 7465 726d 7328  t_gene_go_terms(
-00008010: 7365 6c66 2c20 6765 6e65 3a73 7472 7c69  self, gene:str|i
-00008020: 6e74 2c20 2a2a 6b77 6172 6773 292d 3e44  nt, **kwargs)->D
-00008030: 6174 6146 7261 6d65 3a0a 2020 2020 2020  ataFrame:.      
-00008040: 2020 2222 220a 2020 2020 2020 2020 0a20    """.        . 
-00008050: 2020 2020 2020 203a 7061 7261 6d20 6765         :param ge
-00008060: 6e65 3a20 416e 2065 6e73 656d 626c 2067  ne: An ensembl g
-00008070: 656e 6520 6964 656e 7469 6669 6572 2077  ene identifier w
-00008080: 6869 6368 2074 7970 6963 616c 6c79 2073  hich typically s
-00008090: 7461 7274 7320 7769 7468 200a 2020 2020  tarts with .    
-000080a0: 2020 2020 2020 656e 7367 206f 7220 616e        ensg or an
-000080b0: 206e 6362 6920 6765 6e65 2069 6465 6e74   ncbi gene ident
-000080c0: 6966 6965 7220 6f72 2061 6e20 6f66 6669  ifier or an offi
-000080d0: 6369 616c 2067 656e 6520 7379 6d62 6f6c  cial gene symbol
-000080e0: 2061 7070 726f 7665 6420 6279 0a20 2020   approved by.   
-000080f0: 2020 2020 2020 2068 676e 630a 2020 2020         hgnc.    
-00008100: 2020 2020 3a74 7970 6520 6765 6e65 3a20      :type gene: 
-00008110: 7374 727c 696e 740a 2020 2020 2020 2020  str|int.        
-00008120: 3a70 6172 616d 202a 2a6b 7761 7267 733a  :param **kwargs:
-00008130: 2041 6464 6974 696f 6e61 6c20 6172 6775   Additional argu
-00008140: 6d65 6e74 7320 746f 2070 6173 7320 746f  ments to pass to
-00008150: 2072 6177 2e67 6574 5f67 656e 655f 676f   raw.get_gene_go
-00008160: 5f74 6572 6d73 0a20 2020 2020 2020 203a  _terms.        :
-00008170: 7265 7475 726e 3a20 4120 4461 7461 4672  return: A DataFr
-00008180: 616d 6520 7769 7468 2069 6e66 6f72 6d61  ame with informa
-00008190: 7469 6f6e 2061 626f 7574 2074 6865 2047  tion about the G
-000081a0: 4f20 7465 726d 7320 6173 7369 676e 6564  O terms assigned
-000081b0: 2074 6f20 7468 6520 7175 6572 6965 6420   to the queried 
-000081c0: 6765 6e65 0a20 2020 2020 2020 2020 2054  gene.          T
-000081d0: 6865 2066 6965 6c64 7320 6f66 2074 6865  he fields of the
-000081e0: 206f 7574 7075 7420 4461 7461 4672 616d   output DataFram
-000081f0: 6520 6172 653a 0a20 2020 2020 2020 2020  e are:.         
-00008200: 2020 202d 2074 6572 6d5f 6e61 6d65 3a20     - term_name: 
-00008210: 4e61 6d65 206f 6620 7468 6520 7465 726d  Name of the term
-00008220: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
-00008230: 6572 6d5f 4944 3a20 4944 206f 6620 7468  erm_ID: ID of th
-00008240: 6520 7465 726d 0a20 2020 2020 2020 2020  e term.         
-00008250: 2020 202d 2074 6572 6d5f 5552 493a 2055     - term_URI: U
-00008260: 5249 206f 6620 7468 6520 7465 726d 200a  RI of the term .
-00008270: 2020 2020 2020 2020 3a72 7479 7065 3a20          :rtype: 
-00008280: 4461 7461 4672 616d 650a 0a20 2020 2020  DataFrame..     
-00008290: 2020 2022 2222 0a20 2020 2020 2020 2072     """.        r
-000082a0: 6573 706f 6e73 6520 3d20 7365 6c66 2e72  esponse = self.r
-000082b0: 6177 2e67 6574 5f67 656e 655f 676f 5f74  aw.get_gene_go_t
-000082c0: 6572 6d73 2867 656e 652c 202a 2a6b 7761  erms(gene, **kwa
-000082d0: 7267 7329 0a20 2020 2020 2020 2064 6620  rgs).        df 
-000082e0: 3d20 7073 2e70 726f 6365 7373 5f47 4f28  = ps.process_GO(
-000082f0: 7265 7370 6f6e 7365 2e64 6174 6129 0a20  response.data). 
-00008300: 2020 2020 2020 2072 6574 7572 6e20 6466         return df
-00008310: 0a0a 2020 2020 0a20 2020 2023 2067 656e  ..    .    # gen
-00008320: 6573 2f7b 6765 6e65 7d2f 6c6f 6361 7469  es/{gene}/locati
-00008330: 6f6e 732c 2067 6574 5f67 656e 655f 6c6f  ons, get_gene_lo
-00008340: 6361 7469 6f6e 7320 2d2d 2d2d 0a20 2020  cations ----.   
-00008350: 200a 2020 2020 6465 6620 6765 745f 6765   .    def get_ge
-00008360: 6e65 5f6c 6f63 6174 696f 6e73 2873 656c  ne_locations(sel
-00008370: 662c 2067 656e 653a 7374 727c 696e 742c  f, gene:str|int,
-00008380: 202a 2a6b 7761 7267 7329 2d3e 4461 7461   **kwargs)->Data
-00008390: 4672 616d 653a 0a20 2020 2020 2020 2022  Frame:.        "
-000083a0: 2222 0a20 2020 2020 2020 200a 2020 2020  "".        .    
-000083b0: 2020 2020 3a70 6172 616d 2067 656e 653a      :param gene:
-000083c0: 2044 4553 4352 4950 5449 4f4e 0a20 2020   DESCRIPTION.   
-000083d0: 2020 2020 203a 7479 7065 2067 656e 653a       :type gene:
-000083e0: 2073 7472 7c69 6e74 0a20 2020 2020 2020   str|int.       
-000083f0: 203a 7061 7261 6d20 2a2a 6b77 6172 6773   :param **kwargs
-00008400: 3a20 4441 6464 6974 696f 6e61 6c20 6172  : DAdditional ar
-00008410: 6775 6d65 6e74 7320 746f 2070 6173 7320  guments to pass 
-00008420: 746f 2072 6177 2e67 6574 5f67 656e 655f  to raw.get_gene_
-00008430: 6c6f 6361 7469 6f6e 730a 2020 2020 2020  locations.      
-00008440: 2020 3a74 7970 6520 2a2a 6b77 6172 6773    :type **kwargs
-00008450: 3a20 5459 5045 0a20 2020 2020 2020 203a  : TYPE.        :
-00008460: 7265 7475 726e 3a20 4120 4461 7461 4672  return: A DataFr
-00008470: 616d 6520 7769 7468 2069 6e66 6f72 6d61  ame with informa
-00008480: 7469 6f6e 2061 626f 7574 2074 6865 2070  tion about the p
-00008490: 6879 7369 6361 6c20 6c6f 6361 7469 6f6e  hysical location
-000084a0: 206f 6620 7468 6520 7175 6572 6965 6420   of the queried 
-000084b0: 6765 6e65 0a20 2020 2020 2020 2020 2054  gene.          T
-000084c0: 6865 2066 6965 6c64 7320 6f66 2074 6865  he fields of the
-000084d0: 206f 7574 7075 7420 4461 7461 4672 616d   output DataFram
-000084e0: 6520 6172 653a 0a20 2020 2020 2020 2020  e are:.         
-000084f0: 2020 202d 2063 6872 6f6d 6f73 6f6d 653a     - chromosome:
-00008500: 204e 616d 6520 6f66 2074 6865 2063 6872   Name of the chr
-00008510: 6f6d 6f73 6f6d 6520 7468 6520 6765 6e65  omosome the gene
-00008520: 2069 7320 6c6f 6361 7465 640a 2020 2020   is located.    
-00008530: 2020 2020 2020 2020 2d20 7374 7261 6e64          - strand
-00008540: 3a20 5768 6963 6820 7374 7261 6e64 2074  : Which strand t
-00008550: 6865 2067 656e 6520 6973 206c 6f63 6174  he gene is locat
-00008560: 6564 0a20 2020 2020 2020 2020 2020 202d  ed.            -
-00008570: 206e 7563 6c65 6f74 6964 653a 204e 7563   nucleotide: Nuc
-00008580: 6c65 6f74 6964 6520 6e75 6d62 6572 2066  leotide number f
-00008590: 6f72 2074 6865 2067 656e 650a 2020 2020  or the gene.    
-000085a0: 2020 2020 2020 2020 2d20 6c65 6e67 7468          - length
-000085b0: 3a20 4765 6e65 206c 656e 6774 680a 2020  : Gene length.  
-000085c0: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
-000085d0: 6e5f 6e61 6d65 3a20 4e61 6d65 206f 6620  n_name: Name of 
-000085e0: 7468 6520 7370 6563 6965 730a 2020 2020  the species.    
-000085f0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
-00008600: 7363 6965 6e74 6966 6963 3a20 5363 6965  scientific: Scie
-00008610: 6e74 6966 6963 206e 616d 6520 666f 7220  ntific name for 
-00008620: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
-00008630: 2020 2020 2020 2d20 7461 786f 6e5f 4944        - taxon_ID
-00008640: 3a20 496e 7465 726e 616c 2069 6465 6e74  : Internal ident
-00008650: 6966 6965 7220 6769 7665 6e20 746f 2074  ifier given to t
-00008660: 6865 2073 7065 6369 6573 2062 7920 4765  he species by Ge
-00008670: 6d6d 610a 2020 2020 2020 2020 2020 2020  mma.            
-00008680: 2d20 7461 786f 6e5f 4e43 4249 3a20 4e43  - taxon_NCBI: NC
-00008690: 4249 2049 4420 6f66 2074 6865 2074 6178  BI ID of the tax
-000086a0: 6f6e 0a20 2020 2020 2020 2020 2020 202d  on.            -
-000086b0: 2074 6178 6f6e 5f64 6174 6162 6173 655f   taxon_database_
-000086c0: 6e61 6d65 3a20 556e 6465 726c 7969 6e67  name: Underlying
-000086d0: 2064 6174 6162 6173 6520 7573 6564 2069   database used i
-000086e0: 6e20 4765 6d6d 6120 666f 7220 7468 6520  n Gemma for the 
-000086f0: 7461 786f 6e0a 2020 2020 2020 2020 2020  taxon.          
-00008700: 2020 2d20 7461 786f 6e5f 6461 7461 6261    - taxon_databa
-00008710: 7365 5f49 443a 2049 4420 6f66 2074 6865  se_ID: ID of the
-00008720: 2075 6e64 6572 6c79 696e 6720 6461 7461   underlying data
-00008730: 6261 7365 2075 7365 6420 696e 2047 656d  base used in Gem
-00008740: 6d61 2066 6f72 2074 6865 2074 6178 6f6e  ma for the taxon
-00008750: 2020 2020 2020 0a20 2020 2020 2020 203a        .        :
-00008760: 7274 7970 653a 2044 6174 6146 7261 6d65  rtype: DataFrame
-00008770: 0a0a 2020 2020 2020 2020 2222 220a 0a20  ..        """.. 
-00008780: 2020 2020 2020 2072 6573 706f 6e73 6520         response 
-00008790: 3d20 7365 6c66 2e72 6177 2e67 6574 5f67  = self.raw.get_g
-000087a0: 656e 655f 6c6f 6361 7469 6f6e 7328 6765  ene_locations(ge
-000087b0: 6e65 2c20 2a2a 6b77 6172 6773 290a 2020  ne, **kwargs).  
-000087c0: 2020 2020 2020 6466 203d 2070 732e 7072        df = ps.pr
-000087d0: 6f63 6573 735f 6765 6e65 5f6c 6f63 6174  ocess_gene_locat
-000087e0: 696f 6e28 7265 7370 6f6e 7365 2e64 6174  ion(response.dat
-000087f0: 6129 0a20 2020 2020 2020 2072 6574 7572  a).        retur
-00008800: 6e20 6466 0a20 2020 200a 2020 2020 2320  n df.    .    # 
-00008810: 6765 6e65 732f 7b67 656e 657d 2f70 726f  genes/{gene}/pro
-00008820: 6265 732c 2067 6574 5f67 656e 655f 7072  bes, get_gene_pr
-00008830: 6f62 6573 202d 2d2d 2d2d 0a20 2020 200a  obes -----.    .
-00008840: 2020 2020 6465 6620 6765 745f 6765 6e65      def get_gene
-00008850: 5f70 726f 6265 7328 7365 6c66 2c20 6765  _probes(self, ge
-00008860: 6e65 3a73 7472 7c69 6e74 2c0a 2020 2020  ne:str|int,.    
-00008870: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00008880: 2020 2020 6f66 6673 6574 3a69 6e74 203d      offset:int =
-00008890: 2030 2c0a 2020 2020 2020 2020 2020 2020   0,.            
-000088a0: 2020 2020 2020 2020 2020 2020 6c69 6d69              limi
-000088b0: 743a 696e 7420 3d20 3230 2c0a 2020 2020  t:int = 20,.    
-000088c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000088d0: 2020 2020 2a2a 6b77 6172 6773 292d 3e44      **kwargs)->D
-000088e0: 6174 6146 7261 6d65 3a0a 2020 2020 2020  ataFrame:.      
-000088f0: 2020 2222 2252 6574 7269 6576 6520 7468    """Retrieve th
-00008900: 6520 7072 6f62 6573 2061 7373 6f63 6961  e probes associa
-00008910: 7465 6420 746f 2061 2067 656e 6573 2061  ted to a genes a
-00008920: 6372 6f73 7320 616c 6c20 706c 6174 666f  cross all platfo
-00008930: 726d 730a 2020 2020 2020 2020 0a20 2020  rms.        .   
-00008940: 2020 2020 203a 7061 7261 6d20 6765 6e65       :param gene
-00008950: 3a20 416e 2065 6e73 656d 626c 2067 656e  : An ensembl gen
-00008960: 6520 6964 656e 7469 6669 6572 2077 6869  e identifier whi
-00008970: 6368 2074 7970 6963 616c 6c79 2073 7461  ch typically sta
-00008980: 7274 7320 7769 7468 0a20 2020 2020 2020  rts with.       
-00008990: 2020 2065 6e73 6720 6f72 2061 6e20 6e63     ensg or an nc
-000089a0: 6269 2067 656e 6520 6964 656e 7469 6669  bi gene identifi
-000089b0: 6572 206f 7220 616e 206f 6666 6963 6961  er or an officia
-000089c0: 6c20 6765 6e65 2073 796d 626f 6c20 6170  l gene symbol ap
-000089d0: 7072 6f76 6564 2062 7920 0a20 2020 2020  proved by .     
-000089e0: 2020 2020 2068 676e 630a 2020 2020 2020       hgnc.      
-000089f0: 2020 3a74 7970 6520 6765 6e65 3a20 7374    :type gene: st
-00008a00: 727c 696e 740a 2020 2020 2020 2020 3a70  r|int.        :p
-00008a10: 6172 616d 206f 6666 7365 743a 2054 6865  aram offset: The
-00008a20: 206f 6666 7365 7420 6f66 2074 6865 2066   offset of the f
-00008a30: 6972 7374 2072 6574 7269 6576 6564 2072  irst retrieved r
-00008a40: 6573 756c 742e 2c20 6465 6661 756c 7473  esult., defaults
-00008a50: 2074 6f20 300a 2020 2020 2020 2020 3a74   to 0.        :t
-00008a60: 7970 6520 6f66 6673 6574 3a20 696e 742c  ype offset: int,
-00008a70: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
-00008a80: 2020 3a70 6172 616d 206c 696d 6974 3a20    :param limit: 
-00008a90: 4c69 6d69 7473 2074 6865 2072 6573 756c  Limits the resul
-00008aa0: 7420 746f 2073 7065 6369 6669 6564 2061  t to specified a
-00008ab0: 6d6f 756e 7420 6f66 206f 626a 6563 7473  mount of objects
-00008ac0: 2e0a 2020 2020 2020 2020 2020 4861 7320  ..          Has 
-00008ad0: 6120 6d61 7869 6d75 6d20 7661 6c75 6520  a maximum value 
-00008ae0: 6f66 2031 3030 2e20 5573 6520 746f 6765  of 100. Use toge
-00008af0: 7468 6572 2077 6974 6820 6f66 6673 6574  ther with offset
-00008b00: 2061 6e64 2074 6865 200a 2020 2020 2020   and the .      
-00008b10: 2020 2020 746f 7461 6c5f 656c 656d 656e      total_elemen
-00008b20: 7473 2061 7474 7269 6275 7465 2069 6e20  ts attribute in 
-00008b30: 7468 6520 6f75 7470 7574 2074 6f20 636f  the output to co
-00008b40: 6d70 696c 6520 616c 6c20 6461 7461 2069  mpile all data i
-00008b50: 6620 6e65 6564 6564 2e0a 2020 2020 2020  f needed..      
-00008b60: 2020 2020 416c 7465 726e 6174 6976 656c      Alternativel
-00008b70: 7920 6765 745f 616c 6c5f 7061 6765 7320  y get_all_pages 
-00008b80: 6675 6e63 7469 6f6e 2063 616e 2062 6520  function can be 
-00008b90: 7573 6564 2077 6974 6820 616c 6c20 6675  used with all fu
-00008ba0: 6e63 7469 6f6e 730a 2020 2020 2020 2020  nctions.        
-00008bb0: 2020 696e 636c 7564 696e 6720 6f66 6673    including offs
-00008bc0: 6574 2061 6e64 206c 696d 6974 2070 6172  et and limit par
-00008bd0: 616d 6574 6572 732c 2064 6566 6175 6c74  ameters, default
-00008be0: 7320 746f 2032 300a 2020 2020 2020 2020  s to 20.        
-00008bf0: 3a74 7970 6520 6c69 6d69 743a 2069 6e74  :type limit: int
-00008c00: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
-00008c10: 2020 203a 7061 7261 6d20 2a2a 6b77 6172     :param **kwar
-00008c20: 6773 3a20 4164 6469 7469 6f6e 616c 2061  gs: Additional a
-00008c30: 7267 756d 656e 7473 2074 6f20 7061 7373  rguments to pass
-00008c40: 2074 6f20 7261 772e 6765 745f 6765 6e65   to raw.get_gene
-00008c50: 5f70 726f 6265 730a 2020 2020 2020 2020  _probes.        
-00008c60: 3a72 6574 7572 6e3a 2041 2044 6174 6146  :return: A DataF
-00008c70: 7261 6d65 2077 6974 6820 696e 666f 726d  rame with inform
-00008c80: 6174 696f 6e20 6162 6f75 7420 7468 6520  ation about the 
-00008c90: 7072 6f62 6573 2072 6570 7265 7365 6e74  probes represent
-00008ca0: 696e 6720 6120 0a20 2020 2020 2020 2020  ing a .         
-00008cb0: 2067 656e 6520 6163 726f 7373 2061 6c6c   gene across all
-00008cc0: 2070 6c61 7472 6f66 6d73 2e0a 2020 2020   platrofms..    
-00008cd0: 2020 2020 0a20 2020 2020 2020 2020 2054      .          T
-00008ce0: 6865 2066 6965 6c64 7320 6f66 2074 6865  he fields of the
-00008cf0: 206f 7574 7075 7420 4461 7461 4672 616d   output DataFram
-00008d00: 6520 6172 653a 0a20 2020 2020 2020 2020  e are:.         
-00008d10: 2020 202d 2065 6c65 6d65 6e74 5f6e 616d     - element_nam
-00008d20: 653a 204e 616d 6520 6f66 2074 6865 2065  e: Name of the e
-00008d30: 6c65 6d65 6e74 2e20 5479 7069 6361 6c6c  lement. Typicall
-00008d40: 7920 7468 6520 7072 6f62 6573 6574 206e  y the probeset n
-00008d50: 616d 650a 2020 2020 2020 2020 2020 2020  ame.            
-00008d60: 2d20 656c 656d 656e 745f 6465 7363 7269  - element_descri
-00008d70: 7074 696f 6e3a 2041 2066 7265 6520 7465  ption: A free te
-00008d80: 7874 2066 6965 6c64 2070 726f 7669 6469  xt field providi
-00008d90: 6e67 206f 7074 696f 6e61 6c20 696e 666f  ng optional info
-00008da0: 726d 6174 696f 6e20 6162 6f75 7420 7468  rmation about th
-00008db0: 6520 656c 656d 656e 740a 2020 2020 2020  e element.      
-00008dc0: 2020 2020 2020 2d20 706c 6174 666f 726d        - platform
-00008dd0: 5f73 686f 7274 5f6e 616d 653a 2053 686f  _short_name: Sho
-00008de0: 7274 6e61 6d65 206f 6620 7468 6520 706c  rtname of the pl
-00008df0: 6174 666f 726d 2067 6976 656e 2062 7920  atform given by 
-00008e00: 4765 6d6d 612e 2054 7970 6963 616c 6c79  Gemma. Typically
-00008e10: 2074 6865 2047 504c 2069 6465 6e74 6966   the GPL identif
-00008e20: 6965 722e 0a20 2020 2020 2020 2020 2020  ier..           
-00008e30: 202d 2070 6c61 7466 6f72 6d5f 6e61 6d65   - platform_name
-00008e40: 3a20 4675 6c6c 206e 616d 6520 6f66 2074  : Full name of t
-00008e50: 6865 2070 6c61 7466 6f72 6d0a 2020 2020  he platform.    
-00008e60: 2020 2020 2020 2020 2d20 706c 6174 666f          - platfo
-00008e70: 726d 5f49 443a 2049 6420 6e75 6d62 6572  rm_ID: Id number
-00008e80: 206f 6620 7468 6520 706c 6174 666f 726d   of the platform
-00008e90: 2067 6976 656e 2062 7920 4765 6d6d 610a   given by Gemma.
-00008ea0: 2020 2020 2020 2020 2020 2020 2d20 706c              - pl
-00008eb0: 6174 666f 726d 5f74 7970 653a 2054 7970  atform_type: Typ
-00008ec0: 6520 6f66 2074 6865 2070 6c61 7466 6f72  e of the platfor
-00008ed0: 6d2e 0a20 2020 2020 2020 2020 2020 202d  m..            -
-00008ee0: 2070 6c61 7466 6f72 6d5f 6465 7363 7269   platform_descri
-00008ef0: 7074 696f 6e3a 2046 7265 6520 7465 7874  ption: Free text
-00008f00: 2066 6965 6c64 2064 6573 6372 6962 696e   field describin
-00008f10: 6720 7468 6520 706c 6174 666f 726d 2e0a  g the platform..
-00008f20: 2020 2020 2020 2020 2020 2020 2d20 706c              - pl
-00008f30: 6174 666f 726d 5f74 726f 7562 6c65 643a  atform_troubled:
-00008f40: 2057 6865 7468 6572 2074 6865 2070 6c61   Whether the pla
-00008f50: 7466 6f72 6d20 6973 206d 6172 6b65 6420  tform is marked 
-00008f60: 6173 2074 726f 7562 6c65 6420 6279 2061  as troubled by a
-00008f70: 2047 656d 6d61 2063 7572 6174 6f72 2e0a   Gemma curator..
-00008f80: 2020 2020 2020 2020 2020 2020 2d20 7461              - ta
-00008f90: 786f 6e5f 6e61 6d65 3a20 4e61 6d65 206f  xon_name: Name o
-00008fa0: 6620 7468 6520 7370 6563 6965 730a 2020  f the species.  
-00008fb0: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
-00008fc0: 6e5f 7363 6965 6e74 6966 6963 3a20 5363  n_scientific: Sc
-00008fd0: 6965 6e74 6966 6963 206e 616d 6520 666f  ientific name fo
-00008fe0: 7220 7468 6520 7461 786f 6e0a 2020 2020  r the taxon.    
-00008ff0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
-00009000: 4944 3a20 496e 7465 726e 616c 2069 6465  ID: Internal ide
-00009010: 6e74 6966 6965 7220 6769 7665 6e20 746f  ntifier given to
-00009020: 2074 6865 2073 7065 6369 6573 2062 7920   the species by 
-00009030: 4765 6d6d 610a 2020 2020 2020 2020 2020  Gemma.          
-00009040: 2020 2d20 7461 786f 6e5f 4e43 4249 3a20    - taxon_NCBI: 
-00009050: 4e43 4249 2049 4420 6f66 2074 6865 2074  NCBI ID of the t
-00009060: 6178 6f6e 0a20 2020 2020 2020 2020 2020  axon.           
-00009070: 202d 2074 6178 6f6e 5f64 6174 6162 6173   - taxon_databas
-00009080: 655f 6e61 6d65 3a20 556e 6465 726c 7969  e_name: Underlyi
-00009090: 6e67 2064 6174 6162 6173 6520 7573 6564  ng database used
-000090a0: 2069 6e20 4765 6d6d 6120 666f 7220 7468   in Gemma for th
-000090b0: 6520 7461 786f 6e0a 2020 2020 2020 2020  e taxon.        
-000090c0: 2020 2020 2d20 7461 786f 6e5f 6461 7461      - taxon_data
-000090d0: 6261 7365 5f49 443a 2049 4420 6f66 2074  base_ID: ID of t
-000090e0: 6865 2075 6e64 6572 6c79 696e 6720 6461  he underlying da
-000090f0: 7461 6261 7365 2075 7365 6420 696e 2047  tabase used in G
-00009100: 656d 6d61 2066 6f72 2074 6865 2074 6178  emma for the tax
-00009110: 6f6e 2020 2020 0a20 2020 2020 2020 203a  on    .        :
-00009120: 7274 7970 653a 2044 6174 6146 7261 6d65  rtype: DataFrame
-00009130: 0a0a 2020 2020 2020 2020 2222 220a 2020  ..        """.  
-00009140: 2020 2020 2020 0a20 2020 2020 2020 206b        .        k
-00009150: 7761 7267 7320 3d20 7673 2e72 656d 6f76  wargs = vs.remov
-00009160: 655f 6e6f 6e65 7328 6f66 6673 6574 203d  e_nones(offset =
-00009170: 206f 6666 7365 742c 0a20 2020 2020 2020   offset,.       
-00009180: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00009190: 2020 2020 2020 2020 2020 6c69 6d69 7420            limit 
-000091a0: 3d20 6c69 6d69 742c 0a20 2020 2020 2020  = limit,.       
-000091b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000091c0: 2020 2020 2020 2020 2020 2a2a 6b77 6172            **kwar
-000091d0: 6773 290a 2020 2020 2020 2020 0a20 2020  gs).        .   
-000091e0: 2020 2020 2072 6573 706f 6e73 6520 3d20       response = 
-000091f0: 7365 6c66 2e72 6177 2e67 6574 5f67 656e  self.raw.get_gen
-00009200: 655f 7072 6f62 6573 2867 656e 652c 202a  e_probes(gene, *
-00009210: 2a6b 7761 7267 7329 0a20 2020 2020 2020  *kwargs).       
-00009220: 2064 6620 3d20 7073 2e70 726f 6365 7373   df = ps.process
-00009230: 5f65 6c65 6d65 6e74 7328 7265 7370 6f6e  _elements(respon
-00009240: 7365 2e64 6174 6129 0a20 2020 2020 2020  se.data).       
-00009250: 2070 732e 6174 7461 6368 5f61 7474 7269   ps.attach_attri
-00009260: 6275 7465 7328 6466 2c20 7265 7370 6f6e  butes(df, respon
-00009270: 7365 2e74 6f5f 6469 6374 2829 290a 2020  se.to_dict()).  
-00009280: 2020 2020 2020 7265 7475 726e 2064 660a        return df.
-00009290: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-000092a0: 200a 2020 2020 2320 6765 6e65 732f 7b67   .    # genes/{g
-000092b0: 656e 6573 7d2c 2067 6574 5f67 656e 6573  enes}, get_genes
-000092c0: 2d2d 2d2d 2d2d 2d0a 0a20 2020 2064 6566  -------..    def
-000092d0: 2067 6574 5f67 656e 6573 2873 656c 662c   get_genes(self,
-000092e0: 2067 656e 6573 3a69 6e74 7c73 7472 2c20   genes:int|str, 
-000092f0: 2a2a 6b77 6172 6773 292d 3e44 6174 6146  **kwargs)->DataF
-00009300: 7261 6d65 3a0a 2020 2020 2020 2020 2222  rame:.        ""
-00009310: 2252 6574 7269 6576 6520 6765 6e65 7320  "Retrieve genes 
-00009320: 6d61 7463 6869 6e67 2067 656e 6520 6964  matching gene id
-00009330: 656e 7469 6669 6572 730a 0a20 2020 2020  entifiers..     
-00009340: 2020 203a 7061 7261 6d20 6765 6e65 733a     :param genes:
-00009350: 2041 6e20 656e 7365 6d62 6c20 6765 6e65   An ensembl gene
-00009360: 2069 6465 6e74 6966 6965 7220 7768 6963   identifier whic
-00009370: 6820 7479 7069 6361 6c6c 7920 7374 6172  h typically star
-00009380: 7473 2077 6974 6820 0a20 2020 2020 2020  ts with .       
-00009390: 2020 2065 6e73 6720 6f72 2061 6e20 6e63     ensg or an nc
-000093a0: 6269 2067 656e 6520 6964 656e 7469 6669  bi gene identifi
-000093b0: 6572 206f 7220 616e 206f 6666 6963 6961  er or an officia
-000093c0: 6c20 6765 6e65 2073 796d 626f 6c20 6170  l gene symbol ap
-000093d0: 7072 6f76 6564 2062 7920 6867 6e63 0a20  proved by hgnc. 
-000093e0: 2020 2020 2020 203a 7479 7065 2067 656e         :type gen
-000093f0: 6573 3a20 696e 747c 7374 720a 2020 2020  es: int|str.    
-00009400: 2020 2020 3a70 6172 616d 202a 2a6b 7761      :param **kwa
-00009410: 7267 733a 2041 6464 6974 696f 6e61 6c20  rgs: Additional 
-00009420: 6172 6775 6d65 6e74 7320 746f 2070 6173  arguments to pas
-00009430: 7320 746f 2072 6177 5f67 6574 5f67 656e  s to raw_get_gen
-00009440: 6573 0a20 2020 2020 2020 203a 7265 7475  es.        :retu
-00009450: 726e 3a20 4120 4461 7461 4672 616d 6520  rn: A DataFrame 
-00009460: 7769 7468 2074 6865 2069 6e66 6f72 6d61  with the informa
-00009470: 7469 6f6e 2061 626f 7574 2074 6865 2071  tion about the q
-00009480: 7565 7272 6965 6420 6765 6e65 732e 0a20  uerried genes.. 
-00009490: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-000094a0: 2020 5468 6520 6669 656c 6473 206f 6620    The fields of 
-000094b0: 7468 6520 6f75 7470 7574 2044 6174 6146  the output DataF
-000094c0: 7261 6d65 2061 7265 3a0a 2020 2020 2020  rame are:.      
-000094d0: 2020 2020 2020 2d20 6765 6e65 5f73 796d        - gene_sym
-000094e0: 626f 6c3a 2053 796d 626f 6c20 666f 7220  bol: Symbol for 
-000094f0: 7468 6520 6765 6e65 0a20 2020 2020 2020  the gene.       
-00009500: 2020 2020 202d 2067 656e 655f 656e 7365       - gene_ense
-00009510: 6d62 6c3a 2045 6e73 656d 626c 2049 4420  mbl: Ensembl ID 
-00009520: 666f 7220 7468 6520 6765 6e65 0a20 2020  for the gene.   
-00009530: 2020 2020 2020 2020 202d 2067 656e 655f           - gene_
-00009540: 4e43 4249 3a20 4e43 4249 2069 6420 666f  NCBI: NCBI id fo
-00009550: 7220 7468 6520 6765 6e65 0a20 2020 2020  r the gene.     
-00009560: 2020 2020 2020 202d 2067 656e 655f 6e61         - gene_na
-00009570: 6d65 3a20 4e61 6d65 206f 6620 7468 6520  me: Name of the 
-00009580: 6765 6e65 0a20 2020 2020 2020 2020 2020  gene.           
-00009590: 202d 2067 656e 655f 616c 6961 7365 733a   - gene_aliases:
-000095a0: 2047 656e 6520 616c 6961 7365 732e 2045   Gene aliases. E
-000095b0: 6163 6820 726f 7720 696e 636c 7564 6573  ach row includes
-000095c0: 2061 2076 6563 746f 720a 2020 2020 2020   a vector.      
-000095d0: 2020 2020 2020 2d20 6765 6e65 5f4d 4658        - gene_MFX
-000095e0: 5f72 616e 6b3a 204d 756c 7469 6675 6e63  _rank: Multifunc
-000095f0: 7469 6f6e 616c 6974 7920 7261 6e6b 2066  tionality rank f
-00009600: 6f72 2074 6865 2067 656e 650a 2020 2020  or the gene.    
-00009610: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
-00009620: 6e61 6d65 3a20 4e61 6d65 206f 6620 7468  name: Name of th
-00009630: 6520 7370 6563 6965 730a 2020 2020 2020  e species.      
-00009640: 2020 2020 2020 2d20 7461 786f 6e5f 7363        - taxon_sc
-00009650: 6965 6e74 6966 6963 3a20 5363 6965 6e74  ientific: Scient
-00009660: 6966 6963 206e 616d 6520 666f 7220 7468  ific name for th
-00009670: 6520 7461 786f 6e0a 2020 2020 2020 2020  e taxon.        
-00009680: 2020 2020 2d20 7461 786f 6e5f 4944 3a20      - taxon_ID: 
-00009690: 496e 7465 726e 616c 2069 6465 6e74 6966  Internal identif
-000096a0: 6965 7220 6769 7665 6e20 746f 2074 6865  ier given to the
-000096b0: 2073 7065 6369 6573 2062 7920 4765 6d6d   species by Gemm
-000096c0: 610a 2020 2020 2020 2020 2020 2020 2d20  a.            - 
-000096d0: 7461 786f 6e5f 4e43 4249 3a20 4e43 4249  taxon_NCBI: NCBI
-000096e0: 2049 4420 6f66 2074 6865 2074 6178 6f6e   ID of the taxon
-000096f0: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
-00009700: 6178 6f6e 5f64 6174 6162 6173 655f 6e61  axon_database_na
-00009710: 6d65 3a20 556e 6465 726c 7969 6e67 2064  me: Underlying d
-00009720: 6174 6162 6173 6520 7573 6564 2069 6e20  atabase used in 
-00009730: 4765 6d6d 6120 666f 7220 7468 6520 7461  Gemma for the ta
-00009740: 786f 6e0a 2020 2020 2020 2020 2020 2020  xon.            
-00009750: 2d20 7461 786f 6e5f 6461 7461 6261 7365  - taxon_database
-00009760: 5f49 443a 2049 4420 6f66 2074 6865 2075  _ID: ID of the u
-00009770: 6e64 6572 6c79 696e 6720 6461 7461 6261  nderlying databa
-00009780: 7365 2075 7365 6420 696e 2047 656d 6d61  se used in Gemma
-00009790: 2066 6f72 2074 6865 2074 6178 6f6e 2020   for the taxon  
-000097a0: 2020 0a20 2020 2020 2020 203a 7274 7970    .        :rtyp
-000097b0: 653a 2044 6174 6146 7261 6d65 0a0a 2020  e: DataFrame..  
-000097c0: 2020 2020 2020 2222 220a 0a20 2020 2020        """..     
-000097d0: 2020 2072 6573 706f 6e73 6520 3d20 7365     response = se
-000097e0: 6c66 2e72 6177 2e67 6574 5f67 656e 6573  lf.raw.get_genes
-000097f0: 2867 656e 6573 2c20 2a2a 6b77 6172 6773  (genes, **kwargs
-00009800: 290a 2020 2020 2020 2020 6466 203d 2070  ).        df = p
-00009810: 732e 7072 6f63 6573 735f 6765 6e65 7328  s.process_genes(
-00009820: 7265 7370 6f6e 7365 2e64 6174 6129 0a20  response.data). 
-00009830: 2020 2020 2020 2072 6574 7572 6e20 6466         return df
-00009840: 0a0a 2020 2020 2320 706c 6174 666f 726d  ..    # platform
-00009850: 732f 636f 756e 7420 2d2d 2d2d 2d0a 2020  s/count -----.  
-00009860: 2020 2320 756e 696d 706c 656d 656e 7465    # unimplemente
-00009870: 640a 2020 2020 0a20 2020 2023 2070 6c61  d.    .    # pla
-00009880: 7466 6f72 6d2f 7b70 6c61 7466 6f72 6d7d  tform/{platform}
-00009890: 2f61 6e6e 6f74 6174 696f 6e73 202d 2d2d  /annotations ---
-000098a0: 2d2d 0a20 2020 2023 2069 6e20 6765 6d6d  --.    # in gemm
-000098b0: 612e 5220 7468 6973 2065 6e64 706f 696e  a.R this endpoin
-000098c0: 7420 6973 6e27 7420 696d 706c 656d 656e  t isn't implemen
-000098d0: 7465 6420 616e 6420 7573 6573 2061 2063  ted and uses a c
-000098e0: 6f6e 7665 6e69 656e 6365 2066 756e 6374  onvenience funct
-000098f0: 696f 6e20 696e 7374 6561 640a 2020 2020  ion instead.    
-00009900: 2320 6865 7265 2077 6520 6a75 7374 2075  # here we just u
-00009910: 7365 2074 6865 2065 6e70 6f69 6e74 2073  se the enpoint s
-00009920: 696e 6365 2074 6865 2061 6464 6564 2066  ince the added f
-00009930: 756e 6374 696f 6e61 6c69 7479 2069 736e  unctionality isn
-00009940: 2774 206e 6565 6465 640a 0a20 2020 2023  't needed..    #
-00009950: 2043 6f72 7265 7370 6f6e 6469 6e67 2067   Corresponding g
-00009960: 656d 6d61 2e52 2066 756e 6374 696f 6e20  emma.R function 
-00009970: 646f 6573 6e27 7420 7573 6520 616e 7920  doesn't use any 
-00009980: 656e 6470 6f69 6e74 2028 7573 6573 2073  endpoint (uses s
-00009990: 6f6d 6520 616c 7465 726e 6174 6976 6520  ome alternative 
-000099a0: 5552 4c0a 2020 2020 2320 746f 2067 6574  URL.    # to get
-000099b0: 2069 6e66 6f29 2062 7574 2068 6173 2073   info) but has s
-000099c0: 6576 6572 616c 206f 7074 696f 6e73 2061  everal options a
-000099d0: 6c6c 6f77 696e 6720 746f 2073 656c 6563  llowing to selec
-000099e0: 7420 7468 6520 616e 6e2e 2074 7970 653a  t the ann. type:
-000099f0: 0a20 2020 2023 2061 6e6e 6f74 5479 7065  .    # annotType
-00009a00: 203d 2063 2822 6269 6f50 726f 6365 7373   = c("bioProcess
-00009a10: 222c 2022 6e6f 5061 7265 6e74 7322 2c20  ", "noParents", 
-00009a20: 2261 6c6c 5061 7265 6e74 7322 290a 2020  "allParents").  
-00009a30: 2020 2320 5468 6973 2066 6561 7475 7265    # This feature
-00009a40: 2069 7320 6e6f 7420 696d 706c 656d 656e   is not implemen
-00009a50: 7465 6420 6865 7265 2c20 7468 6520 7265  ted here, the re
-00009a60: 7475 726e 2076 616c 7565 2063 6f72 7265  turn value corre
-00009a70: 7370 6f6e 6473 2074 6f20 226e 6f50 6172  sponds to "noPar
-00009a80: 656e 7473 220a 2020 2020 2320 2861 7320  ents".    # (as 
-00009a90: 6f66 2032 3032 322d 3035 2d31 3929 0a20  of 2022-05-19). 
-00009aa0: 2020 2064 6566 2067 6574 5f70 6c61 7466     def get_platf
-00009ab0: 6f72 6d5f 616e 6e6f 7461 7469 6f6e 7328  orm_annotations(
-00009ac0: 7365 6c66 2c20 706c 6174 666f 726d 3a69  self, platform:i
-00009ad0: 6e74 7c73 7472 2c20 2a2a 6b77 6172 6773  nt|str, **kwargs
-00009ae0: 292d 3e44 6174 6146 7261 6d65 3a0a 2020  )->DataFrame:.  
-00009af0: 2020 2020 2020 2222 2247 6574 7320 4765        """Gets Ge
-00009b00: 6d6d 6127 7320 706c 6174 666f 726d 2061  mma's platform a
-00009b10: 6e6e 6f74 6174 696f 6e73 2069 6e63 6c75  nnotations inclu
-00009b20: 6469 6e67 206d 6170 7069 6e67 7320 6f66  ding mappings of
-00009b30: 206d 6963 726f 6172 7261 7920 7072 6f62   microarray prob
-00009b40: 6573 2074 6f20 6765 6e65 732e 0a20 2020  es to genes..   
-00009b50: 2020 2020 200a 2020 2020 2020 2020 3a70       .        :p
-00009b60: 6172 616d 2070 6c61 7466 6f72 6d3a 2041  aram platform: A
-00009b70: 2070 6c61 7466 6f72 6d20 6e75 6d65 7269   platform numeri
-00009b80: 6361 6c20 6964 656e 7469 6669 6572 206f  cal identifier o
-00009b90: 7220 6120 706c 6174 666f 726d 2073 686f  r a platform sho
-00009ba0: 7274 206e 616d 650a 2020 2020 2020 2020  rt name.        
-00009bb0: 3a74 7970 6520 706c 6174 666f 726d 3a20  :type platform: 
-00009bc0: 696e 747c 7374 720a 2020 2020 2020 2020  int|str.        
-00009bd0: 3a70 6172 616d 202a 2a6b 7761 7267 733a  :param **kwargs:
-00009be0: 2041 6464 6974 696f 6e61 6c20 6172 6775   Additional argu
-00009bf0: 6d65 6e74 7320 746f 2070 6173 7320 746f  ments to pass to
-00009c00: 2072 6177 2e67 6574 5f70 6c61 7466 6f72   raw.get_platfor
-00009c10: 6d5f 616e 6e6f 7461 7469 6f6e 730a 2020  m_annotations.  
-00009c20: 2020 2020 2020 3a74 7970 6520 2a2a 6b77        :type **kw
-00009c30: 6172 6773 3a20 5459 5045 0a20 2020 2020  args: TYPE.     
-00009c40: 2020 203a 7265 7475 726e 3a20 4120 4461     :return: A Da
-00009c50: 7461 4672 616d 6520 6f66 2061 6e6e 6f74  taFrame of annot
-00009c60: 6174 696f 6e73 0a20 2020 2020 2020 200a  ations.        .
-00009c70: 2020 2020 2020 2020 2020 2d20 5072 6f62            - Prob
-00009c80: 654e 616d 653a 2050 726f 6265 7365 7420  eName: Probeset 
-00009c90: 6e61 6d65 7320 7072 6f76 6964 6564 2062  names provided b
-00009ca0: 7920 7468 6520 706c 6174 666f 726d 2e20  y the platform. 
-00009cb0: 4765 6e65 2073 796d 626f 6c73 2066 6f72  Gene symbols for
-00009cc0: 2067 656e 6572 6963 2061 6e6e 6f74 6174   generic annotat
-00009cd0: 696f 6e73 0a20 2020 2020 2020 2020 202d  ions.          -
-00009ce0: 2047 656e 6553 796d 626f 6c73 3a20 4765   GeneSymbols: Ge
-00009cf0: 6e65 7320 7468 6174 2077 6572 6520 666f  nes that were fo
-00009d00: 756e 6420 746f 2062 6520 616c 6967 6e65  und to be aligne
-00009d10: 6420 746f 2074 6865 2070 726f 6265 2073  d to the probe s
-00009d20: 6571 7565 6e63 652e 204e 6f74 6520 7468  equence. Note th
-00009d30: 6174 2069 7420 6973 2070 6f73 7369 626c  at it is possibl
-00009d40: 6520 666f 7220 7072 6f62 6573 2074 6f20  e for probes to 
-00009d50: 6265 206e 6f6e 2d73 7065 6369 6669 632e  be non-specific.
-00009d60: 2041 6c69 676e 6d65 6e74 2074 6f20 6d75   Alignment to mu
-00009d70: 6c74 6970 6c65 2067 656e 6573 2061 7265  ltiple genes are
-00009d80: 2069 6e64 6963 6174 6564 2077 6974 6820   indicated with 
-00009d90: 6765 6e65 2073 796d 626f 6c73 2073 6570  gene symbols sep
-00009da0: 6172 6174 6564 2062 7920 227c 2273 0a20  arated by "|"s. 
-00009db0: 2020 2020 2020 2020 202d 2047 656e 654e           - GeneN
-00009dc0: 616d 6573 3a20 4e61 6d65 206f 6620 7468  ames: Name of th
-00009dd0: 6520 6765 6e65 0a20 2020 2020 2020 2020  e gene.         
-00009de0: 202d 2047 4f54 6572 6d73 3a20 474f 2054   - GOTerms: GO T
-00009df0: 6572 6d73 2061 7373 6f63 6961 7465 6420  erms associated 
-00009e00: 7769 7468 2074 6865 2067 656e 6573 2e20  with the genes. 
-00009e10: 616e 6e6f 7454 7970 6520 6172 6775 6d65  annotType argume
-00009e20: 6e74 2063 616e 2062 6520 7573 6564 2074  nt can be used t
-00009e30: 6f20 6368 6f6f 7365 2077 6869 6368 2074  o choose which t
-00009e40: 6572 6d73 2073 686f 756c 6420 6265 2069  erms should be i
-00009e50: 6e63 6c75 6465 642e 0a20 2020 2020 2020  ncluded..       
-00009e60: 2020 202d 2047 656d 6d61 4944 7320 616e     - GemmaIDs an
-00009e70: 6420 4e43 4249 6964 733a 2072 6573 7065  d NCBIids: respe
-00009e80: 6374 6976 6520 4944 7320 666f 7220 7468  ctive IDs for th
-00009e90: 6520 6765 6e65 732e 0a20 2020 2020 2020  e genes..       
-00009ea0: 203a 7274 7970 653a 2044 6174 6146 7261   :rtype: DataFra
-00009eb0: 6d65 0a0a 2020 2020 2020 2020 2222 220a  me..        """.
-00009ec0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-00009ed0: 2061 7069 5f72 6573 706f 6e73 6520 3d20   api_response = 
-00009ee0: 7365 6c66 2e72 6177 2e67 6574 5f70 6c61  self.raw.get_pla
-00009ef0: 7466 6f72 6d5f 616e 6e6f 7461 7469 6f6e  tform_annotation
-00009f00: 7328 706c 6174 666f 726d 2c20 2a2a 6b77  s(platform, **kw
-00009f10: 6172 6773 290a 2020 2020 2020 2020 756e  args).        un
-00009f20: 636f 6d6d 656e 7420 3d20 6170 695f 7265  comment = api_re
-00009f30: 7370 6f6e 7365 2e73 706c 6974 2822 5c6e  sponse.split("\n
-00009f40: 2322 290a 2020 2020 2020 2020 6170 695f  #").        api_
-00009f50: 7265 7370 6f6e 7365 203d 2075 6e63 6f6d  response = uncom
-00009f60: 6d65 6e74 5b6c 656e 2875 6e63 6f6d 6d65  ment[len(uncomme
-00009f70: 6e74 292d 315d 0a20 2020 2020 2020 2075  nt)-1].        u
-00009f80: 6e63 6f6d 6d65 6e74 203d 2061 7069 5f72  ncomment = api_r
-00009f90: 6573 706f 6e73 652e 7370 6c69 7428 275c  esponse.split('\
-00009fa0: 6e27 2c31 290a 2020 2020 2020 2020 6170  n',1).        ap
-00009fb0: 695f 7265 7370 6f6e 7365 203d 2075 6e63  i_response = unc
-00009fc0: 6f6d 6d65 6e74 5b6c 656e 2875 6e63 6f6d  omment[len(uncom
-00009fd0: 6d65 6e74 292d 315d 0a20 2020 2020 2020  ment)-1].       
-00009fe0: 2072 6574 7572 6e20 7064 2e72 6561 645f   return pd.read_
-00009ff0: 6373 7628 5374 7269 6e67 494f 2861 7069  csv(StringIO(api
-0000a000: 5f72 6573 706f 6e73 6529 2c20 7365 703d  _response), sep=
-0000a010: 275c 7427 290a 0a20 2020 2023 2070 6c61  '\t')..    # pla
-0000a020: 7466 6f72 6d2f 7b70 6c61 7466 6f72 6d7d  tform/{platform}
-0000a030: 2f64 6174 6173 6574 732c 2067 6574 5f70  /datasets, get_p
-0000a040: 6c61 7466 6f72 6d5f 6461 7461 7365 7473  latform_datasets
-0000a050: 202d 2d2d 2d0a 0a20 2020 2064 6566 2067   ----..    def g
-0000a060: 6574 5f70 6c61 7466 6f72 6d5f 6461 7461  et_platform_data
-0000a070: 7365 7473 2873 656c 662c 2070 6c61 7466  sets(self, platf
-0000a080: 6f72 6d3a 7374 727c 696e 742c 0a20 2020  orm:str|int,.   
-0000a090: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000a0a0: 2020 2020 2020 2020 2020 206f 6666 7365             offse
-0000a0b0: 743a 696e 7420 3d20 302c 0a20 2020 2020  t:int = 0,.     
-0000a0c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000a0d0: 2020 2020 2020 2020 206c 696d 6974 3a69           limit:i
-0000a0e0: 6e74 203d 2032 302c 0a20 2020 2020 2020  nt = 20,.       
-0000a0f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000a100: 2020 2020 2020 202a 2a6b 7761 7267 7329         **kwargs)
-0000a110: 2d3e 4461 7461 4672 616d 653a 0a20 2020  ->DataFrame:.   
-0000a120: 2020 2020 2022 2222 5265 7472 6965 7665       """Retrieve
-0000a130: 2061 6c6c 2065 7870 6572 696d 656e 7473   all experiments
-0000a140: 2075 7369 6e67 2061 2067 6976 656e 2070   using a given p
-0000a150: 6c61 7466 6f72 6d0a 0a20 2020 2020 2020  latform..       
-0000a160: 200a 2020 2020 2020 2020 3a70 6172 616d   .        :param
-0000a170: 2070 6c61 7466 6f72 6d3a 2041 2070 6c61   platform: A pla
-0000a180: 7466 6f72 6d20 6e75 6d65 7269 6361 6c20  tform numerical 
-0000a190: 6964 656e 7469 6669 6572 206f 7220 6120  identifier or a 
-0000a1a0: 706c 6174 666f 726d 2073 686f 7274 206e  platform short n
-0000a1b0: 616d 650a 2020 2020 2020 2020 3a74 7970  ame.        :typ
-0000a1c0: 6520 706c 6174 666f 726d 3a20 7374 727c  e platform: str|
-0000a1d0: 696e 740a 2020 2020 2020 2020 3a70 6172  int.        :par
-0000a1e0: 616d 206f 6666 7365 743a 2054 6865 206f  am offset: The o
-0000a1f0: 6666 7365 7420 6f66 2074 6865 2066 6972  ffset of the fir
-0000a200: 7374 2072 6574 7269 6576 6564 2072 6573  st retrieved res
-0000a210: 756c 742e 2c20 6465 6661 756c 7473 2074  ult., defaults t
-0000a220: 6f20 300a 2020 2020 2020 2020 3a74 7970  o 0.        :typ
-0000a230: 6520 6f66 6673 6574 3a20 696e 742c 206f  e offset: int, o
-0000a240: 7074 696f 6e61 6c0a 2020 2020 2020 2020  ptional.        
-0000a250: 3a70 6172 616d 206c 696d 6974 3a20 4c69  :param limit: Li
-0000a260: 6d69 7473 2074 6865 2072 6573 756c 7420  mits the result 
-0000a270: 746f 2073 7065 6369 6669 6564 2061 6d6f  to specified amo
-0000a280: 756e 7420 6f66 206f 626a 6563 7473 2e0a  unt of objects..
-0000a290: 2020 2020 2020 2020 2020 4861 7320 6120            Has a 
-0000a2a0: 6d61 7869 6d75 6d20 7661 6c75 6520 6f66  maximum value of
-0000a2b0: 2031 3030 2e20 5573 6520 746f 6765 7468   100. Use togeth
-0000a2c0: 6572 2077 6974 6820 6f66 6673 6574 2061  er with offset a
-0000a2d0: 6e64 2074 6865 200a 2020 2020 2020 2020  nd the .        
-0000a2e0: 2020 746f 7461 6c5f 656c 656d 656e 7473    total_elements
-0000a2f0: 2061 7474 7269 6275 7465 2069 6e20 7468   attribute in th
-0000a300: 6520 6f75 7470 7574 2074 6f20 636f 6d70  e output to comp
-0000a310: 696c 6520 616c 6c20 6461 7461 2069 6620  ile all data if 
-0000a320: 6e65 6564 6564 2e0a 2020 2020 2020 2020  needed..        
-0000a330: 2020 416c 7465 726e 6174 6976 656c 7920    Alternatively 
-0000a340: 6765 745f 616c 6c5f 7061 6765 7320 6675  get_all_pages fu
-0000a350: 6e63 7469 6f6e 2063 616e 2062 6520 7573  nction can be us
-0000a360: 6564 2077 6974 6820 616c 6c20 6675 6e63  ed with all func
-0000a370: 7469 6f6e 730a 2020 2020 2020 2020 2020  tions.          
-0000a380: 696e 636c 7564 696e 6720 6f66 6673 6574  including offset
-0000a390: 2061 6e64 206c 696d 6974 2070 6172 616d   and limit param
-0000a3a0: 6574 6572 732c 2064 6566 6175 6c74 7320  eters, defaults 
-0000a3b0: 746f 2032 300a 2020 2020 2020 2020 3a74  to 20.        :t
-0000a3c0: 7970 6520 6c69 6d69 743a 2069 6e74 2c20  ype limit: int, 
-0000a3d0: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
-0000a3e0: 203a 7061 7261 6d20 2a2a 6b77 6172 6773   :param **kwargs
-0000a3f0: 3a20 4164 6469 7469 6f6e 616c 2061 7267  : Additional arg
-0000a400: 756d 656e 7473 2074 6f20 7061 7373 2074  uments to pass t
-0000a410: 6f20 7261 772e 6765 745f 706c 6174 666f  o raw.get_platfo
-0000a420: 726d 5f64 6174 6173 6574 730a 2020 2020  rm_datasets.    
-0000a430: 2020 2020 3a72 6574 7572 6e3a 2041 2044      :return: A D
-0000a440: 6174 6146 7261 6d65 2077 6974 6820 696e  ataFrame with in
-0000a450: 666f 726d 6174 696f 6e20 6162 6f75 7420  formation about 
-0000a460: 7468 6520 7175 6572 6965 6420 6461 7461  the queried data
-0000a470: 7365 7428 7329 2e0a 2020 2020 2020 2020  set(s)..        
-0000a480: 0a20 2020 2020 2020 2020 2054 6865 2066  .          The f
-0000a490: 6965 6c64 7320 6f66 2074 6865 2044 6174  ields of the Dat
-0000a4a0: 6146 7261 6d65 2061 7265 3a0a 2020 2020  aFrame are:.    
-0000a4b0: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
-0000a4c0: 6d65 6e74 5f73 686f 7274 5f6e 616d 653a  ment_short_name:
-0000a4d0: 2053 686f 7274 6e61 6d65 2067 6976 656e   Shortname given
-0000a4e0: 2074 6f20 7468 6520 6461 7461 7365 7420   to the dataset 
-0000a4f0: 7769 7468 696e 2047 656d 6d61 2e20 4f66  within Gemma. Of
-0000a500: 7465 6e20 636f 7272 6573 706f 6e64 7320  ten corresponds 
-0000a510: 746f 2061 6363 6573 7369 6f6e 2049 440a  to accession ID.
-0000a520: 2020 2020 2020 2020 2020 2020 2d20 6578              - ex
-0000a530: 7065 7269 6d65 6e74 5f6e 616d 653a 2046  periment_name: F
-0000a540: 756c 6c20 7469 746c 6520 6f66 2074 6865  ull title of the
-0000a550: 2064 6174 6173 6574 0a20 2020 2020 2020   dataset.       
+00007fa0: 7461 7861 202d 2d2d 2d2d 0a20 2020 2023  taxa -----.    #
+00007fb0: 2063 7572 7265 6e74 6c79 2075 6e69 6d70   currently unimp
+00007fc0: 6c65 6d65 6e74 6564 0a0a 2020 2020 2320  lemented..    # 
+00007fd0: 6461 7461 7365 7473 2f63 6f75 6e74 202d  datasets/count -
+00007fe0: 2d2d 2d2d 0a20 2020 2023 2063 7572 7265  ----.    # curre
+00007ff0: 6e74 6c79 2075 6e69 6d70 6c65 6d65 6e74  ntly unimplement
+00008000: 6564 0a0a 2020 2020 2320 6765 6e65 732f  ed..    # genes/
+00008010: 7b67 656e 657d 2f67 6f54 6572 6d73 202d  {gene}/goTerms -
+00008020: 2d2d 2d2d 2d2d 2020 200a 2020 2020 0a20  ------   .    . 
+00008030: 2020 2064 6566 2067 6574 5f67 656e 655f     def get_gene_
+00008040: 676f 5f74 6572 6d73 2873 656c 662c 2067  go_terms(self, g
+00008050: 656e 653a 7374 727c 696e 742c 202a 2a6b  ene:str|int, **k
+00008060: 7761 7267 7329 2d3e 4461 7461 4672 616d  wargs)->DataFram
+00008070: 653a 0a20 2020 2020 2020 2022 2222 0a20  e:.        """. 
+00008080: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+00008090: 3a70 6172 616d 2067 656e 653a 2041 6e20  :param gene: An 
+000080a0: 656e 7365 6d62 6c20 6765 6e65 2069 6465  ensembl gene ide
+000080b0: 6e74 6966 6965 7220 7768 6963 6820 7479  ntifier which ty
+000080c0: 7069 6361 6c6c 7920 7374 6172 7473 2077  pically starts w
+000080d0: 6974 6820 0a20 2020 2020 2020 2020 2065  ith .          e
+000080e0: 6e73 6720 6f72 2061 6e20 6e63 6269 2067  nsg or an ncbi g
+000080f0: 656e 6520 6964 656e 7469 6669 6572 206f  ene identifier o
+00008100: 7220 616e 206f 6666 6963 6961 6c20 6765  r an official ge
+00008110: 6e65 2073 796d 626f 6c20 6170 7072 6f76  ne symbol approv
+00008120: 6564 2062 790a 2020 2020 2020 2020 2020  ed by.          
+00008130: 6867 6e63 0a20 2020 2020 2020 203a 7479  hgnc.        :ty
+00008140: 7065 2067 656e 653a 2073 7472 7c69 6e74  pe gene: str|int
+00008150: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+00008160: 2a2a 6b77 6172 6773 3a20 4164 6469 7469  **kwargs: Additi
+00008170: 6f6e 616c 2061 7267 756d 656e 7473 2074  onal arguments t
+00008180: 6f20 7061 7373 2074 6f20 7261 772e 6765  o pass to raw.ge
+00008190: 745f 6765 6e65 5f67 6f5f 7465 726d 730a  t_gene_go_terms.
+000081a0: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
+000081b0: 2041 2044 6174 6146 7261 6d65 2077 6974   A DataFrame wit
+000081c0: 6820 696e 666f 726d 6174 696f 6e20 6162  h information ab
+000081d0: 6f75 7420 7468 6520 474f 2074 6572 6d73  out the GO terms
+000081e0: 2061 7373 6967 6e65 6420 746f 2074 6865   assigned to the
+000081f0: 2071 7565 7269 6564 2067 656e 650a 2020   queried gene.  
+00008200: 2020 2020 2020 2020 5468 6520 6669 656c          The fiel
+00008210: 6473 206f 6620 7468 6520 6f75 7470 7574  ds of the output
+00008220: 2044 6174 6146 7261 6d65 2061 7265 3a0a   DataFrame are:.
+00008230: 2020 2020 2020 2020 2020 2020 2d20 7465              - te
+00008240: 726d 5f6e 616d 653a 204e 616d 6520 6f66  rm_name: Name of
+00008250: 2074 6865 2074 6572 6d0a 2020 2020 2020   the term.      
+00008260: 2020 2020 2020 2d20 7465 726d 5f49 443a        - term_ID:
+00008270: 2049 4420 6f66 2074 6865 2074 6572 6d0a   ID of the term.
+00008280: 2020 2020 2020 2020 2020 2020 2d20 7465              - te
+00008290: 726d 5f55 5249 3a20 5552 4920 6f66 2074  rm_URI: URI of t
+000082a0: 6865 2074 6572 6d20 0a20 2020 2020 2020  he term .       
+000082b0: 203a 7274 7970 653a 2044 6174 6146 7261   :rtype: DataFra
+000082c0: 6d65 0a0a 2020 2020 2020 2020 2222 220a  me..        """.
+000082d0: 2020 2020 2020 2020 7265 7370 6f6e 7365          response
+000082e0: 203d 2073 656c 662e 7261 772e 6765 745f   = self.raw.get_
+000082f0: 6765 6e65 5f67 6f5f 7465 726d 7328 6765  gene_go_terms(ge
+00008300: 6e65 2c20 2a2a 6b77 6172 6773 290a 2020  ne, **kwargs).  
+00008310: 2020 2020 2020 6466 203d 2070 732e 7072        df = ps.pr
+00008320: 6f63 6573 735f 474f 2872 6573 706f 6e73  ocess_GO(respons
+00008330: 652e 6461 7461 290a 2020 2020 2020 2020  e.data).        
+00008340: 7265 7475 726e 2064 660a 0a20 2020 200a  return df..    .
+00008350: 2020 2020 2320 6765 6e65 732f 7b67 656e      # genes/{gen
+00008360: 657d 2f6c 6f63 6174 696f 6e73 2c20 6765  e}/locations, ge
+00008370: 745f 6765 6e65 5f6c 6f63 6174 696f 6e73  t_gene_locations
+00008380: 202d 2d2d 2d0a 2020 2020 0a20 2020 2064   ----.    .    d
+00008390: 6566 2067 6574 5f67 656e 655f 6c6f 6361  ef get_gene_loca
+000083a0: 7469 6f6e 7328 7365 6c66 2c20 6765 6e65  tions(self, gene
+000083b0: 3a73 7472 7c69 6e74 2c20 2a2a 6b77 6172  :str|int, **kwar
+000083c0: 6773 292d 3e44 6174 6146 7261 6d65 3a0a  gs)->DataFrame:.
+000083d0: 2020 2020 2020 2020 2222 220a 2020 2020          """.    
+000083e0: 2020 2020 0a20 2020 2020 2020 203a 7061      .        :pa
+000083f0: 7261 6d20 6765 6e65 3a20 4445 5343 5249  ram gene: DESCRI
+00008400: 5054 494f 4e0a 2020 2020 2020 2020 3a74  PTION.        :t
+00008410: 7970 6520 6765 6e65 3a20 7374 727c 696e  ype gene: str|in
+00008420: 740a 2020 2020 2020 2020 3a70 6172 616d  t.        :param
+00008430: 202a 2a6b 7761 7267 733a 2044 4164 6469   **kwargs: DAddi
+00008440: 7469 6f6e 616c 2061 7267 756d 656e 7473  tional arguments
+00008450: 2074 6f20 7061 7373 2074 6f20 7261 772e   to pass to raw.
+00008460: 6765 745f 6765 6e65 5f6c 6f63 6174 696f  get_gene_locatio
+00008470: 6e73 0a20 2020 2020 2020 203a 7479 7065  ns.        :type
+00008480: 202a 2a6b 7761 7267 733a 2054 5950 450a   **kwargs: TYPE.
+00008490: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
+000084a0: 2041 2044 6174 6146 7261 6d65 2077 6974   A DataFrame wit
+000084b0: 6820 696e 666f 726d 6174 696f 6e20 6162  h information ab
+000084c0: 6f75 7420 7468 6520 7068 7973 6963 616c  out the physical
+000084d0: 206c 6f63 6174 696f 6e20 6f66 2074 6865   location of the
+000084e0: 2071 7565 7269 6564 2067 656e 650a 2020   queried gene.  
+000084f0: 2020 2020 2020 2020 5468 6520 6669 656c          The fiel
+00008500: 6473 206f 6620 7468 6520 6f75 7470 7574  ds of the output
+00008510: 2044 6174 6146 7261 6d65 2061 7265 3a0a   DataFrame are:.
+00008520: 2020 2020 2020 2020 2020 2020 2d20 6368              - ch
+00008530: 726f 6d6f 736f 6d65 3a20 4e61 6d65 206f  romosome: Name o
+00008540: 6620 7468 6520 6368 726f 6d6f 736f 6d65  f the chromosome
+00008550: 2074 6865 2067 656e 6520 6973 206c 6f63   the gene is loc
+00008560: 6174 6564 0a20 2020 2020 2020 2020 2020  ated.           
+00008570: 202d 2073 7472 616e 643a 2057 6869 6368   - strand: Which
+00008580: 2073 7472 616e 6420 7468 6520 6765 6e65   strand the gene
+00008590: 2069 7320 6c6f 6361 7465 640a 2020 2020   is located.    
+000085a0: 2020 2020 2020 2020 2d20 6e75 636c 656f          - nucleo
+000085b0: 7469 6465 3a20 4e75 636c 656f 7469 6465  tide: Nucleotide
+000085c0: 206e 756d 6265 7220 666f 7220 7468 6520   number for the 
+000085d0: 6765 6e65 0a20 2020 2020 2020 2020 2020  gene.           
+000085e0: 202d 206c 656e 6774 683a 2047 656e 6520   - length: Gene 
+000085f0: 6c65 6e67 7468 0a20 2020 2020 2020 2020  length.         
+00008600: 2020 202d 2074 6178 6f6e 5f6e 616d 653a     - taxon_name:
+00008610: 204e 616d 6520 6f66 2074 6865 2073 7065   Name of the spe
+00008620: 6369 6573 0a20 2020 2020 2020 2020 2020  cies.           
+00008630: 202d 2074 6178 6f6e 5f73 6369 656e 7469   - taxon_scienti
+00008640: 6669 633a 2053 6369 656e 7469 6669 6320  fic: Scientific 
+00008650: 6e61 6d65 2066 6f72 2074 6865 2074 6178  name for the tax
+00008660: 6f6e 0a20 2020 2020 2020 2020 2020 202d  on.            -
+00008670: 2074 6178 6f6e 5f49 443a 2049 6e74 6572   taxon_ID: Inter
+00008680: 6e61 6c20 6964 656e 7469 6669 6572 2067  nal identifier g
+00008690: 6976 656e 2074 6f20 7468 6520 7370 6563  iven to the spec
+000086a0: 6965 7320 6279 2047 656d 6d61 0a20 2020  ies by Gemma.   
+000086b0: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
+000086c0: 5f4e 4342 493a 204e 4342 4920 4944 206f  _NCBI: NCBI ID o
+000086d0: 6620 7468 6520 7461 786f 6e0a 2020 2020  f the taxon.    
+000086e0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
+000086f0: 6461 7461 6261 7365 5f6e 616d 653a 2055  database_name: U
+00008700: 6e64 6572 6c79 696e 6720 6461 7461 6261  nderlying databa
+00008710: 7365 2075 7365 6420 696e 2047 656d 6d61  se used in Gemma
+00008720: 2066 6f72 2074 6865 2074 6178 6f6e 0a20   for the taxon. 
+00008730: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
+00008740: 6f6e 5f64 6174 6162 6173 655f 4944 3a20  on_database_ID: 
+00008750: 4944 206f 6620 7468 6520 756e 6465 726c  ID of the underl
+00008760: 7969 6e67 2064 6174 6162 6173 6520 7573  ying database us
+00008770: 6564 2069 6e20 4765 6d6d 6120 666f 7220  ed in Gemma for 
+00008780: 7468 6520 7461 786f 6e20 2020 2020 200a  the taxon      .
+00008790: 2020 2020 2020 2020 3a72 7479 7065 3a20          :rtype: 
+000087a0: 4461 7461 4672 616d 650a 0a20 2020 2020  DataFrame..     
+000087b0: 2020 2022 2222 0a0a 2020 2020 2020 2020     """..        
+000087c0: 7265 7370 6f6e 7365 203d 2073 656c 662e  response = self.
+000087d0: 7261 772e 6765 745f 6765 6e65 5f6c 6f63  raw.get_gene_loc
+000087e0: 6174 696f 6e73 2867 656e 652c 202a 2a6b  ations(gene, **k
+000087f0: 7761 7267 7329 0a20 2020 2020 2020 2064  wargs).        d
+00008800: 6620 3d20 7073 2e70 726f 6365 7373 5f67  f = ps.process_g
+00008810: 656e 655f 6c6f 6361 7469 6f6e 2872 6573  ene_location(res
+00008820: 706f 6e73 652e 6461 7461 290a 2020 2020  ponse.data).    
+00008830: 2020 2020 7265 7475 726e 2064 660a 2020      return df.  
+00008840: 2020 0a20 2020 2023 2067 656e 6573 2f7b    .    # genes/{
+00008850: 6765 6e65 7d2f 7072 6f62 6573 2c20 6765  gene}/probes, ge
+00008860: 745f 6765 6e65 5f70 726f 6265 7320 2d2d  t_gene_probes --
+00008870: 2d2d 2d0a 2020 2020 0a20 2020 2064 6566  ---.    .    def
+00008880: 2067 6574 5f67 656e 655f 7072 6f62 6573   get_gene_probes
+00008890: 2873 656c 662c 2067 656e 653a 7374 727c  (self, gene:str|
+000088a0: 696e 742c 0a20 2020 2020 2020 2020 2020  int,.           
+000088b0: 2020 2020 2020 2020 2020 2020 206f 6666               off
+000088c0: 7365 743a 696e 7420 3d20 302c 0a20 2020  set:int = 0,.   
+000088d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000088e0: 2020 2020 206c 696d 6974 3a69 6e74 203d       limit:int =
+000088f0: 2032 302c 0a20 2020 2020 2020 2020 2020   20,.           
+00008900: 2020 2020 2020 2020 2020 2020 202a 2a6b               **k
+00008910: 7761 7267 7329 2d3e 4461 7461 4672 616d  wargs)->DataFram
+00008920: 653a 0a20 2020 2020 2020 2022 2222 5265  e:.        """Re
+00008930: 7472 6965 7665 2074 6865 2070 726f 6265  trieve the probe
+00008940: 7320 6173 736f 6369 6174 6564 2074 6f20  s associated to 
+00008950: 6120 6765 6e65 7320 6163 726f 7373 2061  a genes across a
+00008960: 6c6c 2070 6c61 7466 6f72 6d73 0a20 2020  ll platforms.   
+00008970: 2020 2020 200a 2020 2020 2020 2020 3a70       .        :p
+00008980: 6172 616d 2067 656e 653a 2041 6e20 656e  aram gene: An en
+00008990: 7365 6d62 6c20 6765 6e65 2069 6465 6e74  sembl gene ident
+000089a0: 6966 6965 7220 7768 6963 6820 7479 7069  ifier which typi
+000089b0: 6361 6c6c 7920 7374 6172 7473 2077 6974  cally starts wit
+000089c0: 680a 2020 2020 2020 2020 2020 656e 7367  h.          ensg
+000089d0: 206f 7220 616e 206e 6362 6920 6765 6e65   or an ncbi gene
+000089e0: 2069 6465 6e74 6966 6965 7220 6f72 2061   identifier or a
+000089f0: 6e20 6f66 6669 6369 616c 2067 656e 6520  n official gene 
+00008a00: 7379 6d62 6f6c 2061 7070 726f 7665 6420  symbol approved 
+00008a10: 6279 200a 2020 2020 2020 2020 2020 6867  by .          hg
+00008a20: 6e63 0a20 2020 2020 2020 203a 7479 7065  nc.        :type
+00008a30: 2067 656e 653a 2073 7472 7c69 6e74 0a20   gene: str|int. 
+00008a40: 2020 2020 2020 203a 7061 7261 6d20 6f66         :param of
+00008a50: 6673 6574 3a20 5468 6520 6f66 6673 6574  fset: The offset
+00008a60: 206f 6620 7468 6520 6669 7273 7420 7265   of the first re
+00008a70: 7472 6965 7665 6420 7265 7375 6c74 2e2c  trieved result.,
+00008a80: 2064 6566 6175 6c74 7320 746f 2030 0a20   defaults to 0. 
+00008a90: 2020 2020 2020 203a 7479 7065 206f 6666         :type off
+00008aa0: 7365 743a 2069 6e74 2c20 6f70 7469 6f6e  set: int, option
+00008ab0: 616c 0a20 2020 2020 2020 203a 7061 7261  al.        :para
+00008ac0: 6d20 6c69 6d69 743a 204c 696d 6974 7320  m limit: Limits 
+00008ad0: 7468 6520 7265 7375 6c74 2074 6f20 7370  the result to sp
+00008ae0: 6563 6966 6965 6420 616d 6f75 6e74 206f  ecified amount o
+00008af0: 6620 6f62 6a65 6374 732e 0a20 2020 2020  f objects..     
+00008b00: 2020 2020 2048 6173 2061 206d 6178 696d       Has a maxim
+00008b10: 756d 2076 616c 7565 206f 6620 3130 302e  um value of 100.
+00008b20: 2055 7365 2074 6f67 6574 6865 7220 7769   Use together wi
+00008b30: 7468 206f 6666 7365 7420 616e 6420 7468  th offset and th
+00008b40: 6520 0a20 2020 2020 2020 2020 2074 6f74  e .          tot
+00008b50: 616c 5f65 6c65 6d65 6e74 7320 6174 7472  al_elements attr
+00008b60: 6962 7574 6520 696e 2074 6865 206f 7574  ibute in the out
+00008b70: 7075 7420 746f 2063 6f6d 7069 6c65 2061  put to compile a
+00008b80: 6c6c 2064 6174 6120 6966 206e 6565 6465  ll data if neede
+00008b90: 642e 0a20 2020 2020 2020 2020 2041 6c74  d..          Alt
+00008ba0: 6572 6e61 7469 7665 6c79 2067 6574 5f61  ernatively get_a
+00008bb0: 6c6c 5f70 6167 6573 2066 756e 6374 696f  ll_pages functio
+00008bc0: 6e20 6361 6e20 6265 2075 7365 6420 7769  n can be used wi
+00008bd0: 7468 2061 6c6c 2066 756e 6374 696f 6e73  th all functions
+00008be0: 0a20 2020 2020 2020 2020 2069 6e63 6c75  .          inclu
+00008bf0: 6469 6e67 206f 6666 7365 7420 616e 6420  ding offset and 
+00008c00: 6c69 6d69 7420 7061 7261 6d65 7465 7273  limit parameters
+00008c10: 2c20 6465 6661 756c 7473 2074 6f20 3230  , defaults to 20
+00008c20: 0a20 2020 2020 2020 203a 7479 7065 206c  .        :type l
+00008c30: 696d 6974 3a20 696e 742c 206f 7074 696f  imit: int, optio
+00008c40: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
+00008c50: 616d 202a 2a6b 7761 7267 733a 2041 6464  am **kwargs: Add
+00008c60: 6974 696f 6e61 6c20 6172 6775 6d65 6e74  itional argument
+00008c70: 7320 746f 2070 6173 7320 746f 2072 6177  s to pass to raw
+00008c80: 2e67 6574 5f67 656e 655f 7072 6f62 6573  .get_gene_probes
+00008c90: 0a20 2020 2020 2020 203a 7265 7475 726e  .        :return
+00008ca0: 3a20 4120 4461 7461 4672 616d 6520 7769  : A DataFrame wi
+00008cb0: 7468 2069 6e66 6f72 6d61 7469 6f6e 2061  th information a
+00008cc0: 626f 7574 2074 6865 2070 726f 6265 7320  bout the probes 
+00008cd0: 7265 7072 6573 656e 7469 6e67 2061 200a  representing a .
+00008ce0: 2020 2020 2020 2020 2020 6765 6e65 2061            gene a
+00008cf0: 6372 6f73 7320 616c 6c20 706c 6174 726f  cross all platro
+00008d00: 666d 732e 0a20 2020 2020 2020 200a 2020  fms..        .  
+00008d10: 2020 2020 2020 2020 5468 6520 6669 656c          The fiel
+00008d20: 6473 206f 6620 7468 6520 6f75 7470 7574  ds of the output
+00008d30: 2044 6174 6146 7261 6d65 2061 7265 3a0a   DataFrame are:.
+00008d40: 2020 2020 2020 2020 2020 2020 2d20 656c              - el
+00008d50: 656d 656e 745f 6e61 6d65 3a20 4e61 6d65  ement_name: Name
+00008d60: 206f 6620 7468 6520 656c 656d 656e 742e   of the element.
+00008d70: 2054 7970 6963 616c 6c79 2074 6865 2070   Typically the p
+00008d80: 726f 6265 7365 7420 6e61 6d65 0a20 2020  robeset name.   
+00008d90: 2020 2020 2020 2020 202d 2065 6c65 6d65           - eleme
+00008da0: 6e74 5f64 6573 6372 6970 7469 6f6e 3a20  nt_description: 
+00008db0: 4120 6672 6565 2074 6578 7420 6669 656c  A free text fiel
+00008dc0: 6420 7072 6f76 6964 696e 6720 6f70 7469  d providing opti
+00008dd0: 6f6e 616c 2069 6e66 6f72 6d61 7469 6f6e  onal information
+00008de0: 2061 626f 7574 2074 6865 2065 6c65 6d65   about the eleme
+00008df0: 6e74 0a20 2020 2020 2020 2020 2020 202d  nt.            -
+00008e00: 2070 6c61 7466 6f72 6d5f 7368 6f72 745f   platform_short_
+00008e10: 6e61 6d65 3a20 5368 6f72 746e 616d 6520  name: Shortname 
+00008e20: 6f66 2074 6865 2070 6c61 7466 6f72 6d20  of the platform 
+00008e30: 6769 7665 6e20 6279 2047 656d 6d61 2e20  given by Gemma. 
+00008e40: 5479 7069 6361 6c6c 7920 7468 6520 4750  Typically the GP
+00008e50: 4c20 6964 656e 7469 6669 6572 2e0a 2020  L identifier..  
+00008e60: 2020 2020 2020 2020 2020 2d20 706c 6174            - plat
+00008e70: 666f 726d 5f6e 616d 653a 2046 756c 6c20  form_name: Full 
+00008e80: 6e61 6d65 206f 6620 7468 6520 706c 6174  name of the plat
+00008e90: 666f 726d 0a20 2020 2020 2020 2020 2020  form.           
+00008ea0: 202d 2070 6c61 7466 6f72 6d5f 4944 3a20   - platform_ID: 
+00008eb0: 4964 206e 756d 6265 7220 6f66 2074 6865  Id number of the
+00008ec0: 2070 6c61 7466 6f72 6d20 6769 7665 6e20   platform given 
+00008ed0: 6279 2047 656d 6d61 0a20 2020 2020 2020  by Gemma.       
+00008ee0: 2020 2020 202d 2070 6c61 7466 6f72 6d5f       - platform_
+00008ef0: 7479 7065 3a20 5479 7065 206f 6620 7468  type: Type of th
+00008f00: 6520 706c 6174 666f 726d 2e0a 2020 2020  e platform..    
+00008f10: 2020 2020 2020 2020 2d20 706c 6174 666f          - platfo
+00008f20: 726d 5f64 6573 6372 6970 7469 6f6e 3a20  rm_description: 
+00008f30: 4672 6565 2074 6578 7420 6669 656c 6420  Free text field 
+00008f40: 6465 7363 7269 6269 6e67 2074 6865 2070  describing the p
+00008f50: 6c61 7466 6f72 6d2e 0a20 2020 2020 2020  latform..       
+00008f60: 2020 2020 202d 2070 6c61 7466 6f72 6d5f       - platform_
+00008f70: 7472 6f75 626c 6564 3a20 5768 6574 6865  troubled: Whethe
+00008f80: 7220 7468 6520 706c 6174 666f 726d 2069  r the platform i
+00008f90: 7320 6d61 726b 6564 2061 7320 7472 6f75  s marked as trou
+00008fa0: 626c 6564 2062 7920 6120 4765 6d6d 6120  bled by a Gemma 
+00008fb0: 6375 7261 746f 722e 0a20 2020 2020 2020  curator..       
+00008fc0: 2020 2020 202d 2074 6178 6f6e 5f6e 616d       - taxon_nam
+00008fd0: 653a 204e 616d 6520 6f66 2074 6865 2073  e: Name of the s
+00008fe0: 7065 6369 6573 0a20 2020 2020 2020 2020  pecies.         
+00008ff0: 2020 202d 2074 6178 6f6e 5f73 6369 656e     - taxon_scien
+00009000: 7469 6669 633a 2053 6369 656e 7469 6669  tific: Scientifi
+00009010: 6320 6e61 6d65 2066 6f72 2074 6865 2074  c name for the t
+00009020: 6178 6f6e 0a20 2020 2020 2020 2020 2020  axon.           
+00009030: 202d 2074 6178 6f6e 5f49 443a 2049 6e74   - taxon_ID: Int
+00009040: 6572 6e61 6c20 6964 656e 7469 6669 6572  ernal identifier
+00009050: 2067 6976 656e 2074 6f20 7468 6520 7370   given to the sp
+00009060: 6563 6965 7320 6279 2047 656d 6d61 0a20  ecies by Gemma. 
+00009070: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
+00009080: 6f6e 5f4e 4342 493a 204e 4342 4920 4944  on_NCBI: NCBI ID
+00009090: 206f 6620 7468 6520 7461 786f 6e0a 2020   of the taxon.  
+000090a0: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
+000090b0: 6e5f 6461 7461 6261 7365 5f6e 616d 653a  n_database_name:
+000090c0: 2055 6e64 6572 6c79 696e 6720 6461 7461   Underlying data
+000090d0: 6261 7365 2075 7365 6420 696e 2047 656d  base used in Gem
+000090e0: 6d61 2066 6f72 2074 6865 2074 6178 6f6e  ma for the taxon
+000090f0: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
+00009100: 6178 6f6e 5f64 6174 6162 6173 655f 4944  axon_database_ID
+00009110: 3a20 4944 206f 6620 7468 6520 756e 6465  : ID of the unde
+00009120: 726c 7969 6e67 2064 6174 6162 6173 6520  rlying database 
+00009130: 7573 6564 2069 6e20 4765 6d6d 6120 666f  used in Gemma fo
+00009140: 7220 7468 6520 7461 786f 6e20 2020 200a  r the taxon    .
+00009150: 2020 2020 2020 2020 3a72 7479 7065 3a20          :rtype: 
+00009160: 4461 7461 4672 616d 650a 0a20 2020 2020  DataFrame..     
+00009170: 2020 2022 2222 0a20 2020 2020 2020 200a     """.        .
+00009180: 2020 2020 2020 2020 6b77 6172 6773 203d          kwargs =
+00009190: 2076 732e 7265 6d6f 7665 5f6e 6f6e 6573   vs.remove_nones
+000091a0: 286f 6666 7365 7420 3d20 6f66 6673 6574  (offset = offset
+000091b0: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+000091c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000091d0: 2020 206c 696d 6974 203d 206c 696d 6974     limit = limit
+000091e0: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+000091f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00009200: 2020 202a 2a6b 7761 7267 7329 0a20 2020     **kwargs).   
+00009210: 2020 2020 200a 2020 2020 2020 2020 7265       .        re
+00009220: 7370 6f6e 7365 203d 2073 656c 662e 7261  sponse = self.ra
+00009230: 772e 6765 745f 6765 6e65 5f70 726f 6265  w.get_gene_probe
+00009240: 7328 6765 6e65 2c20 2a2a 6b77 6172 6773  s(gene, **kwargs
+00009250: 290a 2020 2020 2020 2020 6466 203d 2070  ).        df = p
+00009260: 732e 7072 6f63 6573 735f 656c 656d 656e  s.process_elemen
+00009270: 7473 2872 6573 706f 6e73 652e 6461 7461  ts(response.data
+00009280: 290a 2020 2020 2020 2020 7073 2e61 7474  ).        ps.att
+00009290: 6163 685f 6174 7472 6962 7574 6573 2864  ach_attributes(d
+000092a0: 662c 2072 6573 706f 6e73 652e 746f 5f64  f, response.to_d
+000092b0: 6963 7428 2929 0a20 2020 2020 2020 2072  ict()).        r
+000092c0: 6574 7572 6e20 6466 0a20 2020 2020 2020  eturn df.       
+000092d0: 200a 2020 2020 2020 2020 0a20 2020 2023   .        .    #
+000092e0: 2067 656e 6573 2f7b 6765 6e65 737d 2c20   genes/{genes}, 
+000092f0: 6765 745f 6765 6e65 732d 2d2d 2d2d 2d2d  get_genes-------
+00009300: 0a0a 2020 2020 6465 6620 6765 745f 6765  ..    def get_ge
+00009310: 6e65 7328 7365 6c66 2c20 6765 6e65 733a  nes(self, genes:
+00009320: 696e 747c 7374 722c 202a 2a6b 7761 7267  int|str, **kwarg
+00009330: 7329 2d3e 4461 7461 4672 616d 653a 0a20  s)->DataFrame:. 
+00009340: 2020 2020 2020 2022 2222 5265 7472 6965         """Retrie
+00009350: 7665 2067 656e 6573 206d 6174 6368 696e  ve genes matchin
+00009360: 6720 6765 6e65 2069 6465 6e74 6966 6965  g gene identifie
+00009370: 7273 0a0a 2020 2020 2020 2020 3a70 6172  rs..        :par
+00009380: 616d 2067 656e 6573 3a20 416e 2065 6e73  am genes: An ens
+00009390: 656d 626c 2067 656e 6520 6964 656e 7469  embl gene identi
+000093a0: 6669 6572 2077 6869 6368 2074 7970 6963  fier which typic
+000093b0: 616c 6c79 2073 7461 7274 7320 7769 7468  ally starts with
+000093c0: 200a 2020 2020 2020 2020 2020 656e 7367   .          ensg
+000093d0: 206f 7220 616e 206e 6362 6920 6765 6e65   or an ncbi gene
+000093e0: 2069 6465 6e74 6966 6965 7220 6f72 2061   identifier or a
+000093f0: 6e20 6f66 6669 6369 616c 2067 656e 6520  n official gene 
+00009400: 7379 6d62 6f6c 2061 7070 726f 7665 6420  symbol approved 
+00009410: 6279 2068 676e 630a 2020 2020 2020 2020  by hgnc.        
+00009420: 3a74 7970 6520 6765 6e65 733a 2069 6e74  :type genes: int
+00009430: 7c73 7472 0a20 2020 2020 2020 203a 7061  |str.        :pa
+00009440: 7261 6d20 2a2a 6b77 6172 6773 3a20 4164  ram **kwargs: Ad
+00009450: 6469 7469 6f6e 616c 2061 7267 756d 656e  ditional argumen
+00009460: 7473 2074 6f20 7061 7373 2074 6f20 7261  ts to pass to ra
+00009470: 775f 6765 745f 6765 6e65 730a 2020 2020  w_get_genes.    
+00009480: 2020 2020 3a72 6574 7572 6e3a 2041 2044      :return: A D
+00009490: 6174 6146 7261 6d65 2077 6974 6820 7468  ataFrame with th
+000094a0: 6520 696e 666f 726d 6174 696f 6e20 6162  e information ab
+000094b0: 6f75 7420 7468 6520 7175 6572 7269 6564  out the querried
+000094c0: 2067 656e 6573 2e0a 2020 2020 2020 2020   genes..        
+000094d0: 0a20 2020 2020 2020 2020 2054 6865 2066  .          The f
+000094e0: 6965 6c64 7320 6f66 2074 6865 206f 7574  ields of the out
+000094f0: 7075 7420 4461 7461 4672 616d 6520 6172  put DataFrame ar
+00009500: 653a 0a20 2020 2020 2020 2020 2020 202d  e:.            -
+00009510: 2067 656e 655f 7379 6d62 6f6c 3a20 5379   gene_symbol: Sy
+00009520: 6d62 6f6c 2066 6f72 2074 6865 2067 656e  mbol for the gen
+00009530: 650a 2020 2020 2020 2020 2020 2020 2d20  e.            - 
+00009540: 6765 6e65 5f65 6e73 656d 626c 3a20 456e  gene_ensembl: En
+00009550: 7365 6d62 6c20 4944 2066 6f72 2074 6865  sembl ID for the
+00009560: 2067 656e 650a 2020 2020 2020 2020 2020   gene.          
+00009570: 2020 2d20 6765 6e65 5f4e 4342 493a 204e    - gene_NCBI: N
+00009580: 4342 4920 6964 2066 6f72 2074 6865 2067  CBI id for the g
+00009590: 656e 650a 2020 2020 2020 2020 2020 2020  ene.            
+000095a0: 2d20 6765 6e65 5f6e 616d 653a 204e 616d  - gene_name: Nam
+000095b0: 6520 6f66 2074 6865 2067 656e 650a 2020  e of the gene.  
+000095c0: 2020 2020 2020 2020 2020 2d20 6765 6e65            - gene
+000095d0: 5f61 6c69 6173 6573 3a20 4765 6e65 2061  _aliases: Gene a
+000095e0: 6c69 6173 6573 2e20 4561 6368 2072 6f77  liases. Each row
+000095f0: 2069 6e63 6c75 6465 7320 6120 7665 6374   includes a vect
+00009600: 6f72 0a20 2020 2020 2020 2020 2020 202d  or.            -
+00009610: 2067 656e 655f 4d46 585f 7261 6e6b 3a20   gene_MFX_rank: 
+00009620: 4d75 6c74 6966 756e 6374 696f 6e61 6c69  Multifunctionali
+00009630: 7479 2072 616e 6b20 666f 7220 7468 6520  ty rank for the 
+00009640: 6765 6e65 0a20 2020 2020 2020 2020 2020  gene.           
+00009650: 202d 2074 6178 6f6e 5f6e 616d 653a 204e   - taxon_name: N
+00009660: 616d 6520 6f66 2074 6865 2073 7065 6369  ame of the speci
+00009670: 6573 0a20 2020 2020 2020 2020 2020 202d  es.            -
+00009680: 2074 6178 6f6e 5f73 6369 656e 7469 6669   taxon_scientifi
+00009690: 633a 2053 6369 656e 7469 6669 6320 6e61  c: Scientific na
+000096a0: 6d65 2066 6f72 2074 6865 2074 6178 6f6e  me for the taxon
+000096b0: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
+000096c0: 6178 6f6e 5f49 443a 2049 6e74 6572 6e61  axon_ID: Interna
+000096d0: 6c20 6964 656e 7469 6669 6572 2067 6976  l identifier giv
+000096e0: 656e 2074 6f20 7468 6520 7370 6563 6965  en to the specie
+000096f0: 7320 6279 2047 656d 6d61 0a20 2020 2020  s by Gemma.     
+00009700: 2020 2020 2020 202d 2074 6178 6f6e 5f4e         - taxon_N
+00009710: 4342 493a 204e 4342 4920 4944 206f 6620  CBI: NCBI ID of 
+00009720: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
+00009730: 2020 2020 2020 2d20 7461 786f 6e5f 6461        - taxon_da
+00009740: 7461 6261 7365 5f6e 616d 653a 2055 6e64  tabase_name: Und
+00009750: 6572 6c79 696e 6720 6461 7461 6261 7365  erlying database
+00009760: 2075 7365 6420 696e 2047 656d 6d61 2066   used in Gemma f
+00009770: 6f72 2074 6865 2074 6178 6f6e 0a20 2020  or the taxon.   
+00009780: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
+00009790: 5f64 6174 6162 6173 655f 4944 3a20 4944  _database_ID: ID
+000097a0: 206f 6620 7468 6520 756e 6465 726c 7969   of the underlyi
+000097b0: 6e67 2064 6174 6162 6173 6520 7573 6564  ng database used
+000097c0: 2069 6e20 4765 6d6d 6120 666f 7220 7468   in Gemma for th
+000097d0: 6520 7461 786f 6e20 2020 200a 2020 2020  e taxon    .    
+000097e0: 2020 2020 3a72 7479 7065 3a20 4461 7461      :rtype: Data
+000097f0: 4672 616d 650a 0a20 2020 2020 2020 2022  Frame..        "
+00009800: 2222 0a0a 2020 2020 2020 2020 7265 7370  ""..        resp
+00009810: 6f6e 7365 203d 2073 656c 662e 7261 772e  onse = self.raw.
+00009820: 6765 745f 6765 6e65 7328 6765 6e65 732c  get_genes(genes,
+00009830: 202a 2a6b 7761 7267 7329 0a20 2020 2020   **kwargs).     
+00009840: 2020 2064 6620 3d20 7073 2e70 726f 6365     df = ps.proce
+00009850: 7373 5f67 656e 6573 2872 6573 706f 6e73  ss_genes(respons
+00009860: 652e 6461 7461 290a 2020 2020 2020 2020  e.data).        
+00009870: 7265 7475 726e 2064 660a 0a20 2020 2023  return df..    #
+00009880: 2070 6c61 7466 6f72 6d73 2f63 6f75 6e74   platforms/count
+00009890: 202d 2d2d 2d2d 0a20 2020 2023 2075 6e69   -----.    # uni
+000098a0: 6d70 6c65 6d65 6e74 6564 0a20 2020 200a  mplemented.    .
+000098b0: 2020 2020 2320 706c 6174 666f 726d 2f7b      # platform/{
+000098c0: 706c 6174 666f 726d 7d2f 616e 6e6f 7461  platform}/annota
+000098d0: 7469 6f6e 7320 2d2d 2d2d 2d0a 2020 2020  tions -----.    
+000098e0: 2320 696e 2067 656d 6d61 2e52 2074 6869  # in gemma.R thi
+000098f0: 7320 656e 6470 6f69 6e74 2069 736e 2774  s endpoint isn't
+00009900: 2069 6d70 6c65 6d65 6e74 6564 2061 6e64   implemented and
+00009910: 2075 7365 7320 6120 636f 6e76 656e 6965   uses a convenie
+00009920: 6e63 6520 6675 6e63 7469 6f6e 2069 6e73  nce function ins
+00009930: 7465 6164 0a20 2020 2023 2068 6572 6520  tead.    # here 
+00009940: 7765 206a 7573 7420 7573 6520 7468 6520  we just use the 
+00009950: 656e 706f 696e 7420 7369 6e63 6520 7468  enpoint since th
+00009960: 6520 6164 6465 6420 6675 6e63 7469 6f6e  e added function
+00009970: 616c 6974 7920 6973 6e27 7420 6e65 6564  ality isn't need
+00009980: 6564 0a0a 2020 2020 2320 436f 7272 6573  ed..    # Corres
+00009990: 706f 6e64 696e 6720 6765 6d6d 612e 5220  ponding gemma.R 
+000099a0: 6675 6e63 7469 6f6e 2064 6f65 736e 2774  function doesn't
+000099b0: 2075 7365 2061 6e79 2065 6e64 706f 696e   use any endpoin
+000099c0: 7420 2875 7365 7320 736f 6d65 2061 6c74  t (uses some alt
+000099d0: 6572 6e61 7469 7665 2055 524c 0a20 2020  ernative URL.   
+000099e0: 2023 2074 6f20 6765 7420 696e 666f 2920   # to get info) 
+000099f0: 6275 7420 6861 7320 7365 7665 7261 6c20  but has several 
+00009a00: 6f70 7469 6f6e 7320 616c 6c6f 7769 6e67  options allowing
+00009a10: 2074 6f20 7365 6c65 6374 2074 6865 2061   to select the a
+00009a20: 6e6e 2e20 7479 7065 3a0a 2020 2020 2320  nn. type:.    # 
+00009a30: 616e 6e6f 7454 7970 6520 3d20 6328 2262  annotType = c("b
+00009a40: 696f 5072 6f63 6573 7322 2c20 226e 6f50  ioProcess", "noP
+00009a50: 6172 656e 7473 222c 2022 616c 6c50 6172  arents", "allPar
+00009a60: 656e 7473 2229 0a20 2020 2023 2054 6869  ents").    # Thi
+00009a70: 7320 6665 6174 7572 6520 6973 206e 6f74  s feature is not
+00009a80: 2069 6d70 6c65 6d65 6e74 6564 2068 6572   implemented her
+00009a90: 652c 2074 6865 2072 6574 7572 6e20 7661  e, the return va
+00009aa0: 6c75 6520 636f 7272 6573 706f 6e64 7320  lue corresponds 
+00009ab0: 746f 2022 6e6f 5061 7265 6e74 7322 0a20  to "noParents". 
+00009ac0: 2020 2023 2028 6173 206f 6620 3230 3232     # (as of 2022
+00009ad0: 2d30 352d 3139 290a 2020 2020 6465 6620  -05-19).    def 
+00009ae0: 6765 745f 706c 6174 666f 726d 5f61 6e6e  get_platform_ann
+00009af0: 6f74 6174 696f 6e73 2873 656c 662c 2070  otations(self, p
+00009b00: 6c61 7466 6f72 6d3a 696e 747c 7374 722c  latform:int|str,
+00009b10: 202a 2a6b 7761 7267 7329 2d3e 4461 7461   **kwargs)->Data
+00009b20: 4672 616d 653a 0a20 2020 2020 2020 2022  Frame:.        "
+00009b30: 2222 4765 7473 2047 656d 6d61 2773 2070  ""Gets Gemma's p
+00009b40: 6c61 7466 6f72 6d20 616e 6e6f 7461 7469  latform annotati
+00009b50: 6f6e 7320 696e 636c 7564 696e 6720 6d61  ons including ma
+00009b60: 7070 696e 6773 206f 6620 6d69 6372 6f61  ppings of microa
+00009b70: 7272 6179 2070 726f 6265 7320 746f 2067  rray probes to g
+00009b80: 656e 6573 2e0a 2020 2020 2020 2020 0a20  enes..        . 
+00009b90: 2020 2020 2020 203a 7061 7261 6d20 706c         :param pl
+00009ba0: 6174 666f 726d 3a20 4120 706c 6174 666f  atform: A platfo
+00009bb0: 726d 206e 756d 6572 6963 616c 2069 6465  rm numerical ide
+00009bc0: 6e74 6966 6965 7220 6f72 2061 2070 6c61  ntifier or a pla
+00009bd0: 7466 6f72 6d20 7368 6f72 7420 6e61 6d65  tform short name
+00009be0: 0a20 2020 2020 2020 203a 7479 7065 2070  .        :type p
+00009bf0: 6c61 7466 6f72 6d3a 2069 6e74 7c73 7472  latform: int|str
+00009c00: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+00009c10: 2a2a 6b77 6172 6773 3a20 4164 6469 7469  **kwargs: Additi
+00009c20: 6f6e 616c 2061 7267 756d 656e 7473 2074  onal arguments t
+00009c30: 6f20 7061 7373 2074 6f20 7261 772e 6765  o pass to raw.ge
+00009c40: 745f 706c 6174 666f 726d 5f61 6e6e 6f74  t_platform_annot
+00009c50: 6174 696f 6e73 0a20 2020 2020 2020 203a  ations.        :
+00009c60: 7479 7065 202a 2a6b 7761 7267 733a 2054  type **kwargs: T
+00009c70: 5950 450a 2020 2020 2020 2020 3a72 6574  YPE.        :ret
+00009c80: 7572 6e3a 2041 2044 6174 6146 7261 6d65  urn: A DataFrame
+00009c90: 206f 6620 616e 6e6f 7461 7469 6f6e 730a   of annotations.
+00009ca0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00009cb0: 2020 202d 2050 726f 6265 4e61 6d65 3a20     - ProbeName: 
+00009cc0: 5072 6f62 6573 6574 206e 616d 6573 2070  Probeset names p
+00009cd0: 726f 7669 6465 6420 6279 2074 6865 2070  rovided by the p
+00009ce0: 6c61 7466 6f72 6d2e 2047 656e 6520 7379  latform. Gene sy
+00009cf0: 6d62 6f6c 7320 666f 7220 6765 6e65 7269  mbols for generi
+00009d00: 6320 616e 6e6f 7461 7469 6f6e 730a 2020  c annotations.  
+00009d10: 2020 2020 2020 2020 2d20 4765 6e65 5379          - GeneSy
+00009d20: 6d62 6f6c 733a 2047 656e 6573 2074 6861  mbols: Genes tha
+00009d30: 7420 7765 7265 2066 6f75 6e64 2074 6f20  t were found to 
+00009d40: 6265 2061 6c69 676e 6564 2074 6f20 7468  be aligned to th
+00009d50: 6520 7072 6f62 6520 7365 7175 656e 6365  e probe sequence
+00009d60: 2e20 4e6f 7465 2074 6861 7420 6974 2069  . Note that it i
+00009d70: 7320 706f 7373 6962 6c65 2066 6f72 2070  s possible for p
+00009d80: 726f 6265 7320 746f 2062 6520 6e6f 6e2d  robes to be non-
+00009d90: 7370 6563 6966 6963 2e20 416c 6967 6e6d  specific. Alignm
+00009da0: 656e 7420 746f 206d 756c 7469 706c 6520  ent to multiple 
+00009db0: 6765 6e65 7320 6172 6520 696e 6469 6361  genes are indica
+00009dc0: 7465 6420 7769 7468 2067 656e 6520 7379  ted with gene sy
+00009dd0: 6d62 6f6c 7320 7365 7061 7261 7465 6420  mbols separated 
+00009de0: 6279 2022 7c22 730a 2020 2020 2020 2020  by "|"s.        
+00009df0: 2020 2d20 4765 6e65 4e61 6d65 733a 204e    - GeneNames: N
+00009e00: 616d 6520 6f66 2074 6865 2067 656e 650a  ame of the gene.
+00009e10: 2020 2020 2020 2020 2020 2d20 474f 5465            - GOTe
+00009e20: 726d 733a 2047 4f20 5465 726d 7320 6173  rms: GO Terms as
+00009e30: 736f 6369 6174 6564 2077 6974 6820 7468  sociated with th
+00009e40: 6520 6765 6e65 732e 2061 6e6e 6f74 5479  e genes. annotTy
+00009e50: 7065 2061 7267 756d 656e 7420 6361 6e20  pe argument can 
+00009e60: 6265 2075 7365 6420 746f 2063 686f 6f73  be used to choos
+00009e70: 6520 7768 6963 6820 7465 726d 7320 7368  e which terms sh
+00009e80: 6f75 6c64 2062 6520 696e 636c 7564 6564  ould be included
+00009e90: 2e0a 2020 2020 2020 2020 2020 2d20 4765  ..          - Ge
+00009ea0: 6d6d 6149 4473 2061 6e64 204e 4342 4969  mmaIDs and NCBIi
+00009eb0: 6473 3a20 7265 7370 6563 7469 7665 2049  ds: respective I
+00009ec0: 4473 2066 6f72 2074 6865 2067 656e 6573  Ds for the genes
+00009ed0: 2e0a 2020 2020 2020 2020 3a72 7479 7065  ..        :rtype
+00009ee0: 3a20 4461 7461 4672 616d 650a 0a20 2020  : DataFrame..   
+00009ef0: 2020 2020 2022 2222 0a20 2020 2020 2020       """.       
+00009f00: 200a 2020 2020 2020 2020 6170 695f 7265   .        api_re
+00009f10: 7370 6f6e 7365 203d 2073 656c 662e 7261  sponse = self.ra
+00009f20: 772e 6765 745f 706c 6174 666f 726d 5f61  w.get_platform_a
+00009f30: 6e6e 6f74 6174 696f 6e73 2870 6c61 7466  nnotations(platf
+00009f40: 6f72 6d2c 202a 2a6b 7761 7267 7329 0a20  orm, **kwargs). 
+00009f50: 2020 2020 2020 2075 6e63 6f6d 6d65 6e74         uncomment
+00009f60: 203d 2061 7069 5f72 6573 706f 6e73 652e   = api_response.
+00009f70: 7370 6c69 7428 225c 6e23 2229 0a20 2020  split("\n#").   
+00009f80: 2020 2020 2061 7069 5f72 6573 706f 6e73       api_respons
+00009f90: 6520 3d20 756e 636f 6d6d 656e 745b 6c65  e = uncomment[le
+00009fa0: 6e28 756e 636f 6d6d 656e 7429 2d31 5d0a  n(uncomment)-1].
+00009fb0: 2020 2020 2020 2020 756e 636f 6d6d 656e          uncommen
+00009fc0: 7420 3d20 6170 695f 7265 7370 6f6e 7365  t = api_response
+00009fd0: 2e73 706c 6974 2827 5c6e 272c 3129 0a20  .split('\n',1). 
+00009fe0: 2020 2020 2020 2061 7069 5f72 6573 706f         api_respo
+00009ff0: 6e73 6520 3d20 756e 636f 6d6d 656e 745b  nse = uncomment[
+0000a000: 6c65 6e28 756e 636f 6d6d 656e 7429 2d31  len(uncomment)-1
+0000a010: 5d0a 2020 2020 2020 2020 7265 7475 726e  ].        return
+0000a020: 2070 642e 7265 6164 5f63 7376 2853 7472   pd.read_csv(Str
+0000a030: 696e 6749 4f28 6170 695f 7265 7370 6f6e  ingIO(api_respon
+0000a040: 7365 292c 2073 6570 3d27 5c74 2729 0a0a  se), sep='\t')..
+0000a050: 2020 2020 2320 706c 6174 666f 726d 2f7b      # platform/{
+0000a060: 706c 6174 666f 726d 7d2f 6461 7461 7365  platform}/datase
+0000a070: 7473 2c20 6765 745f 706c 6174 666f 726d  ts, get_platform
+0000a080: 5f64 6174 6173 6574 7320 2d2d 2d2d 0a0a  _datasets ----..
+0000a090: 2020 2020 6465 6620 6765 745f 706c 6174      def get_plat
+0000a0a0: 666f 726d 5f64 6174 6173 6574 7328 7365  form_datasets(se
+0000a0b0: 6c66 2c20 706c 6174 666f 726d 3a73 7472  lf, platform:str
+0000a0c0: 7c69 6e74 2c0a 2020 2020 2020 2020 2020  |int,.          
+0000a0d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000a0e0: 2020 2020 6f66 6673 6574 3a69 6e74 203d      offset:int =
+0000a0f0: 2030 2c0a 2020 2020 2020 2020 2020 2020   0,.            
+0000a100: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000a110: 2020 6c69 6d69 743a 696e 7420 3d20 3230    limit:int = 20
+0000a120: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+0000a130: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000a140: 2a2a 6b77 6172 6773 292d 3e44 6174 6146  **kwargs)->DataF
+0000a150: 7261 6d65 3a0a 2020 2020 2020 2020 2222  rame:.        ""
+0000a160: 2252 6574 7269 6576 6520 616c 6c20 6578  "Retrieve all ex
+0000a170: 7065 7269 6d65 6e74 7320 7573 696e 6720  periments using 
+0000a180: 6120 6769 7665 6e20 706c 6174 666f 726d  a given platform
+0000a190: 0a0a 2020 2020 2020 2020 0a20 2020 2020  ..        .     
+0000a1a0: 2020 203a 7061 7261 6d20 706c 6174 666f     :param platfo
+0000a1b0: 726d 3a20 4120 706c 6174 666f 726d 206e  rm: A platform n
+0000a1c0: 756d 6572 6963 616c 2069 6465 6e74 6966  umerical identif
+0000a1d0: 6965 7220 6f72 2061 2070 6c61 7466 6f72  ier or a platfor
+0000a1e0: 6d20 7368 6f72 7420 6e61 6d65 0a20 2020  m short name.   
+0000a1f0: 2020 2020 203a 7479 7065 2070 6c61 7466       :type platf
+0000a200: 6f72 6d3a 2073 7472 7c69 6e74 0a20 2020  orm: str|int.   
+0000a210: 2020 2020 203a 7061 7261 6d20 6f66 6673       :param offs
+0000a220: 6574 3a20 5468 6520 6f66 6673 6574 206f  et: The offset o
+0000a230: 6620 7468 6520 6669 7273 7420 7265 7472  f the first retr
+0000a240: 6965 7665 6420 7265 7375 6c74 2e2c 2064  ieved result., d
+0000a250: 6566 6175 6c74 7320 746f 2030 0a20 2020  efaults to 0.   
+0000a260: 2020 2020 203a 7479 7065 206f 6666 7365       :type offse
+0000a270: 743a 2069 6e74 2c20 6f70 7469 6f6e 616c  t: int, optional
+0000a280: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+0000a290: 6c69 6d69 743a 204c 696d 6974 7320 7468  limit: Limits th
+0000a2a0: 6520 7265 7375 6c74 2074 6f20 7370 6563  e result to spec
+0000a2b0: 6966 6965 6420 616d 6f75 6e74 206f 6620  ified amount of 
+0000a2c0: 6f62 6a65 6374 732e 0a20 2020 2020 2020  objects..       
+0000a2d0: 2020 2048 6173 2061 206d 6178 696d 756d     Has a maximum
+0000a2e0: 2076 616c 7565 206f 6620 3130 302e 2055   value of 100. U
+0000a2f0: 7365 2074 6f67 6574 6865 7220 7769 7468  se together with
+0000a300: 206f 6666 7365 7420 616e 6420 7468 6520   offset and the 
+0000a310: 0a20 2020 2020 2020 2020 2074 6f74 616c  .          total
+0000a320: 5f65 6c65 6d65 6e74 7320 6174 7472 6962  _elements attrib
+0000a330: 7574 6520 696e 2074 6865 206f 7574 7075  ute in the outpu
+0000a340: 7420 746f 2063 6f6d 7069 6c65 2061 6c6c  t to compile all
+0000a350: 2064 6174 6120 6966 206e 6565 6465 642e   data if needed.
+0000a360: 0a20 2020 2020 2020 2020 2041 6c74 6572  .          Alter
+0000a370: 6e61 7469 7665 6c79 2067 6574 5f61 6c6c  natively get_all
+0000a380: 5f70 6167 6573 2066 756e 6374 696f 6e20  _pages function 
+0000a390: 6361 6e20 6265 2075 7365 6420 7769 7468  can be used with
+0000a3a0: 2061 6c6c 2066 756e 6374 696f 6e73 0a20   all functions. 
+0000a3b0: 2020 2020 2020 2020 2069 6e63 6c75 6469           includi
+0000a3c0: 6e67 206f 6666 7365 7420 616e 6420 6c69  ng offset and li
+0000a3d0: 6d69 7420 7061 7261 6d65 7465 7273 2c20  mit parameters, 
+0000a3e0: 6465 6661 756c 7473 2074 6f20 3230 0a20  defaults to 20. 
+0000a3f0: 2020 2020 2020 203a 7479 7065 206c 696d         :type lim
+0000a400: 6974 3a20 696e 742c 206f 7074 696f 6e61  it: int, optiona
+0000a410: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
+0000a420: 202a 2a6b 7761 7267 733a 2041 6464 6974   **kwargs: Addit
+0000a430: 696f 6e61 6c20 6172 6775 6d65 6e74 7320  ional arguments 
+0000a440: 746f 2070 6173 7320 746f 2072 6177 2e67  to pass to raw.g
+0000a450: 6574 5f70 6c61 7466 6f72 6d5f 6461 7461  et_platform_data
+0000a460: 7365 7473 0a20 2020 2020 2020 203a 7265  sets.        :re
+0000a470: 7475 726e 3a20 4120 4461 7461 4672 616d  turn: A DataFram
+0000a480: 6520 7769 7468 2069 6e66 6f72 6d61 7469  e with informati
+0000a490: 6f6e 2061 626f 7574 2074 6865 2071 7565  on about the que
+0000a4a0: 7269 6564 2064 6174 6173 6574 2873 292e  ried dataset(s).
+0000a4b0: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
+0000a4c0: 2020 2020 5468 6520 6669 656c 6473 206f      The fields o
+0000a4d0: 6620 7468 6520 4461 7461 4672 616d 6520  f the DataFrame 
+0000a4e0: 6172 653a 0a20 2020 2020 2020 2020 2020  are:.           
+0000a4f0: 202d 2065 7870 6572 696d 656e 745f 7368   - experiment_sh
+0000a500: 6f72 745f 6e61 6d65 3a20 5368 6f72 746e  ort_name: Shortn
+0000a510: 616d 6520 6769 7665 6e20 746f 2074 6865  ame given to the
+0000a520: 2064 6174 6173 6574 2077 6974 6869 6e20   dataset within 
+0000a530: 4765 6d6d 612e 204f 6674 656e 2063 6f72  Gemma. Often cor
+0000a540: 7265 7370 6f6e 6473 2074 6f20 6163 6365  responds to acce
+0000a550: 7373 696f 6e20 4944 0a20 2020 2020 2020  ssion ID.       
 0000a560: 2020 2020 202d 2065 7870 6572 696d 656e       - experimen
-0000a570: 745f 4944 3a20 496e 7465 726e 616c 2049  t_ID: Internal I
-0000a580: 4420 6f66 2074 6865 2064 6174 6173 6574  D of the dataset
-0000a590: 2e0a 2020 2020 2020 2020 2020 2020 2d20  ..            - 
-0000a5a0: 6578 7065 7269 6d65 6e74 5f64 6573 6372  experiment_descr
-0000a5b0: 6970 7469 6f6e 3a20 4465 7363 7269 7074  iption: Descript
-0000a5c0: 696f 6e20 6f66 2074 6865 2064 6174 6173  ion of the datas
-0000a5d0: 6574 0a20 2020 2020 2020 2020 2020 202d  et.            -
-0000a5e0: 2065 7870 6572 696d 656e 745f 7472 6f75   experiment_trou
-0000a5f0: 626c 6564 3a20 4469 6420 616e 2061 7574  bled: Did an aut
-0000a600: 6f6d 6174 6963 2070 726f 6365 7373 2077  omatic process w
-0000a610: 6974 6869 6e20 6765 6d6d 6120 6f72 2061  ithin gemma or a
-0000a620: 2063 7572 6174 6f72 206d 6172 6b20 7468   curator mark th
-0000a630: 6520 6461 7461 7365 7420 6173 2022 7472  e dataset as "tr
-0000a640: 6f75 626c 6564 220a 2020 2020 2020 2020  oubled".        
-0000a650: 2020 2020 2d20 6578 7065 7269 6d65 6e74      - experiment
-0000a660: 5f61 6363 6573 7369 6f6e 3a20 4163 6365  _accession: Acce
-0000a670: 7373 696f 6e20 4944 206f 6620 7468 6520  ssion ID of the 
-0000a680: 6461 7461 7365 7420 696e 2074 6865 2065  dataset in the e
-0000a690: 7874 6572 6e61 6c20 6461 7461 6261 7365  xternal database
-0000a6a0: 2069 7420 7761 7320 7461 6b65 6e20 6672   it was taken fr
-0000a6b0: 6f6d 0a20 2020 2020 2020 2020 2020 202d  om.            -
-0000a6c0: 2065 7870 6572 696d 656e 745f 6461 7461   experiment_data
-0000a6d0: 6261 7365 3a20 5468 6520 6e61 6d65 206f  base: The name o
-0000a6e0: 6620 7468 6520 6461 7461 6261 7365 2077  f the database w
-0000a6f0: 6865 7265 2074 6865 2064 6174 6173 6574  here the dataset
-0000a700: 2077 6173 2074 616b 656e 2066 726f 6d0a   was taken from.
-0000a710: 2020 2020 2020 2020 2020 2020 2d20 6578              - ex
-0000a720: 7065 7269 6d65 6e74 5f55 5249 3a20 5552  periment_URI: UR
-0000a730: 4920 6f66 2074 6865 206f 7269 6769 6e61  I of the origina
-0000a740: 6c20 6461 7461 6261 7365 0a20 2020 2020  l database.     
-0000a750: 2020 2020 2020 202d 2065 7870 6572 696d         - experim
-0000a760: 656e 745f 7361 6d70 6c65 5f63 6f75 6e74  ent_sample_count
-0000a770: 3a20 4e75 6d62 6572 206f 6620 7361 6d70  : Number of samp
-0000a780: 6c65 7320 696e 2074 6865 2064 6174 6173  les in the datas
-0000a790: 6574 0a20 2020 2020 2020 2020 2020 202d  et.            -
-0000a7a0: 2065 7870 6572 696d 656e 745f 6261 7463   experiment_batc
-0000a7b0: 685f 6566 6665 6374 5f74 6578 743a 2041  h_effect_text: A
-0000a7c0: 2074 6578 7420 6669 656c 6420 6465 7363   text field desc
-0000a7d0: 7269 6269 6e67 2077 6865 7468 6572 2074  ribing whether t
-0000a7e0: 6865 2064 6174 6173 6574 2068 6173 2062  he dataset has b
-0000a7f0: 6174 6368 2065 6666 6563 7473 0a20 2020  atch effects.   
-0000a800: 2020 2020 2020 2020 202d 2065 7870 6572           - exper
-0000a810: 696d 656e 5f62 6174 6368 5f63 6f72 7265  imen_batch_corre
-0000a820: 6374 6564 3a20 5768 6574 6865 7220 6261  cted: Whether ba
-0000a830: 7463 6820 636f 7272 6563 7469 6f6e 2068  tch correction h
-0000a840: 6173 2062 6565 6e20 7065 7266 6f72 6d65  as been performe
-0000a850: 6420 6f6e 2074 6865 2064 6174 6173 6574  d on the dataset
-0000a860: 2e0a 2020 2020 2020 2020 2020 2020 2d20  ..            - 
-0000a870: 6578 7065 7269 6d65 6e5f 6261 7463 685f  experimen_batch_
-0000a880: 636f 6e66 6f75 6e64 3a20 3020 6966 2062  confound: 0 if b
-0000a890: 6174 6368 2069 6e66 6f20 6973 6e27 7420  atch info isn't 
-0000a8a0: 6176 6169 6c61 626c 652c 202d 3120 6966  available, -1 if
-0000a8b0: 2062 6174 6368 2063 6f75 6e66 6f75 6420   batch counfoud 
-0000a8c0: 6973 2064 6574 6563 7465 642c 2031 2069  is detected, 1 i
-0000a8d0: 6620 6261 7463 6820 696e 666f 726d 6174  f batch informat
-0000a8e0: 696f 6e20 6973 2061 7661 696c 6162 6c65  ion is available
-0000a8f0: 2061 6e64 206e 6f20 6261 7463 6820 636f   and no batch co
-0000a900: 6e66 6f75 6e64 2066 6f75 6e64 0a20 2020  nfound found.   
-0000a910: 2020 2020 2020 2020 202d 2065 7870 6572           - exper
-0000a920: 696d 656e 5f62 6174 6368 5f65 6666 6563  imen_batch_effec
-0000a930: 743a 202d 3120 6966 2062 6174 6368 2070  t: -1 if batch p
-0000a940: 2076 616c 7565 203c 2030 2e30 3030 312c   value < 0.0001,
-0000a950: 2031 2069 6620 6261 7463 6820 7020 7661   1 if batch p va
-0000a960: 6c75 6520 3e20 302e 312c 2030 2069 6620  lue > 0.1, 0 if 
-0000a970: 6f74 6865 7277 6973 6520 616e 6420 7768  otherwise and wh
-0000a980: 656e 2074 6865 7265 2069 7320 6e6f 2062  en there is no b
-0000a990: 6174 6368 2069 6e66 6f72 6d61 7469 6f6e  atch information
-0000a9a0: 2069 7320 6176 6169 6c61 626c 6520 6f72   is available or
-0000a9b0: 2077 6865 6e20 7468 6520 6461 7461 2069   when the data i
-0000a9c0: 7320 636f 6e66 6f75 6e64 6564 2077 6974  s confounded wit
-0000a9d0: 6820 6261 7463 6865 732e 0a20 2020 2020  h batches..     
-0000a9e0: 2020 2020 2020 202d 2065 7870 6572 696d         - experim
-0000a9f0: 656e 5f72 6177 5f64 6174 613a 202d 3120  en_raw_data: -1 
-0000aa00: 6966 206e 6f20 7261 7720 6461 7461 2061  if no raw data a
-0000aa10: 7661 696c 6162 6c65 2c20 3120 6966 2072  vailable, 1 if r
-0000aa20: 6177 2064 6174 6120 7761 7320 6176 6169  aw data was avai
-0000aa30: 6c61 626c 652e 2057 6865 6e20 6176 6169  lable. When avai
-0000aa40: 6c61 626c 652c 2047 656d 6d61 2072 6570  lable, Gemma rep
-0000aa50: 726f 6365 7373 6573 2072 6177 2064 6174  rocesses raw dat
-0000aa60: 6120 746f 2067 6574 2065 7870 7265 7373  a to get express
-0000aa70: 696f 6e20 7661 6c75 6573 2061 6e64 2062  ion values and b
-0000aa80: 6174 6368 6573 0a20 2020 2020 2020 2020  atches.         
-0000aa90: 2020 202d 2067 6565 715f 715f 7363 6f72     - geeq_q_scor
-0000aaa0: 653a 2053 7569 7461 6269 6c69 7479 2073  e: Suitability s
-0000aab0: 636f 7265 2067 6976 656e 2074 6f20 7468  core given to th
-0000aac0: 6520 6461 7461 7365 7420 6279 2047 656d  e dataset by Gem
-0000aad0: 6d61 2e20 5265 6665 7273 2074 6f20 6661  ma. Refers to fa
-0000aae0: 6374 6f72 7320 6c69 6b65 2062 6174 6368  ctors like batch
-0000aaf0: 6573 2c20 706c 6174 666f 726d 7320 616e  es, platforms an
-0000ab00: 6420 6f74 6865 7220 6173 7065 6374 7320  d other aspects 
-0000ab10: 6f66 2065 7870 6572 696d 656e 7461 6c20  of experimental 
-0000ab20: 6465 7369 676e 0a20 2020 2020 2020 2020  design.         
-0000ab30: 2020 202d 2067 6565 715f 735f 7363 6f72     - geeq_s_scor
-0000ab40: 653a 2044 6174 6120 7175 616c 6974 7920  e: Data quality 
-0000ab50: 7363 6f72 6520 6769 7665 6e20 746f 2074  score given to t
-0000ab60: 6865 2064 6174 6173 6574 2062 7920 4765  he dataset by Ge
-0000ab70: 6d6d 612e 0a20 2020 2020 2020 2020 2020  mma..           
-0000ab80: 202d 2074 6178 6f6e 5f6e 616d 653a 204e   - taxon_name: N
-0000ab90: 616d 6520 6f66 2074 6865 2073 7065 6369  ame of the speci
-0000aba0: 6573 0a20 2020 2020 2020 2020 2020 202d  es.            -
-0000abb0: 2074 6178 6f6e 5f73 6369 656e 7469 6669   taxon_scientifi
-0000abc0: 633a 2053 6369 656e 7469 6669 6320 6e61  c: Scientific na
-0000abd0: 6d65 2066 6f72 2074 6865 2074 6178 6f6e  me for the taxon
-0000abe0: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
-0000abf0: 6178 6f6e 5f49 443a 2049 6e74 6572 6e61  axon_ID: Interna
-0000ac00: 6c20 6964 656e 7469 6669 6572 2067 6976  l identifier giv
-0000ac10: 656e 2074 6f20 7468 6520 7370 6563 6965  en to the specie
-0000ac20: 7320 6279 2047 656d 6d61 0a20 2020 2020  s by Gemma.     
-0000ac30: 2020 2020 2020 202d 2074 6178 6f6e 5f4e         - taxon_N
-0000ac40: 4342 493a 204e 4342 4920 4944 206f 6620  CBI: NCBI ID of 
-0000ac50: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
-0000ac60: 2020 2020 2020 2d20 7461 786f 6e5f 6461        - taxon_da
-0000ac70: 7461 6261 7365 5f6e 616d 653a 2055 6e64  tabase_name: Und
-0000ac80: 6572 6c79 696e 6720 6461 7461 6261 7365  erlying database
-0000ac90: 2075 7365 6420 696e 2047 656d 6d61 2066   used in Gemma f
-0000aca0: 6f72 2074 6865 2074 6178 6f6e 0a20 2020  or the taxon.   
-0000acb0: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
-0000acc0: 5f64 6174 6162 6173 655f 4944 3a20 4944  _database_ID: ID
-0000acd0: 206f 6620 7468 6520 756e 6465 7279 6c69   of the underyli
-0000ace0: 6e67 2064 6174 6162 6173 6520 7573 6564  ng database used
-0000acf0: 2069 6e20 4765 6d6d 6120 666f 7220 7468   in Gemma for th
-0000ad00: 6520 7461 786f 6e0a 2020 2020 2020 2020  e taxon.        
-0000ad10: 3a72 7479 7065 3a20 4461 7461 4672 616d  :rtype: DataFram
-0000ad20: 650a 0a20 2020 2020 2020 2022 2222 0a20  e..        """. 
-0000ad30: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-0000ad40: 6b77 6172 6773 203d 2076 732e 7265 6d6f  kwargs = vs.remo
-0000ad50: 7665 5f6e 6f6e 6573 286f 6666 7365 7420  ve_nones(offset 
-0000ad60: 3d20 6f66 6673 6574 2c0a 2020 2020 2020  = offset,.      
-0000ad70: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ad80: 2020 2020 2020 2020 2020 206c 696d 6974             limit
-0000ad90: 203d 206c 696d 6974 2c0a 2020 2020 2020   = limit,.      
-0000ada0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000adb0: 2020 2020 2020 2020 2020 202a 2a6b 7761             **kwa
-0000adc0: 7267 7329 0a20 2020 2020 2020 200a 2020  rgs).        .  
-0000add0: 2020 2020 2020 0a20 2020 2020 2020 2072        .        r
-0000ade0: 6573 706f 6e73 6520 3d20 7365 6c66 2e72  esponse = self.r
-0000adf0: 6177 2e67 6574 5f70 6c61 7466 6f72 6d5f  aw.get_platform_
-0000ae00: 6461 7461 7365 7473 2870 6c61 7466 6f72  datasets(platfor
-0000ae10: 6d2c 202a 2a6b 7761 7267 7329 0a20 2020  m, **kwargs).   
-0000ae20: 2020 2020 2064 6620 3d20 7073 2e70 726f       df = ps.pro
-0000ae30: 6365 7373 5f64 6174 6173 6574 7328 7265  cess_datasets(re
-0000ae40: 7370 6f6e 7365 2e64 6174 6129 0a20 2020  sponse.data).   
-0000ae50: 2020 2020 2070 732e 6174 7461 6368 5f61       ps.attach_a
-0000ae60: 7474 7269 6275 7465 7328 6466 2c20 7265  ttributes(df, re
-0000ae70: 7370 6f6e 7365 2e74 6f5f 6469 6374 2829  sponse.to_dict()
-0000ae80: 290a 2020 2020 2020 2020 7265 7475 726e  ).        return
-0000ae90: 2064 660a 0a20 2020 2023 2070 6c61 7466   df..    # platf
-0000aea0: 6f72 6d73 2f7b 706c 6174 666f 726d 7d2f  orms/{platform}/
-0000aeb0: 656c 656d 656e 7473 2f7b 7072 6f62 6573  elements/{probes
-0000aec0: 7d20 2d2d 2d2d 2d0a 2020 2020 2320 6e6f  } -----.    # no
-0000aed0: 7420 696d 706c 656d 656e 7465 640a 2020  t implemented.  
-0000aee0: 2020 0a20 2020 2023 2070 6c61 7466 6f72    .    # platfor
-0000aef0: 6d73 2f7b 706c 6174 666f 726d 7d2f 656c  ms/{platform}/el
-0000af00: 656d 656e 7473 2f7b 7072 6f62 657d 2f67  ements/{probe}/g
-0000af10: 656e 6573 2c20 6765 745f 706c 6174 666f  enes, get_platfo
-0000af20: 726d 5f65 6c65 6d65 6e74 5f67 656e 6573  rm_element_genes
-0000af30: 202d 2d2d 2d0a 0a20 2020 2064 6566 2067   ----..    def g
-0000af40: 6574 5f70 6c61 7466 6f72 6d5f 656c 656d  et_platform_elem
-0000af50: 656e 745f 6765 6e65 7328 7365 6c66 2c20  ent_genes(self, 
-0000af60: 706c 6174 666f 726d 3a73 7472 7c69 6e74  platform:str|int
-0000af70: 2c20 0a20 2020 2020 2020 2020 2020 2020  , .             
-0000af80: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000af90: 2020 2020 2020 7072 6f62 653a 7374 727c        probe:str|
-0000afa0: 696e 742c 0a20 2020 2020 2020 2020 2020  int,.           
+0000a570: 745f 6e61 6d65 3a20 4675 6c6c 2074 6974  t_name: Full tit
+0000a580: 6c65 206f 6620 7468 6520 6461 7461 7365  le of the datase
+0000a590: 740a 2020 2020 2020 2020 2020 2020 2d20  t.            - 
+0000a5a0: 6578 7065 7269 6d65 6e74 5f49 443a 2049  experiment_ID: I
+0000a5b0: 6e74 6572 6e61 6c20 4944 206f 6620 7468  nternal ID of th
+0000a5c0: 6520 6461 7461 7365 742e 0a20 2020 2020  e dataset..     
+0000a5d0: 2020 2020 2020 202d 2065 7870 6572 696d         - experim
+0000a5e0: 656e 745f 6465 7363 7269 7074 696f 6e3a  ent_description:
+0000a5f0: 2044 6573 6372 6970 7469 6f6e 206f 6620   Description of 
+0000a600: 7468 6520 6461 7461 7365 740a 2020 2020  the dataset.    
+0000a610: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
+0000a620: 6d65 6e74 5f74 726f 7562 6c65 643a 2044  ment_troubled: D
+0000a630: 6964 2061 6e20 6175 746f 6d61 7469 6320  id an automatic 
+0000a640: 7072 6f63 6573 7320 7769 7468 696e 2067  process within g
+0000a650: 656d 6d61 206f 7220 6120 6375 7261 746f  emma or a curato
+0000a660: 7220 6d61 726b 2074 6865 2064 6174 6173  r mark the datas
+0000a670: 6574 2061 7320 2274 726f 7562 6c65 6422  et as "troubled"
+0000a680: 0a20 2020 2020 2020 2020 2020 202d 2065  .            - e
+0000a690: 7870 6572 696d 656e 745f 6163 6365 7373  xperiment_access
+0000a6a0: 696f 6e3a 2041 6363 6573 7369 6f6e 2049  ion: Accession I
+0000a6b0: 4420 6f66 2074 6865 2064 6174 6173 6574  D of the dataset
+0000a6c0: 2069 6e20 7468 6520 6578 7465 726e 616c   in the external
+0000a6d0: 2064 6174 6162 6173 6520 6974 2077 6173   database it was
+0000a6e0: 2074 616b 656e 2066 726f 6d0a 2020 2020   taken from.    
+0000a6f0: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
+0000a700: 6d65 6e74 5f64 6174 6162 6173 653a 2054  ment_database: T
+0000a710: 6865 206e 616d 6520 6f66 2074 6865 2064  he name of the d
+0000a720: 6174 6162 6173 6520 7768 6572 6520 7468  atabase where th
+0000a730: 6520 6461 7461 7365 7420 7761 7320 7461  e dataset was ta
+0000a740: 6b65 6e20 6672 6f6d 0a20 2020 2020 2020  ken from.       
+0000a750: 2020 2020 202d 2065 7870 6572 696d 656e       - experimen
+0000a760: 745f 5552 493a 2055 5249 206f 6620 7468  t_URI: URI of th
+0000a770: 6520 6f72 6967 696e 616c 2064 6174 6162  e original datab
+0000a780: 6173 650a 2020 2020 2020 2020 2020 2020  ase.            
+0000a790: 2d20 6578 7065 7269 6d65 6e74 5f73 616d  - experiment_sam
+0000a7a0: 706c 655f 636f 756e 743a 204e 756d 6265  ple_count: Numbe
+0000a7b0: 7220 6f66 2073 616d 706c 6573 2069 6e20  r of samples in 
+0000a7c0: 7468 6520 6461 7461 7365 740a 2020 2020  the dataset.    
+0000a7d0: 2020 2020 2020 2020 2d20 6578 7065 7269          - experi
+0000a7e0: 6d65 6e74 5f62 6174 6368 5f65 6666 6563  ment_batch_effec
+0000a7f0: 745f 7465 7874 3a20 4120 7465 7874 2066  t_text: A text f
+0000a800: 6965 6c64 2064 6573 6372 6962 696e 6720  ield describing 
+0000a810: 7768 6574 6865 7220 7468 6520 6461 7461  whether the data
+0000a820: 7365 7420 6861 7320 6261 7463 6820 6566  set has batch ef
+0000a830: 6665 6374 730a 2020 2020 2020 2020 2020  fects.          
+0000a840: 2020 2d20 6578 7065 7269 6d65 6e5f 6261    - experimen_ba
+0000a850: 7463 685f 636f 7272 6563 7465 643a 2057  tch_corrected: W
+0000a860: 6865 7468 6572 2062 6174 6368 2063 6f72  hether batch cor
+0000a870: 7265 6374 696f 6e20 6861 7320 6265 656e  rection has been
+0000a880: 2070 6572 666f 726d 6564 206f 6e20 7468   performed on th
+0000a890: 6520 6461 7461 7365 742e 0a20 2020 2020  e dataset..     
+0000a8a0: 2020 2020 2020 202d 2065 7870 6572 696d         - experim
+0000a8b0: 656e 5f62 6174 6368 5f63 6f6e 666f 756e  en_batch_confoun
+0000a8c0: 643a 2030 2069 6620 6261 7463 6820 696e  d: 0 if batch in
+0000a8d0: 666f 2069 736e 2774 2061 7661 696c 6162  fo isn't availab
+0000a8e0: 6c65 2c20 2d31 2069 6620 6261 7463 6820  le, -1 if batch 
+0000a8f0: 636f 756e 666f 7564 2069 7320 6465 7465  counfoud is dete
+0000a900: 6374 6564 2c20 3120 6966 2062 6174 6368  cted, 1 if batch
+0000a910: 2069 6e66 6f72 6d61 7469 6f6e 2069 7320   information is 
+0000a920: 6176 6169 6c61 626c 6520 616e 6420 6e6f  available and no
+0000a930: 2062 6174 6368 2063 6f6e 666f 756e 6420   batch confound 
+0000a940: 666f 756e 640a 2020 2020 2020 2020 2020  found.          
+0000a950: 2020 2d20 6578 7065 7269 6d65 6e5f 6261    - experimen_ba
+0000a960: 7463 685f 6566 6665 6374 3a20 2d31 2069  tch_effect: -1 i
+0000a970: 6620 6261 7463 6820 7020 7661 6c75 6520  f batch p value 
+0000a980: 3c20 302e 3030 3031 2c20 3120 6966 2062  < 0.0001, 1 if b
+0000a990: 6174 6368 2070 2076 616c 7565 203e 2030  atch p value > 0
+0000a9a0: 2e31 2c20 3020 6966 206f 7468 6572 7769  .1, 0 if otherwi
+0000a9b0: 7365 2061 6e64 2077 6865 6e20 7468 6572  se and when ther
+0000a9c0: 6520 6973 206e 6f20 6261 7463 6820 696e  e is no batch in
+0000a9d0: 666f 726d 6174 696f 6e20 6973 2061 7661  formation is ava
+0000a9e0: 696c 6162 6c65 206f 7220 7768 656e 2074  ilable or when t
+0000a9f0: 6865 2064 6174 6120 6973 2063 6f6e 666f  he data is confo
+0000aa00: 756e 6465 6420 7769 7468 2062 6174 6368  unded with batch
+0000aa10: 6573 2e0a 2020 2020 2020 2020 2020 2020  es..            
+0000aa20: 2d20 6578 7065 7269 6d65 6e5f 7261 775f  - experimen_raw_
+0000aa30: 6461 7461 3a20 2d31 2069 6620 6e6f 2072  data: -1 if no r
+0000aa40: 6177 2064 6174 6120 6176 6169 6c61 626c  aw data availabl
+0000aa50: 652c 2031 2069 6620 7261 7720 6461 7461  e, 1 if raw data
+0000aa60: 2077 6173 2061 7661 696c 6162 6c65 2e20   was available. 
+0000aa70: 5768 656e 2061 7661 696c 6162 6c65 2c20  When available, 
+0000aa80: 4765 6d6d 6120 7265 7072 6f63 6573 7365  Gemma reprocesse
+0000aa90: 7320 7261 7720 6461 7461 2074 6f20 6765  s raw data to ge
+0000aaa0: 7420 6578 7072 6573 7369 6f6e 2076 616c  t expression val
+0000aab0: 7565 7320 616e 6420 6261 7463 6865 730a  ues and batches.
+0000aac0: 2020 2020 2020 2020 2020 2020 2d20 6765              - ge
+0000aad0: 6571 5f71 5f73 636f 7265 3a20 5375 6974  eq_q_score: Suit
+0000aae0: 6162 696c 6974 7920 7363 6f72 6520 6769  ability score gi
+0000aaf0: 7665 6e20 746f 2074 6865 2064 6174 6173  ven to the datas
+0000ab00: 6574 2062 7920 4765 6d6d 612e 2052 6566  et by Gemma. Ref
+0000ab10: 6572 7320 746f 2066 6163 746f 7273 206c  ers to factors l
+0000ab20: 696b 6520 6261 7463 6865 732c 2070 6c61  ike batches, pla
+0000ab30: 7466 6f72 6d73 2061 6e64 206f 7468 6572  tforms and other
+0000ab40: 2061 7370 6563 7473 206f 6620 6578 7065   aspects of expe
+0000ab50: 7269 6d65 6e74 616c 2064 6573 6967 6e0a  rimental design.
+0000ab60: 2020 2020 2020 2020 2020 2020 2d20 6765              - ge
+0000ab70: 6571 5f73 5f73 636f 7265 3a20 4461 7461  eq_s_score: Data
+0000ab80: 2071 7561 6c69 7479 2073 636f 7265 2067   quality score g
+0000ab90: 6976 656e 2074 6f20 7468 6520 6461 7461  iven to the data
+0000aba0: 7365 7420 6279 2047 656d 6d61 2e0a 2020  set by Gemma..  
+0000abb0: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
+0000abc0: 6e5f 6e61 6d65 3a20 4e61 6d65 206f 6620  n_name: Name of 
+0000abd0: 7468 6520 7370 6563 6965 730a 2020 2020  the species.    
+0000abe0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
+0000abf0: 7363 6965 6e74 6966 6963 3a20 5363 6965  scientific: Scie
+0000ac00: 6e74 6966 6963 206e 616d 6520 666f 7220  ntific name for 
+0000ac10: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
+0000ac20: 2020 2020 2020 2d20 7461 786f 6e5f 4944        - taxon_ID
+0000ac30: 3a20 496e 7465 726e 616c 2069 6465 6e74  : Internal ident
+0000ac40: 6966 6965 7220 6769 7665 6e20 746f 2074  ifier given to t
+0000ac50: 6865 2073 7065 6369 6573 2062 7920 4765  he species by Ge
+0000ac60: 6d6d 610a 2020 2020 2020 2020 2020 2020  mma.            
+0000ac70: 2d20 7461 786f 6e5f 4e43 4249 3a20 4e43  - taxon_NCBI: NC
+0000ac80: 4249 2049 4420 6f66 2074 6865 2074 6178  BI ID of the tax
+0000ac90: 6f6e 0a20 2020 2020 2020 2020 2020 202d  on.            -
+0000aca0: 2074 6178 6f6e 5f64 6174 6162 6173 655f   taxon_database_
+0000acb0: 6e61 6d65 3a20 556e 6465 726c 7969 6e67  name: Underlying
+0000acc0: 2064 6174 6162 6173 6520 7573 6564 2069   database used i
+0000acd0: 6e20 4765 6d6d 6120 666f 7220 7468 6520  n Gemma for the 
+0000ace0: 7461 786f 6e0a 2020 2020 2020 2020 2020  taxon.          
+0000acf0: 2020 2d20 7461 786f 6e5f 6461 7461 6261    - taxon_databa
+0000ad00: 7365 5f49 443a 2049 4420 6f66 2074 6865  se_ID: ID of the
+0000ad10: 2075 6e64 6572 796c 696e 6720 6461 7461   underyling data
+0000ad20: 6261 7365 2075 7365 6420 696e 2047 656d  base used in Gem
+0000ad30: 6d61 2066 6f72 2074 6865 2074 6178 6f6e  ma for the taxon
+0000ad40: 0a20 2020 2020 2020 203a 7274 7970 653a  .        :rtype:
+0000ad50: 2044 6174 6146 7261 6d65 0a0a 2020 2020   DataFrame..    
+0000ad60: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
+0000ad70: 0a20 2020 2020 2020 206b 7761 7267 7320  .        kwargs 
+0000ad80: 3d20 7673 2e72 656d 6f76 655f 6e6f 6e65  = vs.remove_none
+0000ad90: 7328 6f66 6673 6574 203d 206f 6666 7365  s(offset = offse
+0000ada0: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
+0000adb0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000adc0: 2020 2020 6c69 6d69 7420 3d20 6c69 6d69      limit = limi
+0000add0: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
+0000ade0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000adf0: 2020 2020 2a2a 6b77 6172 6773 290a 2020      **kwargs).  
+0000ae00: 2020 2020 2020 0a20 2020 2020 2020 200a        .        .
+0000ae10: 2020 2020 2020 2020 7265 7370 6f6e 7365          response
+0000ae20: 203d 2073 656c 662e 7261 772e 6765 745f   = self.raw.get_
+0000ae30: 706c 6174 666f 726d 5f64 6174 6173 6574  platform_dataset
+0000ae40: 7328 706c 6174 666f 726d 2c20 2a2a 6b77  s(platform, **kw
+0000ae50: 6172 6773 290a 2020 2020 2020 2020 6466  args).        df
+0000ae60: 203d 2070 732e 7072 6f63 6573 735f 6461   = ps.process_da
+0000ae70: 7461 7365 7473 2872 6573 706f 6e73 652e  tasets(response.
+0000ae80: 6461 7461 290a 2020 2020 2020 2020 7073  data).        ps
+0000ae90: 2e61 7474 6163 685f 6174 7472 6962 7574  .attach_attribut
+0000aea0: 6573 2864 662c 2072 6573 706f 6e73 652e  es(df, response.
+0000aeb0: 746f 5f64 6963 7428 2929 0a20 2020 2020  to_dict()).     
+0000aec0: 2020 2072 6574 7572 6e20 6466 0a0a 2020     return df..  
+0000aed0: 2020 2320 706c 6174 666f 726d 732f 7b70    # platforms/{p
+0000aee0: 6c61 7466 6f72 6d7d 2f65 6c65 6d65 6e74  latform}/element
+0000aef0: 732f 7b70 726f 6265 737d 202d 2d2d 2d2d  s/{probes} -----
+0000af00: 0a20 2020 2023 206e 6f74 2069 6d70 6c65  .    # not imple
+0000af10: 6d65 6e74 6564 0a20 2020 200a 2020 2020  mented.    .    
+0000af20: 2320 706c 6174 666f 726d 732f 7b70 6c61  # platforms/{pla
+0000af30: 7466 6f72 6d7d 2f65 6c65 6d65 6e74 732f  tform}/elements/
+0000af40: 7b70 726f 6265 7d2f 6765 6e65 732c 2067  {probe}/genes, g
+0000af50: 6574 5f70 6c61 7466 6f72 6d5f 656c 656d  et_platform_elem
+0000af60: 656e 745f 6765 6e65 7320 2d2d 2d2d 0a0a  ent_genes ----..
+0000af70: 2020 2020 6465 6620 6765 745f 706c 6174      def get_plat
+0000af80: 666f 726d 5f65 6c65 6d65 6e74 5f67 656e  form_element_gen
+0000af90: 6573 2873 656c 662c 2070 6c61 7466 6f72  es(self, platfor
+0000afa0: 6d3a 7374 727c 696e 742c 200a 2020 2020  m:str|int, .    
 0000afb0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000afc0: 2020 2020 2020 2020 6f66 6673 6574 3a69          offset:i
-0000afd0: 6e74 203d 2030 2c0a 2020 2020 2020 2020  nt = 0,.        
+0000afc0: 2020 2020 2020 2020 2020 2020 2020 2070                 p
+0000afd0: 726f 6265 3a73 7472 7c69 6e74 2c0a 2020  robe:str|int,.  
 0000afe0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000aff0: 2020 2020 2020 2020 2020 206c 696d 6974             limit
-0000b000: 3a69 6e74 203d 2032 302c 0a20 2020 2020  :int = 20,.     
-0000b010: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000b020: 2020 2020 2020 2020 2020 2020 2020 2a2a                **
-0000b030: 6b77 6172 6773 292d 3e44 6174 6146 7261  kwargs)->DataFra
-0000b040: 6d65 3a0a 2020 2020 2020 2020 0a20 2020  me:.        .   
-0000b050: 2020 2020 200a 2020 2020 2020 2020 2222       .        ""
-0000b060: 2252 6574 7269 6576 6520 7468 6520 6765  "Retrieve the ge
-0000b070: 6e65 7320 6173 736f 6369 6174 6564 2074  nes associated t
-0000b080: 6f20 6120 7072 6f62 6520 696e 2061 2067  o a probe in a g
-0000b090: 6976 656e 2070 6c61 7466 6f72 6d0a 2020  iven platform.  
-0000b0a0: 2020 2020 2020 0a20 2020 2020 2020 203a        .        :
-0000b0b0: 7061 7261 6d20 706c 6174 666f 726d 3a20  param platform: 
-0000b0c0: 4120 706c 6174 666f 726d 206e 756d 6572  A platform numer
-0000b0d0: 6963 616c 2069 6465 6e74 6966 6965 7220  ical identifier 
-0000b0e0: 6f72 2061 2070 6c61 7466 6f72 6d20 7368  or a platform sh
-0000b0f0: 6f72 7420 6e61 6d65 0a20 2020 2020 2020  ort name.       
-0000b100: 203a 7479 7065 2070 6c61 7466 6f72 6d3a   :type platform:
-0000b110: 2073 7472 7c69 6e74 0a20 2020 2020 2020   str|int.       
-0000b120: 203a 7061 7261 6d20 7072 6f62 653a 2041   :param probe: A
-0000b130: 2070 726f 6265 206e 616d 6520 6f72 2069   probe name or i
-0000b140: 7427 7320 6e75 6d65 7269 6361 6c20 6964  t's numerical id
-0000b150: 656e 7469 6669 6572 0a20 2020 2020 2020  entifier.       
-0000b160: 203a 7479 7065 2070 726f 6265 3a20 7374   :type probe: st
-0000b170: 727c 696e 740a 2020 2020 2020 2020 3a70  r|int.        :p
-0000b180: 6172 616d 206f 6666 7365 743a 2054 6865  aram offset: The
-0000b190: 206f 6666 7365 7420 6f66 2074 6865 2066   offset of the f
-0000b1a0: 6972 7374 2072 6574 7269 6576 6564 2072  irst retrieved r
-0000b1b0: 6573 756c 742e 2c20 6465 6661 756c 7473  esult., defaults
-0000b1c0: 2074 6f20 300a 2020 2020 2020 2020 3a74   to 0.        :t
-0000b1d0: 7970 6520 6f66 6673 6574 3a20 696e 742c  ype offset: int,
-0000b1e0: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
-0000b1f0: 2020 3a70 6172 616d 206c 696d 6974 3a20    :param limit: 
-0000b200: 4c69 6d69 7473 2074 6865 2072 6573 756c  Limits the resul
-0000b210: 7420 746f 2073 7065 6369 6669 6564 2061  t to specified a
-0000b220: 6d6f 756e 7420 6f66 206f 626a 6563 7473  mount of objects
-0000b230: 2e0a 2020 2020 2020 2020 2020 4861 7320  ..          Has 
-0000b240: 6120 6d61 7869 6d75 6d20 7661 6c75 6520  a maximum value 
-0000b250: 6f66 2031 3030 2e20 5573 6520 746f 6765  of 100. Use toge
-0000b260: 7468 6572 2077 6974 6820 6f66 6673 6574  ther with offset
-0000b270: 2061 6e64 2074 6865 200a 2020 2020 2020   and the .      
-0000b280: 2020 2020 746f 7461 6c5f 656c 656d 656e      total_elemen
-0000b290: 7473 2061 7474 7269 6275 7465 2069 6e20  ts attribute in 
-0000b2a0: 7468 6520 6f75 7470 7574 2074 6f20 636f  the output to co
-0000b2b0: 6d70 696c 6520 616c 6c20 6461 7461 2069  mpile all data i
-0000b2c0: 6620 6e65 6564 6564 2e0a 2020 2020 2020  f needed..      
-0000b2d0: 2020 2020 416c 7465 726e 6174 6976 656c      Alternativel
-0000b2e0: 7920 6765 745f 616c 6c5f 7061 6765 7320  y get_all_pages 
-0000b2f0: 6675 6e63 7469 6f6e 2063 616e 2062 6520  function can be 
-0000b300: 7573 6564 2077 6974 6820 616c 6c20 6675  used with all fu
-0000b310: 6e63 7469 6f6e 730a 2020 2020 2020 2020  nctions.        
-0000b320: 2020 696e 636c 7564 696e 6720 6f66 6673    including offs
-0000b330: 6574 2061 6e64 206c 696d 6974 2070 6172  et and limit par
-0000b340: 616d 6574 6572 732c 2064 6566 6175 6c74  ameters, default
-0000b350: 7320 746f 2032 300a 2020 2020 2020 2020  s to 20.        
-0000b360: 3a74 7970 6520 6c69 6d69 743a 2069 6e74  :type limit: int
-0000b370: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
-0000b380: 2020 203a 7061 7261 6d20 2a2a 6b77 6172     :param **kwar
-0000b390: 6773 3a20 4164 6469 7469 6f6e 616c 2061  gs: Additional a
-0000b3a0: 7267 756d 656e 7473 2074 6f20 7061 7373  rguments to pass
-0000b3b0: 2074 6f20 7261 772e 6765 745f 706c 6174   to raw.get_plat
-0000b3c0: 666f 726d 5f65 6c65 6d65 6e74 5f67 656e  form_element_gen
-0000b3d0: 6573 0a20 2020 2020 2020 203a 7479 7065  es.        :type
-0000b3e0: 202a 2a6b 7761 7267 733a 2054 5950 450a   **kwargs: TYPE.
-0000b3f0: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
-0000b400: 2041 2044 6174 6146 7261 6d65 2077 6974   A DataFrame wit
-0000b410: 6820 7468 6520 696e 666f 726d 6174 696f  h the informatio
-0000b420: 6e20 6162 6f75 7420 7175 6572 7269 6564  n about querried
-0000b430: 2067 656e 6528 7329 2e0a 2020 2020 2020   gene(s)..      
-0000b440: 2020 2020 5468 6520 6669 656c 6473 206f      The fields o
-0000b450: 6620 7468 6520 6f75 7470 7574 2044 6174  f the output Dat
-0000b460: 6146 7261 6d65 2061 7265 3a0a 2020 2020  aFrame are:.    
-0000b470: 2020 2020 2020 2020 2d20 6765 6e65 5f73          - gene_s
-0000b480: 796d 626f 6c3a 2053 796d 626f 6c20 666f  ymbol: Symbol fo
-0000b490: 7220 7468 6520 6765 6e65 0a20 2020 2020  r the gene.     
-0000b4a0: 2020 2020 2020 202d 2067 656e 655f 656e         - gene_en
-0000b4b0: 7365 6d62 6c3a 2045 6e73 656d 626c 2049  sembl: Ensembl I
-0000b4c0: 4420 666f 7220 7468 6520 6765 6e65 0a20  D for the gene. 
-0000b4d0: 2020 2020 2020 2020 2020 202d 2067 656e             - gen
-0000b4e0: 655f 4e43 4249 3a20 4e43 4249 2069 6420  e_NCBI: NCBI id 
-0000b4f0: 666f 7220 7468 6520 6765 6e65 0a20 2020  for the gene.   
-0000b500: 2020 2020 2020 2020 202d 2067 656e 655f           - gene_
-0000b510: 6e61 6d65 3a20 4e61 6d65 206f 6620 7468  name: Name of th
-0000b520: 6520 6765 6e65 0a20 2020 2020 2020 2020  e gene.         
-0000b530: 2020 202d 2067 656e 655f 616c 6961 7365     - gene_aliase
-0000b540: 733a 2047 656e 6520 616c 6961 7365 732e  s: Gene aliases.
-0000b550: 2045 6163 6820 726f 7720 696e 636c 7564   Each row includ
-0000b560: 6573 2061 2076 6563 746f 720a 2020 2020  es a vector.    
-0000b570: 2020 2020 2020 2020 2d20 6765 6e65 5f4d          - gene_M
-0000b580: 4658 5f72 616e 6b3a 204d 756c 7469 6675  FX_rank: Multifu
-0000b590: 6e63 7469 6f6e 616c 6974 7920 7261 6e6b  nctionality rank
-0000b5a0: 2066 6f72 2074 6865 2067 656e 650a 2020   for the gene.  
-0000b5b0: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
-0000b5c0: 6e5f 6e61 6d65 3a20 4e61 6d65 206f 6620  n_name: Name of 
-0000b5d0: 7468 6520 7370 6563 6965 730a 2020 2020  the species.    
-0000b5e0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
-0000b5f0: 7363 6965 6e74 6966 6963 3a20 5363 6965  scientific: Scie
-0000b600: 6e74 6966 6963 206e 616d 6520 666f 7220  ntific name for 
-0000b610: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
-0000b620: 2020 2020 2020 2d20 7461 786f 6e5f 4944        - taxon_ID
-0000b630: 3a20 496e 7465 726e 616c 2069 6465 6e74  : Internal ident
-0000b640: 6966 6965 7220 6769 7665 6e20 746f 2074  ifier given to t
-0000b650: 6865 2073 7065 6369 6573 2062 7920 4765  he species by Ge
-0000b660: 6d6d 610a 2020 2020 2020 2020 2020 2020  mma.            
-0000b670: 2d20 7461 786f 6e5f 4e43 4249 3a20 4e43  - taxon_NCBI: NC
-0000b680: 4249 2049 4420 6f66 2074 6865 2074 6178  BI ID of the tax
-0000b690: 6f6e 0a20 2020 2020 2020 2020 2020 202d  on.            -
-0000b6a0: 2074 6178 6f6e 5f64 6174 6162 6173 655f   taxon_database_
-0000b6b0: 6e61 6d65 3a20 556e 6465 726c 7969 6e67  name: Underlying
-0000b6c0: 2064 6174 6162 6173 6520 7573 6564 2069   database used i
-0000b6d0: 6e20 4765 6d6d 6120 666f 7220 7468 6520  n Gemma for the 
-0000b6e0: 7461 786f 6e0a 2020 2020 2020 2020 2020  taxon.          
-0000b6f0: 2020 2d20 7461 786f 6e5f 6461 7461 6261    - taxon_databa
-0000b700: 7365 5f49 443a 2049 4420 6f66 2074 6865  se_ID: ID of the
-0000b710: 2075 6e64 6572 6c79 696e 6720 6461 7461   underlying data
-0000b720: 6261 7365 2075 7365 6420 696e 2047 656d  base used in Gem
-0000b730: 6d61 2066 6f72 2074 6865 2074 6178 6f6e  ma for the taxon
-0000b740: 0a20 2020 2020 2020 203a 7274 7970 653a  .        :rtype:
-0000b750: 2044 6174 6146 7261 6d65 0a0a 2020 2020   DataFrame..    
-0000b760: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
-0000b770: 0a20 2020 2020 2020 206b 7761 7267 7320  .        kwargs 
-0000b780: 3d20 7673 2e72 656d 6f76 655f 6e6f 6e65  = vs.remove_none
-0000b790: 7328 6f66 6673 6574 203d 206f 6666 7365  s(offset = offse
-0000b7a0: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
-0000b7b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000b7c0: 2020 2020 6c69 6d69 7420 3d20 6c69 6d69      limit = limi
-0000b7d0: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
+0000aff0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000b000: 206f 6666 7365 743a 696e 7420 3d20 302c   offset:int = 0,
+0000b010: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+0000b020: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000b030: 2020 2020 6c69 6d69 743a 696e 7420 3d20      limit:int = 
+0000b040: 3230 2c0a 2020 2020 2020 2020 2020 2020  20,.            
+0000b050: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000b060: 2020 2020 2020 202a 2a6b 7761 7267 7329         **kwargs)
+0000b070: 2d3e 4461 7461 4672 616d 653a 0a20 2020  ->DataFrame:.   
+0000b080: 2020 2020 200a 2020 2020 2020 2020 0a20       .        . 
+0000b090: 2020 2020 2020 2022 2222 5265 7472 6965         """Retrie
+0000b0a0: 7665 2074 6865 2067 656e 6573 2061 7373  ve the genes ass
+0000b0b0: 6f63 6961 7465 6420 746f 2061 2070 726f  ociated to a pro
+0000b0c0: 6265 2069 6e20 6120 6769 7665 6e20 706c  be in a given pl
+0000b0d0: 6174 666f 726d 0a20 2020 2020 2020 200a  atform.        .
+0000b0e0: 2020 2020 2020 2020 3a70 6172 616d 2070          :param p
+0000b0f0: 6c61 7466 6f72 6d3a 2041 2070 6c61 7466  latform: A platf
+0000b100: 6f72 6d20 6e75 6d65 7269 6361 6c20 6964  orm numerical id
+0000b110: 656e 7469 6669 6572 206f 7220 6120 706c  entifier or a pl
+0000b120: 6174 666f 726d 2073 686f 7274 206e 616d  atform short nam
+0000b130: 650a 2020 2020 2020 2020 3a74 7970 6520  e.        :type 
+0000b140: 706c 6174 666f 726d 3a20 7374 727c 696e  platform: str|in
+0000b150: 740a 2020 2020 2020 2020 3a70 6172 616d  t.        :param
+0000b160: 2070 726f 6265 3a20 4120 7072 6f62 6520   probe: A probe 
+0000b170: 6e61 6d65 206f 7220 6974 2773 206e 756d  name or it's num
+0000b180: 6572 6963 616c 2069 6465 6e74 6966 6965  erical identifie
+0000b190: 720a 2020 2020 2020 2020 3a74 7970 6520  r.        :type 
+0000b1a0: 7072 6f62 653a 2073 7472 7c69 6e74 0a20  probe: str|int. 
+0000b1b0: 2020 2020 2020 203a 7061 7261 6d20 6f66         :param of
+0000b1c0: 6673 6574 3a20 5468 6520 6f66 6673 6574  fset: The offset
+0000b1d0: 206f 6620 7468 6520 6669 7273 7420 7265   of the first re
+0000b1e0: 7472 6965 7665 6420 7265 7375 6c74 2e2c  trieved result.,
+0000b1f0: 2064 6566 6175 6c74 7320 746f 2030 0a20   defaults to 0. 
+0000b200: 2020 2020 2020 203a 7479 7065 206f 6666         :type off
+0000b210: 7365 743a 2069 6e74 2c20 6f70 7469 6f6e  set: int, option
+0000b220: 616c 0a20 2020 2020 2020 203a 7061 7261  al.        :para
+0000b230: 6d20 6c69 6d69 743a 204c 696d 6974 7320  m limit: Limits 
+0000b240: 7468 6520 7265 7375 6c74 2074 6f20 7370  the result to sp
+0000b250: 6563 6966 6965 6420 616d 6f75 6e74 206f  ecified amount o
+0000b260: 6620 6f62 6a65 6374 732e 0a20 2020 2020  f objects..     
+0000b270: 2020 2020 2048 6173 2061 206d 6178 696d       Has a maxim
+0000b280: 756d 2076 616c 7565 206f 6620 3130 302e  um value of 100.
+0000b290: 2055 7365 2074 6f67 6574 6865 7220 7769   Use together wi
+0000b2a0: 7468 206f 6666 7365 7420 616e 6420 7468  th offset and th
+0000b2b0: 6520 0a20 2020 2020 2020 2020 2074 6f74  e .          tot
+0000b2c0: 616c 5f65 6c65 6d65 6e74 7320 6174 7472  al_elements attr
+0000b2d0: 6962 7574 6520 696e 2074 6865 206f 7574  ibute in the out
+0000b2e0: 7075 7420 746f 2063 6f6d 7069 6c65 2061  put to compile a
+0000b2f0: 6c6c 2064 6174 6120 6966 206e 6565 6465  ll data if neede
+0000b300: 642e 0a20 2020 2020 2020 2020 2041 6c74  d..          Alt
+0000b310: 6572 6e61 7469 7665 6c79 2067 6574 5f61  ernatively get_a
+0000b320: 6c6c 5f70 6167 6573 2066 756e 6374 696f  ll_pages functio
+0000b330: 6e20 6361 6e20 6265 2075 7365 6420 7769  n can be used wi
+0000b340: 7468 2061 6c6c 2066 756e 6374 696f 6e73  th all functions
+0000b350: 0a20 2020 2020 2020 2020 2069 6e63 6c75  .          inclu
+0000b360: 6469 6e67 206f 6666 7365 7420 616e 6420  ding offset and 
+0000b370: 6c69 6d69 7420 7061 7261 6d65 7465 7273  limit parameters
+0000b380: 2c20 6465 6661 756c 7473 2074 6f20 3230  , defaults to 20
+0000b390: 0a20 2020 2020 2020 203a 7479 7065 206c  .        :type l
+0000b3a0: 696d 6974 3a20 696e 742c 206f 7074 696f  imit: int, optio
+0000b3b0: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
+0000b3c0: 616d 202a 2a6b 7761 7267 733a 2041 6464  am **kwargs: Add
+0000b3d0: 6974 696f 6e61 6c20 6172 6775 6d65 6e74  itional argument
+0000b3e0: 7320 746f 2070 6173 7320 746f 2072 6177  s to pass to raw
+0000b3f0: 2e67 6574 5f70 6c61 7466 6f72 6d5f 656c  .get_platform_el
+0000b400: 656d 656e 745f 6765 6e65 730a 2020 2020  ement_genes.    
+0000b410: 2020 2020 3a74 7970 6520 2a2a 6b77 6172      :type **kwar
+0000b420: 6773 3a20 5459 5045 0a20 2020 2020 2020  gs: TYPE.       
+0000b430: 203a 7265 7475 726e 3a20 4120 4461 7461   :return: A Data
+0000b440: 4672 616d 6520 7769 7468 2074 6865 2069  Frame with the i
+0000b450: 6e66 6f72 6d61 7469 6f6e 2061 626f 7574  nformation about
+0000b460: 2071 7565 7272 6965 6420 6765 6e65 2873   querried gene(s
+0000b470: 292e 0a20 2020 2020 2020 2020 2054 6865  )..          The
+0000b480: 2066 6965 6c64 7320 6f66 2074 6865 206f   fields of the o
+0000b490: 7574 7075 7420 4461 7461 4672 616d 6520  utput DataFrame 
+0000b4a0: 6172 653a 0a20 2020 2020 2020 2020 2020  are:.           
+0000b4b0: 202d 2067 656e 655f 7379 6d62 6f6c 3a20   - gene_symbol: 
+0000b4c0: 5379 6d62 6f6c 2066 6f72 2074 6865 2067  Symbol for the g
+0000b4d0: 656e 650a 2020 2020 2020 2020 2020 2020  ene.            
+0000b4e0: 2d20 6765 6e65 5f65 6e73 656d 626c 3a20  - gene_ensembl: 
+0000b4f0: 456e 7365 6d62 6c20 4944 2066 6f72 2074  Ensembl ID for t
+0000b500: 6865 2067 656e 650a 2020 2020 2020 2020  he gene.        
+0000b510: 2020 2020 2d20 6765 6e65 5f4e 4342 493a      - gene_NCBI:
+0000b520: 204e 4342 4920 6964 2066 6f72 2074 6865   NCBI id for the
+0000b530: 2067 656e 650a 2020 2020 2020 2020 2020   gene.          
+0000b540: 2020 2d20 6765 6e65 5f6e 616d 653a 204e    - gene_name: N
+0000b550: 616d 6520 6f66 2074 6865 2067 656e 650a  ame of the gene.
+0000b560: 2020 2020 2020 2020 2020 2020 2d20 6765              - ge
+0000b570: 6e65 5f61 6c69 6173 6573 3a20 4765 6e65  ne_aliases: Gene
+0000b580: 2061 6c69 6173 6573 2e20 4561 6368 2072   aliases. Each r
+0000b590: 6f77 2069 6e63 6c75 6465 7320 6120 7665  ow includes a ve
+0000b5a0: 6374 6f72 0a20 2020 2020 2020 2020 2020  ctor.           
+0000b5b0: 202d 2067 656e 655f 4d46 585f 7261 6e6b   - gene_MFX_rank
+0000b5c0: 3a20 4d75 6c74 6966 756e 6374 696f 6e61  : Multifunctiona
+0000b5d0: 6c69 7479 2072 616e 6b20 666f 7220 7468  lity rank for th
+0000b5e0: 6520 6765 6e65 0a20 2020 2020 2020 2020  e gene.         
+0000b5f0: 2020 202d 2074 6178 6f6e 5f6e 616d 653a     - taxon_name:
+0000b600: 204e 616d 6520 6f66 2074 6865 2073 7065   Name of the spe
+0000b610: 6369 6573 0a20 2020 2020 2020 2020 2020  cies.           
+0000b620: 202d 2074 6178 6f6e 5f73 6369 656e 7469   - taxon_scienti
+0000b630: 6669 633a 2053 6369 656e 7469 6669 6320  fic: Scientific 
+0000b640: 6e61 6d65 2066 6f72 2074 6865 2074 6178  name for the tax
+0000b650: 6f6e 0a20 2020 2020 2020 2020 2020 202d  on.            -
+0000b660: 2074 6178 6f6e 5f49 443a 2049 6e74 6572   taxon_ID: Inter
+0000b670: 6e61 6c20 6964 656e 7469 6669 6572 2067  nal identifier g
+0000b680: 6976 656e 2074 6f20 7468 6520 7370 6563  iven to the spec
+0000b690: 6965 7320 6279 2047 656d 6d61 0a20 2020  ies by Gemma.   
+0000b6a0: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
+0000b6b0: 5f4e 4342 493a 204e 4342 4920 4944 206f  _NCBI: NCBI ID o
+0000b6c0: 6620 7468 6520 7461 786f 6e0a 2020 2020  f the taxon.    
+0000b6d0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
+0000b6e0: 6461 7461 6261 7365 5f6e 616d 653a 2055  database_name: U
+0000b6f0: 6e64 6572 6c79 696e 6720 6461 7461 6261  nderlying databa
+0000b700: 7365 2075 7365 6420 696e 2047 656d 6d61  se used in Gemma
+0000b710: 2066 6f72 2074 6865 2074 6178 6f6e 0a20   for the taxon. 
+0000b720: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
+0000b730: 6f6e 5f64 6174 6162 6173 655f 4944 3a20  on_database_ID: 
+0000b740: 4944 206f 6620 7468 6520 756e 6465 726c  ID of the underl
+0000b750: 7969 6e67 2064 6174 6162 6173 6520 7573  ying database us
+0000b760: 6564 2069 6e20 4765 6d6d 6120 666f 7220  ed in Gemma for 
+0000b770: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
+0000b780: 2020 3a72 7479 7065 3a20 4461 7461 4672    :rtype: DataFr
+0000b790: 616d 650a 0a20 2020 2020 2020 2022 2222  ame..        """
+0000b7a0: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
+0000b7b0: 2020 6b77 6172 6773 203d 2076 732e 7265    kwargs = vs.re
+0000b7c0: 6d6f 7665 5f6e 6f6e 6573 286f 6666 7365  move_nones(offse
+0000b7d0: 7420 3d20 6f66 6673 6574 2c0a 2020 2020  t = offset,.    
 0000b7e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000b7f0: 2020 2020 2a2a 6b77 6172 6773 290a 2020      **kwargs).  
-0000b800: 2020 2020 2020 0a20 2020 2020 2020 2072        .        r
-0000b810: 6573 706f 6e73 6520 3d20 2073 656c 662e  esponse =  self.
-0000b820: 7261 772e 6765 745f 706c 6174 666f 726d  raw.get_platform
-0000b830: 5f65 6c65 6d65 6e74 5f67 656e 6573 2870  _element_genes(p
-0000b840: 6c61 7466 6f72 6d2c 2070 726f 6265 2c20  latform, probe, 
-0000b850: 2a2a 6b77 6172 6773 290a 2020 2020 2020  **kwargs).      
-0000b860: 2020 6466 203d 2070 732e 7072 6f63 6573    df = ps.proces
-0000b870: 735f 6765 6e65 7328 7265 7370 6f6e 7365  s_genes(response
-0000b880: 2e64 6174 6129 0a20 2020 2020 2020 2070  .data).        p
-0000b890: 732e 6174 7461 6368 5f61 7474 7269 6275  s.attach_attribu
-0000b8a0: 7465 7328 6466 2c20 7265 7370 6f6e 7365  tes(df, response
-0000b8b0: 2e74 6f5f 6469 6374 2829 290a 2020 2020  .to_dict()).    
-0000b8c0: 2020 2020 7265 7475 726e 2064 660a 2020      return df.  
-0000b8d0: 2020 2320 706c 6174 666f 726d 732f 7b70    # platforms/{p
-0000b8e0: 6c61 7466 6f72 6d7d 2f65 6c65 6d65 6e74  latform}/element
-0000b8f0: 7320 2d2d 2d2d 0a20 2020 2023 2075 6e69  s ----.    # uni
-0000b900: 6d70 6c65 6d65 6e74 6564 0a20 2020 2023  mplemented.    #
-0000b910: 2072 6564 756e 7461 6e74 2077 6974 6820   reduntant with 
-0000b920: 616e 6e6f 7461 7469 6f6e 2066 696c 6573  annotation files
-0000b930: 0a0a 2020 2020 2320 706c 6174 666f 726d  ..    # platform
-0000b940: 7320 6765 745f 706c 6174 666f 726d 7320  s get_platforms 
-0000b950: 2d2d 2d2d 2d0a 2020 2020 2320 756e 696d  -----.    # unim
-0000b960: 706c 656d 656e 7465 6420 696e 2052 2062  plemented in R b
-0000b970: 7574 2065 6173 6965 7220 746f 206b 6565  ut easier to kee
-0000b980: 7020 7468 696e 6773 2073 6570 6172 6174  p things separat
-0000b990: 6520 6865 7265 0a20 2020 2064 6566 2067  e here.    def g
-0000b9a0: 6574 5f70 6c61 7466 6f72 6d73 2873 656c  et_platforms(sel
-0000b9b0: 662c 0a20 2020 2020 2020 2020 2020 2020  f,.             
-0000b9c0: 2020 2020 2020 2020 2066 696c 7465 723a           filter:
-0000b9d0: 7374 7220 3d20 4e6f 6e65 2c0a 2020 2020  str = None,.    
-0000b9e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000b9f0: 2020 7461 7861 3a4c 6973 745b 7374 725d    taxa:List[str]
-0000ba00: 203d 204e 6f6e 652c 0a20 2020 2020 2020   = None,.       
-0000ba10: 2020 2020 2020 2020 2020 2020 2020 206f                 o
-0000ba20: 6666 7365 743a 696e 743d 302c 0a20 2020  ffset:int=0,.   
-0000ba30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ba40: 2020 206c 696d 6974 3a69 6e74 203d 2032     limit:int = 2
-0000ba50: 302c 0a20 2020 2020 2020 2020 2020 2020  0,.             
-0000ba60: 2020 2020 2020 2020 2073 6f72 743a 7374           sort:st
-0000ba70: 723d 222b 6964 222c 0a20 2020 2020 2020  r="+id",.       
-0000ba80: 2020 2020 2020 2020 2020 2020 2020 202a                 *
-0000ba90: 2a6b 7761 7267 7329 2d3e 4461 7461 4672  *kwargs)->DataFr
-0000baa0: 616d 653a 0a20 2020 2020 2020 2022 2222  ame:.        """
-0000bab0: 0a20 2020 2020 2020 2052 6574 7269 6576  .        Retriev
-0000bac0: 6520 616c 6c20 706c 6174 666f 726d 730a  e all platforms.
-0000bad0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-0000bae0: 203a 7061 7261 6d20 6669 6c74 6572 3a20   :param filter: 
-0000baf0: 4669 6c74 6572 2072 6573 756c 7473 2062  Filter results b
-0000bb00: 7920 6d61 7463 6869 6e67 2065 7870 7265  y matching expre
-0000bb10: 7373 696f 6e2e 2055 7365 200a 2020 2020  ssion. Use .    
-0000bb20: 2020 2020 2020 6669 6c74 6572 5f70 726f        filter_pro
-0000bb30: 7065 7274 6965 7320 6675 6e63 7469 6f6e  perties function
-0000bb40: 2074 6f20 6765 7420 6120 6c69 7374 206f   to get a list o
-0000bb50: 6620 616c 6c20 6176 6169 6c61 626c 6520  f all available 
-0000bb60: 7061 7261 6d65 7465 7273 2e0a 2020 2020  parameters..    
-0000bb70: 2020 2020 2020 5468 6573 6520 7072 6f70        These prop
-0000bb80: 6572 7469 6573 2063 616e 2062 6520 636f  erties can be co
-0000bb90: 6d62 696e 6564 2075 7369 6e67 2022 616e  mbined using "an
-0000bba0: 6422 2022 6f72 2220 636c 6175 7365 7320  d" "or" clauses 
-0000bbb0: 616e 6420 6d61 7920 0a20 2020 2020 2020  and may .       
-0000bbc0: 2020 2063 6f6e 7461 696e 2063 6f6d 6d6f     contain commo
-0000bbd0: 6e20 6f70 6572 6174 6f72 7320 7375 6368  n operators such
-0000bbe0: 2061 7320 223d 222c 2022 3c22 206f 7220   as "=", "<" or 
-0000bbf0: 2269 6e22 2e20 2865 2e67 2e20 0a20 2020  "in". (e.g. .   
-0000bc00: 2020 2020 2020 2022 7461 786f 6e2e 636f         "taxon.co
-0000bc10: 6d6d 6f6e 4e61 6d65 203d 2068 756d 616e  mmonName = human
-0000bc20: 222c 2022 7461 786f 6e2e 636f 6d6d 6f6e  ", "taxon.common
-0000bc30: 4e61 6d65 2069 6e20 2868 756d 616e 2c6d  Name in (human,m
-0000bc40: 6f75 7365 292c 200a 2020 2020 2020 2020  ouse), .        
-0000bc50: 2020 2269 6420 3c20 3130 3030 2229 2c20    "id < 1000"), 
-0000bc60: 6465 6661 756c 7473 2074 6f20 4e6f 6e65  defaults to None
-0000bc70: 0a20 2020 2020 2020 203a 7479 7065 2066  .        :type f
-0000bc80: 696c 7465 723a 2073 7472 2c20 6f70 7469  ilter: str, opti
-0000bc90: 6f6e 616c 0a20 2020 2020 2020 203a 7061  onal.        :pa
-0000bca0: 7261 6d20 7461 7861 3a20 4120 6c69 7374  ram taxa: A list
-0000bcb0: 206f 6620 7461 786f 6e20 636f 6d6d 6f6e   of taxon common
-0000bcc0: 206e 616d 6573 2028 652e 672e 2068 756d   names (e.g. hum
-0000bcd0: 616e 2c20 6d6f 7573 652c 2072 6174 292e  an, mouse, rat).
-0000bce0: 200a 2020 2020 2020 2020 2020 5072 6f76   .          Prov
-0000bcf0: 6964 696e 6720 6d75 6c74 6970 6c65 2073  iding multiple s
-0000bd00: 7065 6369 6573 2077 696c 6c20 7265 7475  pecies will retu
-0000bd10: 726e 2072 6573 756c 7473 2066 6f72 2061  rn results for a
-0000bd20: 6c6c 2073 7065 6369 6573 2e20 5468 6573  ll species. Thes
-0000bd30: 6520 0a20 2020 2020 2020 2020 2061 7265  e .          are
-0000bd40: 2061 7070 656e 6465 6420 746f 2074 6865   appended to the
-0000bd50: 2066 696c 7465 7220 616e 6420 6571 7569   filter and equi
-0000bd60: 7661 6c65 6e74 2074 6f20 6669 6c74 6572  valent to filter
-0000bd70: 696e 6720 666f 7220 0a20 2020 2020 2020  ing for .       
-0000bd80: 2020 2074 6178 6f6e 2e63 6f6d 6d6f 6e4e     taxon.commonN
-0000bd90: 616d 6520 7072 6f70 6572 7479 2c20 6465  ame property, de
-0000bda0: 6661 756c 7473 2074 6f20 4e6f 6e65 0a20  faults to None. 
-0000bdb0: 2020 2020 2020 203a 7479 7065 2074 6178         :type tax
-0000bdc0: 613a 204c 6973 745b 7374 725d 2c20 6f70  a: List[str], op
-0000bdd0: 7469 6f6e 616c 0a20 2020 2020 2020 203a  tional.        :
-0000bde0: 7061 7261 6d20 6f66 6673 6574 3a20 5468  param offset: Th
-0000bdf0: 6520 6f66 6673 6574 206f 6620 7468 6520  e offset of the 
-0000be00: 6669 7273 7420 7265 7472 6965 7665 6420  first retrieved 
-0000be10: 7265 7375 6c74 2e2c 2064 6566 6175 6c74  result., default
-0000be20: 7320 746f 2030 0a20 2020 2020 2020 203a  s to 0.        :
-0000be30: 7479 7065 206f 6666 7365 743a 2069 6e74  type offset: int
-0000be40: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
-0000be50: 2020 203a 7061 7261 6d20 6c69 6d69 743a     :param limit:
-0000be60: 204c 696d 6974 7320 7468 6520 7265 7375   Limits the resu
-0000be70: 6c74 2074 6f20 7370 6563 6966 6965 6420  lt to specified 
-0000be80: 616d 6f75 6e74 206f 6620 6f62 6a65 6374  amount of object
-0000be90: 732e 0a20 2020 2020 2020 2020 2048 6173  s..          Has
-0000bea0: 2061 206d 6178 696d 756d 2076 616c 7565   a maximum value
-0000beb0: 206f 6620 3130 302e 2055 7365 2074 6f67   of 100. Use tog
-0000bec0: 6574 6865 7220 7769 7468 206f 6666 7365  ether with offse
-0000bed0: 7420 616e 6420 7468 6520 0a20 2020 2020  t and the .     
-0000bee0: 2020 2020 2074 6f74 616c 5f65 6c65 6d65       total_eleme
-0000bef0: 6e74 7320 6174 7472 6962 7574 6520 696e  nts attribute in
-0000bf00: 2074 6865 206f 7574 7075 7420 746f 2063   the output to c
-0000bf10: 6f6d 7069 6c65 2061 6c6c 2064 6174 6120  ompile all data 
-0000bf20: 6966 206e 6565 6465 642e 0a20 2020 2020  if needed..     
-0000bf30: 2020 2020 2041 6c74 6572 6e61 7469 7665       Alternative
-0000bf40: 6c79 2067 6574 5f61 6c6c 5f70 6167 6573  ly get_all_pages
-0000bf50: 2066 756e 6374 696f 6e20 6361 6e20 6265   function can be
-0000bf60: 2075 7365 6420 7769 7468 2061 6c6c 2066   used with all f
-0000bf70: 756e 6374 696f 6e73 0a20 2020 2020 2020  unctions.       
-0000bf80: 2020 2069 6e63 6c75 6469 6e67 206f 6666     including off
-0000bf90: 7365 7420 616e 6420 6c69 6d69 7420 7061  set and limit pa
-0000bfa0: 7261 6d65 7465 7273 2c20 6465 6661 756c  rameters, defaul
-0000bfb0: 7473 2074 6f20 3230 0a20 2020 2020 2020  ts to 20.       
-0000bfc0: 203a 7479 7065 206c 696d 6974 3a20 696e   :type limit: in
-0000bfd0: 742c 206f 7074 696f 6e61 6c0a 2020 2020  t, optional.    
-0000bfe0: 2020 2020 3a70 6172 616d 2073 6f72 743a      :param sort:
-0000bff0: 204f 7264 6572 2072 6573 756c 7473 2062   Order results b
-0000c000: 7920 7468 6520 6769 7665 6e20 7072 6f70  y the given prop
-0000c010: 6572 7479 2061 6e64 2064 6972 6563 7469  erty and directi
-0000c020: 6f6e 2e20 5468 6520 272b 270a 2020 2020  on. The '+'.    
-0000c030: 2020 2020 2020 7369 676e 2069 6e64 6963        sign indic
-0000c040: 6174 6520 6173 6365 6e64 696e 6720 6f72  ate ascending or
-0000c050: 6465 7220 7768 6572 6561 7320 7468 6520  der whereas the 
-0000c060: 272d 2720 696e 6469 6361 7465 2064 6573  '-' indicate des
-0000c070: 6365 6e64 696e 672c 0a20 2020 2020 2020  cending,.       
-0000c080: 2020 2064 6566 6175 6c74 7320 746f 2022     defaults to "
-0000c090: 2b69 6422 0a20 2020 2020 2020 203a 7479  +id".        :ty
-0000c0a0: 7065 2073 6f72 743a 2073 7472 2c20 6f70  pe sort: str, op
-0000c0b0: 7469 6f6e 616c 0a20 2020 2020 2020 203a  tional.        :
-0000c0c0: 7061 7261 6d20 2a2a 6b77 6172 6773 3a20  param **kwargs: 
-0000c0d0: 4164 6469 7469 6f6e 616c 2061 7267 756d  Additional argum
-0000c0e0: 656e 7473 2074 6f20 7261 772e 6765 745f  ents to raw.get_
-0000c0f0: 706c 6174 666f 726d 735f 6279 5f69 6473  platforms_by_ids
-0000c100: 0a20 2020 2020 2020 203a 7265 7475 726e  .        :return
-0000c110: 3a20 4120 4461 7461 4672 616d 6520 7769  : A DataFrame wi
-0000c120: 7468 2069 6e66 6f72 6d61 7469 6f6e 2061  th information a
-0000c130: 626f 7574 2074 6865 2070 6c61 7466 6f72  bout the platfor
-0000c140: 6d28 7329 2e0a 2020 2020 2020 2020 0a20  m(s)..        . 
-0000c150: 2020 2020 2020 2020 2054 6865 2066 6965           The fie
-0000c160: 6c64 7320 6f66 2074 6865 206f 7574 7075  lds of the outpu
-0000c170: 7420 4461 7461 4672 616d 6520 6172 653a  t DataFrame are:
-0000c180: 0a0a 2020 2020 2020 2020 2020 2020 2d20  ..            - 
-0000c190: 706c 6174 666f 726d 5f49 443a 2049 6e74  platform_ID: Int
-0000c1a0: 6572 6e61 6c20 6964 656e 7469 6669 6572  ernal identifier
-0000c1b0: 206f 6620 7468 6520 706c 6174 666f 726d   of the platform
-0000c1c0: 0a20 2020 2020 2020 2020 2020 202d 2070  .            - p
-0000c1d0: 6c61 7466 6f72 6d5f 7368 6f72 745f 6e61  latform_short_na
-0000c1e0: 6d65 3a20 5368 6f72 746e 616d 6520 6f66  me: Shortname of
-0000c1f0: 2074 6865 2070 6c61 7466 6f72 6d2e 0a20   the platform.. 
-0000c200: 2020 2020 2020 2020 2020 202d 2070 6c61             - pla
-0000c210: 7466 6f72 6d5f 6e61 6d65 3a20 4675 6c6c  tform_name: Full
-0000c220: 206e 616d 6520 6f66 2074 6865 2070 6c61   name of the pla
-0000c230: 7466 6f72 6d2e 0a20 2020 2020 2020 2020  tform..         
-0000c240: 2020 202d 2070 6c61 7466 6f72 6d5f 6465     - platform_de
-0000c250: 7363 7269 7074 696f 6e3a 2046 7265 6520  scription: Free 
-0000c260: 7465 7874 2064 6573 6372 6970 7469 6f6e  text description
-0000c270: 206f 6620 7468 6520 706c 6174 666f 726d   of the platform
-0000c280: 0a20 2020 2020 2020 2020 2020 202d 2070  .            - p
-0000c290: 6c61 7466 6f72 6d5f 7472 6f75 626c 6564  latform_troubled
-0000c2a0: 3a20 5768 6574 6865 7220 6f72 206e 6f74  : Whether or not
-0000c2b0: 2074 6865 2070 6c61 7466 6f72 6d20 7761   the platform wa
-0000c2c0: 7320 6d61 726b 6564 2022 7472 6f75 626c  s marked "troubl
-0000c2d0: 6564 2220 6279 2061 2047 656d 6d61 2070  ed" by a Gemma p
-0000c2e0: 726f 6365 7373 206f 7220 6120 6375 7261  rocess or a cura
-0000c2f0: 746f 720a 2020 2020 2020 2020 2020 2020  tor.            
-0000c300: 2d20 706c 6174 666f 726d 5f65 7870 6572  - platform_exper
-0000c310: 696d 656e 745f 636f 756e 743a 204e 756d  iment_count: Num
-0000c320: 6265 7220 6f66 2065 7870 6572 696d 656e  ber of experimen
-0000c330: 7473 2075 7369 6e67 2074 6865 2070 6c61  ts using the pla
-0000c340: 7466 6f72 6d20 7769 7468 696e 2047 656d  tform within Gem
-0000c350: 6d61 0a20 2020 2020 2020 2020 2020 202d  ma.            -
-0000c360: 2070 6c61 7466 6f72 6d5f 7479 7065 3a20   platform_type: 
-0000c370: 5465 6368 6e6f 6c6f 6779 2074 7970 6520  Technology type 
-0000c380: 666f 7220 7468 6520 706c 6174 666f 726d  for the platform
-0000c390: 2e0a 2020 2020 2020 2020 2020 2020 2d20  ..            - 
-0000c3a0: 7461 786f 6e5f 6e61 6d65 3a20 4e61 6d65  taxon_name: Name
-0000c3b0: 206f 6620 7468 6520 7370 6563 6965 7320   of the species 
-0000c3c0: 706c 6174 666f 726d 2077 6173 206d 6164  platform was mad
-0000c3d0: 6520 666f 720a 2020 2020 2020 2020 2020  e for.          
-0000c3e0: 2020 2d20 7461 786f 6e5f 7363 6965 6e74    - taxon_scient
-0000c3f0: 6966 6963 3a20 5363 6965 6e74 6966 6963  ific: Scientific
-0000c400: 206e 616d 6520 666f 7220 7468 6520 7461   name for the ta
-0000c410: 786f 6e0a 2020 2020 2020 2020 2020 2020  xon.            
-0000c420: 2d20 7461 786f 6e5f 4944 3a20 496e 7465  - taxon_ID: Inte
-0000c430: 726e 616c 2069 6465 6e74 6966 6965 7220  rnal identifier 
-0000c440: 6769 7665 6e20 746f 2074 6865 2073 7065  given to the spe
-0000c450: 6369 6573 2062 7920 4765 6d6d 610a 2020  cies by Gemma.  
-0000c460: 2020 2020 2020 2020 2020 2d20 7461 786f            - taxo
-0000c470: 6e5f 4e43 4249 3a20 4e43 4249 2049 4420  n_NCBI: NCBI ID 
-0000c480: 6f66 2074 6865 2074 6178 6f6e 0a20 2020  of the taxon.   
-0000c490: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
-0000c4a0: 5f64 6174 6162 6173 655f 6e61 6d65 3a20  _database_name: 
-0000c4b0: 556e 6465 726c 7969 6e67 2064 6174 6162  Underlying datab
-0000c4c0: 6173 6520 7573 6564 2069 6e20 4765 6d6d  ase used in Gemm
-0000c4d0: 6120 666f 7220 7468 6520 7461 786f 6e0a  a for the taxon.
-0000c4e0: 2020 2020 2020 2020 2020 2020 2d20 7461              - ta
-0000c4f0: 786f 6e5f 6461 7461 6261 7365 5f49 443a  xon_database_ID:
-0000c500: 2049 4420 6f66 2074 6865 2075 6e64 6572   ID of the under
-0000c510: 796c 696e 6720 6461 7461 6261 7365 2075  yling database u
-0000c520: 7365 6420 696e 2047 656d 6d61 2066 6f72  sed in Gemma for
-0000c530: 2074 6865 2074 6178 6f6e 0a20 2020 2020   the taxon.     
-0000c540: 2020 203a 7274 7970 653a 2044 6174 6146     :rtype: DataF
-0000c550: 7261 6d65 0a0a 2020 2020 2020 2020 2222  rame..        ""
-0000c560: 220a 2020 2020 2020 2020 0a20 2020 2020  ".        .     
-0000c570: 2020 2066 696c 7465 7220 3d20 7673 2e61     filter = vs.a
-0000c580: 6464 5f74 6f5f 6669 6c74 6572 2866 696c  dd_to_filter(fil
-0000c590: 7465 722c 2274 6178 6f6e 2e63 6f6d 6d6f  ter,"taxon.commo
-0000c5a0: 6e4e 616d 6522 2c74 6178 6129 0a20 2020  nName",taxa).   
-0000c5b0: 2020 2020 200a 2020 2020 2020 2020 6b77       .        kw
-0000c5c0: 6172 6773 203d 2076 732e 7265 6d6f 7665  args = vs.remove
-0000c5d0: 5f6e 6f6e 6573 2866 696c 7465 7220 3d20  _nones(filter = 
-0000c5e0: 6669 6c74 6572 2c0a 2020 2020 2020 2020  filter,.        
-0000c5f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c600: 2020 2020 2020 2020 206f 6666 7365 7420           offset 
-0000c610: 3d20 6f66 6673 6574 2c0a 2020 2020 2020  = offset,.      
-0000c620: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c630: 2020 2020 2020 2020 2020 206c 696d 6974             limit
-0000c640: 203d 206c 696d 6974 2c0a 2020 2020 2020   = limit,.      
-0000c650: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c660: 2020 2020 2020 2020 2020 2073 6f72 7420             sort 
-0000c670: 3d20 736f 7274 2c0a 2020 2020 2020 2020  = sort,.        
-0000c680: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c690: 2020 2020 2020 2020 202a 2a6b 7761 7267           **kwarg
-0000c6a0: 7329 0a20 2020 2020 2020 200a 2020 2020  s).        .    
-0000c6b0: 2020 2020 7265 7370 6f6e 7365 203d 2020      response =  
-0000c6c0: 7365 6c66 2e72 6177 2e67 6574 5f70 6c61  self.raw.get_pla
-0000c6d0: 7466 6f72 6d73 282a 2a6b 7761 7267 7329  tforms(**kwargs)
-0000c6e0: 0a20 2020 2020 2020 2064 6620 3d20 7073  .        df = ps
-0000c6f0: 2e70 726f 6365 7373 5f70 6c61 7466 6f72  .process_platfor
-0000c700: 6d73 2872 6573 706f 6e73 652e 6461 7461  ms(response.data
-0000c710: 290a 2020 2020 2020 2020 7073 2e61 7474  ).        ps.att
-0000c720: 6163 685f 6174 7472 6962 7574 6573 2864  ach_attributes(d
-0000c730: 662c 2072 6573 706f 6e73 652e 746f 5f64  f, response.to_d
-0000c740: 6963 7428 2929 0a20 2020 2020 2020 2072  ict()).        r
-0000c750: 6574 7572 6e20 6466 0a0a 2020 2020 2320  eturn df..    # 
-0000c760: 706c 6174 666f 726d 732f 7b70 6c61 7466  platforms/{platf
-0000c770: 6f72 6d7d 2c20 6765 745f 706c 6174 666f  orm}, get_platfo
-0000c780: 726d 735f 6279 5f69 6473 202d 2d2d 2d20  rms_by_ids ---- 
-0000c790: 0a20 2020 2064 6566 2067 6574 5f70 6c61  .    def get_pla
-0000c7a0: 7466 6f72 6d73 5f62 795f 6964 7328 7365  tforms_by_ids(se
-0000c7b0: 6c66 2c20 706c 6174 666f 726d 733a 4c69  lf, platforms:Li
-0000c7c0: 7374 5b73 7472 7c69 6e74 5d2c 200a 2020  st[str|int], .  
-0000c7d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c7e0: 2020 2020 2020 2020 2020 2066 696c 7465             filte
-0000c7f0: 723a 7374 7220 3d20 4e6f 6e65 2c0a 2020  r:str = None,.  
-0000c800: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c810: 2020 2020 2020 2020 2020 2074 6178 613a             taxa:
-0000c820: 4c69 7374 5b73 7472 5d20 3d20 4e6f 6e65  List[str] = None
-0000c830: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
-0000c840: 2020 2020 2020 2020 2020 2020 2020 206f                 o
-0000c850: 6666 7365 743a 696e 743d 302c 0a20 2020  ffset:int=0,.   
-0000c860: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c870: 2020 2020 2020 2020 2020 6c69 6d69 743a            limit:
-0000c880: 696e 7420 3d20 3230 2c0a 2020 2020 2020  int = 20,.      
-0000c890: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c8a0: 2020 2020 2020 2073 6f72 743a 7374 723d         sort:str=
-0000c8b0: 222b 6964 222c 0a20 2020 2020 2020 2020  "+id",.         
-0000c8c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000c8d0: 2020 2020 2a2a 6b77 6172 6773 292d 3e44      **kwargs)->D
-0000c8e0: 6174 6146 7261 6d65 3a0a 2020 200a 2020  ataFrame:.   .  
-0000c8f0: 2020 2020 2020 0a20 2020 2020 2020 200a        .        .
-0000c900: 2020 2020 2020 2020 2222 2252 6574 7269          """Retri
-0000c910: 6576 6520 706c 6174 666f 726d 7320 6279  eve platforms by
-0000c920: 2074 6865 6972 2069 6465 6e74 6966 6965   their identifie
-0000c930: 7273 0a0a 2020 2020 2020 2020 0a20 2020  rs..        .   
-0000c940: 2020 2020 203a 7061 7261 6d20 706c 6174       :param plat
-0000c950: 666f 726d 733a 2050 6c61 7466 6f72 6d20  forms: Platform 
-0000c960: 6e75 6d65 7269 6361 6c20 6964 656e 7469  numerical identi
-0000c970: 6669 6572 7320 6f72 2070 6c61 7466 6f72  fiers or platfor
-0000c980: 6d20 7368 6f72 7420 6e61 6d65 732e 0a20  m short names.. 
-0000c990: 2020 2020 2020 203a 7479 7065 2070 6c61         :type pla
-0000c9a0: 7466 6f72 6d73 3a20 4c69 7374 5b73 7472  tforms: List[str
-0000c9b0: 7c69 6e74 5d0a 2020 2020 2020 2020 3a70  |int].        :p
-0000c9c0: 6172 616d 2066 696c 7465 723a 2046 696c  aram filter: Fil
-0000c9d0: 7465 7220 7265 7375 6c74 7320 6279 206d  ter results by m
-0000c9e0: 6174 6368 696e 6720 6578 7072 6573 7369  atching expressi
-0000c9f0: 6f6e 2e20 5573 6520 0a20 2020 2020 2020  on. Use .       
-0000ca00: 2020 2066 696c 7465 725f 7072 6f70 6572     filter_proper
-0000ca10: 7469 6573 2066 756e 6374 696f 6e20 746f  ties function to
-0000ca20: 2067 6574 2061 206c 6973 7420 6f66 2061   get a list of a
-0000ca30: 6c6c 2061 7661 696c 6162 6c65 2070 6172  ll available par
-0000ca40: 616d 6574 6572 732e 0a20 2020 2020 2020  ameters..       
-0000ca50: 2020 2054 6865 7365 2070 726f 7065 7274     These propert
-0000ca60: 6965 7320 6361 6e20 6265 2063 6f6d 6269  ies can be combi
-0000ca70: 6e65 6420 7573 696e 6720 2261 6e64 2220  ned using "and" 
-0000ca80: 226f 7222 2063 6c61 7573 6573 2061 6e64  "or" clauses and
-0000ca90: 206d 6179 200a 2020 2020 2020 2020 2020   may .          
-0000caa0: 636f 6e74 6169 6e20 636f 6d6d 6f6e 206f  contain common o
-0000cab0: 7065 7261 746f 7273 2073 7563 6820 6173  perators such as
-0000cac0: 2022 3d22 2c20 223c 2220 6f72 2022 696e   "=", "<" or "in
-0000cad0: 222e 2028 652e 672e 200a 2020 2020 2020  ". (e.g. .      
-0000cae0: 2020 2020 2274 6178 6f6e 2e63 6f6d 6d6f      "taxon.commo
-0000caf0: 6e4e 616d 6520 3d20 6875 6d61 6e22 2c20  nName = human", 
-0000cb00: 2274 6178 6f6e 2e63 6f6d 6d6f 6e4e 616d  "taxon.commonNam
-0000cb10: 6520 696e 2028 6875 6d61 6e2c 6d6f 7573  e in (human,mous
-0000cb20: 6529 2c20 0a20 2020 2020 2020 2020 2022  e), .          "
-0000cb30: 6964 203c 2031 3030 3022 292c 2064 6566  id < 1000"), def
-0000cb40: 6175 6c74 7320 746f 204e 6f6e 650a 2020  aults to None.  
-0000cb50: 2020 2020 2020 3a74 7970 6520 6669 6c74        :type filt
-0000cb60: 6572 3a20 7374 722c 206f 7074 696f 6e61  er: str, optiona
-0000cb70: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
-0000cb80: 2074 6178 613a 2041 206c 6973 7420 6f66   taxa: A list of
-0000cb90: 2074 6178 6f6e 2063 6f6d 6d6f 6e20 6e61   taxon common na
-0000cba0: 6d65 7320 2865 2e67 2e20 6875 6d61 6e2c  mes (e.g. human,
-0000cbb0: 206d 6f75 7365 2c20 7261 7429 2e20 0a20   mouse, rat). . 
-0000cbc0: 2020 2020 2020 2020 2050 726f 7669 6469           Providi
-0000cbd0: 6e67 206d 756c 7469 706c 6520 7370 6563  ng multiple spec
-0000cbe0: 6965 7320 7769 6c6c 2072 6574 7572 6e20  ies will return 
-0000cbf0: 7265 7375 6c74 7320 666f 7220 616c 6c20  results for all 
-0000cc00: 7370 6563 6965 732e 2054 6865 7365 200a  species. These .
-0000cc10: 2020 2020 2020 2020 2020 6172 6520 6170            are ap
-0000cc20: 7065 6e64 6564 2074 6f20 7468 6520 6669  pended to the fi
-0000cc30: 6c74 6572 2061 6e64 2065 7175 6976 616c  lter and equival
-0000cc40: 656e 7420 746f 2066 696c 7465 7269 6e67  ent to filtering
-0000cc50: 2066 6f72 200a 2020 2020 2020 2020 2020   for .          
-0000cc60: 7461 786f 6e2e 636f 6d6d 6f6e 4e61 6d65  taxon.commonName
-0000cc70: 2070 726f 7065 7274 792c 2064 6566 6175   property, defau
-0000cc80: 6c74 7320 746f 204e 6f6e 650a 2020 2020  lts to None.    
-0000cc90: 2020 2020 3a74 7970 6520 7461 7861 3a20      :type taxa: 
-0000cca0: 4c69 7374 5b73 7472 5d2c 206f 7074 696f  List[str], optio
-0000ccb0: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
-0000ccc0: 616d 206f 6666 7365 743a 2054 6865 206f  am offset: The o
-0000ccd0: 6666 7365 7420 6f66 2074 6865 2066 6972  ffset of the fir
-0000cce0: 7374 2072 6574 7269 6576 6564 2072 6573  st retrieved res
-0000ccf0: 756c 742e 2c20 6465 6661 756c 7473 2074  ult., defaults t
-0000cd00: 6f20 300a 2020 2020 2020 2020 3a74 7970  o 0.        :typ
-0000cd10: 6520 6f66 6673 6574 3a20 696e 742c 206f  e offset: int, o
-0000cd20: 7074 696f 6e61 6c0a 2020 2020 2020 2020  ptional.        
-0000cd30: 3a70 6172 616d 206c 696d 6974 3a20 4c69  :param limit: Li
-0000cd40: 6d69 7473 2074 6865 2072 6573 756c 7420  mits the result 
-0000cd50: 746f 2073 7065 6369 6669 6564 2061 6d6f  to specified amo
-0000cd60: 756e 7420 6f66 206f 626a 6563 7473 2e0a  unt of objects..
-0000cd70: 2020 2020 2020 2020 2020 4861 7320 6120            Has a 
-0000cd80: 6d61 7869 6d75 6d20 7661 6c75 6520 6f66  maximum value of
-0000cd90: 2031 3030 2e20 5573 6520 746f 6765 7468   100. Use togeth
-0000cda0: 6572 2077 6974 6820 6f66 6673 6574 2061  er with offset a
-0000cdb0: 6e64 2074 6865 200a 2020 2020 2020 2020  nd the .        
-0000cdc0: 2020 746f 7461 6c5f 656c 656d 656e 7473    total_elements
-0000cdd0: 2061 7474 7269 6275 7465 2069 6e20 7468   attribute in th
-0000cde0: 6520 6f75 7470 7574 2074 6f20 636f 6d70  e output to comp
-0000cdf0: 696c 6520 616c 6c20 6461 7461 2069 6620  ile all data if 
-0000ce00: 6e65 6564 6564 2e0a 2020 2020 2020 2020  needed..        
-0000ce10: 2020 416c 7465 726e 6174 6976 656c 7920    Alternatively 
-0000ce20: 6765 745f 616c 6c5f 7061 6765 7320 6675  get_all_pages fu
-0000ce30: 6e63 7469 6f6e 2063 616e 2062 6520 7573  nction can be us
-0000ce40: 6564 2077 6974 6820 616c 6c20 6675 6e63  ed with all func
-0000ce50: 7469 6f6e 730a 2020 2020 2020 2020 2020  tions.          
-0000ce60: 696e 636c 7564 696e 6720 6f66 6673 6574  including offset
-0000ce70: 2061 6e64 206c 696d 6974 2070 6172 616d   and limit param
-0000ce80: 6574 6572 732c 2064 6566 6175 6c74 7320  eters, defaults 
-0000ce90: 746f 2032 300a 2020 2020 2020 2020 3a74  to 20.        :t
-0000cea0: 7970 6520 6c69 6d69 743a 2069 6e74 2c20  ype limit: int, 
-0000ceb0: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
-0000cec0: 203a 7061 7261 6d20 736f 7274 3a20 4f72   :param sort: Or
-0000ced0: 6465 7220 7265 7375 6c74 7320 6279 2074  der results by t
-0000cee0: 6865 2067 6976 656e 2070 726f 7065 7274  he given propert
-0000cef0: 7920 616e 6420 6469 7265 6374 696f 6e2e  y and direction.
-0000cf00: 2054 6865 2027 2b27 0a20 2020 2020 2020   The '+'.       
-0000cf10: 2020 2073 6967 6e20 696e 6469 6361 7465     sign indicate
-0000cf20: 2061 7363 656e 6469 6e67 206f 7264 6572   ascending order
-0000cf30: 2077 6865 7265 6173 2074 6865 2027 2d27   whereas the '-'
-0000cf40: 2069 6e64 6963 6174 6520 6465 7363 656e   indicate descen
-0000cf50: 6469 6e67 2c0a 2020 2020 2020 2020 2020  ding,.          
-0000cf60: 6465 6661 756c 7473 2074 6f20 222b 6964  defaults to "+id
-0000cf70: 220a 2020 2020 2020 2020 3a74 7970 6520  ".        :type 
-0000cf80: 736f 7274 3a20 7374 722c 206f 7074 696f  sort: str, optio
-0000cf90: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
-0000cfa0: 616d 202a 2a6b 7761 7267 733a 2041 6464  am **kwargs: Add
-0000cfb0: 6974 696f 6e61 6c20 6172 6775 6d65 6e74  itional argument
-0000cfc0: 7320 746f 2072 6177 2e67 6574 5f70 6c61  s to raw.get_pla
-0000cfd0: 7466 6f72 6d73 5f62 795f 6964 730a 2020  tforms_by_ids.  
-0000cfe0: 2020 2020 2020 3a72 6574 7572 6e3a 2041        :return: A
-0000cff0: 2044 6174 6146 7261 6d65 2077 6974 6820   DataFrame with 
-0000d000: 696e 666f 726d 6174 696f 6e20 6162 6f75  information abou
-0000d010: 7420 7468 6520 706c 6174 666f 726d 2873  t the platform(s
-0000d020: 292e 0a20 2020 2020 2020 200a 2020 2020  )..        .    
-0000d030: 2020 2020 2020 5468 6520 6669 656c 6473        The fields
-0000d040: 206f 6620 7468 6520 6f75 7470 7574 2044   of the output D
-0000d050: 6174 6146 7261 6d65 2061 7265 3a0a 0a20  ataFrame are:.. 
-0000d060: 2020 2020 2020 2020 2020 202d 2070 6c61             - pla
-0000d070: 7466 6f72 6d5f 4944 3a20 496e 7465 726e  tform_ID: Intern
-0000d080: 616c 2069 6465 6e74 6966 6965 7220 6f66  al identifier of
-0000d090: 2074 6865 2070 6c61 7466 6f72 6d0a 2020   the platform.  
-0000d0a0: 2020 2020 2020 2020 2020 2d20 706c 6174            - plat
-0000d0b0: 666f 726d 5f73 686f 7274 5f6e 616d 653a  form_short_name:
-0000d0c0: 2053 686f 7274 6e61 6d65 206f 6620 7468   Shortname of th
-0000d0d0: 6520 706c 6174 666f 726d 2e0a 2020 2020  e platform..    
-0000d0e0: 2020 2020 2020 2020 2d20 706c 6174 666f          - platfo
-0000d0f0: 726d 5f6e 616d 653a 2046 756c 6c20 6e61  rm_name: Full na
-0000d100: 6d65 206f 6620 7468 6520 706c 6174 666f  me of the platfo
-0000d110: 726d 2e0a 2020 2020 2020 2020 2020 2020  rm..            
-0000d120: 2d20 706c 6174 666f 726d 5f64 6573 6372  - platform_descr
-0000d130: 6970 7469 6f6e 3a20 4672 6565 2074 6578  iption: Free tex
-0000d140: 7420 6465 7363 7269 7074 696f 6e20 6f66  t description of
-0000d150: 2074 6865 2070 6c61 7466 6f72 6d0a 2020   the platform.  
-0000d160: 2020 2020 2020 2020 2020 2d20 706c 6174            - plat
-0000d170: 666f 726d 5f74 726f 7562 6c65 643a 2057  form_troubled: W
-0000d180: 6865 7468 6572 206f 7220 6e6f 7420 7468  hether or not th
-0000d190: 6520 706c 6174 666f 726d 2077 6173 206d  e platform was m
-0000d1a0: 6172 6b65 6420 2274 726f 7562 6c65 6422  arked "troubled"
-0000d1b0: 2062 7920 6120 4765 6d6d 6120 7072 6f63   by a Gemma proc
-0000d1c0: 6573 7320 6f72 2061 2063 7572 6174 6f72  ess or a curator
-0000d1d0: 0a20 2020 2020 2020 2020 2020 202d 2070  .            - p
-0000d1e0: 6c61 7466 6f72 6d5f 6578 7065 7269 6d65  latform_experime
-0000d1f0: 6e74 5f63 6f75 6e74 3a20 4e75 6d62 6572  nt_count: Number
-0000d200: 206f 6620 6578 7065 7269 6d65 6e74 7320   of experiments 
-0000d210: 7573 696e 6720 7468 6520 706c 6174 666f  using the platfo
-0000d220: 726d 2077 6974 6869 6e20 4765 6d6d 610a  rm within Gemma.
-0000d230: 2020 2020 2020 2020 2020 2020 2d20 706c              - pl
-0000d240: 6174 666f 726d 5f74 7970 653a 2054 6563  atform_type: Tec
-0000d250: 686e 6f6c 6f67 7920 7479 7065 2066 6f72  hnology type for
-0000d260: 2074 6865 2070 6c61 7466 6f72 6d2e 0a20   the platform.. 
-0000d270: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
-0000d280: 6f6e 5f6e 616d 653a 204e 616d 6520 6f66  on_name: Name of
-0000d290: 2074 6865 2073 7065 6369 6573 2070 6c61   the species pla
-0000d2a0: 7466 6f72 6d20 7761 7320 6d61 6465 2066  tform was made f
-0000d2b0: 6f72 0a20 2020 2020 2020 2020 2020 202d  or.            -
-0000d2c0: 2074 6178 6f6e 5f73 6369 656e 7469 6669   taxon_scientifi
-0000d2d0: 633a 2053 6369 656e 7469 6669 6320 6e61  c: Scientific na
-0000d2e0: 6d65 2066 6f72 2074 6865 2074 6178 6f6e  me for the taxon
-0000d2f0: 0a20 2020 2020 2020 2020 2020 202d 2074  .            - t
-0000d300: 6178 6f6e 5f49 443a 2049 6e74 6572 6e61  axon_ID: Interna
-0000d310: 6c20 6964 656e 7469 6669 6572 2067 6976  l identifier giv
-0000d320: 656e 2074 6f20 7468 6520 7370 6563 6965  en to the specie
-0000d330: 7320 6279 2047 656d 6d61 0a20 2020 2020  s by Gemma.     
-0000d340: 2020 2020 2020 202d 2074 6178 6f6e 5f4e         - taxon_N
-0000d350: 4342 493a 204e 4342 4920 4944 206f 6620  CBI: NCBI ID of 
-0000d360: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
-0000d370: 2020 2020 2020 2d20 7461 786f 6e5f 6461        - taxon_da
-0000d380: 7461 6261 7365 5f6e 616d 653a 2055 6e64  tabase_name: Und
-0000d390: 6572 6c79 696e 6720 6461 7461 6261 7365  erlying database
-0000d3a0: 2075 7365 6420 696e 2047 656d 6d61 2066   used in Gemma f
-0000d3b0: 6f72 2074 6865 2074 6178 6f6e 0a20 2020  or the taxon.   
-0000d3c0: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
-0000d3d0: 5f64 6174 6162 6173 655f 4944 3a20 4944  _database_ID: ID
-0000d3e0: 206f 6620 7468 6520 756e 6465 7279 6c69   of the underyli
-0000d3f0: 6e67 2064 6174 6162 6173 6520 7573 6564  ng database used
-0000d400: 2069 6e20 4765 6d6d 6120 666f 7220 7468   in Gemma for th
-0000d410: 6520 7461 786f 6e0a 2020 2020 2020 2020  e taxon.        
-0000d420: 3a72 7479 7065 3a20 4461 7461 4672 616d  :rtype: DataFram
-0000d430: 650a 0a20 2020 2020 2020 2022 2222 0a20  e..        """. 
-0000d440: 2020 2020 2020 2066 696c 7465 7220 3d20         filter = 
-0000d450: 7673 2e61 6464 5f74 6f5f 6669 6c74 6572  vs.add_to_filter
-0000d460: 2866 696c 7465 722c 2274 6178 6f6e 2e63  (filter,"taxon.c
-0000d470: 6f6d 6d6f 6e4e 616d 6522 2c74 6178 6129  ommonName",taxa)
-0000d480: 0a20 2020 2020 2020 206b 7761 7267 7320  .        kwargs 
-0000d490: 3d20 7673 2e72 656d 6f76 655f 6e6f 6e65  = vs.remove_none
-0000d4a0: 7328 6669 6c74 6572 203d 2066 696c 7465  s(filter = filte
-0000d4b0: 722c 0a20 2020 2020 2020 2020 2020 2020  r,.             
-0000d4c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000d4d0: 2020 2020 6f66 6673 6574 203d 206f 6666      offset = off
-0000d4e0: 7365 742c 0a20 2020 2020 2020 2020 2020  set,.           
+0000b7f0: 2020 2020 2020 2020 2020 2020 206c 696d               lim
+0000b800: 6974 203d 206c 696d 6974 2c0a 2020 2020  it = limit,.    
+0000b810: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000b820: 2020 2020 2020 2020 2020 2020 202a 2a6b               **k
+0000b830: 7761 7267 7329 0a20 2020 2020 2020 200a  wargs).        .
+0000b840: 2020 2020 2020 2020 7265 7370 6f6e 7365          response
+0000b850: 203d 2020 7365 6c66 2e72 6177 2e67 6574   =  self.raw.get
+0000b860: 5f70 6c61 7466 6f72 6d5f 656c 656d 656e  _platform_elemen
+0000b870: 745f 6765 6e65 7328 706c 6174 666f 726d  t_genes(platform
+0000b880: 2c20 7072 6f62 652c 202a 2a6b 7761 7267  , probe, **kwarg
+0000b890: 7329 0a20 2020 2020 2020 2064 6620 3d20  s).        df = 
+0000b8a0: 7073 2e70 726f 6365 7373 5f67 656e 6573  ps.process_genes
+0000b8b0: 2872 6573 706f 6e73 652e 6461 7461 290a  (response.data).
+0000b8c0: 2020 2020 2020 2020 7073 2e61 7474 6163          ps.attac
+0000b8d0: 685f 6174 7472 6962 7574 6573 2864 662c  h_attributes(df,
+0000b8e0: 2072 6573 706f 6e73 652e 746f 5f64 6963   response.to_dic
+0000b8f0: 7428 2929 0a20 2020 2020 2020 2072 6574  t()).        ret
+0000b900: 7572 6e20 6466 0a20 2020 2023 2070 6c61  urn df.    # pla
+0000b910: 7466 6f72 6d73 2f7b 706c 6174 666f 726d  tforms/{platform
+0000b920: 7d2f 656c 656d 656e 7473 202d 2d2d 2d0a  }/elements ----.
+0000b930: 2020 2020 2320 756e 696d 706c 656d 656e      # unimplemen
+0000b940: 7465 640a 2020 2020 2320 7265 6475 6e74  ted.    # redunt
+0000b950: 616e 7420 7769 7468 2061 6e6e 6f74 6174  ant with annotat
+0000b960: 696f 6e20 6669 6c65 730a 0a20 2020 2023  ion files..    #
+0000b970: 2070 6c61 7466 6f72 6d73 2067 6574 5f70   platforms get_p
+0000b980: 6c61 7466 6f72 6d73 202d 2d2d 2d2d 0a20  latforms -----. 
+0000b990: 2020 2023 2075 6e69 6d70 6c65 6d65 6e74     # unimplement
+0000b9a0: 6564 2069 6e20 5220 6275 7420 6561 7369  ed in R but easi
+0000b9b0: 6572 2074 6f20 6b65 6570 2074 6869 6e67  er to keep thing
+0000b9c0: 7320 7365 7061 7261 7465 2068 6572 650a  s separate here.
+0000b9d0: 2020 2020 6465 6620 6765 745f 706c 6174      def get_plat
+0000b9e0: 666f 726d 7328 7365 6c66 2c0a 2020 2020  forms(self,.    
+0000b9f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000ba00: 2020 6669 6c74 6572 3a73 7472 203d 204e    filter:str = N
+0000ba10: 6f6e 652c 0a20 2020 2020 2020 2020 2020  one,.           
+0000ba20: 2020 2020 2020 2020 2020 2074 6178 613a             taxa:
+0000ba30: 4c69 7374 5b73 7472 5d20 3d20 4e6f 6e65  List[str] = None
+0000ba40: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+0000ba50: 2020 2020 2020 2020 6f66 6673 6574 3a69          offset:i
+0000ba60: 6e74 3d30 2c0a 2020 2020 2020 2020 2020  nt=0,.          
+0000ba70: 2020 2020 2020 2020 2020 2020 6c69 6d69              limi
+0000ba80: 743a 696e 7420 3d20 3230 2c0a 2020 2020  t:int = 20,.    
+0000ba90: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000baa0: 2020 736f 7274 3a73 7472 3d22 2b69 6422    sort:str="+id"
+0000bab0: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+0000bac0: 2020 2020 2020 2020 2a2a 6b77 6172 6773          **kwargs
+0000bad0: 292d 3e44 6174 6146 7261 6d65 3a0a 2020  )->DataFrame:.  
+0000bae0: 2020 2020 2020 2222 220a 2020 2020 2020        """.      
+0000baf0: 2020 5265 7472 6965 7665 2061 6c6c 2070    Retrieve all p
+0000bb00: 6c61 7466 6f72 6d73 0a20 2020 2020 2020  latforms.       
+0000bb10: 200a 2020 2020 2020 2020 3a70 6172 616d   .        :param
+0000bb20: 2066 696c 7465 723a 2046 696c 7465 7220   filter: Filter 
+0000bb30: 7265 7375 6c74 7320 6279 206d 6174 6368  results by match
+0000bb40: 696e 6720 6578 7072 6573 7369 6f6e 2e20  ing expression. 
+0000bb50: 5573 6520 0a20 2020 2020 2020 2020 2066  Use .          f
+0000bb60: 696c 7465 725f 7072 6f70 6572 7469 6573  ilter_properties
+0000bb70: 2066 756e 6374 696f 6e20 746f 2067 6574   function to get
+0000bb80: 2061 206c 6973 7420 6f66 2061 6c6c 2061   a list of all a
+0000bb90: 7661 696c 6162 6c65 2070 6172 616d 6574  vailable paramet
+0000bba0: 6572 732e 0a20 2020 2020 2020 2020 2054  ers..          T
+0000bbb0: 6865 7365 2070 726f 7065 7274 6965 7320  hese properties 
+0000bbc0: 6361 6e20 6265 2063 6f6d 6269 6e65 6420  can be combined 
+0000bbd0: 7573 696e 6720 2261 6e64 2220 226f 7222  using "and" "or"
+0000bbe0: 2063 6c61 7573 6573 2061 6e64 206d 6179   clauses and may
+0000bbf0: 200a 2020 2020 2020 2020 2020 636f 6e74   .          cont
+0000bc00: 6169 6e20 636f 6d6d 6f6e 206f 7065 7261  ain common opera
+0000bc10: 746f 7273 2073 7563 6820 6173 2022 3d22  tors such as "="
+0000bc20: 2c20 223c 2220 6f72 2022 696e 222e 2028  , "<" or "in". (
+0000bc30: 652e 672e 200a 2020 2020 2020 2020 2020  e.g. .          
+0000bc40: 2274 6178 6f6e 2e63 6f6d 6d6f 6e4e 616d  "taxon.commonNam
+0000bc50: 6520 3d20 6875 6d61 6e22 2c20 2274 6178  e = human", "tax
+0000bc60: 6f6e 2e63 6f6d 6d6f 6e4e 616d 6520 696e  on.commonName in
+0000bc70: 2028 6875 6d61 6e2c 6d6f 7573 6529 2c20   (human,mouse), 
+0000bc80: 0a20 2020 2020 2020 2020 2022 6964 203c  .          "id <
+0000bc90: 2031 3030 3022 292c 2064 6566 6175 6c74   1000"), default
+0000bca0: 7320 746f 204e 6f6e 650a 2020 2020 2020  s to None.      
+0000bcb0: 2020 3a74 7970 6520 6669 6c74 6572 3a20    :type filter: 
+0000bcc0: 7374 722c 206f 7074 696f 6e61 6c0a 2020  str, optional.  
+0000bcd0: 2020 2020 2020 3a70 6172 616d 2074 6178        :param tax
+0000bce0: 613a 2041 206c 6973 7420 6f66 2074 6178  a: A list of tax
+0000bcf0: 6f6e 2063 6f6d 6d6f 6e20 6e61 6d65 7320  on common names 
+0000bd00: 2865 2e67 2e20 6875 6d61 6e2c 206d 6f75  (e.g. human, mou
+0000bd10: 7365 2c20 7261 7429 2e20 0a20 2020 2020  se, rat). .     
+0000bd20: 2020 2020 2050 726f 7669 6469 6e67 206d       Providing m
+0000bd30: 756c 7469 706c 6520 7370 6563 6965 7320  ultiple species 
+0000bd40: 7769 6c6c 2072 6574 7572 6e20 7265 7375  will return resu
+0000bd50: 6c74 7320 666f 7220 616c 6c20 7370 6563  lts for all spec
+0000bd60: 6965 732e 2054 6865 7365 200a 2020 2020  ies. These .    
+0000bd70: 2020 2020 2020 6172 6520 6170 7065 6e64        are append
+0000bd80: 6564 2074 6f20 7468 6520 6669 6c74 6572  ed to the filter
+0000bd90: 2061 6e64 2065 7175 6976 616c 656e 7420   and equivalent 
+0000bda0: 746f 2066 696c 7465 7269 6e67 2066 6f72  to filtering for
+0000bdb0: 200a 2020 2020 2020 2020 2020 7461 786f   .          taxo
+0000bdc0: 6e2e 636f 6d6d 6f6e 4e61 6d65 2070 726f  n.commonName pro
+0000bdd0: 7065 7274 792c 2064 6566 6175 6c74 7320  perty, defaults 
+0000bde0: 746f 204e 6f6e 650a 2020 2020 2020 2020  to None.        
+0000bdf0: 3a74 7970 6520 7461 7861 3a20 4c69 7374  :type taxa: List
+0000be00: 5b73 7472 5d2c 206f 7074 696f 6e61 6c0a  [str], optional.
+0000be10: 2020 2020 2020 2020 3a70 6172 616d 206f          :param o
+0000be20: 6666 7365 743a 2054 6865 206f 6666 7365  ffset: The offse
+0000be30: 7420 6f66 2074 6865 2066 6972 7374 2072  t of the first r
+0000be40: 6574 7269 6576 6564 2072 6573 756c 742e  etrieved result.
+0000be50: 2c20 6465 6661 756c 7473 2074 6f20 300a  , defaults to 0.
+0000be60: 2020 2020 2020 2020 3a74 7970 6520 6f66          :type of
+0000be70: 6673 6574 3a20 696e 742c 206f 7074 696f  fset: int, optio
+0000be80: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
+0000be90: 616d 206c 696d 6974 3a20 4c69 6d69 7473  am limit: Limits
+0000bea0: 2074 6865 2072 6573 756c 7420 746f 2073   the result to s
+0000beb0: 7065 6369 6669 6564 2061 6d6f 756e 7420  pecified amount 
+0000bec0: 6f66 206f 626a 6563 7473 2e0a 2020 2020  of objects..    
+0000bed0: 2020 2020 2020 4861 7320 6120 6d61 7869        Has a maxi
+0000bee0: 6d75 6d20 7661 6c75 6520 6f66 2031 3030  mum value of 100
+0000bef0: 2e20 5573 6520 746f 6765 7468 6572 2077  . Use together w
+0000bf00: 6974 6820 6f66 6673 6574 2061 6e64 2074  ith offset and t
+0000bf10: 6865 200a 2020 2020 2020 2020 2020 746f  he .          to
+0000bf20: 7461 6c5f 656c 656d 656e 7473 2061 7474  tal_elements att
+0000bf30: 7269 6275 7465 2069 6e20 7468 6520 6f75  ribute in the ou
+0000bf40: 7470 7574 2074 6f20 636f 6d70 696c 6520  tput to compile 
+0000bf50: 616c 6c20 6461 7461 2069 6620 6e65 6564  all data if need
+0000bf60: 6564 2e0a 2020 2020 2020 2020 2020 416c  ed..          Al
+0000bf70: 7465 726e 6174 6976 656c 7920 6765 745f  ternatively get_
+0000bf80: 616c 6c5f 7061 6765 7320 6675 6e63 7469  all_pages functi
+0000bf90: 6f6e 2063 616e 2062 6520 7573 6564 2077  on can be used w
+0000bfa0: 6974 6820 616c 6c20 6675 6e63 7469 6f6e  ith all function
+0000bfb0: 730a 2020 2020 2020 2020 2020 696e 636c  s.          incl
+0000bfc0: 7564 696e 6720 6f66 6673 6574 2061 6e64  uding offset and
+0000bfd0: 206c 696d 6974 2070 6172 616d 6574 6572   limit parameter
+0000bfe0: 732c 2064 6566 6175 6c74 7320 746f 2032  s, defaults to 2
+0000bff0: 300a 2020 2020 2020 2020 3a74 7970 6520  0.        :type 
+0000c000: 6c69 6d69 743a 2069 6e74 2c20 6f70 7469  limit: int, opti
+0000c010: 6f6e 616c 0a20 2020 2020 2020 203a 7061  onal.        :pa
+0000c020: 7261 6d20 736f 7274 3a20 4f72 6465 7220  ram sort: Order 
+0000c030: 7265 7375 6c74 7320 6279 2074 6865 2067  results by the g
+0000c040: 6976 656e 2070 726f 7065 7274 7920 616e  iven property an
+0000c050: 6420 6469 7265 6374 696f 6e2e 2054 6865  d direction. The
+0000c060: 2027 2b27 0a20 2020 2020 2020 2020 2073   '+'.          s
+0000c070: 6967 6e20 696e 6469 6361 7465 2061 7363  ign indicate asc
+0000c080: 656e 6469 6e67 206f 7264 6572 2077 6865  ending order whe
+0000c090: 7265 6173 2074 6865 2027 2d27 2069 6e64  reas the '-' ind
+0000c0a0: 6963 6174 6520 6465 7363 656e 6469 6e67  icate descending
+0000c0b0: 2c0a 2020 2020 2020 2020 2020 6465 6661  ,.          defa
+0000c0c0: 756c 7473 2074 6f20 222b 6964 220a 2020  ults to "+id".  
+0000c0d0: 2020 2020 2020 3a74 7970 6520 736f 7274        :type sort
+0000c0e0: 3a20 7374 722c 206f 7074 696f 6e61 6c0a  : str, optional.
+0000c0f0: 2020 2020 2020 2020 3a70 6172 616d 202a          :param *
+0000c100: 2a6b 7761 7267 733a 2041 6464 6974 696f  *kwargs: Additio
+0000c110: 6e61 6c20 6172 6775 6d65 6e74 7320 746f  nal arguments to
+0000c120: 2072 6177 2e67 6574 5f70 6c61 7466 6f72   raw.get_platfor
+0000c130: 6d73 5f62 795f 6964 730a 2020 2020 2020  ms_by_ids.      
+0000c140: 2020 3a72 6574 7572 6e3a 2041 2044 6174    :return: A Dat
+0000c150: 6146 7261 6d65 2077 6974 6820 696e 666f  aFrame with info
+0000c160: 726d 6174 696f 6e20 6162 6f75 7420 7468  rmation about th
+0000c170: 6520 706c 6174 666f 726d 2873 292e 0a20  e platform(s).. 
+0000c180: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+0000c190: 2020 5468 6520 6669 656c 6473 206f 6620    The fields of 
+0000c1a0: 7468 6520 6f75 7470 7574 2044 6174 6146  the output DataF
+0000c1b0: 7261 6d65 2061 7265 3a0a 0a20 2020 2020  rame are:..     
+0000c1c0: 2020 2020 2020 202d 2070 6c61 7466 6f72         - platfor
+0000c1d0: 6d5f 4944 3a20 496e 7465 726e 616c 2069  m_ID: Internal i
+0000c1e0: 6465 6e74 6966 6965 7220 6f66 2074 6865  dentifier of the
+0000c1f0: 2070 6c61 7466 6f72 6d0a 2020 2020 2020   platform.      
+0000c200: 2020 2020 2020 2d20 706c 6174 666f 726d        - platform
+0000c210: 5f73 686f 7274 5f6e 616d 653a 2053 686f  _short_name: Sho
+0000c220: 7274 6e61 6d65 206f 6620 7468 6520 706c  rtname of the pl
+0000c230: 6174 666f 726d 2e0a 2020 2020 2020 2020  atform..        
+0000c240: 2020 2020 2d20 706c 6174 666f 726d 5f6e      - platform_n
+0000c250: 616d 653a 2046 756c 6c20 6e61 6d65 206f  ame: Full name o
+0000c260: 6620 7468 6520 706c 6174 666f 726d 2e0a  f the platform..
+0000c270: 2020 2020 2020 2020 2020 2020 2d20 706c              - pl
+0000c280: 6174 666f 726d 5f64 6573 6372 6970 7469  atform_descripti
+0000c290: 6f6e 3a20 4672 6565 2074 6578 7420 6465  on: Free text de
+0000c2a0: 7363 7269 7074 696f 6e20 6f66 2074 6865  scription of the
+0000c2b0: 2070 6c61 7466 6f72 6d0a 2020 2020 2020   platform.      
+0000c2c0: 2020 2020 2020 2d20 706c 6174 666f 726d        - platform
+0000c2d0: 5f74 726f 7562 6c65 643a 2057 6865 7468  _troubled: Wheth
+0000c2e0: 6572 206f 7220 6e6f 7420 7468 6520 706c  er or not the pl
+0000c2f0: 6174 666f 726d 2077 6173 206d 6172 6b65  atform was marke
+0000c300: 6420 2274 726f 7562 6c65 6422 2062 7920  d "troubled" by 
+0000c310: 6120 4765 6d6d 6120 7072 6f63 6573 7320  a Gemma process 
+0000c320: 6f72 2061 2063 7572 6174 6f72 0a20 2020  or a curator.   
+0000c330: 2020 2020 2020 2020 202d 2070 6c61 7466           - platf
+0000c340: 6f72 6d5f 6578 7065 7269 6d65 6e74 5f63  orm_experiment_c
+0000c350: 6f75 6e74 3a20 4e75 6d62 6572 206f 6620  ount: Number of 
+0000c360: 6578 7065 7269 6d65 6e74 7320 7573 696e  experiments usin
+0000c370: 6720 7468 6520 706c 6174 666f 726d 2077  g the platform w
+0000c380: 6974 6869 6e20 4765 6d6d 610a 2020 2020  ithin Gemma.    
+0000c390: 2020 2020 2020 2020 2d20 706c 6174 666f          - platfo
+0000c3a0: 726d 5f74 7970 653a 2054 6563 686e 6f6c  rm_type: Technol
+0000c3b0: 6f67 7920 7479 7065 2066 6f72 2074 6865  ogy type for the
+0000c3c0: 2070 6c61 7466 6f72 6d2e 0a20 2020 2020   platform..     
+0000c3d0: 2020 2020 2020 202d 2074 6178 6f6e 5f6e         - taxon_n
+0000c3e0: 616d 653a 204e 616d 6520 6f66 2074 6865  ame: Name of the
+0000c3f0: 2073 7065 6369 6573 2070 6c61 7466 6f72   species platfor
+0000c400: 6d20 7761 7320 6d61 6465 2066 6f72 0a20  m was made for. 
+0000c410: 2020 2020 2020 2020 2020 202d 2074 6178             - tax
+0000c420: 6f6e 5f73 6369 656e 7469 6669 633a 2053  on_scientific: S
+0000c430: 6369 656e 7469 6669 6320 6e61 6d65 2066  cientific name f
+0000c440: 6f72 2074 6865 2074 6178 6f6e 0a20 2020  or the taxon.   
+0000c450: 2020 2020 2020 2020 202d 2074 6178 6f6e           - taxon
+0000c460: 5f49 443a 2049 6e74 6572 6e61 6c20 6964  _ID: Internal id
+0000c470: 656e 7469 6669 6572 2067 6976 656e 2074  entifier given t
+0000c480: 6f20 7468 6520 7370 6563 6965 7320 6279  o the species by
+0000c490: 2047 656d 6d61 0a20 2020 2020 2020 2020   Gemma.         
+0000c4a0: 2020 202d 2074 6178 6f6e 5f4e 4342 493a     - taxon_NCBI:
+0000c4b0: 204e 4342 4920 4944 206f 6620 7468 6520   NCBI ID of the 
+0000c4c0: 7461 786f 6e0a 2020 2020 2020 2020 2020  taxon.          
+0000c4d0: 2020 2d20 7461 786f 6e5f 6461 7461 6261    - taxon_databa
+0000c4e0: 7365 5f6e 616d 653a 2055 6e64 6572 6c79  se_name: Underly
+0000c4f0: 696e 6720 6461 7461 6261 7365 2075 7365  ing database use
+0000c500: 6420 696e 2047 656d 6d61 2066 6f72 2074  d in Gemma for t
+0000c510: 6865 2074 6178 6f6e 0a20 2020 2020 2020  he taxon.       
+0000c520: 2020 2020 202d 2074 6178 6f6e 5f64 6174       - taxon_dat
+0000c530: 6162 6173 655f 4944 3a20 4944 206f 6620  abase_ID: ID of 
+0000c540: 7468 6520 756e 6465 7279 6c69 6e67 2064  the underyling d
+0000c550: 6174 6162 6173 6520 7573 6564 2069 6e20  atabase used in 
+0000c560: 4765 6d6d 6120 666f 7220 7468 6520 7461  Gemma for the ta
+0000c570: 786f 6e0a 2020 2020 2020 2020 3a72 7479  xon.        :rty
+0000c580: 7065 3a20 4461 7461 4672 616d 650a 0a20  pe: DataFrame.. 
+0000c590: 2020 2020 2020 2022 2222 0a20 2020 2020         """.     
+0000c5a0: 2020 200a 2020 2020 2020 2020 6669 6c74     .        filt
+0000c5b0: 6572 203d 2076 732e 6164 645f 746f 5f66  er = vs.add_to_f
+0000c5c0: 696c 7465 7228 6669 6c74 6572 2c22 7461  ilter(filter,"ta
+0000c5d0: 786f 6e2e 636f 6d6d 6f6e 4e61 6d65 222c  xon.commonName",
+0000c5e0: 7461 7861 290a 2020 2020 2020 2020 0a20  taxa).        . 
+0000c5f0: 2020 2020 2020 206b 7761 7267 7320 3d20         kwargs = 
+0000c600: 7673 2e72 656d 6f76 655f 6e6f 6e65 7328  vs.remove_nones(
+0000c610: 6669 6c74 6572 203d 2066 696c 7465 722c  filter = filter,
+0000c620: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+0000c630: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c640: 2020 6f66 6673 6574 203d 206f 6666 7365    offset = offse
+0000c650: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
+0000c660: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c670: 2020 2020 6c69 6d69 7420 3d20 6c69 6d69      limit = limi
+0000c680: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
+0000c690: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c6a0: 2020 2020 736f 7274 203d 2073 6f72 742c      sort = sort,
+0000c6b0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+0000c6c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c6d0: 2020 2a2a 6b77 6172 6773 290a 2020 2020    **kwargs).    
+0000c6e0: 2020 2020 0a20 2020 2020 2020 2072 6573      .        res
+0000c6f0: 706f 6e73 6520 3d20 2073 656c 662e 7261  ponse =  self.ra
+0000c700: 772e 6765 745f 706c 6174 666f 726d 7328  w.get_platforms(
+0000c710: 2a2a 6b77 6172 6773 290a 2020 2020 2020  **kwargs).      
+0000c720: 2020 6466 203d 2070 732e 7072 6f63 6573    df = ps.proces
+0000c730: 735f 706c 6174 666f 726d 7328 7265 7370  s_platforms(resp
+0000c740: 6f6e 7365 2e64 6174 6129 0a20 2020 2020  onse.data).     
+0000c750: 2020 2070 732e 6174 7461 6368 5f61 7474     ps.attach_att
+0000c760: 7269 6275 7465 7328 6466 2c20 7265 7370  ributes(df, resp
+0000c770: 6f6e 7365 2e74 6f5f 6469 6374 2829 290a  onse.to_dict()).
+0000c780: 2020 2020 2020 2020 7265 7475 726e 2064          return d
+0000c790: 660a 0a20 2020 2023 2070 6c61 7466 6f72  f..    # platfor
+0000c7a0: 6d73 2f7b 706c 6174 666f 726d 7d2c 2067  ms/{platform}, g
+0000c7b0: 6574 5f70 6c61 7466 6f72 6d73 5f62 795f  et_platforms_by_
+0000c7c0: 6964 7320 2d2d 2d2d 200a 2020 2020 6465  ids ---- .    de
+0000c7d0: 6620 6765 745f 706c 6174 666f 726d 735f  f get_platforms_
+0000c7e0: 6279 5f69 6473 2873 656c 662c 2070 6c61  by_ids(self, pla
+0000c7f0: 7466 6f72 6d73 3a4c 6973 745b 7374 727c  tforms:List[str|
+0000c800: 696e 745d 2c20 0a20 2020 2020 2020 2020  int], .         
+0000c810: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c820: 2020 2020 6669 6c74 6572 3a73 7472 203d      filter:str =
+0000c830: 204e 6f6e 652c 0a20 2020 2020 2020 2020   None,.         
+0000c840: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c850: 2020 2020 7461 7861 3a4c 6973 745b 7374      taxa:List[st
+0000c860: 725d 203d 204e 6f6e 652c 0a20 2020 2020  r] = None,.     
+0000c870: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c880: 2020 2020 2020 2020 6f66 6673 6574 3a69          offset:i
+0000c890: 6e74 3d30 2c0a 2020 2020 2020 2020 2020  nt=0,.          
+0000c8a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c8b0: 2020 206c 696d 6974 3a69 6e74 203d 2032     limit:int = 2
+0000c8c0: 302c 0a20 2020 2020 2020 2020 2020 2020  0,.             
+0000c8d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c8e0: 736f 7274 3a73 7472 3d22 2b69 6422 2c0a  sort:str="+id",.
+0000c8f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000c900: 2020 2020 2020 2020 2020 2020 202a 2a6b               **k
+0000c910: 7761 7267 7329 2d3e 4461 7461 4672 616d  wargs)->DataFram
+0000c920: 653a 0a20 2020 0a20 2020 2020 2020 200a  e:.   .        .
+0000c930: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+0000c940: 2022 2222 5265 7472 6965 7665 2070 6c61   """Retrieve pla
+0000c950: 7466 6f72 6d73 2062 7920 7468 6569 7220  tforms by their 
+0000c960: 6964 656e 7469 6669 6572 730a 0a20 2020  identifiers..   
+0000c970: 2020 2020 200a 2020 2020 2020 2020 3a70       .        :p
+0000c980: 6172 616d 2070 6c61 7466 6f72 6d73 3a20  aram platforms: 
+0000c990: 506c 6174 666f 726d 206e 756d 6572 6963  Platform numeric
+0000c9a0: 616c 2069 6465 6e74 6966 6965 7273 206f  al identifiers o
+0000c9b0: 7220 706c 6174 666f 726d 2073 686f 7274  r platform short
+0000c9c0: 206e 616d 6573 2e0a 2020 2020 2020 2020   names..        
+0000c9d0: 3a74 7970 6520 706c 6174 666f 726d 733a  :type platforms:
+0000c9e0: 204c 6973 745b 7374 727c 696e 745d 0a20   List[str|int]. 
+0000c9f0: 2020 2020 2020 203a 7061 7261 6d20 6669         :param fi
+0000ca00: 6c74 6572 3a20 4669 6c74 6572 2072 6573  lter: Filter res
+0000ca10: 756c 7473 2062 7920 6d61 7463 6869 6e67  ults by matching
+0000ca20: 2065 7870 7265 7373 696f 6e2e 2055 7365   expression. Use
+0000ca30: 200a 2020 2020 2020 2020 2020 6669 6c74   .          filt
+0000ca40: 6572 5f70 726f 7065 7274 6965 7320 6675  er_properties fu
+0000ca50: 6e63 7469 6f6e 2074 6f20 6765 7420 6120  nction to get a 
+0000ca60: 6c69 7374 206f 6620 616c 6c20 6176 6169  list of all avai
+0000ca70: 6c61 626c 6520 7061 7261 6d65 7465 7273  lable parameters
+0000ca80: 2e0a 2020 2020 2020 2020 2020 5468 6573  ..          Thes
+0000ca90: 6520 7072 6f70 6572 7469 6573 2063 616e  e properties can
+0000caa0: 2062 6520 636f 6d62 696e 6564 2075 7369   be combined usi
+0000cab0: 6e67 2022 616e 6422 2022 6f72 2220 636c  ng "and" "or" cl
+0000cac0: 6175 7365 7320 616e 6420 6d61 7920 0a20  auses and may . 
+0000cad0: 2020 2020 2020 2020 2063 6f6e 7461 696e           contain
+0000cae0: 2063 6f6d 6d6f 6e20 6f70 6572 6174 6f72   common operator
+0000caf0: 7320 7375 6368 2061 7320 223d 222c 2022  s such as "=", "
+0000cb00: 3c22 206f 7220 2269 6e22 2e20 2865 2e67  <" or "in". (e.g
+0000cb10: 2e20 0a20 2020 2020 2020 2020 2022 7461  . .          "ta
+0000cb20: 786f 6e2e 636f 6d6d 6f6e 4e61 6d65 203d  xon.commonName =
+0000cb30: 2068 756d 616e 222c 2022 7461 786f 6e2e   human", "taxon.
+0000cb40: 636f 6d6d 6f6e 4e61 6d65 2069 6e20 2868  commonName in (h
+0000cb50: 756d 616e 2c6d 6f75 7365 292c 200a 2020  uman,mouse), .  
+0000cb60: 2020 2020 2020 2020 2269 6420 3c20 3130          "id < 10
+0000cb70: 3030 2229 2c20 6465 6661 756c 7473 2074  00"), defaults t
+0000cb80: 6f20 4e6f 6e65 0a20 2020 2020 2020 203a  o None.        :
+0000cb90: 7479 7065 2066 696c 7465 723a 2073 7472  type filter: str
+0000cba0: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
+0000cbb0: 2020 203a 7061 7261 6d20 7461 7861 3a20     :param taxa: 
+0000cbc0: 4120 6c69 7374 206f 6620 7461 786f 6e20  A list of taxon 
+0000cbd0: 636f 6d6d 6f6e 206e 616d 6573 2028 652e  common names (e.
+0000cbe0: 672e 2068 756d 616e 2c20 6d6f 7573 652c  g. human, mouse,
+0000cbf0: 2072 6174 292e 200a 2020 2020 2020 2020   rat). .        
+0000cc00: 2020 5072 6f76 6964 696e 6720 6d75 6c74    Providing mult
+0000cc10: 6970 6c65 2073 7065 6369 6573 2077 696c  iple species wil
+0000cc20: 6c20 7265 7475 726e 2072 6573 756c 7473  l return results
+0000cc30: 2066 6f72 2061 6c6c 2073 7065 6369 6573   for all species
+0000cc40: 2e20 5468 6573 6520 0a20 2020 2020 2020  . These .       
+0000cc50: 2020 2061 7265 2061 7070 656e 6465 6420     are appended 
+0000cc60: 746f 2074 6865 2066 696c 7465 7220 616e  to the filter an
+0000cc70: 6420 6571 7569 7661 6c65 6e74 2074 6f20  d equivalent to 
+0000cc80: 6669 6c74 6572 696e 6720 666f 7220 0a20  filtering for . 
+0000cc90: 2020 2020 2020 2020 2074 6178 6f6e 2e63           taxon.c
+0000cca0: 6f6d 6d6f 6e4e 616d 6520 7072 6f70 6572  ommonName proper
+0000ccb0: 7479 2c20 6465 6661 756c 7473 2074 6f20  ty, defaults to 
+0000ccc0: 4e6f 6e65 0a20 2020 2020 2020 203a 7479  None.        :ty
+0000ccd0: 7065 2074 6178 613a 204c 6973 745b 7374  pe taxa: List[st
+0000cce0: 725d 2c20 6f70 7469 6f6e 616c 0a20 2020  r], optional.   
+0000ccf0: 2020 2020 203a 7061 7261 6d20 6f66 6673       :param offs
+0000cd00: 6574 3a20 5468 6520 6f66 6673 6574 206f  et: The offset o
+0000cd10: 6620 7468 6520 6669 7273 7420 7265 7472  f the first retr
+0000cd20: 6965 7665 6420 7265 7375 6c74 2e2c 2064  ieved result., d
+0000cd30: 6566 6175 6c74 7320 746f 2030 0a20 2020  efaults to 0.   
+0000cd40: 2020 2020 203a 7479 7065 206f 6666 7365       :type offse
+0000cd50: 743a 2069 6e74 2c20 6f70 7469 6f6e 616c  t: int, optional
+0000cd60: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+0000cd70: 6c69 6d69 743a 204c 696d 6974 7320 7468  limit: Limits th
+0000cd80: 6520 7265 7375 6c74 2074 6f20 7370 6563  e result to spec
+0000cd90: 6966 6965 6420 616d 6f75 6e74 206f 6620  ified amount of 
+0000cda0: 6f62 6a65 6374 732e 0a20 2020 2020 2020  objects..       
+0000cdb0: 2020 2048 6173 2061 206d 6178 696d 756d     Has a maximum
+0000cdc0: 2076 616c 7565 206f 6620 3130 302e 2055   value of 100. U
+0000cdd0: 7365 2074 6f67 6574 6865 7220 7769 7468  se together with
+0000cde0: 206f 6666 7365 7420 616e 6420 7468 6520   offset and the 
+0000cdf0: 0a20 2020 2020 2020 2020 2074 6f74 616c  .          total
+0000ce00: 5f65 6c65 6d65 6e74 7320 6174 7472 6962  _elements attrib
+0000ce10: 7574 6520 696e 2074 6865 206f 7574 7075  ute in the outpu
+0000ce20: 7420 746f 2063 6f6d 7069 6c65 2061 6c6c  t to compile all
+0000ce30: 2064 6174 6120 6966 206e 6565 6465 642e   data if needed.
+0000ce40: 0a20 2020 2020 2020 2020 2041 6c74 6572  .          Alter
+0000ce50: 6e61 7469 7665 6c79 2067 6574 5f61 6c6c  natively get_all
+0000ce60: 5f70 6167 6573 2066 756e 6374 696f 6e20  _pages function 
+0000ce70: 6361 6e20 6265 2075 7365 6420 7769 7468  can be used with
+0000ce80: 2061 6c6c 2066 756e 6374 696f 6e73 0a20   all functions. 
+0000ce90: 2020 2020 2020 2020 2069 6e63 6c75 6469           includi
+0000cea0: 6e67 206f 6666 7365 7420 616e 6420 6c69  ng offset and li
+0000ceb0: 6d69 7420 7061 7261 6d65 7465 7273 2c20  mit parameters, 
+0000cec0: 6465 6661 756c 7473 2074 6f20 3230 0a20  defaults to 20. 
+0000ced0: 2020 2020 2020 203a 7479 7065 206c 696d         :type lim
+0000cee0: 6974 3a20 696e 742c 206f 7074 696f 6e61  it: int, optiona
+0000cef0: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
+0000cf00: 2073 6f72 743a 204f 7264 6572 2072 6573   sort: Order res
+0000cf10: 756c 7473 2062 7920 7468 6520 6769 7665  ults by the give
+0000cf20: 6e20 7072 6f70 6572 7479 2061 6e64 2064  n property and d
+0000cf30: 6972 6563 7469 6f6e 2e20 5468 6520 272b  irection. The '+
+0000cf40: 270a 2020 2020 2020 2020 2020 7369 676e  '.          sign
+0000cf50: 2069 6e64 6963 6174 6520 6173 6365 6e64   indicate ascend
+0000cf60: 696e 6720 6f72 6465 7220 7768 6572 6561  ing order wherea
+0000cf70: 7320 7468 6520 272d 2720 696e 6469 6361  s the '-' indica
+0000cf80: 7465 2064 6573 6365 6e64 696e 672c 0a20  te descending,. 
+0000cf90: 2020 2020 2020 2020 2064 6566 6175 6c74           default
+0000cfa0: 7320 746f 2022 2b69 6422 0a20 2020 2020  s to "+id".     
+0000cfb0: 2020 203a 7479 7065 2073 6f72 743a 2073     :type sort: s
+0000cfc0: 7472 2c20 6f70 7469 6f6e 616c 0a20 2020  tr, optional.   
+0000cfd0: 2020 2020 203a 7061 7261 6d20 2a2a 6b77       :param **kw
+0000cfe0: 6172 6773 3a20 4164 6469 7469 6f6e 616c  args: Additional
+0000cff0: 2061 7267 756d 656e 7473 2074 6f20 7261   arguments to ra
+0000d000: 772e 6765 745f 706c 6174 666f 726d 735f  w.get_platforms_
+0000d010: 6279 5f69 6473 0a20 2020 2020 2020 203a  by_ids.        :
+0000d020: 7265 7475 726e 3a20 4120 4461 7461 4672  return: A DataFr
+0000d030: 616d 6520 7769 7468 2069 6e66 6f72 6d61  ame with informa
+0000d040: 7469 6f6e 2061 626f 7574 2074 6865 2070  tion about the p
+0000d050: 6c61 7466 6f72 6d28 7329 2e0a 2020 2020  latform(s)..    
+0000d060: 2020 2020 0a20 2020 2020 2020 2020 2054      .          T
+0000d070: 6865 2066 6965 6c64 7320 6f66 2074 6865  he fields of the
+0000d080: 206f 7574 7075 7420 4461 7461 4672 616d   output DataFram
+0000d090: 6520 6172 653a 0a0a 2020 2020 2020 2020  e are:..        
+0000d0a0: 2020 2020 2d20 706c 6174 666f 726d 5f49      - platform_I
+0000d0b0: 443a 2049 6e74 6572 6e61 6c20 6964 656e  D: Internal iden
+0000d0c0: 7469 6669 6572 206f 6620 7468 6520 706c  tifier of the pl
+0000d0d0: 6174 666f 726d 0a20 2020 2020 2020 2020  atform.         
+0000d0e0: 2020 202d 2070 6c61 7466 6f72 6d5f 7368     - platform_sh
+0000d0f0: 6f72 745f 6e61 6d65 3a20 5368 6f72 746e  ort_name: Shortn
+0000d100: 616d 6520 6f66 2074 6865 2070 6c61 7466  ame of the platf
+0000d110: 6f72 6d2e 0a20 2020 2020 2020 2020 2020  orm..           
+0000d120: 202d 2070 6c61 7466 6f72 6d5f 6e61 6d65   - platform_name
+0000d130: 3a20 4675 6c6c 206e 616d 6520 6f66 2074  : Full name of t
+0000d140: 6865 2070 6c61 7466 6f72 6d2e 0a20 2020  he platform..   
+0000d150: 2020 2020 2020 2020 202d 2070 6c61 7466           - platf
+0000d160: 6f72 6d5f 6465 7363 7269 7074 696f 6e3a  orm_description:
+0000d170: 2046 7265 6520 7465 7874 2064 6573 6372   Free text descr
+0000d180: 6970 7469 6f6e 206f 6620 7468 6520 706c  iption of the pl
+0000d190: 6174 666f 726d 0a20 2020 2020 2020 2020  atform.         
+0000d1a0: 2020 202d 2070 6c61 7466 6f72 6d5f 7472     - platform_tr
+0000d1b0: 6f75 626c 6564 3a20 5768 6574 6865 7220  oubled: Whether 
+0000d1c0: 6f72 206e 6f74 2074 6865 2070 6c61 7466  or not the platf
+0000d1d0: 6f72 6d20 7761 7320 6d61 726b 6564 2022  orm was marked "
+0000d1e0: 7472 6f75 626c 6564 2220 6279 2061 2047  troubled" by a G
+0000d1f0: 656d 6d61 2070 726f 6365 7373 206f 7220  emma process or 
+0000d200: 6120 6375 7261 746f 720a 2020 2020 2020  a curator.      
+0000d210: 2020 2020 2020 2d20 706c 6174 666f 726d        - platform
+0000d220: 5f65 7870 6572 696d 656e 745f 636f 756e  _experiment_coun
+0000d230: 743a 204e 756d 6265 7220 6f66 2065 7870  t: Number of exp
+0000d240: 6572 696d 656e 7473 2075 7369 6e67 2074  eriments using t
+0000d250: 6865 2070 6c61 7466 6f72 6d20 7769 7468  he platform with
+0000d260: 696e 2047 656d 6d61 0a20 2020 2020 2020  in Gemma.       
+0000d270: 2020 2020 202d 2070 6c61 7466 6f72 6d5f       - platform_
+0000d280: 7479 7065 3a20 5465 6368 6e6f 6c6f 6779  type: Technology
+0000d290: 2074 7970 6520 666f 7220 7468 6520 706c   type for the pl
+0000d2a0: 6174 666f 726d 2e0a 2020 2020 2020 2020  atform..        
+0000d2b0: 2020 2020 2d20 7461 786f 6e5f 6e61 6d65      - taxon_name
+0000d2c0: 3a20 4e61 6d65 206f 6620 7468 6520 7370  : Name of the sp
+0000d2d0: 6563 6965 7320 706c 6174 666f 726d 2077  ecies platform w
+0000d2e0: 6173 206d 6164 6520 666f 720a 2020 2020  as made for.    
+0000d2f0: 2020 2020 2020 2020 2d20 7461 786f 6e5f          - taxon_
+0000d300: 7363 6965 6e74 6966 6963 3a20 5363 6965  scientific: Scie
+0000d310: 6e74 6966 6963 206e 616d 6520 666f 7220  ntific name for 
+0000d320: 7468 6520 7461 786f 6e0a 2020 2020 2020  the taxon.      
+0000d330: 2020 2020 2020 2d20 7461 786f 6e5f 4944        - taxon_ID
+0000d340: 3a20 496e 7465 726e 616c 2069 6465 6e74  : Internal ident
+0000d350: 6966 6965 7220 6769 7665 6e20 746f 2074  ifier given to t
+0000d360: 6865 2073 7065 6369 6573 2062 7920 4765  he species by Ge
+0000d370: 6d6d 610a 2020 2020 2020 2020 2020 2020  mma.            
+0000d380: 2d20 7461 786f 6e5f 4e43 4249 3a20 4e43  - taxon_NCBI: NC
+0000d390: 4249 2049 4420 6f66 2074 6865 2074 6178  BI ID of the tax
+0000d3a0: 6f6e 0a20 2020 2020 2020 2020 2020 202d  on.            -
+0000d3b0: 2074 6178 6f6e 5f64 6174 6162 6173 655f   taxon_database_
+0000d3c0: 6e61 6d65 3a20 556e 6465 726c 7969 6e67  name: Underlying
+0000d3d0: 2064 6174 6162 6173 6520 7573 6564 2069   database used i
+0000d3e0: 6e20 4765 6d6d 6120 666f 7220 7468 6520  n Gemma for the 
+0000d3f0: 7461 786f 6e0a 2020 2020 2020 2020 2020  taxon.          
+0000d400: 2020 2d20 7461 786f 6e5f 6461 7461 6261    - taxon_databa
+0000d410: 7365 5f49 443a 2049 4420 6f66 2074 6865  se_ID: ID of the
+0000d420: 2075 6e64 6572 796c 696e 6720 6461 7461   underyling data
+0000d430: 6261 7365 2075 7365 6420 696e 2047 656d  base used in Gem
+0000d440: 6d61 2066 6f72 2074 6865 2074 6178 6f6e  ma for the taxon
+0000d450: 0a20 2020 2020 2020 203a 7274 7970 653a  .        :rtype:
+0000d460: 2044 6174 6146 7261 6d65 0a0a 2020 2020   DataFrame..    
+0000d470: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
+0000d480: 6669 6c74 6572 203d 2076 732e 6164 645f  filter = vs.add_
+0000d490: 746f 5f66 696c 7465 7228 6669 6c74 6572  to_filter(filter
+0000d4a0: 2c22 7461 786f 6e2e 636f 6d6d 6f6e 4e61  ,"taxon.commonNa
+0000d4b0: 6d65 222c 7461 7861 290a 2020 2020 2020  me",taxa).      
+0000d4c0: 2020 6b77 6172 6773 203d 2076 732e 7265    kwargs = vs.re
+0000d4d0: 6d6f 7665 5f6e 6f6e 6573 2866 696c 7465  move_nones(filte
+0000d4e0: 7220 3d20 6669 6c74 6572 2c0a 2020 2020  r = filter,.    
 0000d4f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000d500: 2020 2020 2020 6c69 6d69 7420 3d20 6c69        limit = li
-0000d510: 6d69 742c 0a20 2020 2020 2020 2020 2020  mit,.           
+0000d500: 2020 2020 2020 2020 2020 2020 206f 6666               off
+0000d510: 7365 7420 3d20 6f66 6673 6574 2c0a 2020  set = offset,.  
 0000d520: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000d530: 2020 2020 2020 736f 7274 203d 2073 6f72        sort = sor
-0000d540: 742c 0a20 2020 2020 2020 2020 2020 2020  t,.             
+0000d530: 2020 2020 2020 2020 2020 2020 2020 206c                 l
+0000d540: 696d 6974 203d 206c 696d 6974 2c0a 2020  imit = limit,.  
 0000d550: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000d560: 2020 2020 2a2a 6b77 6172 6773 290a 2020      **kwargs).  
-0000d570: 2020 2020 2020 0a20 2020 2020 2020 2072        .        r
-0000d580: 6573 706f 6e73 6520 3d20 2073 656c 662e  esponse =  self.
-0000d590: 7261 772e 6765 745f 706c 6174 666f 726d  raw.get_platform
-0000d5a0: 735f 6279 5f69 6473 2870 6c61 7466 6f72  s_by_ids(platfor
-0000d5b0: 6d73 2c20 2a2a 6b77 6172 6773 290a 2020  ms, **kwargs).  
-0000d5c0: 2020 2020 2020 6466 203d 2070 732e 7072        df = ps.pr
-0000d5d0: 6f63 6573 735f 706c 6174 666f 726d 7328  ocess_platforms(
-0000d5e0: 7265 7370 6f6e 7365 2e64 6174 6129 0a20  response.data). 
-0000d5f0: 2020 2020 2020 2070 732e 6174 7461 6368         ps.attach
-0000d600: 5f61 7474 7269 6275 7465 7328 6466 2c20  _attributes(df, 
-0000d610: 7265 7370 6f6e 7365 2e74 6f5f 6469 6374  response.to_dict
-0000d620: 2829 290a 2020 2020 2020 2020 7265 7475  ()).        retu
-0000d630: 726e 2064 660a 2020 2020 0a20 2020 200a  rn df.    .    .
-0000d640: 2020 2020 2320 7365 6172 6368 2073 6561      # search sea
-0000d650: 7263 6820 2d2d 2d2d 2d2d 0a20 2020 2023  rch ------.    #
-0000d660: 2074 6869 7320 656e 7064 6f69 6e74 2069   this enpdoint i
-0000d670: 7320 6e6f 7420 7665 7279 2075 7365 6675  s not very usefu
-0000d680: 6c20 7768 656e 2073 7065 6369 6669 6320  l when specific 
-0000d690: 656e 6470 6f69 6e74 7320 6578 6973 7420  endpoints exist 
-0000d6a0: 666f 7220 7370 6563 6966 6963 0a20 2020  for specific.   
-0000d6b0: 2023 2072 6573 756c 7420 7479 7065 732e   # result types.
-0000d6c0: 206b 6565 7069 6e67 2068 6572 6520 666f   keeping here fo
-0000d6d0: 7220 6e6f 7720 666f 7220 636f 6d70 6174  r now for compat
-0000d6e0: 6962 696c 6974 7920 7769 7468 2052 0a20  ibility with R. 
-0000d6f0: 2020 200a 2020 2020 6465 6620 7365 6172     .    def sear
-0000d700: 6368 5f67 656d 6d61 2873 656c 662c 0a20  ch_gemma(self,. 
-0000d710: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000d720: 2020 2020 7175 6572 793a 7374 722c 0a20      query:str,. 
-0000d730: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000d740: 2020 2020 7461 786f 6e3a 4f70 7469 6f6e      taxon:Option
-0000d750: 616c 5b73 7472 7c69 6e74 5d3d 4e6f 6e65  al[str|int]=None
-0000d760: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
-0000d770: 2020 2020 2020 2070 6c61 7466 6f72 6d3a         platform:
-0000d780: 4f70 7469 6f6e 616c 5b73 7472 7c69 6e74  Optional[str|int
-0000d790: 5d20 3d20 4e6f 6e65 2c0a 2020 2020 2020  ] = None,.      
-0000d7a0: 2020 2020 2020 2020 2020 2020 2020 206c                 l
-0000d7b0: 696d 6974 3a69 6e74 203d 2031 3030 2c0a  imit:int = 100,.
-0000d7c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000d7d0: 2020 2020 2072 6573 756c 745f 7479 7065       result_type
-0000d7e0: 3a73 7472 203d 2022 6578 7065 7269 6d65  :str = "experime
-0000d7f0: 6e74 222c 0a20 2020 2020 2020 2020 2020  nt",.           
-0000d800: 2020 2020 2020 2020 2020 2a2a 6b77 6172            **kwar
-0000d810: 6773 292d 3e6c 6973 745b 7364 6b2e 5365  gs)->list[sdk.Se
-0000d820: 6172 6368 5265 7375 6c74 5661 6c75 654f  archResultValueO
-0000d830: 626a 6563 744f 626a 6563 745d 3a0a 0a20  bjectObject]:.. 
-0000d840: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-0000d850: 2222 220a 2020 2020 2020 2020 5365 6172  """.        Sear
-0000d860: 6368 2065 7665 7279 7468 696e 6720 696e  ch everything in
-0000d870: 2047 656d 6d61 0a20 2020 2020 2020 200a   Gemma.        .
-0000d880: 2020 2020 2020 2020 3a70 6172 616d 2071          :param q
-0000d890: 7565 7279 3a20 5468 6520 7365 6172 6368  uery: The search
-0000d8a0: 2071 7565 7279 2e20 4569 7468 6572 2070   query. Either p
-0000d8b0: 6c61 696e 2074 6578 7420 2827 7472 6175  lain text ('trau
-0000d8c0: 6d61 7469 6327 292c 206f 7220 616e 200a  matic'), or an .
-0000d8d0: 2020 2020 2020 2020 2020 6f6e 746f 6c6f            ontolo
-0000d8e0: 6779 2074 6572 6d20 5552 4920 2827 6874  gy term URI ('ht
-0000d8f0: 7470 3a2f 2f70 7572 6c2e 6f62 6f6c 6962  tp://purl.obolib
-0000d900: 7261 7279 2e6f 7267 2f6f 626f 2f55 4245  rary.org/obo/UBE
-0000d910: 524f 4e5f 3030 3032 3034 3827 292e 0a20  RON_0002048').. 
-0000d920: 2020 2020 2020 2020 2044 6174 6173 6574           Dataset
-0000d930: 7320 7468 6174 2063 6f6e 7461 696e 2074  s that contain t
-0000d940: 6865 2067 6976 656e 2073 7472 696e 6720  he given string 
-0000d950: 696e 2074 6865 6972 2073 686f 7274 206f  in their short o
-0000d960: 6620 6675 6c6c 206e 616d 650a 2020 2020  f full name.    
-0000d970: 2020 2020 2020 7769 6c6c 2061 6c73 6f20        will also 
-0000d980: 6265 206d 6174 6368 6564 2e20 4361 6e20  be matched. Can 
-0000d990: 6265 206d 756c 7469 706c 6520 6964 656e  be multiple iden
-0000d9a0: 7469 6669 6572 7320 7365 7061 7261 7465  tifiers separate
-0000d9b0: 6420 6279 2063 6f6d 6d61 732e 0a20 2020  d by commas..   
-0000d9c0: 2020 2020 203a 7479 7065 2071 7565 7279       :type query
-0000d9d0: 3a20 7374 720a 2020 2020 2020 2020 3a70  : str.        :p
-0000d9e0: 6172 616d 2074 6178 6f6e 3a20 4120 6e75  aram taxon: A nu
-0000d9f0: 6d65 7269 6361 6c20 7461 786f 6e20 6964  merical taxon id
-0000da00: 656e 7469 6669 6572 206f 7220 616e 206e  entifier or an n
-0000da10: 6362 6920 7461 786f 6e20 6964 656e 7469  cbi taxon identi
-0000da20: 6669 6572 0a20 2020 2020 2020 2020 206f  fier.          o
-0000da30: 7220 6120 7461 786f 6e20 6964 656e 7469  r a taxon identi
-0000da40: 6669 6572 2074 6861 7420 6d61 7463 6865  fier that matche
-0000da50: 7320 6569 7468 6572 2069 7473 2073 6369  s either its sci
-0000da60: 656e 7469 6669 6320 6f72 2063 6f6d 6d6f  entific or commo
-0000da70: 6e0a 2020 2020 2020 2020 2020 6e61 6d65  n.          name
-0000da80: 2c20 6465 6661 756c 7473 2074 6f20 4e6f  , defaults to No
-0000da90: 6e65 0a20 2020 2020 2020 203a 7479 7065  ne.        :type
-0000daa0: 2074 6178 6f6e 3a20 4f70 7469 6f6e 616c   taxon: Optional
-0000dab0: 5b73 7472 7c69 6e74 5d2c 206f 7074 696f  [str|int], optio
-0000dac0: 6e61 6c0a 2020 2020 2020 2020 3a70 6172  nal.        :par
-0000dad0: 616d 2070 6c61 7466 6f72 6d3a 2041 2070  am platform: A p
-0000dae0: 6c61 7466 6f72 6d20 6e75 6d65 7269 6361  latform numerica
-0000daf0: 6c20 6964 656e 7469 6669 6572 206f 7220  l identifier or 
-0000db00: 6120 706c 6174 666f 726d 2073 686f 7274  a platform short
-0000db10: 200a 2020 2020 2020 2020 2020 6e61 6d65   .          name
-0000db20: 2c20 6465 6661 756c 7473 2074 6f20 4e6f  , defaults to No
-0000db30: 6e65 0a20 2020 2020 2020 203a 7479 7065  ne.        :type
-0000db40: 2070 6c61 7466 6f72 6d3a 204f 7074 696f   platform: Optio
-0000db50: 6e61 6c5b 7374 727c 696e 745d 2c20 6f70  nal[str|int], op
-0000db60: 7469 6f6e 616c 0a20 2020 2020 2020 203a  tional.        :
-0000db70: 7061 7261 6d20 6c69 6d69 743a 2044 6566  param limit: Def
-0000db80: 6175 6c74 7320 746f 2031 3030 2077 6974  aults to 100 wit
-0000db90: 6820 6120 6d61 7869 6d75 6d20 7661 6c75  h a maximum valu
-0000dba0: 6520 6f66 2032 3030 302e 204c 696d 6974  e of 2000. Limit
-0000dbb0: 7320 7468 650a 2020 2020 2020 2020 2020  s the.          
-0000dbc0: 6e75 6d62 6572 206f 6620 7265 7475 726e  number of return
-0000dbd0: 6564 2072 6573 756c 7473 2e20 4e6f 7465  ed results. Note
-0000dbe0: 2074 6861 7420 7468 6973 2066 756e 6374   that this funct
-0000dbf0: 696f 6e20 646f 6573 206e 6f74 2073 7570  ion does not sup
-0000dc00: 706f 7274 0a20 2020 2020 2020 2020 2070  port.          p
-0000dc10: 6167 696e 6174 696f 6e2e 2c20 6465 6661  agination., defa
-0000dc20: 756c 7473 2074 6f20 3130 300a 2020 2020  ults to 100.    
-0000dc30: 2020 2020 3a74 7970 6520 6c69 6d69 743a      :type limit:
-0000dc40: 2069 6e74 2c20 6f70 7469 6f6e 616c 0a20   int, optional. 
-0000dc50: 2020 2020 2020 203a 7061 7261 6d20 7265         :param re
-0000dc60: 7375 6c74 5f74 7970 653a 2054 6865 206b  sult_type: The k
-0000dc70: 696e 6420 6f66 2072 6573 756c 7473 2074  ind of results t
-0000dc80: 6861 7420 7368 6f75 6c64 2062 6520 696e  hat should be in
-0000dc90: 636c 7564 6564 2069 6e20 7468 650a 2020  cluded in the.  
-0000dca0: 2020 2020 2020 2020 6f75 7470 7574 2e20          output. 
-0000dcb0: 4361 6e20 6265 2022 6578 7065 7269 6d65  Can be "experime
-0000dcc0: 6e74 222c 2022 6765 6e65 222c 2022 706c  nt", "gene", "pl
-0000dcd0: 6174 666f 726d 2220 6f72 2061 206c 6f6e  atform" or a lon
-0000dce0: 6720 6f62 6a65 6374 2074 7970 6520 6e61  g object type na
-0000dcf0: 6d65 2c0a 2020 2020 2020 2020 2020 646f  me,.          do
-0000dd00: 6375 6d65 6e74 6564 2069 6e20 7468 6520  cumented in the 
-0000dd10: 4150 4920 646f 6375 6d65 6e74 6174 696f  API documentatio
-0000dd20: 6e2e 2c20 6465 6661 756c 7473 2074 6f20  n., defaults to 
-0000dd30: 2265 7870 6572 696d 656e 7422 0a20 2020  "experiment".   
-0000dd40: 2020 2020 203a 7479 7065 2072 6573 756c       :type resul
-0000dd50: 745f 7479 7065 3a20 7374 722c 206f 7074  t_type: str, opt
-0000dd60: 696f 6e61 6c0a 2020 2020 2020 2020 3a70  ional.        :p
-0000dd70: 6172 616d 202a 2a6b 7761 7267 733a 2041  aram **kwargs: A
-0000dd80: 6464 6974 696f 6e61 6c20 6172 6775 6d65  dditional argume
-0000dd90: 6e74 7320 746f 2072 6177 2e73 6561 7263  nts to raw.searc
-0000dda0: 680a 2020 2020 2020 2020 3a72 6574 7572  h.        :retur
-0000ddb0: 6e3a 2041 206c 6973 7420 636f 6e74 6169  n: A list contai
-0000ddc0: 6e69 6e67 2074 6865 2072 6573 756c 7473  ning the results
-0000ddd0: 2e20 4163 7475 616c 2072 6573 756c 7473  . Actual results
-0000dde0: 2061 7265 2075 6e64 6572 2074 6865 200a   are under the .
-0000ddf0: 2020 2020 2020 2020 2020 7265 7375 6c74            result
-0000de00: 5f6f 626a 6563 7420 636f 6d70 6f6e 656e  _object componen
-0000de10: 7420 6173 2064 6963 7473 0a20 2020 2020  t as dicts.     
-0000de20: 2020 203a 7274 7970 653a 206c 6973 745b     :rtype: list[
-0000de30: 7364 6b2e 5365 6172 6368 5265 7375 6c74  sdk.SearchResult
-0000de40: 5661 6c75 654f 626a 6563 744f 626a 6563  ValueObjectObjec
-0000de50: 745d 0a0a 2020 2020 2020 2020 2222 220a  t]..        """.
-0000de60: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-0000de70: 2072 6573 756c 745f 7479 7065 203d 2076   result_type = v
-0000de80: 732e 6368 6563 6b5f 7265 7375 6c74 5f74  s.check_result_t
-0000de90: 7970 6528 7265 7375 6c74 5f74 7970 6529  ype(result_type)
-0000dea0: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
-0000deb0: 2020 6b77 6172 6773 203d 2076 732e 7265    kwargs = vs.re
-0000dec0: 6d6f 7665 5f6e 6f6e 6573 2871 7565 7279  move_nones(query
-0000ded0: 203d 7175 6572 792c 0a20 2020 2020 2020   =query,.       
-0000dee0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000def0: 2020 2020 2020 2020 2020 7461 786f 6e20            taxon 
-0000df00: 3d20 7461 786f 6e2c 0a20 2020 2020 2020  = taxon,.       
-0000df10: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000df20: 2020 2020 2020 2020 2020 706c 6174 666f            platfo
-0000df30: 726d 203d 2070 6c61 7466 6f72 6d2c 0a20  rm = platform,. 
-0000df40: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000d560: 2020 2020 2020 2020 2020 2020 2020 2073                 s
+0000d570: 6f72 7420 3d20 736f 7274 2c0a 2020 2020  ort = sort,.    
+0000d580: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000d590: 2020 2020 2020 2020 2020 2020 202a 2a6b               **k
+0000d5a0: 7761 7267 7329 0a20 2020 2020 2020 200a  wargs).        .
+0000d5b0: 2020 2020 2020 2020 7265 7370 6f6e 7365          response
+0000d5c0: 203d 2020 7365 6c66 2e72 6177 2e67 6574   =  self.raw.get
+0000d5d0: 5f70 6c61 7466 6f72 6d73 5f62 795f 6964  _platforms_by_id
+0000d5e0: 7328 706c 6174 666f 726d 732c 202a 2a6b  s(platforms, **k
+0000d5f0: 7761 7267 7329 0a20 2020 2020 2020 2064  wargs).        d
+0000d600: 6620 3d20 7073 2e70 726f 6365 7373 5f70  f = ps.process_p
+0000d610: 6c61 7466 6f72 6d73 2872 6573 706f 6e73  latforms(respons
+0000d620: 652e 6461 7461 290a 2020 2020 2020 2020  e.data).        
+0000d630: 7073 2e61 7474 6163 685f 6174 7472 6962  ps.attach_attrib
+0000d640: 7574 6573 2864 662c 2072 6573 706f 6e73  utes(df, respons
+0000d650: 652e 746f 5f64 6963 7428 2929 0a20 2020  e.to_dict()).   
+0000d660: 2020 2020 2072 6574 7572 6e20 6466 0a20       return df. 
+0000d670: 2020 200a 2020 2020 0a20 2020 2023 2073     .    .    # s
+0000d680: 6561 7263 6820 7365 6172 6368 202d 2d2d  earch search ---
+0000d690: 2d2d 2d0a 2020 2020 2320 7468 6973 2065  ---.    # this e
+0000d6a0: 6e70 646f 696e 7420 6973 206e 6f74 2076  npdoint is not v
+0000d6b0: 6572 7920 7573 6566 756c 2077 6865 6e20  ery useful when 
+0000d6c0: 7370 6563 6966 6963 2065 6e64 706f 696e  specific endpoin
+0000d6d0: 7473 2065 7869 7374 2066 6f72 2073 7065  ts exist for spe
+0000d6e0: 6369 6669 630a 2020 2020 2320 7265 7375  cific.    # resu
+0000d6f0: 6c74 2074 7970 6573 2e20 6b65 6570 696e  lt types. keepin
+0000d700: 6720 6865 7265 2066 6f72 206e 6f77 2066  g here for now f
+0000d710: 6f72 2063 6f6d 7061 7469 6269 6c69 7479  or compatibility
+0000d720: 2077 6974 6820 520a 2020 2020 0a20 2020   with R.    .   
+0000d730: 2064 6566 2073 6561 7263 685f 6765 6d6d   def search_gemm
+0000d740: 6128 7365 6c66 2c0a 2020 2020 2020 2020  a(self,.        
+0000d750: 2020 2020 2020 2020 2020 2020 2071 7565               que
+0000d760: 7279 3a73 7472 2c0a 2020 2020 2020 2020  ry:str,.        
+0000d770: 2020 2020 2020 2020 2020 2020 2074 6178               tax
+0000d780: 6f6e 3a4f 7074 696f 6e61 6c5b 7374 727c  on:Optional[str|
+0000d790: 696e 745d 3d4e 6f6e 652c 0a20 2020 2020  int]=None,.     
+0000d7a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000d7b0: 706c 6174 666f 726d 3a4f 7074 696f 6e61  platform:Optiona
+0000d7c0: 6c5b 7374 727c 696e 745d 203d 204e 6f6e  l[str|int] = Non
+0000d7d0: 652c 0a20 2020 2020 2020 2020 2020 2020  e,.             
+0000d7e0: 2020 2020 2020 2020 6c69 6d69 743a 696e          limit:in
+0000d7f0: 7420 3d20 3130 302c 0a20 2020 2020 2020  t = 100,.       
+0000d800: 2020 2020 2020 2020 2020 2020 2020 7265                re
+0000d810: 7375 6c74 5f74 7970 653a 7374 7220 3d20  sult_type:str = 
+0000d820: 2265 7870 6572 696d 656e 7422 2c0a 2020  "experiment",.  
+0000d830: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000d840: 2020 202a 2a6b 7761 7267 7329 2d3e 6c69     **kwargs)->li
+0000d850: 7374 5b73 646b 2e53 6561 7263 6852 6573  st[sdk.SearchRes
+0000d860: 756c 7456 616c 7565 4f62 6a65 6374 4f62  ultValueObjectOb
+0000d870: 6a65 6374 5d3a 0a0a 2020 2020 2020 2020  ject]:..        
+0000d880: 0a20 2020 2020 2020 2022 2222 0a20 2020  .        """.   
+0000d890: 2020 2020 2053 6561 7263 6820 6576 6572       Search ever
+0000d8a0: 7974 6869 6e67 2069 6e20 4765 6d6d 610a  ything in Gemma.
+0000d8b0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+0000d8c0: 203a 7061 7261 6d20 7175 6572 793a 2054   :param query: T
+0000d8d0: 6865 2073 6561 7263 6820 7175 6572 792e  he search query.
+0000d8e0: 2045 6974 6865 7220 706c 6169 6e20 7465   Either plain te
+0000d8f0: 7874 2028 2774 7261 756d 6174 6963 2729  xt ('traumatic')
+0000d900: 2c20 6f72 2061 6e20 0a20 2020 2020 2020  , or an .       
+0000d910: 2020 206f 6e74 6f6c 6f67 7920 7465 726d     ontology term
+0000d920: 2055 5249 2028 2768 7474 703a 2f2f 7075   URI ('http://pu
+0000d930: 726c 2e6f 626f 6c69 6272 6172 792e 6f72  rl.obolibrary.or
+0000d940: 672f 6f62 6f2f 5542 4552 4f4e 5f30 3030  g/obo/UBERON_000
+0000d950: 3230 3438 2729 2e0a 2020 2020 2020 2020  2048')..        
+0000d960: 2020 4461 7461 7365 7473 2074 6861 7420    Datasets that 
+0000d970: 636f 6e74 6169 6e20 7468 6520 6769 7665  contain the give
+0000d980: 6e20 7374 7269 6e67 2069 6e20 7468 6569  n string in thei
+0000d990: 7220 7368 6f72 7420 6f66 2066 756c 6c20  r short of full 
+0000d9a0: 6e61 6d65 0a20 2020 2020 2020 2020 2077  name.          w
+0000d9b0: 696c 6c20 616c 736f 2062 6520 6d61 7463  ill also be matc
+0000d9c0: 6865 642e 2043 616e 2062 6520 6d75 6c74  hed. Can be mult
+0000d9d0: 6970 6c65 2069 6465 6e74 6966 6965 7273  iple identifiers
+0000d9e0: 2073 6570 6172 6174 6564 2062 7920 636f   separated by co
+0000d9f0: 6d6d 6173 2e0a 2020 2020 2020 2020 3a74  mmas..        :t
+0000da00: 7970 6520 7175 6572 793a 2073 7472 0a20  ype query: str. 
+0000da10: 2020 2020 2020 203a 7061 7261 6d20 7461         :param ta
+0000da20: 786f 6e3a 2041 206e 756d 6572 6963 616c  xon: A numerical
+0000da30: 2074 6178 6f6e 2069 6465 6e74 6966 6965   taxon identifie
+0000da40: 7220 6f72 2061 6e20 6e63 6269 2074 6178  r or an ncbi tax
+0000da50: 6f6e 2069 6465 6e74 6966 6965 720a 2020  on identifier.  
+0000da60: 2020 2020 2020 2020 6f72 2061 2074 6178          or a tax
+0000da70: 6f6e 2069 6465 6e74 6966 6965 7220 7468  on identifier th
+0000da80: 6174 206d 6174 6368 6573 2065 6974 6865  at matches eithe
+0000da90: 7220 6974 7320 7363 6965 6e74 6966 6963  r its scientific
+0000daa0: 206f 7220 636f 6d6d 6f6e 0a20 2020 2020   or common.     
+0000dab0: 2020 2020 206e 616d 652c 2064 6566 6175       name, defau
+0000dac0: 6c74 7320 746f 204e 6f6e 650a 2020 2020  lts to None.    
+0000dad0: 2020 2020 3a74 7970 6520 7461 786f 6e3a      :type taxon:
+0000dae0: 204f 7074 696f 6e61 6c5b 7374 727c 696e   Optional[str|in
+0000daf0: 745d 2c20 6f70 7469 6f6e 616c 0a20 2020  t], optional.   
+0000db00: 2020 2020 203a 7061 7261 6d20 706c 6174       :param plat
+0000db10: 666f 726d 3a20 4120 706c 6174 666f 726d  form: A platform
+0000db20: 206e 756d 6572 6963 616c 2069 6465 6e74   numerical ident
+0000db30: 6966 6965 7220 6f72 2061 2070 6c61 7466  ifier or a platf
+0000db40: 6f72 6d20 7368 6f72 7420 0a20 2020 2020  orm short .     
+0000db50: 2020 2020 206e 616d 652c 2064 6566 6175       name, defau
+0000db60: 6c74 7320 746f 204e 6f6e 650a 2020 2020  lts to None.    
+0000db70: 2020 2020 3a74 7970 6520 706c 6174 666f      :type platfo
+0000db80: 726d 3a20 4f70 7469 6f6e 616c 5b73 7472  rm: Optional[str
+0000db90: 7c69 6e74 5d2c 206f 7074 696f 6e61 6c0a  |int], optional.
+0000dba0: 2020 2020 2020 2020 3a70 6172 616d 206c          :param l
+0000dbb0: 696d 6974 3a20 4465 6661 756c 7473 2074  imit: Defaults t
+0000dbc0: 6f20 3130 3020 7769 7468 2061 206d 6178  o 100 with a max
+0000dbd0: 696d 756d 2076 616c 7565 206f 6620 3230  imum value of 20
+0000dbe0: 3030 2e20 4c69 6d69 7473 2074 6865 0a20  00. Limits the. 
+0000dbf0: 2020 2020 2020 2020 206e 756d 6265 7220           number 
+0000dc00: 6f66 2072 6574 7572 6e65 6420 7265 7375  of returned resu
+0000dc10: 6c74 732e 204e 6f74 6520 7468 6174 2074  lts. Note that t
+0000dc20: 6869 7320 6675 6e63 7469 6f6e 2064 6f65  his function doe
+0000dc30: 7320 6e6f 7420 7375 7070 6f72 740a 2020  s not support.  
+0000dc40: 2020 2020 2020 2020 7061 6769 6e61 7469          paginati
+0000dc50: 6f6e 2e2c 2064 6566 6175 6c74 7320 746f  on., defaults to
+0000dc60: 2031 3030 0a20 2020 2020 2020 203a 7479   100.        :ty
+0000dc70: 7065 206c 696d 6974 3a20 696e 742c 206f  pe limit: int, o
+0000dc80: 7074 696f 6e61 6c0a 2020 2020 2020 2020  ptional.        
+0000dc90: 3a70 6172 616d 2072 6573 756c 745f 7479  :param result_ty
+0000dca0: 7065 3a20 5468 6520 6b69 6e64 206f 6620  pe: The kind of 
+0000dcb0: 7265 7375 6c74 7320 7468 6174 2073 686f  results that sho
+0000dcc0: 756c 6420 6265 2069 6e63 6c75 6465 6420  uld be included 
+0000dcd0: 696e 2074 6865 0a20 2020 2020 2020 2020  in the.         
+0000dce0: 206f 7574 7075 742e 2043 616e 2062 6520   output. Can be 
+0000dcf0: 2265 7870 6572 696d 656e 7422 2c20 2267  "experiment", "g
+0000dd00: 656e 6522 2c20 2270 6c61 7466 6f72 6d22  ene", "platform"
+0000dd10: 206f 7220 6120 6c6f 6e67 206f 626a 6563   or a long objec
+0000dd20: 7420 7479 7065 206e 616d 652c 0a20 2020  t type name,.   
+0000dd30: 2020 2020 2020 2064 6f63 756d 656e 7465         documente
+0000dd40: 6420 696e 2074 6865 2041 5049 2064 6f63  d in the API doc
+0000dd50: 756d 656e 7461 7469 6f6e 2e2c 2064 6566  umentation., def
+0000dd60: 6175 6c74 7320 746f 2022 6578 7065 7269  aults to "experi
+0000dd70: 6d65 6e74 220a 2020 2020 2020 2020 3a74  ment".        :t
+0000dd80: 7970 6520 7265 7375 6c74 5f74 7970 653a  ype result_type:
+0000dd90: 2073 7472 2c20 6f70 7469 6f6e 616c 0a20   str, optional. 
+0000dda0: 2020 2020 2020 203a 7061 7261 6d20 2a2a         :param **
+0000ddb0: 6b77 6172 6773 3a20 4164 6469 7469 6f6e  kwargs: Addition
+0000ddc0: 616c 2061 7267 756d 656e 7473 2074 6f20  al arguments to 
+0000ddd0: 7261 772e 7365 6172 6368 0a20 2020 2020  raw.search.     
+0000dde0: 2020 203a 7265 7475 726e 3a20 4120 6c69     :return: A li
+0000ddf0: 7374 2063 6f6e 7461 696e 696e 6720 7468  st containing th
+0000de00: 6520 7265 7375 6c74 732e 2041 6374 7561  e results. Actua
+0000de10: 6c20 7265 7375 6c74 7320 6172 6520 756e  l results are un
+0000de20: 6465 7220 7468 6520 0a20 2020 2020 2020  der the .       
+0000de30: 2020 2072 6573 756c 745f 6f62 6a65 6374     result_object
+0000de40: 2063 6f6d 706f 6e65 6e74 2061 7320 6469   component as di
+0000de50: 6374 730a 2020 2020 2020 2020 3a72 7479  cts.        :rty
+0000de60: 7065 3a20 6c69 7374 5b73 646b 2e53 6561  pe: list[sdk.Sea
+0000de70: 7263 6852 6573 756c 7456 616c 7565 4f62  rchResultValueOb
+0000de80: 6a65 6374 4f62 6a65 6374 5d0a 0a20 2020  jectObject]..   
+0000de90: 2020 2020 2022 2222 0a20 2020 2020 2020       """.       
+0000dea0: 200a 2020 2020 2020 2020 7265 7375 6c74   .        result
+0000deb0: 5f74 7970 6520 3d20 7673 2e63 6865 636b  _type = vs.check
+0000dec0: 5f72 6573 756c 745f 7479 7065 2872 6573  _result_type(res
+0000ded0: 756c 745f 7479 7065 290a 2020 2020 2020  ult_type).      
+0000dee0: 2020 0a20 2020 2020 2020 206b 7761 7267    .        kwarg
+0000def0: 7320 3d20 7673 2e72 656d 6f76 655f 6e6f  s = vs.remove_no
+0000df00: 6e65 7328 7175 6572 7920 3d71 7565 7279  nes(query =query
+0000df10: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+0000df20: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000df30: 2020 2074 6178 6f6e 203d 2074 6178 6f6e     taxon = taxon
+0000df40: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
 0000df50: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000df60: 6c69 6d69 7420 3d20 6c69 6d69 742c 0a20  limit = limit,. 
-0000df70: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000df60: 2020 2070 6c61 7466 6f72 6d20 3d20 706c     platform = pl
+0000df70: 6174 666f 726d 2c0a 2020 2020 2020 2020  atform,.        
 0000df80: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000df90: 7265 7375 6c74 5f74 7970 6573 203d 205b  result_types = [
-0000dfa0: 7265 7375 6c74 5f74 7970 655d 2c0a 2020  result_type],.  
+0000df90: 2020 2020 2020 2020 206c 696d 6974 203d           limit =
+0000dfa0: 206c 696d 6974 2c0a 2020 2020 2020 2020   limit,.        
 0000dfb0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000dfc0: 2020 2020 2020 2020 2020 2020 2020 202a                 *
-0000dfd0: 2a6b 7761 7267 7329 0a20 2020 2020 2020  *kwargs).       
-0000dfe0: 200a 2020 2020 2020 2020 7265 7370 6f6e   .        respon
-0000dff0: 7365 203d 2073 656c 662e 7261 772e 7365  se = self.raw.se
-0000e000: 6172 6368 282a 2a6b 7761 7267 7329 0a20  arch(**kwargs). 
-0000e010: 2020 2020 2020 2023 2064 6620 3d20 7073         # df = ps
-0000e020: 2e70 726f 6365 7373 5f73 6561 7263 6828  .process_search(
-0000e030: 7265 7370 6f6e 7365 2e64 6174 612c 7265  response.data,re
-0000e040: 7375 6c74 5f74 7970 6529 0a20 2020 2020  sult_type).     
-0000e050: 2020 200a 2020 2020 2020 2020 7265 7475     .        retu
-0000e060: 726e 2072 6573 706f 6e73 652e 6461 7461  rn response.data
-0000e070: 0a20 2020 200a 2020 2020 2320 7461 7861  .    .    # taxa
-0000e080: 2f7b 7461 786f 6e7d 2f67 656e 6573 2f7b  /{taxon}/genes/{
-0000e090: 6765 6e65 7d2f 6c6f 6361 7469 6f6e 732d  gene}/locations-
-0000e0a0: 2d2d 2d0a 2020 2020 2320 756e 696d 706c  ---.    # unimpl
-0000e0b0: 656d 656e 7465 642c 2072 6564 756e 6461  emented, redunda
-0000e0c0: 6e74 2077 6974 6820 6765 745f 6765 6e65  nt with get_gene
-0000e0d0: 5f6c 6f63 6174 696f 6e73 0a0a 2020 2020  _locations..    
-0000e0e0: 2320 7461 7861 202d 2d2d 2d0a 2020 2020  # taxa ----.    
-0000e0f0: 6465 6620 6765 745f 7461 7861 2873 656c  def get_taxa(sel
-0000e100: 662c 202a 2a6b 7761 7267 7329 2d3e 4461  f, **kwargs)->Da
-0000e110: 7461 4672 616d 653a 0a20 2020 2020 2020  taFrame:.       
-0000e120: 2022 2222 0a20 2020 2020 2020 2047 6574   """.        Get
-0000e130: 2061 6c6c 2074 6178 6120 7769 7468 696e   all taxa within
-0000e140: 2047 656d 6d61 0a20 2020 2020 2020 200a   Gemma.        .
-0000e150: 2020 2020 2020 2020 3a70 6172 616d 202a          :param *
-0000e160: 2a6b 7761 7267 733a 2041 6464 6974 696f  *kwargs: Additio
-0000e170: 6e61 6c20 6172 6775 6d65 6e74 7320 746f  nal arguments to
-0000e180: 2072 6177 2e67 6574 5f74 6178 610a 2020   raw.get_taxa.  
-0000e190: 2020 2020 2020 3a72 6574 7572 6e3a 2041        :return: A
-0000e1a0: 2044 6174 6146 7261 6d65 2069 6e63 6c75   DataFrame inclu
-0000e1b0: 6469 6e67 2074 6865 206e 616d 6573 2c20  ding the names, 
-0000e1c0: 4944 7320 616e 6420 6461 7461 6261 7365  IDs and database
-0000e1d0: 2069 6e66 6f72 6d61 7469 6f6e 0a20 2020   information.   
-0000e1e0: 2020 2020 2020 2061 626f 7574 2074 6865         about the
-0000e1f0: 2074 6178 6f6e 730a 2020 2020 2020 2020   taxons.        
-0000e200: 3a72 7479 7065 3a20 4461 7461 4672 616d  :rtype: DataFram
-0000e210: 650a 0a20 2020 2020 2020 2022 2222 0a20  e..        """. 
-0000e220: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-0000e230: 7265 7370 6f6e 7365 203d 2020 7365 6c66  response =  self
-0000e240: 2e72 6177 2e67 6574 5f74 6178 6128 2a2a  .raw.get_taxa(**
-0000e250: 6b77 6172 6773 290a 2020 2020 2020 2020  kwargs).        
-0000e260: 0a20 2020 2020 2020 2064 6620 3d20 7073  .        df = ps
-0000e270: 2e70 726f 6365 7373 5f74 6178 6f6e 2872  .process_taxon(r
-0000e280: 6573 706f 6e73 652e 6461 7461 290a 2020  esponse.data).  
-0000e290: 2020 2020 2020 7265 7475 726e 2064 665b        return df[
-0000e2a0: 6466 2e69 736e 756c 6c28 292e 7461 786f  df.isnull().taxo
-0000e2b0: 6e5f 6e61 6d65 2021 3d20 5472 7565 5d0a  n_name != True].
-0000e2c0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-0000e2d0: 200a 2020 2020 0a20 2020 2023 2074 6178   .    .    # tax
-0000e2e0: 612f 7b74 6178 617d 2c20 6765 745f 7461  a/{taxa}, get_ta
-0000e2f0: 7861 5f62 795f 6964 7320 2d2d 2d2d 2d0a  xa_by_ids -----.
-0000e300: 2020 2020 2320 696d 706c 656d 656e 7465      # implemente
-0000e310: 642c 2068 6172 646c 7920 6e65 6564 6564  d, hardly needed
-0000e320: 2077 6974 6820 3320 7461 7861 2020 0a0a   with 3 taxa  ..
-0000e330: 0a0a 2320 4265 6c6f 7720 6172 6520 2243  ..# Below are "C
-0000e340: 6f6e 7665 6e69 656e 6365 2220 2863 6f6d  onvenience" (com
-0000e350: 6269 6e61 7469 6f6e 2920 6675 6e63 7469  bination) functi
-0000e360: 6f6e 730a 0a20 2020 2023 2073 6574 5f67  ons..    # set_g
-0000e370: 656d 6d61 5f75 7365 7220 6973 206e 6f74  emma_user is not
-0000e380: 206e 6565 6465 6420 7369 6e63 6520 6974   needed since it
-0000e390: 2773 2077 7261 7070 6564 2069 6e20 7468  's wrapped in th
-0000e3a0: 6520 4765 6d6d 6150 7920 636c 6173 730a  e GemmaPy class.
-0000e3b0: 2020 2020 2320 6765 745f 706c 6174 666f      # get_platfo
-0000e3c0: 726d 5f61 6e6e 6f74 6174 696f 6e73 2069  rm_annotations i
-0000e3d0: 7320 7468 6520 6465 6661 756c 7420 6765  s the default ge
-0000e3e0: 745f 706c 6174 666f 726d 5f61 6e6e 6f74  t_platform_annot
-0000e3f0: 6174 696f 6e73 0a20 2020 200a 0a20 2020  ations.    ..   
-0000e400: 2064 6566 206d 616b 655f 6465 7369 676e   def make_design
-0000e410: 2873 656c 662c 7361 6d70 6c65 733a 4461  (self,samples:Da
-0000e420: 7461 4672 616d 652c 6d65 7461 5f74 7970  taFrame,meta_typ
-0000e430: 653a 7374 7220 3d20 2774 6578 7427 292d  e:str = 'text')-
-0000e440: 3e44 6174 6146 7261 6d65 3a0a 2020 2020  >DataFrame:.    
-0000e450: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
-0000e460: 5573 696e 6720 6f6e 2074 6865 206f 7574  Using on the out
-0000e470: 7075 7420 6f66 2067 6574 5f64 6174 6173  put of get_datas
-0000e480: 6574 5f73 616d 706c 6573 2c20 7468 6973  et_samples, this
-0000e490: 2066 756e 6374 696f 6e20 6372 6561 7465   function create
-0000e4a0: 7320 610a 2020 2020 2020 2020 7369 6d70  s a.        simp
-0000e4b0: 6c69 6669 6564 2064 6573 6967 6e20 7461  lified design ta
-0000e4c0: 626c 652c 2067 7261 6e74 696e 6720 6f6e  ble, granting on
-0000e4d0: 6520 636f 6c75 6d6e 2074 6f20 6561 6368  e column to each
-0000e4e0: 2065 7870 6572 696d 656e 7461 6c20 7661   experimental va
-0000e4f0: 7269 6162 6c65 0a0a 0a20 2020 2020 2020  riable...       
-0000e500: 203a 7061 7261 6d20 7361 6d70 6c65 733a   :param samples:
-0000e510: 2041 6e20 6f75 7470 7574 2066 726f 6d20   An output from 
-0000e520: 6765 745f 6461 7461 7365 745f 7361 6d70  get_dataset_samp
-0000e530: 6c65 732e 0a20 2020 2020 2020 203a 7479  les..        :ty
-0000e540: 7065 2073 616d 706c 6573 3a20 4461 7461  pe samples: Data
-0000e550: 4672 616d 650a 2020 2020 2020 2020 3a70  Frame.        :p
-0000e560: 6172 616d 206d 6574 615f 7479 7065 3a20  aram meta_type: 
-0000e570: 5479 7065 206f 6620 6d65 7461 6461 7461  Type of metadata
-0000e580: 2074 6f20 696e 636c 7564 6520 696e 2074   to include in t
-0000e590: 6865 206f 7574 7075 742e 2022 7465 7874  he output. "text
-0000e5a0: 222c 0a20 2020 2020 2020 2020 2022 7572  ",.          "ur
-0000e5b0: 6922 206f 7220 2262 6f74 6822 2c20 6465  i" or "both", de
-0000e5c0: 6661 756c 7473 2074 6f20 2774 6578 7427  faults to 'text'
-0000e5d0: 0a20 2020 2020 2020 203a 7479 7065 206d  .        :type m
-0000e5e0: 6574 615f 7479 7065 3a20 7374 722c 206f  eta_type: str, o
-0000e5f0: 7074 696f 6e61 6c0a 2020 2020 2020 2020  ptional.        
-0000e600: 3a72 6574 7572 6e3a 2041 2044 6174 6146  :return: A DataF
-0000e610: 7261 6d65 2069 6e63 6c75 6469 6e67 2074  rame including t
-0000e620: 6865 2064 6573 6967 6e20 7461 626c 6520  he design table 
-0000e630: 666f 7220 7468 6520 6461 7461 7365 740a  for the dataset.
-0000e640: 2020 2020 2020 2020 3a72 7479 7065 3a20          :rtype: 
-0000e650: 4461 7461 4672 616d 650a 0a20 2020 2020  DataFrame..     
-0000e660: 2020 2022 2222 0a20 2020 2020 2020 200a     """.        .
-0000e670: 2020 2020 2020 2020 6361 7465 676f 7269          categori
-0000e680: 6573 203d 2070 642e 636f 6e63 6174 285b  es = pd.concat([
-0000e690: 785b 5b22 6661 6374 6f72 5f49 4422 2c22  x[["factor_ID","
-0000e6a0: 6661 6374 6f72 5f63 6174 6567 6f72 7922  factor_category"
-0000e6b0: 2c22 6661 6374 6f72 5f63 6174 6567 6f72  ,"factor_categor
-0000e6c0: 795f 5552 4922 5d5d 200a 2020 2020 2020  y_URI"]] .      
-0000e6d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000e6e0: 2020 2020 2020 2020 2020 666f 7220 7820            for x 
-0000e6f0: 696e 2073 616d 706c 6573 2e73 616d 706c  in samples.sampl
-0000e700: 655f 6661 6374 6f72 5f76 616c 7565 735d  e_factor_values]
-0000e710: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
-0000e720: 2020 2020 6967 6e6f 7265 5f69 6e64 6578      ignore_index
-0000e730: 203d 2054 7275 6529 2e64 726f 705f 6475   = True).drop_du
-0000e740: 706c 6963 6174 6573 2829 0a20 2020 2020  plicates().     
-0000e750: 2020 200a 2020 2020 2020 2020 0a20 2020     .        .   
-0000e760: 2020 2020 2064 6566 2067 6574 5f76 616c       def get_val
-0000e770: 5f75 7269 2878 293a 0a20 2020 2020 2020  _uri(x):.       
-0000e780: 2020 2020 2072 6574 7572 6e20 5b22 2c22       return [","
-0000e790: 2e6a 6f69 6e28 5b73 7472 287a 2920 6966  .join([str(z) if
-0000e7a0: 207a 2069 7320 6e6f 7420 4e6f 6e65 2065   z is not None e
-0000e7b0: 6c73 6520 2222 200a 2020 2020 2020 2020  lse "" .        
-0000e7c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000e7d0: 2020 2020 2020 666f 7220 7a20 696e 2079        for z in y
-0000e7e0: 5b79 2e66 6163 746f 725f 4944 3d3d 785d  [y.factor_ID==x]
-0000e7f0: 2e76 616c 7565 5f55 5249 5d29 200a 2020  .value_URI]) .  
-0000e800: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000e810: 2020 666f 7220 7920 696e 2073 616d 706c    for y in sampl
-0000e820: 6573 2e73 616d 706c 655f 6661 6374 6f72  es.sample_factor
-0000e830: 5f76 616c 7565 735d 0a20 2020 2020 2020  _values].       
-0000e840: 200a 2020 2020 2020 2020 6661 6374 6f72   .        factor
-0000e850: 5f55 5249 7320 3d20 5b67 6574 5f76 616c  _URIs = [get_val
-0000e860: 5f75 7269 2878 2920 666f 7220 7820 696e  _uri(x) for x in
-0000e870: 2063 6174 6567 6f72 6965 732e 6661 6374   categories.fact
-0000e880: 6f72 5f49 445d 0a20 2020 2020 2020 200a  or_ID].        .
-0000e890: 2020 2020 2020 2020 6465 6620 6765 745f          def get_
-0000e8a0: 7465 7874 2878 293a 0a20 2020 2020 2020  text(x):.       
-0000e8b0: 2020 2020 2064 6566 2067 6574 5f73 756d       def get_sum
-0000e8c0: 6d61 7279 2879 293a 0a20 2020 2020 2020  mary(y):.       
-0000e8d0: 2020 2020 2020 2020 2072 6574 7572 6e20           return 
-0000e8e0: 272c 272e 6a6f 696e 285b 7a5b 315d 2e73  ','.join([z[1].s
-0000e8f0: 756d 6d61 7279 0a20 2020 2020 2020 2020  ummary.         
-0000e900: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000e910: 2020 2020 2020 2020 6966 207a 5b31 5d2e          if z[1].
-0000e920: 7375 6d6d 6172 7920 6973 206e 6f74 204e  summary is not N
-0000e930: 6f6e 6520 656c 7365 207a 5b31 5d2e 7661  one else z[1].va
-0000e940: 6c75 6520 0a20 2020 2020 2020 2020 2020  lue .           
-0000e950: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000e960: 2020 2020 2020 666f 7220 7a20 696e 2079        for z in y
-0000e970: 5b79 2e66 6163 746f 725f 4944 3d3d 785d  [y.factor_ID==x]
-0000e980: 2e69 7465 7272 6f77 7328 295d 290a 2020  .iterrows()]).  
-0000e990: 2020 2020 2020 2020 2020 0a20 2020 2020            .     
-0000e9a0: 2020 2020 2020 2072 6574 7572 6e20 5b67         return [g
-0000e9b0: 6574 5f73 756d 6d61 7279 2879 2920 666f  et_summary(y) fo
-0000e9c0: 7220 7920 696e 2073 616d 706c 6573 2e73  r y in samples.s
-0000e9d0: 616d 706c 655f 6661 6374 6f72 5f76 616c  ample_factor_val
-0000e9e0: 7565 735d 0a20 2020 2020 2020 2020 2020  ues].           
-0000e9f0: 200a 2020 2020 2020 2020 2020 2020 0a20   .            . 
-0000ea00: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-0000ea10: 7465 7874 203d 205b 6765 745f 7465 7874  text = [get_text
-0000ea20: 2878 2920 666f 7220 7820 696e 2063 6174  (x) for x in cat
-0000ea30: 6567 6f72 6965 732e 6661 6374 6f72 5f49  egories.factor_I
-0000ea40: 445d 0a20 2020 2020 2020 200a 2020 2020  D].        .    
-0000ea50: 2020 2020 0a20 2020 2020 2020 2069 6620      .        if 
-0000ea60: 6d65 7461 5f74 7970 6520 3d3d 2774 6578  meta_type =='tex
-0000ea70: 7427 3a0a 2020 2020 2020 2020 2020 2020  t':.            
-0000ea80: 6465 7369 676e 5f66 7261 6d65 203d 2070  design_frame = p
-0000ea90: 642e 4461 7461 4672 616d 6528 7b0a 2020  d.DataFrame({.  
-0000eaa0: 2020 2020 2020 2020 2020 2020 2020 6361                ca
-0000eab0: 7465 676f 7269 6573 2e66 6163 746f 725f  tegories.factor_
-0000eac0: 6361 7465 676f 7279 5b69 5d3a 7465 7874  category[i]:text
-0000ead0: 5b69 5d20 666f 7220 6920 696e 2072 616e  [i] for i in ran
-0000eae0: 6765 286c 656e 2874 6578 7429 290a 2020  ge(len(text)).  
-0000eaf0: 2020 2020 2020 2020 2020 2020 2020 7d29                })
-0000eb00: 0a20 2020 2020 2020 2065 6c69 6620 6d65  .        elif me
-0000eb10: 7461 5f74 7970 6520 3d3d 2027 7572 6927  ta_type == 'uri'
-0000eb20: 3a0a 2020 2020 2020 2020 2020 2020 6465  :.            de
-0000eb30: 7369 676e 5f66 7261 6d65 203d 2070 642e  sign_frame = pd.
-0000eb40: 4461 7461 4672 616d 6528 7b0a 2020 2020  DataFrame({.    
-0000eb50: 2020 2020 2020 2020 2020 2020 6361 7465              cate
-0000eb60: 676f 7269 6573 2e66 6163 746f 725f 6361  gories.factor_ca
-0000eb70: 7465 676f 7279 5f55 5249 5b69 5d3a 6661  tegory_URI[i]:fa
-0000eb80: 6374 6f72 5f55 5249 735b 695d 2066 6f72  ctor_URIs[i] for
-0000eb90: 2069 2069 6e20 7261 6e67 6528 6c65 6e28   i in range(len(
-0000eba0: 6661 6374 6f72 5f55 5249 7329 290a 2020  factor_URIs)).  
-0000ebb0: 2020 2020 2020 2020 2020 2020 2020 7d29                })
-0000ebc0: 0a20 2020 2020 2020 2065 6c69 6620 6d65  .        elif me
-0000ebd0: 7461 5f74 7970 6520 3d3d 2762 6f74 6827  ta_type =='both'
-0000ebe0: 3a0a 2020 2020 2020 2020 2020 2020 6d65  :.            me
-0000ebf0: 7267 6564 5f6e 616d 6520 3d20 5b5b 227c  rged_name = [["|
-0000ec00: 222e 6a6f 696e 285b 6361 7465 676f 7269  ".join([categori
-0000ec10: 6573 2e66 6163 746f 725f 6361 7465 676f  es.factor_catego
-0000ec20: 7279 5b69 5d2c 0a20 2020 2020 2020 2020  ry[i],.         
-0000ec30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ec40: 2020 2020 2020 2020 2020 2020 2063 6174               cat
-0000ec50: 6567 6f72 6965 732e 6661 6374 6f72 5f63  egories.factor_c
-0000ec60: 6174 6567 6f72 795f 5552 495b 695d 5d29  ategory_URI[i]])
-0000ec70: 5d20 666f 7220 6920 696e 2072 616e 6765  ] for i in range
-0000ec80: 286c 656e 2874 6578 7429 295d 0a20 2020  (len(text))].   
-0000ec90: 2020 2020 2020 2020 200a 2020 2020 2020           .      
-0000eca0: 2020 2020 2020 6d65 7267 6564 5f63 6f6c        merged_col
-0000ecb0: 203d 205b 5b22 7c22 2e6a 6f69 6e28 5b74   = [["|".join([t
-0000ecc0: 6578 745b 695d 5b6a 5d2c 6661 6374 6f72  ext[i][j],factor
-0000ecd0: 5f55 5249 735b 695d 5b6a 5d5d 2920 0a20  _URIs[i][j]]) . 
-0000ece0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ecf0: 2020 2020 2020 2020 2020 666f 7220 6a20            for j 
-0000ed00: 696e 2072 616e 6765 286c 656e 2874 6578  in range(len(tex
-0000ed10: 745b 695d 2929 5d20 666f 7220 6920 696e  t[i]))] for i in
-0000ed20: 2072 616e 6765 286c 656e 2874 6578 7429   range(len(text)
-0000ed30: 295d 0a20 2020 2020 2020 2020 2020 200a  )].            .
-0000ed40: 2020 2020 2020 2020 2020 2020 0a20 2020              .   
-0000ed50: 2020 2020 2020 2020 2064 6573 6967 6e5f           design_
-0000ed60: 6672 616d 6520 3d20 7064 2e44 6174 6146  frame = pd.DataF
-0000ed70: 7261 6d65 287b 0a20 2020 2020 2020 2020  rame({.         
-0000ed80: 2020 2020 2020 206d 6572 6765 645f 6e61         merged_na
-0000ed90: 6d65 5b69 5d3a 6d65 7267 6564 5f63 6f6c  me[i]:merged_col
-0000eda0: 5b69 5d0a 2020 2020 2020 2020 2020 2020  [i].            
-0000edb0: 2020 2020 666f 7220 6920 696e 2072 616e      for i in ran
-0000edc0: 6765 286c 656e 2874 6578 7429 297d 290a  ge(len(text))}).
-0000edd0: 2020 2020 2020 2020 2020 2020 0a20 2020              .   
-0000ede0: 2020 2020 2064 6573 6967 6e5f 6672 616d       design_fram
-0000edf0: 652e 696e 7365 7274 286c 6f63 203d 2030  e.insert(loc = 0
-0000ee00: 2c63 6f6c 756d 6e20 3d20 2266 6163 746f  ,column = "facto
-0000ee10: 725f 7661 6c75 6573 222c 0a20 2020 2020  r_values",.     
-0000ee20: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ee30: 2020 2020 2020 2076 616c 7565 203d 2073         value = s
-0000ee40: 616d 706c 6573 2e73 616d 706c 655f 6661  amples.sample_fa
-0000ee50: 6374 6f72 5f76 616c 7565 7329 0a20 2020  ctor_values).   
-0000ee60: 2020 2020 2064 6573 6967 6e5f 6672 616d       design_fram
-0000ee70: 652e 696e 6465 7820 3d20 7361 6d70 6c65  e.index = sample
-0000ee80: 732e 7361 6d70 6c65 5f6e 616d 650a 2020  s.sample_name.  
-0000ee90: 2020 2020 2020 0a20 2020 2020 2020 2072        .        r
-0000eea0: 6574 7572 6e20 6465 7369 676e 5f66 7261  eturn design_fra
-0000eeb0: 6d65 0a20 2020 2020 2020 200a 2020 2020  me.        .    
-0000eec0: 2020 2020 0a20 2020 200a 2020 2020 6465      .    .    de
-0000eed0: 6620 5f5f 7375 6273 6574 5f66 6163 746f  f __subset_facto
-0000eee0: 725f 7661 6c75 6573 2873 656c 662c 0a20  r_values(self,. 
-0000eef0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ef00: 2020 2020 2020 2020 2020 2020 2020 6661                fa
-0000ef10: 6374 6f72 5f76 616c 7565 732c 0a20 2020  ctor_values,.   
-0000ef20: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ef30: 2020 2020 2020 2020 2020 2020 6469 6666              diff
-0000ef40: 6572 656e 7469 616c 5f65 7870 7265 7373  erential_express
-0000ef50: 696f 6e73 3a70 642e 4461 7461 4672 616d  ions:pd.DataFram
-0000ef60: 652c 0a20 2020 2020 2020 2020 2020 2020  e,.             
-0000ef70: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000ef80: 2020 7265 7375 6c74 5f73 6574 2c0a 2020    result_set,.  
-0000ef90: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000efa0: 2020 2020 2020 2020 2020 2020 2063 6f6e               con
-0000efb0: 7472 6173 7429 3a0a 2020 2020 2020 2020  trast):.        
-0000efc0: 6f75 7420 3d20 7375 622e 7265 7028 5472  out = sub.rep(Tr
-0000efd0: 7565 2c6c 656e 2866 6163 746f 725f 7661  ue,len(factor_va
-0000efe0: 6c75 6573 2929 0a20 2020 2020 2020 2069  lues)).        i
-0000eff0: 6620 6469 6666 6572 656e 7469 616c 5f65  f differential_e
-0000f000: 7870 7265 7373 696f 6e73 2069 7320 6e6f  xpressions is no
-0000f010: 7420 4e6f 6e65 3a0a 2020 2020 2020 2020  t None:.        
-0000f020: 2020 2020 7375 6273 6574 203d 2064 6966      subset = dif
-0000f030: 6665 7265 6e74 6961 6c5f 6578 7072 6573  ferential_expres
-0000f040: 7369 6f6e 735b 6469 6666 6572 656e 7469  sions[differenti
-0000f050: 616c 5f65 7870 7265 7373 696f 6e73 2e72  al_expressions.r
-0000f060: 6573 756c 745f 4944 203d 3d20 0a20 2020  esult_ID == .   
-0000f070: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f080: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f090: 2020 7265 7375 6c74 5f73 6574 5d2e 7375    result_set].su
-0000f0a0: 6273 6574 5f66 6163 746f 722e 6472 6f70  bset_factor.drop
-0000f0b0: 5f64 7570 6c69 6361 7465 7328 290a 2020  _duplicates().  
-0000f0c0: 2020 2020 2020 2020 2020 2320 7265 7375            # resu
-0000f0d0: 6c74 2073 6574 2073 686f 756c 6420 6861  lt set should ha
-0000f0e0: 7665 2074 6865 2073 616d 6520 7375 6273  ve the same subs
-0000f0f0: 6574 2066 6f72 2061 6c6c 2063 6f6e 7472  et for all contr
-0000f100: 6173 7473 0a20 2020 2020 2020 2020 2020  asts.           
-0000f110: 2061 7373 6572 7420 6c65 6e28 7375 6273   assert len(subs
-0000f120: 6574 2920 3d3d 2031 0a20 2020 2020 2020  et) == 1.       
-0000f130: 2020 2020 2069 6620 7375 6273 6574 5b30       if subset[0
-0000f140: 5d2e 7368 6170 655b 305d 213d 303a 0a20  ].shape[0]!=0:. 
-0000f150: 2020 2020 2020 2020 2020 2020 2020 2073                 s
-0000f160: 7562 7365 745f 6964 7320 3d20 7375 6273  ubset_ids = subs
-0000f170: 6574 5b30 5d2e 4944 0a20 2020 2020 2020  et[0].ID.       
-0000f180: 2020 2020 2020 2020 200a 2020 2020 2020           .      
-0000f190: 2020 2020 2020 2020 2020 696e 5f73 7562            in_sub
-0000f1a0: 7365 7420 3d20 5b61 6e79 2873 7562 2e6c  set = [any(sub.l
-0000f1b0: 6973 745f 696e 5f6c 6973 7428 782e 4944  ist_in_list(x.ID
-0000f1c0: 2c20 7375 6273 6574 5f69 6473 2929 2066  , subset_ids)) f
-0000f1d0: 6f72 2078 2069 6e20 6661 6374 6f72 5f76  or x in factor_v
-0000f1e0: 616c 7565 735d 0a20 2020 2020 2020 2020  alues].         
-0000f1f0: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-0000f200: 2020 2020 2020 2020 6f75 7420 3d20 6f75          out = ou
-0000f210: 7420 616e 6420 696e 5f73 7562 7365 740a  t and in_subset.
-0000f220: 2020 2020 2020 2020 2020 2020 0a20 2020              .   
-0000f230: 2020 2020 2020 2020 2069 6620 636f 6e74           if cont
-0000f240: 7261 7374 2069 7320 6e6f 7420 4e6f 6e65  rast is not None
-0000f250: 3a0a 2020 2020 2020 2020 2020 2020 2020  :.              
-0000f260: 2020 636e 203d 2064 6966 6665 7265 6e74    cn = different
-0000f270: 6961 6c5f 6578 7072 6573 7369 6f6e 735b  ial_expressions[
-0000f280: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-0000f290: 2020 2020 206c 6973 7428 6469 6666 6572       list(differ
-0000f2a0: 656e 7469 616c 5f65 7870 7265 7373 696f  ential_expressio
-0000f2b0: 6e73 2e72 6573 756c 745f 4944 203d 3d20  ns.result_ID == 
-0000f2c0: 7265 7375 6c74 5f73 6574 2920 616e 6420  result_set) and 
-0000f2d0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-0000f2e0: 2020 2020 206c 6973 7428 6469 6666 6572       list(differ
-0000f2f0: 656e 7469 616c 5f65 7870 7265 7373 696f  ential_expressio
-0000f300: 6e73 2e63 6f6e 7472 6173 745f 4944 203d  ns.contrast_ID =
-0000f310: 3d20 7374 7228 636f 6e74 7261 7374 2929  = str(contrast))
-0000f320: 5d0a 2020 2020 2020 2020 2020 2020 2020  ].              
-0000f330: 2020 0a20 2020 2020 2020 2020 2020 2020    .             
-0000f340: 2020 2062 6173 656c 696e 655f 6964 203d     baseline_id =
-0000f350: 206c 6973 7428 7375 622e 756e 6971 7565   list(sub.unique
-0000f360: 2873 7562 2e62 7265 616b 5f6c 6973 7428  (sub.break_list(
-0000f370: 5b6c 6973 7428 782e 4944 2920 666f 7220  [list(x.ID) for 
-0000f380: 7820 696e 2063 6e2e 6261 7365 6c69 6e65  x in cn.baseline
-0000f390: 5f66 6163 746f 7273 5d29 2929 0a20 2020  _factors]))).   
-0000f3a0: 2020 2020 2020 2020 2020 2020 2062 6173               bas
-0000f3b0: 656c 696e 655f 6661 6374 6f72 5f69 6420  eline_factor_id 
-0000f3c0: 3d20 6c69 7374 2873 7562 2e75 6e69 7175  = list(sub.uniqu
-0000f3d0: 6528 7375 622e 6272 6561 6b5f 6c69 7374  e(sub.break_list
-0000f3e0: 285b 6c69 7374 2878 2e66 6163 746f 725f  ([list(x.factor_
-0000f3f0: 4944 2920 666f 7220 7820 696e 2063 6e2e  ID) for x in cn.
-0000f400: 6261 7365 6c69 6e65 5f66 6163 746f 7273  baseline_factors
-0000f410: 5d29 2929 0a20 2020 2020 2020 2020 2020  ]))).           
-0000f420: 2020 2020 200a 2020 2020 2020 2020 2020       .          
-0000f430: 2020 2020 2020 636f 6e74 7261 7374 5f69        contrast_i
-0000f440: 6420 3d20 206c 6973 7428 7375 622e 756e  d =  list(sub.un
-0000f450: 6971 7565 2873 7562 2e62 7265 616b 5f6c  ique(sub.break_l
-0000f460: 6973 7428 5b6c 6973 7428 782e 4944 2920  ist([list(x.ID) 
-0000f470: 666f 7220 7820 696e 2063 6e2e 6578 7065  for x in cn.expe
-0000f480: 7269 6d65 6e74 616c 5f66 6163 746f 7273  rimental_factors
-0000f490: 5d29 2929 0a20 2020 2020 2020 2020 2020  ]))).           
-0000f4a0: 2020 2020 2063 6f6e 7472 6173 745f 6661       contrast_fa
-0000f4b0: 6374 6f72 5f69 6420 3d20 206c 6973 7428  ctor_id =  list(
-0000f4c0: 7375 622e 756e 6971 7565 2873 7562 2e62  sub.unique(sub.b
-0000f4d0: 7265 616b 5f6c 6973 7428 5b6c 6973 7428  reak_list([list(
-0000f4e0: 782e 6661 6374 6f72 5f49 4429 2066 6f72  x.factor_ID) for
-0000f4f0: 2078 2069 6e20 636e 2e65 7870 6572 696d   x in cn.experim
-0000f500: 656e 7461 6c5f 6661 6374 6f72 735d 2929  ental_factors]))
-0000f510: 290a 2020 2020 2020 2020 2020 2020 2020  ).              
-0000f520: 2020 0a20 2020 2020 2020 2020 2020 2020    .             
-0000f530: 2020 2063 6f6e 7472 6173 745f 6964 203d     contrast_id =
-0000f540: 2073 7562 2e6d 6174 6368 5f62 7928 636f   sub.match_by(co
-0000f550: 6e74 7261 7374 5f69 642c 6261 7365 6c69  ntrast_id,baseli
-0000f560: 6e65 5f66 6163 746f 725f 6964 2c20 636f  ne_factor_id, co
-0000f570: 6e74 7261 7374 5f66 6163 746f 725f 6964  ntrast_factor_id
-0000f580: 290a 2020 2020 2020 2020 2020 2020 2020  ).              
-0000f590: 2020 0a20 2020 2020 2020 2020 2020 2020    .             
-0000f5a0: 2020 2064 6566 2069 6e5f 636f 6e28 6661     def in_con(fa
-0000f5b0: 6374 6f72 5f76 616c 7565 293a 0a20 2020  ctor_value):.   
-0000f5c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f5d0: 2063 6f6e 6431 203d 2061 6c6c 2873 7562   cond1 = all(sub
-0000f5e0: 2e6c 6973 745f 696e 5f6c 6973 7428 636f  .list_in_list(co
-0000f5f0: 6e74 7261 7374 5f69 642c 2066 6163 746f  ntrast_id, facto
-0000f600: 725f 7661 6c75 652e 4944 2929 206f 7220  r_value.ID)) or 
-0000f610: 5c0a 2020 2020 2020 2020 2020 2020 2020  \.              
-0000f620: 2020 2020 2020 2020 2020 2020 2020 616c                al
-0000f630: 6c28 7375 622e 6c69 7374 5f69 6e5f 6c69  l(sub.list_in_li
-0000f640: 7374 2862 6173 656c 696e 655f 6964 2c66  st(baseline_id,f
-0000f650: 6163 746f 725f 7661 6c75 652e 4944 2929  actor_value.ID))
-0000f660: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f670: 2020 200a 2020 2020 2020 2020 2020 2020     .            
-0000f680: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-0000f690: 2020 2020 2020 2020 2020 2020 2069 6620               if 
-0000f6a0: 6c65 6e28 636f 6e74 7261 7374 5f69 6429  len(contrast_id)
-0000f6b0: 3d3d 323a 0a20 2020 2020 2020 2020 2020  ==2:.           
-0000f6c0: 2020 2020 2020 2020 2020 2020 2063 6f6e               con
-0000f6d0: 6432 203d 2028 636f 6e74 7261 7374 5f69  d2 = (contrast_i
-0000f6e0: 645b 305d 2069 6e20 6661 6374 6f72 5f76  d[0] in factor_v
-0000f6f0: 616c 7565 2e49 4420 616e 6420 5c0a 2020  alue.ID and \.  
-0000f700: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f710: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f720: 2020 6261 7365 6c69 6e65 5f69 645b 315d    baseline_id[1]
-0000f730: 2069 6e20 6661 6374 6f72 5f76 616c 7565   in factor_value
-0000f740: 2e49 4429 206f 7220 5c0a 2020 2020 2020  .ID) or \.      
-0000f750: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f760: 2020 2020 2020 2863 6f6e 7472 6173 745f        (contrast_
-0000f770: 6964 5b31 5d20 696e 2066 6163 746f 725f  id[1] in factor_
-0000f780: 7661 6c75 652e 4944 2061 6e64 205c 0a20  value.ID and \. 
-0000f790: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f7a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f7b0: 2020 2020 2020 6261 7365 6c69 6e65 5f69        baseline_i
-0000f7c0: 645b 305d 2069 6e20 6661 6374 6f72 5f76  d[0] in factor_v
-0000f7d0: 616c 7565 2e49 4429 0a20 2020 2020 2020  alue.ID).       
-0000f7e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f7f0: 200a 2020 2020 2020 2020 2020 2020 2020   .              
-0000f800: 2020 2020 2020 2020 2020 636f 6e64 3120            cond1 
-0000f810: 3d20 636f 6e64 3120 6f72 2063 6f6e 6432  = cond1 or cond2
-0000f820: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-0000f830: 2020 2020 200a 2020 2020 2020 2020 2020       .          
-0000f840: 2020 2020 2020 2020 2020 7265 7475 726e            return
-0000f850: 2063 6f6e 6431 0a20 2020 2020 2020 2020   cond1.         
-0000f860: 2020 2020 2020 2020 2020 2020 2020 200a                 .
+0000dfc0: 2020 2020 2020 2020 2072 6573 756c 745f           result_
+0000dfd0: 7479 7065 7320 3d20 5b72 6573 756c 745f  types = [result_
+0000dfe0: 7479 7065 5d2c 0a20 2020 2020 2020 2020  type],.         
+0000dff0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000e000: 2020 2020 2020 2020 2a2a 6b77 6172 6773          **kwargs
+0000e010: 290a 2020 2020 2020 2020 0a20 2020 2020  ).        .     
+0000e020: 2020 2072 6573 706f 6e73 6520 3d20 7365     response = se
+0000e030: 6c66 2e72 6177 2e73 6561 7263 6828 2a2a  lf.raw.search(**
+0000e040: 6b77 6172 6773 290a 2020 2020 2020 2020  kwargs).        
+0000e050: 2320 6466 203d 2070 732e 7072 6f63 6573  # df = ps.proces
+0000e060: 735f 7365 6172 6368 2872 6573 706f 6e73  s_search(respons
+0000e070: 652e 6461 7461 2c72 6573 756c 745f 7479  e.data,result_ty
+0000e080: 7065 290a 2020 2020 2020 2020 0a20 2020  pe).        .   
+0000e090: 2020 2020 2072 6574 7572 6e20 7265 7370       return resp
+0000e0a0: 6f6e 7365 2e64 6174 610a 2020 2020 0a20  onse.data.    . 
+0000e0b0: 2020 2023 2074 6178 612f 7b74 6178 6f6e     # taxa/{taxon
+0000e0c0: 7d2f 6765 6e65 732f 7b67 656e 657d 2f6c  }/genes/{gene}/l
+0000e0d0: 6f63 6174 696f 6e73 2d2d 2d2d 0a20 2020  ocations----.   
+0000e0e0: 2023 2075 6e69 6d70 6c65 6d65 6e74 6564   # unimplemented
+0000e0f0: 2c20 7265 6475 6e64 616e 7420 7769 7468  , redundant with
+0000e100: 2067 6574 5f67 656e 655f 6c6f 6361 7469   get_gene_locati
+0000e110: 6f6e 730a 0a20 2020 2023 2074 6178 6120  ons..    # taxa 
+0000e120: 2d2d 2d2d 0a20 2020 2064 6566 2067 6574  ----.    def get
+0000e130: 5f74 6178 6128 7365 6c66 2c20 2a2a 6b77  _taxa(self, **kw
+0000e140: 6172 6773 292d 3e44 6174 6146 7261 6d65  args)->DataFrame
+0000e150: 3a0a 2020 2020 2020 2020 2222 220a 2020  :.        """.  
+0000e160: 2020 2020 2020 4765 7420 616c 6c20 7461        Get all ta
+0000e170: 7861 2077 6974 6869 6e20 4765 6d6d 610a  xa within Gemma.
+0000e180: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+0000e190: 203a 7061 7261 6d20 2a2a 6b77 6172 6773   :param **kwargs
+0000e1a0: 3a20 4164 6469 7469 6f6e 616c 2061 7267  : Additional arg
+0000e1b0: 756d 656e 7473 2074 6f20 7261 772e 6765  uments to raw.ge
+0000e1c0: 745f 7461 7861 0a20 2020 2020 2020 203a  t_taxa.        :
+0000e1d0: 7265 7475 726e 3a20 4120 4461 7461 4672  return: A DataFr
+0000e1e0: 616d 6520 696e 636c 7564 696e 6720 7468  ame including th
+0000e1f0: 6520 6e61 6d65 732c 2049 4473 2061 6e64  e names, IDs and
+0000e200: 2064 6174 6162 6173 6520 696e 666f 726d   database inform
+0000e210: 6174 696f 6e0a 2020 2020 2020 2020 2020  ation.          
+0000e220: 6162 6f75 7420 7468 6520 7461 786f 6e73  about the taxons
+0000e230: 0a20 2020 2020 2020 203a 7274 7970 653a  .        :rtype:
+0000e240: 2044 6174 6146 7261 6d65 0a0a 2020 2020   DataFrame..    
+0000e250: 2020 2020 2222 220a 2020 2020 2020 2020      """.        
+0000e260: 0a20 2020 2020 2020 2072 6573 706f 6e73  .        respons
+0000e270: 6520 3d20 2073 656c 662e 7261 772e 6765  e =  self.raw.ge
+0000e280: 745f 7461 7861 282a 2a6b 7761 7267 7329  t_taxa(**kwargs)
+0000e290: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
+0000e2a0: 2020 6466 203d 2070 732e 7072 6f63 6573    df = ps.proces
+0000e2b0: 735f 7461 786f 6e28 7265 7370 6f6e 7365  s_taxon(response
+0000e2c0: 2e64 6174 6129 0a20 2020 2020 2020 2072  .data).        r
+0000e2d0: 6574 7572 6e20 6466 5b64 662e 6973 6e75  eturn df[df.isnu
+0000e2e0: 6c6c 2829 2e74 6178 6f6e 5f6e 616d 6520  ll().taxon_name 
+0000e2f0: 213d 2054 7275 655d 0a20 2020 2020 2020  != True].       
+0000e300: 200a 2020 2020 2020 2020 0a20 2020 200a   .        .    .
+0000e310: 2020 2020 2320 7461 7861 2f7b 7461 7861      # taxa/{taxa
+0000e320: 7d2c 2067 6574 5f74 6178 615f 6279 5f69  }, get_taxa_by_i
+0000e330: 6473 202d 2d2d 2d2d 0a20 2020 2023 2069  ds -----.    # i
+0000e340: 6d70 6c65 6d65 6e74 6564 2c20 6861 7264  mplemented, hard
+0000e350: 6c79 206e 6565 6465 6420 7769 7468 2033  ly needed with 3
+0000e360: 2074 6178 6120 200a 0a0a 0a23 2042 656c   taxa  ....# Bel
+0000e370: 6f77 2061 7265 2022 436f 6e76 656e 6965  ow are "Convenie
+0000e380: 6e63 6522 2028 636f 6d62 696e 6174 696f  nce" (combinatio
+0000e390: 6e29 2066 756e 6374 696f 6e73 0a0a 2020  n) functions..  
+0000e3a0: 2020 2320 7365 745f 6765 6d6d 615f 7573    # set_gemma_us
+0000e3b0: 6572 2069 7320 6e6f 7420 6e65 6564 6564  er is not needed
+0000e3c0: 2073 696e 6365 2069 7427 7320 7772 6170   since it's wrap
+0000e3d0: 7065 6420 696e 2074 6865 2047 656d 6d61  ped in the Gemma
+0000e3e0: 5079 2063 6c61 7373 0a20 2020 2023 2067  Py class.    # g
+0000e3f0: 6574 5f70 6c61 7466 6f72 6d5f 616e 6e6f  et_platform_anno
+0000e400: 7461 7469 6f6e 7320 6973 2074 6865 2064  tations is the d
+0000e410: 6566 6175 6c74 2067 6574 5f70 6c61 7466  efault get_platf
+0000e420: 6f72 6d5f 616e 6e6f 7461 7469 6f6e 730a  orm_annotations.
+0000e430: 2020 2020 0a0a 2020 2020 6465 6620 6d61      ..    def ma
+0000e440: 6b65 5f64 6573 6967 6e28 7365 6c66 2c73  ke_design(self,s
+0000e450: 616d 706c 6573 3a44 6174 6146 7261 6d65  amples:DataFrame
+0000e460: 2c6d 6574 615f 7479 7065 3a73 7472 203d  ,meta_type:str =
+0000e470: 2027 7465 7874 2729 2d3e 4461 7461 4672   'text')->DataFr
+0000e480: 616d 653a 0a20 2020 2020 2020 2022 2222  ame:.        """
+0000e490: 0a20 2020 2020 2020 2055 7369 6e67 206f  .        Using o
+0000e4a0: 6e20 7468 6520 6f75 7470 7574 206f 6620  n the output of 
+0000e4b0: 6765 745f 6461 7461 7365 745f 7361 6d70  get_dataset_samp
+0000e4c0: 6c65 732c 2074 6869 7320 6675 6e63 7469  les, this functi
+0000e4d0: 6f6e 2063 7265 6174 6573 2061 0a20 2020  on creates a.   
+0000e4e0: 2020 2020 2073 696d 706c 6966 6965 6420       simplified 
+0000e4f0: 6465 7369 676e 2074 6162 6c65 2c20 6772  design table, gr
+0000e500: 616e 7469 6e67 206f 6e65 2063 6f6c 756d  anting one colum
+0000e510: 6e20 746f 2065 6163 6820 6578 7065 7269  n to each experi
+0000e520: 6d65 6e74 616c 2076 6172 6961 626c 650a  mental variable.
+0000e530: 0a0a 2020 2020 2020 2020 3a70 6172 616d  ..        :param
+0000e540: 2073 616d 706c 6573 3a20 416e 206f 7574   samples: An out
+0000e550: 7075 7420 6672 6f6d 2067 6574 5f64 6174  put from get_dat
+0000e560: 6173 6574 5f73 616d 706c 6573 2e0a 2020  aset_samples..  
+0000e570: 2020 2020 2020 3a74 7970 6520 7361 6d70        :type samp
+0000e580: 6c65 733a 2044 6174 6146 7261 6d65 0a20  les: DataFrame. 
+0000e590: 2020 2020 2020 203a 7061 7261 6d20 6d65         :param me
+0000e5a0: 7461 5f74 7970 653a 2054 7970 6520 6f66  ta_type: Type of
+0000e5b0: 206d 6574 6164 6174 6120 746f 2069 6e63   metadata to inc
+0000e5c0: 6c75 6465 2069 6e20 7468 6520 6f75 7470  lude in the outp
+0000e5d0: 7574 2e20 2274 6578 7422 2c0a 2020 2020  ut. "text",.    
+0000e5e0: 2020 2020 2020 2275 7269 2220 6f72 2022        "uri" or "
+0000e5f0: 626f 7468 222c 2064 6566 6175 6c74 7320  both", defaults 
+0000e600: 746f 2027 7465 7874 270a 2020 2020 2020  to 'text'.      
+0000e610: 2020 3a74 7970 6520 6d65 7461 5f74 7970    :type meta_typ
+0000e620: 653a 2073 7472 2c20 6f70 7469 6f6e 616c  e: str, optional
+0000e630: 0a20 2020 2020 2020 203a 7265 7475 726e  .        :return
+0000e640: 3a20 4120 4461 7461 4672 616d 6520 696e  : A DataFrame in
+0000e650: 636c 7564 696e 6720 7468 6520 6465 7369  cluding the desi
+0000e660: 676e 2074 6162 6c65 2066 6f72 2074 6865  gn table for the
+0000e670: 2064 6174 6173 6574 0a20 2020 2020 2020   dataset.       
+0000e680: 203a 7274 7970 653a 2044 6174 6146 7261   :rtype: DataFra
+0000e690: 6d65 0a0a 2020 2020 2020 2020 2222 220a  me..        """.
+0000e6a0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+0000e6b0: 2063 6174 6567 6f72 6965 7320 3d20 7064   categories = pd
+0000e6c0: 2e63 6f6e 6361 7428 5b78 5b5b 2266 6163  .concat([x[["fac
+0000e6d0: 746f 725f 4944 222c 2266 6163 746f 725f  tor_ID","factor_
+0000e6e0: 6361 7465 676f 7279 222c 2266 6163 746f  category","facto
+0000e6f0: 725f 6361 7465 676f 7279 5f55 5249 225d  r_category_URI"]
+0000e700: 5d20 0a20 2020 2020 2020 2020 2020 2020  ] .             
+0000e710: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000e720: 2020 2066 6f72 2078 2069 6e20 7361 6d70     for x in samp
+0000e730: 6c65 732e 7361 6d70 6c65 5f66 6163 746f  les.sample_facto
+0000e740: 725f 7661 6c75 6573 5d2c 0a20 2020 2020  r_values],.     
+0000e750: 2020 2020 2020 2020 2020 2020 2069 676e               ign
+0000e760: 6f72 655f 696e 6465 7820 3d20 5472 7565  ore_index = True
+0000e770: 292e 6472 6f70 5f64 7570 6c69 6361 7465  ).drop_duplicate
+0000e780: 7328 290a 2020 2020 2020 2020 0a20 2020  s().        .   
+0000e790: 2020 2020 200a 2020 2020 2020 2020 6465       .        de
+0000e7a0: 6620 6765 745f 7661 6c5f 7572 6928 7829  f get_val_uri(x)
+0000e7b0: 3a0a 2020 2020 2020 2020 2020 2020 7265  :.            re
+0000e7c0: 7475 726e 205b 222c 222e 6a6f 696e 285b  turn [",".join([
+0000e7d0: 7374 7228 7a29 2069 6620 7a20 6973 206e  str(z) if z is n
+0000e7e0: 6f74 204e 6f6e 6520 656c 7365 2022 2220  ot None else "" 
+0000e7f0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+0000e800: 2020 2020 2020 2020 2020 2020 2020 2066                 f
+0000e810: 6f72 207a 2069 6e20 795b 792e 6661 6374  or z in y[y.fact
+0000e820: 6f72 5f49 443d 3d78 5d2e 7661 6c75 655f  or_ID==x].value_
+0000e830: 5552 495d 2920 0a20 2020 2020 2020 2020  URI]) .         
+0000e840: 2020 2020 2020 2020 2020 2066 6f72 2079             for y
+0000e850: 2069 6e20 7361 6d70 6c65 732e 7361 6d70   in samples.samp
+0000e860: 6c65 5f66 6163 746f 725f 7661 6c75 6573  le_factor_values
+0000e870: 5d0a 2020 2020 2020 2020 0a20 2020 2020  ].        .     
+0000e880: 2020 2066 6163 746f 725f 5552 4973 203d     factor_URIs =
+0000e890: 205b 6765 745f 7661 6c5f 7572 6928 7829   [get_val_uri(x)
+0000e8a0: 2066 6f72 2078 2069 6e20 6361 7465 676f   for x in catego
+0000e8b0: 7269 6573 2e66 6163 746f 725f 4944 5d0a  ries.factor_ID].
+0000e8c0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+0000e8d0: 2064 6566 2067 6574 5f74 6578 7428 7829   def get_text(x)
+0000e8e0: 3a0a 2020 2020 2020 2020 2020 2020 6465  :.            de
+0000e8f0: 6620 6765 745f 7375 6d6d 6172 7928 7929  f get_summary(y)
+0000e900: 3a0a 2020 2020 2020 2020 2020 2020 2020  :.              
+0000e910: 2020 7265 7475 726e 2027 2c27 2e6a 6f69    return ','.joi
+0000e920: 6e28 5b7a 5b31 5d2e 7375 6d6d 6172 790a  n([z[1].summary.
+0000e930: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000e940: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000e950: 2069 6620 7a5b 315d 2e73 756d 6d61 7279   if z[1].summary
+0000e960: 2069 7320 6e6f 7420 4e6f 6e65 2065 6c73   is not None els
+0000e970: 6520 7a5b 315d 2e76 616c 7565 200a 2020  e z[1].value .  
+0000e980: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000e990: 2020 2020 2020 2020 2020 2020 2020 2066                 f
+0000e9a0: 6f72 207a 2069 6e20 795b 792e 6661 6374  or z in y[y.fact
+0000e9b0: 6f72 5f49 443d 3d78 5d2e 6974 6572 726f  or_ID==x].iterro
+0000e9c0: 7773 2829 5d29 0a20 2020 2020 2020 2020  ws()]).         
+0000e9d0: 2020 200a 2020 2020 2020 2020 2020 2020     .            
+0000e9e0: 7265 7475 726e 205b 6765 745f 7375 6d6d  return [get_summ
+0000e9f0: 6172 7928 7929 2066 6f72 2079 2069 6e20  ary(y) for y in 
+0000ea00: 7361 6d70 6c65 732e 7361 6d70 6c65 5f66  samples.sample_f
+0000ea10: 6163 746f 725f 7661 6c75 6573 5d0a 2020  actor_values].  
+0000ea20: 2020 2020 2020 2020 2020 0a20 2020 2020            .     
+0000ea30: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+0000ea40: 0a20 2020 2020 2020 2074 6578 7420 3d20  .        text = 
+0000ea50: 5b67 6574 5f74 6578 7428 7829 2066 6f72  [get_text(x) for
+0000ea60: 2078 2069 6e20 6361 7465 676f 7269 6573   x in categories
+0000ea70: 2e66 6163 746f 725f 4944 5d0a 2020 2020  .factor_ID].    
+0000ea80: 2020 2020 0a20 2020 2020 2020 200a 2020      .        .  
+0000ea90: 2020 2020 2020 6966 206d 6574 615f 7479        if meta_ty
+0000eaa0: 7065 203d 3d27 7465 7874 273a 0a20 2020  pe =='text':.   
+0000eab0: 2020 2020 2020 2020 2064 6573 6967 6e5f           design_
+0000eac0: 6672 616d 6520 3d20 7064 2e44 6174 6146  frame = pd.DataF
+0000ead0: 7261 6d65 287b 0a20 2020 2020 2020 2020  rame({.         
+0000eae0: 2020 2020 2020 2063 6174 6567 6f72 6965         categorie
+0000eaf0: 732e 6661 6374 6f72 5f63 6174 6567 6f72  s.factor_categor
+0000eb00: 795b 695d 3a74 6578 745b 695d 2066 6f72  y[i]:text[i] for
+0000eb10: 2069 2069 6e20 7261 6e67 6528 6c65 6e28   i in range(len(
+0000eb20: 7465 7874 2929 0a20 2020 2020 2020 2020  text)).         
+0000eb30: 2020 2020 2020 207d 290a 2020 2020 2020         }).      
+0000eb40: 2020 656c 6966 206d 6574 615f 7479 7065    elif meta_type
+0000eb50: 203d 3d20 2775 7269 273a 0a20 2020 2020   == 'uri':.     
+0000eb60: 2020 2020 2020 2064 6573 6967 6e5f 6672         design_fr
+0000eb70: 616d 6520 3d20 7064 2e44 6174 6146 7261  ame = pd.DataFra
+0000eb80: 6d65 287b 0a20 2020 2020 2020 2020 2020  me({.           
+0000eb90: 2020 2020 2063 6174 6567 6f72 6965 732e       categories.
+0000eba0: 6661 6374 6f72 5f63 6174 6567 6f72 795f  factor_category_
+0000ebb0: 5552 495b 695d 3a66 6163 746f 725f 5552  URI[i]:factor_UR
+0000ebc0: 4973 5b69 5d20 666f 7220 6920 696e 2072  Is[i] for i in r
+0000ebd0: 616e 6765 286c 656e 2866 6163 746f 725f  ange(len(factor_
+0000ebe0: 5552 4973 2929 0a20 2020 2020 2020 2020  URIs)).         
+0000ebf0: 2020 2020 2020 207d 290a 2020 2020 2020         }).      
+0000ec00: 2020 656c 6966 206d 6574 615f 7479 7065    elif meta_type
+0000ec10: 203d 3d27 626f 7468 273a 0a20 2020 2020   =='both':.     
+0000ec20: 2020 2020 2020 206d 6572 6765 645f 6e61         merged_na
+0000ec30: 6d65 203d 205b 5b22 7c22 2e6a 6f69 6e28  me = [["|".join(
+0000ec40: 5b63 6174 6567 6f72 6965 732e 6661 6374  [categories.fact
+0000ec50: 6f72 5f63 6174 6567 6f72 795b 695d 2c0a  or_category[i],.
+0000ec60: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000ec70: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000ec80: 2020 2020 2020 6361 7465 676f 7269 6573        categories
+0000ec90: 2e66 6163 746f 725f 6361 7465 676f 7279  .factor_category
+0000eca0: 5f55 5249 5b69 5d5d 295d 2066 6f72 2069  _URI[i]])] for i
+0000ecb0: 2069 6e20 7261 6e67 6528 6c65 6e28 7465   in range(len(te
+0000ecc0: 7874 2929 5d0a 2020 2020 2020 2020 2020  xt))].          
+0000ecd0: 2020 0a20 2020 2020 2020 2020 2020 206d    .            m
+0000ece0: 6572 6765 645f 636f 6c20 3d20 5b5b 227c  erged_col = [["|
+0000ecf0: 222e 6a6f 696e 285b 7465 7874 5b69 5d5b  ".join([text[i][
+0000ed00: 6a5d 2c66 6163 746f 725f 5552 4973 5b69  j],factor_URIs[i
+0000ed10: 5d5b 6a5d 5d29 200a 2020 2020 2020 2020  ][j]]) .        
+0000ed20: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000ed30: 2020 2066 6f72 206a 2069 6e20 7261 6e67     for j in rang
+0000ed40: 6528 6c65 6e28 7465 7874 5b69 5d29 295d  e(len(text[i]))]
+0000ed50: 2066 6f72 2069 2069 6e20 7261 6e67 6528   for i in range(
+0000ed60: 6c65 6e28 7465 7874 2929 5d0a 2020 2020  len(text))].    
+0000ed70: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+0000ed80: 2020 2020 200a 2020 2020 2020 2020 2020       .          
+0000ed90: 2020 6465 7369 676e 5f66 7261 6d65 203d    design_frame =
+0000eda0: 2070 642e 4461 7461 4672 616d 6528 7b0a   pd.DataFrame({.
+0000edb0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000edc0: 6d65 7267 6564 5f6e 616d 655b 695d 3a6d  merged_name[i]:m
+0000edd0: 6572 6765 645f 636f 6c5b 695d 0a20 2020  erged_col[i].   
+0000ede0: 2020 2020 2020 2020 2020 2020 2066 6f72               for
+0000edf0: 2069 2069 6e20 7261 6e67 6528 6c65 6e28   i in range(len(
+0000ee00: 7465 7874 2929 7d29 0a20 2020 2020 2020  text))}).       
+0000ee10: 2020 2020 200a 2020 2020 2020 2020 6465       .        de
+0000ee20: 7369 676e 5f66 7261 6d65 2e69 6e73 6572  sign_frame.inser
+0000ee30: 7428 6c6f 6320 3d20 302c 636f 6c75 6d6e  t(loc = 0,column
+0000ee40: 203d 2022 6661 6374 6f72 5f76 616c 7565   = "factor_value
+0000ee50: 7322 2c0a 2020 2020 2020 2020 2020 2020  s",.            
+0000ee60: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000ee70: 7661 6c75 6520 3d20 7361 6d70 6c65 732e  value = samples.
+0000ee80: 7361 6d70 6c65 5f66 6163 746f 725f 7661  sample_factor_va
+0000ee90: 6c75 6573 290a 2020 2020 2020 2020 6465  lues).        de
+0000eea0: 7369 676e 5f66 7261 6d65 2e69 6e64 6578  sign_frame.index
+0000eeb0: 203d 2073 616d 706c 6573 2e73 616d 706c   = samples.sampl
+0000eec0: 655f 6e61 6d65 0a20 2020 2020 2020 200a  e_name.        .
+0000eed0: 2020 2020 2020 2020 7265 7475 726e 2064          return d
+0000eee0: 6573 6967 6e5f 6672 616d 650a 2020 2020  esign_frame.    
+0000eef0: 2020 2020 0a20 2020 2020 2020 200a 2020      .        .  
+0000ef00: 2020 0a20 2020 2064 6566 205f 5f73 7562    .    def __sub
+0000ef10: 7365 745f 6661 6374 6f72 5f76 616c 7565  set_factor_value
+0000ef20: 7328 7365 6c66 2c0a 2020 2020 2020 2020  s(self,.        
+0000ef30: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000ef40: 2020 2020 2020 2066 6163 746f 725f 7661         factor_va
+0000ef50: 6c75 6573 2c0a 2020 2020 2020 2020 2020  lues,.          
+0000ef60: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000ef70: 2020 2020 2064 6966 6665 7265 6e74 6961       differentia
+0000ef80: 6c5f 6578 7072 6573 7369 6f6e 733a 7064  l_expressions:pd
+0000ef90: 2e44 6174 6146 7261 6d65 2c0a 2020 2020  .DataFrame,.    
+0000efa0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000efb0: 2020 2020 2020 2020 2020 2072 6573 756c             resul
+0000efc0: 745f 7365 742c 0a20 2020 2020 2020 2020  t_set,.         
+0000efd0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000efe0: 2020 2020 2020 636f 6e74 7261 7374 293a        contrast):
+0000eff0: 0a20 2020 2020 2020 206f 7574 203d 2073  .        out = s
+0000f000: 7562 2e72 6570 2854 7275 652c 6c65 6e28  ub.rep(True,len(
+0000f010: 6661 6374 6f72 5f76 616c 7565 7329 290a  factor_values)).
+0000f020: 2020 2020 2020 2020 6966 2064 6966 6665          if diffe
+0000f030: 7265 6e74 6961 6c5f 6578 7072 6573 7369  rential_expressi
+0000f040: 6f6e 7320 6973 206e 6f74 204e 6f6e 653a  ons is not None:
+0000f050: 0a20 2020 2020 2020 2020 2020 2073 7562  .            sub
+0000f060: 7365 7420 3d20 6469 6666 6572 656e 7469  set = differenti
+0000f070: 616c 5f65 7870 7265 7373 696f 6e73 5b64  al_expressions[d
+0000f080: 6966 6665 7265 6e74 6961 6c5f 6578 7072  ifferential_expr
+0000f090: 6573 7369 6f6e 732e 7265 7375 6c74 5f49  essions.result_I
+0000f0a0: 4420 3d3d 200a 2020 2020 2020 2020 2020  D == .          
+0000f0b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f0c0: 2020 2020 2020 2020 2020 2072 6573 756c             resul
+0000f0d0: 745f 7365 745d 2e73 7562 7365 745f 6661  t_set].subset_fa
+0000f0e0: 6374 6f72 2e64 726f 705f 6475 706c 6963  ctor.drop_duplic
+0000f0f0: 6174 6573 2829 0a20 2020 2020 2020 2020  ates().         
+0000f100: 2020 2023 2072 6573 756c 7420 7365 7420     # result set 
+0000f110: 7368 6f75 6c64 2068 6176 6520 7468 6520  should have the 
+0000f120: 7361 6d65 2073 7562 7365 7420 666f 7220  same subset for 
+0000f130: 616c 6c20 636f 6e74 7261 7374 730a 2020  all contrasts.  
+0000f140: 2020 2020 2020 2020 2020 6173 7365 7274            assert
+0000f150: 206c 656e 2873 7562 7365 7429 203d 3d20   len(subset) == 
+0000f160: 310a 2020 2020 2020 2020 2020 2020 6966  1.            if
+0000f170: 2073 7562 7365 745b 305d 2e73 6861 7065   subset[0].shape
+0000f180: 5b30 5d21 3d30 3a0a 2020 2020 2020 2020  [0]!=0:.        
+0000f190: 2020 2020 2020 2020 7375 6273 6574 5f69          subset_i
+0000f1a0: 6473 203d 2073 7562 7365 745b 305d 2e49  ds = subset[0].I
+0000f1b0: 440a 2020 2020 2020 2020 2020 2020 2020  D.              
+0000f1c0: 2020 0a20 2020 2020 2020 2020 2020 2020    .             
+0000f1d0: 2020 2069 6e5f 7375 6273 6574 203d 205b     in_subset = [
+0000f1e0: 616e 7928 7375 622e 6c69 7374 5f69 6e5f  any(sub.list_in_
+0000f1f0: 6c69 7374 2878 2e49 442c 2073 7562 7365  list(x.ID, subse
+0000f200: 745f 6964 7329 2920 666f 7220 7820 696e  t_ids)) for x in
+0000f210: 2066 6163 746f 725f 7661 6c75 6573 5d0a   factor_values].
+0000f220: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f230: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+0000f240: 206f 7574 203d 206f 7574 2061 6e64 2069   out = out and i
+0000f250: 6e5f 7375 6273 6574 0a20 2020 2020 2020  n_subset.       
+0000f260: 2020 2020 200a 2020 2020 2020 2020 2020       .          
+0000f270: 2020 6966 2063 6f6e 7472 6173 7420 6973    if contrast is
+0000f280: 206e 6f74 204e 6f6e 653a 0a20 2020 2020   not None:.     
+0000f290: 2020 2020 2020 2020 2020 2063 6e20 3d20             cn = 
+0000f2a0: 6469 6666 6572 656e 7469 616c 5f65 7870  differential_exp
+0000f2b0: 7265 7373 696f 6e73 5b0a 2020 2020 2020  ressions[.      
+0000f2c0: 2020 2020 2020 2020 2020 2020 2020 6c69                li
+0000f2d0: 7374 2864 6966 6665 7265 6e74 6961 6c5f  st(differential_
+0000f2e0: 6578 7072 6573 7369 6f6e 732e 7265 7375  expressions.resu
+0000f2f0: 6c74 5f49 4420 3d3d 2072 6573 756c 745f  lt_ID == result_
+0000f300: 7365 7429 2061 6e64 200a 2020 2020 2020  set) and .      
+0000f310: 2020 2020 2020 2020 2020 2020 2020 6c69                li
+0000f320: 7374 2864 6966 6665 7265 6e74 6961 6c5f  st(differential_
+0000f330: 6578 7072 6573 7369 6f6e 732e 636f 6e74  expressions.cont
+0000f340: 7261 7374 5f49 4420 3d3d 2073 7472 2863  rast_ID == str(c
+0000f350: 6f6e 7472 6173 7429 295d 0a20 2020 2020  ontrast))].     
+0000f360: 2020 2020 2020 2020 2020 200a 2020 2020             .    
+0000f370: 2020 2020 2020 2020 2020 2020 6261 7365              base
+0000f380: 6c69 6e65 5f69 6420 3d20 6c69 7374 2873  line_id = list(s
+0000f390: 7562 2e75 6e69 7175 6528 7375 622e 6272  ub.unique(sub.br
+0000f3a0: 6561 6b5f 6c69 7374 285b 6c69 7374 2878  eak_list([list(x
+0000f3b0: 2e49 4429 2066 6f72 2078 2069 6e20 636e  .ID) for x in cn
+0000f3c0: 2e62 6173 656c 696e 655f 6661 6374 6f72  .baseline_factor
+0000f3d0: 735d 2929 290a 2020 2020 2020 2020 2020  s]))).          
+0000f3e0: 2020 2020 2020 6261 7365 6c69 6e65 5f66        baseline_f
+0000f3f0: 6163 746f 725f 6964 203d 206c 6973 7428  actor_id = list(
+0000f400: 7375 622e 756e 6971 7565 2873 7562 2e62  sub.unique(sub.b
+0000f410: 7265 616b 5f6c 6973 7428 5b6c 6973 7428  reak_list([list(
+0000f420: 782e 6661 6374 6f72 5f49 4429 2066 6f72  x.factor_ID) for
+0000f430: 2078 2069 6e20 636e 2e62 6173 656c 696e   x in cn.baselin
+0000f440: 655f 6661 6374 6f72 735d 2929 290a 2020  e_factors]))).  
+0000f450: 2020 2020 2020 2020 2020 2020 2020 0a20                . 
+0000f460: 2020 2020 2020 2020 2020 2020 2020 2063                 c
+0000f470: 6f6e 7472 6173 745f 6964 203d 2020 6c69  ontrast_id =  li
+0000f480: 7374 2873 7562 2e75 6e69 7175 6528 7375  st(sub.unique(su
+0000f490: 622e 6272 6561 6b5f 6c69 7374 285b 6c69  b.break_list([li
+0000f4a0: 7374 2878 2e49 4429 2066 6f72 2078 2069  st(x.ID) for x i
+0000f4b0: 6e20 636e 2e65 7870 6572 696d 656e 7461  n cn.experimenta
+0000f4c0: 6c5f 6661 6374 6f72 735d 2929 290a 2020  l_factors]))).  
+0000f4d0: 2020 2020 2020 2020 2020 2020 2020 636f                co
+0000f4e0: 6e74 7261 7374 5f66 6163 746f 725f 6964  ntrast_factor_id
+0000f4f0: 203d 2020 6c69 7374 2873 7562 2e75 6e69   =  list(sub.uni
+0000f500: 7175 6528 7375 622e 6272 6561 6b5f 6c69  que(sub.break_li
+0000f510: 7374 285b 6c69 7374 2878 2e66 6163 746f  st([list(x.facto
+0000f520: 725f 4944 2920 666f 7220 7820 696e 2063  r_ID) for x in c
+0000f530: 6e2e 6578 7065 7269 6d65 6e74 616c 5f66  n.experimental_f
+0000f540: 6163 746f 7273 5d29 2929 0a20 2020 2020  actors]))).     
+0000f550: 2020 2020 2020 2020 2020 200a 2020 2020             .    
+0000f560: 2020 2020 2020 2020 2020 2020 636f 6e74              cont
+0000f570: 7261 7374 5f69 6420 3d20 7375 622e 6d61  rast_id = sub.ma
+0000f580: 7463 685f 6279 2863 6f6e 7472 6173 745f  tch_by(contrast_
+0000f590: 6964 2c62 6173 656c 696e 655f 6661 6374  id,baseline_fact
+0000f5a0: 6f72 5f69 642c 2063 6f6e 7472 6173 745f  or_id, contrast_
+0000f5b0: 6661 6374 6f72 5f69 6429 0a20 2020 2020  factor_id).     
+0000f5c0: 2020 2020 2020 2020 2020 200a 2020 2020             .    
+0000f5d0: 2020 2020 2020 2020 2020 2020 6465 6620              def 
+0000f5e0: 696e 5f63 6f6e 2866 6163 746f 725f 7661  in_con(factor_va
+0000f5f0: 6c75 6529 3a0a 2020 2020 2020 2020 2020  lue):.          
+0000f600: 2020 2020 2020 2020 2020 636f 6e64 3120            cond1 
+0000f610: 3d20 616c 6c28 7375 622e 6c69 7374 5f69  = all(sub.list_i
+0000f620: 6e5f 6c69 7374 2863 6f6e 7472 6173 745f  n_list(contrast_
+0000f630: 6964 2c20 6661 6374 6f72 5f76 616c 7565  id, factor_value
+0000f640: 2e49 4429 2920 6f72 205c 0a20 2020 2020  .ID)) or \.     
+0000f650: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f660: 2020 2020 2020 2061 6c6c 2873 7562 2e6c         all(sub.l
+0000f670: 6973 745f 696e 5f6c 6973 7428 6261 7365  ist_in_list(base
+0000f680: 6c69 6e65 5f69 642c 6661 6374 6f72 5f76  line_id,factor_v
+0000f690: 616c 7565 2e49 4429 2920 2020 2020 2020  alue.ID))       
+0000f6a0: 2020 2020 2020 2020 2020 2020 0a20 2020              .   
+0000f6b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f6c0: 200a 2020 2020 2020 2020 2020 2020 2020   .              
+0000f6d0: 2020 2020 2020 6966 206c 656e 2863 6f6e        if len(con
+0000f6e0: 7472 6173 745f 6964 293d 3d32 3a0a 2020  trast_id)==2:.  
+0000f6f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f700: 2020 2020 2020 636f 6e64 3220 3d20 2863        cond2 = (c
+0000f710: 6f6e 7472 6173 745f 6964 5b30 5d20 696e  ontrast_id[0] in
+0000f720: 2066 6163 746f 725f 7661 6c75 652e 4944   factor_value.ID
+0000f730: 2061 6e64 205c 0a20 2020 2020 2020 2020   and \.         
+0000f740: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f750: 2020 2020 2020 2020 2020 2062 6173 656c             basel
+0000f760: 696e 655f 6964 5b31 5d20 696e 2066 6163  ine_id[1] in fac
+0000f770: 746f 725f 7661 6c75 652e 4944 2920 6f72  tor_value.ID) or
+0000f780: 205c 0a20 2020 2020 2020 2020 2020 2020   \.             
+0000f790: 2020 2020 2020 2020 2020 2020 2020 2028                 (
+0000f7a0: 636f 6e74 7261 7374 5f69 645b 315d 2069  contrast_id[1] i
+0000f7b0: 6e20 6661 6374 6f72 5f76 616c 7565 2e49  n factor_value.I
+0000f7c0: 4420 616e 6420 5c0a 2020 2020 2020 2020  D and \.        
+0000f7d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f7e0: 2020 2020 2020 2020 2020 2020 2020 2062                 b
+0000f7f0: 6173 656c 696e 655f 6964 5b30 5d20 696e  aseline_id[0] in
+0000f800: 2066 6163 746f 725f 7661 6c75 652e 4944   factor_value.ID
+0000f810: 290a 2020 2020 2020 2020 2020 2020 2020  ).              
+0000f820: 2020 2020 2020 2020 2020 0a20 2020 2020            .     
+0000f830: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f840: 2020 2063 6f6e 6431 203d 2063 6f6e 6431     cond1 = cond1
+0000f850: 206f 7220 636f 6e64 320a 2020 2020 2020   or cond2.      
+0000f860: 2020 2020 2020 2020 2020 2020 2020 0a20                . 
 0000f870: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f880: 696e 5f63 6f6e 7472 6173 7420 3d20 5b69  in_contrast = [i
-0000f890: 6e5f 636f 6e28 7829 2066 6f72 2078 2069  n_con(x) for x i
-0000f8a0: 6e20 6661 6374 6f72 5f76 616c 7565 735d  n factor_values]
-0000f8b0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-0000f8c0: 200a 2020 2020 2020 2020 2020 2020 2020   .              
-0000f8d0: 2020 6f75 7420 3d20 6f75 7420 616e 6420    out = out and 
-0000f8e0: 696e 5f63 6f6e 7472 6173 740a 2020 2020  in_contrast.    
-0000f8f0: 2020 2020 0a20 2020 2020 2020 2072 6574      .        ret
-0000f900: 7572 6e20 6f75 740a 2020 2020 2020 2020  urn out.        
-0000f910: 0a0a 0a20 2020 2064 6566 2067 6574 5f64  ...    def get_d
-0000f920: 6174 6173 6574 5f6f 626a 6563 7428 7365  ataset_object(se
-0000f930: 6c66 2c20 6461 7461 7365 7473 3a4c 6973  lf, datasets:Lis
-0000f940: 745b 7374 727c 696e 745d 2c0a 2020 2020  t[str|int],.    
-0000f950: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f960: 2020 2020 2020 2067 656e 6573 3a4f 7074         genes:Opt
-0000f970: 696f 6e61 6c5b 4c69 7374 5b73 7472 7c69  ional[List[str|i
-0000f980: 6e74 5d5d 203d 204e 6f6e 652c 0a20 2020  nt]] = None,.   
+0000f880: 2020 2072 6574 7572 6e20 636f 6e64 310a     return cond1.
+0000f890: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f8a0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+0000f8b0: 2020 2020 2020 2020 2069 6e5f 636f 6e74           in_cont
+0000f8c0: 7261 7374 203d 205b 696e 5f63 6f6e 2878  rast = [in_con(x
+0000f8d0: 2920 666f 7220 7820 696e 2066 6163 746f  ) for x in facto
+0000f8e0: 725f 7661 6c75 6573 5d0a 2020 2020 2020  r_values].      
+0000f8f0: 2020 2020 2020 2020 2020 0a20 2020 2020            .     
+0000f900: 2020 2020 2020 2020 2020 206f 7574 203d             out =
+0000f910: 206f 7574 2061 6e64 2069 6e5f 636f 6e74   out and in_cont
+0000f920: 7261 7374 0a20 2020 2020 2020 200a 2020  rast.        .  
+0000f930: 2020 2020 2020 7265 7475 726e 206f 7574        return out
+0000f940: 0a20 2020 2020 2020 200a 0a0a 2020 2020  .        ...    
+0000f950: 6465 6620 6765 745f 6461 7461 7365 745f  def get_dataset_
+0000f960: 6f62 6a65 6374 2873 656c 662c 2064 6174  object(self, dat
+0000f970: 6173 6574 733a 4c69 7374 5b73 7472 7c69  asets:List[str|i
+0000f980: 6e74 5d2c 0a20 2020 2020 2020 2020 2020  nt],.           
 0000f990: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000f9a0: 2020 2020 2020 2020 6b65 6570 5f6e 6f6e          keep_non
-0000f9b0: 5f73 7065 6369 6669 6320 3d20 4661 6c73  _specific = Fals
-0000f9c0: 652c 0a20 2020 2020 2020 2020 2020 2020  e,.             
-0000f9d0: 2020 2020 2020 2020 2020 2020 2020 636f                co
-0000f9e0: 6e73 6f6c 6964 6174 653a 4f70 7469 6f6e  nsolidate:Option
-0000f9f0: 616c 5b73 7472 5d20 3d20 4e6f 6e65 2c0a  al[str] = None,.
+0000f9a0: 6765 6e65 733a 4f70 7469 6f6e 616c 5b4c  genes:Optional[L
+0000f9b0: 6973 745b 7374 727c 696e 745d 5d20 3d20  ist[str|int]] = 
+0000f9c0: 4e6f 6e65 2c0a 2020 2020 2020 2020 2020  None,.          
+0000f9d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000f9e0: 206b 6565 705f 6e6f 6e5f 7370 6563 6966   keep_non_specif
+0000f9f0: 6963 203d 2046 616c 7365 2c0a 2020 2020  ic = False,.    
 0000fa00: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000fa10: 2020 2020 2020 2020 2020 2072 6573 756c             resul
-0000fa20: 745f 7365 7473 3a4f 7074 696f 6e61 6c5b  t_sets:Optional[
-0000fa30: 4c69 7374 5b69 6e74 5d5d 203d 204e 6f6e  List[int]] = Non
-0000fa40: 652c 0a20 2020 2020 2020 2020 2020 2020  e,.             
-0000fa50: 2020 2020 2020 2020 2020 2020 2020 636f                co
-0000fa60: 6e74 7261 7374 733a 4f70 7469 6f6e 616c  ntrasts:Optional
-0000fa70: 5b4c 6973 745b 7374 725d 5d20 3d20 4e6f  [List[str]] = No
-0000fa80: 6e65 2c0a 2020 2020 2020 2020 2020 2020  ne,.            
-0000fa90: 2020 2020 2020 2020 2020 2020 2020 206d                 m
-0000faa0: 6574 615f 7479 7065 3a73 7472 203d 2027  eta_type:str = '
-0000fab0: 7465 7874 272c 0a20 2020 2020 2020 2020  text',.         
+0000fa10: 2020 2020 2020 2063 6f6e 736f 6c69 6461         consolida
+0000fa20: 7465 3a4f 7074 696f 6e61 6c5b 7374 725d  te:Optional[str]
+0000fa30: 203d 204e 6f6e 652c 0a20 2020 2020 2020   = None,.       
+0000fa40: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000fa50: 2020 2020 7265 7375 6c74 5f73 6574 733a      result_sets:
+0000fa60: 4f70 7469 6f6e 616c 5b4c 6973 745b 696e  Optional[List[in
+0000fa70: 745d 5d20 3d20 4e6f 6e65 2c0a 2020 2020  t]] = None,.    
+0000fa80: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000fa90: 2020 2020 2020 2063 6f6e 7472 6173 7473         contrasts
+0000faa0: 3a4f 7074 696f 6e61 6c5b 4c69 7374 5b73  :Optional[List[s
+0000fab0: 7472 5d5d 203d 204e 6f6e 652c 0a20 2020  tr]] = None,.   
 0000fac0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000fad0: 2020 6f75 7470 7574 5f74 7970 653a 7374    output_type:st
-0000fae0: 7220 3d20 2761 6e6e 6461 7461 272c 0a20  r = 'anndata',. 
+0000fad0: 2020 2020 2020 2020 6d65 7461 5f74 7970          meta_typ
+0000fae0: 653a 7374 7220 3d20 2774 6578 7427 2c0a  e:str = 'text',.
 0000faf0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-0000fb00: 2020 2020 2020 2020 2020 2a2a 6b77 6172            **kwar
-0000fb10: 6773 292d 3e64 6963 745b 696e 743a 6469  gs)->dict[int:di
-0000fb20: 6374 7c41 6e6e 4461 7461 5d3a 0a20 2020  ct|AnnData]:.   
-0000fb30: 2020 2020 200a 2020 2020 2020 2020 2222       .        ""
-0000fb40: 2252 6574 7572 6e20 6120 6461 7461 2073  "Return a data s
-0000fb50: 7472 7563 7475 7265 2069 6e63 6c75 6469  tructure includi
-0000fb60: 6e67 2061 6c6c 2072 656c 6576 616e 7420  ng all relevant 
-0000fb70: 6461 7461 2072 656c 6174 6564 2074 6f0a  data related to.
-0000fb80: 2020 2020 2020 2020 6765 6e65 2065 7870          gene exp
-0000fb90: 7265 7373 696f 6e20 696e 2061 2064 6174  ression in a dat
-0000fba0: 6173 6574 2e20 4569 7468 6572 2072 6574  aset. Either ret
-0000fbb0: 7572 6e73 2061 6e20 616e 6e64 6174 6120  urns an anndata 
-0000fbc0: 6f62 6a65 6374 206f 7220 0a20 2020 2020  object or .     
-0000fbd0: 2020 2061 2064 6963 7469 6f6e 6172 7920     a dictionary 
-0000fbe0: 7769 7468 2061 6c6c 2074 6865 206e 6565  with all the nee
-0000fbf0: 6465 6420 6669 656c 6473 2e0a 2020 2020  ded fields..    
-0000fc00: 2020 2020 0a20 2020 2020 2020 200a 2020      .        .  
-0000fc10: 2020 2020 2020 3a70 6172 616d 2064 6174        :param dat
-0000fc20: 6173 6574 733a 204e 756d 6572 6963 616c  asets: Numerical
-0000fc30: 2064 6174 6173 6574 2069 6465 6e74 6966   dataset identif
-0000fc40: 6965 720a 2020 2020 2020 2020 2020 6461  ier.          da
-0000fc50: 7461 7365 7420 7368 6f72 7420 6e61 6d65  taset short name
-0000fc60: 730a 2020 2020 2020 2020 3a74 7970 6520  s.        :type 
-0000fc70: 6461 7461 7365 7473 3a20 4c69 7374 5b73  datasets: List[s
-0000fc80: 7472 7c69 6e74 5d0a 2020 2020 2020 2020  tr|int].        
-0000fc90: 3a70 6172 616d 2067 656e 6573 3a20 416e  :param genes: An
-0000fca0: 206e 6362 6920 6765 6e65 2069 6465 6e74   ncbi gene ident
-0000fcb0: 6966 6965 7220 616e 2c20 0a20 2020 2020  ifier an, .     
-0000fcc0: 2020 2020 2065 6e73 656d 626c 2067 656e       ensembl gen
-0000fcd0: 6520 6964 656e 7469 6669 6572 2077 6869  e identifier whi
-0000fce0: 6368 2074 7970 6963 616c 6c79 2073 7461  ch typically sta
-0000fcf0: 7274 7320 7769 7468 2065 6e73 6720 6f72  rts with ensg or
-0000fd00: 2061 6e20 0a20 2020 2020 2020 2020 206f   an .          o
-0000fd10: 6666 6963 6961 6c20 6765 6e65 2073 796d  fficial gene sym
-0000fd20: 626f 6c20 6170 7072 6f76 6564 2062 7920  bol approved by 
-0000fd30: 6867 6e63 2c20 6465 6661 756c 7473 2074  hgnc, defaults t
-0000fd40: 6f20 4e6f 6e65 0a20 2020 2020 2020 203a  o None.        :
-0000fd50: 7479 7065 2067 656e 6573 3a20 4f70 7469  type genes: Opti
-0000fd60: 6f6e 616c 5b4c 6973 745b 7374 727c 696e  onal[List[str|in
-0000fd70: 745d 5d2c 206f 7074 696f 6e61 6c0a 2020  t]], optional.  
-0000fd80: 2020 2020 2020 3a70 6172 616d 206b 6565        :param kee
-0000fd90: 705f 6e6f 6e5f 7370 6563 6966 6963 3a20  p_non_specific: 
-0000fda0: 4966 2054 7275 652c 2072 6573 756c 7473  If True, results
-0000fdb0: 2066 726f 6d20 0a20 2020 2020 2020 2020   from .         
-0000fdc0: 2070 726f 6265 7365 7473 2074 6861 7420   probesets that 
-0000fdd0: 6172 6520 6e6f 7420 7370 6563 6966 6963  are not specific
-0000fde0: 2074 6f20 7468 6520 6765 6e65 2077 696c   to the gene wil
-0000fdf0: 6c20 616c 736f 2062 6520 7265 7475 726e  l also be return
-0000fe00: 6564 2c0a 2020 2020 2020 2020 2020 6465  ed,.          de
-0000fe10: 6661 756c 7473 2074 6f20 4661 6c73 650a  faults to False.
-0000fe20: 2020 2020 2020 2020 3a74 7970 6520 6b65          :type ke
-0000fe30: 6570 5f6e 6f6e 5f73 7065 6369 6669 633a  ep_non_specific:
-0000fe40: 2054 5950 452c 206f 7074 696f 6e61 6c0a   TYPE, optional.
-0000fe50: 2020 2020 2020 2020 3a70 6172 616d 2063          :param c
-0000fe60: 6f6e 736f 6c69 6461 7465 3a20 4445 5343  onsolidate: DESC
-0000fe70: 5249 5054 494f 4e2c 2041 6e20 6f70 7469  RIPTION, An opti
-0000fe80: 6f6e 2066 6f72 2067 656e 6520 6578 7072  on for gene expr
-0000fe90: 6573 7369 6f6e 206c 6576 656c 2063 6f6e  ession level con
-0000fea0: 736f 6c69 6461 7469 6f6e 2e20 0a20 2020  solidation. .   
-0000feb0: 2020 2020 2020 2049 6620 656d 7074 792c         If empty,
-0000fec0: 2077 696c 6c20 7265 7475 726e 2065 7665   will return eve
-0000fed0: 7279 2070 726f 6265 2066 6f72 2074 6865  ry probe for the
-0000fee0: 2067 656e 6573 2e20 2270 6963 6b6d 6178   genes. "pickmax
-0000fef0: 2220 746f 2070 6963 6b20 0a20 2020 2020  " to pick .     
-0000ff00: 2020 2020 2074 6865 2070 726f 6265 2077       the probe w
-0000ff10: 6974 6820 7468 6520 6869 6768 6573 7420  ith the highest 
-0000ff20: 6578 7072 6573 7369 6f6e 2c20 2270 6963  expression, "pic
-0000ff30: 6b76 6172 2220 746f 2070 6963 6b20 7468  kvar" to pick th
-0000ff40: 6520 7072 6f76 6520 0a20 2020 2020 2020  e prove .       
-0000ff50: 2020 2077 6974 6820 7468 6520 6869 6768     with the high
-0000ff60: 6573 7420 7661 7269 616e 6365 2061 6e64  est variance and
-0000ff70: 2022 6176 6572 6167 6522 2066 6f72 2072   "average" for r
-0000ff80: 6574 7572 6e69 6e67 2074 6865 2061 7665  eturning the ave
-0000ff90: 7261 6765 200a 2020 2020 2020 2020 2020  rage .          
-0000ffa0: 6578 7072 6573 7369 6f6e 2074 6f20 4e6f  expression to No
-0000ffb0: 6e65 0a20 2020 2020 2020 203a 7479 7065  ne.        :type
-0000ffc0: 2063 6f6e 736f 6c69 6461 7465 3a20 4f70   consolidate: Op
-0000ffd0: 7469 6f6e 616c 5b73 7472 5d2c 206f 7074  tional[str], opt
-0000ffe0: 696f 6e61 6c0a 2020 2020 2020 2020 3a70  ional.        :p
-0000fff0: 6172 616d 2072 6573 756c 745f 7365 7473  aram result_sets
-00010000: 3a20 5265 7375 6c74 2073 6574 2049 4473  : Result set IDs
-00010010: 206f 6620 7468 6520 6120 0a20 2020 2020   of the a .     
-00010020: 2020 2020 2064 6966 6665 7265 6e74 6961       differentia
-00010030: 6c20 6578 7072 6573 7369 6f6e 2061 6e61  l expression ana
-00010040: 6c79 7369 732e 2049 6620 7072 6f76 6964  lysis. If provid
-00010050: 6564 2c20 7468 6520 6f75 7470 7574 2077  ed, the output w
-00010060: 696c 6c20 6f6e 6c79 200a 2020 2020 2020  ill only .      
-00010070: 2020 2020 696e 636c 7564 6520 7468 6520      include the 
-00010080: 7361 6d70 6c65 7320 6672 6f6d 2074 6865  samples from the
-00010090: 2073 7562 7365 7420 7573 6564 2069 6e20   subset used in 
-000100a0: 7468 6520 7265 7375 6c74 2073 6574 2049  the result set I
-000100b0: 442e 204d 7573 7420 0a20 2020 2020 2020  D. Must .       
-000100c0: 2020 2062 6520 7468 6520 7361 6d65 206c     be the same l
-000100d0: 656e 6774 6820 6173 2064 6174 6173 6574  ength as dataset
-000100e0: 732c 2064 6566 6175 6c74 7320 746f 204e  s, defaults to N
-000100f0: 6f6e 650a 2020 2020 2020 2020 3a74 7970  one.        :typ
-00010100: 6520 7265 7375 6c74 5f73 6574 733a 204f  e result_sets: O
-00010110: 7074 696f 6e61 6c5b 4c69 7374 5b69 6e74  ptional[List[int
-00010120: 5d5d 2c20 6f70 7469 6f6e 616c 0a20 2020  ]], optional.   
-00010130: 2020 2020 203a 7061 7261 6d20 636f 6e74       :param cont
-00010140: 7261 7374 733a 2043 6f6e 7472 6173 7420  rasts: Contrast 
-00010150: 4944 7320 6f66 2061 2064 6966 6665 7265  IDs of a differe
-00010160: 6e74 6961 6c20 0a20 2020 2020 2020 2020  ntial .         
-00010170: 2065 7870 7265 7373 696f 6e20 636f 6e74   expression cont
-00010180: 7261 7374 2e20 4e65 6564 2072 6573 756c  rast. Need resul
-00010190: 745f 7365 7473 2074 6f20 6265 2064 6566  t_sets to be def
-000101a0: 696e 6564 2074 6f20 776f 726b 2e20 4966  ined to work. If
-000101b0: 200a 2020 2020 2020 2020 2020 7072 6f76   .          prov
-000101c0: 6964 6564 2c20 7468 6520 6f75 7470 7574  ided, the output
-000101d0: 2077 696c 6c20 6f6e 6c79 2069 6e63 6c75   will only inclu
-000101e0: 6465 2073 616d 706c 6573 2072 656c 6576  de samples relev
-000101f0: 616e 7420 746f 2074 6865 2027 0a20 2020  ant to the '.   
-00010200: 2020 2020 2020 2073 7065 6369 6669 6320         specific 
-00010210: 636f 6e74 7261 7473 2e20 4d75 7374 2062  contrats. Must b
-00010220: 6520 7468 6520 7361 6d65 206c 656e 6774  e the same lengt
-00010230: 6820 6173 2064 6174 6173 6574 732e 0a20  h as datasets.. 
-00010240: 2020 2020 2020 203a 7061 7261 6d20 7374         :param st
-00010250: 7220 6d65 7461 5f74 7970 653a 2048 6f77  r meta_type: How
-00010260: 2073 686f 756c 6420 7468 6520 6d65 7461   should the meta
-00010270: 6461 7461 2069 6e66 6f72 6d61 7469 6f6e  data information
-00010280: 2073 686f 756c 6420 6265 200a 2020 2020   should be .    
-00010290: 2020 2020 2020 696e 636c 7564 6564 2e20        included. 
-000102a0: 4361 6e20 6265 2022 7465 7874 222c 2022  Can be "text", "
-000102b0: 7572 6922 206f 7220 2262 6f74 6822 2e20  uri" or "both". 
-000102c0: 2274 6578 7422 2061 6e64 2022 7572 6922  "text" and "uri"
-000102d0: 206f 7074 696f 6e73 2c20 6465 6661 756c   options, defaul
-000102e0: 7473 2074 6f20 4e6f 6e65 0a20 2020 2020  ts to None.     
-000102f0: 2020 203a 7479 7065 2063 6f6e 7472 6173     :type contras
-00010300: 7473 3a20 4f70 7469 6f6e 616c 5b4c 6973  ts: Optional[Lis
-00010310: 745b 7374 725d 5d2c 206f 7074 696f 6e61  t[str]], optiona
-00010320: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
-00010330: 206d 6574 615f 7479 7065 3a20 486f 7720   meta_type: How 
-00010340: 7368 6f75 6c64 2074 6865 206d 6574 6164  should the metad
-00010350: 6174 6120 696e 666f 726d 6174 696f 6e20  ata information 
-00010360: 7368 6f75 6c64 2062 6520 0a20 2020 2020  should be .     
-00010370: 2020 2020 2069 6e63 6c75 6465 642e 2043       included. C
-00010380: 616e 2062 6520 2274 6578 7422 2c20 2275  an be "text", "u
-00010390: 7269 2220 6f72 2022 626f 7468 222e 2022  ri" or "both". "
-000103a0: 7465 7874 2220 616e 6420 2275 7269 2220  text" and "uri" 
-000103b0: 6f70 7469 6f6e 732c 2064 6566 6175 6c74  options, default
-000103c0: 7320 746f 2027 7465 7874 270a 2020 2020  s to 'text'.    
-000103d0: 2020 2020 3a74 7970 6520 6d65 7461 5f74      :type meta_t
-000103e0: 7970 653a 2073 7472 2c20 6f70 7469 6f6e  ype: str, option
-000103f0: 616c 0a20 2020 2020 2020 203a 7061 7261  al.        :para
-00010400: 6d20 6f75 7470 7574 5f74 7970 653a 2054  m output_type: T
-00010410: 7970 6520 6f66 2074 6865 2072 6574 7572  ype of the retur
-00010420: 6e65 6420 6f62 6a65 6374 2e20 2261 6e6e  ned object. "ann
-00010430: 6461 7461 2220 666f 7220 616e 200a 2020  data" for an .  
-00010440: 2020 2020 2020 2020 416e 6e44 6174 6120          AnnData 
-00010450: 6f62 6a65 6374 2061 6e64 2022 6469 6374  object and "dict
-00010460: 2220 666f 7220 6120 6469 6374 696f 6e61  " for a dictiona
-00010470: 7279 2070 6f70 756c 6174 6564 2077 6974  ry populated wit
-00010480: 6820 4461 7461 4672 616d 6573 2c0a 2020  h DataFrames,.  
-00010490: 2020 2020 2020 2020 6465 6661 756c 7473          defaults
-000104a0: 2074 6f20 2761 6e6e 6461 7461 270a 2020   to 'anndata'.  
-000104b0: 2020 2020 2020 3a74 7970 6520 6f75 7470        :type outp
-000104c0: 7574 5f74 7970 653a 2073 7472 2c20 6f70  ut_type: str, op
-000104d0: 7469 6f6e 616c 0a20 2020 2020 2020 203a  tional.        :
-000104e0: 7061 7261 6d20 2a2a 6b77 6172 6773 3a20  param **kwargs: 
-000104f0: 4445 5343 5249 5054 494f 4e0a 2020 2020  DESCRIPTION.    
-00010500: 2020 2020 3a74 7970 6520 2a2a 6b77 6172      :type **kwar
-00010510: 6773 3a20 5459 5045 0a20 2020 2020 2020  gs: TYPE.       
-00010520: 203a 7261 6973 6573 2056 616c 7565 4572   :raises ValueEr
-00010530: 726f 723a 2044 4553 4352 4950 5449 4f4e  ror: DESCRIPTION
-00010540: 0a20 2020 2020 2020 203a 7265 7475 726e  .        :return
-00010550: 3a20 4120 6469 6374 696f 6e61 7279 2063  : A dictionary c
-00010560: 6f6e 7461 696e 696e 6720 416e 6e44 6174  ontaining AnnDat
-00010570: 6120 6f62 6a65 6374 7320 6f72 206e 6573  a objects or nes
-00010580: 7465 6420 6469 6374 696f 6e61 7269 6573  ted dictionaries
-00010590: 0a20 2020 2020 2020 2020 2074 6861 7420  .          that 
-000105a0: 636f 6e74 6169 6e20 6578 7072 6573 7369  contain expressi
-000105b0: 6f6e 2061 6e64 2073 616d 706c 6520 6d65  on and sample me
-000105c0: 7461 6461 206f 6620 7468 6520 7265 7175  tada of the requ
-000105d0: 6573 7465 6420 6578 7065 7269 6d65 6e74  ested experiment
-000105e0: 730a 2020 2020 2020 2020 3a72 7479 7065  s.        :rtype
-000105f0: 3a20 6469 6374 5b69 6e74 3a64 6963 747c  : dict[int:dict|
-00010600: 416e 6e44 6174 615d 0a0a 2020 2020 2020  AnnData]..      
-00010610: 2020 2222 220a 2020 2020 2020 2020 0a0a    """.        ..
-00010620: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-00010630: 200a 0a20 2020 2020 2020 200a 2020 2020   ..        .    
-00010640: 2020 2020 6966 206f 7574 7075 745f 7479      if output_ty
-00010650: 7065 206e 6f74 2069 6e20 5b22 616e 6e64  pe not in ["annd
-00010660: 6174 6122 2c22 7469 6479 222c 2264 6963  ata","tidy","dic
-00010670: 7422 5d3a 0a20 2020 2020 2020 2020 2020  t"]:.           
-00010680: 2072 6169 7365 2056 616c 7565 4572 726f   raise ValueErro
-00010690: 7228 2750 6c65 6173 6520 656e 7465 7220  r('Please enter 
-000106a0: 6120 7661 6c69 6420 6f75 7470 7574 5f74  a valid output_t
-000106b0: 7970 652e 2061 6e6e 6461 7461 2066 6f72  ype. anndata for
-000106c0: 270a 2020 2020 2020 2020 2020 2020 2020  '.              
-000106d0: 2020 2020 2020 2020 2020 2020 2020 2027                 '
-000106e0: 2261 6e6e 6461 7461 2220 6f62 6a65 6374  "anndata" object
-000106f0: 732c 2022 7469 6479 2220 666f 7220 6c6f  s, "tidy" for lo
-00010700: 6e67 2066 6f72 6d20 7061 6e64 6173 270a  ng form pandas'.
-00010710: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010720: 2020 2020 2020 2020 2020 2020 2027 4461               'Da
-00010730: 7461 4672 616d 6573 2c20 2264 6963 7422  taFrames, "dict"
-00010740: 2066 6f72 2064 6963 7469 6f6e 6172 6965   for dictionarie
-00010750: 7320 7769 7468 2073 6570 6172 6174 6527  s with separate'
-00010760: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00010770: 2020 2020 2020 2020 2020 2020 2020 2765                'e
-00010780: 7870 7265 7373 696f 6e20 616e 6420 6d65  xpression and me
-00010790: 7461 6461 7461 2066 6965 6c64 7327 0a20  tadata fields'. 
+0000fb00: 2020 2020 2020 2020 2020 206f 7574 7075             outpu
+0000fb10: 745f 7479 7065 3a73 7472 203d 2027 616e  t_type:str = 'an
+0000fb20: 6e64 6174 6127 2c0a 2020 2020 2020 2020  ndata',.        
+0000fb30: 2020 2020 2020 2020 2020 2020 2020 2020                  
+0000fb40: 2020 202a 2a6b 7761 7267 7329 2d3e 6469     **kwargs)->di
+0000fb50: 6374 5b69 6e74 3a64 6963 747c 416e 6e44  ct[int:dict|AnnD
+0000fb60: 6174 615d 3a0a 2020 2020 2020 2020 0a20  ata]:.        . 
+0000fb70: 2020 2020 2020 2022 2222 5265 7475 726e         """Return
+0000fb80: 2061 2064 6174 6120 7374 7275 6374 7572   a data structur
+0000fb90: 6520 696e 636c 7564 696e 6720 616c 6c20  e including all 
+0000fba0: 7265 6c65 7661 6e74 2064 6174 6120 7265  relevant data re
+0000fbb0: 6c61 7465 6420 746f 0a20 2020 2020 2020  lated to.       
+0000fbc0: 2067 656e 6520 6578 7072 6573 7369 6f6e   gene expression
+0000fbd0: 2069 6e20 6120 6461 7461 7365 742e 2045   in a dataset. E
+0000fbe0: 6974 6865 7220 7265 7475 726e 7320 616e  ither returns an
+0000fbf0: 2061 6e6e 6461 7461 206f 626a 6563 7420   anndata object 
+0000fc00: 6f72 200a 2020 2020 2020 2020 6120 6469  or .        a di
+0000fc10: 6374 696f 6e61 7279 2077 6974 6820 616c  ctionary with al
+0000fc20: 6c20 7468 6520 6e65 6564 6564 2066 6965  l the needed fie
+0000fc30: 6c64 732e 0a20 2020 2020 2020 200a 2020  lds..        .  
+0000fc40: 2020 2020 2020 0a20 2020 2020 2020 203a        .        :
+0000fc50: 7061 7261 6d20 6461 7461 7365 7473 3a20  param datasets: 
+0000fc60: 4e75 6d65 7269 6361 6c20 6461 7461 7365  Numerical datase
+0000fc70: 7420 6964 656e 7469 6669 6572 0a20 2020  t identifier.   
+0000fc80: 2020 2020 2020 2064 6174 6173 6574 2073         dataset s
+0000fc90: 686f 7274 206e 616d 6573 0a20 2020 2020  hort names.     
+0000fca0: 2020 203a 7479 7065 2064 6174 6173 6574     :type dataset
+0000fcb0: 733a 204c 6973 745b 7374 727c 696e 745d  s: List[str|int]
+0000fcc0: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
+0000fcd0: 6765 6e65 733a 2041 6e20 6e63 6269 2067  genes: An ncbi g
+0000fce0: 656e 6520 6964 656e 7469 6669 6572 2061  ene identifier a
+0000fcf0: 6e2c 200a 2020 2020 2020 2020 2020 656e  n, .          en
+0000fd00: 7365 6d62 6c20 6765 6e65 2069 6465 6e74  sembl gene ident
+0000fd10: 6966 6965 7220 7768 6963 6820 7479 7069  ifier which typi
+0000fd20: 6361 6c6c 7920 7374 6172 7473 2077 6974  cally starts wit
+0000fd30: 6820 656e 7367 206f 7220 616e 200a 2020  h ensg or an .  
+0000fd40: 2020 2020 2020 2020 6f66 6669 6369 616c          official
+0000fd50: 2067 656e 6520 7379 6d62 6f6c 2061 7070   gene symbol app
+0000fd60: 726f 7665 6420 6279 2068 676e 632c 2064  roved by hgnc, d
+0000fd70: 6566 6175 6c74 7320 746f 204e 6f6e 650a  efaults to None.
+0000fd80: 2020 2020 2020 2020 3a74 7970 6520 6765          :type ge
+0000fd90: 6e65 733a 204f 7074 696f 6e61 6c5b 4c69  nes: Optional[Li
+0000fda0: 7374 5b73 7472 7c69 6e74 5d5d 2c20 6f70  st[str|int]], op
+0000fdb0: 7469 6f6e 616c 0a20 2020 2020 2020 203a  tional.        :
+0000fdc0: 7061 7261 6d20 6b65 6570 5f6e 6f6e 5f73  param keep_non_s
+0000fdd0: 7065 6369 6669 633a 2049 6620 5472 7565  pecific: If True
+0000fde0: 2c20 7265 7375 6c74 7320 6672 6f6d 200a  , results from .
+0000fdf0: 2020 2020 2020 2020 2020 7072 6f62 6573            probes
+0000fe00: 6574 7320 7468 6174 2061 7265 206e 6f74  ets that are not
+0000fe10: 2073 7065 6369 6669 6320 746f 2074 6865   specific to the
+0000fe20: 2067 656e 6520 7769 6c6c 2061 6c73 6f20   gene will also 
+0000fe30: 6265 2072 6574 7572 6e65 642c 0a20 2020  be returned,.   
+0000fe40: 2020 2020 2020 2064 6566 6175 6c74 7320         defaults 
+0000fe50: 746f 2046 616c 7365 0a20 2020 2020 2020  to False.       
+0000fe60: 203a 7479 7065 206b 6565 705f 6e6f 6e5f   :type keep_non_
+0000fe70: 7370 6563 6966 6963 3a20 5459 5045 2c20  specific: TYPE, 
+0000fe80: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
+0000fe90: 203a 7061 7261 6d20 636f 6e73 6f6c 6964   :param consolid
+0000fea0: 6174 653a 2044 4553 4352 4950 5449 4f4e  ate: DESCRIPTION
+0000feb0: 2c20 416e 206f 7074 696f 6e20 666f 7220  , An option for 
+0000fec0: 6765 6e65 2065 7870 7265 7373 696f 6e20  gene expression 
+0000fed0: 6c65 7665 6c20 636f 6e73 6f6c 6964 6174  level consolidat
+0000fee0: 696f 6e2e 200a 2020 2020 2020 2020 2020  ion. .          
+0000fef0: 4966 2065 6d70 7479 2c20 7769 6c6c 2072  If empty, will r
+0000ff00: 6574 7572 6e20 6576 6572 7920 7072 6f62  eturn every prob
+0000ff10: 6520 666f 7220 7468 6520 6765 6e65 732e  e for the genes.
+0000ff20: 2022 7069 636b 6d61 7822 2074 6f20 7069   "pickmax" to pi
+0000ff30: 636b 200a 2020 2020 2020 2020 2020 7468  ck .          th
+0000ff40: 6520 7072 6f62 6520 7769 7468 2074 6865  e probe with the
+0000ff50: 2068 6967 6865 7374 2065 7870 7265 7373   highest express
+0000ff60: 696f 6e2c 2022 7069 636b 7661 7222 2074  ion, "pickvar" t
+0000ff70: 6f20 7069 636b 2074 6865 2070 726f 7665  o pick the prove
+0000ff80: 200a 2020 2020 2020 2020 2020 7769 7468   .          with
+0000ff90: 2074 6865 2068 6967 6865 7374 2076 6172   the highest var
+0000ffa0: 6961 6e63 6520 616e 6420 2261 7665 7261  iance and "avera
+0000ffb0: 6765 2220 666f 7220 7265 7475 726e 696e  ge" for returnin
+0000ffc0: 6720 7468 6520 6176 6572 6167 6520 0a20  g the average . 
+0000ffd0: 2020 2020 2020 2020 2065 7870 7265 7373           express
+0000ffe0: 696f 6e20 746f 204e 6f6e 650a 2020 2020  ion to None.    
+0000fff0: 2020 2020 3a74 7970 6520 636f 6e73 6f6c      :type consol
+00010000: 6964 6174 653a 204f 7074 696f 6e61 6c5b  idate: Optional[
+00010010: 7374 725d 2c20 6f70 7469 6f6e 616c 0a20  str], optional. 
+00010020: 2020 2020 2020 203a 7061 7261 6d20 7265         :param re
+00010030: 7375 6c74 5f73 6574 733a 2052 6573 756c  sult_sets: Resul
+00010040: 7420 7365 7420 4944 7320 6f66 2074 6865  t set IDs of the
+00010050: 2061 200a 2020 2020 2020 2020 2020 6469   a .          di
+00010060: 6666 6572 656e 7469 616c 2065 7870 7265  fferential expre
+00010070: 7373 696f 6e20 616e 616c 7973 6973 2e20  ssion analysis. 
+00010080: 4966 2070 726f 7669 6465 642c 2074 6865  If provided, the
+00010090: 206f 7574 7075 7420 7769 6c6c 206f 6e6c   output will onl
+000100a0: 7920 0a20 2020 2020 2020 2020 2069 6e63  y .          inc
+000100b0: 6c75 6465 2074 6865 2073 616d 706c 6573  lude the samples
+000100c0: 2066 726f 6d20 7468 6520 7375 6273 6574   from the subset
+000100d0: 2075 7365 6420 696e 2074 6865 2072 6573   used in the res
+000100e0: 756c 7420 7365 7420 4944 2e20 4d75 7374  ult set ID. Must
+000100f0: 200a 2020 2020 2020 2020 2020 6265 2074   .          be t
+00010100: 6865 2073 616d 6520 6c65 6e67 7468 2061  he same length a
+00010110: 7320 6461 7461 7365 7473 2c20 6465 6661  s datasets, defa
+00010120: 756c 7473 2074 6f20 4e6f 6e65 0a20 2020  ults to None.   
+00010130: 2020 2020 203a 7479 7065 2072 6573 756c       :type resul
+00010140: 745f 7365 7473 3a20 4f70 7469 6f6e 616c  t_sets: Optional
+00010150: 5b4c 6973 745b 696e 745d 5d2c 206f 7074  [List[int]], opt
+00010160: 696f 6e61 6c0a 2020 2020 2020 2020 3a70  ional.        :p
+00010170: 6172 616d 2063 6f6e 7472 6173 7473 3a20  aram contrasts: 
+00010180: 436f 6e74 7261 7374 2049 4473 206f 6620  Contrast IDs of 
+00010190: 6120 6469 6666 6572 656e 7469 616c 200a  a differential .
+000101a0: 2020 2020 2020 2020 2020 6578 7072 6573            expres
+000101b0: 7369 6f6e 2063 6f6e 7472 6173 742e 204e  sion contrast. N
+000101c0: 6565 6420 7265 7375 6c74 5f73 6574 7320  eed result_sets 
+000101d0: 746f 2062 6520 6465 6669 6e65 6420 746f  to be defined to
+000101e0: 2077 6f72 6b2e 2049 6620 0a20 2020 2020   work. If .     
+000101f0: 2020 2020 2070 726f 7669 6465 642c 2074       provided, t
+00010200: 6865 206f 7574 7075 7420 7769 6c6c 206f  he output will o
+00010210: 6e6c 7920 696e 636c 7564 6520 7361 6d70  nly include samp
+00010220: 6c65 7320 7265 6c65 7661 6e74 2074 6f20  les relevant to 
+00010230: 7468 6520 270a 2020 2020 2020 2020 2020  the '.          
+00010240: 7370 6563 6966 6963 2063 6f6e 7472 6174  specific contrat
+00010250: 732e 204d 7573 7420 6265 2074 6865 2073  s. Must be the s
+00010260: 616d 6520 6c65 6e67 7468 2061 7320 6461  ame length as da
+00010270: 7461 7365 7473 2e0a 2020 2020 2020 2020  tasets..        
+00010280: 3a70 6172 616d 2073 7472 206d 6574 615f  :param str meta_
+00010290: 7479 7065 3a20 486f 7720 7368 6f75 6c64  type: How should
+000102a0: 2074 6865 206d 6574 6164 6174 6120 696e   the metadata in
+000102b0: 666f 726d 6174 696f 6e20 7368 6f75 6c64  formation should
+000102c0: 2062 6520 0a20 2020 2020 2020 2020 2069   be .          i
+000102d0: 6e63 6c75 6465 642e 2043 616e 2062 6520  ncluded. Can be 
+000102e0: 2274 6578 7422 2c20 2275 7269 2220 6f72  "text", "uri" or
+000102f0: 2022 626f 7468 222e 2022 7465 7874 2220   "both". "text" 
+00010300: 616e 6420 2275 7269 2220 6f70 7469 6f6e  and "uri" option
+00010310: 732c 2064 6566 6175 6c74 7320 746f 204e  s, defaults to N
+00010320: 6f6e 650a 2020 2020 2020 2020 3a74 7970  one.        :typ
+00010330: 6520 636f 6e74 7261 7374 733a 204f 7074  e contrasts: Opt
+00010340: 696f 6e61 6c5b 4c69 7374 5b73 7472 5d5d  ional[List[str]]
+00010350: 2c20 6f70 7469 6f6e 616c 0a20 2020 2020  , optional.     
+00010360: 2020 203a 7061 7261 6d20 6d65 7461 5f74     :param meta_t
+00010370: 7970 653a 2048 6f77 2073 686f 756c 6420  ype: How should 
+00010380: 7468 6520 6d65 7461 6461 7461 2069 6e66  the metadata inf
+00010390: 6f72 6d61 7469 6f6e 2073 686f 756c 6420  ormation should 
+000103a0: 6265 200a 2020 2020 2020 2020 2020 696e  be .          in
+000103b0: 636c 7564 6564 2e20 4361 6e20 6265 2022  cluded. Can be "
+000103c0: 7465 7874 222c 2022 7572 6922 206f 7220  text", "uri" or 
+000103d0: 2262 6f74 6822 2e20 2274 6578 7422 2061  "both". "text" a
+000103e0: 6e64 2022 7572 6922 206f 7074 696f 6e73  nd "uri" options
+000103f0: 2c20 6465 6661 756c 7473 2074 6f20 2774  , defaults to 't
+00010400: 6578 7427 0a20 2020 2020 2020 203a 7479  ext'.        :ty
+00010410: 7065 206d 6574 615f 7479 7065 3a20 7374  pe meta_type: st
+00010420: 722c 206f 7074 696f 6e61 6c0a 2020 2020  r, optional.    
+00010430: 2020 2020 3a70 6172 616d 206f 7574 7075      :param outpu
+00010440: 745f 7479 7065 3a20 5479 7065 206f 6620  t_type: Type of 
+00010450: 7468 6520 7265 7475 726e 6564 206f 626a  the returned obj
+00010460: 6563 742e 2022 616e 6e64 6174 6122 2066  ect. "anndata" f
+00010470: 6f72 2061 6e20 0a20 2020 2020 2020 2020  or an .         
+00010480: 2041 6e6e 4461 7461 206f 626a 6563 7420   AnnData object 
+00010490: 616e 6420 2264 6963 7422 2066 6f72 2061  and "dict" for a
+000104a0: 2064 6963 7469 6f6e 6172 7920 706f 7075   dictionary popu
+000104b0: 6c61 7465 6420 7769 7468 2044 6174 6146  lated with DataF
+000104c0: 7261 6d65 732c 0a20 2020 2020 2020 2020  rames,.         
+000104d0: 2064 6566 6175 6c74 7320 746f 2027 616e   defaults to 'an
+000104e0: 6e64 6174 6127 0a20 2020 2020 2020 203a  ndata'.        :
+000104f0: 7479 7065 206f 7574 7075 745f 7479 7065  type output_type
+00010500: 3a20 7374 722c 206f 7074 696f 6e61 6c0a  : str, optional.
+00010510: 2020 2020 2020 2020 3a70 6172 616d 202a          :param *
+00010520: 2a6b 7761 7267 733a 2044 4553 4352 4950  *kwargs: DESCRIP
+00010530: 5449 4f4e 0a20 2020 2020 2020 203a 7479  TION.        :ty
+00010540: 7065 202a 2a6b 7761 7267 733a 2054 5950  pe **kwargs: TYP
+00010550: 450a 2020 2020 2020 2020 3a72 6169 7365  E.        :raise
+00010560: 7320 5661 6c75 6545 7272 6f72 3a20 4445  s ValueError: DE
+00010570: 5343 5249 5054 494f 4e0a 2020 2020 2020  SCRIPTION.      
+00010580: 2020 3a72 6574 7572 6e3a 2041 2064 6963    :return: A dic
+00010590: 7469 6f6e 6172 7920 636f 6e74 6169 6e69  tionary containi
+000105a0: 6e67 2041 6e6e 4461 7461 206f 626a 6563  ng AnnData objec
+000105b0: 7473 206f 7220 6e65 7374 6564 2064 6963  ts or nested dic
+000105c0: 7469 6f6e 6172 6965 730a 2020 2020 2020  tionaries.      
+000105d0: 2020 2020 7468 6174 2063 6f6e 7461 696e      that contain
+000105e0: 2065 7870 7265 7373 696f 6e20 616e 6420   expression and 
+000105f0: 7361 6d70 6c65 206d 6574 6164 6120 6f66  sample metada of
+00010600: 2074 6865 2072 6571 7565 7374 6564 2065   the requested e
+00010610: 7870 6572 696d 656e 7473 0a20 2020 2020  xperiments.     
+00010620: 2020 203a 7274 7970 653a 2064 6963 745b     :rtype: dict[
+00010630: 696e 743a 6469 6374 7c41 6e6e 4461 7461  int:dict|AnnData
+00010640: 5d0a 0a20 2020 2020 2020 2022 2222 0a20  ]..        """. 
+00010650: 2020 2020 2020 200a 0a20 2020 2020 2020         ..       
+00010660: 200a 2020 2020 2020 2020 0a0a 2020 2020   .        ..    
+00010670: 2020 2020 0a20 2020 2020 2020 2069 6620      .        if 
+00010680: 6f75 7470 7574 5f74 7970 6520 6e6f 7420  output_type not 
+00010690: 696e 205b 2261 6e6e 6461 7461 222c 2274  in ["anndata","t
+000106a0: 6964 7922 2c22 6469 6374 225d 3a0a 2020  idy","dict"]:.  
+000106b0: 2020 2020 2020 2020 2020 7261 6973 6520            raise 
+000106c0: 5661 6c75 6545 7272 6f72 2827 506c 6561  ValueError('Plea
+000106d0: 7365 2065 6e74 6572 2061 2076 616c 6964  se enter a valid
+000106e0: 206f 7574 7075 745f 7479 7065 2e20 616e   output_type. an
+000106f0: 6e64 6174 6120 666f 7227 0a20 2020 2020  ndata for'.     
+00010700: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010710: 2020 2020 2020 2020 2722 616e 6e64 6174          '"anndat
+00010720: 6122 206f 626a 6563 7473 2c20 2274 6964  a" objects, "tid
+00010730: 7922 2066 6f72 206c 6f6e 6720 666f 726d  y" for long form
+00010740: 2070 616e 6461 7327 0a20 2020 2020 2020   pandas'.       
+00010750: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010760: 2020 2020 2020 2744 6174 6146 7261 6d65        'DataFrame
+00010770: 732c 2022 6469 6374 2220 666f 7220 6469  s, "dict" for di
+00010780: 6374 696f 6e61 7269 6573 2077 6974 6820  ctionaries with 
+00010790: 7365 7061 7261 7465 270a 2020 2020 2020  separate'.      
 000107a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000107b0: 2020 2020 2020 2020 2020 2020 2029 0a20               ). 
-000107c0: 2020 2020 2020 2020 2020 200a 2020 2020             .    
-000107d0: 2020 2020 756e 6971 7565 5f73 6574 7320      unique_sets 
-000107e0: 3d20 6c69 7374 2873 6574 2864 6174 6173  = list(set(datas
-000107f0: 6574 7329 290a 2020 2020 2020 2020 0a20  ets)).        . 
-00010800: 2020 2020 2020 206d 6574 6164 6174 6120         metadata 
-00010810: 3d20 7b6b 3a73 656c 662e 6765 745f 6461  = {k:self.get_da
-00010820: 7461 7365 745f 7361 6d70 6c65 7328 6b29  taset_samples(k)
-00010830: 2066 6f72 206b 2069 6e20 756e 6971 7565   for k in unique
-00010840: 5f73 6574 737d 0a20 2020 2020 2020 200a  _sets}.        .
-00010850: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
-00010860: 2069 6620 6765 6e65 7320 6973 204e 6f6e   if genes is Non
-00010870: 653a 0a20 2020 2020 2020 2020 2020 2064  e:.            d
-00010880: 6566 2067 6574 5f65 7870 2864 6174 6173  ef get_exp(datas
-00010890: 6574 293a 0a20 2020 2020 2020 2020 2020  et):.           
-000108a0: 2020 2020 2065 7870 203d 2073 656c 662e       exp = self.
-000108b0: 6765 745f 6461 7461 7365 745f 7072 6f63  get_dataset_proc
-000108c0: 6573 7365 645f 6578 7072 6573 7369 6f6e  essed_expression
-000108d0: 2864 6174 6173 6574 290a 2020 2020 2020  (dataset).      
-000108e0: 2020 2020 2020 2020 2020 6d65 7461 203d            meta =
-000108f0: 206d 6574 6164 6174 615b 6461 7461 7365   metadata[datase
-00010900: 745d 0a20 2020 2020 2020 2020 2020 2020  t].             
-00010910: 2020 200a 2020 2020 2020 2020 2020 2020     .            
-00010920: 2020 2020 6966 206e 6f74 206b 6565 705f      if not keep_
-00010930: 6e6f 6e5f 7370 6563 6966 6963 3a0a 2020  non_specific:.  
-00010940: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010950: 2020 6578 7020 3d20 6578 705b 7e65 7870    exp = exp[~exp
-00010960: 2e47 656e 6553 796d 626f 6c2e 7374 722e  .GeneSymbol.str.
-00010970: 636f 6e74 6169 6e73 2822 7c22 2c72 6567  contains("|",reg
-00010980: 6578 203d 2046 616c 7365 2c6e 6120 3d20  ex = False,na = 
-00010990: 5472 7565 295d 0a20 2020 2020 2020 2020  True)].         
-000109a0: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-000109b0: 2020 2020 2020 2020 6966 2063 6f6e 736f          if conso
-000109c0: 6c69 6461 7465 2069 7320 6e6f 7420 4e6f  lidate is not No
-000109d0: 6e65 2061 6e64 2063 6f6e 736f 6c69 6461  ne and consolida
-000109e0: 7465 203d 3d27 7069 636b 6d61 7827 3a0a  te =='pickmax':.
-000109f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010a00: 2020 2020 6d65 616e 5f65 7870 203d 2065      mean_exp = e
-00010a10: 7870 5b6d 6574 612e 7361 6d70 6c65 5f6e  xp[meta.sample_n
-00010a20: 616d 655d 2e6d 6561 6e28 6178 6973 3d31  ame].mean(axis=1
-00010a30: 2c73 6b69 706e 613d 5472 7565 290a 2020  ,skipna=True).  
-00010a40: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010a50: 2020 6578 7020 3d20 6578 702e 696c 6f63    exp = exp.iloc
-00010a60: 5b6c 6973 7428 7375 622e 6f72 6465 7228  [list(sub.order(
-00010a70: 6d65 616e 5f65 7870 2c64 6563 7265 6173  mean_exp,decreas
-00010a80: 696e 6720 3d20 5472 7565 2929 5d0a 2020  ing = True))].  
-00010a90: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010aa0: 2020 6578 7020 3d20 6578 705b 7e65 7870    exp = exp[~exp
-00010ab0: 2e64 7570 6c69 6361 7465 6428 2747 656e  .duplicated('Gen
-00010ac0: 6553 796d 626f 6c27 295d 0a20 2020 2020  eSymbol')].     
-00010ad0: 2020 2020 2020 2020 2020 2065 6c69 6620             elif 
-00010ae0: 636f 6e73 6f6c 6964 6174 6520 6973 206e  consolidate is n
-00010af0: 6f74 204e 6f6e 6520 616e 6420 636f 6e73  ot None and cons
-00010b00: 6f6c 6964 6174 6520 3d3d 2027 7069 636b  olidate == 'pick
-00010b10: 7661 7227 3a0a 2020 2020 2020 2020 2020  var':.          
-00010b20: 2020 2020 2020 2020 2020 6578 705f 7661            exp_va
-00010b30: 7220 3d20 6578 705b 6d65 7461 2e73 616d  r = exp[meta.sam
-00010b40: 706c 655f 6e61 6d65 5d2e 7661 7228 6178  ple_name].var(ax
-00010b50: 6973 203d 2031 2c20 736b 6970 6e61 3d20  is = 1, skipna= 
-00010b60: 5472 7565 290a 2020 2020 2020 2020 2020  True).          
-00010b70: 2020 2020 2020 2020 2020 6578 7020 3d20            exp = 
-00010b80: 6578 702e 696c 6f63 5b6c 6973 7428 7375  exp.iloc[list(su
-00010b90: 622e 6f72 6465 7228 6578 705f 7661 722c  b.order(exp_var,
-00010ba0: 6465 6372 6561 7369 6e67 203d 2054 7275  decreasing = Tru
-00010bb0: 6529 295d 0a20 2020 2020 2020 2020 2020  e))].           
-00010bc0: 2020 2020 2020 2020 2065 7870 203d 2065           exp = e
-00010bd0: 7870 5b7e 6578 702e 6475 706c 6963 6174  xp[~exp.duplicat
-00010be0: 6564 2827 4765 6e65 5379 6d62 6f6c 2729  ed('GeneSymbol')
-00010bf0: 5d0a 2020 2020 2020 2020 2020 2020 2020  ].              
-00010c00: 2020 656c 6966 2063 6f6e 736f 6c69 6461    elif consolida
-00010c10: 7465 2069 7320 6e6f 7420 4e6f 6e65 2061  te is not None a
-00010c20: 6e64 2063 6f6e 736f 6c69 6461 7465 203d  nd consolidate =
-00010c30: 3d20 2761 7665 7261 6765 273a 0a20 2020  = 'average':.   
-00010c40: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010c50: 2064 7570 7320 3d20 6c69 7374 280a 2020   dups = list(.  
-00010c60: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010c70: 2020 2020 2020 7365 7428 6c69 7374 280a        set(list(.
-00010c80: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010c90: 2020 2020 2020 2020 2020 2020 6578 705b              exp[
-00010ca0: 6578 702e 6475 706c 6963 6174 6564 2822  exp.duplicated("
-00010cb0: 4765 6e65 5379 6d62 6f6c 2229 5d5b 2247  GeneSymbol")]["G
-00010cc0: 656e 6553 796d 626f 6c22 5d0a 2020 2020  eneSymbol"].    
-00010cd0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010ce0: 2020 2020 2020 2020 2929 0a20 2020 2020          )).     
-00010cf0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010d00: 2020 2029 0a20 2020 2020 2020 2020 2020     ).           
-00010d10: 2020 2020 2020 2020 200a 2020 2020 2020           .      
-00010d20: 2020 2020 2020 2020 2020 2020 2020 6465                de
-00010d30: 6620 6765 745f 6d65 616e 2864 7570 293a  f get_mean(dup):
-00010d40: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00010d50: 2020 2020 2020 2020 2064 7570 5f73 7562           dup_sub
-00010d60: 7365 7420 3d20 6578 705b 6578 702e 4765  set = exp[exp.Ge
-00010d70: 6e65 5379 6d62 6f6c 203d 3d20 6475 705d  neSymbol == dup]
-00010d80: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00010d90: 2020 2020 2020 2020 2064 7570 5f6d 6561           dup_mea
-00010da0: 6e20 3d20 2065 7870 5b6d 6574 612e 7361  n =  exp[meta.sa
-00010db0: 6d70 6c65 5f6e 616d 655d 2e6d 6561 6e28  mple_name].mean(
-00010dc0: 6178 6973 203d 2030 290a 2020 2020 2020  axis = 0).      
-00010dd0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010de0: 2020 7072 6f62 6520 3d20 2241 7665 7261    probe = "Avera
-00010df0: 6765 6420 6672 6f6d 2022 202b 2022 2022  ged from " + " "
-00010e00: 2e6a 6f69 6e28 6475 705f 7375 6273 6574  .join(dup_subset
-00010e10: 2e50 726f 6265 290a 2020 2020 2020 2020  .Probe).        
-00010e20: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010e30: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00010e40: 2020 2020 2020 2020 2067 656e 655f 696e           gene_in
-00010e50: 666f 203d 2064 7570 5f73 7562 7365 742e  fo = dup_subset.
-00010e60: 5c0a 2020 2020 2020 2020 2020 2020 2020  \.              
-00010e70: 2020 2020 2020 2020 2020 2020 2020 6c6f                lo
-00010e80: 635b 3a2c 0a20 2020 2020 2020 2020 2020  c[:,.           
-00010e90: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010ea0: 2020 2020 206e 702e 6172 7261 7928 7e6e       np.array(~n
-00010eb0: 702e 6172 7261 7928 0a20 2020 2020 2020  p.array(.       
+000107b0: 2020 2020 2020 2027 6578 7072 6573 7369         'expressi
+000107c0: 6f6e 2061 6e64 206d 6574 6164 6174 6120  on and metadata 
+000107d0: 6669 656c 6473 270a 2020 2020 2020 2020  fields'.        
+000107e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000107f0: 2020 2020 2020 290a 2020 2020 2020 2020        ).        
+00010800: 2020 2020 0a20 2020 2020 2020 2075 6e69      .        uni
+00010810: 7175 655f 7365 7473 203d 206c 6973 7428  que_sets = list(
+00010820: 7365 7428 6461 7461 7365 7473 2929 0a20  set(datasets)). 
+00010830: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+00010840: 6d65 7461 6461 7461 203d 207b 6b3a 7365  metadata = {k:se
+00010850: 6c66 2e67 6574 5f64 6174 6173 6574 5f73  lf.get_dataset_s
+00010860: 616d 706c 6573 286b 2920 666f 7220 6b20  amples(k) for k 
+00010870: 696e 2075 6e69 7175 655f 7365 7473 7d0a  in unique_sets}.
+00010880: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00010890: 200a 2020 2020 2020 2020 6966 2067 656e   .        if gen
+000108a0: 6573 2069 7320 4e6f 6e65 3a0a 2020 2020  es is None:.    
+000108b0: 2020 2020 2020 2020 6465 6620 6765 745f          def get_
+000108c0: 6578 7028 6461 7461 7365 7429 3a0a 2020  exp(dataset):.  
+000108d0: 2020 2020 2020 2020 2020 2020 2020 6578                ex
+000108e0: 7020 3d20 7365 6c66 2e67 6574 5f64 6174  p = self.get_dat
+000108f0: 6173 6574 5f70 726f 6365 7373 6564 5f65  aset_processed_e
+00010900: 7870 7265 7373 696f 6e28 6461 7461 7365  xpression(datase
+00010910: 7429 0a20 2020 2020 2020 2020 2020 2020  t).             
+00010920: 2020 206d 6574 6120 3d20 6d65 7461 6461     meta = metada
+00010930: 7461 5b64 6174 6173 6574 5d0a 2020 2020  ta[dataset].    
+00010940: 2020 2020 2020 2020 2020 2020 0a20 2020              .   
+00010950: 2020 2020 2020 2020 2020 2020 2069 6620               if 
+00010960: 6e6f 7420 6b65 6570 5f6e 6f6e 5f73 7065  not keep_non_spe
+00010970: 6369 6669 633a 0a20 2020 2020 2020 2020  cific:.         
+00010980: 2020 2020 2020 2020 2020 2065 7870 203d             exp =
+00010990: 2065 7870 5b7e 6578 702e 4765 6e65 5379   exp[~exp.GeneSy
+000109a0: 6d62 6f6c 2e73 7472 2e63 6f6e 7461 696e  mbol.str.contain
+000109b0: 7328 227c 222c 7265 6765 7820 3d20 4661  s("|",regex = Fa
+000109c0: 6c73 652c 6e61 203d 2054 7275 6529 5d0a  lse,na = True)].
+000109d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000109e0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+000109f0: 2069 6620 636f 6e73 6f6c 6964 6174 6520   if consolidate 
+00010a00: 6973 206e 6f74 204e 6f6e 6520 616e 6420  is not None and 
+00010a10: 636f 6e73 6f6c 6964 6174 6520 3d3d 2770  consolidate =='p
+00010a20: 6963 6b6d 6178 273a 0a20 2020 2020 2020  ickmax':.       
+00010a30: 2020 2020 2020 2020 2020 2020 206d 6561               mea
+00010a40: 6e5f 6578 7020 3d20 6578 705b 6d65 7461  n_exp = exp[meta
+00010a50: 2e73 616d 706c 655f 6e61 6d65 5d2e 6d65  .sample_name].me
+00010a60: 616e 2861 7869 733d 312c 736b 6970 6e61  an(axis=1,skipna
+00010a70: 3d54 7275 6529 0a20 2020 2020 2020 2020  =True).         
+00010a80: 2020 2020 2020 2020 2020 2065 7870 203d             exp =
+00010a90: 2065 7870 2e69 6c6f 635b 6c69 7374 2873   exp.iloc[list(s
+00010aa0: 7562 2e6f 7264 6572 286d 6561 6e5f 6578  ub.order(mean_ex
+00010ab0: 702c 6465 6372 6561 7369 6e67 203d 2054  p,decreasing = T
+00010ac0: 7275 6529 295d 0a20 2020 2020 2020 2020  rue))].         
+00010ad0: 2020 2020 2020 2020 2020 2065 7870 203d             exp =
+00010ae0: 2065 7870 5b7e 6578 702e 6475 706c 6963   exp[~exp.duplic
+00010af0: 6174 6564 2827 4765 6e65 5379 6d62 6f6c  ated('GeneSymbol
+00010b00: 2729 5d0a 2020 2020 2020 2020 2020 2020  ')].            
+00010b10: 2020 2020 656c 6966 2063 6f6e 736f 6c69      elif consoli
+00010b20: 6461 7465 2069 7320 6e6f 7420 4e6f 6e65  date is not None
+00010b30: 2061 6e64 2063 6f6e 736f 6c69 6461 7465   and consolidate
+00010b40: 203d 3d20 2770 6963 6b76 6172 273a 0a20   == 'pickvar':. 
+00010b50: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010b60: 2020 2065 7870 5f76 6172 203d 2065 7870     exp_var = exp
+00010b70: 5b6d 6574 612e 7361 6d70 6c65 5f6e 616d  [meta.sample_nam
+00010b80: 655d 2e76 6172 2861 7869 7320 3d20 312c  e].var(axis = 1,
+00010b90: 2073 6b69 706e 613d 2054 7275 6529 0a20   skipna= True). 
+00010ba0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010bb0: 2020 2065 7870 203d 2065 7870 2e69 6c6f     exp = exp.ilo
+00010bc0: 635b 6c69 7374 2873 7562 2e6f 7264 6572  c[list(sub.order
+00010bd0: 2865 7870 5f76 6172 2c64 6563 7265 6173  (exp_var,decreas
+00010be0: 696e 6720 3d20 5472 7565 2929 5d0a 2020  ing = True))].  
+00010bf0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010c00: 2020 6578 7020 3d20 6578 705b 7e65 7870    exp = exp[~exp
+00010c10: 2e64 7570 6c69 6361 7465 6428 2747 656e  .duplicated('Gen
+00010c20: 6553 796d 626f 6c27 295d 0a20 2020 2020  eSymbol')].     
+00010c30: 2020 2020 2020 2020 2020 2065 6c69 6620             elif 
+00010c40: 636f 6e73 6f6c 6964 6174 6520 6973 206e  consolidate is n
+00010c50: 6f74 204e 6f6e 6520 616e 6420 636f 6e73  ot None and cons
+00010c60: 6f6c 6964 6174 6520 3d3d 2027 6176 6572  olidate == 'aver
+00010c70: 6167 6527 3a0a 2020 2020 2020 2020 2020  age':.          
+00010c80: 2020 2020 2020 2020 2020 6475 7073 203d            dups =
+00010c90: 206c 6973 7428 0a20 2020 2020 2020 2020   list(.         
+00010ca0: 2020 2020 2020 2020 2020 2020 2020 2073                 s
+00010cb0: 6574 286c 6973 7428 0a20 2020 2020 2020  et(list(.       
+00010cc0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010cd0: 2020 2020 2065 7870 5b65 7870 2e64 7570       exp[exp.dup
+00010ce0: 6c69 6361 7465 6428 2247 656e 6553 796d  licated("GeneSym
+00010cf0: 626f 6c22 295d 5b22 4765 6e65 5379 6d62  bol")]["GeneSymb
+00010d00: 6f6c 225d 0a20 2020 2020 2020 2020 2020  ol"].           
+00010d10: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010d20: 2029 290a 2020 2020 2020 2020 2020 2020   )).            
+00010d30: 2020 2020 2020 2020 2020 2020 290a 2020              ).  
+00010d40: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010d50: 2020 0a20 2020 2020 2020 2020 2020 2020    .             
+00010d60: 2020 2020 2020 2064 6566 2067 6574 5f6d         def get_m
+00010d70: 6561 6e28 6475 7029 3a0a 2020 2020 2020  ean(dup):.      
+00010d80: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010d90: 2020 6475 705f 7375 6273 6574 203d 2065    dup_subset = e
+00010da0: 7870 5b65 7870 2e47 656e 6553 796d 626f  xp[exp.GeneSymbo
+00010db0: 6c20 3d3d 2064 7570 5d0a 2020 2020 2020  l == dup].      
+00010dc0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010dd0: 2020 6475 705f 6d65 616e 203d 2020 6578    dup_mean =  ex
+00010de0: 705b 6d65 7461 2e73 616d 706c 655f 6e61  p[meta.sample_na
+00010df0: 6d65 5d2e 6d65 616e 2861 7869 7320 3d20  me].mean(axis = 
+00010e00: 3029 0a20 2020 2020 2020 2020 2020 2020  0).             
+00010e10: 2020 2020 2020 2020 2020 2070 726f 6265             probe
+00010e20: 203d 2022 4176 6572 6167 6564 2066 726f   = "Averaged fro
+00010e30: 6d20 2220 2b20 2220 222e 6a6f 696e 2864  m " + " ".join(d
+00010e40: 7570 5f73 7562 7365 742e 5072 6f62 6529  up_subset.Probe)
+00010e50: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+00010e60: 2020 2020 2020 2020 200a 2020 2020 2020           .      
+00010e70: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010e80: 2020 6765 6e65 5f69 6e66 6f20 3d20 6475    gene_info = du
+00010e90: 705f 7375 6273 6574 2e5c 0a20 2020 2020  p_subset.\.     
+00010ea0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010eb0: 2020 2020 2020 206c 6f63 5b3a 2c0a 2020         loc[:,.  
 00010ec0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010ed0: 2020 2020 2020 2020 2020 2020 2073 7562               sub
-00010ee0: 2e6c 6973 745f 696e 5f6c 6973 7428 0a20  .list_in_list(. 
-00010ef0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010ed0: 2020 2020 2020 2020 2020 2020 2020 6e70                np
+00010ee0: 2e61 7272 6179 287e 6e70 2e61 7272 6179  .array(~np.array
+00010ef0: 280a 2020 2020 2020 2020 2020 2020 2020  (.              
 00010f00: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010f10: 2020 2020 2020 2064 7570 5f73 7562 7365         dup_subse
-00010f20: 742e 636f 6c75 6d6e 732c 5b22 5072 6f62  t.columns,["Prob
-00010f30: 6522 5d20 2b5c 0a20 2020 2020 2020 2020  e"] +\.         
+00010f10: 2020 2020 2020 7375 622e 6c69 7374 5f69        sub.list_i
+00010f20: 6e5f 6c69 7374 280a 2020 2020 2020 2020  n_list(.        
+00010f30: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00010f40: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010f50: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010f60: 2020 206c 6973 7428 6d65 7461 2e73 616d     list(meta.sam
-00010f70: 706c 655f 6e61 6d65 2929 2929 5d2e 696c  ple_name))))].il
-00010f80: 6f63 5b30 5d2e 5c0a 2020 2020 2020 2020  oc[0].\.        
-00010f90: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010fa0: 2020 2020 2020 2020 746f 5f66 7261 6d65          to_frame
-00010fb0: 2829 2e54 0a20 2020 2020 2020 2020 2020  ().T.           
-00010fc0: 2020 2020 2020 2020 2020 2020 200a 2020               .  
+00010f50: 6475 705f 7375 6273 6574 2e63 6f6c 756d  dup_subset.colum
+00010f60: 6e73 2c5b 2250 726f 6265 225d 202b 5c0a  ns,["Probe"] +\.
+00010f70: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010f80: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00010f90: 2020 2020 2020 2020 2020 2020 6c69 7374              list
+00010fa0: 286d 6574 612e 7361 6d70 6c65 5f6e 616d  (meta.sample_nam
+00010fb0: 6529 2929 295d 2e69 6c6f 635b 305d 2e5c  e))))].iloc[0].\
+00010fc0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
 00010fd0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00010fe0: 2020 2020 2020 7072 6f62 6520 3d20 7064        probe = pd
-00010ff0: 2e44 6174 6146 7261 6d65 287b 0a20 2020  .DataFrame({.   
-00011000: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011010: 2020 2020 2020 2020 2022 5072 6f62 6522           "Probe"
-00011020: 3a20 5b70 726f 6265 5d7d 290a 2020 2020  : [probe]}).    
-00011030: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011040: 2020 2020 6461 7461 203d 2064 7570 5f6d      data = dup_m
-00011050: 6561 6e2e 746f 5f66 7261 6d65 2829 2e54  ean.to_frame().T
-00011060: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00011070: 2020 2020 2020 2020 2072 6574 7572 6e20           return 
-00011080: 7064 2e63 6f6e 6361 7428 5b70 726f 6265  pd.concat([probe
-00011090: 2c67 656e 655f 696e 666f 2e72 6573 6574  ,gene_info.reset
-000110a0: 5f69 6e64 6578 2864 726f 7020 3d20 5472  _index(drop = Tr
-000110b0: 7565 292c 6461 7461 5d2c 0a20 2020 2020  ue),data],.     
-000110c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000110d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000110e0: 2020 2020 6178 6973 203d 2031 290a 2020      axis = 1).  
-000110f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011100: 2020 0a20 2020 2020 2020 2020 2020 2020    .             
-00011110: 2020 2020 2020 2064 7570 5f6d 6561 6e73         dup_means
-00011120: 203d 205b 6765 745f 6d65 616e 2864 7570   = [get_mean(dup
-00011130: 2920 666f 7220 6475 7020 696e 2064 7570  ) for dup in dup
-00011140: 735d 0a20 2020 2020 2020 2020 2020 2020  s].             
-00011150: 2020 2020 2020 2065 7870 203d 2070 642e         exp = pd.
-00011160: 636f 6e63 6174 285b 6578 705b 7e65 7870  concat([exp[~exp
-00011170: 2e47 656e 6553 796d 626f 6c2e 6973 696e  .GeneSymbol.isin
-00011180: 2864 7570 7329 5d5d 202b 2064 7570 5f6d  (dups)]] + dup_m
-00011190: 6561 6e73 2c20 0a20 2020 2020 2020 2020  eans, .         
-000111a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000111b0: 2020 2020 2020 2020 2020 2069 676e 6f72             ignor
-000111c0: 655f 696e 6465 7820 3d20 5472 7565 290a  e_index = True).
+00010fe0: 2074 6f5f 6672 616d 6528 292e 540a 2020   to_frame().T.  
+00010ff0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011000: 2020 2020 2020 0a20 2020 2020 2020 2020        .         
+00011010: 2020 2020 2020 2020 2020 2020 2020 2070                 p
+00011020: 726f 6265 203d 2070 642e 4461 7461 4672  robe = pd.DataFr
+00011030: 616d 6528 7b0a 2020 2020 2020 2020 2020  ame({.          
+00011040: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011050: 2020 2250 726f 6265 223a 205b 7072 6f62    "Probe": [prob
+00011060: 655d 7d29 0a20 2020 2020 2020 2020 2020  e]}).           
+00011070: 2020 2020 2020 2020 2020 2020 2064 6174               dat
+00011080: 6120 3d20 6475 705f 6d65 616e 2e74 6f5f  a = dup_mean.to_
+00011090: 6672 616d 6528 292e 540a 2020 2020 2020  frame().T.      
+000110a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000110b0: 2020 7265 7475 726e 2070 642e 636f 6e63    return pd.conc
+000110c0: 6174 285b 7072 6f62 652c 6765 6e65 5f69  at([probe,gene_i
+000110d0: 6e66 6f2e 7265 7365 745f 696e 6465 7828  nfo.reset_index(
+000110e0: 6472 6f70 203d 2054 7275 6529 2c64 6174  drop = True),dat
+000110f0: 615d 2c0a 2020 2020 2020 2020 2020 2020  a],.            
+00011100: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011110: 2020 2020 2020 2020 2020 2020 2061 7869               axi
+00011120: 7320 3d20 3129 0a20 2020 2020 2020 2020  s = 1).         
+00011130: 2020 2020 2020 2020 2020 200a 2020 2020             .    
+00011140: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011150: 6475 705f 6d65 616e 7320 3d20 5b67 6574  dup_means = [get
+00011160: 5f6d 6561 6e28 6475 7029 2066 6f72 2064  _mean(dup) for d
+00011170: 7570 2069 6e20 6475 7073 5d0a 2020 2020  up in dups].    
+00011180: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011190: 6578 7020 3d20 7064 2e63 6f6e 6361 7428  exp = pd.concat(
+000111a0: 5b65 7870 5b7e 6578 702e 4765 6e65 5379  [exp[~exp.GeneSy
+000111b0: 6d62 6f6c 2e69 7369 6e28 6475 7073 295d  mbol.isin(dups)]
+000111c0: 5d20 2b20 6475 705f 6d65 616e 732c 200a  ] + dup_means, .
 000111d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000111e0: 7265 7475 726e 2065 7870 0a20 2020 2020  return exp.     
-000111f0: 2020 2020 2020 2023 6765 745f 6578 700a         #get_exp.
-00011200: 2020 2020 2020 2020 2020 2020 0a20 2020              .   
-00011210: 2020 2020 2020 2020 2065 7870 7265 7373           express
-00011220: 696f 6e20 3d20 7b6b 3a67 6574 5f65 7870  ion = {k:get_exp
-00011230: 286b 2920 666f 7220 6b20 696e 2075 6e69  (k) for k in uni
-00011240: 7175 655f 7365 7473 7d0a 2020 2020 2020  que_sets}.      
-00011250: 2020 2020 2020 0a20 2020 2020 2020 2065        .        e
-00011260: 6c73 653a 0a20 2020 2020 2020 2020 2020  lse:.           
-00011270: 2065 7870 7265 7373 696f 6e20 3d20 7365   expression = se
-00011280: 6c66 2e5c 0a20 2020 2020 2020 2020 2020  lf.\.           
-00011290: 2020 2020 2067 6574 5f64 6174 6173 6574       get_dataset
-000112a0: 5f65 7870 7265 7373 696f 6e5f 666f 725f  _expression_for_
-000112b0: 6765 6e65 7328 756e 6971 7565 5f73 6574  genes(unique_set
-000112c0: 732c 0a20 2020 2020 2020 2020 2020 2020  s,.             
-000112d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000112e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000112f0: 2020 2020 6765 6e65 7320 3d20 6765 6e65      genes = gene
-00011300: 732c 0a20 2020 2020 2020 2020 2020 2020  s,.             
+000111e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000111f0: 2020 2020 6967 6e6f 7265 5f69 6e64 6578      ignore_index
+00011200: 203d 2054 7275 6529 0a20 2020 2020 2020   = True).       
+00011210: 2020 2020 2020 2020 2072 6574 7572 6e20           return 
+00011220: 6578 700a 2020 2020 2020 2020 2020 2020  exp.            
+00011230: 2367 6574 5f65 7870 0a20 2020 2020 2020  #get_exp.       
+00011240: 2020 2020 200a 2020 2020 2020 2020 2020       .          
+00011250: 2020 6578 7072 6573 7369 6f6e 203d 207b    expression = {
+00011260: 6b3a 6765 745f 6578 7028 6b29 2066 6f72  k:get_exp(k) for
+00011270: 206b 2069 6e20 756e 6971 7565 5f73 6574   k in unique_set
+00011280: 737d 0a20 2020 2020 2020 2020 2020 200a  s}.            .
+00011290: 2020 2020 2020 2020 656c 7365 3a0a 2020          else:.  
+000112a0: 2020 2020 2020 2020 2020 6578 7072 6573            expres
+000112b0: 7369 6f6e 203d 2073 656c 662e 5c0a 2020  sion = self.\.  
+000112c0: 2020 2020 2020 2020 2020 2020 2020 6765                ge
+000112d0: 745f 6461 7461 7365 745f 6578 7072 6573  t_dataset_expres
+000112e0: 7369 6f6e 5f66 6f72 5f67 656e 6573 2875  sion_for_genes(u
+000112f0: 6e69 7175 655f 7365 7473 2c0a 2020 2020  nique_sets,.    
+00011300: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00011310: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011320: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011330: 2020 2020 6b65 6570 5f6e 6f6e 5f73 7065      keep_non_spe
-00011340: 6369 6669 6320 3d20 6b65 6570 5f6e 6f6e  cific = keep_non
-00011350: 5f73 7065 6369 6669 632c 0a20 2020 2020  _specific,.     
-00011360: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011370: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011380: 2020 2020 2020 2020 2020 2020 636f 6e73              cons
-00011390: 6f6c 6964 6174 6520 3d20 636f 6e73 6f6c  olidate = consol
-000113a0: 6964 6174 6529 0a20 2020 2020 2020 200a  idate).        .
-000113b0: 2020 2020 2020 2020 6465 7369 676e 7320          designs 
-000113c0: 3d20 7b6b 3a73 656c 662e 6d61 6b65 5f64  = {k:self.make_d
-000113d0: 6573 6967 6e28 6d65 7461 6461 7461 5b6b  esign(metadata[k
-000113e0: 5d29 2066 6f72 206b 2069 6e20 6d65 7461  ]) for k in meta
-000113f0: 6461 7461 2e6b 6579 7328 297d 0a20 2020  data.keys()}.   
-00011400: 2020 2020 2064 6174 203d 2073 656c 662e       dat = self.
-00011410: 6765 745f 6461 7461 7365 7473 5f62 795f  get_datasets_by_
-00011420: 6964 7328 756e 6971 7565 5f73 6574 7329  ids(unique_sets)
-00011430: 0a20 2020 2020 2020 2064 6566 2070 6163  .        def pac
-00011440: 6b5f 6461 7461 2869 293a 0a20 2020 2020  k_data(i):.     
-00011450: 2020 2020 2020 2064 6174 6173 6574 203d         dataset =
-00011460: 2064 6174 6173 6574 735b 695d 0a20 2020   datasets[i].   
-00011470: 2020 2020 2020 2020 2070 6163 6b65 645f           packed_
-00011480: 696e 666f 203d 207b 0a20 2020 2020 2020  info = {.       
-00011490: 2020 2020 2020 2020 2022 6465 7369 676e           "design
-000114a0: 223a 6465 7369 676e 735b 6461 7461 7365  ":designs[datase
-000114b0: 745d 2e63 6f70 7928 292c 0a20 2020 2020  t].copy(),.     
-000114c0: 2020 2020 2020 2020 2020 2022 6578 7022             "exp"
-000114d0: 3a20 6578 7072 6573 7369 6f6e 5b64 6174  : expression[dat
-000114e0: 6173 6574 5d2e 636f 7079 2829 2c0a 2020  aset].copy(),.  
-000114f0: 2020 2020 2020 2020 2020 2020 2020 2264                "d
-00011500: 6174 223a 2064 6174 5b28 6461 742e 6578  at": dat[(dat.ex
-00011510: 7065 7269 6d65 6e74 5f49 4420 3d3d 2064  periment_ID == d
-00011520: 6174 6173 6574 2920 7c20 5c0a 2020 2020  ataset) | \.    
-00011530: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011540: 2020 2020 2020 2028 6461 742e 6578 7065         (dat.expe
-00011550: 7269 6d65 6e74 5f73 686f 7274 5f6e 616d  riment_short_nam
-00011560: 6520 3d3d 2064 6174 6173 6574 295d 2e63  e == dataset)].c
-00011570: 6f70 7928 292e 7265 7365 745f 696e 6465  opy().reset_inde
-00011580: 7828 292c 0a20 2020 2020 2020 2020 2020  x(),.           
-00011590: 2020 2020 2022 7265 7375 6c74 5f73 6574       "result_set
-000115a0: 223a 204e 6f6e 652c 0a20 2020 2020 2020  ": None,.       
-000115b0: 2020 2020 2020 2020 2022 636f 6e74 7261           "contra
-000115c0: 7374 7322 3a20 4e6f 6e65 0a20 2020 2020  sts": None.     
-000115d0: 2020 2020 2020 2020 2020 207d 0a20 2020             }.   
-000115e0: 2020 2020 2020 2020 200a 2020 2020 2020           .      
-000115f0: 2020 2020 2020 0a20 2020 2020 2020 2020        .         
-00011600: 2020 2069 6620 7265 7375 6c74 5f73 6574     if result_set
-00011610: 7320 6973 206e 6f74 204e 6f6e 653a 0a20  s is not None:. 
-00011620: 2020 2020 2020 2020 2020 2020 2020 2070                 p
-00011630: 6163 6b65 645f 696e 666f 5b27 7265 7375  acked_info['resu
-00011640: 6c74 5f73 6574 275d 203d 2072 6573 756c  lt_set'] = resul
-00011650: 745f 7365 7473 5b69 5d0a 2020 2020 2020  t_sets[i].      
-00011660: 2020 2020 2020 6966 2063 6f6e 7472 6173        if contras
-00011670: 7473 2069 7320 6e6f 7420 4e6f 6e65 3a0a  ts is not None:.
-00011680: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011690: 7061 636b 6564 5f69 6e66 6f5b 2763 6f6e  packed_info['con
-000116a0: 7472 6173 7473 275d 203d 2063 6f6e 7472  trasts'] = contr
-000116b0: 6173 7473 5b69 5d0a 2020 2020 2020 2020  asts[i].        
-000116c0: 2020 2020 2320 7265 6f72 6465 7269 6e67      # reordering
-000116d0: 2074 6f20 6d61 7463 6820 6578 7072 6573   to match expres
-000116e0: 7369 6f6e 2f6d 6574 6164 6174 6120 6e6f  sion/metadata no
-000116f0: 206c 6f6e 6765 7220 6e65 6365 7361 7272   longer necesarr
-00011700: 790a 2020 2020 2020 2020 2020 2020 0a20  y.            . 
-00011710: 2020 2020 2020 2020 2020 2064 6966 6620             diff 
-00011720: 3d20 7365 6c66 2e67 6574 5f64 6174 6173  = self.get_datas
-00011730: 6574 5f64 6966 6665 7265 6e74 6961 6c5f  et_differential_
-00011740: 6578 7072 6573 7369 6f6e 5f61 6e61 6c79  expression_analy
-00011750: 7365 7328 6461 7461 7365 7429 0a20 2020  ses(dataset).   
-00011760: 2020 2020 2020 2020 200a 2020 2020 2020           .      
-00011770: 2020 2020 2020 6966 2072 6573 756c 745f        if result_
-00011780: 7365 7473 2069 7320 6e6f 7420 4e6f 6e65  sets is not None
-00011790: 3a0a 2020 2020 2020 2020 2020 2020 2020  :.              
-000117a0: 2020 6765 6e65 5f69 6e66 6f20 3d20 7061    gene_info = pa
-000117b0: 636b 6564 5f69 6e66 6f5b 2765 7870 275d  cked_info['exp']
-000117c0: 2e5c 0a20 2020 2020 2020 2020 2020 2020  .\.             
-000117d0: 2020 2020 2020 2063 6f6c 756d 6e73 5b5b         columns[[
-000117e0: 6e6f 7420 7820 0a20 2020 2020 2020 2020  not x .         
-000117f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011800: 2020 2020 666f 7220 7820 696e 2073 7562      for x in sub
-00011810: 2e6c 6973 745f 696e 5f6c 6973 7428 7061  .list_in_list(pa
-00011820: 636b 6564 5f69 6e66 6f5b 2765 7870 275d  cked_info['exp']
-00011830: 2e63 6f6c 756d 6e73 2c0a 2020 2020 2020  .columns,.      
-00011840: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011850: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011860: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011870: 2070 6163 6b65 645f 696e 666f 5b27 6465   packed_info['de
-00011880: 7369 676e 275d 2e69 6e64 6578 295d 5d0a  sign'].index)]].
+00011320: 2020 2020 2020 2020 2020 2020 2067 656e               gen
+00011330: 6573 203d 2067 656e 6573 2c0a 2020 2020  es = genes,.    
+00011340: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011350: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011360: 2020 2020 2020 2020 2020 2020 206b 6565               kee
+00011370: 705f 6e6f 6e5f 7370 6563 6966 6963 203d  p_non_specific =
+00011380: 206b 6565 705f 6e6f 6e5f 7370 6563 6966   keep_non_specif
+00011390: 6963 2c0a 2020 2020 2020 2020 2020 2020  ic,.            
+000113a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000113b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000113c0: 2020 2020 2063 6f6e 736f 6c69 6461 7465       consolidate
+000113d0: 203d 2063 6f6e 736f 6c69 6461 7465 290a   = consolidate).
+000113e0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+000113f0: 2064 6573 6967 6e73 203d 207b 6b3a 7365   designs = {k:se
+00011400: 6c66 2e6d 616b 655f 6465 7369 676e 286d  lf.make_design(m
+00011410: 6574 6164 6174 615b 6b5d 2920 666f 7220  etadata[k]) for 
+00011420: 6b20 696e 206d 6574 6164 6174 612e 6b65  k in metadata.ke
+00011430: 7973 2829 7d0a 2020 2020 2020 2020 6461  ys()}.        da
+00011440: 7420 3d20 7365 6c66 2e67 6574 5f64 6174  t = self.get_dat
+00011450: 6173 6574 735f 6279 5f69 6473 2875 6e69  asets_by_ids(uni
+00011460: 7175 655f 7365 7473 290a 2020 2020 2020  que_sets).      
+00011470: 2020 6465 6620 7061 636b 5f64 6174 6128    def pack_data(
+00011480: 6929 3a0a 2020 2020 2020 2020 2020 2020  i):.            
+00011490: 6461 7461 7365 7420 3d20 6461 7461 7365  dataset = datase
+000114a0: 7473 5b69 5d0a 2020 2020 2020 2020 2020  ts[i].          
+000114b0: 2020 7061 636b 6564 5f69 6e66 6f20 3d20    packed_info = 
+000114c0: 7b0a 2020 2020 2020 2020 2020 2020 2020  {.              
+000114d0: 2020 2264 6573 6967 6e22 3a64 6573 6967    "design":desig
+000114e0: 6e73 5b64 6174 6173 6574 5d2e 636f 7079  ns[dataset].copy
+000114f0: 2829 2c0a 2020 2020 2020 2020 2020 2020  (),.            
+00011500: 2020 2020 2265 7870 223a 2065 7870 7265      "exp": expre
+00011510: 7373 696f 6e5b 6461 7461 7365 745d 2e63  ssion[dataset].c
+00011520: 6f70 7928 292c 0a20 2020 2020 2020 2020  opy(),.         
+00011530: 2020 2020 2020 2022 6461 7422 3a20 6461         "dat": da
+00011540: 745b 2864 6174 2e65 7870 6572 696d 656e  t[(dat.experimen
+00011550: 745f 4944 203d 3d20 6461 7461 7365 7429  t_ID == dataset)
+00011560: 207c 205c 0a20 2020 2020 2020 2020 2020   | \.           
+00011570: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011580: 2864 6174 2e65 7870 6572 696d 656e 745f  (dat.experiment_
+00011590: 7368 6f72 745f 6e61 6d65 203d 3d20 6461  short_name == da
+000115a0: 7461 7365 7429 5d2e 636f 7079 2829 2e72  taset)].copy().r
+000115b0: 6573 6574 5f69 6e64 6578 2829 2c0a 2020  eset_index(),.  
+000115c0: 2020 2020 2020 2020 2020 2020 2020 2272                "r
+000115d0: 6573 756c 745f 7365 7422 3a20 4e6f 6e65  esult_set": None
+000115e0: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+000115f0: 2020 2263 6f6e 7472 6173 7473 223a 204e    "contrasts": N
+00011600: 6f6e 650a 2020 2020 2020 2020 2020 2020  one.            
+00011610: 2020 2020 7d0a 2020 2020 2020 2020 2020      }.          
+00011620: 2020 0a20 2020 2020 2020 2020 2020 200a    .            .
+00011630: 2020 2020 2020 2020 2020 2020 6966 2072              if r
+00011640: 6573 756c 745f 7365 7473 2069 7320 6e6f  esult_sets is no
+00011650: 7420 4e6f 6e65 3a0a 2020 2020 2020 2020  t None:.        
+00011660: 2020 2020 2020 2020 7061 636b 6564 5f69          packed_i
+00011670: 6e66 6f5b 2772 6573 756c 745f 7365 7427  nfo['result_set'
+00011680: 5d20 3d20 7265 7375 6c74 5f73 6574 735b  ] = result_sets[
+00011690: 695d 0a20 2020 2020 2020 2020 2020 2069  i].            i
+000116a0: 6620 636f 6e74 7261 7374 7320 6973 206e  f contrasts is n
+000116b0: 6f74 204e 6f6e 653a 0a20 2020 2020 2020  ot None:.       
+000116c0: 2020 2020 2020 2020 2070 6163 6b65 645f           packed_
+000116d0: 696e 666f 5b27 636f 6e74 7261 7374 7327  info['contrasts'
+000116e0: 5d20 3d20 636f 6e74 7261 7374 735b 695d  ] = contrasts[i]
+000116f0: 0a20 2020 2020 2020 2020 2020 2023 2072  .            # r
+00011700: 656f 7264 6572 696e 6720 746f 206d 6174  eordering to mat
+00011710: 6368 2065 7870 7265 7373 696f 6e2f 6d65  ch expression/me
+00011720: 7461 6461 7461 206e 6f20 6c6f 6e67 6572  tadata no longer
+00011730: 206e 6563 6573 6172 7279 0a20 2020 2020   necesarry.     
+00011740: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+00011750: 2020 2020 6469 6666 203d 2073 656c 662e      diff = self.
+00011760: 6765 745f 6461 7461 7365 745f 6469 6666  get_dataset_diff
+00011770: 6572 656e 7469 616c 5f65 7870 7265 7373  erential_express
+00011780: 696f 6e5f 616e 616c 7973 6573 2864 6174  ion_analyses(dat
+00011790: 6173 6574 290a 2020 2020 2020 2020 2020  aset).          
+000117a0: 2020 0a20 2020 2020 2020 2020 2020 2069    .            i
+000117b0: 6620 7265 7375 6c74 5f73 6574 7320 6973  f result_sets is
+000117c0: 206e 6f74 204e 6f6e 653a 0a20 2020 2020   not None:.     
+000117d0: 2020 2020 2020 2020 2020 2067 656e 655f             gene_
+000117e0: 696e 666f 203d 2070 6163 6b65 645f 696e  info = packed_in
+000117f0: 666f 5b27 6578 7027 5d2e 5c0a 2020 2020  fo['exp'].\.    
+00011800: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011810: 636f 6c75 6d6e 735b 5b6e 6f74 2078 200a  columns[[not x .
+00011820: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011830: 2020 2020 2020 2020 2020 2020 2066 6f72               for
+00011840: 2078 2069 6e20 7375 622e 6c69 7374 5f69   x in sub.list_i
+00011850: 6e5f 6c69 7374 2870 6163 6b65 645f 696e  n_list(packed_in
+00011860: 666f 5b27 6578 7027 5d2e 636f 6c75 6d6e  fo['exp'].column
+00011870: 732c 0a20 2020 2020 2020 2020 2020 2020  s,.             
+00011880: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00011890: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000118a0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-000118b0: 2063 6f6e 7320 3d20 4e6f 6e65 2069 6620   cons = None if 
-000118c0: 636f 6e74 7261 7374 7320 6973 204e 6f6e  contrasts is Non
-000118d0: 6520 656c 7365 2063 6f6e 7472 6173 7473  e else contrasts
-000118e0: 5b69 5d0a 2020 2020 2020 2020 2020 2020  [i].            
-000118f0: 2020 2020 0a20 2020 2020 2020 2020 2020      .           
-00011900: 2020 2020 2072 656c 6576 616e 7420 3d20       relevant = 
-00011910: 7365 6c66 2e5f 5f73 7562 7365 745f 6661  self.__subset_fa
-00011920: 6374 6f72 5f76 616c 7565 7328 7061 636b  ctor_values(pack
-00011930: 6564 5f69 6e66 6f5b 2764 6573 6967 6e27  ed_info['design'
-00011940: 5d2e 5c0a 2020 2020 2020 2020 2020 2020  ].\.            
-00011950: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011960: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011970: 2020 2020 2020 2020 2020 2066 6163 746f             facto
-00011980: 725f 7661 6c75 6573 2c0a 2020 2020 2020  r_values,.      
+000118a0: 2020 2020 2020 2020 2020 7061 636b 6564            packed
+000118b0: 5f69 6e66 6f5b 2764 6573 6967 6e27 5d2e  _info['design'].
+000118c0: 696e 6465 7829 5d5d 0a20 2020 2020 2020  index)]].       
+000118d0: 2020 2020 2020 2020 200a 2020 2020 2020           .      
+000118e0: 2020 2020 2020 2020 2020 636f 6e73 203d            cons =
+000118f0: 204e 6f6e 6520 6966 2063 6f6e 7472 6173   None if contras
+00011900: 7473 2069 7320 4e6f 6e65 2065 6c73 6520  ts is None else 
+00011910: 636f 6e74 7261 7374 735b 695d 0a20 2020  contrasts[i].   
+00011920: 2020 2020 2020 2020 2020 2020 200a 2020               .  
+00011930: 2020 2020 2020 2020 2020 2020 2020 7265                re
+00011940: 6c65 7661 6e74 203d 2073 656c 662e 5f5f  levant = self.__
+00011950: 7375 6273 6574 5f66 6163 746f 725f 7661  subset_factor_va
+00011960: 6c75 6573 2870 6163 6b65 645f 696e 666f  lues(packed_info
+00011970: 5b27 6465 7369 676e 275d 2e5c 0a20 2020  ['design'].\.   
+00011980: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00011990: 2020 2020 2020 2020 2020 2020 2020 2020                  
 000119a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000119b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000119c0: 2064 6966 662c 0a20 2020 2020 2020 2020   diff,.         
+000119b0: 2020 2020 6661 6374 6f72 5f76 616c 7565      factor_value
+000119c0: 732c 0a20 2020 2020 2020 2020 2020 2020  s,.             
 000119d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
 000119e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000119f0: 2020 2020 2020 2020 2020 2020 2020 7265                re
-00011a00: 7375 6c74 5f73 6574 735b 695d 2c0a 2020  sult_sets[i],.  
+000119f0: 2020 2020 2020 2020 2020 6469 6666 2c0a            diff,.
+00011a00: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00011a10: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00011a20: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011a30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011a40: 2020 2020 2063 6f6e 7329 0a20 2020 2020       cons).     
-00011a50: 2020 2020 2020 2020 2020 2070 6163 6b65             packe
-00011a60: 645f 696e 666f 5b27 6465 7369 676e 275d  d_info['design']
-00011a70: 203d 2070 6163 6b65 645f 696e 666f 5b27   = packed_info['
-00011a80: 6465 7369 676e 275d 5b72 656c 6576 616e  design'][relevan
-00011a90: 745d 0a20 2020 2020 2020 2020 2020 2020  t].             
-00011aa0: 2020 2070 6163 6b65 645f 696e 666f 5b27     packed_info['
-00011ab0: 6578 7027 5d20 3d20 7061 636b 6564 5f69  exp'] = packed_i
-00011ac0: 6e66 6f5b 2765 7870 275d 5b67 656e 655f  nfo['exp'][gene_
-00011ad0: 696e 666f 2e61 7070 656e 6428 7061 636b  info.append(pack
-00011ae0: 6564 5f69 6e66 6f5b 2764 6573 6967 6e27  ed_info['design'
-00011af0: 5d2e 696e 6465 7829 5d0a 2020 2020 2020  ].index)].      
-00011b00: 2020 2020 2020 2020 2020 0a20 2020 2020            .     
-00011b10: 2020 2020 2020 2020 2020 2020 2020 200a                 .
-00011b20: 2020 2020 2020 2020 2020 2020 7265 7475              retu
-00011b30: 726e 2070 6163 6b65 645f 696e 666f 0a20  rn packed_info. 
-00011b40: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-00011b50: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
-00011b60: 2020 2320 7061 636b 6564 5f64 6174 6120    # packed_data 
-00011b70: 3d20 5b70 6163 6b5f 6461 7461 2869 2920  = [pack_data(i) 
-00011b80: 666f 7220 6920 696e 2072 616e 6765 286c  for i in range(l
-00011b90: 656e 2864 6174 6173 6574 7329 295d 0a20  en(datasets))]. 
-00011ba0: 2020 2020 2020 2070 6163 6b65 645f 6461         packed_da
-00011bb0: 7461 203d 205b 7061 636b 5f64 6174 6128  ta = [pack_data(
-00011bc0: 6929 2066 6f72 2069 2069 6e20 7261 6e67  i) for i in rang
-00011bd0: 6528 6c65 6e28 6461 7461 7365 7473 2929  e(len(datasets))
-00011be0: 5d0a 2020 2020 2020 2020 6b65 7973 203d  ].        keys =
-00011bf0: 205b 7374 7228 785b 2764 6174 275d 2e65   [str(x['dat'].e
-00011c00: 7870 6572 696d 656e 745f 4944 5b30 5d29  xperiment_ID[0])
-00011c10: 2066 6f72 2078 2069 6e20 7061 636b 6564   for x in packed
-00011c20: 5f64 6174 615d 0a0a 2020 2020 2020 2020  _data]..        
-00011c30: 6966 2072 6573 756c 745f 7365 7473 2069  if result_sets i
-00011c40: 7320 6e6f 7420 4e6f 6e65 3a0a 2020 2020  s not None:.    
-00011c50: 2020 2020 2020 2020 6b65 7973 203d 205b          keys = [
-00011c60: 6b65 7973 5b69 5d20 2b20 225f 2220 2b20  keys[i] + "_" + 
-00011c70: 7374 7228 7265 7375 6c74 5f73 6574 735b  str(result_sets[
-00011c80: 695d 2920 666f 7220 6920 696e 2072 616e  i]) for i in ran
-00011c90: 6765 286c 656e 2864 6174 6173 6574 7329  ge(len(datasets)
-00011ca0: 295d 0a20 2020 2020 2020 2020 2020 2069  )].            i
-00011cb0: 6620 636f 6e74 7261 7374 7320 6973 206e  f contrasts is n
-00011cc0: 6f74 204e 6f6e 653a 0a20 2020 2020 2020  ot None:.       
-00011cd0: 2020 2020 2020 2020 206b 6579 7320 3d20           keys = 
-00011ce0: 5b6b 6579 735b 695d 202b 2022 5f22 202b  [keys[i] + "_" +
-00011cf0: 2073 7472 2863 6f6e 7472 6173 7473 5b69   str(contrasts[i
-00011d00: 5d29 2066 6f72 2069 2069 6e20 7261 6e67  ]) for i in rang
-00011d10: 6528 6c65 6e28 6461 7461 7365 7473 2929  e(len(datasets))
-00011d20: 5d20 0a20 2020 2020 2020 200a 2020 2020  ] .        .    
-00011d30: 2020 2020 7061 636b 6564 5f64 6174 6120      packed_data 
-00011d40: 3d20 7b6b 6579 735b 695d 3a70 6163 6b65  = {keys[i]:packe
-00011d50: 645f 6461 7461 5b69 5d20 666f 7220 6920  d_data[i] for i 
-00011d60: 696e 2072 616e 6765 286c 656e 2864 6174  in range(len(dat
-00011d70: 6173 6574 7329 297d 0a20 2020 2020 2020  asets))}.       
-00011d80: 200a 2020 2020 2020 2020 0a20 2020 2020   .        .     
-00011d90: 2020 200a 2020 2020 2020 2020 6966 206f     .        if o
-00011da0: 7574 7075 745f 7479 7065 203d 3d20 2761  utput_type == 'a
-00011db0: 6e6e 6461 7461 273a 0a20 2020 2020 2020  nndata':.       
-00011dc0: 2020 2020 2064 6566 206d 616b 655f 616e       def make_an
-00011dd0: 6e64 6174 6128 7061 636b 293a 0a20 2020  ndata(pack):.   
-00011de0: 2020 2020 2020 2020 2020 2020 2070 6163               pac
-00011df0: 6b5b 2765 7870 275d 2e69 6e64 6578 203d  k['exp'].index =
-00011e00: 2070 6163 6b5b 2765 7870 275d 5b27 5072   pack['exp']['Pr
-00011e10: 6f62 6527 5d0a 2020 2020 2020 2020 2020  obe'].          
-00011e20: 2020 2020 2020 7472 793a 200a 2020 2020        try: .    
-00011e30: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011e40: 6765 6e65 5f64 6174 6120 3d20 7061 636b  gene_data = pack
-00011e50: 5b27 6578 7027 5d5b 5b27 4765 6e65 5379  ['exp'][['GeneSy
-00011e60: 6d62 6f6c 272c 2027 4e43 4249 6964 275d  mbol', 'NCBIid']
-00011e70: 5d0a 2020 2020 2020 2020 2020 2020 2020  ].              
-00011e80: 2020 6578 6365 7074 204b 6579 4572 726f    except KeyErro
-00011e90: 723a 0a20 2020 2020 2020 2020 2020 2020  r:.             
-00011ea0: 2020 2020 2020 2077 6172 6e69 6e67 732e         warnings.
-00011eb0: 7761 726e 2822 5741 524e 494e 473a 204f  warn("WARNING: O
-00011ec0: 6e65 206f 7220 6d6f 7265 2067 656e 6520  ne or more gene 
-00011ed0: 6465 7363 7269 7074 696f 6e73 2061 7265  descriptions are
-00011ee0: 206d 6973 7369 6e67 2069 6e20 4578 7072   missing in Expr
-00011ef0: 6573 7369 6f6e 2074 6162 6c65 2229 0a20  ession table"). 
-00011f00: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011f10: 2020 2067 656e 655f 6461 7461 203d 204e     gene_data = N
-00011f20: 6f6e 650a 2020 2020 2020 2020 2020 2020  one.            
-00011f30: 2020 2020 0a20 2020 2020 2020 2020 2020      .           
-00011f40: 2020 2020 206d 6461 203d 207b 0a20 2020       mda = {.   
-00011f50: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00011f60: 2027 7469 746c 6527 3a20 7061 636b 5b27   'title': pack['
-00011f70: 6461 7427 5d2e 6578 7065 7269 6d65 6e74  dat'].experiment
-00011f80: 5f6e 616d 655b 305d 2c0a 2020 2020 2020  _name[0],.      
-00011f90: 2020 2020 2020 2020 2020 2020 2020 2761                'a
-00011fa0: 6273 7472 6163 7427 3a20 7061 636b 5b27  bstract': pack['
-00011fb0: 6461 7427 5d2e 6578 7065 7269 6d65 6e74  dat'].experiment
-00011fc0: 5f64 6573 6372 6970 7469 6f6e 5b30 5d2c  _description[0],
-00011fd0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00011fe0: 2020 2020 2027 7572 6c27 3a20 2768 7474       'url': 'htt
-00011ff0: 7073 3a2f 2f67 656d 6d61 2e6d 736c 2e75  ps://gemma.msl.u
-00012000: 6263 2e63 612f 6578 7072 6573 7369 6f6e  bc.ca/expression
-00012010: 4578 7065 7269 6d65 6e74 2f73 686f 7745  Experiment/showE
-00012020: 7870 7265 7373 696f 6e45 7870 6572 696d  xpressionExperim
-00012030: 656e 742e 6874 6d6c 3f69 643d 272b 205c  ent.html?id='+ \
-00012040: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00012050: 2020 2020 2020 2020 2073 7472 2870 6163           str(pac
-00012060: 6b5b 2764 6174 275d 2e65 7870 6572 696d  k['dat'].experim
-00012070: 656e 745f 4944 5b30 5d29 2c0a 2020 2020  ent_ID[0]),.    
+00011a30: 2020 2020 2020 2072 6573 756c 745f 7365         result_se
+00011a40: 7473 5b69 5d2c 0a20 2020 2020 2020 2020  ts[i],.         
+00011a50: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011a60: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011a70: 2020 2020 2020 2020 2020 2020 2020 636f                co
+00011a80: 6e73 290a 2020 2020 2020 2020 2020 2020  ns).            
+00011a90: 2020 2020 7061 636b 6564 5f69 6e66 6f5b      packed_info[
+00011aa0: 2764 6573 6967 6e27 5d20 3d20 7061 636b  'design'] = pack
+00011ab0: 6564 5f69 6e66 6f5b 2764 6573 6967 6e27  ed_info['design'
+00011ac0: 5d5b 7265 6c65 7661 6e74 5d0a 2020 2020  ][relevant].    
+00011ad0: 2020 2020 2020 2020 2020 2020 7061 636b              pack
+00011ae0: 6564 5f69 6e66 6f5b 2765 7870 275d 203d  ed_info['exp'] =
+00011af0: 2070 6163 6b65 645f 696e 666f 5b27 6578   packed_info['ex
+00011b00: 7027 5d5b 6765 6e65 5f69 6e66 6f2e 6170  p'][gene_info.ap
+00011b10: 7065 6e64 2870 6163 6b65 645f 696e 666f  pend(packed_info
+00011b20: 5b27 6465 7369 676e 275d 2e69 6e64 6578  ['design'].index
+00011b30: 295d 0a20 2020 2020 2020 2020 2020 2020  )].             
+00011b40: 2020 200a 2020 2020 2020 2020 2020 2020     .            
+00011b50: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00011b60: 2020 2020 2072 6574 7572 6e20 7061 636b       return pack
+00011b70: 6564 5f69 6e66 6f0a 2020 2020 2020 2020  ed_info.        
+00011b80: 0a20 2020 2020 2020 200a 2020 2020 2020  .        .      
+00011b90: 2020 0a20 2020 2020 2020 2023 2070 6163    .        # pac
+00011ba0: 6b65 645f 6461 7461 203d 205b 7061 636b  ked_data = [pack
+00011bb0: 5f64 6174 6128 6929 2066 6f72 2069 2069  _data(i) for i i
+00011bc0: 6e20 7261 6e67 6528 6c65 6e28 6461 7461  n range(len(data
+00011bd0: 7365 7473 2929 5d0a 2020 2020 2020 2020  sets))].        
+00011be0: 7061 636b 6564 5f64 6174 6120 3d20 5b70  packed_data = [p
+00011bf0: 6163 6b5f 6461 7461 2869 2920 666f 7220  ack_data(i) for 
+00011c00: 6920 696e 2072 616e 6765 286c 656e 2864  i in range(len(d
+00011c10: 6174 6173 6574 7329 295d 0a20 2020 2020  atasets))].     
+00011c20: 2020 206b 6579 7320 3d20 5b73 7472 2878     keys = [str(x
+00011c30: 5b27 6461 7427 5d2e 6578 7065 7269 6d65  ['dat'].experime
+00011c40: 6e74 5f49 445b 305d 2920 666f 7220 7820  nt_ID[0]) for x 
+00011c50: 696e 2070 6163 6b65 645f 6461 7461 5d0a  in packed_data].
+00011c60: 0a20 2020 2020 2020 2069 6620 7265 7375  .        if resu
+00011c70: 6c74 5f73 6574 7320 6973 206e 6f74 204e  lt_sets is not N
+00011c80: 6f6e 653a 0a20 2020 2020 2020 2020 2020  one:.           
+00011c90: 206b 6579 7320 3d20 5b6b 6579 735b 695d   keys = [keys[i]
+00011ca0: 202b 2022 5f22 202b 2073 7472 2872 6573   + "_" + str(res
+00011cb0: 756c 745f 7365 7473 5b69 5d29 2066 6f72  ult_sets[i]) for
+00011cc0: 2069 2069 6e20 7261 6e67 6528 6c65 6e28   i in range(len(
+00011cd0: 6461 7461 7365 7473 2929 5d0a 2020 2020  datasets))].    
+00011ce0: 2020 2020 2020 2020 6966 2063 6f6e 7472          if contr
+00011cf0: 6173 7473 2069 7320 6e6f 7420 4e6f 6e65  asts is not None
+00011d00: 3a0a 2020 2020 2020 2020 2020 2020 2020  :.              
+00011d10: 2020 6b65 7973 203d 205b 6b65 7973 5b69    keys = [keys[i
+00011d20: 5d20 2b20 225f 2220 2b20 7374 7228 636f  ] + "_" + str(co
+00011d30: 6e74 7261 7374 735b 695d 2920 666f 7220  ntrasts[i]) for 
+00011d40: 6920 696e 2072 616e 6765 286c 656e 2864  i in range(len(d
+00011d50: 6174 6173 6574 7329 295d 200a 2020 2020  atasets))] .    
+00011d60: 2020 2020 0a20 2020 2020 2020 2070 6163      .        pac
+00011d70: 6b65 645f 6461 7461 203d 207b 6b65 7973  ked_data = {keys
+00011d80: 5b69 5d3a 7061 636b 6564 5f64 6174 615b  [i]:packed_data[
+00011d90: 695d 2066 6f72 2069 2069 6e20 7261 6e67  i] for i in rang
+00011da0: 6528 6c65 6e28 6461 7461 7365 7473 2929  e(len(datasets))
+00011db0: 7d0a 2020 2020 2020 2020 0a20 2020 2020  }.        .     
+00011dc0: 2020 200a 2020 2020 2020 2020 0a20 2020     .        .   
+00011dd0: 2020 2020 2069 6620 6f75 7470 7574 5f74       if output_t
+00011de0: 7970 6520 3d3d 2027 616e 6e64 6174 6127  ype == 'anndata'
+00011df0: 3a0a 2020 2020 2020 2020 2020 2020 6465  :.            de
+00011e00: 6620 6d61 6b65 5f61 6e6e 6461 7461 2870  f make_anndata(p
+00011e10: 6163 6b29 3a0a 2020 2020 2020 2020 2020  ack):.          
+00011e20: 2020 2020 2020 7061 636b 5b27 6578 7027        pack['exp'
+00011e30: 5d2e 696e 6465 7820 3d20 7061 636b 5b27  ].index = pack['
+00011e40: 6578 7027 5d5b 2750 726f 6265 275d 0a20  exp']['Probe']. 
+00011e50: 2020 2020 2020 2020 2020 2020 2020 2074                 t
+00011e60: 7279 3a20 0a20 2020 2020 2020 2020 2020  ry: .           
+00011e70: 2020 2020 2020 2020 2067 656e 655f 6461           gene_da
+00011e80: 7461 203d 2070 6163 6b5b 2765 7870 275d  ta = pack['exp']
+00011e90: 5b5b 2747 656e 6553 796d 626f 6c27 2c20  [['GeneSymbol', 
+00011ea0: 274e 4342 4969 6427 5d5d 0a20 2020 2020  'NCBIid']].     
+00011eb0: 2020 2020 2020 2020 2020 2065 7863 6570             excep
+00011ec0: 7420 4b65 7945 7272 6f72 3a0a 2020 2020  t KeyError:.    
+00011ed0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00011ee0: 7761 726e 696e 6773 2e77 6172 6e28 2257  warnings.warn("W
+00011ef0: 4152 4e49 4e47 3a20 4f6e 6520 6f72 206d  ARNING: One or m
+00011f00: 6f72 6520 6765 6e65 2064 6573 6372 6970  ore gene descrip
+00011f10: 7469 6f6e 7320 6172 6520 6d69 7373 696e  tions are missin
+00011f20: 6720 696e 2045 7870 7265 7373 696f 6e20  g in Expression 
+00011f30: 7461 626c 6522 290a 2020 2020 2020 2020  table").        
+00011f40: 2020 2020 2020 2020 2020 2020 6765 6e65              gene
+00011f50: 5f64 6174 6120 3d20 4e6f 6e65 0a20 2020  _data = None.   
+00011f60: 2020 2020 2020 2020 2020 2020 200a 2020               .  
+00011f70: 2020 2020 2020 2020 2020 2020 2020 6d64                md
+00011f80: 6120 3d20 7b0a 2020 2020 2020 2020 2020  a = {.          
+00011f90: 2020 2020 2020 2020 2020 2774 6974 6c65            'title
+00011fa0: 273a 2070 6163 6b5b 2764 6174 275d 2e65  ': pack['dat'].e
+00011fb0: 7870 6572 696d 656e 745f 6e61 6d65 5b30  xperiment_name[0
+00011fc0: 5d2c 0a20 2020 2020 2020 2020 2020 2020  ],.             
+00011fd0: 2020 2020 2020 2027 6162 7374 7261 6374         'abstract
+00011fe0: 273a 2070 6163 6b5b 2764 6174 275d 2e65  ': pack['dat'].e
+00011ff0: 7870 6572 696d 656e 745f 6465 7363 7269  xperiment_descri
+00012000: 7074 696f 6e5b 305d 2c0a 2020 2020 2020  ption[0],.      
+00012010: 2020 2020 2020 2020 2020 2020 2020 2775                'u
+00012020: 726c 273a 2027 6874 7470 733a 2f2f 6765  rl': 'https://ge
+00012030: 6d6d 612e 6d73 6c2e 7562 632e 6361 2f65  mma.msl.ubc.ca/e
+00012040: 7870 7265 7373 696f 6e45 7870 6572 696d  xpressionExperim
+00012050: 656e 742f 7368 6f77 4578 7072 6573 7369  ent/showExpressi
+00012060: 6f6e 4578 7065 7269 6d65 6e74 2e68 746d  onExperiment.htm
+00012070: 6c3f 6964 3d27 2b20 5c0a 2020 2020 2020  l?id='+ \.      
 00012080: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00012090: 2764 6174 6162 6173 6527 3a20 7061 636b  'database': pack
-000120a0: 5b27 6461 7427 5d2e 6578 7065 7269 6d65  ['dat'].experime
-000120b0: 6e74 5f64 6174 6162 6173 655b 305d 2c0a  nt_database[0],.
-000120c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000120d0: 2020 2020 2761 6363 6573 696f 6e27 3a20      'accesion': 
-000120e0: 2070 6163 6b5b 2764 6174 275d 2e65 7870   pack['dat'].exp
-000120f0: 6572 696d 656e 745f 6163 6365 7373 696f  eriment_accessio
-00012100: 6e5b 305d 2c0a 2020 2020 2020 2020 2020  n[0],.          
-00012110: 2020 2020 2020 2020 2020 2247 656d 6d61            "Gemma
-00012120: 5175 616c 6974 7953 636f 7265 223a 2020  QualityScore":  
-00012130: 7061 636b 5b27 6461 7427 5d2e 6765 6571  pack['dat'].geeq
-00012140: 5f71 5f73 636f 7265 5b30 5d2c 0a20 2020  _q_score[0],.   
-00012150: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00012160: 2022 4765 6d6d 6153 7569 7461 6269 6c69   "GemmaSuitabili
-00012170: 7479 5363 6f72 6522 3a20 2070 6163 6b5b  tyScore":  pack[
-00012180: 2764 6174 275d 2e67 6565 715f 735f 7363  'dat'].geeq_s_sc
-00012190: 6f72 655b 305d 2c0a 2020 2020 2020 2020  ore[0],.        
-000121a0: 2020 2020 2020 2020 2020 2020 2274 6178              "tax
-000121b0: 6f6e 223a 2020 7061 636b 5b27 6461 7427  on":  pack['dat'
-000121c0: 5d2e 7461 786f 6e5f 6e61 6d65 5b30 5d0a  ].taxon_name[0].
-000121d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000121e0: 2020 2020 7d0a 2020 2020 2020 2020 2020      }.          
-000121f0: 2020 2020 2020 0a20 2020 2020 2020 2020        .         
-00012200: 2020 2020 2020 2065 7870 203d 2070 6163         exp = pac
-00012210: 6b5b 2765 7870 275d 5b70 6163 6b5b 2764  k['exp'][pack['d
-00012220: 6573 6967 6e27 5d2e 696e 6465 785d 0a20  esign'].index]. 
-00012230: 2020 2020 2020 2020 2020 2020 2020 2061                 a
-00012240: 6461 7461 203d 2061 642e 416e 6e44 6174  data = ad.AnnDat
-00012250: 6128 6578 7029 0a20 2020 2020 2020 2020  a(exp).         
-00012260: 2020 2020 2020 2069 6620 6e6f 7420 2867         if not (g
-00012270: 656e 655f 6461 7461 2069 7320 4e6f 6e65  ene_data is None
-00012280: 293a 0a20 2020 2020 2020 2020 2020 2020  ):.             
-00012290: 2020 2020 2020 2061 6461 7461 2e6f 6273         adata.obs
-000122a0: 203d 2061 6461 7461 2e6f 6273 2e6a 6f69   = adata.obs.joi
-000122b0: 6e28 6765 6e65 5f64 6174 6129 0a20 2020  n(gene_data).   
+00012090: 2020 7374 7228 7061 636b 5b27 6461 7427    str(pack['dat'
+000120a0: 5d2e 6578 7065 7269 6d65 6e74 5f49 445b  ].experiment_ID[
+000120b0: 305d 292c 0a20 2020 2020 2020 2020 2020  0]),.           
+000120c0: 2020 2020 2020 2020 2027 6461 7461 6261           'databa
+000120d0: 7365 273a 2070 6163 6b5b 2764 6174 275d  se': pack['dat']
+000120e0: 2e65 7870 6572 696d 656e 745f 6461 7461  .experiment_data
+000120f0: 6261 7365 5b30 5d2c 0a20 2020 2020 2020  base[0],.       
+00012100: 2020 2020 2020 2020 2020 2020 2027 6163               'ac
+00012110: 6365 7369 6f6e 273a 2020 7061 636b 5b27  cesion':  pack['
+00012120: 6461 7427 5d2e 6578 7065 7269 6d65 6e74  dat'].experiment
+00012130: 5f61 6363 6573 7369 6f6e 5b30 5d2c 0a20  _accession[0],. 
+00012140: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00012150: 2020 2022 4765 6d6d 6151 7561 6c69 7479     "GemmaQuality
+00012160: 5363 6f72 6522 3a20 2070 6163 6b5b 2764  Score":  pack['d
+00012170: 6174 275d 2e67 6565 715f 715f 7363 6f72  at'].geeq_q_scor
+00012180: 655b 305d 2c0a 2020 2020 2020 2020 2020  e[0],.          
+00012190: 2020 2020 2020 2020 2020 2247 656d 6d61            "Gemma
+000121a0: 5375 6974 6162 696c 6974 7953 636f 7265  SuitabilityScore
+000121b0: 223a 2020 7061 636b 5b27 6461 7427 5d2e  ":  pack['dat'].
+000121c0: 6765 6571 5f73 5f73 636f 7265 5b30 5d2c  geeq_s_score[0],
+000121d0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+000121e0: 2020 2020 2022 7461 786f 6e22 3a20 2070       "taxon":  p
+000121f0: 6163 6b5b 2764 6174 275d 2e74 6178 6f6e  ack['dat'].taxon
+00012200: 5f6e 616d 655b 305d 0a20 2020 2020 2020  _name[0].       
+00012210: 2020 2020 2020 2020 2020 2020 207d 0a20               }. 
+00012220: 2020 2020 2020 2020 2020 2020 2020 200a                 .
+00012230: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00012240: 6578 7020 3d20 7061 636b 5b27 6578 7027  exp = pack['exp'
+00012250: 5d5b 7061 636b 5b27 6465 7369 676e 275d  ][pack['design']
+00012260: 2e69 6e64 6578 5d0a 2020 2020 2020 2020  .index].        
+00012270: 2020 2020 2020 2020 6164 6174 6120 3d20          adata = 
+00012280: 6164 2e41 6e6e 4461 7461 2865 7870 290a  ad.AnnData(exp).
+00012290: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000122a0: 6966 206e 6f74 2028 6765 6e65 5f64 6174  if not (gene_dat
+000122b0: 6120 6973 204e 6f6e 6529 3a0a 2020 2020  a is None):.    
 000122c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000122d0: 200a 2020 2020 2020 2020 2020 2020 2020   .              
-000122e0: 2020 0a20 2020 2020 2020 2020 2020 2020    .             
-000122f0: 2020 2061 6461 7461 2e76 6172 203d 2061     adata.var = a
-00012300: 6461 7461 2e76 6172 2e6a 6f69 6e28 7061  data.var.join(pa
-00012310: 636b 5b27 6465 7369 676e 275d 290a 2020  ck['design']).  
-00012320: 2020 2020 2020 2020 2020 2020 2020 6164                ad
-00012330: 6174 612e 756e 7320 3d20 6d64 610a 2020  ata.uns = mda.  
-00012340: 2020 2020 2020 2020 2020 2020 2020 7265                re
-00012350: 7475 726e 2061 6461 7461 0a20 2020 2020  turn adata.     
-00012360: 2020 2020 2020 2023 206d 616b 655f 616e         # make_an
-00012370: 6e64 6174 610a 2020 2020 2020 2020 2020  ndata.          
-00012380: 2020 0a20 2020 2020 2020 2020 2020 200a    .            .
-00012390: 2020 2020 2020 2020 2020 2020 6f75 7420              out 
-000123a0: 3d20 7b6b 3a6d 616b 655f 616e 6e64 6174  = {k:make_anndat
-000123b0: 6128 7061 636b 6564 5f64 6174 615b 6b5d  a(packed_data[k]
-000123c0: 2920 666f 7220 6b20 696e 2070 6163 6b65  ) for k in packe
-000123d0: 645f 6461 7461 2e6b 6579 7328 297d 0a20  d_data.keys()}. 
-000123e0: 2020 2020 2020 2065 6c69 6620 6f75 7470         elif outp
-000123f0: 7574 5f74 7970 6520 3d3d 2027 6469 6374  ut_type == 'dict
-00012400: 273a 0a20 2020 2020 2020 2020 2020 206f  ':.            o
-00012410: 7574 203d 2070 6163 6b65 645f 6461 7461  ut = packed_data
-00012420: 0a20 2020 2020 2020 2065 6c69 6620 6f75  .        elif ou
-00012430: 7470 7574 5f74 7970 6520 3d3d 2022 7469  tput_type == "ti
-00012440: 6479 223a 0a20 2020 2020 2020 2020 2020  dy":.           
-00012450: 2070 6173 730a 2020 2020 2020 2020 7265   pass.        re
-00012460: 7475 726e 206f 7574 0a0a 2020 2020 6465  turn out..    de
-00012470: 6620 6765 745f 6469 6666 6572 656e 7469  f get_differenti
-00012480: 616c 5f65 7870 7265 7373 696f 6e5f 7661  al_expression_va
-00012490: 6c75 6573 2873 656c 662c 200a 2020 2020  lues(self, .    
-000124a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000124b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000124c0: 2020 2020 2020 2064 6174 6173 6574 3a4f         dataset:O
-000124d0: 7074 696f 6e61 6c5b 7374 727c 696e 745d  ptional[str|int]
-000124e0: 203d 204e 6f6e 652c 200a 2020 2020 2020   = None, .      
+000122d0: 6164 6174 612e 6f62 7320 3d20 6164 6174  adata.obs = adat
+000122e0: 612e 6f62 732e 6a6f 696e 2867 656e 655f  a.obs.join(gene_
+000122f0: 6461 7461 290a 2020 2020 2020 2020 2020  data).          
+00012300: 2020 2020 2020 2020 2020 0a20 2020 2020            .     
+00012310: 2020 2020 2020 2020 2020 200a 2020 2020             .    
+00012320: 2020 2020 2020 2020 2020 2020 6164 6174              adat
+00012330: 612e 7661 7220 3d20 6164 6174 612e 7661  a.var = adata.va
+00012340: 722e 6a6f 696e 2870 6163 6b5b 2764 6573  r.join(pack['des
+00012350: 6967 6e27 5d29 0a20 2020 2020 2020 2020  ign']).         
+00012360: 2020 2020 2020 2061 6461 7461 2e75 6e73         adata.uns
+00012370: 203d 206d 6461 0a20 2020 2020 2020 2020   = mda.         
+00012380: 2020 2020 2020 2072 6574 7572 6e20 6164         return ad
+00012390: 6174 610a 2020 2020 2020 2020 2020 2020  ata.            
+000123a0: 2320 6d61 6b65 5f61 6e6e 6461 7461 0a20  # make_anndata. 
+000123b0: 2020 2020 2020 2020 2020 200a 2020 2020             .    
+000123c0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+000123d0: 2020 2020 206f 7574 203d 207b 6b3a 6d61       out = {k:ma
+000123e0: 6b65 5f61 6e6e 6461 7461 2870 6163 6b65  ke_anndata(packe
+000123f0: 645f 6461 7461 5b6b 5d29 2066 6f72 206b  d_data[k]) for k
+00012400: 2069 6e20 7061 636b 6564 5f64 6174 612e   in packed_data.
+00012410: 6b65 7973 2829 7d0a 2020 2020 2020 2020  keys()}.        
+00012420: 656c 6966 206f 7574 7075 745f 7479 7065  elif output_type
+00012430: 203d 3d20 2764 6963 7427 3a0a 2020 2020   == 'dict':.    
+00012440: 2020 2020 2020 2020 6f75 7420 3d20 7061          out = pa
+00012450: 636b 6564 5f64 6174 610a 2020 2020 2020  cked_data.      
+00012460: 2020 656c 6966 206f 7574 7075 745f 7479    elif output_ty
+00012470: 7065 203d 3d20 2274 6964 7922 3a0a 2020  pe == "tidy":.  
+00012480: 2020 2020 2020 2020 2020 7061 7373 0a20            pass. 
+00012490: 2020 2020 2020 2072 6574 7572 6e20 6f75         return ou
+000124a0: 740a 0a20 2020 2064 6566 2067 6574 5f64  t..    def get_d
+000124b0: 6966 6665 7265 6e74 6961 6c5f 6578 7072  ifferential_expr
+000124c0: 6573 7369 6f6e 5f76 616c 7565 7328 7365  ession_values(se
+000124d0: 6c66 2c20 0a20 2020 2020 2020 2020 2020  lf, .           
+000124e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
 000124f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00012500: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00012510: 2020 2020 2072 6573 756c 745f 7365 7473       result_sets
-00012520: 3a4f 7074 696f 6e61 6c5b 4c69 7374 5b73  :Optional[List[s
-00012530: 7472 7c69 6e74 5d5d 203d 204e 6f6e 652c  tr|int]] = None,
-00012540: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00012550: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00012560: 2020 2020 2020 2020 2020 2020 7265 6164              read
-00012570: 6162 6c65 5f63 6f6e 7472 6173 7473 3a62  able_contrasts:b
-00012580: 6f6f 6c20 3d20 4661 6c73 652c 200a 2020  ool = False, .  
+00012500: 6461 7461 7365 743a 4f70 7469 6f6e 616c  dataset:Optional
+00012510: 5b73 7472 7c69 6e74 5d20 3d20 4e6f 6e65  [str|int] = None
+00012520: 2c20 0a20 2020 2020 2020 2020 2020 2020  , .             
+00012530: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00012540: 2020 2020 2020 2020 2020 2020 2020 7265                re
+00012550: 7375 6c74 5f73 6574 733a 4f70 7469 6f6e  sult_sets:Option
+00012560: 616c 5b4c 6973 745b 7374 727c 696e 745d  al[List[str|int]
+00012570: 5d20 3d20 4e6f 6e65 2c0a 2020 2020 2020  ] = None,.      
+00012580: 2020 2020 2020 2020 2020 2020 2020 2020                  
 00012590: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000125a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000125b0: 2020 2020 2020 2020 202a 2a6b 7761 7267           **kwarg
-000125c0: 7329 2d3e 4c69 7374 5b44 6174 6146 7261  s)->List[DataFra
-000125d0: 6d65 5d3a 0a20 2020 2020 2020 2022 2222  me]:.        """
-000125e0: 0a20 2020 2020 2020 2052 6574 7269 6576  .        Retriev
-000125f0: 6573 2074 6865 2064 6966 6665 7265 6e74  es the different
-00012600: 6961 6c20 6578 7072 6573 7369 6f6e 2072  ial expression r
-00012610: 6573 756c 7453 6574 2873 2920 6173 736f  esultSet(s) asso
-00012620: 6369 6174 6564 2077 6974 6820 7468 6520  ciated with the 
-00012630: 6461 7461 7365 742e 0a20 2020 2020 2020  dataset..       
-00012640: 2049 6620 7468 6572 6520 6973 206d 6f72   If there is mor
-00012650: 6520 7468 616e 206f 6e65 2072 6573 756c  e than one resul
-00012660: 7453 6574 2c20 7573 6520 6765 745f 7265  tSet, use get_re
-00012670: 7375 6c74 5f73 6574 7328 2920 746f 2073  sult_sets() to s
-00012680: 6565 2074 6865 206f 7074 696f 6e73 0a20  ee the options. 
-00012690: 2020 2020 2020 2061 6e64 2067 6574 2074         and get t
-000126a0: 6865 2049 4420 796f 7520 7761 6e74 2e20  he ID you want. 
-000126b0: 416c 7465 726e 6174 6976 656c 792c 2079  Alternatively, y
-000126c0: 6f75 2063 616e 2071 7565 7279 2074 6865  ou can query the
-000126d0: 2072 6573 756c 7453 6574 2064 6972 6563   resultSet direc
-000126e0: 746c 790a 2020 2020 2020 2020 6966 2079  tly.        if y
-000126f0: 6f75 206b 6e6f 7720 6974 7320 4944 2062  ou know its ID b
-00012700: 6566 6f72 6568 616e 642e 0a20 2020 2020  eforehand..     
-00012710: 2020 200a 2020 2020 2020 2020 496e 2047     .        In G
-00012720: 656d 6d61 2065 6163 6820 7265 7375 6c74  emma each result
-00012730: 2073 6574 2063 6f72 7265 7370 6f6e 6473   set corresponds
-00012740: 2074 6f20 7468 6520 6573 7469 6d61 7465   to the estimate
-00012750: 6420 6566 6665 6374 7320 0a20 2020 2020  d effects .     
-00012760: 2020 2061 7373 6f63 6961 7465 6420 7769     associated wi
-00012770: 7468 2061 2073 696e 676c 6520 6661 6374  th a single fact
-00012780: 6f72 2069 6e20 7468 6520 6465 7369 676e  or in the design
-00012790: 2c20 616e 6420 6561 6368 2063 616e 2068  , and each can h
-000127a0: 6176 6520 0a20 2020 2020 2020 206d 756c  ave .        mul
-000127b0: 7469 706c 6520 636f 6e74 7261 7374 7320  tiple contrasts 
-000127c0: 2866 6f72 2065 6163 6820 6c65 7665 6c20  (for each level 
-000127d0: 636f 6d70 6172 6564 2074 6f20 6261 7365  compared to base
-000127e0: 6c69 6e65 292e 2054 6875 7320 6120 0a20  line). Thus a . 
-000127f0: 2020 2020 2020 2064 6174 6173 6574 2077         dataset w
-00012800: 6974 6820 6120 3278 3320 6661 6374 6f72  ith a 2x3 factor
-00012810: 6961 6c20 6465 7369 676e 2077 696c 6c20  ial design will 
-00012820: 6861 7665 2074 776f 2072 6573 756c 7420  have two result 
-00012830: 7365 7473 2c20 6f6e 6520 6f66 200a 2020  sets, one of .  
-00012840: 2020 2020 2020 7768 6963 6820 7769 6c6c        which will
-00012850: 2068 6176 6520 6f6e 6520 636f 6e74 7261   have one contra
-00012860: 7374 2c20 616e 6420 6f6e 6520 6861 7669  st, and one havi
-00012870: 6e67 2074 776f 2063 6f6e 7472 6173 7473  ng two contrasts
-00012880: 2e0a 2020 2020 2020 2020 0a20 2020 2020  ..        .     
-00012890: 2020 2054 6865 206d 6574 686f 646f 6c6f     The methodolo
-000128a0: 6779 2066 6f72 2064 6966 6665 7265 6e74  gy for different
-000128b0: 6961 6c20 6578 7072 6573 7369 6f6e 2069  ial expression i
-000128c0: 7320 6578 706c 6169 6e65 6420 696e 2060  s explained in `
-000128d0: 4375 7261 7469 6f6e 200a 2020 2020 2020  Curation .      
-000128e0: 2020 6f66 206f 7665 7220 3130 3030 3020    of over 10000 
-000128f0: 7472 616e 7363 7269 7074 6f6d 6963 2073  transcriptomic s
-00012900: 7475 6469 6573 2074 6f20 656e 6162 6c65  tudies to enable
-00012910: 2064 6174 6120 7265 7573 6520 3c68 7474   data reuse <htt
-00012920: 7073 3a2f 2f64 6f69 2e6f 7267 2f31 302e  ps://doi.org/10.
-00012930: 3130 3933 2f64 6174 6162 6173 652f 6261  1093/database/ba
-00012940: 6162 3030 363e 605f 2e0a 2020 2020 2020  ab006>`_..      
-00012950: 2020 4272 6965 666c 792c 2064 6966 6665    Briefly, diffe
-00012960: 7265 6e74 6961 6c20 6578 7072 6573 7369  rential expressi
-00012970: 6f6e 2061 6e61 6c79 7369 7320 6973 2070  on analysis is p
-00012980: 6572 666f 726d 6564 206f 6e20 7468 6520  erformed on the 
-00012990: 6461 7461 7365 740a 2020 2020 2020 2020  dataset.        
-000129a0: 6261 7365 6420 6f6e 2074 6865 2061 6e6e  based on the ann
-000129b0: 6f74 6174 6564 2065 7870 6572 696d 656e  otated experimen
-000129c0: 7461 6c20 6465 7369 676e 2077 6974 6820  tal design with 
-000129d0: 7570 2074 776f 2074 6872 6565 2070 6f74  up two three pot
-000129e0: 656e 7469 616c 6c79 0a20 2020 2020 2020  entially.       
-000129f0: 206e 6573 7465 6420 6661 6374 6f72 732e   nested factors.
-00012a00: 2047 656d 6d61 2061 7474 656d 7074 7320   Gemma attempts 
-00012a10: 746f 2061 7574 6f6d 6174 6963 616c 6c79  to automatically
-00012a20: 2061 7373 6967 6e20 6261 7365 6c69 6e65   assign baseline
-00012a30: 2063 6f6e 6469 7469 6f6e 730a 2020 2020   conditions.    
-00012a40: 2020 2020 666f 7220 6561 6368 2066 6163      for each fac
-00012a50: 746f 722e 2049 6e20 7468 6520 6162 7365  tor. In the abse
-00012a60: 6e63 6520 6f66 2061 2063 6c65 6172 2063  nce of a clear c
-00012a70: 6f6e 7472 6f6c 2063 6f6e 6469 7469 6f6e  ontrol condition
-00012a80: 2c20 6120 6261 7365 6c69 6e65 200a 2020  , a baseline .  
-00012a90: 2020 2020 2020 6973 2061 7262 6974 7261        is arbitra
-00012aa0: 7269 6c79 2073 656c 6563 7465 642e 2041  rily selected. A
-00012ab0: 2067 656e 6572 616c 697a 6564 206c 696e   generalized lin
-00012ac0: 6561 7220 6d6f 6465 6c20 7769 7468 2065  ear model with e
-00012ad0: 6d70 6972 6963 616c 2042 6179 6573 0a20  mpirical Bayes. 
-00012ae0: 2020 2020 2020 2073 6872 696e 6b61 6765         shrinkage
-00012af0: 206f 6620 742d 7374 6174 6973 7469 6373   of t-statistics
-00012b00: 2069 7320 6669 7420 746f 2074 6865 2064   is fit to the d
-00012b10: 6174 6120 666f 7220 6561 6368 2070 6c61  ata for each pla
-00012b20: 7466 6f72 6d20 656c 656d 656e 7420 0a20  tform element . 
-00012b30: 2020 2020 2020 2028 7072 6f62 652f 6765         (probe/ge
-00012b40: 6e65 2920 7573 696e 6720 616e 2069 6d70  ne) using an imp
-00012b50: 6c65 6d65 6e74 6174 696f 6e20 6f66 2074  lementation of t
-00012b60: 6865 206c 696d 6d61 2061 6c67 6f72 6974  he limma algorit
-00012b70: 686d 2e20 466f 7220 0a20 2020 2020 2020  hm. For .       
-00012b80: 2052 4e41 2d73 6571 2064 6174 612c 2077   RNA-seq data, w
-00012b90: 6520 7573 6520 7765 6967 6874 6564 2072  e use weighted r
-00012ba0: 6567 7265 7373 696f 6e2c 2061 7070 6c79  egression, apply
-00012bb0: 696e 6720 7468 6520 766f 6f6d 2061 6c67  ing the voom alg
-00012bc0: 6f72 6974 686d 0a20 2020 2020 2020 2074  orithm.        t
-00012bd0: 6f20 636f 6d70 7574 6520 7765 6967 6874  o compute weight
-00012be0: 7320 6672 6f6d 2074 6865 206d 6561 6ee2  s from the mean.
-00012bf0: 8093 7661 7269 616e 6365 2072 656c 6174  ..variance relat
-00012c00: 696f 6e73 6869 7020 6f66 2074 6865 2064  ionship of the d
-00012c10: 6174 612e 200a 2020 2020 2020 2020 436f  ata. .        Co
-00012c20: 6e74 7261 7374 7320 6f66 2065 6163 6820  ntrasts of each 
-00012c30: 636f 6e64 6974 696f 6e20 6172 6520 7468  condition are th
-00012c40: 656e 2063 6f6d 7075 7465 6420 636f 6d70  en computed comp
-00012c50: 6172 6564 2074 6f20 7468 6520 7365 6c65  ared to the sele
-00012c60: 6374 6564 200a 2020 2020 2020 2020 6261  cted .        ba
-00012c70: 7365 6c69 6e65 2e20 496e 2073 6f6d 6520  seline. In some 
-00012c80: 7369 7475 6174 696f 6e73 2c20 4765 6d6d  situations, Gemm
-00012c90: 6120 7769 6c6c 2073 706c 6974 2074 6865  a will split the
-00012ca0: 2064 6174 6120 696e 746f 2073 7562 7365   data into subse
-00012cb0: 7473 2066 6f72 0a20 2020 2020 2020 2061  ts for.        a
-00012cc0: 6e61 6c79 7369 732e 2041 2074 7970 6963  nalysis. A typic
-00012cd0: 616c 2073 7563 6820 7369 7475 6174 696f  al such situatio
-00012ce0: 6e20 6973 2077 6865 6e20 6120 e280 9862  n is when a ...b
-00012cf0: 6174 6368 e280 9920 6661 6374 6f72 2069  atch... factor i
-00012d00: 7320 7072 6573 656e 7420 0a20 2020 2020  s present .     
-00012d10: 2020 2061 6e64 2063 6f6e 666f 756e 6465     and confounde
-00012d20: 6420 7769 7468 2061 6e6f 7468 6572 2066  d with another f
-00012d30: 6163 746f 722c 2074 6865 2073 7562 7365  actor, the subse
-00012d40: 7473 2062 6569 6e67 2064 6574 6572 6d69  ts being determi
-00012d50: 6e65 6420 6279 2074 6865 0a20 2020 2020  ned by the.     
-00012d60: 2020 206c 6576 656c 7320 6f66 2074 6865     levels of the
-00012d70: 2063 6f6e 666f 756e 6469 6e67 2066 6163   confounding fac
-00012d80: 746f 722e 0a20 2020 2020 2020 200a 2020  tor..        .  
-00012d90: 2020 2020 2020 0a20 2020 2020 2020 203a        .        :
-00012da0: 7061 7261 6d20 6461 7461 7365 743a 2041  param dataset: A
-00012db0: 2064 6174 6173 6574 2069 6465 6e74 6966   dataset identif
-00012dc0: 6965 722c 2064 6566 6175 6c74 7320 746f  ier, defaults to
-00012dd0: 204e 6f6e 650a 2020 2020 2020 2020 3a74   None.        :t
-00012de0: 7970 6520 6461 7461 7365 743a 204f 7074  ype dataset: Opt
-00012df0: 696f 6e61 6c5b 7374 727c 696e 745d 2c20  ional[str|int], 
-00012e00: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
-00012e10: 203a 7061 7261 6d20 7265 7375 6c74 5f73   :param result_s
-00012e20: 6574 733a 2072 6573 756c 7420 7365 7420  ets: result set 
-00012e30: 6964 656e 7469 6669 6572 732e 2049 6620  identifiers. If 
-00012e40: 6120 6461 7461 7365 740a 2020 2020 2020  a dataset.      
-00012e50: 2020 2020 6973 206e 6f74 2070 726f 7669      is not provi
-00012e60: 6465 642c 2061 6c6c 2072 6573 756c 7420  ded, all result 
-00012e70: 7365 7473 2077 696c 6c20 6265 2064 6f77  sets will be dow
-00012e80: 6e6c 6f61 6465 642e 2049 6620 6974 2069  nloaded. If it i
-00012e90: 7320 7072 6f76 6964 6564 0a20 2020 2020  s provided.     
-00012ea0: 2020 2020 2069 7420 7769 6c6c 206f 6e6c       it will onl
-00012eb0: 7920 6265 2075 7365 6420 746f 2065 6e73  y be used to ens
-00012ec0: 7572 6520 616c 6c20 7265 7375 6c74 2073  ure all result s
-00012ed0: 6574 7320 6265 6c6f 6e67 2074 6f20 7468  ets belong to th
-00012ee0: 6520 6461 7461 7365 742c 2064 6566 6175  e dataset, defau
-00012ef0: 6c74 7320 746f 204e 6f6e 650a 2020 2020  lts to None.    
-00012f00: 2020 2020 3a74 7970 6520 7265 7375 6c74      :type result
-00012f10: 5f73 6574 733a 204f 7074 696f 6e61 6c5b  _sets: Optional[
-00012f20: 4c69 7374 5b73 7472 7c69 6e74 5d5d 2c20  List[str|int]], 
-00012f30: 6f70 7469 6f6e 616c 0a20 2020 2020 2020  optional.       
-00012f40: 203a 7061 7261 6d20 7265 6164 6162 6c65   :param readable
-00012f50: 5f63 6f6e 7472 6173 7473 3a20 4966 2046  _contrasts: If F
-00012f60: 616c 7365 2028 6465 6661 756c 7429 2c20  alse (default), 
-00012f70: 7468 6520 7265 7475 726e 6564 2063 6f6c  the returned col
-00012f80: 756d 6e73 0a20 2020 2020 2020 2020 2077  umns.          w
-00012f90: 696c 6c20 7573 6520 696e 7465 726e 616c  ill use internal
-00012fa0: 2063 6f6e 7374 7261 7374 7320 4944 7320   constrasts IDs 
-00012fb0: 6173 206e 616d 6573 2e20 4465 7461 696c  as names. Detail
-00012fc0: 7320 6162 6f75 7420 7468 6520 636f 6e74  s about the cont
-00012fd0: 7261 7374 7320 6361 6e20 6265 0a20 2020  rasts can be.   
-00012fe0: 2020 2020 2020 2061 6363 6573 7365 6420         accessed 
-00012ff0: 7573 696e 6720 6765 745f 6461 7461 7365  using get_datase
-00013000: 745f 6469 6666 6572 656e 7469 616c 5f65  t_differential_e
-00013010: 7870 7265 7373 696f 6e5f 616e 616c 7973  xpression_analys
-00013020: 6573 2829 2e20 4966 2054 7275 6520 4944  es(). If True ID
-00013030: 730a 2020 2020 2020 2020 2020 7769 6c6c  s.          will
-00013040: 2062 6520 7265 706c 6163 6564 2077 6974   be replaced wit
-00013050: 6820 6875 6d61 6e20 7265 6164 6162 6c65  h human readable
-00013060: 2063 6f6e 7472 6173 7420 696e 666f 726d   contrast inform
-00013070: 6174 696f 6e2c 2064 6566 6175 6c74 7320  ation, defaults 
-00013080: 746f 2046 616c 7365 0a20 2020 2020 2020  to False.       
-00013090: 203a 7479 7065 2072 6561 6461 626c 655f   :type readable_
-000130a0: 636f 6e74 7261 7374 733a 2062 6f6f 6c2c  contrasts: bool,
-000130b0: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
-000130c0: 2020 3a70 6172 616d 202a 2a6b 7761 7267    :param **kwarg
-000130d0: 733a 200a 2020 2020 2020 2020 3a74 7970  s: .        :typ
-000130e0: 6520 2a2a 6b77 6172 6773 3a20 5459 5045  e **kwargs: TYPE
-000130f0: 0a20 2020 2020 2020 203a 7261 6973 6573  .        :raises
-00013100: 2056 616c 7565 4572 726f 723a 2057 696c   ValueError: Wil
-00013110: 6c20 7265 7475 726e 2061 2076 616c 7565  l return a value
-00013120: 2065 7272 6f72 2069 6620 6e65 6974 6865   error if neithe
-00013130: 7220 7265 7375 6c74 5f73 6574 7320 6e6f  r result_sets no
-00013140: 7220 6120 6461 7461 7365 740a 2020 2020  r a dataset.    
-00013150: 2020 2020 2020 6973 2070 726f 7669 6465        is provide
-00013160: 640a 2020 2020 2020 2020 3a72 6574 7572  d.        :retur
-00013170: 6e3a 2041 206c 6973 7420 6f66 2064 6174  n: A list of dat
-00013180: 6120 6672 616d 6573 2077 6974 6820 6469  a frames with di
-00013190: 6666 6572 656e 7469 616c 2065 7870 7265  fferential expre
-000131a0: 7373 696f 6e20 7661 6c75 6573 2070 6572  ssion values per
-000131b0: 2072 6573 756c 7420 7365 742e 0a20 2020   result set..   
-000131c0: 2020 2020 203a 7274 7970 653a 204c 6973       :rtype: Lis
-000131d0: 745b 4461 7461 4672 616d 655d 0a0a 2020  t[DataFrame]..  
-000131e0: 2020 2020 2020 2222 220a 2020 2020 2020        """.      
-000131f0: 2020 6966 2064 6174 6173 6574 2069 7320    if dataset is 
-00013200: 6e6f 7420 4e6f 6e65 2061 6e64 2072 6573  not None and res
-00013210: 756c 745f 7365 7473 2069 7320 6e6f 7420  ult_sets is not 
-00013220: 4e6f 6e65 3a0a 2020 2020 2020 2020 2020  None:.          
-00013230: 2020 6469 6666 7320 3d20 2073 656c 662e    diffs =  self.
-00013240: 6765 745f 6461 7461 7365 745f 6469 6666  get_dataset_diff
-00013250: 6572 656e 7469 616c 5f65 7870 7265 7373  erential_express
-00013260: 696f 6e5f 616e 616c 7973 6573 2864 6174  ion_analyses(dat
-00013270: 6173 6574 290a 2020 2020 2020 2020 2020  aset).          
-00013280: 2020 7273 7320 3d20 6469 6666 732e 7265    rss = diffs.re
-00013290: 7375 6c74 5f49 440a 2020 2020 2020 2020  sult_ID.        
-000132a0: 2020 2020 6966 206e 6f74 2061 6c6c 2873      if not all(s
-000132b0: 7562 2e6c 6973 745f 696e 5f6c 6973 7428  ub.list_in_list(
-000132c0: 7265 7375 6c74 5f73 6574 732c 2072 7373  result_sets, rss
-000132d0: 2929 3a0a 2020 2020 2020 2020 2020 2020  )):.            
-000132e0: 2020 2020 7761 726e 696e 6773 2e77 6172      warnings.war
-000132f0: 6e28 2754 6865 2071 7565 7269 6564 2072  n('The queried r
-00013300: 6573 756c 7453 6574 2069 7320 6e6f 7420  esultSet is not 
-00013310: 6465 7269 7665 6420 6672 6f6d 2074 6869  derived from thi
-00013320: 7320 6461 7461 7365 742e 2027 200a 2020  s dataset. ' .  
-00013330: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00013340: 2020 2020 2020 2020 2020 2020 2743 6865              'Che
-00013350: 636b 2074 6865 2061 7661 696c 6162 6c65  ck the available
-00013360: 2072 6573 756c 7453 6574 7320 7769 7468   resultSets with
-00013370: 2022 6765 745f 7265 7375 6c74 5f73 6574   "get_result_set
-00013380: 7328 2922 2027 0a20 2020 2020 2020 2020  s()" '.         
-00013390: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000133a0: 2020 2020 2027 6f72 2071 7565 7279 2077       'or query w
-000133b0: 6974 686f 7574 2074 6865 2064 6174 6173  ithout the datas
-000133c0: 6574 2070 6172 616d 6574 6572 2e27 290a  et parameter.').
-000133d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000133e0: 7265 7475 726e 0a20 2020 2020 2020 2065  return.        e
-000133f0: 6c69 6620 6461 7461 7365 7420 6973 206e  lif dataset is n
-00013400: 6f74 204e 6f6e 6520 616e 6420 7265 7375  ot None and resu
-00013410: 6c74 5f73 6574 7320 6973 204e 6f6e 653a  lt_sets is None:
-00013420: 0a20 2020 2020 2020 2020 2020 2064 6966  .            dif
-00013430: 6673 203d 2020 7365 6c66 2e67 6574 5f64  fs =  self.get_d
-00013440: 6174 6173 6574 5f64 6966 6665 7265 6e74  ataset_different
-00013450: 6961 6c5f 6578 7072 6573 7369 6f6e 5f61  ial_expression_a
-00013460: 6e61 6c79 7365 7328 6461 7461 7365 7429  nalyses(dataset)
-00013470: 0a20 2020 2020 2020 2020 2020 2072 6573  .            res
-00013480: 756c 745f 7365 7473 203d 2064 6966 6673  ult_sets = diffs
-00013490: 2e72 6573 756c 745f 4944 2e75 6e69 7175  .result_ID.uniqu
-000134a0: 6528 290a 2020 2020 2020 2020 656c 6966  e().        elif
-000134b0: 2064 6174 6173 6574 2069 7320 4e6f 6e65   dataset is None
-000134c0: 2061 6e64 2072 6573 756c 745f 7365 7473   and result_sets
-000134d0: 2069 7320 4e6f 6e65 3a0a 2020 2020 2020   is None:.      
-000134e0: 2020 2020 2020 7261 6973 6520 5661 6c75        raise Valu
-000134f0: 6545 7272 6f72 2827 596f 7520 6d75 7374  eError('You must
-00013500: 2073 7065 6369 6679 2061 2064 6174 6173   specify a datas
-00013510: 6574 206f 7220 7265 7375 6c74 5f73 6574  et or result_set
-00013520: 7327 290a 2020 2020 2020 2020 2020 2020  s').            
-00013530: 0a20 2020 2020 2020 2072 7373 203d 207b  .        rss = {
-00013540: 7d0a 2020 2020 2020 2020 6966 2072 6561  }.        if rea
-00013550: 6461 626c 655f 636f 6e74 7261 7374 733a  dable_contrasts:
-00013560: 0a20 2020 2020 2020 2020 2020 2061 6c6c  .            all
-00013570: 5f66 6163 746f 7273 203d 2020 7365 6c66  _factors =  self
-00013580: 2e67 6574 5f72 6573 756c 745f 7365 7473  .get_result_sets
-00013590: 2872 6573 756c 745f 7365 7473 203d 2072  (result_sets = r
-000135a0: 6573 756c 745f 7365 7473 290a 2020 2020  esult_sets).    
-000135b0: 2020 2020 0a20 2020 2020 2020 2066 6f72      .        for
-000135c0: 2072 7320 696e 2072 6573 756c 745f 7365   rs in result_se
-000135d0: 7473 3a0a 2020 2020 2020 2020 2020 2020  ts:.            
-000135e0: 6466 203d 2073 656c 662e 5f5f 6765 745f  df = self.__get_
-000135f0: 7265 7375 6c74 5f73 6574 2872 7329 0a20  result_set(rs). 
-00013600: 2020 2020 2020 2020 2020 2069 6620 7265             if re
-00013610: 6164 6162 6c65 5f63 6f6e 7472 6173 7473  adable_contrasts
-00013620: 3a0a 2020 2020 2020 2020 2020 2020 2020  :.              
-00013630: 2020 6661 6374 6f72 7320 3d20 7064 2e63    factors = pd.c
-00013640: 6f6e 6361 7428 0a20 2020 2020 2020 2020  oncat(.         
-00013650: 2020 2020 2020 2020 2020 206c 6973 7428             list(
-00013660: 616c 6c5f 6661 6374 6f72 735b 616c 6c5f  all_factors[all_
-00013670: 6661 6374 6f72 732e 7265 7375 6c74 5f49  factors.result_I
-00013680: 4420 3d3d 2072 735d 2e65 7870 6572 696d  D == rs].experim
-00013690: 656e 7461 6c5f 6661 6374 6f72 7329 0a20  ental_factors). 
-000136a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000136b0: 2020 2029 2e64 726f 705f 6475 706c 6963     ).drop_duplic
-000136c0: 6174 6573 2829 0a20 2020 2020 2020 2020  ates().         
-000136d0: 2020 2020 2020 2063 6f6c 7320 3d20 5b73         cols = [s
-000136e0: 2e72 6570 6c61 6365 2827 6c6f 6732 6663  .replace('log2fc
-000136f0: 272c 276c 6f67 466f 6c64 4368 616e 6765  ','logFoldChange
-00013700: 2729 2066 6f72 2073 2069 6e20 6466 2e63  ') for s in df.c
-00013710: 6f6c 756d 6e73 5d0a 2020 2020 2020 2020  olumns].        
-00013720: 2020 2020 2020 2020 666f 7220 6920 696e          for i in
-00013730: 2072 616e 6765 2866 6163 746f 7273 2e73   range(factors.s
-00013740: 6861 7065 5b30 5d29 3a0a 2020 2020 2020  hape[0]):.      
-00013750: 2020 2020 2020 2020 2020 2020 2020 636f                co
-00013760: 6c73 203d 205b 732e 7265 706c 6163 6528  ls = [s.replace(
-00013770: 7374 7228 6c69 7374 2866 6163 746f 7273  str(list(factors
-00013780: 2e49 4429 5b69 5d29 2c6c 6973 7428 6661  .ID)[i]),list(fa
-00013790: 6374 6f72 732e 7375 6d6d 6172 7929 5b69  ctors.summary)[i
-000137a0: 5d29 2066 6f72 2073 2069 6e20 636f 6c73  ]) for s in cols
-000137b0: 5d0a 2020 2020 2020 2020 2020 2020 2020  ].              
-000137c0: 2020 6466 2e63 6f6c 756d 6e73 203d 2063    df.columns = c
-000137d0: 6f6c 730a 2020 2020 2020 2020 2020 2020  ols.            
-000137e0: 7273 735b 7273 5d20 3d20 6466 0a20 2020  rss[rs] = df.   
-000137f0: 2020 2020 2072 6574 7572 6e20 7273 730a       return rss.
-00013800: 2020 2020 2020 2020 2020 2020 0a0a 2020              ..  
-00013810: 2020 2320 6765 745f 7461 7861 2069 7320    # get_taxa is 
-00013820: 6d6f 7665 6420 746f 2062 6173 652e 206f  moved to base. o
-00013830: 6e6c 7920 7265 6d6f 7665 7320 7468 6520  nly removes the 
-00013840: 6e61 6d65 6c65 7373 2072 6174 206e 6f77  nameless rat now
-00013850: 0a0a 2020 2020 2320 6765 6d6d 615f 6361  ..    # gemma_ca
-00013860: 6c6c 2075 6e69 6d70 6c65 6d65 6e74 6564  ll unimplemented
-00013870: 2c20 6e6f 7420 6e65 6564 6564 2020 2020  , not needed    
-00013880: 0a0a 2020 2020 6465 6620 6765 745f 616c  ..    def get_al
-00013890: 6c5f 7061 6765 7328 7365 6c66 2c66 756e  l_pages(self,fun
-000138a0: 3a43 616c 6c61 626c 652c 7374 6570 5f73  :Callable,step_s
-000138b0: 697a 653a 696e 7420 3d20 3130 302c 2a2a  ize:int = 100,**
-000138c0: 6b77 6172 6773 292d 3e6c 6973 747c 4461  kwargs)->list|Da
-000138d0: 7461 4672 616d 653a 0a20 2020 2020 2020  taFrame:.       
-000138e0: 2022 2222 0a20 2020 2020 2020 2041 2063   """.        A c
-000138f0: 6f6e 7665 6e69 656e 6365 2066 756e 6374  onvenience funct
-00013900: 696f 6e20 746f 2061 6c6c 6f77 2065 6173  ion to allow eas
-00013910: 7920 6974 6572 6174 696f 6e20 6f76 6572  y iteration over
-00013920: 2070 6167 696e 6174 6564 206f 7574 7075   paginated outpu
-00013930: 7473 2e0a 2020 2020 2020 2020 4966 2074  ts..        If t
-00013940: 6865 2066 756e 6374 696f 6e20 7265 7475  he function retu
-00013950: 726e 7320 6120 4461 7461 4672 616d 6520  rns a DataFrame 
-00013960: 6f75 7470 7574 2077 696c 6c20 6265 206d  output will be m
-00013970: 6572 6765 6420 6279 2074 6865 2072 6f77  erged by the row
-00013980: 732c 0a20 2020 2020 2020 2069 6620 7468  s,.        if th
-00013990: 6520 6675 6e63 7469 6f6e 2072 6574 7572  e function retur
-000139a0: 6e73 2061 206c 6973 7420 2865 672e 2027  ns a list (eg. '
-000139b0: 7261 7727 2066 756e 6374 696f 6e73 2920  raw' functions) 
-000139c0: 6120 636f 6e63 6174 616e 6174 6564 206c  a concatanated l
-000139d0: 6973 740a 2020 2020 2020 2020 7769 6c6c  ist.        will
-000139e0: 2062 6520 7265 7475 726e 6564 0a20 2020   be returned.   
-000139f0: 2020 2020 200a 2020 2020 2020 2020 0a20       .        . 
-00013a00: 2020 2020 2020 203a 7061 7261 6d20 6675         :param fu
-00013a10: 6e3a 2041 2063 616c 6c61 626c 6520 6672  n: A callable fr
-00013a20: 6f6d 2067 656d 6d61 7079 2077 6974 6820  om gemmapy with 
-00013a30: 6f66 6673 6574 2061 6e64 206c 696d 6974  offset and limit
-00013a40: 0a20 2020 2020 2020 2020 2066 756e 6374  .          funct
-00013a50: 696f 6e73 0a20 2020 2020 2020 203a 7479  ions.        :ty
-00013a60: 7065 2066 756e 3a20 4361 6c6c 6162 6c65  pe fun: Callable
-00013a70: 0a20 2020 2020 2020 203a 7061 7261 6d20  .        :param 
-00013a80: 7374 6570 5f73 697a 653a 2053 697a 6520  step_size: Size 
-00013a90: 6f66 2069 6e64 6976 6964 7561 6c20 6361  of individual ca
-00013aa0: 6c6c 7320 746f 2074 6865 2073 6572 7665  lls to the serve
-00013ab0: 722e 2031 3030 2069 7320 0a20 2020 2020  r. 100 is .     
-00013ac0: 2020 2020 2074 6865 206d 6178 696d 756d       the maximum
-00013ad0: 2076 616c 7565 2061 6e64 2074 6865 2064   value and the d
-00013ae0: 6566 6175 6c74 2e0a 2020 2020 2020 2020  efault..        
-00013af0: 3a74 7970 6520 7374 6570 5f73 697a 653a  :type step_size:
-00013b00: 2069 6e74 2c20 6f70 7469 6f6e 616c 0a20   int, optional. 
-00013b10: 2020 2020 2020 203a 7061 7261 6d20 2a2a         :param **
-00013b20: 6b77 6172 6773 3a20 6172 6775 6d65 6e74  kwargs: argument
-00013b30: 7320 666f 7220 7468 6520 6361 6c6c 6162  s for the callab
-00013b40: 6c65 2066 756e 0a20 2020 2020 2020 203a  le fun.        :
-00013b50: 7265 7475 726e 3a20 4120 4461 7461 4672  return: A DataFr
-00013b60: 616d 6520 6f72 2061 206c 6973 7420 636f  ame or a list co
-00013b70: 6e74 6169 6e69 6e67 2061 6c6c 2074 6865  ntaining all the
-00013b80: 206f 7574 7075 7420 6465 7065 6e64 696e   output dependin
-00013b90: 6720 6f6e 200a 2020 2020 2020 2020 2020  g on .          
-00013ba0: 6f75 7470 7574 206f 6620 7468 6520 6361  output of the ca
-00013bb0: 6c6c 6162 6c65 0a20 2020 2020 2020 203a  llable.        :
-00013bc0: 7274 7970 653a 206c 6973 747c 4461 7461  rtype: list|Data
-00013bd0: 4672 616d 650a 2020 2020 2020 2020 2222  Frame.        ""
-00013be0: 220a 0a20 2020 2020 2020 206f 7574 203d  "..        out =
-00013bf0: 205b 5d20 2020 2020 2020 200a 2020 2020   []        .    
-00013c00: 2020 2020 706f 6b65 5f63 616c 6c20 3d20      poke_call = 
-00013c10: 6675 6e28 6c69 6d69 7420 3d31 2c2a 2a6b  fun(limit =1,**k
-00013c20: 7761 7267 7329 0a20 2020 2020 2020 200a  wargs).        .
-00013c30: 2020 2020 2020 2020 6966 2074 7970 6528          if type(
-00013c40: 706f 6b65 5f63 616c 6c29 203d 3d20 7064  poke_call) == pd
-00013c50: 2e63 6f72 652e 6672 616d 652e 4461 7461  .core.frame.Data
-00013c60: 4672 616d 653a 0a20 2020 2020 2020 2020  Frame:.         
-00013c70: 2020 2063 6f75 6e74 203d 2070 6f6b 655f     count = poke_
-00013c80: 6361 6c6c 2e61 7474 7269 6275 7465 735b  call.attributes[
-00013c90: 2274 6f74 616c 5f65 6c65 6d65 6e74 7322  "total_elements"
-00013ca0: 5d0a 2020 2020 2020 2020 656c 7365 3a0a  ].        else:.
-00013cb0: 2020 2020 2020 2020 2020 2020 636f 756e              coun
-00013cc0: 7420 3d20 706f 6b65 5f63 616c 6c2e 746f  t = poke_call.to
-00013cd0: 7461 6c5f 656c 656d 656e 7473 0a20 2020  tal_elements.   
-00013ce0: 2020 2020 200a 2020 2020 2020 2020 666f       .        fo
-00013cf0: 7220 6920 696e 2072 616e 6765 2830 2c63  r i in range(0,c
-00013d00: 6f75 6e74 2c73 7465 705f 7369 7a65 293a  ount,step_size):
-00013d10: 0a20 2020 2020 2020 2020 2020 206f 7574  .            out
-00013d20: 2e61 7070 656e 6428 6675 6e28 6c69 6d69  .append(fun(limi
-00013d30: 7420 3d20 7374 6570 5f73 697a 652c 6f66  t = step_size,of
-00013d40: 6673 6574 203d 2069 2c2a 2a6b 7761 7267  fset = i,**kwarg
-00013d50: 7329 290a 2020 2020 2020 2020 0a20 2020  s)).        .   
-00013d60: 2020 2020 2069 6620 7479 7065 2870 6f6b       if type(pok
-00013d70: 655f 6361 6c6c 2920 3d3d 2070 642e 636f  e_call) == pd.co
-00013d80: 7265 2e66 7261 6d65 2e44 6174 6146 7261  re.frame.DataFra
-00013d90: 6d65 3a0a 2020 2020 2020 2020 2020 2020  me:.            
-00013da0: 7265 7475 726e 2070 642e 636f 6e63 6174  return pd.concat
-00013db0: 286f 7574 2c69 676e 6f72 655f 696e 6465  (out,ignore_inde
-00013dc0: 7820 3d20 5472 7565 290a 2020 2020 2020  x = True).      
-00013dd0: 2020 656c 7365 3a0a 2020 2020 2020 2020    else:.        
-00013de0: 2020 2020 7265 7475 726e 2073 7562 2e62      return sub.b
-00013df0: 7265 616b 5f6c 6973 7428 5b78 2e64 6174  reak_list([x.dat
-00013e00: 6120 666f 7220 7820 696e 206f 7574 5d29  a for x in out])
-00013e10: 0a20 2020 2020 2020 200a 0a20 2020 200a  .        ..    .
-00013e20: 2020 2020 6465 6620 6669 6c74 6572 5f70      def filter_p
-00013e30: 726f 7065 7274 6965 7328 7365 6c66 2c20  roperties(self, 
-00013e40: 6f75 7470 7574 5f74 7970 653a 7374 7220  output_type:str 
-00013e50: 3d20 2744 6174 6146 7261 6d65 2729 2d3e  = 'DataFrame')->
-00013e60: 6469 6374 7c44 6174 6146 7261 6d65 3a0a  dict|DataFrame:.
-00013e70: 2020 2020 2020 2020 2222 220a 2020 2020          """.    
-00013e80: 2020 2020 536f 6d65 2066 756e 6374 696f      Some functio
-00013e90: 6e73 2073 7563 6820 6173 2067 6574 5f64  ns such as get_d
-00013ea0: 6174 6173 6574 7320 616e 6420 6765 745f  atasets and get_
-00013eb0: 706c 6174 666f 726d 7320 696e 636c 7564  platforms includ
-00013ec0: 6520 6120 6669 6c74 6572 0a20 2020 2020  e a filter.     
-00013ed0: 2020 2061 7267 756d 656e 7420 7468 6174     argument that
-00013ee0: 2061 6c6c 6f77 7320 6372 6561 7469 6f6e   allows creation
-00013ef0: 206f 6620 6d6f 7265 2063 6f6d 706c 6578   of more complex
-00013f00: 2071 7565 7269 6573 2e20 5468 6973 2066   queries. This f
-00013f10: 756e 6374 696f 6e20 0a20 2020 2020 2020  unction .       
-00013f20: 2072 6574 7572 6e73 2061 206c 6973 7420   returns a list 
-00013f30: 6f66 2073 7570 706f 7274 6564 2070 726f  of supported pro
-00013f40: 7065 7274 6965 7320 746f 2062 6520 7573  perties to be us
-00013f50: 6564 2069 6e20 7468 6f73 6520 6669 6c74  ed in those filt
-00013f60: 6572 730a 2020 2020 2020 2020 0a20 2020  ers.        .   
-00013f70: 2020 2020 203a 7061 7261 6d20 6f75 7470       :param outp
-00013f80: 7574 5f74 7970 653a 2054 7970 6520 746f  ut_type: Type to
-00013f90: 2072 6574 7572 6e2e 2022 4461 7461 4672   return. "DataFr
-00013fa0: 616d 6522 206f 7220 2264 6963 7422 2c20  ame" or "dict", 
-00013fb0: 6465 6661 756c 7473 2074 6f20 2744 6174  defaults to 'Dat
-00013fc0: 6146 7261 6d65 270a 2020 2020 2020 2020  aFrame'.        
-00013fd0: 3a74 7970 6520 6f75 7470 7574 5f74 7970  :type output_typ
-00013fe0: 653a 2073 7472 2c20 6f70 7469 6f6e 616c  e: str, optional
-00013ff0: 0a20 2020 2020 2020 203a 7265 7475 726e  .        :return
-00014000: 3a20 4461 7461 4672 616d 6520 6f72 2064  : DataFrame or d
-00014010: 6963 7420 636f 6e74 6169 6e69 6e67 2073  ict containing s
-00014020: 7570 706f 7274 6564 2070 726f 7065 7274  upported propert
-00014030: 6965 7320 616e 6420 7468 6569 7220 6461  ies and their da
-00014040: 7461 2074 7970 6573 0a20 2020 2020 2020  ta types.       
-00014050: 203a 7274 7970 653a 2064 6963 747c 4461   :rtype: dict|Da
-00014060: 7461 4672 616d 650a 0a20 2020 2020 2020  taFrame..       
-00014070: 2022 2222 0a20 2020 2020 2020 200a 2020   """.        .  
-00014080: 2020 2020 2020 6420 3d20 7365 6c66 2e72        d = self.r
-00014090: 6177 2e61 7069 5f63 6c69 656e 742e 7265  aw.api_client.re
-000140a0: 7374 5f63 6c69 656e 742e 4745 5428 2268  st_client.GET("h
-000140b0: 7474 7073 3a2f 2f67 656d 6d61 2e6d 736c  ttps://gemma.msl
-000140c0: 2e75 6263 2e63 612f 7265 7374 2f76 322f  .ubc.ca/rest/v2/
-000140d0: 6f70 656e 6170 692e 6a73 6f6e 2229 2e75  openapi.json").u
-000140e0: 726c 6c69 6233 5f72 6573 706f 6e73 650a  rllib3_response.
-000140f0: 2020 2020 2020 2020 6170 695f 6669 6c65          api_file
-00014100: 203d 206a 736f 6e2e 6c6f 6164 7328 642e   = json.loads(d.
-00014110: 6461 7461 290a 2020 2020 2020 2020 0a20  data).        . 
-00014120: 2020 2020 2020 2064 6174 6173 6574 5f66         dataset_f
-00014130: 696c 7465 7220 3d20 6170 695f 6669 6c65  ilter = api_file
-00014140: 5b22 636f 6d70 6f6e 656e 7473 225d 5b22  ["components"]["
-00014150: 7363 6865 6d61 7322 5d5b 2246 696c 7465  schemas"]["Filte
-00014160: 7241 7267 4578 7072 6573 7369 6f6e 4578  rArgExpressionEx
-00014170: 7065 7269 6d65 6e74 225d 5b22 782d 6765  periment"]["x-ge
-00014180: 6d6d 612d 6669 6c74 6572 6162 6c65 2d70  mma-filterable-p
-00014190: 726f 7065 7274 6965 7322 5d0a 2020 2020  roperties"].    
-000141a0: 2020 2020 0a20 2020 2020 2020 2070 6c61      .        pla
-000141b0: 7466 6f72 6d5f 6669 6c74 6572 203d 2061  tform_filter = a
-000141c0: 7069 5f66 696c 655b 2263 6f6d 706f 6e65  pi_file["compone
-000141d0: 6e74 7322 5d5b 2273 6368 656d 6173 225d  nts"]["schemas"]
-000141e0: 5b22 4669 6c74 6572 4172 6741 7272 6179  ["FilterArgArray
-000141f0: 4465 7369 676e 225d 5b22 782d 6765 6d6d  Design"]["x-gemm
-00014200: 612d 6669 6c74 6572 6162 6c65 2d70 726f  a-filterable-pro
-00014210: 7065 7274 6965 7322 5d0a 2020 2020 2020  perties"].      
-00014220: 2020 0a20 2020 2020 2020 2072 6573 756c    .        resul
-00014230: 745f 7365 745f 6669 6c74 6572 203d 2061  t_set_filter = a
-00014240: 7069 5f66 696c 655b 2263 6f6d 706f 6e65  pi_file["compone
-00014250: 6e74 7322 5d5b 2273 6368 656d 6173 225d  nts"]["schemas"]
-00014260: 5b22 4669 6c74 6572 4172 6745 7870 7265  ["FilterArgExpre
-00014270: 7373 696f 6e41 6e61 6c79 7369 7352 6573  ssionAnalysisRes
-00014280: 756c 7453 6574 225d 5b22 782d 6765 6d6d  ultSet"]["x-gemm
-00014290: 612d 6669 6c74 6572 6162 6c65 2d70 726f  a-filterable-pro
-000142a0: 7065 7274 6965 7322 5d0a 2020 2020 2020  perties"].      
-000142b0: 2020 0a20 2020 2020 2020 2069 6620 6f75    .        if ou
-000142c0: 7470 7574 5f74 7970 6520 3d3d 2027 4461  tput_type == 'Da
-000142d0: 7461 4672 616d 6527 3a0a 2020 2020 2020  taFrame':.      
-000142e0: 2020 2020 2020 7265 7475 726e 207b 0a20        return {. 
-000142f0: 2020 2020 2020 2020 2020 2020 2020 2022                 "
-00014300: 6461 7461 7365 7422 3a70 642e 4461 7461  dataset":pd.Data
-00014310: 4672 616d 6528 7b0a 2020 2020 2020 2020  Frame({.        
-00014320: 2020 2020 2020 2020 2020 2020 226e 616d              "nam
-00014330: 6522 3a20 7375 622e 6669 656c 645f 696e  e": sub.field_in
-00014340: 5f6c 6973 7428 6461 7461 7365 745f 6669  _list(dataset_fi
-00014350: 6c74 6572 2c27 6e61 6d65 2729 2c0a 2020  lter,'name'),.  
-00014360: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014370: 2020 2274 7970 6522 3a20 7375 622e 6669    "type": sub.fi
-00014380: 656c 645f 696e 5f6c 6973 7428 6461 7461  eld_in_list(data
-00014390: 7365 745f 6669 6c74 6572 2c27 7479 7065  set_filter,'type
-000143a0: 2729 2c0a 2020 2020 2020 2020 2020 2020  '),.            
-000143b0: 2020 2020 2020 2020 2264 6573 6372 6970          "descrip
-000143c0: 7469 6f6e 223a 2073 7562 2e66 6965 6c64  tion": sub.field
-000143d0: 5f69 6e5f 6c69 7374 2864 6174 6173 6574  _in_list(dataset
-000143e0: 5f66 696c 7465 722c 2764 6573 6372 6970  _filter,'descrip
-000143f0: 7469 6f6e 2729 0a20 2020 2020 2020 2020  tion').         
-00014400: 2020 2020 2020 2020 2020 207d 292c 0a20             }),. 
-00014410: 2020 2020 2020 2020 2020 2020 2020 2022                 "
-00014420: 706c 6174 666f 726d 223a 7064 2e44 6174  platform":pd.Dat
-00014430: 6146 7261 6d65 287b 0a20 2020 2020 2020  aFrame({.       
-00014440: 2020 2020 2020 2020 2020 2020 2022 6e61               "na
-00014450: 6d65 223a 2073 7562 2e66 6965 6c64 5f69  me": sub.field_i
-00014460: 6e5f 6c69 7374 2870 6c61 7466 6f72 6d5f  n_list(platform_
-00014470: 6669 6c74 6572 2c27 6e61 6d65 2729 2c0a  filter,'name'),.
-00014480: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014490: 2020 2020 2274 7970 6522 3a20 7375 622e      "type": sub.
-000144a0: 6669 656c 645f 696e 5f6c 6973 7428 706c  field_in_list(pl
-000144b0: 6174 666f 726d 5f66 696c 7465 722c 2774  atform_filter,'t
-000144c0: 7970 6527 292c 0a20 2020 2020 2020 2020  ype'),.         
-000144d0: 2020 2020 2020 2020 2020 2022 6465 7363             "desc
-000144e0: 7269 7074 696f 6e22 3a20 7375 622e 6669  ription": sub.fi
-000144f0: 656c 645f 696e 5f6c 6973 7428 706c 6174  eld_in_list(plat
-00014500: 666f 726d 5f66 696c 7465 722c 2764 6573  form_filter,'des
-00014510: 6372 6970 7469 6f6e 2729 0a20 2020 2020  cription').     
-00014520: 2020 2020 2020 2020 2020 2020 2020 207d                 }
-00014530: 292c 0a20 2020 2020 2020 2020 2020 2020  ),.             
-00014540: 2020 2022 7265 7375 6c74 5f73 6574 223a     "result_set":
-00014550: 7064 2e44 6174 6146 7261 6d65 287b 0a20  pd.DataFrame({. 
-00014560: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014570: 2020 2022 6e61 6d65 223a 2073 7562 2e66     "name": sub.f
-00014580: 6965 6c64 5f69 6e5f 6c69 7374 2872 6573  ield_in_list(res
-00014590: 756c 745f 7365 745f 6669 6c74 6572 2c27  ult_set_filter,'
-000145a0: 6e61 6d65 2729 2c0a 2020 2020 2020 2020  name'),.        
-000145b0: 2020 2020 2020 2020 2020 2020 2274 7970              "typ
-000145c0: 6522 3a20 7375 622e 6669 656c 645f 696e  e": sub.field_in
-000145d0: 5f6c 6973 7428 7265 7375 6c74 5f73 6574  _list(result_set
-000145e0: 5f66 696c 7465 722c 2774 7970 6527 292c  _filter,'type'),
-000145f0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00014600: 2020 2020 2022 6465 7363 7269 7074 696f       "descriptio
-00014610: 6e22 3a20 7375 622e 6669 656c 645f 696e  n": sub.field_in
-00014620: 5f6c 6973 7428 7265 7375 6c74 5f73 6574  _list(result_set
-00014630: 5f66 696c 7465 722c 2764 6573 6372 6970  _filter,'descrip
-00014640: 7469 6f6e 2729 0a20 2020 2020 2020 2020  tion').         
-00014650: 2020 2020 2020 2020 2020 207d 290a 2020             }).  
-00014660: 2020 2020 2020 2020 2020 2020 2020 7d0a                }.
-00014670: 2020 2020 2020 2020 656c 6966 206f 7574          elif out
-00014680: 7075 745f 7479 7065 203d 3d20 2764 6963  put_type == 'dic
-00014690: 7427 3a0a 2020 2020 2020 2020 2020 2020  t':.            
-000146a0: 7265 7475 726e 207b 0a20 2020 2020 2020  return {.       
-000146b0: 2020 2020 2020 2020 2022 6461 7461 7365           "datase
-000146c0: 7422 3a20 7b78 5b27 6e61 6d65 275d 3a7b  t": {x['name']:{
-000146d0: 2274 7970 6522 3a78 5b27 7479 7065 275d  "type":x['type']
-000146e0: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
-000146f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014700: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014710: 2264 6573 6372 6970 7469 6f6e 223a 7375  "description":su
-00014720: 622e 6163 6365 7373 5f66 6965 6c64 2878  b.access_field(x
-00014730: 2c27 6465 7363 7269 7074 696f 6e27 2c4e  ,'description',N
-00014740: 6f6e 6529 7d20 666f 7220 7820 696e 2064  one)} for x in d
-00014750: 6174 6173 6574 5f66 696c 7465 727d 2c0a  ataset_filter},.
-00014760: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014770: 2770 6c61 7466 6f72 6d27 3a7b 785b 276e  'platform':{x['n
-00014780: 616d 6527 5d3a 7b22 7479 7065 223a 785b  ame']:{"type":x[
-00014790: 2774 7970 6527 5d2c 0a20 2020 2020 2020  'type'],.       
-000147a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000147b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000147c0: 2020 2020 2020 2022 6465 7363 7269 7074         "descript
-000147d0: 696f 6e22 3a73 7562 2e61 6363 6573 735f  ion":sub.access_
-000147e0: 6669 656c 6428 782c 2764 6573 6372 6970  field(x,'descrip
-000147f0: 7469 6f6e 272c 4e6f 6e65 297d 2066 6f72  tion',None)} for
-00014800: 2078 2069 6e20 706c 6174 666f 726d 5f66   x in platform_f
-00014810: 696c 7465 727d 2c0a 2020 2020 2020 2020  ilter},.        
-00014820: 2020 2020 2020 2020 2772 6573 756c 745f          'result_
-00014830: 7365 7427 3a7b 785b 276e 616d 6527 5d3a  set':{x['name']:
-00014840: 7b22 7479 7065 223a 785b 2774 7970 6527  {"type":x['type'
-00014850: 5d2c 0a20 2020 2020 2020 2020 2020 2020  ],.             
-00014860: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014870: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00014880: 2022 6465 7363 7269 7074 696f 6e22 3a73   "description":s
-00014890: 7562 2e61 6363 6573 735f 6669 656c 6428  ub.access_field(
-000148a0: 782c 2764 6573 6372 6970 7469 6f6e 272c  x,'description',
-000148b0: 4e6f 6e65 297d 2066 6f72 2078 2069 6e20  None)} for x in 
-000148c0: 7265 7375 6c74 5f73 6574 5f66 696c 7465  result_set_filte
-000148d0: 727d 0a20 2020 2020 2020 2020 2020 2020  r}.             
-000148e0: 2020 200a 2020 2020 2020 2020 2020 2020     .            
-000148f0: 2020 2020 7d0a 2020 2020 2020 2020 2020      }.          
-00014900: 2020 0a20 2020 2020 2020 200a 2020 2020    .        .    
-00014910: 0a20 2020 2064 6566 2067 6574 5f63 6869  .    def get_chi
-00014920: 6c64 5f74 6572 6d73 2873 656c 662c 7465  ld_terms(self,te
-00014930: 726d 733a 4c69 7374 5b73 7472 5d29 2d3e  rms:List[str])->
-00014940: 4c69 7374 5b73 7472 5d3a 0a20 2020 2020  List[str]:.     
-00014950: 2020 2022 2222 0a20 2020 2020 2020 2057     """.        W
-00014960: 6865 6e20 7175 6572 7969 6e67 2066 6f72  hen querying for
-00014970: 206f 6e74 6f6c 6f67 7920 7465 726d 732c   ontology terms,
-00014980: 2047 656d 6d61 2070 726f 7061 6761 7465   Gemma propagate
-00014990: 7320 7468 6573 6520 7465 726d 7320 746f  s these terms to
-000149a0: 200a 2020 2020 2020 2020 696e 636c 7564   .        includ
-000149b0: 6520 616e 7920 6461 7461 7365 7473 2077  e any datasets w
-000149c0: 6974 6820 7468 6569 7220 6368 696c 6420  ith their child 
-000149d0: 7465 726d 7320 696e 2074 6865 2072 6573  terms in the res
-000149e0: 756c 7473 2e20 5468 6973 200a 2020 2020  ults. This .    
-000149f0: 2020 2020 6675 6e63 7469 6f6e 2072 6574      function ret
-00014a00: 7572 6e73 2074 6865 7365 2063 6869 6c64  urns these child
-00014a10: 7265 6e20 666f 7220 616e 7920 6e75 6d62  ren for any numb
-00014a20: 6572 206f 6620 7465 726d 732c 2069 6e63  er of terms, inc
-00014a30: 6c75 6469 6e67 2061 6c6c 200a 2020 2020  luding all .    
-00014a40: 2020 2020 6368 696c 6472 656e 2061 6e64      children and
-00014a50: 2074 6865 2074 6572 6d73 2069 7473 656c   the terms itsel
-00014a60: 6620 696e 2074 6865 206f 7574 7075 7420  f in the output 
-00014a70: 7665 6374 6f72 0a20 2020 2020 2020 200a  vector.        .
-00014a80: 2020 2020 2020 2020 3a70 6172 616d 2074          :param t
-00014a90: 6572 6d73 3a20 4120 6c69 7374 206f 6620  erms: A list of 
-00014aa0: 6f6e 746f 6c6f 6779 2074 6572 6d73 0a20  ontology terms. 
-00014ab0: 2020 2020 2020 203a 7479 7065 2074 6572         :type ter
-00014ac0: 6d73 3a20 4c69 7374 5b73 7472 5d0a 2020  ms: List[str].  
-00014ad0: 2020 2020 2020 3a72 6574 7572 6e3a 2041        :return: A
-00014ae0: 6e20 6172 7261 7920 636f 6e74 6169 6e69  n array containi
-00014af0: 6e67 2064 6573 6365 6e64 656e 6473 206f  ng descendends o
-00014b00: 6620 7468 6520 616e 6e6f 7461 7469 6f6e  f the annotation
-00014b10: 2074 6572 6d73 2c20 0a20 2020 2020 2020   terms, .       
-00014b20: 2069 6e63 6c75 6469 6e67 2074 6865 2074   including the t
-00014b30: 6572 6d73 2074 6865 6d73 656c 7665 730a  erms themselves.
-00014b40: 2020 2020 2020 2020 3a72 7479 7065 3a20          :rtype: 
-00014b50: 4c69 7374 5b73 7472 5d0a 0a20 2020 2020  List[str]..     
-00014b60: 2020 2022 2222 0a20 2020 2020 2020 200a     """.        .
-00014b70: 2020 2020 2020 2020 6f75 7470 7574 203d          output =
-00014b80: 2073 656c 662e 6765 745f 6461 7461 7365   self.get_datase
-00014b90: 7473 2875 7269 733d 7465 726d 7329 0a20  ts(uris=terms). 
-00014ba0: 2020 2020 2020 200a 2020 2020 2020 2020         .        
-00014bb0: 7265 7475 726e 2072 652e 6669 6e64 616c  return re.findal
-00014bc0: 6c28 7227 6874 7470 2e2a 3f28 3f3d 2c7c  l(r'http.*?(?=,|
-00014bd0: 5c29 2927 2c6f 7574 7075 742e 6174 7472  \))',output.attr
-00014be0: 6962 7574 6573 5b27 6669 6c74 6572 275d  ibutes['filter']
-00014bf0: 290a 2020 2020 0a20 2020 2023 2075 7064  ).    .    # upd
-00014c00: 6174 655f 7265 7375 6c74 7320 6375 7272  ate_results curr
-00014c10: 656e 7479 2075 6e69 6d70 6c65 6d65 6e74  enty unimplement
-00014c20: 6564 2061 6e64 206e 6f74 2072 6561 6c6c  ed and not reall
-00014c30: 7920 6573 7365 6e74 6961 6c0a 0a0a 2320  y essential...# 
-00014c40: 5465 7374 730a 6966 205f 5f6e 616d 655f  Tests.if __name_
-00014c50: 5f20 3d3d 2027 5f5f 6d61 696e 5f5f 273a  _ == '__main__':
-00014c60: 0a20 2020 2023 2064 7620 7465 6d70 0a20  .    # dv temp. 
-00014c70: 2020 2069 6d70 6f72 7420 6874 7470 2e63     import http.c
-00014c80: 6c69 656e 740a 2020 2020 6874 7470 2e63  lient.    http.c
-00014c90: 6c69 656e 742e 4854 5450 436f 6e6e 6563  lient.HTTPConnec
-00014ca0: 7469 6f6e 2e64 6562 7567 6c65 7665 6c20  tion.debuglevel 
-00014cb0: 3d20 3020 2320 7075 7420 3520 746f 2064  = 0 # put 5 to d
-00014cc0: 6562 7567 0a0a 2020 2020 696d 706f 7274  ebug..    import
-00014cd0: 2073 7973 0a20 2020 2066 726f 6d20 7070   sys.    from pp
-00014ce0: 7269 6e74 2069 6d70 6f72 7420 7070 7269  rint import ppri
-00014cf0: 6e74 0a0a 2020 2020 6170 695f 696e 7374  nt..    api_inst
-00014d00: 616e 6365 203d 2047 656d 6d61 5079 2829  ance = GemmaPy()
-00014d10: 0a0a 2020 2020 6465 6620 6765 745f 6461  ..    def get_da
-00014d20: 7461 7365 745f 616e 6e6f 7461 7469 6f6e  taset_annotation
-00014d30: 735f 7465 7374 2829 3a0a 2020 2020 2020  s_test():.      
-00014d40: 2020 7072 696e 7428 2754 6573 7469 6e67    print('Testing
-00014d50: 2067 6574 5f64 6174 6173 6574 5f61 6e6e   get_dataset_ann
-00014d60: 6f74 6174 696f 6e73 2066 756e 6374 696f  otations functio
-00014d70: 6e3a 2729 0a20 2020 2020 2020 2061 7069  n:').        api
-00014d80: 5f72 6573 706f 6e73 6520 3d20 6170 695f  _response = api_
-00014d90: 696e 7374 616e 6365 2e67 6574 5f64 6174  instance.get_dat
-00014da0: 6173 6574 5f61 6e6e 6f74 6174 696f 6e73  aset_annotations
-00014db0: 2822 4753 4534 3634 3136 2229 0a20 2020  ("GSE46416").   
-00014dc0: 2020 2020 2070 7072 696e 7428 6170 695f       pprint(api_
-00014dd0: 7265 7370 6f6e 7365 2e64 6174 6129 0a20  response.data). 
-00014de0: 2020 2020 2020 2070 7269 6e74 2827 2729         print('')
-00014df0: 0a0a 2020 2020 6465 6620 7365 6172 6368  ..    def search
-00014e00: 5f64 6174 6173 6574 735f 7465 7374 2829  _datasets_test()
-00014e10: 3a0a 2020 2020 2020 2020 7072 696e 7428  :.        print(
-00014e20: 2754 6573 7469 6e67 2073 6561 7263 685f  'Testing search_
-00014e30: 6461 7461 7365 7473 2066 756e 6374 696f  datasets functio
-00014e40: 6e3a 2729 0a20 2020 2020 2020 2061 7069  n:').        api
-00014e50: 5f72 6573 706f 6e73 6520 3d20 6170 695f  _response = api_
-00014e60: 696e 7374 616e 6365 2e73 6561 7263 685f  instance.search_
-00014e70: 6461 7461 7365 7473 285b 2262 6970 6f6c  datasets(["bipol
-00014e80: 6172 225d 2c20 7461 786f 6e3d 2268 756d  ar"], taxon="hum
-00014e90: 616e 222c 206c 696d 6974 3d31 3030 290a  an", limit=100).
-00014ea0: 2020 2020 2020 2020 666f 7220 6420 696e          for d in
-00014eb0: 2061 7069 5f72 6573 706f 6e73 652e 6461   api_response.da
-00014ec0: 7461 3a0a 2020 2020 2020 2020 2020 2020  ta:.            
-00014ed0: 6966 2064 2e67 6565 7120 6973 206e 6f74  if d.geeq is not
-00014ee0: 204e 6f6e 6520 616e 6420 2064 2e67 6565   None and  d.gee
-00014ef0: 712e 6261 7463 685f 636f 7272 6563 7465  q.batch_correcte
-00014f00: 643a 0a20 2020 2020 2020 2020 2020 2020  d:.             
-00014f10: 2020 2070 7269 6e74 2864 2e73 686f 7274     print(d.short
-00014f20: 5f6e 616d 652c 2064 2e6e 616d 652c 2064  _name, d.name, d
-00014f30: 2e62 696f 5f61 7373 6179 5f63 6f75 6e74  .bio_assay_count
-00014f40: 290a 2020 2020 2020 2020 7072 696e 7428  ).        print(
-00014f50: 2727 290a 0a20 2020 2064 6566 2067 6574  '')..    def get
-00014f60: 5f64 6174 6173 6574 735f 6279 5f69 6473  _datasets_by_ids
-00014f70: 5f74 6573 7428 293a 0a20 2020 2020 2020  _test():.       
-00014f80: 2070 7269 6e74 2827 5465 7374 696e 6720   print('Testing 
-00014f90: 6765 745f 6461 7461 7365 7473 5f62 795f  get_datasets_by_
-00014fa0: 6964 7320 6675 6e63 7469 6f6e 3a27 290a  ids function:').
-00014fb0: 2020 2020 2020 2020 6170 695f 7265 7370          api_resp
-00014fc0: 6f6e 7365 203d 2061 7069 5f69 6e73 7461  onse = api_insta
-00014fd0: 6e63 652e 6765 745f 6461 7461 7365 7473  nce.get_datasets
-00014fe0: 5f62 795f 6964 7328 5b22 4753 4534 3634  _by_ids(["GSE464
-00014ff0: 3136 225d 290a 2020 2020 2020 2020 666f  16"]).        fo
-00015000: 7220 6420 696e 2061 7069 5f72 6573 706f  r d in api_respo
-00015010: 6e73 652e 6461 7461 3a0a 2020 2020 2020  nse.data:.      
-00015020: 2020 2020 2020 7072 696e 7428 642e 7368        print(d.sh
-00015030: 6f72 745f 6e61 6d65 2c20 642e 6e61 6d65  ort_name, d.name
-00015040: 2c20 642e 6964 2c20 642e 6465 7363 7269  , d.id, d.descri
-00015050: 7074 696f 6e29 0a20 2020 2020 2020 2070  ption).        p
-00015060: 7269 6e74 2827 2729 0a0a 2020 2020 6465  rint('')..    de
-00015070: 6620 7365 6172 6368 5f61 6e6e 6f74 6174  f search_annotat
-00015080: 696f 6e73 5f74 6573 7428 293a 0a20 2020  ions_test():.   
-00015090: 2020 2020 2070 7269 6e74 2827 5465 7374       print('Test
-000150a0: 696e 6720 7365 6172 6368 5f61 6e6e 6f74  ing search_annot
-000150b0: 6174 696f 6e73 2066 756e 6374 696f 6e3a  ations function:
-000150c0: 2729 0a20 2020 2020 2020 2061 7069 5f72  ').        api_r
-000150d0: 6573 706f 6e73 6520 3d20 6170 695f 696e  esponse = api_in
-000150e0: 7374 616e 6365 2e73 6561 7263 685f 616e  stance.search_an
-000150f0: 6e6f 7461 7469 6f6e 7328 5b22 4753 4534  notations(["GSE4
-00015100: 3634 3136 225d 290a 2020 2020 2020 2020  6416"]).        
-00015110: 7072 696e 7428 6170 695f 7265 7370 6f6e  print(api_respon
-00015120: 7365 2e64 6174 6129 0a0a 2020 2020 6465  se.data)..    de
-00015130: 6620 6765 745f 6461 7461 7365 745f 6f62  f get_dataset_ob
-00015140: 6a65 6374 5f74 6573 7428 293a 0a20 2020  ject_test():.   
-00015150: 2020 2020 2070 7269 6e74 2827 5465 7374       print('Test
-00015160: 696e 6720 6765 745f 6461 7461 7365 745f  ing get_dataset_
-00015170: 6f62 6a65 6374 2066 756e 6374 696f 6e3a  object function:
-00015180: 2729 0a20 2020 2020 2020 2061 6461 7461  ').        adata
-00015190: 203d 2061 7069 5f69 6e73 7461 6e63 652e   = api_instance.
-000151a0: 6765 745f 6461 7461 7365 745f 6f62 6a65  get_dataset_obje
-000151b0: 6374 2822 4753 4534 3634 3136 2229 0a20  ct("GSE46416"). 
-000151c0: 2020 2020 2020 2070 7269 6e74 2861 6461         print(ada
-000151d0: 7461 290a 2020 2020 2020 2020 7072 696e  ta).        prin
-000151e0: 7428 275c 6e55 6e73 7472 7563 7475 7265  t('\nUnstructure
-000151f0: 6420 496e 666f 3a27 290a 2020 2020 2020  d Info:').      
-00015200: 2020 666f 7220 6c20 696e 2061 6461 7461    for l in adata
-00015210: 2e75 6e73 3a20 7072 696e 7428 2725 733a  .uns: print('%s:
-00015220: 2025 7327 2025 2028 6c2c 2061 6461 7461   %s' % (l, adata
-00015230: 2e75 6e73 5b6c 5d29 290a 2020 2020 2020  .uns[l])).      
-00015240: 2020 7072 696e 7428 2727 290a 2020 2020    print('').    
-00015250: 2020 2020 7072 696e 7428 274f 6273 6572      print('Obser
-00015260: 7661 7469 6f6e 206d 6574 6164 6174 613a  vation metadata:
-00015270: 2729 0a20 2020 2020 2020 2070 7269 6e74  ').        print
-00015280: 2861 6461 7461 2e6f 6273 2e69 6c6f 635b  (adata.obs.iloc[
-00015290: 313a 3130 5d29 0a20 2020 2020 2020 2070  1:10]).        p
-000152a0: 7269 6e74 2827 2729 0a20 2020 2020 2020  rint('').       
-000152b0: 2070 7269 6e74 2827 5661 7273 206d 6574   print('Vars met
-000152c0: 6164 6174 613a 2729 0a20 2020 2020 2020  adata:').       
-000152d0: 2070 7269 6e74 2861 6461 7461 2e76 6172   print(adata.var
-000152e0: 2e69 6c6f 635b 313a 3130 5d29 0a20 2020  .iloc[1:10]).   
-000152f0: 2020 2020 2070 7269 6e74 2827 2729 0a20       print(''). 
-00015300: 2020 2020 2020 2070 7269 6e74 2827 4461         print('Da
-00015310: 7461 3a27 290a 2020 2020 2020 2020 7072  ta:').        pr
-00015320: 696e 7428 6164 6174 612e 5829 0a20 2020  int(adata.X).   
-00015330: 2020 2020 2070 7269 6e74 2827 2729 0a0a       print('')..
-00015340: 2020 2020 2020 2020 7072 696e 7428 2753          print('S
-00015350: 7562 7365 7474 696e 6720 7468 6520 4578  ubsetting the Ex
-00015360: 7065 7269 6d65 6e74 5c6e 4469 7365 6173  periment\nDiseas
-00015370: 6520 6c65 7665 6c73 2729 0a20 2020 2020  e levels').     
-00015380: 2020 2070 7269 6e74 2861 6461 7461 2e76     print(adata.v
-00015390: 6172 5b27 6469 7365 6173 6527 5d2e 756e  ar['disease'].un
-000153a0: 6971 7565 2829 290a 2020 2020 2020 2020  ique()).        
-000153b0: 7072 696e 7428 275c 6e53 7562 7365 743a  print('\nSubset:
-000153c0: 2070 6174 6965 6e74 7320 6475 7269 6e67   patients during
-000153d0: 206d 616e 6963 2070 6861 7365 2061 6e64   manic phase and
-000153e0: 2063 6f6e 7472 6f6c 7327 290a 2020 2020   controls').    
-000153f0: 2020 2020 6d61 6e69 633d 6164 6174 615b      manic=adata[
-00015400: 3a2c 2861 6461 7461 2e76 6172 5b27 6469  :,(adata.var['di
-00015410: 7365 6173 6527 5d20 3d3d 2027 7265 6665  sease'] == 'refe
-00015420: 7265 6e63 655f 7375 626a 6563 745f 726f  rence_subject_ro
-00015430: 6c65 2729 207c 0a20 2020 2020 2020 2020  le') |.         
-00015440: 2020 2020 2020 2020 2020 2020 2028 6164               (ad
-00015450: 6174 612e 7661 725b 2764 6973 6561 7365  ata.var['disease
-00015460: 275d 203d 3d20 2762 6970 6f6c 6172 5f64  '] == 'bipolar_d
-00015470: 6973 6f72 6465 725f 7c5f 6d61 6e69 635f  isorder_|_manic_
-00015480: 7068 6173 655f 7c27 295d 2e63 6f70 7928  phase_|')].copy(
-00015490: 290a 2020 2020 2020 2020 7072 696e 7428  ).        print(
-000154a0: 6d61 6e69 6329 0a20 2020 2020 2020 2070  manic).        p
-000154b0: 7269 6e74 286d 616e 6963 2e76 6172 290a  rint(manic.var).
-000154c0: 0a20 2020 2064 6566 2067 6574 5f72 6573  .    def get_res
-000154d0: 756c 745f 7365 7473 5f74 6573 7428 293a  ult_sets_test():
-000154e0: 0a20 2020 2020 2020 2070 7269 6e74 2827  .        print('
-000154f0: 5465 7374 696e 6720 6765 745f 7265 7375  Testing get_resu
-00015500: 6c74 5f73 6574 7320 6675 6e63 7469 6f6e  lt_sets function
-00015510: 3a27 290a 2020 2020 2020 2020 6466 203d  :').        df =
-00015520: 2061 7069 5f69 6e73 7461 6e63 652e 6765   api_instance.ge
-00015530: 745f 7265 7375 6c74 5f73 6574 7328 2747  t_result_sets('G
-00015540: 5345 3637 3131 2729 0a20 2020 2020 2020  SE6711').       
-00015550: 2070 7269 6e74 2864 6629 0a20 2020 2020   print(df).     
-00015560: 2020 2064 6632 203d 2061 7069 5f69 6e73     df2 = api_ins
-00015570: 7461 6e63 652e 6765 745f 7265 7375 6c74  tance.get_result
-00015580: 5f73 6574 2834 3835 3430 3629 0a20 2020  _set(485406).   
-00015590: 2020 2020 2070 7269 6e74 2864 6632 290a       print(df2).
-000155a0: 0a20 2020 2064 6566 2067 6574 5f64 6966  .    def get_dif
-000155b0: 6665 7265 6e74 6961 6c5f 6578 7072 6573  ferential_expres
-000155c0: 7369 6f6e 5f76 616c 7565 735f 7465 7374  sion_values_test
-000155d0: 2829 3a0a 2020 2020 2020 2020 6465 203d  ():.        de =
-000155e0: 2061 7069 5f69 6e73 7461 6e63 652e 6765   api_instance.ge
-000155f0: 745f 6469 6666 6572 656e 7469 616c 5f65  t_differential_e
-00015600: 7870 7265 7373 696f 6e5f 7661 6c75 6573  xpression_values
-00015610: 2827 4753 4534 3634 3136 2729 0a20 2020  ('GSE46416').   
-00015620: 2020 2020 2064 655b 2764 6966 6665 7870       de['diffexp
-00015630: 7227 5d20 3d20 274e 6f27 2020 2023 2061  r'] = 'No'   # a
-00015640: 6464 2065 7874 7261 2063 6f6c 756d 6e0a  dd extra column.
-00015650: 2020 2020 2020 2020 2364 655b 2764 6966          #de['dif
-00015660: 6665 7870 7227 5d5b 2864 655b 2763 6f6e  fexpr'][(de['con
-00015670: 7472 6173 745f 6269 706f 6c61 7220 6469  trast_bipolar di
-00015680: 736f 7264 6572 2c20 6d61 6e69 6320 7068  sorder, manic ph
-00015690: 6173 655f 6c6f 6746 6f6c 6443 6861 6e67  ase_logFoldChang
-000156a0: 6527 5d3e 312e 3029 2026 0a20 2020 2020  e']>1.0) &.     
-000156b0: 2020 2023 2020 2020 2020 2020 2020 2020     #            
-000156c0: 2020 2028 6465 5b27 636f 6e74 7261 7374     (de['contrast
-000156d0: 5f62 6970 6f6c 6172 2064 6973 6f72 6465  _bipolar disorde
-000156e0: 722c 206d 616e 6963 2070 6861 7365 5f70  r, manic phase_p
-000156f0: 7661 6c75 6527 5d3c 302e 3035 295d 203d  value']<0.05)] =
-00015700: 2027 5570 270a 2020 2020 2020 2020 6465   'Up'.        de
-00015710: 2e6c 6f63 5b28 6465 5b27 636f 6e74 7261  .loc[(de['contra
-00015720: 7374 5f62 6970 6f6c 6172 2064 6973 6f72  st_bipolar disor
-00015730: 6465 722c 206d 616e 6963 2070 6861 7365  der, manic phase
-00015740: 5f6c 6f67 466f 6c64 4368 616e 6765 275d  _logFoldChange']
-00015750: 203e 2031 2e30 2920 260a 2020 2020 2020   > 1.0) &.      
-00015760: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00015770: 2028 6465 5b27 636f 6e74 7261 7374 5f62   (de['contrast_b
-00015780: 6970 6f6c 6172 2064 6973 6f72 6465 722c  ipolar disorder,
-00015790: 206d 616e 6963 2070 6861 7365 5f70 7661   manic phase_pva
-000157a0: 6c75 6527 5d20 3c20 302e 3035 292c 2764  lue'] < 0.05),'d
-000157b0: 6966 6665 7870 7227 5d20 3d20 2755 7027  iffexpr'] = 'Up'
-000157c0: 0a20 2020 2020 2020 2064 652e 6c6f 635b  .        de.loc[
-000157d0: 2864 655b 2763 6f6e 7472 6173 745f 6269  (de['contrast_bi
-000157e0: 706f 6c61 7220 6469 736f 7264 6572 2c20  polar disorder, 
-000157f0: 6d61 6e69 6320 7068 6173 655f 6c6f 6746  manic phase_logF
-00015800: 6f6c 6443 6861 6e67 6527 5d20 3c20 2d31  oldChange'] < -1
-00015810: 2e30 2920 260a 2020 2020 2020 2020 2020  .0) &.          
-00015820: 2020 2020 2020 2020 2020 2020 2028 6465               (de
-00015830: 5b27 636f 6e74 7261 7374 5f62 6970 6f6c  ['contrast_bipol
-00015840: 6172 2064 6973 6f72 6465 722c 206d 616e  ar disorder, man
-00015850: 6963 2070 6861 7365 5f70 7661 6c75 6527  ic phase_pvalue'
-00015860: 5d20 3c20 302e 3035 292c 2764 6966 6665  ] < 0.05),'diffe
-00015870: 7870 7227 5d20 3d20 2744 6f77 6e27 0a20  xpr'] = 'Down'. 
-00015880: 2020 2020 2020 2064 655f 7570 203d 2064         de_up = d
-00015890: 655b 6465 5b27 6469 6666 6578 7072 275d  e[de['diffexpr']
-000158a0: 3d3d 2755 7027 5d0a 2020 2020 2020 2020  =='Up'].        
-000158b0: 6465 5f75 7020 3d20 6465 5f75 705b 5b27  de_up = de_up[['
-000158c0: 5072 6f62 6527 2c27 4765 6e65 5379 6d62  Probe','GeneSymb
-000158d0: 6f6c 272c 2027 636f 6e74 7261 7374 5f62  ol', 'contrast_b
-000158e0: 6970 6f6c 6172 2064 6973 6f72 6465 722c  ipolar disorder,
-000158f0: 206d 616e 6963 2070 6861 7365 5f70 7661   manic phase_pva
-00015900: 6c75 6527 2c0a 2020 2020 2020 2020 2020  lue',.          
-00015910: 2020 2020 2020 2020 2020 2020 2027 636f               'co
-00015920: 6e74 7261 7374 5f62 6970 6f6c 6172 2064  ntrast_bipolar d
-00015930: 6973 6f72 6465 722c 206d 616e 6963 2070  isorder, manic p
-00015940: 6861 7365 5f6c 6f67 466f 6c64 4368 616e  hase_logFoldChan
-00015950: 6765 275d 5d2e 736f 7274 5f76 616c 7565  ge']].sort_value
-00015960: 7328 0a20 2020 2020 2020 2020 2020 2020  s(.             
-00015970: 2020 2027 636f 6e74 7261 7374 5f62 6970     'contrast_bip
-00015980: 6f6c 6172 2064 6973 6f72 6465 722c 206d  olar disorder, m
-00015990: 616e 6963 2070 6861 7365 5f70 7661 6c75  anic phase_pvalu
-000159a0: 6527 290a 2020 2020 2020 2020 7072 696e  e').        prin
-000159b0: 7428 2755 7072 6567 756c 6174 6564 2070  t('Upregulated p
-000159c0: 726f 6265 7327 290a 2020 2020 2020 2020  robes').        
-000159d0: 7769 7468 2070 642e 6f70 7469 6f6e 5f63  with pd.option_c
-000159e0: 6f6e 7465 7874 2827 6469 7370 6c61 792e  ontext('display.
-000159f0: 6d61 785f 726f 7773 272c 204e 6f6e 652c  max_rows', None,
-00015a00: 2027 6469 7370 6c61 792e 6d61 785f 636f   'display.max_co
-00015a10: 6c75 6d6e 7327 2c20 4e6f 6e65 293a 0a20  lumns', None):. 
-00015a20: 2020 2020 2020 2020 2020 2070 7269 6e74             print
-00015a30: 2864 655f 7570 5b3a 3130 5d29 0a0a 2020  (de_up[:10])..  
-00015a40: 2020 2020 2020 6465 5f64 6e20 3d20 6465        de_dn = de
-00015a50: 5b64 655b 2764 6966 6665 7870 7227 5d3d  [de['diffexpr']=
-00015a60: 3d27 446f 776e 275d 0a20 2020 2020 2020  ='Down'].       
-00015a70: 2064 655f 646e 203d 2064 655f 646e 5b5b   de_dn = de_dn[[
-00015a80: 2750 726f 6265 272c 2747 656e 6553 796d  'Probe','GeneSym
-00015a90: 626f 6c27 2c20 2763 6f6e 7472 6173 745f  bol', 'contrast_
-00015aa0: 6269 706f 6c61 7220 6469 736f 7264 6572  bipolar disorder
-00015ab0: 2c20 6d61 6e69 6320 7068 6173 655f 7076  , manic phase_pv
-00015ac0: 616c 7565 272c 0a20 2020 2020 2020 2020  alue',.         
-00015ad0: 2020 2020 2020 2020 2020 2020 2020 2763                'c
-00015ae0: 6f6e 7472 6173 745f 6269 706f 6c61 7220  ontrast_bipolar 
-00015af0: 6469 736f 7264 6572 2c20 6d61 6e69 6320  disorder, manic 
-00015b00: 7068 6173 655f 6c6f 6746 6f6c 6443 6861  phase_logFoldCha
-00015b10: 6e67 6527 5d5d 2e73 6f72 745f 7661 6c75  nge']].sort_valu
-00015b20: 6573 280a 2020 2020 2020 2020 2020 2020  es(.            
-00015b30: 2020 2020 2763 6f6e 7472 6173 745f 6269      'contrast_bi
-00015b40: 706f 6c61 7220 6469 736f 7264 6572 2c20  polar disorder, 
-00015b50: 6d61 6e69 6320 7068 6173 655f 7076 616c  manic phase_pval
-00015b60: 7565 2729 0a20 2020 2020 2020 2070 7269  ue').        pri
-00015b70: 6e74 2827 5c6e 446f 776e 7265 6775 6c61  nt('\nDownregula
-00015b80: 7465 6420 7072 6f62 6573 2729 0a20 2020  ted probes').   
-00015b90: 2020 2020 2077 6974 6820 7064 2e6f 7074       with pd.opt
-00015ba0: 696f 6e5f 636f 6e74 6578 7428 2764 6973  ion_context('dis
-00015bb0: 706c 6179 2e6d 6178 5f72 6f77 7327 2c20  play.max_rows', 
-00015bc0: 4e6f 6e65 2c20 2764 6973 706c 6179 2e6d  None, 'display.m
-00015bd0: 6178 5f63 6f6c 756d 6e73 272c 204e 6f6e  ax_columns', Non
-00015be0: 6529 3a0a 2020 2020 2020 2020 2020 2020  e):.            
-00015bf0: 7072 696e 7428 6465 5f64 6e5b 3a31 305d  print(de_dn[:10]
-00015c00: 290a 0a20 2020 2064 6566 2067 6574 5f70  )..    def get_p
-00015c10: 6c61 7466 6f72 6d5f 616e 6e6f 7461 7469  latform_annotati
-00015c20: 6f6e 735f 7465 7374 2829 3a0a 2020 2020  ons_test():.    
-00015c30: 2020 2020 7072 696e 7428 2754 6573 7469      print('Testi
-00015c40: 6e67 2067 6574 5f70 6c61 7466 6f72 6d5f  ng get_platform_
-00015c50: 616e 6e6f 7461 7469 6f6e 7320 6675 6e63  annotations func
-00015c60: 7469 6f6e 2729 0a20 2020 2020 2020 2064  tion').        d
-00015c70: 6620 3d20 6170 695f 696e 7374 616e 6365  f = api_instance
-00015c80: 2e67 6574 5f70 6c61 7466 6f72 6d5f 616e  .get_platform_an
-00015c90: 6e6f 7461 7469 6f6e 7328 2747 504c 3936  notations('GPL96
-00015ca0: 2729 0a20 2020 2020 2020 2070 7269 6e74  ').        print
-00015cb0: 2864 6629 0a0a 2020 2020 6966 206c 656e  (df)..    if len
-00015cc0: 2873 7973 2e61 7267 7629 203e 2031 3a0a  (sys.argv) > 1:.
-00015cd0: 2020 2020 2020 2020 666e 616d 6520 3d20          fname = 
-00015ce0: 7379 732e 6172 6776 5b31 5d0a 2020 2020  sys.argv[1].    
-00015cf0: 2020 2020 6469 6374 203d 2076 6172 7328      dict = vars(
-00015d00: 290a 2020 2020 2020 2020 6966 2066 6e61  ).        if fna
-00015d10: 6d65 2069 6e20 6469 6374 3a0a 2020 2020  me in dict:.    
-00015d20: 2020 2020 2020 2020 2370 7269 6e74 2822          #print("
-00015d30: 4361 6c6c 696e 6722 2c66 6e61 6d65 290a  Calling",fname).
-00015d40: 2020 2020 2020 2020 2020 2020 6675 6e63              func
-00015d50: 203d 2064 6963 745b 666e 616d 655d 0a20   = dict[fname]. 
-00015d60: 2020 2020 2020 2020 2020 2066 756e 6328             func(
-00015d70: 290a                                     ).
+000125a0: 2020 2020 2072 6561 6461 626c 655f 636f       readable_co
+000125b0: 6e74 7261 7374 733a 626f 6f6c 203d 2046  ntrasts:bool = F
+000125c0: 616c 7365 2c20 0a20 2020 2020 2020 2020  alse, .         
+000125d0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000125e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000125f0: 2020 2a2a 6b77 6172 6773 292d 3e4c 6973    **kwargs)->Lis
+00012600: 745b 4461 7461 4672 616d 655d 3a0a 2020  t[DataFrame]:.  
+00012610: 2020 2020 2020 2222 220a 2020 2020 2020        """.      
+00012620: 2020 5265 7472 6965 7665 7320 7468 6520    Retrieves the 
+00012630: 6469 6666 6572 656e 7469 616c 2065 7870  differential exp
+00012640: 7265 7373 696f 6e20 7265 7375 6c74 5365  ression resultSe
+00012650: 7428 7329 2061 7373 6f63 6961 7465 6420  t(s) associated 
+00012660: 7769 7468 2074 6865 2064 6174 6173 6574  with the dataset
+00012670: 2e0a 2020 2020 2020 2020 4966 2074 6865  ..        If the
+00012680: 7265 2069 7320 6d6f 7265 2074 6861 6e20  re is more than 
+00012690: 6f6e 6520 7265 7375 6c74 5365 742c 2075  one resultSet, u
+000126a0: 7365 2067 6574 5f72 6573 756c 745f 7365  se get_result_se
+000126b0: 7473 2829 2074 6f20 7365 6520 7468 6520  ts() to see the 
+000126c0: 6f70 7469 6f6e 730a 2020 2020 2020 2020  options.        
+000126d0: 616e 6420 6765 7420 7468 6520 4944 2079  and get the ID y
+000126e0: 6f75 2077 616e 742e 2041 6c74 6572 6e61  ou want. Alterna
+000126f0: 7469 7665 6c79 2c20 796f 7520 6361 6e20  tively, you can 
+00012700: 7175 6572 7920 7468 6520 7265 7375 6c74  query the result
+00012710: 5365 7420 6469 7265 6374 6c79 0a20 2020  Set directly.   
+00012720: 2020 2020 2069 6620 796f 7520 6b6e 6f77       if you know
+00012730: 2069 7473 2049 4420 6265 666f 7265 6861   its ID beforeha
+00012740: 6e64 2e0a 2020 2020 2020 2020 0a20 2020  nd..        .   
+00012750: 2020 2020 2049 6e20 4765 6d6d 6120 6561       In Gemma ea
+00012760: 6368 2072 6573 756c 7420 7365 7420 636f  ch result set co
+00012770: 7272 6573 706f 6e64 7320 746f 2074 6865  rresponds to the
+00012780: 2065 7374 696d 6174 6564 2065 6666 6563   estimated effec
+00012790: 7473 200a 2020 2020 2020 2020 6173 736f  ts .        asso
+000127a0: 6369 6174 6564 2077 6974 6820 6120 7369  ciated with a si
+000127b0: 6e67 6c65 2066 6163 746f 7220 696e 2074  ngle factor in t
+000127c0: 6865 2064 6573 6967 6e2c 2061 6e64 2065  he design, and e
+000127d0: 6163 6820 6361 6e20 6861 7665 200a 2020  ach can have .  
+000127e0: 2020 2020 2020 6d75 6c74 6970 6c65 2063        multiple c
+000127f0: 6f6e 7472 6173 7473 2028 666f 7220 6561  ontrasts (for ea
+00012800: 6368 206c 6576 656c 2063 6f6d 7061 7265  ch level compare
+00012810: 6420 746f 2062 6173 656c 696e 6529 2e20  d to baseline). 
+00012820: 5468 7573 2061 200a 2020 2020 2020 2020  Thus a .        
+00012830: 6461 7461 7365 7420 7769 7468 2061 2032  dataset with a 2
+00012840: 7833 2066 6163 746f 7269 616c 2064 6573  x3 factorial des
+00012850: 6967 6e20 7769 6c6c 2068 6176 6520 7477  ign will have tw
+00012860: 6f20 7265 7375 6c74 2073 6574 732c 206f  o result sets, o
+00012870: 6e65 206f 6620 0a20 2020 2020 2020 2077  ne of .        w
+00012880: 6869 6368 2077 696c 6c20 6861 7665 206f  hich will have o
+00012890: 6e65 2063 6f6e 7472 6173 742c 2061 6e64  ne contrast, and
+000128a0: 206f 6e65 2068 6176 696e 6720 7477 6f20   one having two 
+000128b0: 636f 6e74 7261 7374 732e 0a20 2020 2020  contrasts..     
+000128c0: 2020 200a 2020 2020 2020 2020 5468 6520     .        The 
+000128d0: 6d65 7468 6f64 6f6c 6f67 7920 666f 7220  methodology for 
+000128e0: 6469 6666 6572 656e 7469 616c 2065 7870  differential exp
+000128f0: 7265 7373 696f 6e20 6973 2065 7870 6c61  ression is expla
+00012900: 696e 6564 2069 6e20 6043 7572 6174 696f  ined in `Curatio
+00012910: 6e20 0a20 2020 2020 2020 206f 6620 6f76  n .        of ov
+00012920: 6572 2031 3030 3030 2074 7261 6e73 6372  er 10000 transcr
+00012930: 6970 746f 6d69 6320 7374 7564 6965 7320  iptomic studies 
+00012940: 746f 2065 6e61 626c 6520 6461 7461 2072  to enable data r
+00012950: 6575 7365 203c 6874 7470 733a 2f2f 646f  euse <https://do
+00012960: 692e 6f72 672f 3130 2e31 3039 332f 6461  i.org/10.1093/da
+00012970: 7461 6261 7365 2f62 6161 6230 3036 3e60  tabase/baab006>`
+00012980: 5f2e 0a20 2020 2020 2020 2042 7269 6566  _..        Brief
+00012990: 6c79 2c20 6469 6666 6572 656e 7469 616c  ly, differential
+000129a0: 2065 7870 7265 7373 696f 6e20 616e 616c   expression anal
+000129b0: 7973 6973 2069 7320 7065 7266 6f72 6d65  ysis is performe
+000129c0: 6420 6f6e 2074 6865 2064 6174 6173 6574  d on the dataset
+000129d0: 0a20 2020 2020 2020 2062 6173 6564 206f  .        based o
+000129e0: 6e20 7468 6520 616e 6e6f 7461 7465 6420  n the annotated 
+000129f0: 6578 7065 7269 6d65 6e74 616c 2064 6573  experimental des
+00012a00: 6967 6e20 7769 7468 2075 7020 7477 6f20  ign with up two 
+00012a10: 7468 7265 6520 706f 7465 6e74 6961 6c6c  three potentiall
+00012a20: 790a 2020 2020 2020 2020 6e65 7374 6564  y.        nested
+00012a30: 2066 6163 746f 7273 2e20 4765 6d6d 6120   factors. Gemma 
+00012a40: 6174 7465 6d70 7473 2074 6f20 6175 746f  attempts to auto
+00012a50: 6d61 7469 6361 6c6c 7920 6173 7369 676e  matically assign
+00012a60: 2062 6173 656c 696e 6520 636f 6e64 6974   baseline condit
+00012a70: 696f 6e73 0a20 2020 2020 2020 2066 6f72  ions.        for
+00012a80: 2065 6163 6820 6661 6374 6f72 2e20 496e   each factor. In
+00012a90: 2074 6865 2061 6273 656e 6365 206f 6620   the absence of 
+00012aa0: 6120 636c 6561 7220 636f 6e74 726f 6c20  a clear control 
+00012ab0: 636f 6e64 6974 696f 6e2c 2061 2062 6173  condition, a bas
+00012ac0: 656c 696e 6520 0a20 2020 2020 2020 2069  eline .        i
+00012ad0: 7320 6172 6269 7472 6172 696c 7920 7365  s arbitrarily se
+00012ae0: 6c65 6374 6564 2e20 4120 6765 6e65 7261  lected. A genera
+00012af0: 6c69 7a65 6420 6c69 6e65 6172 206d 6f64  lized linear mod
+00012b00: 656c 2077 6974 6820 656d 7069 7269 6361  el with empirica
+00012b10: 6c20 4261 7965 730a 2020 2020 2020 2020  l Bayes.        
+00012b20: 7368 7269 6e6b 6167 6520 6f66 2074 2d73  shrinkage of t-s
+00012b30: 7461 7469 7374 6963 7320 6973 2066 6974  tatistics is fit
+00012b40: 2074 6f20 7468 6520 6461 7461 2066 6f72   to the data for
+00012b50: 2065 6163 6820 706c 6174 666f 726d 2065   each platform e
+00012b60: 6c65 6d65 6e74 200a 2020 2020 2020 2020  lement .        
+00012b70: 2870 726f 6265 2f67 656e 6529 2075 7369  (probe/gene) usi
+00012b80: 6e67 2061 6e20 696d 706c 656d 656e 7461  ng an implementa
+00012b90: 7469 6f6e 206f 6620 7468 6520 6c69 6d6d  tion of the limm
+00012ba0: 6120 616c 676f 7269 7468 6d2e 2046 6f72  a algorithm. For
+00012bb0: 200a 2020 2020 2020 2020 524e 412d 7365   .        RNA-se
+00012bc0: 7120 6461 7461 2c20 7765 2075 7365 2077  q data, we use w
+00012bd0: 6569 6768 7465 6420 7265 6772 6573 7369  eighted regressi
+00012be0: 6f6e 2c20 6170 706c 7969 6e67 2074 6865  on, applying the
+00012bf0: 2076 6f6f 6d20 616c 676f 7269 7468 6d0a   voom algorithm.
+00012c00: 2020 2020 2020 2020 746f 2063 6f6d 7075          to compu
+00012c10: 7465 2077 6569 6768 7473 2066 726f 6d20  te weights from 
+00012c20: 7468 6520 6d65 616e e280 9376 6172 6961  the mean...varia
+00012c30: 6e63 6520 7265 6c61 7469 6f6e 7368 6970  nce relationship
+00012c40: 206f 6620 7468 6520 6461 7461 2e20 0a20   of the data. . 
+00012c50: 2020 2020 2020 2043 6f6e 7472 6173 7473         Contrasts
+00012c60: 206f 6620 6561 6368 2063 6f6e 6469 7469   of each conditi
+00012c70: 6f6e 2061 7265 2074 6865 6e20 636f 6d70  on are then comp
+00012c80: 7574 6564 2063 6f6d 7061 7265 6420 746f  uted compared to
+00012c90: 2074 6865 2073 656c 6563 7465 6420 0a20   the selected . 
+00012ca0: 2020 2020 2020 2062 6173 656c 696e 652e         baseline.
+00012cb0: 2049 6e20 736f 6d65 2073 6974 7561 7469   In some situati
+00012cc0: 6f6e 732c 2047 656d 6d61 2077 696c 6c20  ons, Gemma will 
+00012cd0: 7370 6c69 7420 7468 6520 6461 7461 2069  split the data i
+00012ce0: 6e74 6f20 7375 6273 6574 7320 666f 720a  nto subsets for.
+00012cf0: 2020 2020 2020 2020 616e 616c 7973 6973          analysis
+00012d00: 2e20 4120 7479 7069 6361 6c20 7375 6368  . A typical such
+00012d10: 2073 6974 7561 7469 6f6e 2069 7320 7768   situation is wh
+00012d20: 656e 2061 20e2 8098 6261 7463 68e2 8099  en a ...batch...
+00012d30: 2066 6163 746f 7220 6973 2070 7265 7365   factor is prese
+00012d40: 6e74 200a 2020 2020 2020 2020 616e 6420  nt .        and 
+00012d50: 636f 6e66 6f75 6e64 6564 2077 6974 6820  confounded with 
+00012d60: 616e 6f74 6865 7220 6661 6374 6f72 2c20  another factor, 
+00012d70: 7468 6520 7375 6273 6574 7320 6265 696e  the subsets bein
+00012d80: 6720 6465 7465 726d 696e 6564 2062 7920  g determined by 
+00012d90: 7468 650a 2020 2020 2020 2020 6c65 7665  the.        leve
+00012da0: 6c73 206f 6620 7468 6520 636f 6e66 6f75  ls of the confou
+00012db0: 6e64 696e 6720 6661 6374 6f72 2e0a 2020  nding factor..  
+00012dc0: 2020 2020 2020 0a20 2020 2020 2020 200a        .        .
+00012dd0: 2020 2020 2020 2020 3a70 6172 616d 2064          :param d
+00012de0: 6174 6173 6574 3a20 4120 6461 7461 7365  ataset: A datase
+00012df0: 7420 6964 656e 7469 6669 6572 2c20 6465  t identifier, de
+00012e00: 6661 756c 7473 2074 6f20 4e6f 6e65 0a20  faults to None. 
+00012e10: 2020 2020 2020 203a 7479 7065 2064 6174         :type dat
+00012e20: 6173 6574 3a20 4f70 7469 6f6e 616c 5b73  aset: Optional[s
+00012e30: 7472 7c69 6e74 5d2c 206f 7074 696f 6e61  tr|int], optiona
+00012e40: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
+00012e50: 2072 6573 756c 745f 7365 7473 3a20 7265   result_sets: re
+00012e60: 7375 6c74 2073 6574 2069 6465 6e74 6966  sult set identif
+00012e70: 6965 7273 2e20 4966 2061 2064 6174 6173  iers. If a datas
+00012e80: 6574 0a20 2020 2020 2020 2020 2069 7320  et.          is 
+00012e90: 6e6f 7420 7072 6f76 6964 6564 2c20 616c  not provided, al
+00012ea0: 6c20 7265 7375 6c74 2073 6574 7320 7769  l result sets wi
+00012eb0: 6c6c 2062 6520 646f 776e 6c6f 6164 6564  ll be downloaded
+00012ec0: 2e20 4966 2069 7420 6973 2070 726f 7669  . If it is provi
+00012ed0: 6465 640a 2020 2020 2020 2020 2020 6974  ded.          it
+00012ee0: 2077 696c 6c20 6f6e 6c79 2062 6520 7573   will only be us
+00012ef0: 6564 2074 6f20 656e 7375 7265 2061 6c6c  ed to ensure all
+00012f00: 2072 6573 756c 7420 7365 7473 2062 656c   result sets bel
+00012f10: 6f6e 6720 746f 2074 6865 2064 6174 6173  ong to the datas
+00012f20: 6574 2c20 6465 6661 756c 7473 2074 6f20  et, defaults to 
+00012f30: 4e6f 6e65 0a20 2020 2020 2020 203a 7479  None.        :ty
+00012f40: 7065 2072 6573 756c 745f 7365 7473 3a20  pe result_sets: 
+00012f50: 4f70 7469 6f6e 616c 5b4c 6973 745b 7374  Optional[List[st
+00012f60: 727c 696e 745d 5d2c 206f 7074 696f 6e61  r|int]], optiona
+00012f70: 6c0a 2020 2020 2020 2020 3a70 6172 616d  l.        :param
+00012f80: 2072 6561 6461 626c 655f 636f 6e74 7261   readable_contra
+00012f90: 7374 733a 2049 6620 4661 6c73 6520 2864  sts: If False (d
+00012fa0: 6566 6175 6c74 292c 2074 6865 2072 6574  efault), the ret
+00012fb0: 7572 6e65 6420 636f 6c75 6d6e 730a 2020  urned columns.  
+00012fc0: 2020 2020 2020 2020 7769 6c6c 2075 7365          will use
+00012fd0: 2069 6e74 6572 6e61 6c20 636f 6e73 7472   internal constr
+00012fe0: 6173 7473 2049 4473 2061 7320 6e61 6d65  asts IDs as name
+00012ff0: 732e 2044 6574 6169 6c73 2061 626f 7574  s. Details about
+00013000: 2074 6865 2063 6f6e 7472 6173 7473 2063   the contrasts c
+00013010: 616e 2062 650a 2020 2020 2020 2020 2020  an be.          
+00013020: 6163 6365 7373 6564 2075 7369 6e67 2067  accessed using g
+00013030: 6574 5f64 6174 6173 6574 5f64 6966 6665  et_dataset_diffe
+00013040: 7265 6e74 6961 6c5f 6578 7072 6573 7369  rential_expressi
+00013050: 6f6e 5f61 6e61 6c79 7365 7328 292e 2049  on_analyses(). I
+00013060: 6620 5472 7565 2049 4473 0a20 2020 2020  f True IDs.     
+00013070: 2020 2020 2077 696c 6c20 6265 2072 6570       will be rep
+00013080: 6c61 6365 6420 7769 7468 2068 756d 616e  laced with human
+00013090: 2072 6561 6461 626c 6520 636f 6e74 7261   readable contra
+000130a0: 7374 2069 6e66 6f72 6d61 7469 6f6e 2c20  st information, 
+000130b0: 6465 6661 756c 7473 2074 6f20 4661 6c73  defaults to Fals
+000130c0: 650a 2020 2020 2020 2020 3a74 7970 6520  e.        :type 
+000130d0: 7265 6164 6162 6c65 5f63 6f6e 7472 6173  readable_contras
+000130e0: 7473 3a20 626f 6f6c 2c20 6f70 7469 6f6e  ts: bool, option
+000130f0: 616c 0a20 2020 2020 2020 203a 7061 7261  al.        :para
+00013100: 6d20 2a2a 6b77 6172 6773 3a20 0a20 2020  m **kwargs: .   
+00013110: 2020 2020 203a 7479 7065 202a 2a6b 7761       :type **kwa
+00013120: 7267 733a 2054 5950 450a 2020 2020 2020  rgs: TYPE.      
+00013130: 2020 3a72 6169 7365 7320 5661 6c75 6545    :raises ValueE
+00013140: 7272 6f72 3a20 5769 6c6c 2072 6574 7572  rror: Will retur
+00013150: 6e20 6120 7661 6c75 6520 6572 726f 7220  n a value error 
+00013160: 6966 206e 6569 7468 6572 2072 6573 756c  if neither resul
+00013170: 745f 7365 7473 206e 6f72 2061 2064 6174  t_sets nor a dat
+00013180: 6173 6574 0a20 2020 2020 2020 2020 2069  aset.          i
+00013190: 7320 7072 6f76 6964 6564 0a20 2020 2020  s provided.     
+000131a0: 2020 203a 7265 7475 726e 3a20 4120 6c69     :return: A li
+000131b0: 7374 206f 6620 6461 7461 2066 7261 6d65  st of data frame
+000131c0: 7320 7769 7468 2064 6966 6665 7265 6e74  s with different
+000131d0: 6961 6c20 6578 7072 6573 7369 6f6e 2076  ial expression v
+000131e0: 616c 7565 7320 7065 7220 7265 7375 6c74  alues per result
+000131f0: 2073 6574 2e0a 2020 2020 2020 2020 3a72   set..        :r
+00013200: 7479 7065 3a20 4c69 7374 5b44 6174 6146  type: List[DataF
+00013210: 7261 6d65 5d0a 0a20 2020 2020 2020 2022  rame]..        "
+00013220: 2222 0a20 2020 2020 2020 2069 6620 6461  "".        if da
+00013230: 7461 7365 7420 6973 206e 6f74 204e 6f6e  taset is not Non
+00013240: 6520 616e 6420 7265 7375 6c74 5f73 6574  e and result_set
+00013250: 7320 6973 206e 6f74 204e 6f6e 653a 0a20  s is not None:. 
+00013260: 2020 2020 2020 2020 2020 2064 6966 6673             diffs
+00013270: 203d 2020 7365 6c66 2e67 6574 5f64 6174   =  self.get_dat
+00013280: 6173 6574 5f64 6966 6665 7265 6e74 6961  aset_differentia
+00013290: 6c5f 6578 7072 6573 7369 6f6e 5f61 6e61  l_expression_ana
+000132a0: 6c79 7365 7328 6461 7461 7365 7429 0a20  lyses(dataset). 
+000132b0: 2020 2020 2020 2020 2020 2072 7373 203d             rss =
+000132c0: 2064 6966 6673 2e72 6573 756c 745f 4944   diffs.result_ID
+000132d0: 0a20 2020 2020 2020 2020 2020 2069 6620  .            if 
+000132e0: 6e6f 7420 616c 6c28 7375 622e 6c69 7374  not all(sub.list
+000132f0: 5f69 6e5f 6c69 7374 2872 6573 756c 745f  _in_list(result_
+00013300: 7365 7473 2c20 7273 7329 293a 0a20 2020  sets, rss)):.   
+00013310: 2020 2020 2020 2020 2020 2020 2077 6172               war
+00013320: 6e69 6e67 732e 7761 726e 2827 5468 6520  nings.warn('The 
+00013330: 7175 6572 6965 6420 7265 7375 6c74 5365  queried resultSe
+00013340: 7420 6973 206e 6f74 2064 6572 6976 6564  t is not derived
+00013350: 2066 726f 6d20 7468 6973 2064 6174 6173   from this datas
+00013360: 6574 2e20 2720 0a20 2020 2020 2020 2020  et. ' .         
+00013370: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00013380: 2020 2020 2027 4368 6563 6b20 7468 6520       'Check the 
+00013390: 6176 6169 6c61 626c 6520 7265 7375 6c74  available result
+000133a0: 5365 7473 2077 6974 6820 2267 6574 5f72  Sets with "get_r
+000133b0: 6573 756c 745f 7365 7473 2829 2220 270a  esult_sets()" '.
+000133c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000133d0: 2020 2020 2020 2020 2020 2020 2020 276f                'o
+000133e0: 7220 7175 6572 7920 7769 7468 6f75 7420  r query without 
+000133f0: 7468 6520 6461 7461 7365 7420 7061 7261  the dataset para
+00013400: 6d65 7465 722e 2729 0a20 2020 2020 2020  meter.').       
+00013410: 2020 2020 2020 2020 2072 6574 7572 6e0a           return.
+00013420: 2020 2020 2020 2020 656c 6966 2064 6174          elif dat
+00013430: 6173 6574 2069 7320 6e6f 7420 4e6f 6e65  aset is not None
+00013440: 2061 6e64 2072 6573 756c 745f 7365 7473   and result_sets
+00013450: 2069 7320 4e6f 6e65 3a0a 2020 2020 2020   is None:.      
+00013460: 2020 2020 2020 6469 6666 7320 3d20 2073        diffs =  s
+00013470: 656c 662e 6765 745f 6461 7461 7365 745f  elf.get_dataset_
+00013480: 6469 6666 6572 656e 7469 616c 5f65 7870  differential_exp
+00013490: 7265 7373 696f 6e5f 616e 616c 7973 6573  ression_analyses
+000134a0: 2864 6174 6173 6574 290a 2020 2020 2020  (dataset).      
+000134b0: 2020 2020 2020 7265 7375 6c74 5f73 6574        result_set
+000134c0: 7320 3d20 6469 6666 732e 7265 7375 6c74  s = diffs.result
+000134d0: 5f49 442e 756e 6971 7565 2829 0a20 2020  _ID.unique().   
+000134e0: 2020 2020 2065 6c69 6620 6461 7461 7365       elif datase
+000134f0: 7420 6973 204e 6f6e 6520 616e 6420 7265  t is None and re
+00013500: 7375 6c74 5f73 6574 7320 6973 204e 6f6e  sult_sets is Non
+00013510: 653a 0a20 2020 2020 2020 2020 2020 2072  e:.            r
+00013520: 6169 7365 2056 616c 7565 4572 726f 7228  aise ValueError(
+00013530: 2759 6f75 206d 7573 7420 7370 6563 6966  'You must specif
+00013540: 7920 6120 6461 7461 7365 7420 6f72 2072  y a dataset or r
+00013550: 6573 756c 745f 7365 7473 2729 0a20 2020  esult_sets').   
+00013560: 2020 2020 2020 2020 200a 2020 2020 2020           .      
+00013570: 2020 7273 7320 3d20 7b7d 0a20 2020 2020    rss = {}.     
+00013580: 2020 2069 6620 7265 6164 6162 6c65 5f63     if readable_c
+00013590: 6f6e 7472 6173 7473 3a0a 2020 2020 2020  ontrasts:.      
+000135a0: 2020 2020 2020 616c 6c5f 6661 6374 6f72        all_factor
+000135b0: 7320 3d20 2073 656c 662e 6765 745f 7265  s =  self.get_re
+000135c0: 7375 6c74 5f73 6574 7328 7265 7375 6c74  sult_sets(result
+000135d0: 5f73 6574 7320 3d20 7265 7375 6c74 5f73  _sets = result_s
+000135e0: 6574 7329 0a20 2020 2020 2020 200a 2020  ets).        .  
+000135f0: 2020 2020 2020 666f 7220 7273 2069 6e20        for rs in 
+00013600: 7265 7375 6c74 5f73 6574 733a 0a20 2020  result_sets:.   
+00013610: 2020 2020 2020 2020 2064 6620 3d20 7365           df = se
+00013620: 6c66 2e5f 5f67 6574 5f72 6573 756c 745f  lf.__get_result_
+00013630: 7365 7428 7273 290a 2020 2020 2020 2020  set(rs).        
+00013640: 2020 2020 6966 2072 6561 6461 626c 655f      if readable_
+00013650: 636f 6e74 7261 7374 733a 0a20 2020 2020  contrasts:.     
+00013660: 2020 2020 2020 2020 2020 2066 6163 746f             facto
+00013670: 7273 203d 2070 642e 636f 6e63 6174 280a  rs = pd.concat(.
+00013680: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00013690: 2020 2020 6c69 7374 2861 6c6c 5f66 6163      list(all_fac
+000136a0: 746f 7273 5b61 6c6c 5f66 6163 746f 7273  tors[all_factors
+000136b0: 2e72 6573 756c 745f 4944 203d 3d20 7273  .result_ID == rs
+000136c0: 5d2e 6578 7065 7269 6d65 6e74 616c 5f66  ].experimental_f
+000136d0: 6163 746f 7273 290a 2020 2020 2020 2020  actors).        
+000136e0: 2020 2020 2020 2020 2020 2020 292e 6472              ).dr
+000136f0: 6f70 5f64 7570 6c69 6361 7465 7328 290a  op_duplicates().
+00013700: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00013710: 636f 6c73 203d 205b 732e 7265 706c 6163  cols = [s.replac
+00013720: 6528 276c 6f67 3266 6327 2c27 6c6f 6746  e('log2fc','logF
+00013730: 6f6c 6443 6861 6e67 6527 2920 666f 7220  oldChange') for 
+00013740: 7320 696e 2064 662e 636f 6c75 6d6e 735d  s in df.columns]
+00013750: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+00013760: 2066 6f72 2069 2069 6e20 7261 6e67 6528   for i in range(
+00013770: 6661 6374 6f72 732e 7368 6170 655b 305d  factors.shape[0]
+00013780: 293a 0a20 2020 2020 2020 2020 2020 2020  ):.             
+00013790: 2020 2020 2020 2063 6f6c 7320 3d20 5b73         cols = [s
+000137a0: 2e72 6570 6c61 6365 2873 7472 286c 6973  .replace(str(lis
+000137b0: 7428 6661 6374 6f72 732e 4944 295b 695d  t(factors.ID)[i]
+000137c0: 292c 6c69 7374 2866 6163 746f 7273 2e73  ),list(factors.s
+000137d0: 756d 6d61 7279 295b 695d 2920 666f 7220  ummary)[i]) for 
+000137e0: 7320 696e 2063 6f6c 735d 0a20 2020 2020  s in cols].     
+000137f0: 2020 2020 2020 2020 2020 2064 662e 636f             df.co
+00013800: 6c75 6d6e 7320 3d20 636f 6c73 0a20 2020  lumns = cols.   
+00013810: 2020 2020 2020 2020 2072 7373 5b72 735d           rss[rs]
+00013820: 203d 2064 660a 2020 2020 2020 2020 7265   = df.        re
+00013830: 7475 726e 2072 7373 0a20 2020 2020 2020  turn rss.       
+00013840: 2020 2020 200a 0a20 2020 2023 2067 6574       ..    # get
+00013850: 5f74 6178 6120 6973 206d 6f76 6564 2074  _taxa is moved t
+00013860: 6f20 6261 7365 2e20 6f6e 6c79 2072 656d  o base. only rem
+00013870: 6f76 6573 2074 6865 206e 616d 656c 6573  oves the nameles
+00013880: 7320 7261 7420 6e6f 770a 0a20 2020 2023  s rat now..    #
+00013890: 2067 656d 6d61 5f63 616c 6c20 756e 696d   gemma_call unim
+000138a0: 706c 656d 656e 7465 642c 206e 6f74 206e  plemented, not n
+000138b0: 6565 6465 6420 2020 200a 0a20 2020 2064  eeded    ..    d
+000138c0: 6566 2067 6574 5f61 6c6c 5f70 6167 6573  ef get_all_pages
+000138d0: 2873 656c 662c 6675 6e3a 4361 6c6c 6162  (self,fun:Callab
+000138e0: 6c65 2c73 7465 705f 7369 7a65 3a69 6e74  le,step_size:int
+000138f0: 203d 2031 3030 2c2a 2a6b 7761 7267 7329   = 100,**kwargs)
+00013900: 2d3e 6c69 7374 7c44 6174 6146 7261 6d65  ->list|DataFrame
+00013910: 3a0a 2020 2020 2020 2020 2222 220a 2020  :.        """.  
+00013920: 2020 2020 2020 4120 636f 6e76 656e 6965        A convenie
+00013930: 6e63 6520 6675 6e63 7469 6f6e 2074 6f20  nce function to 
+00013940: 616c 6c6f 7720 6561 7379 2069 7465 7261  allow easy itera
+00013950: 7469 6f6e 206f 7665 7220 7061 6769 6e61  tion over pagina
+00013960: 7465 6420 6f75 7470 7574 732e 0a20 2020  ted outputs..   
+00013970: 2020 2020 2049 6620 7468 6520 6675 6e63       If the func
+00013980: 7469 6f6e 2072 6574 7572 6e73 2061 2044  tion returns a D
+00013990: 6174 6146 7261 6d65 206f 7574 7075 7420  ataFrame output 
+000139a0: 7769 6c6c 2062 6520 6d65 7267 6564 2062  will be merged b
+000139b0: 7920 7468 6520 726f 7773 2c0a 2020 2020  y the rows,.    
+000139c0: 2020 2020 6966 2074 6865 2066 756e 6374      if the funct
+000139d0: 696f 6e20 7265 7475 726e 7320 6120 6c69  ion returns a li
+000139e0: 7374 2028 6567 2e20 2772 6177 2720 6675  st (eg. 'raw' fu
+000139f0: 6e63 7469 6f6e 7329 2061 2063 6f6e 6361  nctions) a conca
+00013a00: 7461 6e61 7465 6420 6c69 7374 0a20 2020  tanated list.   
+00013a10: 2020 2020 2077 696c 6c20 6265 2072 6574       will be ret
+00013a20: 7572 6e65 640a 2020 2020 2020 2020 0a20  urned.        . 
+00013a30: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+00013a40: 3a70 6172 616d 2066 756e 3a20 4120 6361  :param fun: A ca
+00013a50: 6c6c 6162 6c65 2066 726f 6d20 6765 6d6d  llable from gemm
+00013a60: 6170 7920 7769 7468 206f 6666 7365 7420  apy with offset 
+00013a70: 616e 6420 6c69 6d69 740a 2020 2020 2020  and limit.      
+00013a80: 2020 2020 6675 6e63 7469 6f6e 730a 2020      functions.  
+00013a90: 2020 2020 2020 3a74 7970 6520 6675 6e3a        :type fun:
+00013aa0: 2043 616c 6c61 626c 650a 2020 2020 2020   Callable.      
+00013ab0: 2020 3a70 6172 616d 2073 7465 705f 7369    :param step_si
+00013ac0: 7a65 3a20 5369 7a65 206f 6620 696e 6469  ze: Size of indi
+00013ad0: 7669 6475 616c 2063 616c 6c73 2074 6f20  vidual calls to 
+00013ae0: 7468 6520 7365 7276 6572 2e20 3130 3020  the server. 100 
+00013af0: 6973 200a 2020 2020 2020 2020 2020 7468  is .          th
+00013b00: 6520 6d61 7869 6d75 6d20 7661 6c75 6520  e maximum value 
+00013b10: 616e 6420 7468 6520 6465 6661 756c 742e  and the default.
+00013b20: 0a20 2020 2020 2020 203a 7479 7065 2073  .        :type s
+00013b30: 7465 705f 7369 7a65 3a20 696e 742c 206f  tep_size: int, o
+00013b40: 7074 696f 6e61 6c0a 2020 2020 2020 2020  ptional.        
+00013b50: 3a70 6172 616d 202a 2a6b 7761 7267 733a  :param **kwargs:
+00013b60: 2061 7267 756d 656e 7473 2066 6f72 2074   arguments for t
+00013b70: 6865 2063 616c 6c61 626c 6520 6675 6e0a  he callable fun.
+00013b80: 2020 2020 2020 2020 3a72 6574 7572 6e3a          :return:
+00013b90: 2041 2044 6174 6146 7261 6d65 206f 7220   A DataFrame or 
+00013ba0: 6120 6c69 7374 2063 6f6e 7461 696e 696e  a list containin
+00013bb0: 6720 616c 6c20 7468 6520 6f75 7470 7574  g all the output
+00013bc0: 2064 6570 656e 6469 6e67 206f 6e20 0a20   depending on . 
+00013bd0: 2020 2020 2020 2020 206f 7574 7075 7420           output 
+00013be0: 6f66 2074 6865 2063 616c 6c61 626c 650a  of the callable.
+00013bf0: 2020 2020 2020 2020 3a72 7479 7065 3a20          :rtype: 
+00013c00: 6c69 7374 7c44 6174 6146 7261 6d65 0a20  list|DataFrame. 
+00013c10: 2020 2020 2020 2022 2222 0a0a 2020 2020         """..    
+00013c20: 2020 2020 6f75 7420 3d20 5b5d 2020 2020      out = []    
+00013c30: 2020 2020 0a20 2020 2020 2020 2070 6f6b      .        pok
+00013c40: 655f 6361 6c6c 203d 2066 756e 286c 696d  e_call = fun(lim
+00013c50: 6974 203d 312c 2a2a 6b77 6172 6773 290a  it =1,**kwargs).
+00013c60: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00013c70: 2069 6620 7479 7065 2870 6f6b 655f 6361   if type(poke_ca
+00013c80: 6c6c 2920 3d3d 2070 642e 636f 7265 2e66  ll) == pd.core.f
+00013c90: 7261 6d65 2e44 6174 6146 7261 6d65 3a0a  rame.DataFrame:.
+00013ca0: 2020 2020 2020 2020 2020 2020 636f 756e              coun
+00013cb0: 7420 3d20 706f 6b65 5f63 616c 6c2e 6174  t = poke_call.at
+00013cc0: 7472 6962 7574 6573 5b22 746f 7461 6c5f  tributes["total_
+00013cd0: 656c 656d 656e 7473 225d 0a20 2020 2020  elements"].     
+00013ce0: 2020 2065 6c73 653a 0a20 2020 2020 2020     else:.       
+00013cf0: 2020 2020 2063 6f75 6e74 203d 2070 6f6b       count = pok
+00013d00: 655f 6361 6c6c 2e74 6f74 616c 5f65 6c65  e_call.total_ele
+00013d10: 6d65 6e74 730a 2020 2020 2020 2020 0a20  ments.        . 
+00013d20: 2020 2020 2020 2066 6f72 2069 2069 6e20         for i in 
+00013d30: 7261 6e67 6528 302c 636f 756e 742c 7374  range(0,count,st
+00013d40: 6570 5f73 697a 6529 3a0a 2020 2020 2020  ep_size):.      
+00013d50: 2020 2020 2020 6f75 742e 6170 7065 6e64        out.append
+00013d60: 2866 756e 286c 696d 6974 203d 2073 7465  (fun(limit = ste
+00013d70: 705f 7369 7a65 2c6f 6666 7365 7420 3d20  p_size,offset = 
+00013d80: 692c 2a2a 6b77 6172 6773 2929 0a20 2020  i,**kwargs)).   
+00013d90: 2020 2020 200a 2020 2020 2020 2020 6966       .        if
+00013da0: 2074 7970 6528 706f 6b65 5f63 616c 6c29   type(poke_call)
+00013db0: 203d 3d20 7064 2e63 6f72 652e 6672 616d   == pd.core.fram
+00013dc0: 652e 4461 7461 4672 616d 653a 0a20 2020  e.DataFrame:.   
+00013dd0: 2020 2020 2020 2020 2072 6574 7572 6e20           return 
+00013de0: 7064 2e63 6f6e 6361 7428 6f75 742c 6967  pd.concat(out,ig
+00013df0: 6e6f 7265 5f69 6e64 6578 203d 2054 7275  nore_index = Tru
+00013e00: 6529 0a20 2020 2020 2020 2065 6c73 653a  e).        else:
+00013e10: 0a20 2020 2020 2020 2020 2020 2072 6574  .            ret
+00013e20: 7572 6e20 7375 622e 6272 6561 6b5f 6c69  urn sub.break_li
+00013e30: 7374 285b 782e 6461 7461 2066 6f72 2078  st([x.data for x
+00013e40: 2069 6e20 6f75 745d 290a 2020 2020 2020   in out]).      
+00013e50: 2020 0a0a 2020 2020 0a20 2020 2064 6566    ..    .    def
+00013e60: 2066 696c 7465 725f 7072 6f70 6572 7469   filter_properti
+00013e70: 6573 2873 656c 662c 206f 7574 7075 745f  es(self, output_
+00013e80: 7479 7065 3a73 7472 203d 2027 4461 7461  type:str = 'Data
+00013e90: 4672 616d 6527 292d 3e64 6963 747c 4461  Frame')->dict|Da
+00013ea0: 7461 4672 616d 653a 0a20 2020 2020 2020  taFrame:.       
+00013eb0: 2022 2222 0a20 2020 2020 2020 2053 6f6d   """.        Som
+00013ec0: 6520 6675 6e63 7469 6f6e 7320 7375 6368  e functions such
+00013ed0: 2061 7320 6765 745f 6461 7461 7365 7473   as get_datasets
+00013ee0: 2061 6e64 2067 6574 5f70 6c61 7466 6f72   and get_platfor
+00013ef0: 6d73 2069 6e63 6c75 6465 2061 2066 696c  ms include a fil
+00013f00: 7465 720a 2020 2020 2020 2020 6172 6775  ter.        argu
+00013f10: 6d65 6e74 2074 6861 7420 616c 6c6f 7773  ment that allows
+00013f20: 2063 7265 6174 696f 6e20 6f66 206d 6f72   creation of mor
+00013f30: 6520 636f 6d70 6c65 7820 7175 6572 6965  e complex querie
+00013f40: 732e 2054 6869 7320 6675 6e63 7469 6f6e  s. This function
+00013f50: 200a 2020 2020 2020 2020 7265 7475 726e   .        return
+00013f60: 7320 6120 6c69 7374 206f 6620 7375 7070  s a list of supp
+00013f70: 6f72 7465 6420 7072 6f70 6572 7469 6573  orted properties
+00013f80: 2074 6f20 6265 2075 7365 6420 696e 2074   to be used in t
+00013f90: 686f 7365 2066 696c 7465 7273 0a20 2020  hose filters.   
+00013fa0: 2020 2020 200a 2020 2020 2020 2020 3a70       .        :p
+00013fb0: 6172 616d 206f 7574 7075 745f 7479 7065  aram output_type
+00013fc0: 3a20 5479 7065 2074 6f20 7265 7475 726e  : Type to return
+00013fd0: 2e20 2244 6174 6146 7261 6d65 2220 6f72  . "DataFrame" or
+00013fe0: 2022 6469 6374 222c 2064 6566 6175 6c74   "dict", default
+00013ff0: 7320 746f 2027 4461 7461 4672 616d 6527  s to 'DataFrame'
+00014000: 0a20 2020 2020 2020 203a 7479 7065 206f  .        :type o
+00014010: 7574 7075 745f 7479 7065 3a20 7374 722c  utput_type: str,
+00014020: 206f 7074 696f 6e61 6c0a 2020 2020 2020   optional.      
+00014030: 2020 3a72 6574 7572 6e3a 2044 6174 6146    :return: DataF
+00014040: 7261 6d65 206f 7220 6469 6374 2063 6f6e  rame or dict con
+00014050: 7461 696e 696e 6720 7375 7070 6f72 7465  taining supporte
+00014060: 6420 7072 6f70 6572 7469 6573 2061 6e64  d properties and
+00014070: 2074 6865 6972 2064 6174 6120 7479 7065   their data type
+00014080: 730a 2020 2020 2020 2020 3a72 7479 7065  s.        :rtype
+00014090: 3a20 6469 6374 7c44 6174 6146 7261 6d65  : dict|DataFrame
+000140a0: 0a0a 2020 2020 2020 2020 2222 220a 2020  ..        """.  
+000140b0: 2020 2020 2020 0a20 2020 2020 2020 2064        .        d
+000140c0: 203d 2073 656c 662e 7261 772e 6170 695f   = self.raw.api_
+000140d0: 636c 6965 6e74 2e72 6573 745f 636c 6965  client.rest_clie
+000140e0: 6e74 2e47 4554 2822 6874 7470 733a 2f2f  nt.GET("https://
+000140f0: 6765 6d6d 612e 6d73 6c2e 7562 632e 6361  gemma.msl.ubc.ca
+00014100: 2f72 6573 742f 7632 2f6f 7065 6e61 7069  /rest/v2/openapi
+00014110: 2e6a 736f 6e22 292e 7572 6c6c 6962 335f  .json").urllib3_
+00014120: 7265 7370 6f6e 7365 0a20 2020 2020 2020  response.       
+00014130: 2061 7069 5f66 696c 6520 3d20 6a73 6f6e   api_file = json
+00014140: 2e6c 6f61 6473 2864 2e64 6174 6129 0a20  .loads(d.data). 
+00014150: 2020 2020 2020 200a 2020 2020 2020 2020         .        
+00014160: 6461 7461 7365 745f 6669 6c74 6572 203d  dataset_filter =
+00014170: 2061 7069 5f66 696c 655b 2263 6f6d 706f   api_file["compo
+00014180: 6e65 6e74 7322 5d5b 2273 6368 656d 6173  nents"]["schemas
+00014190: 225d 5b22 4669 6c74 6572 4172 6745 7870  "]["FilterArgExp
+000141a0: 7265 7373 696f 6e45 7870 6572 696d 656e  ressionExperimen
+000141b0: 7422 5d5b 2278 2d67 656d 6d61 2d66 696c  t"]["x-gemma-fil
+000141c0: 7465 7261 626c 652d 7072 6f70 6572 7469  terable-properti
+000141d0: 6573 225d 0a20 2020 2020 2020 200a 2020  es"].        .  
+000141e0: 2020 2020 2020 706c 6174 666f 726d 5f66        platform_f
+000141f0: 696c 7465 7220 3d20 6170 695f 6669 6c65  ilter = api_file
+00014200: 5b22 636f 6d70 6f6e 656e 7473 225d 5b22  ["components"]["
+00014210: 7363 6865 6d61 7322 5d5b 2246 696c 7465  schemas"]["Filte
+00014220: 7241 7267 4172 7261 7944 6573 6967 6e22  rArgArrayDesign"
+00014230: 5d5b 2278 2d67 656d 6d61 2d66 696c 7465  ]["x-gemma-filte
+00014240: 7261 626c 652d 7072 6f70 6572 7469 6573  rable-properties
+00014250: 225d 0a20 2020 2020 2020 200a 2020 2020  "].        .    
+00014260: 2020 2020 7265 7375 6c74 5f73 6574 5f66      result_set_f
+00014270: 696c 7465 7220 3d20 6170 695f 6669 6c65  ilter = api_file
+00014280: 5b22 636f 6d70 6f6e 656e 7473 225d 5b22  ["components"]["
+00014290: 7363 6865 6d61 7322 5d5b 2246 696c 7465  schemas"]["Filte
+000142a0: 7241 7267 4578 7072 6573 7369 6f6e 416e  rArgExpressionAn
+000142b0: 616c 7973 6973 5265 7375 6c74 5365 7422  alysisResultSet"
+000142c0: 5d5b 2278 2d67 656d 6d61 2d66 696c 7465  ]["x-gemma-filte
+000142d0: 7261 626c 652d 7072 6f70 6572 7469 6573  rable-properties
+000142e0: 225d 0a20 2020 2020 2020 200a 2020 2020  "].        .    
+000142f0: 2020 2020 6966 206f 7574 7075 745f 7479      if output_ty
+00014300: 7065 203d 3d20 2744 6174 6146 7261 6d65  pe == 'DataFrame
+00014310: 273a 0a20 2020 2020 2020 2020 2020 2072  ':.            r
+00014320: 6574 7572 6e20 7b0a 2020 2020 2020 2020  eturn {.        
+00014330: 2020 2020 2020 2020 2264 6174 6173 6574          "dataset
+00014340: 223a 7064 2e44 6174 6146 7261 6d65 287b  ":pd.DataFrame({
+00014350: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+00014360: 2020 2020 2022 6e61 6d65 223a 2073 7562       "name": sub
+00014370: 2e66 6965 6c64 5f69 6e5f 6c69 7374 2864  .field_in_list(d
+00014380: 6174 6173 6574 5f66 696c 7465 722c 276e  ataset_filter,'n
+00014390: 616d 6527 292c 0a20 2020 2020 2020 2020  ame'),.         
+000143a0: 2020 2020 2020 2020 2020 2022 7479 7065             "type
+000143b0: 223a 2073 7562 2e66 6965 6c64 5f69 6e5f  ": sub.field_in_
+000143c0: 6c69 7374 2864 6174 6173 6574 5f66 696c  list(dataset_fil
+000143d0: 7465 722c 2774 7970 6527 292c 0a20 2020  ter,'type'),.   
+000143e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000143f0: 2022 6465 7363 7269 7074 696f 6e22 3a20   "description": 
+00014400: 7375 622e 6669 656c 645f 696e 5f6c 6973  sub.field_in_lis
+00014410: 7428 6461 7461 7365 745f 6669 6c74 6572  t(dataset_filter
+00014420: 2c27 6465 7363 7269 7074 696f 6e27 290a  ,'description').
+00014430: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00014440: 2020 2020 7d29 2c0a 2020 2020 2020 2020      }),.        
+00014450: 2020 2020 2020 2020 2270 6c61 7466 6f72          "platfor
+00014460: 6d22 3a70 642e 4461 7461 4672 616d 6528  m":pd.DataFrame(
+00014470: 7b0a 2020 2020 2020 2020 2020 2020 2020  {.              
+00014480: 2020 2020 2020 226e 616d 6522 3a20 7375        "name": su
+00014490: 622e 6669 656c 645f 696e 5f6c 6973 7428  b.field_in_list(
+000144a0: 706c 6174 666f 726d 5f66 696c 7465 722c  platform_filter,
+000144b0: 276e 616d 6527 292c 0a20 2020 2020 2020  'name'),.       
+000144c0: 2020 2020 2020 2020 2020 2020 2022 7479               "ty
+000144d0: 7065 223a 2073 7562 2e66 6965 6c64 5f69  pe": sub.field_i
+000144e0: 6e5f 6c69 7374 2870 6c61 7466 6f72 6d5f  n_list(platform_
+000144f0: 6669 6c74 6572 2c27 7479 7065 2729 2c0a  filter,'type'),.
+00014500: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00014510: 2020 2020 2264 6573 6372 6970 7469 6f6e      "description
+00014520: 223a 2073 7562 2e66 6965 6c64 5f69 6e5f  ": sub.field_in_
+00014530: 6c69 7374 2870 6c61 7466 6f72 6d5f 6669  list(platform_fi
+00014540: 6c74 6572 2c27 6465 7363 7269 7074 696f  lter,'descriptio
+00014550: 6e27 290a 2020 2020 2020 2020 2020 2020  n').            
+00014560: 2020 2020 2020 2020 7d29 2c0a 2020 2020          }),.    
+00014570: 2020 2020 2020 2020 2020 2020 2272 6573              "res
+00014580: 756c 745f 7365 7422 3a70 642e 4461 7461  ult_set":pd.Data
+00014590: 4672 616d 6528 7b0a 2020 2020 2020 2020  Frame({.        
+000145a0: 2020 2020 2020 2020 2020 2020 226e 616d              "nam
+000145b0: 6522 3a20 7375 622e 6669 656c 645f 696e  e": sub.field_in
+000145c0: 5f6c 6973 7428 7265 7375 6c74 5f73 6574  _list(result_set
+000145d0: 5f66 696c 7465 722c 276e 616d 6527 292c  _filter,'name'),
+000145e0: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+000145f0: 2020 2020 2022 7479 7065 223a 2073 7562       "type": sub
+00014600: 2e66 6965 6c64 5f69 6e5f 6c69 7374 2872  .field_in_list(r
+00014610: 6573 756c 745f 7365 745f 6669 6c74 6572  esult_set_filter
+00014620: 2c27 7479 7065 2729 2c0a 2020 2020 2020  ,'type'),.      
+00014630: 2020 2020 2020 2020 2020 2020 2020 2264                "d
+00014640: 6573 6372 6970 7469 6f6e 223a 2073 7562  escription": sub
+00014650: 2e66 6965 6c64 5f69 6e5f 6c69 7374 2872  .field_in_list(r
+00014660: 6573 756c 745f 7365 745f 6669 6c74 6572  esult_set_filter
+00014670: 2c27 6465 7363 7269 7074 696f 6e27 290a  ,'description').
+00014680: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00014690: 2020 2020 7d29 0a20 2020 2020 2020 2020      }).         
+000146a0: 2020 2020 2020 207d 0a20 2020 2020 2020         }.       
+000146b0: 2065 6c69 6620 6f75 7470 7574 5f74 7970   elif output_typ
+000146c0: 6520 3d3d 2027 6469 6374 273a 0a20 2020  e == 'dict':.   
+000146d0: 2020 2020 2020 2020 2072 6574 7572 6e20           return 
+000146e0: 7b0a 2020 2020 2020 2020 2020 2020 2020  {.              
+000146f0: 2020 2264 6174 6173 6574 223a 207b 785b    "dataset": {x[
+00014700: 276e 616d 6527 5d3a 7b22 7479 7065 223a  'name']:{"type":
+00014710: 785b 2774 7970 6527 5d2c 0a20 2020 2020  x['type'],.     
+00014720: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00014730: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00014740: 2020 2020 2020 2020 2022 6465 7363 7269           "descri
+00014750: 7074 696f 6e22 3a73 7562 2e61 6363 6573  ption":sub.acces
+00014760: 735f 6669 656c 6428 782c 2764 6573 6372  s_field(x,'descr
+00014770: 6970 7469 6f6e 272c 4e6f 6e65 297d 2066  iption',None)} f
+00014780: 6f72 2078 2069 6e20 6461 7461 7365 745f  or x in dataset_
+00014790: 6669 6c74 6572 7d2c 0a20 2020 2020 2020  filter},.       
+000147a0: 2020 2020 2020 2020 2027 706c 6174 666f           'platfo
+000147b0: 726d 273a 7b78 5b27 6e61 6d65 275d 3a7b  rm':{x['name']:{
+000147c0: 2274 7970 6522 3a78 5b27 7479 7065 275d  "type":x['type']
+000147d0: 2c0a 2020 2020 2020 2020 2020 2020 2020  ,.              
+000147e0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000147f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00014800: 2264 6573 6372 6970 7469 6f6e 223a 7375  "description":su
+00014810: 622e 6163 6365 7373 5f66 6965 6c64 2878  b.access_field(x
+00014820: 2c27 6465 7363 7269 7074 696f 6e27 2c4e  ,'description',N
+00014830: 6f6e 6529 7d20 666f 7220 7820 696e 2070  one)} for x in p
+00014840: 6c61 7466 6f72 6d5f 6669 6c74 6572 7d2c  latform_filter},
+00014850: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
+00014860: 2027 7265 7375 6c74 5f73 6574 273a 7b78   'result_set':{x
+00014870: 5b27 6e61 6d65 275d 3a7b 2274 7970 6522  ['name']:{"type"
+00014880: 3a78 5b27 7479 7065 275d 2c0a 2020 2020  :x['type'],.    
+00014890: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000148a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000148b0: 2020 2020 2020 2020 2020 2264 6573 6372            "descr
+000148c0: 6970 7469 6f6e 223a 7375 622e 6163 6365  iption":sub.acce
+000148d0: 7373 5f66 6965 6c64 2878 2c27 6465 7363  ss_field(x,'desc
+000148e0: 7269 7074 696f 6e27 2c4e 6f6e 6529 7d20  ription',None)} 
+000148f0: 666f 7220 7820 696e 2072 6573 756c 745f  for x in result_
+00014900: 7365 745f 6669 6c74 6572 7d0a 2020 2020  set_filter}.    
+00014910: 2020 2020 2020 2020 2020 2020 0a20 2020              .   
+00014920: 2020 2020 2020 2020 2020 2020 207d 0a20               }. 
+00014930: 2020 2020 2020 2020 2020 200a 2020 2020             .    
+00014940: 2020 2020 0a20 2020 200a 2020 2020 6465      .    .    de
+00014950: 6620 6765 745f 6368 696c 645f 7465 726d  f get_child_term
+00014960: 7328 7365 6c66 2c74 6572 6d73 3a4c 6973  s(self,terms:Lis
+00014970: 745b 7374 725d 292d 3e4c 6973 745b 7374  t[str])->List[st
+00014980: 725d 3a0a 2020 2020 2020 2020 2222 220a  r]:.        """.
+00014990: 2020 2020 2020 2020 5768 656e 2071 7565          When que
+000149a0: 7279 696e 6720 666f 7220 6f6e 746f 6c6f  rying for ontolo
+000149b0: 6779 2074 6572 6d73 2c20 4765 6d6d 6120  gy terms, Gemma 
+000149c0: 7072 6f70 6167 6174 6573 2074 6865 7365  propagates these
+000149d0: 2074 6572 6d73 2074 6f20 0a20 2020 2020   terms to .     
+000149e0: 2020 2069 6e63 6c75 6465 2061 6e79 2064     include any d
+000149f0: 6174 6173 6574 7320 7769 7468 2074 6865  atasets with the
+00014a00: 6972 2063 6869 6c64 2074 6572 6d73 2069  ir child terms i
+00014a10: 6e20 7468 6520 7265 7375 6c74 732e 2054  n the results. T
+00014a20: 6869 7320 0a20 2020 2020 2020 2066 756e  his .        fun
+00014a30: 6374 696f 6e20 7265 7475 726e 7320 7468  ction returns th
+00014a40: 6573 6520 6368 696c 6472 656e 2066 6f72  ese children for
+00014a50: 2061 6e79 206e 756d 6265 7220 6f66 2074   any number of t
+00014a60: 6572 6d73 2c20 696e 636c 7564 696e 6720  erms, including 
+00014a70: 616c 6c20 0a20 2020 2020 2020 2063 6869  all .        chi
+00014a80: 6c64 7265 6e20 616e 6420 7468 6520 7465  ldren and the te
+00014a90: 726d 7320 6974 7365 6c66 2069 6e20 7468  rms itself in th
+00014aa0: 6520 6f75 7470 7574 2076 6563 746f 720a  e output vector.
+00014ab0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00014ac0: 203a 7061 7261 6d20 7465 726d 733a 2041   :param terms: A
+00014ad0: 206c 6973 7420 6f66 206f 6e74 6f6c 6f67   list of ontolog
+00014ae0: 7920 7465 726d 730a 2020 2020 2020 2020  y terms.        
+00014af0: 3a74 7970 6520 7465 726d 733a 204c 6973  :type terms: Lis
+00014b00: 745b 7374 725d 0a20 2020 2020 2020 203a  t[str].        :
+00014b10: 7265 7475 726e 3a20 416e 2061 7272 6179  return: An array
+00014b20: 2063 6f6e 7461 696e 696e 6720 6465 7363   containing desc
+00014b30: 656e 6465 6e64 7320 6f66 2074 6865 2061  endends of the a
+00014b40: 6e6e 6f74 6174 696f 6e20 7465 726d 732c  nnotation terms,
+00014b50: 200a 2020 2020 2020 2020 696e 636c 7564   .        includ
+00014b60: 696e 6720 7468 6520 7465 726d 7320 7468  ing the terms th
+00014b70: 656d 7365 6c76 6573 0a20 2020 2020 2020  emselves.       
+00014b80: 203a 7274 7970 653a 204c 6973 745b 7374   :rtype: List[st
+00014b90: 725d 0a0a 2020 2020 2020 2020 2222 220a  r]..        """.
+00014ba0: 2020 2020 2020 2020 0a20 2020 2020 2020          .       
+00014bb0: 206f 7574 7075 7420 3d20 7365 6c66 2e67   output = self.g
+00014bc0: 6574 5f64 6174 6173 6574 7328 7572 6973  et_datasets(uris
+00014bd0: 3d74 6572 6d73 290a 2020 2020 2020 2020  =terms).        
+00014be0: 0a20 2020 2020 2020 2072 6574 7572 6e20  .        return 
+00014bf0: 7265 2e66 696e 6461 6c6c 2872 2768 7474  re.findall(r'htt
+00014c00: 702e 2a3f 283f 3d2c 7c5c 2929 272c 6f75  p.*?(?=,|\))',ou
+00014c10: 7470 7574 2e61 7474 7269 6275 7465 735b  tput.attributes[
+00014c20: 2766 696c 7465 7227 5d29 0a20 2020 200a  'filter']).    .
+00014c30: 2020 2020 2320 7570 6461 7465 5f72 6573      # update_res
+00014c40: 756c 7473 2063 7572 7265 6e74 7920 756e  ults currenty un
+00014c50: 696d 706c 656d 656e 7465 6420 616e 6420  implemented and 
+00014c60: 6e6f 7420 7265 616c 6c79 2065 7373 656e  not really essen
+00014c70: 7469 616c 0a0a 0a23 2054 6573 7473 0a69  tial...# Tests.i
+00014c80: 6620 5f5f 6e61 6d65 5f5f 203d 3d20 275f  f __name__ == '_
+00014c90: 5f6d 6169 6e5f 5f27 3a0a 2020 2020 2320  _main__':.    # 
+00014ca0: 6476 2074 656d 700a 2020 2020 696d 706f  dv temp.    impo
+00014cb0: 7274 2068 7474 702e 636c 6965 6e74 0a20  rt http.client. 
+00014cc0: 2020 2068 7474 702e 636c 6965 6e74 2e48     http.client.H
+00014cd0: 5454 5043 6f6e 6e65 6374 696f 6e2e 6465  TTPConnection.de
+00014ce0: 6275 676c 6576 656c 203d 2030 2023 2070  buglevel = 0 # p
+00014cf0: 7574 2035 2074 6f20 6465 6275 670a 0a20  ut 5 to debug.. 
+00014d00: 2020 2069 6d70 6f72 7420 7379 730a 2020     import sys.  
+00014d10: 2020 6672 6f6d 2070 7072 696e 7420 696d    from pprint im
+00014d20: 706f 7274 2070 7072 696e 740a 0a20 2020  port pprint..   
+00014d30: 2061 7069 5f69 6e73 7461 6e63 6520 3d20   api_instance = 
+00014d40: 4765 6d6d 6150 7928 290a 0a20 2020 2064  GemmaPy()..    d
+00014d50: 6566 2067 6574 5f64 6174 6173 6574 5f61  ef get_dataset_a
+00014d60: 6e6e 6f74 6174 696f 6e73 5f74 6573 7428  nnotations_test(
+00014d70: 293a 0a20 2020 2020 2020 2070 7269 6e74  ):.        print
+00014d80: 2827 5465 7374 696e 6720 6765 745f 6461  ('Testing get_da
+00014d90: 7461 7365 745f 616e 6e6f 7461 7469 6f6e  taset_annotation
+00014da0: 7320 6675 6e63 7469 6f6e 3a27 290a 2020  s function:').  
+00014db0: 2020 2020 2020 6170 695f 7265 7370 6f6e        api_respon
+00014dc0: 7365 203d 2061 7069 5f69 6e73 7461 6e63  se = api_instanc
+00014dd0: 652e 6765 745f 6461 7461 7365 745f 616e  e.get_dataset_an
+00014de0: 6e6f 7461 7469 6f6e 7328 2247 5345 3436  notations("GSE46
+00014df0: 3431 3622 290a 2020 2020 2020 2020 7070  416").        pp
+00014e00: 7269 6e74 2861 7069 5f72 6573 706f 6e73  rint(api_respons
+00014e10: 652e 6461 7461 290a 2020 2020 2020 2020  e.data).        
+00014e20: 7072 696e 7428 2727 290a 0a20 2020 2064  print('')..    d
+00014e30: 6566 2073 6561 7263 685f 6461 7461 7365  ef search_datase
+00014e40: 7473 5f74 6573 7428 293a 0a20 2020 2020  ts_test():.     
+00014e50: 2020 2070 7269 6e74 2827 5465 7374 696e     print('Testin
+00014e60: 6720 7365 6172 6368 5f64 6174 6173 6574  g search_dataset
+00014e70: 7320 6675 6e63 7469 6f6e 3a27 290a 2020  s function:').  
+00014e80: 2020 2020 2020 6170 695f 7265 7370 6f6e        api_respon
+00014e90: 7365 203d 2061 7069 5f69 6e73 7461 6e63  se = api_instanc
+00014ea0: 652e 7365 6172 6368 5f64 6174 6173 6574  e.search_dataset
+00014eb0: 7328 5b22 6269 706f 6c61 7222 5d2c 2074  s(["bipolar"], t
+00014ec0: 6178 6f6e 3d22 6875 6d61 6e22 2c20 6c69  axon="human", li
+00014ed0: 6d69 743d 3130 3029 0a20 2020 2020 2020  mit=100).       
+00014ee0: 2066 6f72 2064 2069 6e20 6170 695f 7265   for d in api_re
+00014ef0: 7370 6f6e 7365 2e64 6174 613a 0a20 2020  sponse.data:.   
+00014f00: 2020 2020 2020 2020 2069 6620 642e 6765           if d.ge
+00014f10: 6571 2069 7320 6e6f 7420 4e6f 6e65 2061  eq is not None a
+00014f20: 6e64 2020 642e 6765 6571 2e62 6174 6368  nd  d.geeq.batch
+00014f30: 5f63 6f72 7265 6374 6564 3a0a 2020 2020  _corrected:.    
+00014f40: 2020 2020 2020 2020 2020 2020 7072 696e              prin
+00014f50: 7428 642e 7368 6f72 745f 6e61 6d65 2c20  t(d.short_name, 
+00014f60: 642e 6e61 6d65 2c20 642e 6269 6f5f 6173  d.name, d.bio_as
+00014f70: 7361 795f 636f 756e 7429 0a20 2020 2020  say_count).     
+00014f80: 2020 2070 7269 6e74 2827 2729 0a0a 2020     print('')..  
+00014f90: 2020 6465 6620 6765 745f 6461 7461 7365    def get_datase
+00014fa0: 7473 5f62 795f 6964 735f 7465 7374 2829  ts_by_ids_test()
+00014fb0: 3a0a 2020 2020 2020 2020 7072 696e 7428  :.        print(
+00014fc0: 2754 6573 7469 6e67 2067 6574 5f64 6174  'Testing get_dat
+00014fd0: 6173 6574 735f 6279 5f69 6473 2066 756e  asets_by_ids fun
+00014fe0: 6374 696f 6e3a 2729 0a20 2020 2020 2020  ction:').       
+00014ff0: 2061 7069 5f72 6573 706f 6e73 6520 3d20   api_response = 
+00015000: 6170 695f 696e 7374 616e 6365 2e67 6574  api_instance.get
+00015010: 5f64 6174 6173 6574 735f 6279 5f69 6473  _datasets_by_ids
+00015020: 285b 2247 5345 3436 3431 3622 5d29 0a20  (["GSE46416"]). 
+00015030: 2020 2020 2020 2066 6f72 2064 2069 6e20         for d in 
+00015040: 6170 695f 7265 7370 6f6e 7365 2e64 6174  api_response.dat
+00015050: 613a 0a20 2020 2020 2020 2020 2020 2070  a:.            p
+00015060: 7269 6e74 2864 2e73 686f 7274 5f6e 616d  rint(d.short_nam
+00015070: 652c 2064 2e6e 616d 652c 2064 2e69 642c  e, d.name, d.id,
+00015080: 2064 2e64 6573 6372 6970 7469 6f6e 290a   d.description).
+00015090: 2020 2020 2020 2020 7072 696e 7428 2727          print(''
+000150a0: 290a 0a20 2020 2064 6566 2073 6561 7263  )..    def searc
+000150b0: 685f 616e 6e6f 7461 7469 6f6e 735f 7465  h_annotations_te
+000150c0: 7374 2829 3a0a 2020 2020 2020 2020 7072  st():.        pr
+000150d0: 696e 7428 2754 6573 7469 6e67 2073 6561  int('Testing sea
+000150e0: 7263 685f 616e 6e6f 7461 7469 6f6e 7320  rch_annotations 
+000150f0: 6675 6e63 7469 6f6e 3a27 290a 2020 2020  function:').    
+00015100: 2020 2020 6170 695f 7265 7370 6f6e 7365      api_response
+00015110: 203d 2061 7069 5f69 6e73 7461 6e63 652e   = api_instance.
+00015120: 7365 6172 6368 5f61 6e6e 6f74 6174 696f  search_annotatio
+00015130: 6e73 285b 2247 5345 3436 3431 3622 5d29  ns(["GSE46416"])
+00015140: 0a20 2020 2020 2020 2070 7269 6e74 2861  .        print(a
+00015150: 7069 5f72 6573 706f 6e73 652e 6461 7461  pi_response.data
+00015160: 290a 0a20 2020 2064 6566 2067 6574 5f64  )..    def get_d
+00015170: 6174 6173 6574 5f6f 626a 6563 745f 7465  ataset_object_te
+00015180: 7374 2829 3a0a 2020 2020 2020 2020 7072  st():.        pr
+00015190: 696e 7428 2754 6573 7469 6e67 2067 6574  int('Testing get
+000151a0: 5f64 6174 6173 6574 5f6f 626a 6563 7420  _dataset_object 
+000151b0: 6675 6e63 7469 6f6e 3a27 290a 2020 2020  function:').    
+000151c0: 2020 2020 6164 6174 6120 3d20 6170 695f      adata = api_
+000151d0: 696e 7374 616e 6365 2e67 6574 5f64 6174  instance.get_dat
+000151e0: 6173 6574 5f6f 626a 6563 7428 2247 5345  aset_object("GSE
+000151f0: 3436 3431 3622 290a 2020 2020 2020 2020  46416").        
+00015200: 7072 696e 7428 6164 6174 6129 0a20 2020  print(adata).   
+00015210: 2020 2020 2070 7269 6e74 2827 5c6e 556e       print('\nUn
+00015220: 7374 7275 6374 7572 6564 2049 6e66 6f3a  structured Info:
+00015230: 2729 0a20 2020 2020 2020 2066 6f72 206c  ').        for l
+00015240: 2069 6e20 6164 6174 612e 756e 733a 2070   in adata.uns: p
+00015250: 7269 6e74 2827 2573 3a20 2573 2720 2520  rint('%s: %s' % 
+00015260: 286c 2c20 6164 6174 612e 756e 735b 6c5d  (l, adata.uns[l]
+00015270: 2929 0a20 2020 2020 2020 2070 7269 6e74  )).        print
+00015280: 2827 2729 0a20 2020 2020 2020 2070 7269  ('').        pri
+00015290: 6e74 2827 4f62 7365 7276 6174 696f 6e20  nt('Observation 
+000152a0: 6d65 7461 6461 7461 3a27 290a 2020 2020  metadata:').    
+000152b0: 2020 2020 7072 696e 7428 6164 6174 612e      print(adata.
+000152c0: 6f62 732e 696c 6f63 5b31 3a31 305d 290a  obs.iloc[1:10]).
+000152d0: 2020 2020 2020 2020 7072 696e 7428 2727          print(''
+000152e0: 290a 2020 2020 2020 2020 7072 696e 7428  ).        print(
+000152f0: 2756 6172 7320 6d65 7461 6461 7461 3a27  'Vars metadata:'
+00015300: 290a 2020 2020 2020 2020 7072 696e 7428  ).        print(
+00015310: 6164 6174 612e 7661 722e 696c 6f63 5b31  adata.var.iloc[1
+00015320: 3a31 305d 290a 2020 2020 2020 2020 7072  :10]).        pr
+00015330: 696e 7428 2727 290a 2020 2020 2020 2020  int('').        
+00015340: 7072 696e 7428 2744 6174 613a 2729 0a20  print('Data:'). 
+00015350: 2020 2020 2020 2070 7269 6e74 2861 6461         print(ada
+00015360: 7461 2e58 290a 2020 2020 2020 2020 7072  ta.X).        pr
+00015370: 696e 7428 2727 290a 0a20 2020 2020 2020  int('')..       
+00015380: 2070 7269 6e74 2827 5375 6273 6574 7469   print('Subsetti
+00015390: 6e67 2074 6865 2045 7870 6572 696d 656e  ng the Experimen
+000153a0: 745c 6e44 6973 6561 7365 206c 6576 656c  t\nDisease level
+000153b0: 7327 290a 2020 2020 2020 2020 7072 696e  s').        prin
+000153c0: 7428 6164 6174 612e 7661 725b 2764 6973  t(adata.var['dis
+000153d0: 6561 7365 275d 2e75 6e69 7175 6528 2929  ease'].unique())
+000153e0: 0a20 2020 2020 2020 2070 7269 6e74 2827  .        print('
+000153f0: 5c6e 5375 6273 6574 3a20 7061 7469 656e  \nSubset: patien
+00015400: 7473 2064 7572 696e 6720 6d61 6e69 6320  ts during manic 
+00015410: 7068 6173 6520 616e 6420 636f 6e74 726f  phase and contro
+00015420: 6c73 2729 0a20 2020 2020 2020 206d 616e  ls').        man
+00015430: 6963 3d61 6461 7461 5b3a 2c28 6164 6174  ic=adata[:,(adat
+00015440: 612e 7661 725b 2764 6973 6561 7365 275d  a.var['disease']
+00015450: 203d 3d20 2772 6566 6572 656e 6365 5f73   == 'reference_s
+00015460: 7562 6a65 6374 5f72 6f6c 6527 2920 7c0a  ubject_role') |.
+00015470: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00015480: 2020 2020 2020 2861 6461 7461 2e76 6172        (adata.var
+00015490: 5b27 6469 7365 6173 6527 5d20 3d3d 2027  ['disease'] == '
+000154a0: 6269 706f 6c61 725f 6469 736f 7264 6572  bipolar_disorder
+000154b0: 5f7c 5f6d 616e 6963 5f70 6861 7365 5f7c  _|_manic_phase_|
+000154c0: 2729 5d2e 636f 7079 2829 0a20 2020 2020  ')].copy().     
+000154d0: 2020 2070 7269 6e74 286d 616e 6963 290a     print(manic).
+000154e0: 2020 2020 2020 2020 7072 696e 7428 6d61          print(ma
+000154f0: 6e69 632e 7661 7229 0a0a 2020 2020 6465  nic.var)..    de
+00015500: 6620 6765 745f 7265 7375 6c74 5f73 6574  f get_result_set
+00015510: 735f 7465 7374 2829 3a0a 2020 2020 2020  s_test():.      
+00015520: 2020 7072 696e 7428 2754 6573 7469 6e67    print('Testing
+00015530: 2067 6574 5f72 6573 756c 745f 7365 7473   get_result_sets
+00015540: 2066 756e 6374 696f 6e3a 2729 0a20 2020   function:').   
+00015550: 2020 2020 2064 6620 3d20 6170 695f 696e       df = api_in
+00015560: 7374 616e 6365 2e67 6574 5f72 6573 756c  stance.get_resul
+00015570: 745f 7365 7473 2827 4753 4536 3731 3127  t_sets('GSE6711'
+00015580: 290a 2020 2020 2020 2020 7072 696e 7428  ).        print(
+00015590: 6466 290a 2020 2020 2020 2020 6466 3220  df).        df2 
+000155a0: 3d20 6170 695f 696e 7374 616e 6365 2e67  = api_instance.g
+000155b0: 6574 5f72 6573 756c 745f 7365 7428 3438  et_result_set(48
+000155c0: 3534 3036 290a 2020 2020 2020 2020 7072  5406).        pr
+000155d0: 696e 7428 6466 3229 0a0a 2020 2020 6465  int(df2)..    de
+000155e0: 6620 6765 745f 6469 6666 6572 656e 7469  f get_differenti
+000155f0: 616c 5f65 7870 7265 7373 696f 6e5f 7661  al_expression_va
+00015600: 6c75 6573 5f74 6573 7428 293a 0a20 2020  lues_test():.   
+00015610: 2020 2020 2064 6520 3d20 6170 695f 696e       de = api_in
+00015620: 7374 616e 6365 2e67 6574 5f64 6966 6665  stance.get_diffe
+00015630: 7265 6e74 6961 6c5f 6578 7072 6573 7369  rential_expressi
+00015640: 6f6e 5f76 616c 7565 7328 2747 5345 3436  on_values('GSE46
+00015650: 3431 3627 290a 2020 2020 2020 2020 6465  416').        de
+00015660: 5b27 6469 6666 6578 7072 275d 203d 2027  ['diffexpr'] = '
+00015670: 4e6f 2720 2020 2320 6164 6420 6578 7472  No'   # add extr
+00015680: 6120 636f 6c75 6d6e 0a20 2020 2020 2020  a column.       
+00015690: 2023 6465 5b27 6469 6666 6578 7072 275d   #de['diffexpr']
+000156a0: 5b28 6465 5b27 636f 6e74 7261 7374 5f62  [(de['contrast_b
+000156b0: 6970 6f6c 6172 2064 6973 6f72 6465 722c  ipolar disorder,
+000156c0: 206d 616e 6963 2070 6861 7365 5f6c 6f67   manic phase_log
+000156d0: 466f 6c64 4368 616e 6765 275d 3e31 2e30  FoldChange']>1.0
+000156e0: 2920 260a 2020 2020 2020 2020 2320 2020  ) &.        #   
+000156f0: 2020 2020 2020 2020 2020 2020 2864 655b              (de[
+00015700: 2763 6f6e 7472 6173 745f 6269 706f 6c61  'contrast_bipola
+00015710: 7220 6469 736f 7264 6572 2c20 6d61 6e69  r disorder, mani
+00015720: 6320 7068 6173 655f 7076 616c 7565 275d  c phase_pvalue']
+00015730: 3c30 2e30 3529 5d20 3d20 2755 7027 0a20  <0.05)] = 'Up'. 
+00015740: 2020 2020 2020 2064 652e 6c6f 635b 2864         de.loc[(d
+00015750: 655b 2763 6f6e 7472 6173 745f 6269 706f  e['contrast_bipo
+00015760: 6c61 7220 6469 736f 7264 6572 2c20 6d61  lar disorder, ma
+00015770: 6e69 6320 7068 6173 655f 6c6f 6746 6f6c  nic phase_logFol
+00015780: 6443 6861 6e67 6527 5d20 3e20 312e 3029  dChange'] > 1.0)
+00015790: 2026 0a20 2020 2020 2020 2020 2020 2020   &.             
+000157a0: 2020 2020 2020 2020 2020 2864 655b 2763            (de['c
+000157b0: 6f6e 7472 6173 745f 6269 706f 6c61 7220  ontrast_bipolar 
+000157c0: 6469 736f 7264 6572 2c20 6d61 6e69 6320  disorder, manic 
+000157d0: 7068 6173 655f 7076 616c 7565 275d 203c  phase_pvalue'] <
+000157e0: 2030 2e30 3529 2c27 6469 6666 6578 7072   0.05),'diffexpr
+000157f0: 275d 203d 2027 5570 270a 2020 2020 2020  '] = 'Up'.      
+00015800: 2020 6465 2e6c 6f63 5b28 6465 5b27 636f    de.loc[(de['co
+00015810: 6e74 7261 7374 5f62 6970 6f6c 6172 2064  ntrast_bipolar d
+00015820: 6973 6f72 6465 722c 206d 616e 6963 2070  isorder, manic p
+00015830: 6861 7365 5f6c 6f67 466f 6c64 4368 616e  hase_logFoldChan
+00015840: 6765 275d 203c 202d 312e 3029 2026 0a20  ge'] < -1.0) &. 
+00015850: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00015860: 2020 2020 2020 2864 655b 2763 6f6e 7472        (de['contr
+00015870: 6173 745f 6269 706f 6c61 7220 6469 736f  ast_bipolar diso
+00015880: 7264 6572 2c20 6d61 6e69 6320 7068 6173  rder, manic phas
+00015890: 655f 7076 616c 7565 275d 203c 2030 2e30  e_pvalue'] < 0.0
+000158a0: 3529 2c27 6469 6666 6578 7072 275d 203d  5),'diffexpr'] =
+000158b0: 2027 446f 776e 270a 2020 2020 2020 2020   'Down'.        
+000158c0: 6465 5f75 7020 3d20 6465 5b64 655b 2764  de_up = de[de['d
+000158d0: 6966 6665 7870 7227 5d3d 3d27 5570 275d  iffexpr']=='Up']
+000158e0: 0a20 2020 2020 2020 2064 655f 7570 203d  .        de_up =
+000158f0: 2064 655f 7570 5b5b 2750 726f 6265 272c   de_up[['Probe',
+00015900: 2747 656e 6553 796d 626f 6c27 2c20 2763  'GeneSymbol', 'c
+00015910: 6f6e 7472 6173 745f 6269 706f 6c61 7220  ontrast_bipolar 
+00015920: 6469 736f 7264 6572 2c20 6d61 6e69 6320  disorder, manic 
+00015930: 7068 6173 655f 7076 616c 7565 272c 0a20  phase_pvalue',. 
+00015940: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00015950: 2020 2020 2020 2763 6f6e 7472 6173 745f        'contrast_
+00015960: 6269 706f 6c61 7220 6469 736f 7264 6572  bipolar disorder
+00015970: 2c20 6d61 6e69 6320 7068 6173 655f 6c6f  , manic phase_lo
+00015980: 6746 6f6c 6443 6861 6e67 6527 5d5d 2e73  gFoldChange']].s
+00015990: 6f72 745f 7661 6c75 6573 280a 2020 2020  ort_values(.    
+000159a0: 2020 2020 2020 2020 2020 2020 2763 6f6e              'con
+000159b0: 7472 6173 745f 6269 706f 6c61 7220 6469  trast_bipolar di
+000159c0: 736f 7264 6572 2c20 6d61 6e69 6320 7068  sorder, manic ph
+000159d0: 6173 655f 7076 616c 7565 2729 0a20 2020  ase_pvalue').   
+000159e0: 2020 2020 2070 7269 6e74 2827 5570 7265       print('Upre
+000159f0: 6775 6c61 7465 6420 7072 6f62 6573 2729  gulated probes')
+00015a00: 0a20 2020 2020 2020 2077 6974 6820 7064  .        with pd
+00015a10: 2e6f 7074 696f 6e5f 636f 6e74 6578 7428  .option_context(
+00015a20: 2764 6973 706c 6179 2e6d 6178 5f72 6f77  'display.max_row
+00015a30: 7327 2c20 4e6f 6e65 2c20 2764 6973 706c  s', None, 'displ
+00015a40: 6179 2e6d 6178 5f63 6f6c 756d 6e73 272c  ay.max_columns',
+00015a50: 204e 6f6e 6529 3a0a 2020 2020 2020 2020   None):.        
+00015a60: 2020 2020 7072 696e 7428 6465 5f75 705b      print(de_up[
+00015a70: 3a31 305d 290a 0a20 2020 2020 2020 2064  :10])..        d
+00015a80: 655f 646e 203d 2064 655b 6465 5b27 6469  e_dn = de[de['di
+00015a90: 6666 6578 7072 275d 3d3d 2744 6f77 6e27  ffexpr']=='Down'
+00015aa0: 5d0a 2020 2020 2020 2020 6465 5f64 6e20  ].        de_dn 
+00015ab0: 3d20 6465 5f64 6e5b 5b27 5072 6f62 6527  = de_dn[['Probe'
+00015ac0: 2c27 4765 6e65 5379 6d62 6f6c 272c 2027  ,'GeneSymbol', '
+00015ad0: 636f 6e74 7261 7374 5f62 6970 6f6c 6172  contrast_bipolar
+00015ae0: 2064 6973 6f72 6465 722c 206d 616e 6963   disorder, manic
+00015af0: 2070 6861 7365 5f70 7661 6c75 6527 2c0a   phase_pvalue',.
+00015b00: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00015b10: 2020 2020 2020 2027 636f 6e74 7261 7374         'contrast
+00015b20: 5f62 6970 6f6c 6172 2064 6973 6f72 6465  _bipolar disorde
+00015b30: 722c 206d 616e 6963 2070 6861 7365 5f6c  r, manic phase_l
+00015b40: 6f67 466f 6c64 4368 616e 6765 275d 5d2e  ogFoldChange']].
+00015b50: 736f 7274 5f76 616c 7565 7328 0a20 2020  sort_values(.   
+00015b60: 2020 2020 2020 2020 2020 2020 2027 636f               'co
+00015b70: 6e74 7261 7374 5f62 6970 6f6c 6172 2064  ntrast_bipolar d
+00015b80: 6973 6f72 6465 722c 206d 616e 6963 2070  isorder, manic p
+00015b90: 6861 7365 5f70 7661 6c75 6527 290a 2020  hase_pvalue').  
+00015ba0: 2020 2020 2020 7072 696e 7428 275c 6e44        print('\nD
+00015bb0: 6f77 6e72 6567 756c 6174 6564 2070 726f  ownregulated pro
+00015bc0: 6265 7327 290a 2020 2020 2020 2020 7769  bes').        wi
+00015bd0: 7468 2070 642e 6f70 7469 6f6e 5f63 6f6e  th pd.option_con
+00015be0: 7465 7874 2827 6469 7370 6c61 792e 6d61  text('display.ma
+00015bf0: 785f 726f 7773 272c 204e 6f6e 652c 2027  x_rows', None, '
+00015c00: 6469 7370 6c61 792e 6d61 785f 636f 6c75  display.max_colu
+00015c10: 6d6e 7327 2c20 4e6f 6e65 293a 0a20 2020  mns', None):.   
+00015c20: 2020 2020 2020 2020 2070 7269 6e74 2864           print(d
+00015c30: 655f 646e 5b3a 3130 5d29 0a0a 2020 2020  e_dn[:10])..    
+00015c40: 6465 6620 6765 745f 706c 6174 666f 726d  def get_platform
+00015c50: 5f61 6e6e 6f74 6174 696f 6e73 5f74 6573  _annotations_tes
+00015c60: 7428 293a 0a20 2020 2020 2020 2070 7269  t():.        pri
+00015c70: 6e74 2827 5465 7374 696e 6720 6765 745f  nt('Testing get_
+00015c80: 706c 6174 666f 726d 5f61 6e6e 6f74 6174  platform_annotat
+00015c90: 696f 6e73 2066 756e 6374 696f 6e27 290a  ions function').
+00015ca0: 2020 2020 2020 2020 6466 203d 2061 7069          df = api
+00015cb0: 5f69 6e73 7461 6e63 652e 6765 745f 706c  _instance.get_pl
+00015cc0: 6174 666f 726d 5f61 6e6e 6f74 6174 696f  atform_annotatio
+00015cd0: 6e73 2827 4750 4c39 3627 290a 2020 2020  ns('GPL96').    
+00015ce0: 2020 2020 7072 696e 7428 6466 290a 0a20      print(df).. 
+00015cf0: 2020 2069 6620 6c65 6e28 7379 732e 6172     if len(sys.ar
+00015d00: 6776 2920 3e20 313a 0a20 2020 2020 2020  gv) > 1:.       
+00015d10: 2066 6e61 6d65 203d 2073 7973 2e61 7267   fname = sys.arg
+00015d20: 765b 315d 0a20 2020 2020 2020 2064 6963  v[1].        dic
+00015d30: 7420 3d20 7661 7273 2829 0a20 2020 2020  t = vars().     
+00015d40: 2020 2069 6620 666e 616d 6520 696e 2064     if fname in d
+00015d50: 6963 743a 0a20 2020 2020 2020 2020 2020  ict:.           
+00015d60: 2023 7072 696e 7428 2243 616c 6c69 6e67   #print("Calling
+00015d70: 222c 666e 616d 6529 0a20 2020 2020 2020  ",fname).       
+00015d80: 2020 2020 2066 756e 6320 3d20 6469 6374       func = dict
+00015d90: 5b66 6e61 6d65 5d0a 2020 2020 2020 2020  [fname].        
+00015da0: 2020 2020 6675 6e63 2829 0a                  func().
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/__init__.py` & `gemmapy-1.0.1/gemmapy/sdk/models/__init__.py`

 * *Files 4% similar despite different names*

```diff
@@ -1,29 +1,23 @@
 # coding: utf-8
 
 # flake8: noqa
-
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 from __future__ import absolute_import
 
-# import apis into sdk package
-from gemmapy.sdk.api.default_api import DefaultApi
-# import ApiClient
-from gemmapy.sdk.api_client import ApiClient
-from gemmapy.sdk.configuration import Configuration
-# import models into sdk package
+# import models into model package
 from gemmapy.sdk.models.annotation_search_result_value_object import AnnotationSearchResultValueObject
 from gemmapy.sdk.models.annotation_value_object import AnnotationValueObject
 from gemmapy.sdk.models.annotation_with_usage_statistics_value_object import AnnotationWithUsageStatisticsValueObject
 from gemmapy.sdk.models.api_info_value_object import ApiInfoValueObject
 from gemmapy.sdk.models.array_design_value_object import ArrayDesignValueObject
 from gemmapy.sdk.models.array_design_with_usage_statistics_value_object import ArrayDesignWithUsageStatisticsValueObject
 from gemmapy.sdk.models.audit_event_value_object import AuditEventValueObject
@@ -63,56 +57,59 @@
 from gemmapy.sdk.models.expression_experiment_with_search_result_value_object import ExpressionExperimentWithSearchResultValueObject
 from gemmapy.sdk.models.external_database_value_object import ExternalDatabaseValueObject
 from gemmapy.sdk.models.factor_value_basic_value_object import FactorValueBasicValueObject
 from gemmapy.sdk.models.factor_value_value_object import FactorValueValueObject
 from gemmapy.sdk.models.filter_arg_array_design import FilterArgArrayDesign
 from gemmapy.sdk.models.filter_arg_expression_analysis_result_set import FilterArgExpressionAnalysisResultSet
 from gemmapy.sdk.models.filter_arg_expression_experiment import FilterArgExpressionExperiment
+from gemmapy.sdk.models.filtered_and_inferred_and_paginated_response_data_object_expression_experiment_value_object import FilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_array_design_value_object import FilteredAndPaginatedResponseDataObjectArrayDesignValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_composite_sequence_value_object import FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_differential_expression_analysis_result_set_value_object import FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_expression_experiment_value_object import FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_gene_value_object import FilteredAndPaginatedResponseDataObjectGeneValueObject
 from gemmapy.sdk.models.geeq_value_object import GeeqValueObject
 from gemmapy.sdk.models.gene import Gene
 from gemmapy.sdk.models.gene1 import Gene1
 from gemmapy.sdk.models.gene2 import Gene2
 from gemmapy.sdk.models.gene3 import Gene3
 from gemmapy.sdk.models.gene4 import Gene4
+from gemmapy.sdk.models.gene5 import Gene5
 from gemmapy.sdk.models.gene_arg import GeneArg
 from gemmapy.sdk.models.gene_element_expressions_value_object import GeneElementExpressionsValueObject
 from gemmapy.sdk.models.gene_ontology_term_value_object import GeneOntologyTermValueObject
 from gemmapy.sdk.models.gene_set_value_object import GeneSetValueObject
 from gemmapy.sdk.models.gene_value_object import GeneValueObject
-from gemmapy.sdk.models.limited_response_data_object_annotation_with_usage_statistics_value_object import LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
-from gemmapy.sdk.models.limited_response_data_object_array_design_with_usage_statistics_value_object import LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
 from gemmapy.sdk.models.measurement_value_object import MeasurementValueObject
 from gemmapy.sdk.models.one_of_search_result_value_object_object_result_object import OneOfSearchResultValueObjectObjectResultObject
 from gemmapy.sdk.models.ontology_term_value_object import OntologyTermValueObject
 from gemmapy.sdk.models.paginated_response_data_object_composite_sequence_value_object import PaginatedResponseDataObjectCompositeSequenceValueObject
 from gemmapy.sdk.models.paginated_response_data_object_expression_experiment_value_object import PaginatedResponseDataObjectExpressionExperimentValueObject
+from gemmapy.sdk.models.paginated_response_data_object_gene_value_object import PaginatedResponseDataObjectGeneValueObject
+from gemmapy.sdk.models.paginated_results_response_data_object_differential_expression_analysis_result_set_value_object import PaginatedResultsResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
 from gemmapy.sdk.models.physical_location_value_object import PhysicalLocationValueObject
 from gemmapy.sdk.models.platform import Platform
 from gemmapy.sdk.models.platform1 import Platform1
 from gemmapy.sdk.models.platform2 import Platform2
 from gemmapy.sdk.models.platform3 import Platform3
 from gemmapy.sdk.models.platform4 import Platform4
 from gemmapy.sdk.models.platform5 import Platform5
 from gemmapy.sdk.models.platform_arg import PlatformArg
 from gemmapy.sdk.models.probe import Probe
 from gemmapy.sdk.models.quantitation_type import QuantitationType
 from gemmapy.sdk.models.quantitation_type_arg import QuantitationTypeArg
 from gemmapy.sdk.models.quantitation_type_value_object import QuantitationTypeValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_limited_response_data_object_annotation_with_usage_statistics_value_object import QueriedAndFilteredAndInferredAndLimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_limited_response_data_object_array_design_with_usage_statistics_value_object import QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_limited_response_data_object_category_with_usage_statistics_value_object import QueriedAndFilteredAndInferredAndLimitedResponseDataObjectCategoryWithUsageStatisticsValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_paginated_response_data_object_expression_experiment_with_search_result_value_object import QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_response_data_object_taxon_with_usage_statistics_value_object import QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject
 from gemmapy.sdk.models.queried_and_filtered_and_paginated_response_data_object_expression_experiment_value_object import QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
-from gemmapy.sdk.models.queried_and_filtered_and_paginated_response_data_object_expression_experiment_with_search_result_value_object import QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
-from gemmapy.sdk.models.queried_and_filtered_response_data_object_category_with_usage_statistics_value_object import QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject
-from gemmapy.sdk.models.queried_and_filtered_response_data_object_taxon_with_usage_statistics_value_object import QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject
 from gemmapy.sdk.models.query_arg import QueryArg
 from gemmapy.sdk.models.response_data_object_api_info_value_object import ResponseDataObjectApiInfoValueObject
-from gemmapy.sdk.models.response_data_object_differential_expression_analysis_result_set_value_object import ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
 from gemmapy.sdk.models.response_data_object_list_annotation_search_result_value_object import ResponseDataObjectListAnnotationSearchResultValueObject
 from gemmapy.sdk.models.response_data_object_list_array_design_value_object import ResponseDataObjectListArrayDesignValueObject
 from gemmapy.sdk.models.response_data_object_list_bio_assay_value_object import ResponseDataObjectListBioAssayValueObject
 from gemmapy.sdk.models.response_data_object_list_differential_expression_analysis_value_object import ResponseDataObjectListDifferentialExpressionAnalysisValueObject
 from gemmapy.sdk.models.response_data_object_list_experiment_expression_levels_value_object import ResponseDataObjectListExperimentExpressionLevelsValueObject
 from gemmapy.sdk.models.response_data_object_list_gene_ontology_term_value_object import ResponseDataObjectListGeneOntologyTermValueObject
 from gemmapy.sdk.models.response_data_object_list_gene_value_object import ResponseDataObjectListGeneValueObject
@@ -128,22 +125,25 @@
 from gemmapy.sdk.models.search_result_value_object_object import SearchResultValueObjectObject
 from gemmapy.sdk.models.search_results_response_data_object import SearchResultsResponseDataObject
 from gemmapy.sdk.models.search_settings_value_object import SearchSettingsValueObject
 from gemmapy.sdk.models.simple_svd_value_object import SimpleSVDValueObject
 from gemmapy.sdk.models.sort_arg_array_design import SortArgArrayDesign
 from gemmapy.sdk.models.sort_arg_expression_analysis_result_set import SortArgExpressionAnalysisResultSet
 from gemmapy.sdk.models.sort_arg_expression_experiment import SortArgExpressionExperiment
-from gemmapy.sdk.models.sort_arg_taxon import SortArgTaxon
 from gemmapy.sdk.models.sort_value_object import SortValueObject
 from gemmapy.sdk.models.statement_value_object import StatementValueObject
+from gemmapy.sdk.models.taxa import Taxa
 from gemmapy.sdk.models.taxon import Taxon
 from gemmapy.sdk.models.taxon1 import Taxon1
 from gemmapy.sdk.models.taxon2 import Taxon2
 from gemmapy.sdk.models.taxon3 import Taxon3
 from gemmapy.sdk.models.taxon4 import Taxon4
 from gemmapy.sdk.models.taxon5 import Taxon5
 from gemmapy.sdk.models.taxon6 import Taxon6
+from gemmapy.sdk.models.taxon7 import Taxon7
+from gemmapy.sdk.models.taxon8 import Taxon8
+from gemmapy.sdk.models.taxon9 import Taxon9
 from gemmapy.sdk.models.taxon_arg import TaxonArg
 from gemmapy.sdk.models.taxon_value_object import TaxonValueObject
 from gemmapy.sdk.models.taxon_with_usage_statistics_value_object import TaxonWithUsageStatisticsValueObject
 from gemmapy.sdk.models.vector_element_value_object import VectorElementValueObject
 from gemmapy.sdk.models.well_composed_error_body import WellComposedErrorBody
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/api/default_api.py` & `gemmapy-1.0.1/gemmapy/sdk/api/default_api.py`

 * *Files 4% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 from __future__ import absolute_import
 
 import re  # noqa: F401
@@ -69,14 +69,16 @@
         """
 
         all_params = []  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_api_info" % key
@@ -93,29 +95,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectApiInfoValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectApiInfoValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -156,14 +164,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_annotations" % key
@@ -186,29 +196,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectSetAnnotationValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/annotations', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectSetAnnotationValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -249,14 +265,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_design" % key
@@ -279,29 +297,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+            response_type = 'str' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/design', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='str',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -352,14 +376,16 @@
         """
 
         all_params = ['datasets', 'diff_ex_set', 'threshold', 'limit', 'keep_non_specific', 'consolidate']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_differential_expression" % key
@@ -393,29 +419,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListExperimentExpressionLevelsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{datasets}/expressions/differential', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListExperimentExpressionLevelsValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -460,14 +492,16 @@
         """
 
         all_params = ['dataset', 'offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_differential_expression_analyses" % key
@@ -494,29 +528,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListDifferentialExpressionAnalysisValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/analyses/differential', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListDifferentialExpressionAnalysisValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -557,14 +597,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_differential_expression_analyses_result_sets" % key
@@ -587,29 +629,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = None # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/analyses/differential/resultSets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type=None,  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -654,14 +702,16 @@
         """
 
         all_params = ['dataset', 'filter']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_expression" % key
@@ -686,29 +736,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+            response_type = None # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/data', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type=None,  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -755,14 +811,16 @@
         """
 
         all_params = ['datasets', 'genes', 'keep_non_specific', 'consolidate']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_expression_for_genes" % key
@@ -797,29 +855,162 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListExperimentExpressionLevelsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{datasets}/expressions/genes/{genes}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListExperimentExpressionLevelsValueObject',  # noqa: E501
+            response_type=response_type,
+            auth_settings=auth_settings,
+            async_req=params.get('async_req'),
+            _return_http_data_only=params.get('_return_http_data_only'),
+            _preload_content=params.get('_preload_content', True),
+            _request_timeout=params.get('_request_timeout'),
+            collection_formats=collection_formats)
+
+    def get_dataset_expression_for_genes_in_taxa(self, datasets, taxa, genes, **kwargs):  # noqa: E501
+        """Retrieve the expression data matrix of a set of datasets and genes  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_dataset_expression_for_genes_in_taxa(datasets, taxa, genes, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param list[object] datasets: (required)
+        :param Taxa taxa: (required)
+        :param list[object] genes: (required)
+        :param bool keep_non_specific:
+        :param str consolidate:
+        :return: ResponseDataObjectListExperimentExpressionLevelsValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+        kwargs['_return_http_data_only'] = True
+        if kwargs.get('async_req'):
+            return self.get_dataset_expression_for_genes_in_taxa_with_http_info(datasets, taxa, genes, **kwargs)  # noqa: E501
+        else:
+            (data) = self.get_dataset_expression_for_genes_in_taxa_with_http_info(datasets, taxa, genes, **kwargs)  # noqa: E501
+            return data
+
+    def get_dataset_expression_for_genes_in_taxa_with_http_info(self, datasets, taxa, genes, **kwargs):  # noqa: E501
+        """Retrieve the expression data matrix of a set of datasets and genes  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_dataset_expression_for_genes_in_taxa_with_http_info(datasets, taxa, genes, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param list[object] datasets: (required)
+        :param Taxa taxa: (required)
+        :param list[object] genes: (required)
+        :param bool keep_non_specific:
+        :param str consolidate:
+        :return: ResponseDataObjectListExperimentExpressionLevelsValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+
+        all_params = ['datasets', 'taxa', 'genes', 'keep_non_specific', 'consolidate']  # noqa: E501
+        all_params.append('async_req')
+        all_params.append('_return_http_data_only')
+        all_params.append('_preload_content')
+        all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
+
+        params = locals()
+        for key, val in six.iteritems(params['kwargs']):
+            if key not in all_params:
+                raise TypeError(
+                    "Got an unexpected keyword argument '%s'"
+                    " to method get_dataset_expression_for_genes_in_taxa" % key
+                )
+            params[key] = val
+        del params['kwargs']
+        # verify the required parameter 'datasets' is set
+        if ('datasets' not in params or
+                params['datasets'] is None):
+            raise ValueError("Missing the required parameter `datasets` when calling `get_dataset_expression_for_genes_in_taxa`")  # noqa: E501
+        # verify the required parameter 'taxa' is set
+        if ('taxa' not in params or
+                params['taxa'] is None):
+            raise ValueError("Missing the required parameter `taxa` when calling `get_dataset_expression_for_genes_in_taxa`")  # noqa: E501
+        # verify the required parameter 'genes' is set
+        if ('genes' not in params or
+                params['genes'] is None):
+            raise ValueError("Missing the required parameter `genes` when calling `get_dataset_expression_for_genes_in_taxa`")  # noqa: E501
+
+        collection_formats = {}
+
+        path_params = {}
+        if 'datasets' in params:
+            path_params['datasets'] = params['datasets']  # noqa: E501
+            collection_formats['datasets'] = ''  # noqa: E501
+        if 'taxa' in params:
+            path_params['taxa'] = params['taxa']  # noqa: E501
+        if 'genes' in params:
+            path_params['genes'] = params['genes']  # noqa: E501
+            collection_formats['genes'] = ''  # noqa: E501
+
+        query_params = []
+        if 'keep_non_specific' in params:
+            query_params.append(('keepNonSpecific', params['keep_non_specific']))  # noqa: E501
+        if 'consolidate' in params:
+            query_params.append(('consolidate', params['consolidate']))  # noqa: E501
+
+        header_params = {}
+
+        form_params = []
+        local_var_files = {}
+
+        body_params = None
+        # HTTP header `Accept`
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListExperimentExpressionLevelsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
+
+        # Authentication setting
+        auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
+
+        return self.api_client.call_api(
+            '/datasets/{datasets}/expressions/taxa/{taxa}/genes/{genes}', 'GET',
+            path_params,
+            query_params,
+            header_params,
+            body=body_params,
+            post_params=form_params,
+            files=local_var_files,
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -868,14 +1059,16 @@
         """
 
         all_params = ['datasets', 'component', 'limit', 'keep_non_specific', 'consolidate']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_expression_pca" % key
@@ -907,29 +1100,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListExperimentExpressionLevelsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{datasets}/expressions/pca', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListExperimentExpressionLevelsValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -970,14 +1169,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_platforms" % key
@@ -1000,29 +1201,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListArrayDesignValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/platforms', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListArrayDesignValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1063,14 +1270,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_processed_expression" % key
@@ -1093,29 +1302,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+            response_type = 'str' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/data/processed', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='str',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1156,14 +1371,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_quantitation_types" % key
@@ -1186,29 +1403,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectSetQuantitationTypeValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/quantitationTypes', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectSetQuantitationTypeValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1251,14 +1474,16 @@
         """
 
         all_params = ['dataset', 'quantitation_type']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_raw_expression" % key
@@ -1283,29 +1508,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
+            response_type = 'str' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/data/raw', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='str',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1346,14 +1577,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_samples" % key
@@ -1376,29 +1609,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListBioAssayValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/samples', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListBioAssayValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1439,14 +1678,16 @@
         """
 
         all_params = ['dataset']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_dataset_svd" % key
@@ -1469,29 +1710,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectSimpleSVDValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}/svd', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectSimpleSVDValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1505,15 +1752,15 @@
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param int offset:
         :param int limit:
         :param SortArgExpressionExperiment sort:
-        :return: QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
+        :return: QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_datasets_with_http_info(**kwargs)  # noqa: E501
         else:
@@ -1530,24 +1777,26 @@
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param int offset:
         :param int limit:
         :param SortArgExpressionExperiment sort:
-        :return: QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
+        :return: QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['query', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_datasets" % key
@@ -1574,29 +1823,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1612,15 +1867,15 @@
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param list[str] exclude: List of fields to exclude from the payload. Only `parentTerms` can be excluded.
         :param int limit: Maximum number of annotations to returned; capped at 5000.
         :param int min_frequency: Minimum number of associated datasets to report an annotation. If used, the limit will default to 5000.
         :param str category: A category URI to restrict reported annotations. If unspecified, annotations from all categories are reported. If empty, uncategorized terms are reported.
-        :return: LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredAndLimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_datasets_annotations_usage_statistics_with_http_info(**kwargs)  # noqa: E501
         else:
@@ -1639,24 +1894,26 @@
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param list[str] exclude: List of fields to exclude from the payload. Only `parentTerms` can be excluded.
         :param int limit: Maximum number of annotations to returned; capped at 5000.
         :param int min_frequency: Minimum number of associated datasets to report an annotation. If used, the limit will default to 5000.
         :param str category: A category URI to restrict reported annotations. If unspecified, annotations from all categories are reported. If empty, uncategorized terms are reported.
-        :return: LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredAndLimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['query', 'filter', 'exclude', 'limit', 'min_frequency', 'category']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_datasets_annotations_usage_statistics" % key
@@ -1686,29 +1943,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndInferredAndLimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/annotations', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1722,15 +1985,15 @@
 
         :param async_req bool
         :param list[object] dataset: (required)
         :param FilterArgExpressionExperiment filter:
         :param int offset:
         :param int limit:
         :param SortArgExpressionExperiment sort:
-        :return: FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
+        :return: FilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_datasets_by_ids_with_http_info(dataset, **kwargs)  # noqa: E501
         else:
@@ -1747,24 +2010,26 @@
 
         :param async_req bool
         :param list[object] dataset: (required)
         :param FilterArgExpressionExperiment filter:
         :param int offset:
         :param int limit:
         :param SortArgExpressionExperiment sort:
-        :return: FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
+        :return: FilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['dataset', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_datasets_by_ids" % key
@@ -1796,29 +2061,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'FilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/{dataset}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1831,15 +2102,15 @@
         >>> thread = api.get_datasets_categories_usage_statistics(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param int limit:
-        :return: QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredAndLimitedResponseDataObjectCategoryWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_datasets_categories_usage_statistics_with_http_info(**kwargs)  # noqa: E501
         else:
@@ -1855,24 +2126,26 @@
         >>> thread = api.get_datasets_categories_usage_statistics_with_http_info(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param int limit:
-        :return: QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredAndLimitedResponseDataObjectCategoryWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['query', 'filter', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_datasets_categories_usage_statistics" % key
@@ -1895,29 +2168,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndInferredAndLimitedResponseDataObjectCategoryWithUsageStatisticsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/categories', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -1930,15 +2209,15 @@
         >>> thread = api.get_datasets_platforms_usage_statistics(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param int limit:
-        :return: LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_datasets_platforms_usage_statistics_with_http_info(**kwargs)  # noqa: E501
         else:
@@ -1954,24 +2233,26 @@
         >>> thread = api.get_datasets_platforms_usage_statistics_with_http_info(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
         :param int limit:
-        :return: LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['query', 'filter', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_datasets_platforms_usage_statistics" % key
@@ -1994,29 +2275,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/platforms', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -2027,15 +2314,15 @@
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_datasets_taxa_usage_statistics(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
-        :return: QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_datasets_taxa_usage_statistics_with_http_info(**kwargs)  # noqa: E501
         else:
@@ -2049,24 +2336,26 @@
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_datasets_taxa_usage_statistics_with_http_info(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param QueryArg query:
         :param FilterArgExpressionExperiment filter:
-        :return: QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject
+        :return: QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['query', 'filter']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_datasets_taxa_usage_statistics" % key
@@ -2087,29 +2376,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/taxa', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -2150,14 +2445,16 @@
         """
 
         all_params = ['gene']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_gene_go_terms" % key
@@ -2180,29 +2477,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListGeneOntologyTermValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/genes/{gene}/goTerms', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListGeneOntologyTermValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -2243,14 +2546,16 @@
         """
 
         all_params = ['gene']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_gene_locations" % key
@@ -2273,325 +2578,347 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListPhysicalLocationValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/genes/{gene}/locations', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListPhysicalLocationValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
-    def get_gene_locations_in_taxon(self, taxon, gene, **kwargs):  # noqa: E501
-        """Retrieve physical locations for a given gene and taxon  # noqa: E501
+    def get_gene_probes(self, gene, **kwargs):  # noqa: E501
+        """Retrieve the probes associated to a genes across all platforms  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_gene_locations_in_taxon(taxon, gene, async_req=True)
+        >>> thread = api.get_gene_probes(gene, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Taxon3 taxon: (required)
-        :param Gene3 gene: (required)
-        :return: ResponseDataObjectListPhysicalLocationValueObject
+        :param Gene2 gene: (required)
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectCompositeSequenceValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
-            return self.get_gene_locations_in_taxon_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+            return self.get_gene_probes_with_http_info(gene, **kwargs)  # noqa: E501
         else:
-            (data) = self.get_gene_locations_in_taxon_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+            (data) = self.get_gene_probes_with_http_info(gene, **kwargs)  # noqa: E501
             return data
 
-    def get_gene_locations_in_taxon_with_http_info(self, taxon, gene, **kwargs):  # noqa: E501
-        """Retrieve physical locations for a given gene and taxon  # noqa: E501
+    def get_gene_probes_with_http_info(self, gene, **kwargs):  # noqa: E501
+        """Retrieve the probes associated to a genes across all platforms  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_gene_locations_in_taxon_with_http_info(taxon, gene, async_req=True)
+        >>> thread = api.get_gene_probes_with_http_info(gene, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Taxon3 taxon: (required)
-        :param Gene3 gene: (required)
-        :return: ResponseDataObjectListPhysicalLocationValueObject
+        :param Gene2 gene: (required)
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectCompositeSequenceValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
-        all_params = ['taxon', 'gene']  # noqa: E501
+        all_params = ['gene', 'offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
-                    " to method get_gene_locations_in_taxon" % key
+                    " to method get_gene_probes" % key
                 )
             params[key] = val
         del params['kwargs']
-        # verify the required parameter 'taxon' is set
-        if ('taxon' not in params or
-                params['taxon'] is None):
-            raise ValueError("Missing the required parameter `taxon` when calling `get_gene_locations_in_taxon`")  # noqa: E501
         # verify the required parameter 'gene' is set
         if ('gene' not in params or
                 params['gene'] is None):
-            raise ValueError("Missing the required parameter `gene` when calling `get_gene_locations_in_taxon`")  # noqa: E501
+            raise ValueError("Missing the required parameter `gene` when calling `get_gene_probes`")  # noqa: E501
 
         collection_formats = {}
 
         path_params = {}
-        if 'taxon' in params:
-            path_params['taxon'] = params['taxon']  # noqa: E501
         if 'gene' in params:
             path_params['gene'] = params['gene']  # noqa: E501
 
         query_params = []
+        if 'offset' in params:
+            query_params.append(('offset', params['offset']))  # noqa: E501
+        if 'limit' in params:
+            query_params.append(('limit', params['limit']))  # noqa: E501
 
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'PaginatedResponseDataObjectCompositeSequenceValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
-            '/taxa/{taxon}/genes/{gene}/locations', 'GET',
+            '/genes/{gene}/probes', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListPhysicalLocationValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
-    def get_gene_probes(self, gene, **kwargs):  # noqa: E501
-        """Retrieve the probes associated to a genes across all platforms  # noqa: E501
+    def get_genes(self, genes, **kwargs):  # noqa: E501
+        """Retrieve genes matching gene identifiers  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_gene_probes(gene, async_req=True)
+        >>> thread = api.get_genes(genes, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Gene2 gene: (required)
-        :param int offset:
-        :param int limit:
-        :return: FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject
+        :param list[object] genes: (required)
+        :return: ResponseDataObjectListGeneValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
-            return self.get_gene_probes_with_http_info(gene, **kwargs)  # noqa: E501
+            return self.get_genes_with_http_info(genes, **kwargs)  # noqa: E501
         else:
-            (data) = self.get_gene_probes_with_http_info(gene, **kwargs)  # noqa: E501
+            (data) = self.get_genes_with_http_info(genes, **kwargs)  # noqa: E501
             return data
 
-    def get_gene_probes_with_http_info(self, gene, **kwargs):  # noqa: E501
-        """Retrieve the probes associated to a genes across all platforms  # noqa: E501
+    def get_genes_with_http_info(self, genes, **kwargs):  # noqa: E501
+        """Retrieve genes matching gene identifiers  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_gene_probes_with_http_info(gene, async_req=True)
+        >>> thread = api.get_genes_with_http_info(genes, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Gene2 gene: (required)
-        :param int offset:
-        :param int limit:
-        :return: FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject
+        :param list[object] genes: (required)
+        :return: ResponseDataObjectListGeneValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
-        all_params = ['gene', 'offset', 'limit']  # noqa: E501
+        all_params = ['genes']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
-                    " to method get_gene_probes" % key
+                    " to method get_genes" % key
                 )
             params[key] = val
         del params['kwargs']
-        # verify the required parameter 'gene' is set
-        if ('gene' not in params or
-                params['gene'] is None):
-            raise ValueError("Missing the required parameter `gene` when calling `get_gene_probes`")  # noqa: E501
+        # verify the required parameter 'genes' is set
+        if ('genes' not in params or
+                params['genes'] is None):
+            raise ValueError("Missing the required parameter `genes` when calling `get_genes`")  # noqa: E501
 
         collection_formats = {}
 
         path_params = {}
-        if 'gene' in params:
-            path_params['gene'] = params['gene']  # noqa: E501
+        if 'genes' in params:
+            path_params['genes'] = params['genes']  # noqa: E501
+            collection_formats['genes'] = ''  # noqa: E501
 
         query_params = []
-        if 'offset' in params:
-            query_params.append(('offset', params['offset']))  # noqa: E501
-        if 'limit' in params:
-            query_params.append(('limit', params['limit']))  # noqa: E501
 
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListGeneValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
-            '/genes/{gene}/probes', 'GET',
+            '/genes/{genes}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
-    def get_genes(self, genes, **kwargs):  # noqa: E501
-        """Retrieve genes matching gene identifiers  # noqa: E501
+    def get_genes1(self, **kwargs):  # noqa: E501
+        """Retrieve all genes  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_genes(genes, async_req=True)
+        >>> thread = api.get_genes1(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param list[object] genes: (required)
-        :return: ResponseDataObjectListGeneValueObject
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectGeneValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
-            return self.get_genes_with_http_info(genes, **kwargs)  # noqa: E501
+            return self.get_genes1_with_http_info(**kwargs)  # noqa: E501
         else:
-            (data) = self.get_genes_with_http_info(genes, **kwargs)  # noqa: E501
+            (data) = self.get_genes1_with_http_info(**kwargs)  # noqa: E501
             return data
 
-    def get_genes_with_http_info(self, genes, **kwargs):  # noqa: E501
-        """Retrieve genes matching gene identifiers  # noqa: E501
+    def get_genes1_with_http_info(self, **kwargs):  # noqa: E501
+        """Retrieve all genes  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_genes_with_http_info(genes, async_req=True)
+        >>> thread = api.get_genes1_with_http_info(async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param list[object] genes: (required)
-        :return: ResponseDataObjectListGeneValueObject
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectGeneValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
-        all_params = ['genes']  # noqa: E501
+        all_params = ['offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
-                    " to method get_genes" % key
+                    " to method get_genes1" % key
                 )
             params[key] = val
         del params['kwargs']
-        # verify the required parameter 'genes' is set
-        if ('genes' not in params or
-                params['genes'] is None):
-            raise ValueError("Missing the required parameter `genes` when calling `get_genes`")  # noqa: E501
 
         collection_formats = {}
 
         path_params = {}
-        if 'genes' in params:
-            path_params['genes'] = params['genes']  # noqa: E501
-            collection_formats['genes'] = ''  # noqa: E501
 
         query_params = []
+        if 'offset' in params:
+            query_params.append(('offset', params['offset']))  # noqa: E501
+        if 'limit' in params:
+            query_params.append(('limit', params['limit']))  # noqa: E501
 
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'PaginatedResponseDataObjectGeneValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
-            '/genes/{genes}', 'GET',
+            '/genes', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListGeneValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -2634,14 +2961,16 @@
         """
 
         all_params = ['query', 'filter']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_number_of_datasets" % key
@@ -2662,29 +2991,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectLong' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/datasets/count', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectLong',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -2725,14 +3060,16 @@
         """
 
         all_params = ['filter']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_number_of_platforms" % key
@@ -2751,29 +3088,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectLong' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms/count', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectLong',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -2814,14 +3157,16 @@
         """
 
         all_params = ['filter']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_number_of_result_sets" % key
@@ -2840,29 +3185,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectLong' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/resultSets/count', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectLong',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -2903,14 +3254,16 @@
         """
 
         all_params = ['platform']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_platform_annotations" % key
@@ -2933,29 +3286,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['text/tab-separated-values; charset=UTF-8'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['text/tab-separated-values; charset=UTF-8'])  # noqa: E501
+            response_type = 'str' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms/{platform}/annotations', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='str',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3000,14 +3359,16 @@
         """
 
         all_params = ['platform', 'offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_platform_datasets" % key
@@ -3034,29 +3395,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'PaginatedResponseDataObjectExpressionExperimentValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms/{platform}/datasets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='PaginatedResponseDataObjectExpressionExperimentValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3103,14 +3470,16 @@
         """
 
         all_params = ['platform', 'probes', 'offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_platform_element" % key
@@ -3144,29 +3513,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms/{platform}/elements/{probes}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3213,14 +3588,16 @@
         """
 
         all_params = ['platform', 'probe', 'offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_platform_element_genes" % key
@@ -3253,29 +3630,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'FilteredAndPaginatedResponseDataObjectGeneValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms/{platform}/elements/{probe}/genes', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectGeneValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3320,14 +3703,16 @@
         """
 
         all_params = ['platform', 'offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_platform_elements" % key
@@ -3354,29 +3739,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'PaginatedResponseDataObjectCompositeSequenceValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms/{platform}/elements', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='PaginatedResponseDataObjectCompositeSequenceValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3423,14 +3814,16 @@
         """
 
         all_params = ['filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_platforms" % key
@@ -3455,29 +3848,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'FilteredAndPaginatedResponseDataObjectArrayDesignValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectArrayDesignValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3526,14 +3925,16 @@
         """
 
         all_params = ['platform', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_platforms_by_ids" % key
@@ -3565,77 +3966,93 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'FilteredAndPaginatedResponseDataObjectArrayDesignValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/platforms/{platform}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectArrayDesignValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
     def get_result_set(self, result_set, **kwargs):  # noqa: E501
         """Retrieve a single analysis result set by its identifier  # noqa: E501
 
+        A slice or results can be retrieved by specifying the `offset` and `limit` parameters. This is only applicable to the JSON representation.  # noqa: E501
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_result_set(result_set, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param int result_set: (required)
-        :return: ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
+        :param float threshold:
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResultsResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_result_set_with_http_info(result_set, **kwargs)  # noqa: E501
         else:
             (data) = self.get_result_set_with_http_info(result_set, **kwargs)  # noqa: E501
             return data
 
     def get_result_set_with_http_info(self, result_set, **kwargs):  # noqa: E501
         """Retrieve a single analysis result set by its identifier  # noqa: E501
 
+        A slice or results can be retrieved by specifying the `offset` and `limit` parameters. This is only applicable to the JSON representation.  # noqa: E501
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_result_set_with_http_info(result_set, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param int result_set: (required)
-        :return: ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
+        :param float threshold:
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResultsResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
-        all_params = ['result_set']  # noqa: E501
+        all_params = ['result_set', 'threshold', 'offset', 'limit']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_result_set" % key
@@ -3650,130 +4067,49 @@
         collection_formats = {}
 
         path_params = {}
         if 'result_set' in params:
             path_params['resultSet'] = params['result_set']  # noqa: E501
 
         query_params = []
+        if 'threshold' in params:
+            query_params.append(('threshold', params['threshold']))  # noqa: E501
+        if 'offset' in params:
+            query_params.append(('offset', params['offset']))  # noqa: E501
+        if 'limit' in params:
+            query_params.append(('limit', params['limit']))  # noqa: E501
 
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json', 'text/tab-separated-values; charset=UTF-8'])  # noqa: E501
+            response_type = 'PaginatedResultsResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/resultSets/{resultSet}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject',  # noqa: E501
-            auth_settings=auth_settings,
-            async_req=params.get('async_req'),
-            _return_http_data_only=params.get('_return_http_data_only'),
-            _preload_content=params.get('_preload_content', True),
-            _request_timeout=params.get('_request_timeout'),
-            collection_formats=collection_formats)
-
-    def get_result_set_as_tsv(self, result_set_, **kwargs):  # noqa: E501
-        """Retrieve a single analysis result set by its identifier as a tab-separated values  # noqa: E501
-
-        This method makes a synchronous HTTP request by default. To make an
-        asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_result_set_as_tsv(result_set_, async_req=True)
-        >>> result = thread.get()
-
-        :param async_req bool
-        :param int result_set_: (required)
-        :return: str
-                 If the method is called asynchronously,
-                 returns the request thread.
-        """
-        kwargs['_return_http_data_only'] = True
-        if kwargs.get('async_req'):
-            return self.get_result_set_as_tsv_with_http_info(result_set_, **kwargs)  # noqa: E501
-        else:
-            (data) = self.get_result_set_as_tsv_with_http_info(result_set_, **kwargs)  # noqa: E501
-            return data
-
-    def get_result_set_as_tsv_with_http_info(self, result_set_, **kwargs):  # noqa: E501
-        """Retrieve a single analysis result set by its identifier as a tab-separated values  # noqa: E501
-
-        This method makes a synchronous HTTP request by default. To make an
-        asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_result_set_as_tsv_with_http_info(result_set_, async_req=True)
-        >>> result = thread.get()
-
-        :param async_req bool
-        :param int result_set_: (required)
-        :return: str
-                 If the method is called asynchronously,
-                 returns the request thread.
-        """
-
-        all_params = ['result_set_']  # noqa: E501
-        all_params.append('async_req')
-        all_params.append('_return_http_data_only')
-        all_params.append('_preload_content')
-        all_params.append('_request_timeout')
-
-        params = locals()
-        for key, val in six.iteritems(params['kwargs']):
-            if key not in all_params:
-                raise TypeError(
-                    "Got an unexpected keyword argument '%s'"
-                    " to method get_result_set_as_tsv" % key
-                )
-            params[key] = val
-        del params['kwargs']
-        # verify the required parameter 'result_set_' is set
-        if ('result_set_' not in params or
-                params['result_set_'] is None):
-            raise ValueError("Missing the required parameter `result_set_` when calling `get_result_set_as_tsv`")  # noqa: E501
-
-        collection_formats = {}
-
-        path_params = {}
-        if 'result_set_' in params:
-            path_params['resultSet_'] = params['result_set_']  # noqa: E501
-
-        query_params = []
-
-        header_params = {}
-
-        form_params = []
-        local_var_files = {}
-
-        body_params = None
-        # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['text/tab-separated-values; charset=UTF-8', 'application/json'])  # noqa: E501
-
-        # Authentication setting
-        auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
-
-        return self.api_client.call_api(
-            '/resultSets/{resultSet_}', 'GET',
-            path_params,
-            query_params,
-            header_params,
-            body=body_params,
-            post_params=form_params,
-            files=local_var_files,
-            response_type='str',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3824,14 +4160,16 @@
         """
 
         all_params = ['datasets', 'database_entries', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_result_sets" % key
@@ -3862,29 +4200,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/resultSets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -3923,14 +4267,16 @@
         """
 
         all_params = []  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_taxa" % key
@@ -3947,29 +4293,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListTaxonValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/taxa', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListTaxonValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4010,14 +4362,16 @@
         """
 
         all_params = ['taxa']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_taxa_by_ids" % key
@@ -4041,29 +4395,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListTaxonValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/taxa/{taxa}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListTaxonValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4072,19 +4432,19 @@
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_taxon_datasets(taxon, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Taxon4 taxon: (required)
+        :param Taxon3 taxon: (required)
         :param FilterArgExpressionExperiment filter:
         :param int offset:
         :param int limit:
-        :param SortArgTaxon sort:
+        :param SortArgExpressionExperiment sort:
         :return: FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
             return self.get_taxon_datasets_with_http_info(taxon, **kwargs)  # noqa: E501
@@ -4097,29 +4457,31 @@
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_taxon_datasets_with_http_info(taxon, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Taxon4 taxon: (required)
+        :param Taxon3 taxon: (required)
         :param FilterArgExpressionExperiment filter:
         :param int offset:
         :param int limit:
-        :param SortArgTaxon sort:
+        :param SortArgExpressionExperiment sort:
         :return: FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['taxon', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_taxon_datasets" % key
@@ -4150,130 +4512,589 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/taxa/{taxon}/datasets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
-    def get_taxon_genes(self, taxon, gene, **kwargs):  # noqa: E501
-        """Retrieve all genes in a given taxon  # noqa: E501
+    def get_taxon_gene_go_terms(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve the GO terms associated to a gene in a given taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_gene_go_terms(taxon, gene, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon4 taxon: (required)
+        :param Gene3 gene: (required)
+        :return: ResponseDataObjectListGeneOntologyTermValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+        kwargs['_return_http_data_only'] = True
+        if kwargs.get('async_req'):
+            return self.get_taxon_gene_go_terms_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+        else:
+            (data) = self.get_taxon_gene_go_terms_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+            return data
+
+    def get_taxon_gene_go_terms_with_http_info(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve the GO terms associated to a gene in a given taxon  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_taxon_genes(taxon, gene, async_req=True)
+        >>> thread = api.get_taxon_gene_go_terms_with_http_info(taxon, gene, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon4 taxon: (required)
+        :param Gene3 gene: (required)
+        :return: ResponseDataObjectListGeneOntologyTermValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+
+        all_params = ['taxon', 'gene']  # noqa: E501
+        all_params.append('async_req')
+        all_params.append('_return_http_data_only')
+        all_params.append('_preload_content')
+        all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
+
+        params = locals()
+        for key, val in six.iteritems(params['kwargs']):
+            if key not in all_params:
+                raise TypeError(
+                    "Got an unexpected keyword argument '%s'"
+                    " to method get_taxon_gene_go_terms" % key
+                )
+            params[key] = val
+        del params['kwargs']
+        # verify the required parameter 'taxon' is set
+        if ('taxon' not in params or
+                params['taxon'] is None):
+            raise ValueError("Missing the required parameter `taxon` when calling `get_taxon_gene_go_terms`")  # noqa: E501
+        # verify the required parameter 'gene' is set
+        if ('gene' not in params or
+                params['gene'] is None):
+            raise ValueError("Missing the required parameter `gene` when calling `get_taxon_gene_go_terms`")  # noqa: E501
+
+        collection_formats = {}
+
+        path_params = {}
+        if 'taxon' in params:
+            path_params['taxon'] = params['taxon']  # noqa: E501
+        if 'gene' in params:
+            path_params['gene'] = params['gene']  # noqa: E501
+
+        query_params = []
+
+        header_params = {}
+
+        form_params = []
+        local_var_files = {}
+
+        body_params = None
+        # HTTP header `Accept`
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListGeneOntologyTermValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
+
+        # Authentication setting
+        auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
+
+        return self.api_client.call_api(
+            '/taxa/{taxon}/genes/{gene}/goTerms', 'GET',
+            path_params,
+            query_params,
+            header_params,
+            body=body_params,
+            post_params=form_params,
+            files=local_var_files,
+            response_type=response_type,
+            auth_settings=auth_settings,
+            async_req=params.get('async_req'),
+            _return_http_data_only=params.get('_return_http_data_only'),
+            _preload_content=params.get('_preload_content', True),
+            _request_timeout=params.get('_request_timeout'),
+            collection_formats=collection_formats)
+
+    def get_taxon_gene_locations(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve physical locations for a given gene and taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_gene_locations(taxon, gene, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param Taxon5 taxon: (required)
         :param Gene4 gene: (required)
-        :return: ResponseDataObjectListGeneValueObject
+        :return: ResponseDataObjectListPhysicalLocationValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
         kwargs['_return_http_data_only'] = True
         if kwargs.get('async_req'):
-            return self.get_taxon_genes_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+            return self.get_taxon_gene_locations_with_http_info(taxon, gene, **kwargs)  # noqa: E501
         else:
-            (data) = self.get_taxon_genes_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+            (data) = self.get_taxon_gene_locations_with_http_info(taxon, gene, **kwargs)  # noqa: E501
             return data
 
-    def get_taxon_genes_with_http_info(self, taxon, gene, **kwargs):  # noqa: E501
-        """Retrieve all genes in a given taxon  # noqa: E501
+    def get_taxon_gene_locations_with_http_info(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve physical locations for a given gene and taxon  # noqa: E501
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
-        >>> thread = api.get_taxon_genes_with_http_info(taxon, gene, async_req=True)
+        >>> thread = api.get_taxon_gene_locations_with_http_info(taxon, gene, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
         :param Taxon5 taxon: (required)
         :param Gene4 gene: (required)
-        :return: ResponseDataObjectListGeneValueObject
+        :return: ResponseDataObjectListPhysicalLocationValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['taxon', 'gene']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
+
+        params = locals()
+        for key, val in six.iteritems(params['kwargs']):
+            if key not in all_params:
+                raise TypeError(
+                    "Got an unexpected keyword argument '%s'"
+                    " to method get_taxon_gene_locations" % key
+                )
+            params[key] = val
+        del params['kwargs']
+        # verify the required parameter 'taxon' is set
+        if ('taxon' not in params or
+                params['taxon'] is None):
+            raise ValueError("Missing the required parameter `taxon` when calling `get_taxon_gene_locations`")  # noqa: E501
+        # verify the required parameter 'gene' is set
+        if ('gene' not in params or
+                params['gene'] is None):
+            raise ValueError("Missing the required parameter `gene` when calling `get_taxon_gene_locations`")  # noqa: E501
+
+        collection_formats = {}
+
+        path_params = {}
+        if 'taxon' in params:
+            path_params['taxon'] = params['taxon']  # noqa: E501
+        if 'gene' in params:
+            path_params['gene'] = params['gene']  # noqa: E501
+
+        query_params = []
+
+        header_params = {}
+
+        form_params = []
+        local_var_files = {}
+
+        body_params = None
+        # HTTP header `Accept`
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListPhysicalLocationValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
+
+        # Authentication setting
+        auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
+
+        return self.api_client.call_api(
+            '/taxa/{taxon}/genes/{gene}/locations', 'GET',
+            path_params,
+            query_params,
+            header_params,
+            body=body_params,
+            post_params=form_params,
+            files=local_var_files,
+            response_type=response_type,
+            auth_settings=auth_settings,
+            async_req=params.get('async_req'),
+            _return_http_data_only=params.get('_return_http_data_only'),
+            _preload_content=params.get('_preload_content', True),
+            _request_timeout=params.get('_request_timeout'),
+            collection_formats=collection_formats)
+
+    def get_taxon_gene_probes(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve the probes associated to a genes across all platforms in a given taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_gene_probes(taxon, gene, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon6 taxon: (required)
+        :param Gene5 gene: (required)
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectCompositeSequenceValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+        kwargs['_return_http_data_only'] = True
+        if kwargs.get('async_req'):
+            return self.get_taxon_gene_probes_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+        else:
+            (data) = self.get_taxon_gene_probes_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+            return data
+
+    def get_taxon_gene_probes_with_http_info(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve the probes associated to a genes across all platforms in a given taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_gene_probes_with_http_info(taxon, gene, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon6 taxon: (required)
+        :param Gene5 gene: (required)
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectCompositeSequenceValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+
+        all_params = ['taxon', 'gene', 'offset', 'limit']  # noqa: E501
+        all_params.append('async_req')
+        all_params.append('_return_http_data_only')
+        all_params.append('_preload_content')
+        all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
+
+        params = locals()
+        for key, val in six.iteritems(params['kwargs']):
+            if key not in all_params:
+                raise TypeError(
+                    "Got an unexpected keyword argument '%s'"
+                    " to method get_taxon_gene_probes" % key
+                )
+            params[key] = val
+        del params['kwargs']
+        # verify the required parameter 'taxon' is set
+        if ('taxon' not in params or
+                params['taxon'] is None):
+            raise ValueError("Missing the required parameter `taxon` when calling `get_taxon_gene_probes`")  # noqa: E501
+        # verify the required parameter 'gene' is set
+        if ('gene' not in params or
+                params['gene'] is None):
+            raise ValueError("Missing the required parameter `gene` when calling `get_taxon_gene_probes`")  # noqa: E501
+
+        collection_formats = {}
+
+        path_params = {}
+        if 'taxon' in params:
+            path_params['taxon'] = params['taxon']  # noqa: E501
+        if 'gene' in params:
+            path_params['gene'] = params['gene']  # noqa: E501
+
+        query_params = []
+        if 'offset' in params:
+            query_params.append(('offset', params['offset']))  # noqa: E501
+        if 'limit' in params:
+            query_params.append(('limit', params['limit']))  # noqa: E501
+
+        header_params = {}
+
+        form_params = []
+        local_var_files = {}
+
+        body_params = None
+        # HTTP header `Accept`
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'PaginatedResponseDataObjectCompositeSequenceValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
+
+        # Authentication setting
+        auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
+
+        return self.api_client.call_api(
+            '/taxa/{taxon}/genes/{gene}/probes', 'GET',
+            path_params,
+            query_params,
+            header_params,
+            body=body_params,
+            post_params=form_params,
+            files=local_var_files,
+            response_type=response_type,
+            auth_settings=auth_settings,
+            async_req=params.get('async_req'),
+            _return_http_data_only=params.get('_return_http_data_only'),
+            _preload_content=params.get('_preload_content', True),
+            _request_timeout=params.get('_request_timeout'),
+            collection_formats=collection_formats)
+
+    def get_taxon_genes(self, taxon, **kwargs):  # noqa: E501
+        """Retrieve all genes in a given taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_genes(taxon, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon7 taxon: (required)
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectGeneValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+        kwargs['_return_http_data_only'] = True
+        if kwargs.get('async_req'):
+            return self.get_taxon_genes_with_http_info(taxon, **kwargs)  # noqa: E501
+        else:
+            (data) = self.get_taxon_genes_with_http_info(taxon, **kwargs)  # noqa: E501
+            return data
+
+    def get_taxon_genes_with_http_info(self, taxon, **kwargs):  # noqa: E501
+        """Retrieve all genes in a given taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_genes_with_http_info(taxon, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon7 taxon: (required)
+        :param int offset:
+        :param int limit:
+        :return: PaginatedResponseDataObjectGeneValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+
+        all_params = ['taxon', 'offset', 'limit']  # noqa: E501
+        all_params.append('async_req')
+        all_params.append('_return_http_data_only')
+        all_params.append('_preload_content')
+        all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_taxon_genes" % key
                 )
             params[key] = val
         del params['kwargs']
         # verify the required parameter 'taxon' is set
         if ('taxon' not in params or
                 params['taxon'] is None):
             raise ValueError("Missing the required parameter `taxon` when calling `get_taxon_genes`")  # noqa: E501
+
+        collection_formats = {}
+
+        path_params = {}
+        if 'taxon' in params:
+            path_params['taxon'] = params['taxon']  # noqa: E501
+
+        query_params = []
+        if 'offset' in params:
+            query_params.append(('offset', params['offset']))  # noqa: E501
+        if 'limit' in params:
+            query_params.append(('limit', params['limit']))  # noqa: E501
+
+        header_params = {}
+
+        form_params = []
+        local_var_files = {}
+
+        body_params = None
+        # HTTP header `Accept`
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'PaginatedResponseDataObjectGeneValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
+
+        # Authentication setting
+        auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
+
+        return self.api_client.call_api(
+            '/taxa/{taxon}/genes', 'GET',
+            path_params,
+            query_params,
+            header_params,
+            body=body_params,
+            post_params=form_params,
+            files=local_var_files,
+            response_type=response_type,
+            auth_settings=auth_settings,
+            async_req=params.get('async_req'),
+            _return_http_data_only=params.get('_return_http_data_only'),
+            _preload_content=params.get('_preload_content', True),
+            _request_timeout=params.get('_request_timeout'),
+            collection_formats=collection_formats)
+
+    def get_taxon_genes_by_ids(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve genes matching gene identifiers in a given taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_genes_by_ids(taxon, gene, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon8 taxon: (required)
+        :param list[object] gene: (required)
+        :return: ResponseDataObjectListGeneValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+        kwargs['_return_http_data_only'] = True
+        if kwargs.get('async_req'):
+            return self.get_taxon_genes_by_ids_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+        else:
+            (data) = self.get_taxon_genes_by_ids_with_http_info(taxon, gene, **kwargs)  # noqa: E501
+            return data
+
+    def get_taxon_genes_by_ids_with_http_info(self, taxon, gene, **kwargs):  # noqa: E501
+        """Retrieve genes matching gene identifiers in a given taxon  # noqa: E501
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+        >>> thread = api.get_taxon_genes_by_ids_with_http_info(taxon, gene, async_req=True)
+        >>> result = thread.get()
+
+        :param async_req bool
+        :param Taxon8 taxon: (required)
+        :param list[object] gene: (required)
+        :return: ResponseDataObjectListGeneValueObject
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+
+        all_params = ['taxon', 'gene']  # noqa: E501
+        all_params.append('async_req')
+        all_params.append('_return_http_data_only')
+        all_params.append('_preload_content')
+        all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
+
+        params = locals()
+        for key, val in six.iteritems(params['kwargs']):
+            if key not in all_params:
+                raise TypeError(
+                    "Got an unexpected keyword argument '%s'"
+                    " to method get_taxon_genes_by_ids" % key
+                )
+            params[key] = val
+        del params['kwargs']
+        # verify the required parameter 'taxon' is set
+        if ('taxon' not in params or
+                params['taxon'] is None):
+            raise ValueError("Missing the required parameter `taxon` when calling `get_taxon_genes_by_ids`")  # noqa: E501
         # verify the required parameter 'gene' is set
         if ('gene' not in params or
                 params['gene'] is None):
-            raise ValueError("Missing the required parameter `gene` when calling `get_taxon_genes`")  # noqa: E501
+            raise ValueError("Missing the required parameter `gene` when calling `get_taxon_genes_by_ids`")  # noqa: E501
 
         collection_formats = {}
 
         path_params = {}
         if 'taxon' in params:
             path_params['taxon'] = params['taxon']  # noqa: E501
         if 'gene' in params:
             path_params['gene'] = params['gene']  # noqa: E501
+            collection_formats['gene'] = ''  # noqa: E501
 
         query_params = []
 
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListGeneValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/taxa/{taxon}/genes/{gene}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListGeneValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4282,15 +5103,15 @@
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_taxon_genes_overlapping_chromosome(taxon, chromosome, start, size, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Taxon6 taxon: (required)
+        :param Taxon9 taxon: (required)
         :param str chromosome: (required)
         :param int start: (required)
         :param int size: (required)
         :param str strand:
         :return: ResponseDataObjectListGeneValueObject
                  If the method is called asynchronously,
                  returns the request thread.
@@ -4307,29 +5128,31 @@
 
         This method makes a synchronous HTTP request by default. To make an
         asynchronous HTTP request, please pass async_req=True
         >>> thread = api.get_taxon_genes_overlapping_chromosome_with_http_info(taxon, chromosome, start, size, async_req=True)
         >>> result = thread.get()
 
         :param async_req bool
-        :param Taxon6 taxon: (required)
+        :param Taxon9 taxon: (required)
         :param str chromosome: (required)
         :param int start: (required)
         :param int size: (required)
         :param str strand:
         :return: ResponseDataObjectListGeneValueObject
                  If the method is called asynchronously,
                  returns the request thread.
         """
 
         all_params = ['taxon', 'chromosome', 'start', 'size', 'strand']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method get_taxon_genes_overlapping_chromosome" % key
@@ -4372,29 +5195,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListGeneValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/taxa/{taxon}/chromosomes/{chromosome}/genes', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListGeneValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4445,14 +5274,16 @@
         """
 
         all_params = ['query', 'taxon', 'platform', 'result_types', 'limit', 'exclude']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method search" % key
@@ -4483,29 +5314,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'SearchResultsResponseDataObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/search', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='SearchResultsResponseDataObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4546,14 +5383,16 @@
         """
 
         all_params = ['query']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method search_annotations" % key
@@ -4573,29 +5412,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListAnnotationSearchResultValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/annotations/search', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListAnnotationSearchResultValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4638,14 +5483,16 @@
         """
 
         all_params = ['query']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method search_annotations_by_path_query" % key
@@ -4669,29 +5516,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'ResponseDataObjectListAnnotationSearchResultValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/annotations/search/{query}', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='ResponseDataObjectListAnnotationSearchResultValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4742,14 +5595,16 @@
         """
 
         all_params = ['query', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method search_datasets" % key
@@ -4777,29 +5632,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/annotations/search/datasets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4850,14 +5711,16 @@
         """
 
         all_params = ['query', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method search_datasets_by_query_in_path" % key
@@ -4889,29 +5752,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/annotations/search/{query}/datasets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -4964,14 +5833,16 @@
         """
 
         all_params = ['taxon', 'query', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method search_taxon_datasets" % key
@@ -5005,29 +5876,35 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/annotations/{taxon}/search/datasets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
 
@@ -5080,14 +5957,16 @@
         """
 
         all_params = ['taxon', 'query', 'filter', 'offset', 'limit', 'sort']  # noqa: E501
         all_params.append('async_req')
         all_params.append('_return_http_data_only')
         all_params.append('_preload_content')
         all_params.append('_request_timeout')
+        all_params.append('_force_table')
+
 
         params = locals()
         for key, val in six.iteritems(params['kwargs']):
             if key not in all_params:
                 raise TypeError(
                     "Got an unexpected keyword argument '%s'"
                     " to method search_taxon_datasets_by_query_in_path" % key
@@ -5125,28 +6004,34 @@
         header_params = {}
 
         form_params = []
         local_var_files = {}
 
         body_params = None
         # HTTP header `Accept`
-        header_params['Accept'] = self.api_client.select_header_accept(
-            ['application/json'])  # noqa: E501
+        if not kwargs.get('_force_table'):
+            header_params['Accept'] = self.api_client.select_header_accept(
+                ['application/json'])  # noqa: E501
+            response_type = 'QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject' # noqa: E501
+        else:
+            header_params['Accept'] = 'text/tab-separated-values; charset=UTF-8'
+            response_type = 'str'
+            
 
         # Authentication setting
         auth_settings = ['basicAuth', 'cookieAuth']  # noqa: E501
 
         return self.api_client.call_api(
             '/annotations/{taxon}/search/{query}/datasets', 'GET',
             path_params,
             query_params,
             header_params,
             body=body_params,
             post_params=form_params,
             files=local_var_files,
-            response_type='QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject',  # noqa: E501
+            response_type=response_type,
             auth_settings=auth_settings,
             async_req=params.get('async_req'),
             _return_http_data_only=params.get('_return_http_data_only'),
             _preload_content=params.get('_preload_content', True),
             _request_timeout=params.get('_request_timeout'),
             collection_formats=collection_formats)
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/api_client.py` & `gemmapy-1.0.1/gemmapy/sdk/api_client.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,14 +1,14 @@
 # coding: utf-8
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 from __future__ import absolute_import
 
 import datetime
 import json
@@ -69,15 +69,15 @@
         self.pool = ThreadPool()
         self.rest_client = rest.RESTClientObject(configuration)
         self.default_headers = {}
         if header_name is not None:
             self.default_headers[header_name] = header_value
         self.cookie = cookie
         # Set default User-Agent.
-        self.user_agent = 'gemmapy/0.0.2'
+        self.user_agent = 'gemmapy/1.0.1'
 
     def __del__(self):
         self.pool.close()
         self.pool.join()
 
     @property
     def user_agent(self):
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/configuration.py` & `gemmapy-1.0.1/gemmapy/sdk/configuration.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 from __future__ import absolute_import
 
 import copy
@@ -252,10 +252,10 @@
         """Gets the essential information for debugging.
 
         :return: The report for debugging.
         """
         return "Python SDK Debug Report:\n"\
                "OS: {env}\n"\
                "Python Version: {pyversion}\n"\
-               "Version of the API: 2.7.4\n"\
+               "Version of the API: 2.7.5\n"\
                "SDK Package Version: 1.0.0".\
                format(env=sys.platform, pyversion=sys.version)
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/__init__.py` & `gemmapy-1.0.1/gemmapy/sdk/__init__.py`

 * *Files 5% similar despite different names*

```diff
@@ -1,23 +1,29 @@
 # coding: utf-8
 
 # flake8: noqa
+
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 from __future__ import absolute_import
 
-# import models into model package
+# import apis into sdk package
+from gemmapy.sdk.api.default_api import DefaultApi
+# import ApiClient
+from gemmapy.sdk.api_client import ApiClient
+from gemmapy.sdk.configuration import Configuration
+# import models into sdk package
 from gemmapy.sdk.models.annotation_search_result_value_object import AnnotationSearchResultValueObject
 from gemmapy.sdk.models.annotation_value_object import AnnotationValueObject
 from gemmapy.sdk.models.annotation_with_usage_statistics_value_object import AnnotationWithUsageStatisticsValueObject
 from gemmapy.sdk.models.api_info_value_object import ApiInfoValueObject
 from gemmapy.sdk.models.array_design_value_object import ArrayDesignValueObject
 from gemmapy.sdk.models.array_design_with_usage_statistics_value_object import ArrayDesignWithUsageStatisticsValueObject
 from gemmapy.sdk.models.audit_event_value_object import AuditEventValueObject
@@ -57,56 +63,59 @@
 from gemmapy.sdk.models.expression_experiment_with_search_result_value_object import ExpressionExperimentWithSearchResultValueObject
 from gemmapy.sdk.models.external_database_value_object import ExternalDatabaseValueObject
 from gemmapy.sdk.models.factor_value_basic_value_object import FactorValueBasicValueObject
 from gemmapy.sdk.models.factor_value_value_object import FactorValueValueObject
 from gemmapy.sdk.models.filter_arg_array_design import FilterArgArrayDesign
 from gemmapy.sdk.models.filter_arg_expression_analysis_result_set import FilterArgExpressionAnalysisResultSet
 from gemmapy.sdk.models.filter_arg_expression_experiment import FilterArgExpressionExperiment
+from gemmapy.sdk.models.filtered_and_inferred_and_paginated_response_data_object_expression_experiment_value_object import FilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_array_design_value_object import FilteredAndPaginatedResponseDataObjectArrayDesignValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_composite_sequence_value_object import FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_differential_expression_analysis_result_set_value_object import FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_expression_experiment_value_object import FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
 from gemmapy.sdk.models.filtered_and_paginated_response_data_object_gene_value_object import FilteredAndPaginatedResponseDataObjectGeneValueObject
 from gemmapy.sdk.models.geeq_value_object import GeeqValueObject
 from gemmapy.sdk.models.gene import Gene
 from gemmapy.sdk.models.gene1 import Gene1
 from gemmapy.sdk.models.gene2 import Gene2
 from gemmapy.sdk.models.gene3 import Gene3
 from gemmapy.sdk.models.gene4 import Gene4
+from gemmapy.sdk.models.gene5 import Gene5
 from gemmapy.sdk.models.gene_arg import GeneArg
 from gemmapy.sdk.models.gene_element_expressions_value_object import GeneElementExpressionsValueObject
 from gemmapy.sdk.models.gene_ontology_term_value_object import GeneOntologyTermValueObject
 from gemmapy.sdk.models.gene_set_value_object import GeneSetValueObject
 from gemmapy.sdk.models.gene_value_object import GeneValueObject
-from gemmapy.sdk.models.limited_response_data_object_annotation_with_usage_statistics_value_object import LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
-from gemmapy.sdk.models.limited_response_data_object_array_design_with_usage_statistics_value_object import LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
 from gemmapy.sdk.models.measurement_value_object import MeasurementValueObject
 from gemmapy.sdk.models.one_of_search_result_value_object_object_result_object import OneOfSearchResultValueObjectObjectResultObject
 from gemmapy.sdk.models.ontology_term_value_object import OntologyTermValueObject
 from gemmapy.sdk.models.paginated_response_data_object_composite_sequence_value_object import PaginatedResponseDataObjectCompositeSequenceValueObject
 from gemmapy.sdk.models.paginated_response_data_object_expression_experiment_value_object import PaginatedResponseDataObjectExpressionExperimentValueObject
+from gemmapy.sdk.models.paginated_response_data_object_gene_value_object import PaginatedResponseDataObjectGeneValueObject
+from gemmapy.sdk.models.paginated_results_response_data_object_differential_expression_analysis_result_set_value_object import PaginatedResultsResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
 from gemmapy.sdk.models.physical_location_value_object import PhysicalLocationValueObject
 from gemmapy.sdk.models.platform import Platform
 from gemmapy.sdk.models.platform1 import Platform1
 from gemmapy.sdk.models.platform2 import Platform2
 from gemmapy.sdk.models.platform3 import Platform3
 from gemmapy.sdk.models.platform4 import Platform4
 from gemmapy.sdk.models.platform5 import Platform5
 from gemmapy.sdk.models.platform_arg import PlatformArg
 from gemmapy.sdk.models.probe import Probe
 from gemmapy.sdk.models.quantitation_type import QuantitationType
 from gemmapy.sdk.models.quantitation_type_arg import QuantitationTypeArg
 from gemmapy.sdk.models.quantitation_type_value_object import QuantitationTypeValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_limited_response_data_object_annotation_with_usage_statistics_value_object import QueriedAndFilteredAndInferredAndLimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_limited_response_data_object_array_design_with_usage_statistics_value_object import QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_limited_response_data_object_category_with_usage_statistics_value_object import QueriedAndFilteredAndInferredAndLimitedResponseDataObjectCategoryWithUsageStatisticsValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_and_paginated_response_data_object_expression_experiment_with_search_result_value_object import QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
+from gemmapy.sdk.models.queried_and_filtered_and_inferred_response_data_object_taxon_with_usage_statistics_value_object import QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject
 from gemmapy.sdk.models.queried_and_filtered_and_paginated_response_data_object_expression_experiment_value_object import QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject
-from gemmapy.sdk.models.queried_and_filtered_and_paginated_response_data_object_expression_experiment_with_search_result_value_object import QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject
-from gemmapy.sdk.models.queried_and_filtered_response_data_object_category_with_usage_statistics_value_object import QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject
-from gemmapy.sdk.models.queried_and_filtered_response_data_object_taxon_with_usage_statistics_value_object import QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject
 from gemmapy.sdk.models.query_arg import QueryArg
 from gemmapy.sdk.models.response_data_object_api_info_value_object import ResponseDataObjectApiInfoValueObject
-from gemmapy.sdk.models.response_data_object_differential_expression_analysis_result_set_value_object import ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject
 from gemmapy.sdk.models.response_data_object_list_annotation_search_result_value_object import ResponseDataObjectListAnnotationSearchResultValueObject
 from gemmapy.sdk.models.response_data_object_list_array_design_value_object import ResponseDataObjectListArrayDesignValueObject
 from gemmapy.sdk.models.response_data_object_list_bio_assay_value_object import ResponseDataObjectListBioAssayValueObject
 from gemmapy.sdk.models.response_data_object_list_differential_expression_analysis_value_object import ResponseDataObjectListDifferentialExpressionAnalysisValueObject
 from gemmapy.sdk.models.response_data_object_list_experiment_expression_levels_value_object import ResponseDataObjectListExperimentExpressionLevelsValueObject
 from gemmapy.sdk.models.response_data_object_list_gene_ontology_term_value_object import ResponseDataObjectListGeneOntologyTermValueObject
 from gemmapy.sdk.models.response_data_object_list_gene_value_object import ResponseDataObjectListGeneValueObject
@@ -122,22 +131,25 @@
 from gemmapy.sdk.models.search_result_value_object_object import SearchResultValueObjectObject
 from gemmapy.sdk.models.search_results_response_data_object import SearchResultsResponseDataObject
 from gemmapy.sdk.models.search_settings_value_object import SearchSettingsValueObject
 from gemmapy.sdk.models.simple_svd_value_object import SimpleSVDValueObject
 from gemmapy.sdk.models.sort_arg_array_design import SortArgArrayDesign
 from gemmapy.sdk.models.sort_arg_expression_analysis_result_set import SortArgExpressionAnalysisResultSet
 from gemmapy.sdk.models.sort_arg_expression_experiment import SortArgExpressionExperiment
-from gemmapy.sdk.models.sort_arg_taxon import SortArgTaxon
 from gemmapy.sdk.models.sort_value_object import SortValueObject
 from gemmapy.sdk.models.statement_value_object import StatementValueObject
+from gemmapy.sdk.models.taxa import Taxa
 from gemmapy.sdk.models.taxon import Taxon
 from gemmapy.sdk.models.taxon1 import Taxon1
 from gemmapy.sdk.models.taxon2 import Taxon2
 from gemmapy.sdk.models.taxon3 import Taxon3
 from gemmapy.sdk.models.taxon4 import Taxon4
 from gemmapy.sdk.models.taxon5 import Taxon5
 from gemmapy.sdk.models.taxon6 import Taxon6
+from gemmapy.sdk.models.taxon7 import Taxon7
+from gemmapy.sdk.models.taxon8 import Taxon8
+from gemmapy.sdk.models.taxon9 import Taxon9
 from gemmapy.sdk.models.taxon_arg import TaxonArg
 from gemmapy.sdk.models.taxon_value_object import TaxonValueObject
 from gemmapy.sdk.models.taxon_with_usage_statistics_value_object import TaxonWithUsageStatisticsValueObject
 from gemmapy.sdk.models.vector_element_value_object import VectorElementValueObject
 from gemmapy.sdk.models.well_composed_error_body import WellComposedErrorBody
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/annotation_search_result_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/annotation_search_result_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/annotation_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/annotation_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/annotation_with_usage_statistics_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/annotation_with_usage_statistics_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/api_info_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/api_info_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/array_design_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/array_design_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -43,16 +43,16 @@
         'is_mergee': 'bool',
         'name': 'str',
         'short_name': 'str',
         'technology_type': 'str',
         'release_version': 'str',
         'release_url': 'str',
         'external_references': 'list[DatabaseEntryValueObject]',
-        'taxon_id': 'int',
         'number_of_expression_experiments': 'int',
+        'taxon_id': 'int',
         'trouble_details': 'str',
         'number_of_switched_expression_experiments': 'int',
         'taxon': 'TaxonValueObject'
     }
 
     attribute_map = {
         'id': 'id',
@@ -70,22 +70,22 @@
         'is_mergee': 'isMergee',
         'name': 'name',
         'short_name': 'shortName',
         'technology_type': 'technologyType',
         'release_version': 'releaseVersion',
         'release_url': 'releaseUrl',
         'external_references': 'externalReferences',
-        'taxon_id': 'taxonID',
         'number_of_expression_experiments': 'numberOfExpressionExperiments',
+        'taxon_id': 'taxonID',
         'trouble_details': 'troubleDetails',
         'number_of_switched_expression_experiments': 'numberOfSwitchedExpressionExperiments',
         'taxon': 'taxon'
     }
 
-    def __init__(self, id=None, last_updated=None, troubled=None, last_troubled_event=None, needs_attention=None, last_needs_attention_event=None, curation_note=None, last_note_update_event=None, color=None, description=None, expression_experiment_count=None, is_merged=None, is_mergee=None, name=None, short_name=None, technology_type=None, release_version=None, release_url=None, external_references=None, taxon_id=None, number_of_expression_experiments=None, trouble_details=None, number_of_switched_expression_experiments=None, taxon=None):  # noqa: E501
+    def __init__(self, id=None, last_updated=None, troubled=None, last_troubled_event=None, needs_attention=None, last_needs_attention_event=None, curation_note=None, last_note_update_event=None, color=None, description=None, expression_experiment_count=None, is_merged=None, is_mergee=None, name=None, short_name=None, technology_type=None, release_version=None, release_url=None, external_references=None, number_of_expression_experiments=None, taxon_id=None, trouble_details=None, number_of_switched_expression_experiments=None, taxon=None):  # noqa: E501
         """ArrayDesignValueObject - a model defined in Swagger"""  # noqa: E501
         self._id = None
         self._last_updated = None
         self._troubled = None
         self._last_troubled_event = None
         self._needs_attention = None
         self._last_needs_attention_event = None
@@ -98,16 +98,16 @@
         self._is_mergee = None
         self._name = None
         self._short_name = None
         self._technology_type = None
         self._release_version = None
         self._release_url = None
         self._external_references = None
-        self._taxon_id = None
         self._number_of_expression_experiments = None
+        self._taxon_id = None
         self._trouble_details = None
         self._number_of_switched_expression_experiments = None
         self._taxon = None
         self.discriminator = None
         if id is not None:
             self.id = id
         if last_updated is not None:
@@ -142,18 +142,18 @@
             self.technology_type = technology_type
         if release_version is not None:
             self.release_version = release_version
         if release_url is not None:
             self.release_url = release_url
         if external_references is not None:
             self.external_references = external_references
-        if taxon_id is not None:
-            self.taxon_id = taxon_id
         if number_of_expression_experiments is not None:
             self.number_of_expression_experiments = number_of_expression_experiments
+        if taxon_id is not None:
+            self.taxon_id = taxon_id
         if trouble_details is not None:
             self.trouble_details = trouble_details
         if number_of_switched_expression_experiments is not None:
             self.number_of_switched_expression_experiments = number_of_switched_expression_experiments
         if taxon is not None:
             self.taxon = taxon
 
@@ -565,54 +565,54 @@
         :param external_references: The external_references of this ArrayDesignValueObject.  # noqa: E501
         :type: list[DatabaseEntryValueObject]
         """
 
         self._external_references = external_references
 
     @property
-    def taxon_id(self):
-        """Gets the taxon_id of this ArrayDesignValueObject.  # noqa: E501
+    def number_of_expression_experiments(self):
+        """Gets the number_of_expression_experiments of this ArrayDesignValueObject.  # noqa: E501
 
 
-        :return: The taxon_id of this ArrayDesignValueObject.  # noqa: E501
+        :return: The number_of_expression_experiments of this ArrayDesignValueObject.  # noqa: E501
         :rtype: int
         """
-        return self._taxon_id
+        return self._number_of_expression_experiments
 
-    @taxon_id.setter
-    def taxon_id(self, taxon_id):
-        """Sets the taxon_id of this ArrayDesignValueObject.
+    @number_of_expression_experiments.setter
+    def number_of_expression_experiments(self, number_of_expression_experiments):
+        """Sets the number_of_expression_experiments of this ArrayDesignValueObject.
 
 
-        :param taxon_id: The taxon_id of this ArrayDesignValueObject.  # noqa: E501
+        :param number_of_expression_experiments: The number_of_expression_experiments of this ArrayDesignValueObject.  # noqa: E501
         :type: int
         """
 
-        self._taxon_id = taxon_id
+        self._number_of_expression_experiments = number_of_expression_experiments
 
     @property
-    def number_of_expression_experiments(self):
-        """Gets the number_of_expression_experiments of this ArrayDesignValueObject.  # noqa: E501
+    def taxon_id(self):
+        """Gets the taxon_id of this ArrayDesignValueObject.  # noqa: E501
 
 
-        :return: The number_of_expression_experiments of this ArrayDesignValueObject.  # noqa: E501
+        :return: The taxon_id of this ArrayDesignValueObject.  # noqa: E501
         :rtype: int
         """
-        return self._number_of_expression_experiments
+        return self._taxon_id
 
-    @number_of_expression_experiments.setter
-    def number_of_expression_experiments(self, number_of_expression_experiments):
-        """Sets the number_of_expression_experiments of this ArrayDesignValueObject.
+    @taxon_id.setter
+    def taxon_id(self, taxon_id):
+        """Sets the taxon_id of this ArrayDesignValueObject.
 
 
-        :param number_of_expression_experiments: The number_of_expression_experiments of this ArrayDesignValueObject.  # noqa: E501
+        :param taxon_id: The taxon_id of this ArrayDesignValueObject.  # noqa: E501
         :type: int
         """
 
-        self._number_of_expression_experiments = number_of_expression_experiments
+        self._taxon_id = taxon_id
 
     @property
     def trouble_details(self):
         """Gets the trouble_details of this ArrayDesignValueObject.  # noqa: E501
 
 
         :return: The trouble_details of this ArrayDesignValueObject.  # noqa: E501
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/array_design_with_usage_statistics_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/array_design_with_usage_statistics_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/audit_event_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/audit_event_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -30,56 +30,56 @@
     swagger_types = {
         'id': 'int',
         'performer': 'str',
         '_date': 'datetime',
         'action': 'str',
         'note': 'str',
         'detail': 'str',
-        'event_type_name': 'str',
-        'action_name': 'str'
+        'action_name': 'str',
+        'event_type_name': 'str'
     }
 
     attribute_map = {
         'id': 'id',
         'performer': 'performer',
         '_date': 'date',
         'action': 'action',
         'note': 'note',
         'detail': 'detail',
-        'event_type_name': 'eventTypeName',
-        'action_name': 'actionName'
+        'action_name': 'actionName',
+        'event_type_name': 'eventTypeName'
     }
 
-    def __init__(self, id=None, performer=None, _date=None, action=None, note=None, detail=None, event_type_name=None, action_name=None):  # noqa: E501
+    def __init__(self, id=None, performer=None, _date=None, action=None, note=None, detail=None, action_name=None, event_type_name=None):  # noqa: E501
         """AuditEventValueObject - a model defined in Swagger"""  # noqa: E501
         self._id = None
         self._performer = None
         self.__date = None
         self._action = None
         self._note = None
         self._detail = None
-        self._event_type_name = None
         self._action_name = None
+        self._event_type_name = None
         self.discriminator = None
         if id is not None:
             self.id = id
         if performer is not None:
             self.performer = performer
         if _date is not None:
             self._date = _date
         if action is not None:
             self.action = action
         if note is not None:
             self.note = note
         if detail is not None:
             self.detail = detail
-        if event_type_name is not None:
-            self.event_type_name = event_type_name
         if action_name is not None:
             self.action_name = action_name
+        if event_type_name is not None:
+            self.event_type_name = event_type_name
 
     @property
     def id(self):
         """Gets the id of this AuditEventValueObject.  # noqa: E501
 
 
         :return: The id of this AuditEventValueObject.  # noqa: E501
@@ -206,54 +206,54 @@
         :param detail: The detail of this AuditEventValueObject.  # noqa: E501
         :type: str
         """
 
         self._detail = detail
 
     @property
-    def event_type_name(self):
-        """Gets the event_type_name of this AuditEventValueObject.  # noqa: E501
+    def action_name(self):
+        """Gets the action_name of this AuditEventValueObject.  # noqa: E501
 
 
-        :return: The event_type_name of this AuditEventValueObject.  # noqa: E501
+        :return: The action_name of this AuditEventValueObject.  # noqa: E501
         :rtype: str
         """
-        return self._event_type_name
+        return self._action_name
 
-    @event_type_name.setter
-    def event_type_name(self, event_type_name):
-        """Sets the event_type_name of this AuditEventValueObject.
+    @action_name.setter
+    def action_name(self, action_name):
+        """Sets the action_name of this AuditEventValueObject.
 
 
-        :param event_type_name: The event_type_name of this AuditEventValueObject.  # noqa: E501
+        :param action_name: The action_name of this AuditEventValueObject.  # noqa: E501
         :type: str
         """
 
-        self._event_type_name = event_type_name
+        self._action_name = action_name
 
     @property
-    def action_name(self):
-        """Gets the action_name of this AuditEventValueObject.  # noqa: E501
+    def event_type_name(self):
+        """Gets the event_type_name of this AuditEventValueObject.  # noqa: E501
 
 
-        :return: The action_name of this AuditEventValueObject.  # noqa: E501
+        :return: The event_type_name of this AuditEventValueObject.  # noqa: E501
         :rtype: str
         """
-        return self._action_name
+        return self._event_type_name
 
-    @action_name.setter
-    def action_name(self, action_name):
-        """Sets the action_name of this AuditEventValueObject.
+    @event_type_name.setter
+    def event_type_name(self, event_type_name):
+        """Sets the event_type_name of this AuditEventValueObject.
 
 
-        :param action_name: The action_name of this AuditEventValueObject.  # noqa: E501
+        :param event_type_name: The event_type_name of this AuditEventValueObject.  # noqa: E501
         :type: str
         """
 
-        self._action_name = action_name
+        self._event_type_name = event_type_name
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/bibliographic_phenotypes_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/bibliographic_phenotypes_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/bibliographic_reference_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/bibliographic_reference_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/bio_assay_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/bio_assay_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/bio_material_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/bio_material_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/bio_sequence_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/bio_sequence_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/category_with_usage_statistics_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/category_with_usage_statistics_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/characteristic_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/characteristic_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/citation_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/citation_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/composite_sequence_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/composite_sequence_arg.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/composite_sequence_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/composite_sequence_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/contrast_result_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/contrast_result_value_object.py`

 * *Files 2% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/database_entry_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/database_entry_arg.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/database_entry_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/database_entry_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset1.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset1.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset10.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset10.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset2.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset2.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset3.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset3.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset4.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset4.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset5.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset5.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset6.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset6.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset7.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset7.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset8.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset8.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset9.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset9.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/dataset_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/dataset_arg.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/diff_ex_result_set_summary_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/diff_ex_result_set_summary_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/differential_expression_analysis_result_set_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/differential_expression_analysis_result_value_object.py`

 * *Files 25% similar despite different names*

```diff
@@ -1,175 +1,253 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class DifferentialExpressionAnalysisResultSetValueObject(object):
+class DifferentialExpressionAnalysisResultValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'id': 'int',
-        'analysis': 'DifferentialExpressionAnalysisValueObject',
-        'experimental_factors': 'list[ExperimentalFactorValueObject]',
-        'baseline_group': 'FactorValueBasicValueObject',
-        'results': 'list[DifferentialExpressionAnalysisResultValueObject]'
+        'probe_id': 'int',
+        'probe_name': 'str',
+        'genes': 'list[GeneValueObject]',
+        'corrected_pvalue': 'float',
+        'rank': 'float',
+        'contrasts': 'list[ContrastResultValueObject]',
+        'pvalue': 'float'
     }
 
     attribute_map = {
         'id': 'id',
-        'analysis': 'analysis',
-        'experimental_factors': 'experimentalFactors',
-        'baseline_group': 'baselineGroup',
-        'results': 'results'
+        'probe_id': 'probeId',
+        'probe_name': 'probeName',
+        'genes': 'genes',
+        'corrected_pvalue': 'correctedPvalue',
+        'rank': 'rank',
+        'contrasts': 'contrasts',
+        'pvalue': 'pvalue'
     }
 
-    def __init__(self, id=None, analysis=None, experimental_factors=None, baseline_group=None, results=None):  # noqa: E501
-        """DifferentialExpressionAnalysisResultSetValueObject - a model defined in Swagger"""  # noqa: E501
+    def __init__(self, id=None, probe_id=None, probe_name=None, genes=None, corrected_pvalue=None, rank=None, contrasts=None, pvalue=None):  # noqa: E501
+        """DifferentialExpressionAnalysisResultValueObject - a model defined in Swagger"""  # noqa: E501
         self._id = None
-        self._analysis = None
-        self._experimental_factors = None
-        self._baseline_group = None
-        self._results = None
+        self._probe_id = None
+        self._probe_name = None
+        self._genes = None
+        self._corrected_pvalue = None
+        self._rank = None
+        self._contrasts = None
+        self._pvalue = None
         self.discriminator = None
         if id is not None:
             self.id = id
-        if analysis is not None:
-            self.analysis = analysis
-        if experimental_factors is not None:
-            self.experimental_factors = experimental_factors
-        if baseline_group is not None:
-            self.baseline_group = baseline_group
-        if results is not None:
-            self.results = results
+        if probe_id is not None:
+            self.probe_id = probe_id
+        if probe_name is not None:
+            self.probe_name = probe_name
+        if genes is not None:
+            self.genes = genes
+        if corrected_pvalue is not None:
+            self.corrected_pvalue = corrected_pvalue
+        if rank is not None:
+            self.rank = rank
+        if contrasts is not None:
+            self.contrasts = contrasts
+        if pvalue is not None:
+            self.pvalue = pvalue
 
     @property
     def id(self):
-        """Gets the id of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+        """Gets the id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
 
 
-        :return: The id of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+        :return: The id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
         :rtype: int
         """
         return self._id
 
     @id.setter
     def id(self, id):
-        """Sets the id of this DifferentialExpressionAnalysisResultSetValueObject.
+        """Sets the id of this DifferentialExpressionAnalysisResultValueObject.
 
 
-        :param id: The id of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+        :param id: The id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
         :type: int
         """
 
         self._id = id
 
     @property
-    def analysis(self):
-        """Gets the analysis of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+    def probe_id(self):
+        """Gets the probe_id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
 
 
-        :return: The analysis of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :rtype: DifferentialExpressionAnalysisValueObject
+        :return: The probe_id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :rtype: int
+        """
+        return self._probe_id
+
+    @probe_id.setter
+    def probe_id(self, probe_id):
+        """Sets the probe_id of this DifferentialExpressionAnalysisResultValueObject.
+
+
+        :param probe_id: The probe_id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :type: int
+        """
+
+        self._probe_id = probe_id
+
+    @property
+    def probe_name(self):
+        """Gets the probe_name of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+
+
+        :return: The probe_name of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._probe_name
+
+    @probe_name.setter
+    def probe_name(self, probe_name):
+        """Sets the probe_name of this DifferentialExpressionAnalysisResultValueObject.
+
+
+        :param probe_name: The probe_name of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._probe_name = probe_name
+
+    @property
+    def genes(self):
+        """Gets the genes of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+
+
+        :return: The genes of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :rtype: list[GeneValueObject]
+        """
+        return self._genes
+
+    @genes.setter
+    def genes(self, genes):
+        """Sets the genes of this DifferentialExpressionAnalysisResultValueObject.
+
+
+        :param genes: The genes of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :type: list[GeneValueObject]
+        """
+
+        self._genes = genes
+
+    @property
+    def corrected_pvalue(self):
+        """Gets the corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+
+
+        :return: The corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :rtype: float
         """
-        return self._analysis
+        return self._corrected_pvalue
 
-    @analysis.setter
-    def analysis(self, analysis):
-        """Sets the analysis of this DifferentialExpressionAnalysisResultSetValueObject.
+    @corrected_pvalue.setter
+    def corrected_pvalue(self, corrected_pvalue):
+        """Sets the corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.
 
 
-        :param analysis: The analysis of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :type: DifferentialExpressionAnalysisValueObject
+        :param corrected_pvalue: The corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :type: float
         """
 
-        self._analysis = analysis
+        self._corrected_pvalue = corrected_pvalue
 
     @property
-    def experimental_factors(self):
-        """Gets the experimental_factors of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+    def rank(self):
+        """Gets the rank of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
 
 
-        :return: The experimental_factors of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :rtype: list[ExperimentalFactorValueObject]
+        :return: The rank of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :rtype: float
         """
-        return self._experimental_factors
+        return self._rank
 
-    @experimental_factors.setter
-    def experimental_factors(self, experimental_factors):
-        """Sets the experimental_factors of this DifferentialExpressionAnalysisResultSetValueObject.
+    @rank.setter
+    def rank(self, rank):
+        """Sets the rank of this DifferentialExpressionAnalysisResultValueObject.
 
 
-        :param experimental_factors: The experimental_factors of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :type: list[ExperimentalFactorValueObject]
+        :param rank: The rank of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :type: float
         """
 
-        self._experimental_factors = experimental_factors
+        self._rank = rank
 
     @property
-    def baseline_group(self):
-        """Gets the baseline_group of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+    def contrasts(self):
+        """Gets the contrasts of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
 
 
-        :return: The baseline_group of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :rtype: FactorValueBasicValueObject
+        :return: The contrasts of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :rtype: list[ContrastResultValueObject]
         """
-        return self._baseline_group
+        return self._contrasts
 
-    @baseline_group.setter
-    def baseline_group(self, baseline_group):
-        """Sets the baseline_group of this DifferentialExpressionAnalysisResultSetValueObject.
+    @contrasts.setter
+    def contrasts(self, contrasts):
+        """Sets the contrasts of this DifferentialExpressionAnalysisResultValueObject.
 
 
-        :param baseline_group: The baseline_group of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :type: FactorValueBasicValueObject
+        :param contrasts: The contrasts of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :type: list[ContrastResultValueObject]
         """
 
-        self._baseline_group = baseline_group
+        self._contrasts = contrasts
 
     @property
-    def results(self):
-        """Gets the results of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+    def pvalue(self):
+        """Gets the pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
 
 
-        :return: The results of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :rtype: list[DifferentialExpressionAnalysisResultValueObject]
+        :return: The pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :rtype: float
         """
-        return self._results
+        return self._pvalue
 
-    @results.setter
-    def results(self, results):
-        """Sets the results of this DifferentialExpressionAnalysisResultSetValueObject.
+    @pvalue.setter
+    def pvalue(self, pvalue):
+        """Sets the pvalue of this DifferentialExpressionAnalysisResultValueObject.
 
 
-        :param results: The results of this DifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :type: list[DifferentialExpressionAnalysisResultValueObject]
+        :param pvalue: The pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+        :type: float
         """
 
-        self._results = results
+        self._pvalue = pvalue
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -184,15 +262,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(DifferentialExpressionAnalysisResultSetValueObject, dict):
+        if issubclass(DifferentialExpressionAnalysisResultValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -200,15 +278,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, DifferentialExpressionAnalysisResultSetValueObject):
+        if not isinstance(other, DifferentialExpressionAnalysisResultValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/differential_expression_analysis_result_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/paginated_response_data_object_composite_sequence_value_object.py`

 * *Files 26% similar despite different names*

```diff
@@ -1,253 +1,201 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class DifferentialExpressionAnalysisResultValueObject(object):
+class PaginatedResponseDataObjectCompositeSequenceValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'id': 'int',
-        'probe_id': 'int',
-        'probe_name': 'str',
-        'genes': 'list[GeneValueObject]',
-        'corrected_pvalue': 'float',
-        'rank': 'float',
-        'contrasts': 'list[ContrastResultValueObject]',
-        'pvalue': 'float'
+        'data': 'list[CompositeSequenceValueObject]',
+        'group_by': 'list[str]',
+        'sort': 'SortValueObject',
+        'offset': 'int',
+        'limit': 'int',
+        'total_elements': 'int'
     }
 
     attribute_map = {
-        'id': 'id',
-        'probe_id': 'probeId',
-        'probe_name': 'probeName',
-        'genes': 'genes',
-        'corrected_pvalue': 'correctedPvalue',
-        'rank': 'rank',
-        'contrasts': 'contrasts',
-        'pvalue': 'pvalue'
+        'data': 'data',
+        'group_by': 'groupBy',
+        'sort': 'sort',
+        'offset': 'offset',
+        'limit': 'limit',
+        'total_elements': 'totalElements'
     }
 
-    def __init__(self, id=None, probe_id=None, probe_name=None, genes=None, corrected_pvalue=None, rank=None, contrasts=None, pvalue=None):  # noqa: E501
-        """DifferentialExpressionAnalysisResultValueObject - a model defined in Swagger"""  # noqa: E501
-        self._id = None
-        self._probe_id = None
-        self._probe_name = None
-        self._genes = None
-        self._corrected_pvalue = None
-        self._rank = None
-        self._contrasts = None
-        self._pvalue = None
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
+        """PaginatedResponseDataObjectCompositeSequenceValueObject - a model defined in Swagger"""  # noqa: E501
+        self._data = None
+        self._group_by = None
+        self._sort = None
+        self._offset = None
+        self._limit = None
+        self._total_elements = None
         self.discriminator = None
-        if id is not None:
-            self.id = id
-        if probe_id is not None:
-            self.probe_id = probe_id
-        if probe_name is not None:
-            self.probe_name = probe_name
-        if genes is not None:
-            self.genes = genes
-        if corrected_pvalue is not None:
-            self.corrected_pvalue = corrected_pvalue
-        if rank is not None:
-            self.rank = rank
-        if contrasts is not None:
-            self.contrasts = contrasts
-        if pvalue is not None:
-            self.pvalue = pvalue
+        if data is not None:
+            self.data = data
+        if group_by is not None:
+            self.group_by = group_by
+        if sort is not None:
+            self.sort = sort
+        if offset is not None:
+            self.offset = offset
+        if limit is not None:
+            self.limit = limit
+        if total_elements is not None:
+            self.total_elements = total_elements
 
     @property
-    def id(self):
-        """Gets the id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+    def data(self):
+        """Gets the data of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
-        :return: The id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: int
-        """
-        return self._id
-
-    @id.setter
-    def id(self, id):
-        """Sets the id of this DifferentialExpressionAnalysisResultValueObject.
-
-
-        :param id: The id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: int
-        """
-
-        self._id = id
-
-    @property
-    def probe_id(self):
-        """Gets the probe_id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-
-
-        :return: The probe_id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: int
-        """
-        return self._probe_id
-
-    @probe_id.setter
-    def probe_id(self, probe_id):
-        """Sets the probe_id of this DifferentialExpressionAnalysisResultValueObject.
-
-
-        :param probe_id: The probe_id of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: int
+        :return: The data of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :rtype: list[CompositeSequenceValueObject]
         """
+        return self._data
 
-        self._probe_id = probe_id
-
-    @property
-    def probe_name(self):
-        """Gets the probe_name of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+    @data.setter
+    def data(self, data):
+        """Sets the data of this PaginatedResponseDataObjectCompositeSequenceValueObject.
 
 
-        :return: The probe_name of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: str
+        :param data: The data of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :type: list[CompositeSequenceValueObject]
         """
-        return self._probe_name
-
-    @probe_name.setter
-    def probe_name(self, probe_name):
-        """Sets the probe_name of this DifferentialExpressionAnalysisResultValueObject.
 
-
-        :param probe_name: The probe_name of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._probe_name = probe_name
+        self._data = data
 
     @property
-    def genes(self):
-        """Gets the genes of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+    def group_by(self):
+        """Gets the group_by of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
-        :return: The genes of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: list[GeneValueObject]
+        :return: The group_by of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :rtype: list[str]
         """
-        return self._genes
+        return self._group_by
 
-    @genes.setter
-    def genes(self, genes):
-        """Sets the genes of this DifferentialExpressionAnalysisResultValueObject.
+    @group_by.setter
+    def group_by(self, group_by):
+        """Sets the group_by of this PaginatedResponseDataObjectCompositeSequenceValueObject.
 
 
-        :param genes: The genes of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: list[GeneValueObject]
+        :param group_by: The group_by of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :type: list[str]
         """
 
-        self._genes = genes
+        self._group_by = group_by
 
     @property
-    def corrected_pvalue(self):
-        """Gets the corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+    def sort(self):
+        """Gets the sort of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
-        :return: The corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: float
+        :return: The sort of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :rtype: SortValueObject
         """
-        return self._corrected_pvalue
+        return self._sort
 
-    @corrected_pvalue.setter
-    def corrected_pvalue(self, corrected_pvalue):
-        """Sets the corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.
+    @sort.setter
+    def sort(self, sort):
+        """Sets the sort of this PaginatedResponseDataObjectCompositeSequenceValueObject.
 
 
-        :param corrected_pvalue: The corrected_pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: float
+        :param sort: The sort of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :type: SortValueObject
         """
 
-        self._corrected_pvalue = corrected_pvalue
+        self._sort = sort
 
     @property
-    def rank(self):
-        """Gets the rank of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+    def offset(self):
+        """Gets the offset of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
-        :return: The rank of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: float
+        :return: The offset of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._rank
+        return self._offset
 
-    @rank.setter
-    def rank(self, rank):
-        """Sets the rank of this DifferentialExpressionAnalysisResultValueObject.
+    @offset.setter
+    def offset(self, offset):
+        """Sets the offset of this PaginatedResponseDataObjectCompositeSequenceValueObject.
 
 
-        :param rank: The rank of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: float
+        :param offset: The offset of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._rank = rank
+        self._offset = offset
 
     @property
-    def contrasts(self):
-        """Gets the contrasts of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+    def limit(self):
+        """Gets the limit of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
-        :return: The contrasts of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: list[ContrastResultValueObject]
+        :return: The limit of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._contrasts
+        return self._limit
 
-    @contrasts.setter
-    def contrasts(self, contrasts):
-        """Sets the contrasts of this DifferentialExpressionAnalysisResultValueObject.
+    @limit.setter
+    def limit(self, limit):
+        """Sets the limit of this PaginatedResponseDataObjectCompositeSequenceValueObject.
 
 
-        :param contrasts: The contrasts of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: list[ContrastResultValueObject]
+        :param limit: The limit of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._contrasts = contrasts
+        self._limit = limit
 
     @property
-    def pvalue(self):
-        """Gets the pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
+    def total_elements(self):
+        """Gets the total_elements of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
-        :return: The pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :rtype: float
+        :return: The total_elements of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._pvalue
+        return self._total_elements
 
-    @pvalue.setter
-    def pvalue(self, pvalue):
-        """Sets the pvalue of this DifferentialExpressionAnalysisResultValueObject.
+    @total_elements.setter
+    def total_elements(self, total_elements):
+        """Sets the total_elements of this PaginatedResponseDataObjectCompositeSequenceValueObject.
 
 
-        :param pvalue: The pvalue of this DifferentialExpressionAnalysisResultValueObject.  # noqa: E501
-        :type: float
+        :param total_elements: The total_elements of this PaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._pvalue = pvalue
+        self._total_elements = total_elements
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -262,15 +210,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(DifferentialExpressionAnalysisResultValueObject, dict):
+        if issubclass(PaginatedResponseDataObjectCompositeSequenceValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -278,15 +226,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, DifferentialExpressionAnalysisResultValueObject):
+        if not isinstance(other, PaginatedResponseDataObjectCompositeSequenceValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/differential_expression_analysis_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/differential_expression_analysis_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/experiment_expression_levels_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/experiment_expression_levels_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/experimental_factor_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/experimental_factor_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/expression_experiment_set_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/expression_experiment_set_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/expression_experiment_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/expression_experiment_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/expression_experiment_with_search_result_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/expression_experiment_with_search_result_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/external_database_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/external_database_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/factor_value_basic_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/factor_value_basic_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/factor_value_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/factor_value_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filter_arg_array_design.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filter_arg_array_design.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filter_arg_expression_analysis_result_set.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filter_arg_expression_analysis_result_set.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filter_arg_expression_experiment.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filter_arg_expression_experiment.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_array_design_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_array_design_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -25,56 +25,56 @@
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[ArrayDesignValueObject]',
-        'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
         'offset': 'int',
         'limit': 'int',
-        'total_elements': 'int'
+        'total_elements': 'int',
+        'filter': 'str'
     }
 
     attribute_map = {
         'data': 'data',
-        'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
         'offset': 'offset',
         'limit': 'limit',
-        'total_elements': 'totalElements'
+        'total_elements': 'totalElements',
+        'filter': 'filter'
     }
 
-    def __init__(self, data=None, filter=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None, filter=None):  # noqa: E501
         """FilteredAndPaginatedResponseDataObjectArrayDesignValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._filter = None
         self._group_by = None
         self._sort = None
         self._offset = None
         self._limit = None
         self._total_elements = None
+        self._filter = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if filter is not None:
-            self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if offset is not None:
             self.offset = offset
         if limit is not None:
             self.limit = limit
         if total_elements is not None:
             self.total_elements = total_elements
+        if filter is not None:
+            self.filter = filter
 
     @property
     def data(self):
         """Gets the data of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
 
 
         :return: The data of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
@@ -90,35 +90,14 @@
         :param data: The data of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
         :type: list[ArrayDesignValueObject]
         """
 
         self._data = data
 
     @property
-    def filter(self):
-        """Gets the filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
-
-
-        :return: The filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._filter
-
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.
-
-
-        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._filter = filter
-
-    @property
     def group_by(self):
         """Gets the group_by of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
 
 
         :return: The group_by of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
         :rtype: list[str]
         """
@@ -215,14 +194,35 @@
 
         :param total_elements: The total_elements of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
         :type: int
         """
 
         self._total_elements = total_elements
 
+    @property
+    def filter(self):
+        """Gets the filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
+
+
+        :return: The filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        """Sets the filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.
+
+
+        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectArrayDesignValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._filter = filter
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_composite_sequence_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_composite_sequence_value_object.py`

 * *Files 2% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -25,56 +25,56 @@
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[CompositeSequenceValueObject]',
-        'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
         'offset': 'int',
         'limit': 'int',
-        'total_elements': 'int'
+        'total_elements': 'int',
+        'filter': 'str'
     }
 
     attribute_map = {
         'data': 'data',
-        'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
         'offset': 'offset',
         'limit': 'limit',
-        'total_elements': 'totalElements'
+        'total_elements': 'totalElements',
+        'filter': 'filter'
     }
 
-    def __init__(self, data=None, filter=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None, filter=None):  # noqa: E501
         """FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._filter = None
         self._group_by = None
         self._sort = None
         self._offset = None
         self._limit = None
         self._total_elements = None
+        self._filter = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if filter is not None:
-            self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if offset is not None:
             self.offset = offset
         if limit is not None:
             self.limit = limit
         if total_elements is not None:
             self.total_elements = total_elements
+        if filter is not None:
+            self.filter = filter
 
     @property
     def data(self):
         """Gets the data of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
         :return: The data of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
@@ -90,35 +90,14 @@
         :param data: The data of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
         :type: list[CompositeSequenceValueObject]
         """
 
         self._data = data
 
     @property
-    def filter(self):
-        """Gets the filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
-
-
-        :return: The filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._filter
-
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.
-
-
-        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._filter = filter
-
-    @property
     def group_by(self):
         """Gets the group_by of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
 
 
         :return: The group_by of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
         :rtype: list[str]
         """
@@ -215,14 +194,35 @@
 
         :param total_elements: The total_elements of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
         :type: int
         """
 
         self._total_elements = total_elements
 
+    @property
+    def filter(self):
+        """Gets the filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+
+
+        :return: The filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        """Sets the filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.
+
+
+        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectCompositeSequenceValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._filter = filter
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_differential_expression_analysis_result_set_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_differential_expression_analysis_result_set_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -25,56 +25,56 @@
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[DifferentialExpressionAnalysisResultSetValueObject]',
-        'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
         'offset': 'int',
         'limit': 'int',
-        'total_elements': 'int'
+        'total_elements': 'int',
+        'filter': 'str'
     }
 
     attribute_map = {
         'data': 'data',
-        'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
         'offset': 'offset',
         'limit': 'limit',
-        'total_elements': 'totalElements'
+        'total_elements': 'totalElements',
+        'filter': 'filter'
     }
 
-    def __init__(self, data=None, filter=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None, filter=None):  # noqa: E501
         """FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._filter = None
         self._group_by = None
         self._sort = None
         self._offset = None
         self._limit = None
         self._total_elements = None
+        self._filter = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if filter is not None:
-            self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if offset is not None:
             self.offset = offset
         if limit is not None:
             self.limit = limit
         if total_elements is not None:
             self.total_elements = total_elements
+        if filter is not None:
+            self.filter = filter
 
     @property
     def data(self):
         """Gets the data of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
 
 
         :return: The data of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
@@ -90,35 +90,14 @@
         :param data: The data of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
         :type: list[DifferentialExpressionAnalysisResultSetValueObject]
         """
 
         self._data = data
 
     @property
-    def filter(self):
-        """Gets the filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-
-
-        :return: The filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._filter
-
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.
-
-
-        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._filter = filter
-
-    @property
     def group_by(self):
         """Gets the group_by of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
 
 
         :return: The group_by of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
         :rtype: list[str]
         """
@@ -215,14 +194,35 @@
 
         :param total_elements: The total_elements of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
         :type: int
         """
 
         self._total_elements = total_elements
 
+    @property
+    def filter(self):
+        """Gets the filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+
+
+        :return: The filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        """Sets the filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.
+
+
+        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._filter = filter
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_expression_experiment_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_expression_experiment_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -25,56 +25,56 @@
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[ExpressionExperimentValueObject]',
-        'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
         'offset': 'int',
         'limit': 'int',
-        'total_elements': 'int'
+        'total_elements': 'int',
+        'filter': 'str'
     }
 
     attribute_map = {
         'data': 'data',
-        'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
         'offset': 'offset',
         'limit': 'limit',
-        'total_elements': 'totalElements'
+        'total_elements': 'totalElements',
+        'filter': 'filter'
     }
 
-    def __init__(self, data=None, filter=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None, filter=None):  # noqa: E501
         """FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._filter = None
         self._group_by = None
         self._sort = None
         self._offset = None
         self._limit = None
         self._total_elements = None
+        self._filter = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if filter is not None:
-            self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if offset is not None:
             self.offset = offset
         if limit is not None:
             self.limit = limit
         if total_elements is not None:
             self.total_elements = total_elements
+        if filter is not None:
+            self.filter = filter
 
     @property
     def data(self):
         """Gets the data of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
 
 
         :return: The data of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
@@ -90,35 +90,14 @@
         :param data: The data of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
         :type: list[ExpressionExperimentValueObject]
         """
 
         self._data = data
 
     @property
-    def filter(self):
-        """Gets the filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-
-
-        :return: The filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._filter
-
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.
-
-
-        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._filter = filter
-
-    @property
     def group_by(self):
         """Gets the group_by of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
 
 
         :return: The group_by of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
         :rtype: list[str]
         """
@@ -215,14 +194,35 @@
 
         :param total_elements: The total_elements of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
         :type: int
         """
 
         self._total_elements = total_elements
 
+    @property
+    def filter(self):
+        """Gets the filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+
+
+        :return: The filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        """Sets the filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.
+
+
+        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._filter = filter
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/filtered_and_paginated_response_data_object_gene_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/filtered_and_paginated_response_data_object_gene_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -25,56 +25,56 @@
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[GeneValueObject]',
-        'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
         'offset': 'int',
         'limit': 'int',
-        'total_elements': 'int'
+        'total_elements': 'int',
+        'filter': 'str'
     }
 
     attribute_map = {
         'data': 'data',
-        'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
         'offset': 'offset',
         'limit': 'limit',
-        'total_elements': 'totalElements'
+        'total_elements': 'totalElements',
+        'filter': 'filter'
     }
 
-    def __init__(self, data=None, filter=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None, filter=None):  # noqa: E501
         """FilteredAndPaginatedResponseDataObjectGeneValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._filter = None
         self._group_by = None
         self._sort = None
         self._offset = None
         self._limit = None
         self._total_elements = None
+        self._filter = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if filter is not None:
-            self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if offset is not None:
             self.offset = offset
         if limit is not None:
             self.limit = limit
         if total_elements is not None:
             self.total_elements = total_elements
+        if filter is not None:
+            self.filter = filter
 
     @property
     def data(self):
         """Gets the data of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
 
 
         :return: The data of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
@@ -90,35 +90,14 @@
         :param data: The data of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
         :type: list[GeneValueObject]
         """
 
         self._data = data
 
     @property
-    def filter(self):
-        """Gets the filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
-
-
-        :return: The filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._filter
-
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.
-
-
-        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._filter = filter
-
-    @property
     def group_by(self):
         """Gets the group_by of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
 
 
         :return: The group_by of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
         :rtype: list[str]
         """
@@ -215,14 +194,35 @@
 
         :param total_elements: The total_elements of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
         :type: int
         """
 
         self._total_elements = total_elements
 
+    @property
+    def filter(self):
+        """Gets the filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
+
+
+        :return: The filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        """Sets the filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.
+
+
+        :param filter: The filter of this FilteredAndPaginatedResponseDataObjectGeneValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._filter = filter
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/geeq_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/geeq_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene1.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene1.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene2.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene2.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene3.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene3.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene4.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene4.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene_arg.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene_element_expressions_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene_element_expressions_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene_ontology_term_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene_ontology_term_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene_set_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene_set_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/gene_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/gene_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/limited_response_data_object_annotation_with_usage_statistics_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/physical_location_value_object.py`

 * *Files 22% similar despite different names*

```diff
@@ -1,201 +1,227 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject(object):
+class PhysicalLocationValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'data': 'list[AnnotationWithUsageStatisticsValueObject]',
-        'query': 'str',
-        'filter': 'str',
-        'group_by': 'list[str]',
-        'sort': 'SortValueObject',
-        'limit': 'int'
+        'id': 'int',
+        'nucleotide': 'int',
+        'nucleotide_length': 'int',
+        'strand': 'str',
+        'bin': 'int',
+        'chromosome': 'str',
+        'taxon': 'TaxonValueObject'
     }
 
     attribute_map = {
-        'data': 'data',
-        'query': 'query',
-        'filter': 'filter',
-        'group_by': 'groupBy',
-        'sort': 'sort',
-        'limit': 'limit'
+        'id': 'id',
+        'nucleotide': 'nucleotide',
+        'nucleotide_length': 'nucleotideLength',
+        'strand': 'strand',
+        'bin': 'bin',
+        'chromosome': 'chromosome',
+        'taxon': 'taxon'
     }
 
-    def __init__(self, data=None, query=None, filter=None, group_by=None, sort=None, limit=None):  # noqa: E501
-        """LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject - a model defined in Swagger"""  # noqa: E501
-        self._data = None
-        self._query = None
-        self._filter = None
-        self._group_by = None
-        self._sort = None
-        self._limit = None
+    def __init__(self, id=None, nucleotide=None, nucleotide_length=None, strand=None, bin=None, chromosome=None, taxon=None):  # noqa: E501
+        """PhysicalLocationValueObject - a model defined in Swagger"""  # noqa: E501
+        self._id = None
+        self._nucleotide = None
+        self._nucleotide_length = None
+        self._strand = None
+        self._bin = None
+        self._chromosome = None
+        self._taxon = None
         self.discriminator = None
-        if data is not None:
-            self.data = data
-        if query is not None:
-            self.query = query
-        if filter is not None:
-            self.filter = filter
-        if group_by is not None:
-            self.group_by = group_by
-        if sort is not None:
-            self.sort = sort
-        if limit is not None:
-            self.limit = limit
+        if id is not None:
+            self.id = id
+        if nucleotide is not None:
+            self.nucleotide = nucleotide
+        if nucleotide_length is not None:
+            self.nucleotide_length = nucleotide_length
+        if strand is not None:
+            self.strand = strand
+        if bin is not None:
+            self.bin = bin
+        if chromosome is not None:
+            self.chromosome = chromosome
+        if taxon is not None:
+            self.taxon = taxon
 
     @property
-    def data(self):
-        """Gets the data of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+    def id(self):
+        """Gets the id of this PhysicalLocationValueObject.  # noqa: E501
 
 
-        :return: The data of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: list[AnnotationWithUsageStatisticsValueObject]
+        :return: The id of this PhysicalLocationValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._data
+        return self._id
 
-    @data.setter
-    def data(self, data):
-        """Sets the data of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.
+    @id.setter
+    def id(self, id):
+        """Sets the id of this PhysicalLocationValueObject.
 
 
-        :param data: The data of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :type: list[AnnotationWithUsageStatisticsValueObject]
+        :param id: The id of this PhysicalLocationValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._data = data
+        self._id = id
 
     @property
-    def query(self):
-        """Gets the query of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+    def nucleotide(self):
+        """Gets the nucleotide of this PhysicalLocationValueObject.  # noqa: E501
 
 
-        :return: The query of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: str
+        :return: The nucleotide of this PhysicalLocationValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._query
+        return self._nucleotide
 
-    @query.setter
-    def query(self, query):
-        """Sets the query of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.
+    @nucleotide.setter
+    def nucleotide(self, nucleotide):
+        """Sets the nucleotide of this PhysicalLocationValueObject.
 
 
-        :param query: The query of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :type: str
+        :param nucleotide: The nucleotide of this PhysicalLocationValueObject.  # noqa: E501
+        :type: int
+        """
+
+        self._nucleotide = nucleotide
+
+    @property
+    def nucleotide_length(self):
+        """Gets the nucleotide_length of this PhysicalLocationValueObject.  # noqa: E501
+
+
+        :return: The nucleotide_length of this PhysicalLocationValueObject.  # noqa: E501
+        :rtype: int
+        """
+        return self._nucleotide_length
+
+    @nucleotide_length.setter
+    def nucleotide_length(self, nucleotide_length):
+        """Sets the nucleotide_length of this PhysicalLocationValueObject.
+
+
+        :param nucleotide_length: The nucleotide_length of this PhysicalLocationValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._query = query
+        self._nucleotide_length = nucleotide_length
 
     @property
-    def filter(self):
-        """Gets the filter of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+    def strand(self):
+        """Gets the strand of this PhysicalLocationValueObject.  # noqa: E501
 
 
-        :return: The filter of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The strand of this PhysicalLocationValueObject.  # noqa: E501
         :rtype: str
         """
-        return self._filter
+        return self._strand
 
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.
+    @strand.setter
+    def strand(self, strand):
+        """Sets the strand of this PhysicalLocationValueObject.
 
 
-        :param filter: The filter of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+        :param strand: The strand of this PhysicalLocationValueObject.  # noqa: E501
         :type: str
         """
 
-        self._filter = filter
+        self._strand = strand
 
     @property
-    def group_by(self):
-        """Gets the group_by of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+    def bin(self):
+        """Gets the bin of this PhysicalLocationValueObject.  # noqa: E501
 
 
-        :return: The group_by of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: list[str]
+        :return: The bin of this PhysicalLocationValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._group_by
+        return self._bin
 
-    @group_by.setter
-    def group_by(self, group_by):
-        """Sets the group_by of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.
+    @bin.setter
+    def bin(self, bin):
+        """Sets the bin of this PhysicalLocationValueObject.
 
 
-        :param group_by: The group_by of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :type: list[str]
+        :param bin: The bin of this PhysicalLocationValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._group_by = group_by
+        self._bin = bin
 
     @property
-    def sort(self):
-        """Gets the sort of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+    def chromosome(self):
+        """Gets the chromosome of this PhysicalLocationValueObject.  # noqa: E501
 
 
-        :return: The sort of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: SortValueObject
+        :return: The chromosome of this PhysicalLocationValueObject.  # noqa: E501
+        :rtype: str
         """
-        return self._sort
+        return self._chromosome
 
-    @sort.setter
-    def sort(self, sort):
-        """Sets the sort of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.
+    @chromosome.setter
+    def chromosome(self, chromosome):
+        """Sets the chromosome of this PhysicalLocationValueObject.
 
 
-        :param sort: The sort of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :type: SortValueObject
+        :param chromosome: The chromosome of this PhysicalLocationValueObject.  # noqa: E501
+        :type: str
         """
 
-        self._sort = sort
+        self._chromosome = chromosome
 
     @property
-    def limit(self):
-        """Gets the limit of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
+    def taxon(self):
+        """Gets the taxon of this PhysicalLocationValueObject.  # noqa: E501
 
 
-        :return: The limit of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: int
+        :return: The taxon of this PhysicalLocationValueObject.  # noqa: E501
+        :rtype: TaxonValueObject
         """
-        return self._limit
+        return self._taxon
 
-    @limit.setter
-    def limit(self, limit):
-        """Sets the limit of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.
+    @taxon.setter
+    def taxon(self, taxon):
+        """Sets the taxon of this PhysicalLocationValueObject.
 
 
-        :param limit: The limit of this LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject.  # noqa: E501
-        :type: int
+        :param taxon: The taxon of this PhysicalLocationValueObject.  # noqa: E501
+        :type: TaxonValueObject
         """
 
-        self._limit = limit
+        self._taxon = taxon
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -210,15 +236,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject, dict):
+        if issubclass(PhysicalLocationValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -226,15 +252,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, LimitedResponseDataObjectAnnotationWithUsageStatisticsValueObject):
+        if not isinstance(other, PhysicalLocationValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/limited_response_data_object_array_design_with_usage_statistics_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_and_limited_response_data_object_array_design_with_usage_statistics_value_object.py`

 * *Files 21% similar despite different names*

```diff
@@ -1,202 +1,228 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject(object):
+class QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[ArrayDesignWithUsageStatisticsValueObject]',
-        'query': 'str',
         'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
-        'limit': 'int'
+        'limit': 'int',
+        'query': 'str',
+        'inferred_terms': 'list[CharacteristicValueObject]'
     }
 
     attribute_map = {
         'data': 'data',
-        'query': 'query',
         'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
-        'limit': 'limit'
+        'limit': 'limit',
+        'query': 'query',
+        'inferred_terms': 'inferredTerms'
     }
 
-    def __init__(self, data=None, query=None, filter=None, group_by=None, sort=None, limit=None):  # noqa: E501
-        """LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject - a model defined in Swagger"""  # noqa: E501
+    def __init__(self, data=None, filter=None, group_by=None, sort=None, limit=None, query=None, inferred_terms=None):  # noqa: E501
+        """QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._query = None
         self._filter = None
         self._group_by = None
         self._sort = None
         self._limit = None
+        self._query = None
+        self._inferred_terms = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if query is not None:
-            self.query = query
         if filter is not None:
             self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if limit is not None:
             self.limit = limit
+        if query is not None:
+            self.query = query
+        if inferred_terms is not None:
+            self.inferred_terms = inferred_terms
 
     @property
     def data(self):
-        """Gets the data of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the data of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The data of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The data of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: list[ArrayDesignWithUsageStatisticsValueObject]
         """
         return self._data
 
     @data.setter
     def data(self, data):
-        """Sets the data of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
+        """Sets the data of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
 
 
-        :param data: The data of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :param data: The data of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :type: list[ArrayDesignWithUsageStatisticsValueObject]
         """
 
         self._data = data
 
     @property
-    def query(self):
-        """Gets the query of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
-
-
-        :return: The query of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._query
-
-    @query.setter
-    def query(self, query):
-        """Sets the query of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
-
-
-        :param query: The query of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._query = query
-
-    @property
     def filter(self):
-        """Gets the filter of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the filter of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The filter of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The filter of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: str
         """
         return self._filter
 
     @filter.setter
     def filter(self, filter):
-        """Sets the filter of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
+        """Sets the filter of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
 
 
-        :param filter: The filter of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :param filter: The filter of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :type: str
         """
 
         self._filter = filter
 
     @property
     def group_by(self):
-        """Gets the group_by of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the group_by of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The group_by of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The group_by of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: list[str]
         """
         return self._group_by
 
     @group_by.setter
     def group_by(self, group_by):
-        """Sets the group_by of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
+        """Sets the group_by of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
 
 
-        :param group_by: The group_by of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :param group_by: The group_by of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :type: list[str]
         """
 
         self._group_by = group_by
 
     @property
     def sort(self):
-        """Gets the sort of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the sort of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The sort of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The sort of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: SortValueObject
         """
         return self._sort
 
     @sort.setter
     def sort(self, sort):
-        """Sets the sort of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
+        """Sets the sort of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
 
 
-        :param sort: The sort of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :param sort: The sort of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :type: SortValueObject
         """
 
         self._sort = sort
 
     @property
     def limit(self):
-        """Gets the limit of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the limit of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The limit of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The limit of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: int
         """
         return self._limit
 
     @limit.setter
     def limit(self, limit):
-        """Sets the limit of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
+        """Sets the limit of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
 
 
-        :param limit: The limit of this LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :param limit: The limit of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
         :type: int
         """
 
         self._limit = limit
 
+    @property
+    def query(self):
+        """Gets the query of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+
+
+        :return: The query of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._query
+
+    @query.setter
+    def query(self, query):
+        """Sets the query of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
+
+
+        :param query: The query of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._query = query
+
+    @property
+    def inferred_terms(self):
+        """Gets the inferred_terms of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+
+
+        :return: The inferred_terms of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: list[CharacteristicValueObject]
+        """
+        return self._inferred_terms
+
+    @inferred_terms.setter
+    def inferred_terms(self, inferred_terms):
+        """Sets the inferred_terms of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.
+
+
+        :param inferred_terms: The inferred_terms of this QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject.  # noqa: E501
+        :type: list[CharacteristicValueObject]
+        """
+
+        self._inferred_terms = inferred_terms
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
@@ -210,15 +236,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject, dict):
+        if issubclass(QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -226,15 +252,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, LimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject):
+        if not isinstance(other, QueriedAndFilteredAndInferredAndLimitedResponseDataObjectArrayDesignWithUsageStatisticsValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/measurement_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/measurement_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/one_of_search_result_value_object_object_result_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/one_of_search_result_value_object_object_result_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/ontology_term_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/ontology_term_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/paginated_response_data_object_expression_experiment_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/paginated_response_data_object_expression_experiment_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/physical_location_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/search_settings_value_object.py`

 * *Files 23% similar despite different names*

```diff
@@ -1,227 +1,175 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class PhysicalLocationValueObject(object):
+class SearchSettingsValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'id': 'int',
-        'nucleotide': 'int',
-        'nucleotide_length': 'int',
-        'strand': 'str',
-        'bin': 'int',
-        'chromosome': 'str',
-        'taxon': 'TaxonValueObject'
+        'query': 'str',
+        'result_types': 'list[SearchResultType]',
+        'taxon': 'TaxonValueObject',
+        'platform': 'ArrayDesignValueObject',
+        'max_results': 'int'
     }
 
     attribute_map = {
-        'id': 'id',
-        'nucleotide': 'nucleotide',
-        'nucleotide_length': 'nucleotideLength',
-        'strand': 'strand',
-        'bin': 'bin',
-        'chromosome': 'chromosome',
-        'taxon': 'taxon'
+        'query': 'query',
+        'result_types': 'resultTypes',
+        'taxon': 'taxon',
+        'platform': 'platform',
+        'max_results': 'maxResults'
     }
 
-    def __init__(self, id=None, nucleotide=None, nucleotide_length=None, strand=None, bin=None, chromosome=None, taxon=None):  # noqa: E501
-        """PhysicalLocationValueObject - a model defined in Swagger"""  # noqa: E501
-        self._id = None
-        self._nucleotide = None
-        self._nucleotide_length = None
-        self._strand = None
-        self._bin = None
-        self._chromosome = None
+    def __init__(self, query=None, result_types=None, taxon=None, platform=None, max_results=None):  # noqa: E501
+        """SearchSettingsValueObject - a model defined in Swagger"""  # noqa: E501
+        self._query = None
+        self._result_types = None
         self._taxon = None
+        self._platform = None
+        self._max_results = None
         self.discriminator = None
-        if id is not None:
-            self.id = id
-        if nucleotide is not None:
-            self.nucleotide = nucleotide
-        if nucleotide_length is not None:
-            self.nucleotide_length = nucleotide_length
-        if strand is not None:
-            self.strand = strand
-        if bin is not None:
-            self.bin = bin
-        if chromosome is not None:
-            self.chromosome = chromosome
+        if query is not None:
+            self.query = query
+        if result_types is not None:
+            self.result_types = result_types
         if taxon is not None:
             self.taxon = taxon
+        if platform is not None:
+            self.platform = platform
+        if max_results is not None:
+            self.max_results = max_results
 
     @property
-    def id(self):
-        """Gets the id of this PhysicalLocationValueObject.  # noqa: E501
+    def query(self):
+        """Gets the query of this SearchSettingsValueObject.  # noqa: E501
 
 
-        :return: The id of this PhysicalLocationValueObject.  # noqa: E501
-        :rtype: int
+        :return: The query of this SearchSettingsValueObject.  # noqa: E501
+        :rtype: str
         """
-        return self._id
+        return self._query
 
-    @id.setter
-    def id(self, id):
-        """Sets the id of this PhysicalLocationValueObject.
+    @query.setter
+    def query(self, query):
+        """Sets the query of this SearchSettingsValueObject.
 
 
-        :param id: The id of this PhysicalLocationValueObject.  # noqa: E501
-        :type: int
+        :param query: The query of this SearchSettingsValueObject.  # noqa: E501
+        :type: str
         """
 
-        self._id = id
+        self._query = query
 
     @property
-    def nucleotide(self):
-        """Gets the nucleotide of this PhysicalLocationValueObject.  # noqa: E501
+    def result_types(self):
+        """Gets the result_types of this SearchSettingsValueObject.  # noqa: E501
 
 
-        :return: The nucleotide of this PhysicalLocationValueObject.  # noqa: E501
-        :rtype: int
+        :return: The result_types of this SearchSettingsValueObject.  # noqa: E501
+        :rtype: list[SearchResultType]
         """
-        return self._nucleotide
+        return self._result_types
 
-    @nucleotide.setter
-    def nucleotide(self, nucleotide):
-        """Sets the nucleotide of this PhysicalLocationValueObject.
+    @result_types.setter
+    def result_types(self, result_types):
+        """Sets the result_types of this SearchSettingsValueObject.
 
 
-        :param nucleotide: The nucleotide of this PhysicalLocationValueObject.  # noqa: E501
-        :type: int
+        :param result_types: The result_types of this SearchSettingsValueObject.  # noqa: E501
+        :type: list[SearchResultType]
         """
 
-        self._nucleotide = nucleotide
+        self._result_types = result_types
 
     @property
-    def nucleotide_length(self):
-        """Gets the nucleotide_length of this PhysicalLocationValueObject.  # noqa: E501
+    def taxon(self):
+        """Gets the taxon of this SearchSettingsValueObject.  # noqa: E501
 
 
-        :return: The nucleotide_length of this PhysicalLocationValueObject.  # noqa: E501
-        :rtype: int
+        :return: The taxon of this SearchSettingsValueObject.  # noqa: E501
+        :rtype: TaxonValueObject
         """
-        return self._nucleotide_length
+        return self._taxon
 
-    @nucleotide_length.setter
-    def nucleotide_length(self, nucleotide_length):
-        """Sets the nucleotide_length of this PhysicalLocationValueObject.
+    @taxon.setter
+    def taxon(self, taxon):
+        """Sets the taxon of this SearchSettingsValueObject.
 
 
-        :param nucleotide_length: The nucleotide_length of this PhysicalLocationValueObject.  # noqa: E501
-        :type: int
+        :param taxon: The taxon of this SearchSettingsValueObject.  # noqa: E501
+        :type: TaxonValueObject
         """
 
-        self._nucleotide_length = nucleotide_length
+        self._taxon = taxon
 
     @property
-    def strand(self):
-        """Gets the strand of this PhysicalLocationValueObject.  # noqa: E501
+    def platform(self):
+        """Gets the platform of this SearchSettingsValueObject.  # noqa: E501
 
 
-        :return: The strand of this PhysicalLocationValueObject.  # noqa: E501
-        :rtype: str
+        :return: The platform of this SearchSettingsValueObject.  # noqa: E501
+        :rtype: ArrayDesignValueObject
         """
-        return self._strand
+        return self._platform
 
-    @strand.setter
-    def strand(self, strand):
-        """Sets the strand of this PhysicalLocationValueObject.
+    @platform.setter
+    def platform(self, platform):
+        """Sets the platform of this SearchSettingsValueObject.
 
 
-        :param strand: The strand of this PhysicalLocationValueObject.  # noqa: E501
-        :type: str
+        :param platform: The platform of this SearchSettingsValueObject.  # noqa: E501
+        :type: ArrayDesignValueObject
         """
 
-        self._strand = strand
+        self._platform = platform
 
     @property
-    def bin(self):
-        """Gets the bin of this PhysicalLocationValueObject.  # noqa: E501
+    def max_results(self):
+        """Gets the max_results of this SearchSettingsValueObject.  # noqa: E501
 
 
-        :return: The bin of this PhysicalLocationValueObject.  # noqa: E501
+        :return: The max_results of this SearchSettingsValueObject.  # noqa: E501
         :rtype: int
         """
-        return self._bin
+        return self._max_results
 
-    @bin.setter
-    def bin(self, bin):
-        """Sets the bin of this PhysicalLocationValueObject.
+    @max_results.setter
+    def max_results(self, max_results):
+        """Sets the max_results of this SearchSettingsValueObject.
 
 
-        :param bin: The bin of this PhysicalLocationValueObject.  # noqa: E501
+        :param max_results: The max_results of this SearchSettingsValueObject.  # noqa: E501
         :type: int
         """
 
-        self._bin = bin
-
-    @property
-    def chromosome(self):
-        """Gets the chromosome of this PhysicalLocationValueObject.  # noqa: E501
-
-
-        :return: The chromosome of this PhysicalLocationValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._chromosome
-
-    @chromosome.setter
-    def chromosome(self, chromosome):
-        """Sets the chromosome of this PhysicalLocationValueObject.
-
-
-        :param chromosome: The chromosome of this PhysicalLocationValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._chromosome = chromosome
-
-    @property
-    def taxon(self):
-        """Gets the taxon of this PhysicalLocationValueObject.  # noqa: E501
-
-
-        :return: The taxon of this PhysicalLocationValueObject.  # noqa: E501
-        :rtype: TaxonValueObject
-        """
-        return self._taxon
-
-    @taxon.setter
-    def taxon(self, taxon):
-        """Sets the taxon of this PhysicalLocationValueObject.
-
-
-        :param taxon: The taxon of this PhysicalLocationValueObject.  # noqa: E501
-        :type: TaxonValueObject
-        """
-
-        self._taxon = taxon
+        self._max_results = max_results
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -236,15 +184,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(PhysicalLocationValueObject, dict):
+        if issubclass(SearchSettingsValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -252,15 +200,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, PhysicalLocationValueObject):
+        if not isinstance(other, SearchSettingsValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/platform.py` & `gemmapy-1.0.1/gemmapy/sdk/models/platform.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/platform1.py` & `gemmapy-1.0.1/gemmapy/sdk/models/platform1.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/platform2.py` & `gemmapy-1.0.1/gemmapy/sdk/models/platform2.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/platform3.py` & `gemmapy-1.0.1/gemmapy/sdk/models/platform_arg.py`

 * *Files 2% similar despite different names*

```diff
@@ -1,25 +1,25 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class Platform3(object):
+class PlatformArg(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
@@ -30,15 +30,15 @@
     swagger_types = {
     }
 
     attribute_map = {
     }
 
     def __init__(self):  # noqa: E501
-        """Platform3 - a model defined in Swagger"""  # noqa: E501
+        """PlatformArg - a model defined in Swagger"""  # noqa: E501
         self.discriminator = None
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
@@ -54,15 +54,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(Platform3, dict):
+        if issubclass(PlatformArg, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -70,15 +70,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, Platform3):
+        if not isinstance(other, PlatformArg):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/platform4.py` & `gemmapy-1.0.1/gemmapy/sdk/models/platform5.py`

 * *Files 2% similar despite different names*

```diff
@@ -1,25 +1,25 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class Platform4(object):
+class Platform5(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
@@ -30,15 +30,15 @@
     swagger_types = {
     }
 
     attribute_map = {
     }
 
     def __init__(self):  # noqa: E501
-        """Platform4 - a model defined in Swagger"""  # noqa: E501
+        """Platform5 - a model defined in Swagger"""  # noqa: E501
         self.discriminator = None
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
@@ -54,15 +54,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(Platform4, dict):
+        if issubclass(Platform5, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -70,15 +70,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, Platform4):
+        if not isinstance(other, Platform5):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/platform5.py` & `gemmapy-1.0.1/gemmapy/sdk/models/platform3.py`

 * *Files 3% similar despite different names*

```diff
@@ -1,25 +1,25 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class Platform5(object):
+class Platform3(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
@@ -30,15 +30,15 @@
     swagger_types = {
     }
 
     attribute_map = {
     }
 
     def __init__(self):  # noqa: E501
-        """Platform5 - a model defined in Swagger"""  # noqa: E501
+        """Platform3 - a model defined in Swagger"""  # noqa: E501
         self.discriminator = None
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
@@ -54,15 +54,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(Platform5, dict):
+        if issubclass(Platform3, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -70,15 +70,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, Platform5):
+        if not isinstance(other, Platform3):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/platform_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/platform4.py`

 * *Files 3% similar despite different names*

```diff
@@ -1,25 +1,25 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class PlatformArg(object):
+class Platform4(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
@@ -30,15 +30,15 @@
     swagger_types = {
     }
 
     attribute_map = {
     }
 
     def __init__(self):  # noqa: E501
-        """PlatformArg - a model defined in Swagger"""  # noqa: E501
+        """Platform4 - a model defined in Swagger"""  # noqa: E501
         self.discriminator = None
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
@@ -54,15 +54,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(PlatformArg, dict):
+        if issubclass(Platform4, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -70,15 +70,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, PlatformArg):
+        if not isinstance(other, Platform4):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/probe.py` & `gemmapy-1.0.1/gemmapy/sdk/models/probe.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/quantitation_type.py` & `gemmapy-1.0.1/gemmapy/sdk/models/quantitation_type.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/quantitation_type_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/quantitation_type_arg.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/quantitation_type_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/quantitation_type_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
@@ -25,61 +25,61 @@
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[ExpressionExperimentValueObject]',
-        'query': 'str',
-        'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
         'offset': 'int',
         'limit': 'int',
-        'total_elements': 'int'
+        'total_elements': 'int',
+        'filter': 'str',
+        'query': 'str'
     }
 
     attribute_map = {
         'data': 'data',
-        'query': 'query',
-        'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
         'offset': 'offset',
         'limit': 'limit',
-        'total_elements': 'totalElements'
+        'total_elements': 'totalElements',
+        'filter': 'filter',
+        'query': 'query'
     }
 
-    def __init__(self, data=None, query=None, filter=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None, filter=None, query=None):  # noqa: E501
         """QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._query = None
-        self._filter = None
         self._group_by = None
         self._sort = None
         self._offset = None
         self._limit = None
         self._total_elements = None
+        self._filter = None
+        self._query = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if query is not None:
-            self.query = query
-        if filter is not None:
-            self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if offset is not None:
             self.offset = offset
         if limit is not None:
             self.limit = limit
         if total_elements is not None:
             self.total_elements = total_elements
+        if filter is not None:
+            self.filter = filter
+        if query is not None:
+            self.query = query
 
     @property
     def data(self):
         """Gets the data of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
 
 
         :return: The data of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
@@ -95,56 +95,14 @@
         :param data: The data of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
         :type: list[ExpressionExperimentValueObject]
         """
 
         self._data = data
 
     @property
-    def query(self):
-        """Gets the query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-
-
-        :return: The query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._query
-
-    @query.setter
-    def query(self, query):
-        """Sets the query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.
-
-
-        :param query: The query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._query = query
-
-    @property
-    def filter(self):
-        """Gets the filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-
-
-        :return: The filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._filter
-
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.
-
-
-        :param filter: The filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._filter = filter
-
-    @property
     def group_by(self):
         """Gets the group_by of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
 
 
         :return: The group_by of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
         :rtype: list[str]
         """
@@ -241,14 +199,56 @@
 
         :param total_elements: The total_elements of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
         :type: int
         """
 
         self._total_elements = total_elements
 
+    @property
+    def filter(self):
+        """Gets the filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+
+
+        :return: The filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        """Sets the filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.
+
+
+        :param filter: The filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._filter = filter
+
+    @property
+    def query(self):
+        """Gets the query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+
+
+        :return: The query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._query
+
+    @query.setter
+    def query(self, query):
+        """Sets the query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.
+
+
+        :param query: The query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._query = query
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_with_search_result_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_and_paginated_response_data_object_expression_experiment_with_search_result_value_object.py`

 * *Files 15% similar despite different names*

```diff
@@ -1,254 +1,280 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject(object):
+class QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[ExpressionExperimentWithSearchResultValueObject]',
-        'query': 'str',
-        'filter': 'str',
         'group_by': 'list[str]',
         'sort': 'SortValueObject',
         'offset': 'int',
         'limit': 'int',
-        'total_elements': 'int'
+        'total_elements': 'int',
+        'filter': 'str',
+        'query': 'str',
+        'inferred_terms': 'list[CharacteristicValueObject]'
     }
 
     attribute_map = {
         'data': 'data',
-        'query': 'query',
-        'filter': 'filter',
         'group_by': 'groupBy',
         'sort': 'sort',
         'offset': 'offset',
         'limit': 'limit',
-        'total_elements': 'totalElements'
+        'total_elements': 'totalElements',
+        'filter': 'filter',
+        'query': 'query',
+        'inferred_terms': 'inferredTerms'
     }
 
-    def __init__(self, data=None, query=None, filter=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None):  # noqa: E501
-        """QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject - a model defined in Swagger"""  # noqa: E501
+    def __init__(self, data=None, group_by=None, sort=None, offset=None, limit=None, total_elements=None, filter=None, query=None, inferred_terms=None):  # noqa: E501
+        """QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._query = None
-        self._filter = None
         self._group_by = None
         self._sort = None
         self._offset = None
         self._limit = None
         self._total_elements = None
+        self._filter = None
+        self._query = None
+        self._inferred_terms = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if query is not None:
-            self.query = query
-        if filter is not None:
-            self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
         if offset is not None:
             self.offset = offset
         if limit is not None:
             self.limit = limit
         if total_elements is not None:
             self.total_elements = total_elements
+        if filter is not None:
+            self.filter = filter
+        if query is not None:
+            self.query = query
+        if inferred_terms is not None:
+            self.inferred_terms = inferred_terms
 
     @property
     def data(self):
-        """Gets the data of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        """Gets the data of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
 
 
-        :return: The data of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :return: The data of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :rtype: list[ExpressionExperimentWithSearchResultValueObject]
         """
         return self._data
 
     @data.setter
     def data(self, data):
-        """Sets the data of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+        """Sets the data of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
 
 
-        :param data: The data of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :param data: The data of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :type: list[ExpressionExperimentWithSearchResultValueObject]
         """
 
         self._data = data
 
     @property
-    def query(self):
-        """Gets the query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
-
-
-        :return: The query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._query
-
-    @query.setter
-    def query(self, query):
-        """Sets the query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
-
-
-        :param query: The query of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._query = query
-
-    @property
-    def filter(self):
-        """Gets the filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
-
-
-        :return: The filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._filter
-
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
-
-
-        :param filter: The filter of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._filter = filter
-
-    @property
     def group_by(self):
-        """Gets the group_by of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        """Gets the group_by of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
 
 
-        :return: The group_by of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :return: The group_by of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :rtype: list[str]
         """
         return self._group_by
 
     @group_by.setter
     def group_by(self, group_by):
-        """Sets the group_by of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+        """Sets the group_by of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
 
 
-        :param group_by: The group_by of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :param group_by: The group_by of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :type: list[str]
         """
 
         self._group_by = group_by
 
     @property
     def sort(self):
-        """Gets the sort of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        """Gets the sort of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
 
 
-        :return: The sort of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :return: The sort of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :rtype: SortValueObject
         """
         return self._sort
 
     @sort.setter
     def sort(self, sort):
-        """Sets the sort of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+        """Sets the sort of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
 
 
-        :param sort: The sort of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :param sort: The sort of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :type: SortValueObject
         """
 
         self._sort = sort
 
     @property
     def offset(self):
-        """Gets the offset of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        """Gets the offset of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
 
 
-        :return: The offset of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :return: The offset of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :rtype: int
         """
         return self._offset
 
     @offset.setter
     def offset(self, offset):
-        """Sets the offset of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+        """Sets the offset of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
 
 
-        :param offset: The offset of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :param offset: The offset of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :type: int
         """
 
         self._offset = offset
 
     @property
     def limit(self):
-        """Gets the limit of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        """Gets the limit of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
 
 
-        :return: The limit of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :return: The limit of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :rtype: int
         """
         return self._limit
 
     @limit.setter
     def limit(self, limit):
-        """Sets the limit of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+        """Sets the limit of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
 
 
-        :param limit: The limit of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :param limit: The limit of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :type: int
         """
 
         self._limit = limit
 
     @property
     def total_elements(self):
-        """Gets the total_elements of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        """Gets the total_elements of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
 
 
-        :return: The total_elements of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :return: The total_elements of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :rtype: int
         """
         return self._total_elements
 
     @total_elements.setter
     def total_elements(self, total_elements):
-        """Sets the total_elements of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+        """Sets the total_elements of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
 
 
-        :param total_elements: The total_elements of this QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :param total_elements: The total_elements of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
         :type: int
         """
 
         self._total_elements = total_elements
 
+    @property
+    def filter(self):
+        """Gets the filter of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+
+
+        :return: The filter of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._filter
+
+    @filter.setter
+    def filter(self, filter):
+        """Sets the filter of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+
+
+        :param filter: The filter of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._filter = filter
+
+    @property
+    def query(self):
+        """Gets the query of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+
+
+        :return: The query of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._query
+
+    @query.setter
+    def query(self, query):
+        """Sets the query of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+
+
+        :param query: The query of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._query = query
+
+    @property
+    def inferred_terms(self):
+        """Gets the inferred_terms of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+
+
+        :return: The inferred_terms of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :rtype: list[CharacteristicValueObject]
+        """
+        return self._inferred_terms
+
+    @inferred_terms.setter
+    def inferred_terms(self, inferred_terms):
+        """Sets the inferred_terms of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.
+
+
+        :param inferred_terms: The inferred_terms of this QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject.  # noqa: E501
+        :type: list[CharacteristicValueObject]
+        """
+
+        self._inferred_terms = inferred_terms
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
@@ -262,15 +288,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject, dict):
+        if issubclass(QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -278,15 +304,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, QueriedAndFilteredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject):
+        if not isinstance(other, QueriedAndFilteredAndInferredAndPaginatedResponseDataObjectExpressionExperimentWithSearchResultValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_response_data_object_category_with_usage_statistics_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon_with_usage_statistics_value_object.py`

 * *Files 27% similar despite different names*

```diff
@@ -1,175 +1,201 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject(object):
+class TaxonWithUsageStatisticsValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'data': 'list[CategoryWithUsageStatisticsValueObject]',
-        'query': 'str',
-        'filter': 'str',
-        'group_by': 'list[str]',
-        'sort': 'str'
+        'id': 'int',
+        'scientific_name': 'str',
+        'common_name': 'str',
+        'ncbi_id': 'int',
+        'external_database': 'ExternalDatabaseValueObject',
+        'number_of_expression_experiments': 'int'
     }
 
     attribute_map = {
-        'data': 'data',
-        'query': 'query',
-        'filter': 'filter',
-        'group_by': 'groupBy',
-        'sort': 'sort'
+        'id': 'id',
+        'scientific_name': 'scientificName',
+        'common_name': 'commonName',
+        'ncbi_id': 'ncbiId',
+        'external_database': 'externalDatabase',
+        'number_of_expression_experiments': 'numberOfExpressionExperiments'
     }
 
-    def __init__(self, data=None, query=None, filter=None, group_by=None, sort=None):  # noqa: E501
-        """QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject - a model defined in Swagger"""  # noqa: E501
-        self._data = None
-        self._query = None
-        self._filter = None
-        self._group_by = None
-        self._sort = None
+    def __init__(self, id=None, scientific_name=None, common_name=None, ncbi_id=None, external_database=None, number_of_expression_experiments=None):  # noqa: E501
+        """TaxonWithUsageStatisticsValueObject - a model defined in Swagger"""  # noqa: E501
+        self._id = None
+        self._scientific_name = None
+        self._common_name = None
+        self._ncbi_id = None
+        self._external_database = None
+        self._number_of_expression_experiments = None
         self.discriminator = None
-        if data is not None:
-            self.data = data
-        if query is not None:
-            self.query = query
-        if filter is not None:
-            self.filter = filter
-        if group_by is not None:
-            self.group_by = group_by
-        if sort is not None:
-            self.sort = sort
+        if id is not None:
+            self.id = id
+        if scientific_name is not None:
+            self.scientific_name = scientific_name
+        if common_name is not None:
+            self.common_name = common_name
+        if ncbi_id is not None:
+            self.ncbi_id = ncbi_id
+        if external_database is not None:
+            self.external_database = external_database
+        if number_of_expression_experiments is not None:
+            self.number_of_expression_experiments = number_of_expression_experiments
 
     @property
-    def data(self):
-        """Gets the data of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+    def id(self):
+        """Gets the id of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The data of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: list[CategoryWithUsageStatisticsValueObject]
+        :return: The id of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._data
+        return self._id
 
-    @data.setter
-    def data(self, data):
-        """Sets the data of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.
+    @id.setter
+    def id(self, id):
+        """Sets the id of this TaxonWithUsageStatisticsValueObject.
 
 
-        :param data: The data of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
-        :type: list[CategoryWithUsageStatisticsValueObject]
+        :param id: The id of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._data = data
+        self._id = id
 
     @property
-    def query(self):
-        """Gets the query of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+    def scientific_name(self):
+        """Gets the scientific_name of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The query of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The scientific_name of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: str
         """
-        return self._query
+        return self._scientific_name
 
-    @query.setter
-    def query(self, query):
-        """Sets the query of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.
+    @scientific_name.setter
+    def scientific_name(self, scientific_name):
+        """Sets the scientific_name of this TaxonWithUsageStatisticsValueObject.
 
 
-        :param query: The query of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+        :param scientific_name: The scientific_name of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
         :type: str
         """
 
-        self._query = query
+        self._scientific_name = scientific_name
 
     @property
-    def filter(self):
-        """Gets the filter of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+    def common_name(self):
+        """Gets the common_name of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The filter of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The common_name of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: str
         """
-        return self._filter
+        return self._common_name
 
-    @filter.setter
-    def filter(self, filter):
-        """Sets the filter of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.
+    @common_name.setter
+    def common_name(self, common_name):
+        """Sets the common_name of this TaxonWithUsageStatisticsValueObject.
 
 
-        :param filter: The filter of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+        :param common_name: The common_name of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
         :type: str
         """
 
-        self._filter = filter
+        self._common_name = common_name
 
     @property
-    def group_by(self):
-        """Gets the group_by of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+    def ncbi_id(self):
+        """Gets the ncbi_id of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The group_by of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: list[str]
+        :return: The ncbi_id of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: int
         """
-        return self._group_by
+        return self._ncbi_id
 
-    @group_by.setter
-    def group_by(self, group_by):
-        """Sets the group_by of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.
+    @ncbi_id.setter
+    def ncbi_id(self, ncbi_id):
+        """Sets the ncbi_id of this TaxonWithUsageStatisticsValueObject.
 
 
-        :param group_by: The group_by of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
-        :type: list[str]
+        :param ncbi_id: The ncbi_id of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._group_by = group_by
+        self._ncbi_id = ncbi_id
 
     @property
-    def sort(self):
-        """Gets the sort of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
+    def external_database(self):
+        """Gets the external_database of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The sort of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: str
+        :return: The external_database of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: ExternalDatabaseValueObject
         """
-        return self._sort
+        return self._external_database
 
-    @sort.setter
-    def sort(self, sort):
-        """Sets the sort of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.
+    @external_database.setter
+    def external_database(self, external_database):
+        """Sets the external_database of this TaxonWithUsageStatisticsValueObject.
 
 
-        :param sort: The sort of this QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject.  # noqa: E501
-        :type: str
+        :param external_database: The external_database of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :type: ExternalDatabaseValueObject
+        """
+
+        self._external_database = external_database
+
+    @property
+    def number_of_expression_experiments(self):
+        """Gets the number_of_expression_experiments of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+
+
+        :return: The number_of_expression_experiments of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: int
+        """
+        return self._number_of_expression_experiments
+
+    @number_of_expression_experiments.setter
+    def number_of_expression_experiments(self, number_of_expression_experiments):
+        """Sets the number_of_expression_experiments of this TaxonWithUsageStatisticsValueObject.
+
+
+        :param number_of_expression_experiments: The number_of_expression_experiments of this TaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :type: int
         """
 
-        self._sort = sort
+        self._number_of_expression_experiments = number_of_expression_experiments
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -184,15 +210,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject, dict):
+        if issubclass(TaxonWithUsageStatisticsValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -200,15 +226,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, QueriedAndFilteredResponseDataObjectCategoryWithUsageStatisticsValueObject):
+        if not isinstance(other, TaxonWithUsageStatisticsValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/queried_and_filtered_response_data_object_taxon_with_usage_statistics_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/queried_and_filtered_and_inferred_response_data_object_taxon_with_usage_statistics_value_object.py`

 * *Files 19% similar despite different names*

```diff
@@ -1,176 +1,202 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject(object):
+class QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
         'data': 'list[TaxonWithUsageStatisticsValueObject]',
-        'query': 'str',
         'filter': 'str',
         'group_by': 'list[str]',
-        'sort': 'str'
+        'sort': 'SortValueObject',
+        'query': 'str',
+        'inferred_terms': 'list[CharacteristicValueObject]'
     }
 
     attribute_map = {
         'data': 'data',
-        'query': 'query',
         'filter': 'filter',
         'group_by': 'groupBy',
-        'sort': 'sort'
+        'sort': 'sort',
+        'query': 'query',
+        'inferred_terms': 'inferredTerms'
     }
 
-    def __init__(self, data=None, query=None, filter=None, group_by=None, sort=None):  # noqa: E501
-        """QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject - a model defined in Swagger"""  # noqa: E501
+    def __init__(self, data=None, filter=None, group_by=None, sort=None, query=None, inferred_terms=None):  # noqa: E501
+        """QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
-        self._query = None
         self._filter = None
         self._group_by = None
         self._sort = None
+        self._query = None
+        self._inferred_terms = None
         self.discriminator = None
         if data is not None:
             self.data = data
-        if query is not None:
-            self.query = query
         if filter is not None:
             self.filter = filter
         if group_by is not None:
             self.group_by = group_by
         if sort is not None:
             self.sort = sort
+        if query is not None:
+            self.query = query
+        if inferred_terms is not None:
+            self.inferred_terms = inferred_terms
 
     @property
     def data(self):
-        """Gets the data of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the data of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The data of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The data of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: list[TaxonWithUsageStatisticsValueObject]
         """
         return self._data
 
     @data.setter
     def data(self, data):
-        """Sets the data of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.
+        """Sets the data of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.
 
 
-        :param data: The data of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :param data: The data of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
         :type: list[TaxonWithUsageStatisticsValueObject]
         """
 
         self._data = data
 
     @property
-    def query(self):
-        """Gets the query of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
-
-
-        :return: The query of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: str
-        """
-        return self._query
-
-    @query.setter
-    def query(self, query):
-        """Sets the query of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.
-
-
-        :param query: The query of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
-        :type: str
-        """
-
-        self._query = query
-
-    @property
     def filter(self):
-        """Gets the filter of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the filter of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The filter of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The filter of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: str
         """
         return self._filter
 
     @filter.setter
     def filter(self, filter):
-        """Sets the filter of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.
+        """Sets the filter of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.
 
 
-        :param filter: The filter of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :param filter: The filter of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
         :type: str
         """
 
         self._filter = filter
 
     @property
     def group_by(self):
-        """Gets the group_by of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the group_by of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The group_by of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :return: The group_by of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
         :rtype: list[str]
         """
         return self._group_by
 
     @group_by.setter
     def group_by(self, group_by):
-        """Sets the group_by of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.
+        """Sets the group_by of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.
 
 
-        :param group_by: The group_by of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :param group_by: The group_by of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
         :type: list[str]
         """
 
         self._group_by = group_by
 
     @property
     def sort(self):
-        """Gets the sort of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        """Gets the sort of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
 
 
-        :return: The sort of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
-        :rtype: str
+        :return: The sort of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: SortValueObject
         """
         return self._sort
 
     @sort.setter
     def sort(self, sort):
-        """Sets the sort of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.
+        """Sets the sort of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.
 
 
-        :param sort: The sort of this QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
-        :type: str
+        :param sort: The sort of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :type: SortValueObject
         """
 
         self._sort = sort
 
+    @property
+    def query(self):
+        """Gets the query of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+
+
+        :return: The query of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._query
+
+    @query.setter
+    def query(self, query):
+        """Sets the query of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.
+
+
+        :param query: The query of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :type: str
+        """
+
+        self._query = query
+
+    @property
+    def inferred_terms(self):
+        """Gets the inferred_terms of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+
+
+        :return: The inferred_terms of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :rtype: list[CharacteristicValueObject]
+        """
+        return self._inferred_terms
+
+    @inferred_terms.setter
+    def inferred_terms(self, inferred_terms):
+        """Sets the inferred_terms of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.
+
+
+        :param inferred_terms: The inferred_terms of this QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject.  # noqa: E501
+        :type: list[CharacteristicValueObject]
+        """
+
+        self._inferred_terms = inferred_terms
+
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
             if isinstance(value, list):
@@ -184,15 +210,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject, dict):
+        if issubclass(QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -200,15 +226,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, QueriedAndFilteredResponseDataObjectTaxonWithUsageStatisticsValueObject):
+        if not isinstance(other, QueriedAndFilteredAndInferredResponseDataObjectTaxonWithUsageStatisticsValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/query_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/query_arg.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_api_info_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_api_info_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_differential_expression_analysis_result_set_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_differential_expression_analysis_value_object.py`

 * *Files 12% similar despite different names*

```diff
@@ -1,68 +1,68 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject(object):
+class ResponseDataObjectListDifferentialExpressionAnalysisValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'data': 'DifferentialExpressionAnalysisResultSetValueObject'
+        'data': 'list[DifferentialExpressionAnalysisValueObject]'
     }
 
     attribute_map = {
         'data': 'data'
     }
 
     def __init__(self, data=None):  # noqa: E501
-        """ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject - a model defined in Swagger"""  # noqa: E501
+        """ResponseDataObjectListDifferentialExpressionAnalysisValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
         self.discriminator = None
         if data is not None:
             self.data = data
 
     @property
     def data(self):
-        """Gets the data of this ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
+        """Gets the data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.  # noqa: E501
 
 
-        :return: The data of this ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :rtype: DifferentialExpressionAnalysisResultSetValueObject
+        :return: The data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.  # noqa: E501
+        :rtype: list[DifferentialExpressionAnalysisValueObject]
         """
         return self._data
 
     @data.setter
     def data(self, data):
-        """Sets the data of this ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.
+        """Sets the data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.
 
 
-        :param data: The data of this ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject.  # noqa: E501
-        :type: DifferentialExpressionAnalysisResultSetValueObject
+        :param data: The data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.  # noqa: E501
+        :type: list[DifferentialExpressionAnalysisValueObject]
         """
 
         self._data = data
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
@@ -80,15 +80,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject, dict):
+        if issubclass(ResponseDataObjectListDifferentialExpressionAnalysisValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -96,15 +96,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, ResponseDataObjectDifferentialExpressionAnalysisResultSetValueObject):
+        if not isinstance(other, ResponseDataObjectListDifferentialExpressionAnalysisValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_annotation_search_result_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_annotation_search_result_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_array_design_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_array_design_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_bio_assay_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_bio_assay_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_differential_expression_analysis_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_simple_svd_value_object.py`

 * *Files 11% similar despite different names*

```diff
@@ -1,68 +1,68 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class ResponseDataObjectListDifferentialExpressionAnalysisValueObject(object):
+class ResponseDataObjectSimpleSVDValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'data': 'list[DifferentialExpressionAnalysisValueObject]'
+        'data': 'SimpleSVDValueObject'
     }
 
     attribute_map = {
         'data': 'data'
     }
 
     def __init__(self, data=None):  # noqa: E501
-        """ResponseDataObjectListDifferentialExpressionAnalysisValueObject - a model defined in Swagger"""  # noqa: E501
+        """ResponseDataObjectSimpleSVDValueObject - a model defined in Swagger"""  # noqa: E501
         self._data = None
         self.discriminator = None
         if data is not None:
             self.data = data
 
     @property
     def data(self):
-        """Gets the data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.  # noqa: E501
+        """Gets the data of this ResponseDataObjectSimpleSVDValueObject.  # noqa: E501
 
 
-        :return: The data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.  # noqa: E501
-        :rtype: list[DifferentialExpressionAnalysisValueObject]
+        :return: The data of this ResponseDataObjectSimpleSVDValueObject.  # noqa: E501
+        :rtype: SimpleSVDValueObject
         """
         return self._data
 
     @data.setter
     def data(self, data):
-        """Sets the data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.
+        """Sets the data of this ResponseDataObjectSimpleSVDValueObject.
 
 
-        :param data: The data of this ResponseDataObjectListDifferentialExpressionAnalysisValueObject.  # noqa: E501
-        :type: list[DifferentialExpressionAnalysisValueObject]
+        :param data: The data of this ResponseDataObjectSimpleSVDValueObject.  # noqa: E501
+        :type: SimpleSVDValueObject
         """
 
         self._data = data
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
@@ -80,15 +80,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(ResponseDataObjectListDifferentialExpressionAnalysisValueObject, dict):
+        if issubclass(ResponseDataObjectSimpleSVDValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -96,15 +96,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, ResponseDataObjectListDifferentialExpressionAnalysisValueObject):
+        if not isinstance(other, ResponseDataObjectSimpleSVDValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_experiment_expression_levels_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_experiment_expression_levels_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_gene_ontology_term_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_gene_ontology_term_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_gene_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_gene_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_physical_location_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_physical_location_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_list_taxon_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_list_taxon_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_long.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_long.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_set_annotation_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_set_annotation_value_object.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_set_quantitation_type_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_data_object_set_quantitation_type_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_data_object_simple_svd_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/response_error_object.py`

 * *Files 21% similar despite different names*

```diff
@@ -1,71 +1,97 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class ResponseDataObjectSimpleSVDValueObject(object):
+class ResponseErrorObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'data': 'SimpleSVDValueObject'
+        'error': 'WellComposedErrorBody',
+        'api_version': 'str'
     }
 
     attribute_map = {
-        'data': 'data'
+        'error': 'error',
+        'api_version': 'apiVersion'
     }
 
-    def __init__(self, data=None):  # noqa: E501
-        """ResponseDataObjectSimpleSVDValueObject - a model defined in Swagger"""  # noqa: E501
-        self._data = None
+    def __init__(self, error=None, api_version=None):  # noqa: E501
+        """ResponseErrorObject - a model defined in Swagger"""  # noqa: E501
+        self._error = None
+        self._api_version = None
         self.discriminator = None
-        if data is not None:
-            self.data = data
+        if error is not None:
+            self.error = error
+        if api_version is not None:
+            self.api_version = api_version
 
     @property
-    def data(self):
-        """Gets the data of this ResponseDataObjectSimpleSVDValueObject.  # noqa: E501
+    def error(self):
+        """Gets the error of this ResponseErrorObject.  # noqa: E501
 
 
-        :return: The data of this ResponseDataObjectSimpleSVDValueObject.  # noqa: E501
-        :rtype: SimpleSVDValueObject
+        :return: The error of this ResponseErrorObject.  # noqa: E501
+        :rtype: WellComposedErrorBody
         """
-        return self._data
+        return self._error
 
-    @data.setter
-    def data(self, data):
-        """Sets the data of this ResponseDataObjectSimpleSVDValueObject.
+    @error.setter
+    def error(self, error):
+        """Sets the error of this ResponseErrorObject.
 
 
-        :param data: The data of this ResponseDataObjectSimpleSVDValueObject.  # noqa: E501
-        :type: SimpleSVDValueObject
+        :param error: The error of this ResponseErrorObject.  # noqa: E501
+        :type: WellComposedErrorBody
         """
 
-        self._data = data
+        self._error = error
+
+    @property
+    def api_version(self):
+        """Gets the api_version of this ResponseErrorObject.  # noqa: E501
+
+
+        :return: The api_version of this ResponseErrorObject.  # noqa: E501
+        :rtype: str
+        """
+        return self._api_version
+
+    @api_version.setter
+    def api_version(self, api_version):
+        """Sets the api_version of this ResponseErrorObject.
+
+
+        :param api_version: The api_version of this ResponseErrorObject.  # noqa: E501
+        :type: str
+        """
+
+        self._api_version = api_version
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -80,15 +106,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(ResponseDataObjectSimpleSVDValueObject, dict):
+        if issubclass(ResponseErrorObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -96,15 +122,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, ResponseDataObjectSimpleSVDValueObject):
+        if not isinstance(other, ResponseErrorObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/response_error_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/sort_value_object.py`

 * *Files 18% similar despite different names*

```diff
@@ -1,97 +1,97 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class ResponseErrorObject(object):
+class SortValueObject(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'error': 'WellComposedErrorBody',
-        'api_version': 'str'
+        'order_by': 'str',
+        'direction': 'str'
     }
 
     attribute_map = {
-        'error': 'error',
-        'api_version': 'apiVersion'
+        'order_by': 'orderBy',
+        'direction': 'direction'
     }
 
-    def __init__(self, error=None, api_version=None):  # noqa: E501
-        """ResponseErrorObject - a model defined in Swagger"""  # noqa: E501
-        self._error = None
-        self._api_version = None
+    def __init__(self, order_by=None, direction=None):  # noqa: E501
+        """SortValueObject - a model defined in Swagger"""  # noqa: E501
+        self._order_by = None
+        self._direction = None
         self.discriminator = None
-        if error is not None:
-            self.error = error
-        if api_version is not None:
-            self.api_version = api_version
+        if order_by is not None:
+            self.order_by = order_by
+        if direction is not None:
+            self.direction = direction
 
     @property
-    def error(self):
-        """Gets the error of this ResponseErrorObject.  # noqa: E501
+    def order_by(self):
+        """Gets the order_by of this SortValueObject.  # noqa: E501
 
 
-        :return: The error of this ResponseErrorObject.  # noqa: E501
-        :rtype: WellComposedErrorBody
+        :return: The order_by of this SortValueObject.  # noqa: E501
+        :rtype: str
         """
-        return self._error
+        return self._order_by
 
-    @error.setter
-    def error(self, error):
-        """Sets the error of this ResponseErrorObject.
+    @order_by.setter
+    def order_by(self, order_by):
+        """Sets the order_by of this SortValueObject.
 
 
-        :param error: The error of this ResponseErrorObject.  # noqa: E501
-        :type: WellComposedErrorBody
+        :param order_by: The order_by of this SortValueObject.  # noqa: E501
+        :type: str
         """
 
-        self._error = error
+        self._order_by = order_by
 
     @property
-    def api_version(self):
-        """Gets the api_version of this ResponseErrorObject.  # noqa: E501
+    def direction(self):
+        """Gets the direction of this SortValueObject.  # noqa: E501
 
 
-        :return: The api_version of this ResponseErrorObject.  # noqa: E501
+        :return: The direction of this SortValueObject.  # noqa: E501
         :rtype: str
         """
-        return self._api_version
+        return self._direction
 
-    @api_version.setter
-    def api_version(self, api_version):
-        """Sets the api_version of this ResponseErrorObject.
+    @direction.setter
+    def direction(self, direction):
+        """Sets the direction of this SortValueObject.
 
 
-        :param api_version: The api_version of this ResponseErrorObject.  # noqa: E501
+        :param direction: The direction of this SortValueObject.  # noqa: E501
         :type: str
         """
 
-        self._api_version = api_version
+        self._direction = direction
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -106,15 +106,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(ResponseErrorObject, dict):
+        if issubclass(SortValueObject, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -122,15 +122,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, ResponseErrorObject):
+        if not isinstance(other, SortValueObject):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/search_result_type.py` & `gemmapy-1.0.1/gemmapy/sdk/models/search_result_type.py`

 * *Files 1% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/search_result_value_object_expression_experiment_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/search_result_value_object_expression_experiment_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/search_result_value_object_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/search_result_value_object_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/search_results_response_data_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/search_results_response_data_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/simple_svd_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/simple_svd_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_array_design.py` & `gemmapy-1.0.1/gemmapy/sdk/models/sort_arg_array_design.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_expression_analysis_result_set.py` & `gemmapy-1.0.1/gemmapy/sdk/models/sort_arg_expression_analysis_result_set.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_expression_experiment.py` & `gemmapy-1.0.1/gemmapy/sdk/models/sort_arg_expression_experiment.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/sort_arg_taxon.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon_arg.py`

 * *Files 4% similar despite different names*

```diff
@@ -1,25 +1,25 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class SortArgTaxon(object):
+class TaxonArg(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
@@ -30,15 +30,15 @@
     swagger_types = {
     }
 
     attribute_map = {
     }
 
     def __init__(self):  # noqa: E501
-        """SortArgTaxon - a model defined in Swagger"""  # noqa: E501
+        """TaxonArg - a model defined in Swagger"""  # noqa: E501
         self.discriminator = None
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
@@ -54,15 +54,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(SortArgTaxon, dict):
+        if issubclass(TaxonArg, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -70,15 +70,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, SortArgTaxon):
+        if not isinstance(other, TaxonArg):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/sort_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/well_composed_error_body.py`

 * *Files 22% similar despite different names*

```diff
@@ -1,97 +1,123 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class SortValueObject(object):
+class WellComposedErrorBody(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
                             and the value is attribute type.
       attribute_map (dict): The key is attribute name
                             and the value is json key in definition.
     """
     swagger_types = {
-        'order_by': 'str',
-        'direction': 'str'
+        'code': 'int',
+        'message': 'str',
+        'errors': 'dict(str, str)'
     }
 
     attribute_map = {
-        'order_by': 'orderBy',
-        'direction': 'direction'
+        'code': 'code',
+        'message': 'message',
+        'errors': 'errors'
     }
 
-    def __init__(self, order_by=None, direction=None):  # noqa: E501
-        """SortValueObject - a model defined in Swagger"""  # noqa: E501
-        self._order_by = None
-        self._direction = None
+    def __init__(self, code=None, message=None, errors=None):  # noqa: E501
+        """WellComposedErrorBody - a model defined in Swagger"""  # noqa: E501
+        self._code = None
+        self._message = None
+        self._errors = None
         self.discriminator = None
-        if order_by is not None:
-            self.order_by = order_by
-        if direction is not None:
-            self.direction = direction
+        if code is not None:
+            self.code = code
+        if message is not None:
+            self.message = message
+        if errors is not None:
+            self.errors = errors
 
     @property
-    def order_by(self):
-        """Gets the order_by of this SortValueObject.  # noqa: E501
+    def code(self):
+        """Gets the code of this WellComposedErrorBody.  # noqa: E501
 
 
-        :return: The order_by of this SortValueObject.  # noqa: E501
-        :rtype: str
+        :return: The code of this WellComposedErrorBody.  # noqa: E501
+        :rtype: int
         """
-        return self._order_by
+        return self._code
 
-    @order_by.setter
-    def order_by(self, order_by):
-        """Sets the order_by of this SortValueObject.
+    @code.setter
+    def code(self, code):
+        """Sets the code of this WellComposedErrorBody.
 
 
-        :param order_by: The order_by of this SortValueObject.  # noqa: E501
-        :type: str
+        :param code: The code of this WellComposedErrorBody.  # noqa: E501
+        :type: int
         """
 
-        self._order_by = order_by
+        self._code = code
 
     @property
-    def direction(self):
-        """Gets the direction of this SortValueObject.  # noqa: E501
+    def message(self):
+        """Gets the message of this WellComposedErrorBody.  # noqa: E501
 
 
-        :return: The direction of this SortValueObject.  # noqa: E501
+        :return: The message of this WellComposedErrorBody.  # noqa: E501
         :rtype: str
         """
-        return self._direction
+        return self._message
 
-    @direction.setter
-    def direction(self, direction):
-        """Sets the direction of this SortValueObject.
+    @message.setter
+    def message(self, message):
+        """Sets the message of this WellComposedErrorBody.
 
 
-        :param direction: The direction of this SortValueObject.  # noqa: E501
+        :param message: The message of this WellComposedErrorBody.  # noqa: E501
         :type: str
         """
 
-        self._direction = direction
+        self._message = message
+
+    @property
+    def errors(self):
+        """Gets the errors of this WellComposedErrorBody.  # noqa: E501
+
+
+        :return: The errors of this WellComposedErrorBody.  # noqa: E501
+        :rtype: dict(str, str)
+        """
+        return self._errors
+
+    @errors.setter
+    def errors(self, errors):
+        """Sets the errors of this WellComposedErrorBody.
+
+
+        :param errors: The errors of this WellComposedErrorBody.  # noqa: E501
+        :type: dict(str, str)
+        """
+
+        self._errors = errors
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
             value = getattr(self, attr)
@@ -106,15 +132,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(SortValueObject, dict):
+        if issubclass(WellComposedErrorBody, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -122,15 +148,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, SortValueObject):
+        if not isinstance(other, WellComposedErrorBody):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/statement_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/statement_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon1.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon1.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon2.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon2.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon3.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon3.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon4.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon4.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon5.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon5.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon6.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon6.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon_arg.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon7.py`

 * *Files 2% similar despite different names*

```diff
@@ -1,25 +1,25 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
 
 import six
 
-class TaxonArg(object):
+class Taxon7(object):
     """NOTE: This class is auto generated by the swagger code generator program.
 
     Do not edit the class manually.
     """
     """
     Attributes:
       swagger_types (dict): The key is attribute name
@@ -30,15 +30,15 @@
     swagger_types = {
     }
 
     attribute_map = {
     }
 
     def __init__(self):  # noqa: E501
-        """TaxonArg - a model defined in Swagger"""  # noqa: E501
+        """Taxon7 - a model defined in Swagger"""  # noqa: E501
         self.discriminator = None
 
     def to_dict(self):
         """Returns the model properties as a dict"""
         result = {}
 
         for attr, _ in six.iteritems(self.swagger_types):
@@ -54,15 +54,15 @@
                 result[attr] = dict(map(
                     lambda item: (item[0], item[1].to_dict())
                     if hasattr(item[1], "to_dict") else item,
                     value.items()
                 ))
             else:
                 result[attr] = value
-        if issubclass(TaxonArg, dict):
+        if issubclass(Taxon7, dict):
             for key, value in self.items():
                 result[key] = value
 
         return result
 
     def to_str(self):
         """Returns the string representation of the model"""
@@ -70,15 +70,15 @@
 
     def __repr__(self):
         """For `print` and `pprint`"""
         return self.to_str()
 
     def __eq__(self, other):
         """Returns true if both objects are equal"""
-        if not isinstance(other, TaxonArg):
+        if not isinstance(other, Taxon7):
             return False
 
         return self.__dict__ == other.__dict__
 
     def __ne__(self, other):
         """Returns true if both objects are not equal"""
         return not self == other
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/taxon_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/taxon_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/models/vector_element_value_object.py` & `gemmapy-1.0.1/gemmapy/sdk/models/vector_element_value_object.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 import pprint
 import re  # noqa: F401
```

### Comparing `gemmapy-1.0.0/gemmapy/sdk/rest.py` & `gemmapy-1.0.1/gemmapy/sdk/rest.py`

 * *Files 0% similar despite different names*

```diff
@@ -1,15 +1,15 @@
 # coding: utf-8
 
 """
     Gemma RESTful API
 
     This website documents the usage of the [Gemma RESTful API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  Use of this webpage and the Gemma Web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system.  You can [consult the CHANGELOG.md file](https://gemma.msl.ubc.ca/resources/restapidocs/CHANGELOG.md) to view  release notes and recent changes to the Gemma RESTful API.   # noqa: E501
 
-    OpenAPI spec version: 2.7.4
+    OpenAPI spec version: 2.7.5
     Contact: pavlab-support@msl.ubc.ca
     Generated by: https://github.com/swagger-api/swagger-codegen.git
 """
 
 from __future__ import absolute_import
 
 import io
```

### Comparing `gemmapy-1.0.0/gemmapy.egg-info/PKG-INFO` & `gemmapy-1.0.1/gemmapy.egg-info/PKG-INFO`

 * *Files 0% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: gemmapy
-Version: 1.0.0
+Version: 1.0.1
 Summary: a Python Wrapper for the Gemma API
 Keywords: gemma,bioinformatics
 Requires-Python: >3.10
 License-File: LICENSE.md
 Requires-Dist: certifi>=14.05.14
 Requires-Dist: six>=1.10
 Requires-Dist: python_dateutil>=2.5.3
```

### Comparing `gemmapy-1.0.0/gemmapy.egg-info/SOURCES.txt` & `gemmapy-1.0.1/gemmapy.egg-info/SOURCES.txt`

 * *Files 7% similar despite different names*

```diff
@@ -62,56 +62,59 @@
 gemmapy/sdk/models/expression_experiment_with_search_result_value_object.py
 gemmapy/sdk/models/external_database_value_object.py
 gemmapy/sdk/models/factor_value_basic_value_object.py
 gemmapy/sdk/models/factor_value_value_object.py
 gemmapy/sdk/models/filter_arg_array_design.py
 gemmapy/sdk/models/filter_arg_expression_analysis_result_set.py
 gemmapy/sdk/models/filter_arg_expression_experiment.py
+gemmapy/sdk/models/filtered_and_inferred_and_paginated_response_data_object_expression_experiment_value_object.py
 gemmapy/sdk/models/filtered_and_paginated_response_data_object_array_design_value_object.py
 gemmapy/sdk/models/filtered_and_paginated_response_data_object_composite_sequence_value_object.py
 gemmapy/sdk/models/filtered_and_paginated_response_data_object_differential_expression_analysis_result_set_value_object.py
 gemmapy/sdk/models/filtered_and_paginated_response_data_object_expression_experiment_value_object.py
 gemmapy/sdk/models/filtered_and_paginated_response_data_object_gene_value_object.py
 gemmapy/sdk/models/geeq_value_object.py
 gemmapy/sdk/models/gene.py
 gemmapy/sdk/models/gene1.py
 gemmapy/sdk/models/gene2.py
 gemmapy/sdk/models/gene3.py
 gemmapy/sdk/models/gene4.py
+gemmapy/sdk/models/gene5.py
 gemmapy/sdk/models/gene_arg.py
 gemmapy/sdk/models/gene_element_expressions_value_object.py
 gemmapy/sdk/models/gene_ontology_term_value_object.py
 gemmapy/sdk/models/gene_set_value_object.py
 gemmapy/sdk/models/gene_value_object.py
-gemmapy/sdk/models/limited_response_data_object_annotation_with_usage_statistics_value_object.py
-gemmapy/sdk/models/limited_response_data_object_array_design_with_usage_statistics_value_object.py
 gemmapy/sdk/models/measurement_value_object.py
 gemmapy/sdk/models/one_of_search_result_value_object_object_result_object.py
 gemmapy/sdk/models/ontology_term_value_object.py
 gemmapy/sdk/models/paginated_response_data_object_composite_sequence_value_object.py
 gemmapy/sdk/models/paginated_response_data_object_expression_experiment_value_object.py
+gemmapy/sdk/models/paginated_response_data_object_gene_value_object.py
+gemmapy/sdk/models/paginated_results_response_data_object_differential_expression_analysis_result_set_value_object.py
 gemmapy/sdk/models/physical_location_value_object.py
 gemmapy/sdk/models/platform.py
 gemmapy/sdk/models/platform1.py
 gemmapy/sdk/models/platform2.py
 gemmapy/sdk/models/platform3.py
 gemmapy/sdk/models/platform4.py
 gemmapy/sdk/models/platform5.py
 gemmapy/sdk/models/platform_arg.py
 gemmapy/sdk/models/probe.py
 gemmapy/sdk/models/quantitation_type.py
 gemmapy/sdk/models/quantitation_type_arg.py
 gemmapy/sdk/models/quantitation_type_value_object.py
+gemmapy/sdk/models/queried_and_filtered_and_inferred_and_limited_response_data_object_annotation_with_usage_statistics_value_object.py
+gemmapy/sdk/models/queried_and_filtered_and_inferred_and_limited_response_data_object_array_design_with_usage_statistics_value_object.py
+gemmapy/sdk/models/queried_and_filtered_and_inferred_and_limited_response_data_object_category_with_usage_statistics_value_object.py
+gemmapy/sdk/models/queried_and_filtered_and_inferred_and_paginated_response_data_object_expression_experiment_with_search_result_value_object.py
+gemmapy/sdk/models/queried_and_filtered_and_inferred_response_data_object_taxon_with_usage_statistics_value_object.py
 gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_value_object.py
-gemmapy/sdk/models/queried_and_filtered_and_paginated_response_data_object_expression_experiment_with_search_result_value_object.py
-gemmapy/sdk/models/queried_and_filtered_response_data_object_category_with_usage_statistics_value_object.py
-gemmapy/sdk/models/queried_and_filtered_response_data_object_taxon_with_usage_statistics_value_object.py
 gemmapy/sdk/models/query_arg.py
 gemmapy/sdk/models/response_data_object_api_info_value_object.py
-gemmapy/sdk/models/response_data_object_differential_expression_analysis_result_set_value_object.py
 gemmapy/sdk/models/response_data_object_list_annotation_search_result_value_object.py
 gemmapy/sdk/models/response_data_object_list_array_design_value_object.py
 gemmapy/sdk/models/response_data_object_list_bio_assay_value_object.py
 gemmapy/sdk/models/response_data_object_list_differential_expression_analysis_value_object.py
 gemmapy/sdk/models/response_data_object_list_experiment_expression_levels_value_object.py
 gemmapy/sdk/models/response_data_object_list_gene_ontology_term_value_object.py
 gemmapy/sdk/models/response_data_object_list_gene_value_object.py
@@ -127,23 +130,26 @@
 gemmapy/sdk/models/search_result_value_object_object.py
 gemmapy/sdk/models/search_results_response_data_object.py
 gemmapy/sdk/models/search_settings_value_object.py
 gemmapy/sdk/models/simple_svd_value_object.py
 gemmapy/sdk/models/sort_arg_array_design.py
 gemmapy/sdk/models/sort_arg_expression_analysis_result_set.py
 gemmapy/sdk/models/sort_arg_expression_experiment.py
-gemmapy/sdk/models/sort_arg_taxon.py
 gemmapy/sdk/models/sort_value_object.py
 gemmapy/sdk/models/statement_value_object.py
+gemmapy/sdk/models/taxa.py
 gemmapy/sdk/models/taxon.py
 gemmapy/sdk/models/taxon1.py
 gemmapy/sdk/models/taxon2.py
 gemmapy/sdk/models/taxon3.py
 gemmapy/sdk/models/taxon4.py
 gemmapy/sdk/models/taxon5.py
 gemmapy/sdk/models/taxon6.py
+gemmapy/sdk/models/taxon7.py
+gemmapy/sdk/models/taxon8.py
+gemmapy/sdk/models/taxon9.py
 gemmapy/sdk/models/taxon_arg.py
 gemmapy/sdk/models/taxon_value_object.py
 gemmapy/sdk/models/taxon_with_usage_statistics_value_object.py
 gemmapy/sdk/models/vector_element_value_object.py
 gemmapy/sdk/models/well_composed_error_body.py
 tests/test_basic.py
```

### Comparing `gemmapy-1.0.0/tests/test_basic.py` & `gemmapy-1.0.1/tests/test_basic.py`

 * *Files identical despite different names*

